---
title: "Rsubread"
author: "Holly"
date: "7/11/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r load packages}
#source("https://bioconductor.org/biocLite.R")
#        biocLite("Rsubread")
library(Rsubread)
```

## Including Plots

You can also embed plots, for example:

```{r load files, echo=FALSE}
bam_directory <- "/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Indiv_Genes/Rsubread/bams"
bam_files <- list.files(bam_directory, full.names = TRUE)

gtffile = "/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Indiv_Genes/Rsubread/genes.gtf"

count_data <- featureCounts( files = bam_files, annot.ext = gtffile, isGTFAnnotationFile = TRUE,GTF.featureType = "exon", GTF.attrType = "gene_id", strandSpecific = 1)

```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
