---
title: "DC_workflow_old_MA"
author: "Holly"
date: "11/30/2017"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r load in required pakages}
library(moments)
library("Hmisc")
library(pBrackets)
library(data.table)
library(Rsamtools)
library(splitstackshape)
library(ggplot2)
library(pgirmess)
library(FSA)
library(dunn.test)
library(rcompanion)
```

``` {r load in data,make matrices,echo=FALSE,include=TRUE}
sc_counts.old <- read.csv("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Chrom_wide_Analyses/Cuffnorm_Output/CuffnormJune19/genes.fpkm_table.joined.rmcom.split.commas.rmvmito.chgchr.rmhead.nwhd.shrt_old.csv",header=T)
class(sc_counts.old)
str(sc_counts.old)
colnames(sc_counts.old)[1] <- "chr"; colnames(sc_counts.old)[2] <- "tracking_id"
sc_counts.z.old <- sc_counts.old
sc_counts.z.old[,3:50] <- sc_counts.z.old[,3:50]+.5 #add .5 to dataset so the zeroes dont mess up
#we want to exclude genes that are very lowly expressed
gene.means.old <- rowMeans(sc_counts.z.old[,3:50]) #average across the sample
anc.means.MA.old <- rowMeans(sc_counts.z.old[,3:5]) #average in the ancestor

#if the average fpkm is less than 5 in the ancestor or less than 5 in the overall, remove the gene
sc_counts.final.old <- sc_counts.z.old[-which(gene.means.old<5 | anc.means.MA.old<5),]

#final matrix for ancestor and descendans
sc_anc.old <- sc_counts.final.old[,c(1,2,3,4,5)]
sc_des.MA.old <- sc_counts.final.old[,c(1,2,6:50)] #average for each MA line

index1.old <- which(sc_anc.old[,1]==1)
index2.old <- which(sc_anc.old[,1]==2)
index3.old <- which(sc_anc.old[,1]==3)
index4.old <- which(sc_anc.old[,1]==4)
index5.old <- which(sc_anc.old[,1]==5)
index6.old <- which(sc_anc.old[,1]==6)
index7.old <- which(sc_anc.old[,1]==7)
index8.old <- which(sc_anc.old[,1]==8)
index9.old <- which(sc_anc.old[,1]==9)
index10.old <- which(sc_anc.old[,1]==10)
index11.old <- which(sc_anc.old[,1]==11)
index12.old <- which(sc_anc.old[,1]==12)
index13.old <- which(sc_anc.old[,1]==13)
index14.old <- which(sc_anc.old[,1]==14)
index15.old <- which(sc_anc.old[,1]==15)
index16.old <- which(sc_anc.old[,1]==16)

# FOLD CHANGES: average of line over average of ancestor
###ancestor average
#PROBLEM HERE! I was using the old code and not taking into account that I had two ancestors! now I have 3...
ancAvg.MA.old <- rowMeans(sc_anc.old[,3:5],na.rm=T)
ancAvg.MA.old <- cbind(sc_anc.old[,1:2],ancAvg.MA.old)

```

``` {r automated functions,echo=FALSE,include=TRUE}
#first write some functions that help automate the process
##########
#this gets the average FPKM for a given line from the sc_des matrix
getAvgFPKM.MA.old <- function(line) { #for line as "###" string
  r1 <- paste(c("SC",line,"_0"),collapse="") #triplicates are named X[LINE]_[rep]
  r2 <- paste(c("SC",line,"_1"),collapse="")
  r3 <- paste(c("SC",line,"_2"),collapse="")
  reps <- sc_des.MA.old[,c("chr","tracking_id",r1,r2,r3)] #get those 4 columns out of the descendant dataset
  reps[reps==0] <- NA #replace 0 values with NA as before
  desAvg.MA.old <- rowMeans(reps[,c(-1,-2)],na.rm=T)
  desAvg.MA.old <- cbind(sc_des.MA.old[,c(1,2)],desAvg.MA.old)
  #colnames(desAvg) <- c("chr","gene",paste(c(line,"_Avg"),collapse=""))
  return(desAvg.MA.old) #returned is a matrix equivalent to ancAvg: chrm, gene, and line average FPKM 
}

getFPKMRatio.MA.old <- function(line) { #line must be a 3 digit character
  desAvg.MA.old <- getAvgFPKM.MA.old(line) #first call avg fpkm script, which returns a 3 column matrix
  ratioByGene.MA <- desAvg.MA.old[,3]/ancAvg.MA.old[,3]
  rat.MA.old <- cbind(sc_des.MA.old[,1:2],ratioByGene.MA) #add gene and chrm values
  colnames(rat.MA.old) <- c("chr","tracking_id","ratio")
  return(rat.MA.old)
}

getChrmRatio.MA.old <- function(line, chr){ #input line as "XXX", chr as a number
  rat.MA <- getFPKMRatio.MA.old(line) #get list of ratios which includes chrm and gene
  rat.c.MA <- rat.MA[(rat.MA[,1]==chr),] #get just the rows for the appropriate chrom
  rat.cs.MA.old <- rat.c.MA[order(rat.c.MA[,3]),]#order by column 3
  return(rat.cs.MA.old)
}


y.MA.old=NULL
rep.MA.old <- c("001","002","003","004","005","006","007","008","009","011","015","028","088","108","119")
#View(rep.MA)
for(line in rep.MA.old){
  for(chr in 1:16){
    y.MA.old <- c(y.MA.old,median(getChrmRatio.MA.old(line,chr)[,3],na.rm=T))
  }
}

ratMatMed.MA.old <- matrix(y.MA.old,nrow=16,ncol=15)
colnames(ratMatMed.MA.old) <- rep.MA.old
ratMatMed.MA.old <- cbind(c(1:16),ratMatMed.MA.old)
colnames(ratMatMed.MA.old)[1] <- "chr"

#remove the ratios that are about 1
ratMatMedRed.MA.old <- ratMatMed.MA.old
ratMatMedRed.MA.old[ratMatMedRed.MA.old<1.35] <- NA
#ratMatMedRed.MA[9,15] <- ratMatMedRed.MA[9,15]

#make a matrix that is the LOG2 RATIO
yl.MA.old=NULL
for(line in rep.MA.old){
  for(chrm in 1:16){
    yl.MA.old <- c(yl.MA.old,median(log2(getChrmRatio.MA.old(line,chrm)[,3])))
  }
}

warnings()
logMat.MA.old <- matrix(yl.MA.old,nrow=16,ncol=15)
colnames(logMat.MA.old) <- rep.MA.old
logMat.MA.old <- data.frame(logMat.MA.old)
chrms <- c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16)
logMat2.MA.old <-cbind(chrms,logMat.MA.old)


#this makes a matrix with the log2 ratio between ancestor and sample of each gene 

getChrmRatio.MA.test.old <- function(line, chr){ #input line as "XXX", chr as a number
  rat.MA <- getFPKMRatio.MA.old(line) #get list of ratios which includes chrm and gene
  rat.cs.MA.old <- rat.MA[order(rat.MA[,3]),]#order by column 3
  return(rat.cs.MA.old)
}

yl.MA.old=NULL
for(line in rep.MA.old){
    y.MA.old <- c(yl.MA.old,log2(getChrmRatio.MA.test.old(line,chrm)[,3]))
}

warnings()
logMat.MA.test.old<- matrix(y.MA.old,nrow=6037,ncol=15)
colnames(logMat.MA.test.old) <- rep.MA.old
logMat.MA.test.old <- data.frame(logMat.MA.test.old)
genes <- ancAvg.MA.old[,2]
logMat2.MA.test.old <-cbind(genes,logMat.MA.test.old)

##make a matrix of the standard deviations for each line and each chromosome
ysd.MA.old=NULL
for(line in rep.MA.old){
  for(chrm in 1:16){
    ysd.MA.old <- c(ysd.MA.old,sd(log2(getChrmRatio.MA.old(line,chrm)[,3])))
  }
}

logMatsd.MA.old <- matrix(ysd.MA.old,nrow=16,ncol=15)
colnames(logMatsd.MA.old) <- rep.MA.old
logMatsd.MA.old <- data.frame(logMatsd.MA.old)
chrms <- c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16)
logMatsd2.MA.old <-cbind(chrms,logMatsd.MA.old)



```

``` {r graph:line log2 ratio per chr,,echo=FALSE,include=TRUE}
######################################################################################
#how to make a graph of the chromosomes and their respective error bars 
#changes the margins to the standard margin width
setwd = "/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/January2018_Final_Graphs"
par(mar=c(5, 4, 4, 2) + 0.1)
par(mfrow=c(1,1))
#line 1
X001 <- subset(logMat.MA.old, select=c("X001"))
#get the standard deviations for each chromosome for line 112 only
X001.sd <- subset(logMatsd2.MA.old, select=c("X001"))
#merge the stdev with the mean and the chromosome number
X001.all <- cbind(chrms,X001,X001.sd)
#change the column names appropriately
colnames(X001.all)[1] <- "chr"; colnames(X001.all)[2] <- "mean";colnames(X001.all)[3] <- "sd"

#plots the chromosomes with error bars for each one 
pdf("X001_chroms.pdf")
with (
  data = X001.all
  , expr = errbar(main="MA Line 1",chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a title to the graph
title(main="MA Line 1")
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()

#line 2
par(mar=c(5, 4, 4, 2) + 0.1)
#get the standard deviations for each chromosome for line 112 only
X2 <- subset(logMat.MA.old, select=c("X002"))
X2.sd <- subset(logMatsd2.MA.old, select=c("X002"))
#merge the stdev with the mean and the chromosome number
X2.all <- cbind(chrms,X2,X2.sd)
#change the column names appropriately
colnames(X2.all)[1] <- "chr"; colnames(X2.all)[2] <- "mean";colnames(X2.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
pdf("X002_chroms.pdf")
with (
  data = X2.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="MA Line 2")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()

#line 3
#get the standard deviations for each chromosome for line 112 only
X3 <- subset(logMat.MA.old, select=c("X003"))
X3.sd <- subset(logMatsd2.MA.old, select=c("X003"))
#merge the stdev with the mean and the chromosome number
X3.all <- cbind(chrms,X3,X3.sd)
#change the column names appropriately
colnames(X3.all)[1] <- "chr"; colnames(X3.all)[2] <- "mean";colnames(X3.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
pdf("X003_chroms.pdf")
with (
  data = X3.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="MA Line 3")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()

#line 4
X4 <- subset(logMat.MA.old, select=c("X004"))
X4.sd <- subset(logMatsd2.MA.old, select=c("X004"))
#merge the stdev with the mean and the chromosome number
X4.all <- cbind(chrms,X4,X4.sd)
#change the column names appropriately
colnames(X4.all)[1] <- "chr"; colnames(X4.all)[2] <- "mean";colnames(X4.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
pdf("X004_chroms.pdf")
with (
  data = X4.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="MA Line 4")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()


#line 5
X5 <- subset(logMat.MA.old, select=c("X005"))
X5.sd <- subset(logMatsd2.MA.old, select=c("X005"))
#merge the stdev with the mean and the chromosome number
X5.all <- cbind(chrms,X5,X5.sd)
#change the column names appropriately
colnames(X5.all)[1] <- "chr"; colnames(X5.all)[2] <- "mean";colnames(X5.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
pdf("X005_chroms.pdf")
with (
  data = X5.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="MA Line 5")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()

#line 6
X6 <- subset(logMat.MA.old, select=c("X006"))
X6.sd <- subset(logMatsd2.MA.old, select=c("X006"))
#merge the stdev with the mean and the chromosome number
X6.all <- cbind(chrms,X6,X6.sd)
#change the column names appropriately
colnames(X6.all)[1] <- "chr"; colnames(X6.all)[2] <- "mean";colnames(X6.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
pdf("X006_chroms.pdf")
with (
  data = X6.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="MA Line 6")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()

#line 7
X7 <- subset(logMat.MA.old, select=c("X007"))
X7.sd <- subset(logMatsd2.MA.old, select=c("X007"))
#merge the stdev with the mean and the chromosome number
X7.all <- cbind(chrms,X7,X7.sd)
#change the column names appropriately
colnames(X7.all)[1] <- "chr"; colnames(X7.all)[2] <- "mean";colnames(X7.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
pdf("X007_chroms.pdf")
with (
  data = X7.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="MA Line 7")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()

#line 8
X8 <- subset(logMat.MA.old, select=c("X008"))
X8.sd <- subset(logMatsd2.MA.old, select=c("X008"))
#merge the stdev with the mean and the chromosome number
X8.all <- cbind(chrms,X8,X8.sd)
#change the column names appropriately
colnames(X8.all)[1] <- "chr"; colnames(X8.all)[2] <- "mean";colnames(X8.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
pdf("X008_chroms.pdf")
with (
  data = X8.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="MA Line 8")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()

#line 9
X9 <- subset(logMat.MA.old, select=c("X009"))
X9.sd <- subset(logMatsd2.MA.old, select=c("X009"))
#merge the stdev with the mean and the chromosome number
X9.all <- cbind(chrms,X9,X9.sd)
#change the column names appropriately
colnames(X9.all)[1] <- "chr"; colnames(X9.all)[2] <- "mean";colnames(X9.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
pdf("X009_chroms.pdf")
with (
  data = X9.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="MA Line 9")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()

#line 11
X11 <- subset(logMat.MA.old, select=c("X011"))
X11.sd <- subset(logMatsd2.MA.old, select=c("X011"))
#merge the stdev with the mean and the chromosome number
X11.all <- cbind(chrms,X11,X11.sd)
#change the column names appropriately
colnames(X11.all)[1] <- "chr"; colnames(X11.all)[2] <- "mean";colnames(X11.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
pdf("X011_chroms.pdf")
with (
  data = X11.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="MA Line 11")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()

#line 15
X15 <- subset(logMat.MA.old, select=c("X015"))
X15.sd <- subset(logMatsd2.MA.old, select=c("X015"))
#merge the stdev with the mean and the chromosome number
X15.all <- cbind(chrms,X15,X15.sd)
#change the column names appropriately
colnames(X15.all)[1] <- "chr"; colnames(X15.all)[2] <- "mean";colnames(X15.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
pdf("X015_chroms.pdf")
with (
  data = X15.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="MA Line 15")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()

#line 28
X28 <- subset(logMat.MA.old, select=c("X028"))
X28.sd <- subset(logMatsd2.MA.old, select=c("X028"))
#merge the stdev with the mean and the chromosome number
X28.all <- cbind(chrms,X28,X28.sd)
#change the column names appropriately
colnames(X28.all)[1] <- "chr"; colnames(X28.all)[2] <- "mean";colnames(X28.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
pdf("X028_chroms.pdf")
with (
  data = X28.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="MA Line 28")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()

#line 88
X88 <- subset(logMat.MA.old, select=c("X088"))
X88.sd <- subset(logMatsd2.MA.old, select=c("X088"))
#merge the stdev with the mean and the chromosome number
X88.all <- cbind(chrms,X88,X88.sd)
#change the column names appropriately
colnames(X88.all)[1] <- "chr"; colnames(X88.all)[2] <- "mean";colnames(X88.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
pdf("X088_chroms.pdf")
with (
  data = X88.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="MA Line 88")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()

#line 108
X108 <- subset(logMat.MA.old, select=c("X108"))
X108.sd <- subset(logMatsd2.MA.old, select=c("X108"))
#merge the stdev with the mean and the chromosome number
X108.all <- cbind(chrms,X108,X108.sd)
#change the column names appropriately
colnames(X108.all)[1] <- "chr"; colnames(X108.all)[2] <- "mean";colnames(X108.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
pdf("X108_chroms.pdf")
with (
  data = X108.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="MA Line 108")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()

#line 119
X119 <- subset(logMat.MA.old, select=c("X119"))
X119.sd <- subset(logMatsd2.MA.old, select=c("X119"))
#merge the stdev with the mean and the chromosome number
X119.all <- cbind(chrms,X119,X119.sd)
#change the column names appropriately
colnames(X119.all)[1] <- "chr"; colnames(X119.all)[2] <- "mean";colnames(X119.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
pdf("X119_chroms.pdf")
with (
  data = X119.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="MA Line 119")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()
```

``` {r ratio files, echo=FALSE,echo=FALSE,include=TRUE}
#get ratios
setwd = "/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/R_graphs"

l001.1.old <- getChrmRatio.MA.old("001",1)
l001.1.red.old <- l001.1.old[which(l001.1.old[,3]>0.1),]
l002.1.old <- getChrmRatio.MA.old("002",1)
l002.1.red.old <- l002.1.old[which(l002.1.old[,3]>0.1),]
l003.1.old <- getChrmRatio.MA.old("003",1)
l003.1.red.old <- l003.1.old[which(l003.1.old[,3]>0.1),]
l004.1.old <- getChrmRatio.MA.old("004",1)
l004.1.red.old <- l004.1.old[which(l004.1.old[,3]>0.1),]
l005.1.old <- getChrmRatio.MA.old("005",1)
l005.1.red.old <- l005.1.old[which(l005.1.old[,3]>0.1),]
l006.1.old <- getChrmRatio.MA.old("006",1)
l006.1.red.old <- l006.1.old[which(l006.1.old[,3]>0.1),]
l007.1.old <- getChrmRatio.MA.old("007",1)
l007.1.red.old <- l007.1.old[which(l007.1.old[,3]>0.1),]
l008.1.old <- getChrmRatio.MA.old("008",1)
l008.1.red.old <- l008.1.old[which(l008.1.old[,3]>0.1),]
l009.1.old <- getChrmRatio.MA.old("009",1)
l009.1.red.old <- l009.1.old[which(l009.1.old[,3]>0.1),]
l011.1.old <- getChrmRatio.MA.old("011",1)
l011.1.red.old <- l011.1.old[which(l011.1.old[,3]>0.1),]
l015.1 <- getChrmRatio.MA.old("015",1)
l015.1.red <- l015.1[which(l015.1[,3]>0.1),]
l028.1.old <- getChrmRatio.MA.old("028",1)
l028.1.red.old <- l028.1.old[which(l028.1.old[,3]>0.1),]
l088.1 <- getChrmRatio.MA.old("088",1)
l088.1.red <- l088.1[which(l088.1[,3]>0.1),]
l108.1 <- getChrmRatio.MA.old("108",1)
l108.1.red <- l108.1[which(l108.1[,3]>0.1),]
l119.1 <- getChrmRatio.MA.old("119",1)
l119.1.red <- l119.1[which(l119.1[,3]>0.1),]

l001.2.old <- getChrmRatio.MA.old("001",2)
l001.2.red.old <- l001.2.old[which(l001.2.old[,3]>0.1),]
l002.2.old <- getChrmRatio.MA.old("002",2)
l002.2.red.old <- l002.2.old[which(l002.2.old[,3]>0.1),]
l003.2.old <- getChrmRatio.MA.old("003",2)
l003.2.red.old <- l003.2.old[which(l003.2.old[,3]>0.1),]
l004.2.old <- getChrmRatio.MA.old("004",2)
l004.2.red.old <- l004.2.old[which(l004.2.old[,3]>0.1),]
l005.2.old <- getChrmRatio.MA.old("005",2)
l005.2.red.old <- l005.2.old[which(l005.2.old[,3]>0.1),]
l006.2.old <- getChrmRatio.MA.old("006",2)
l006.2.red.old <- l006.2.old[which(l006.2.old[,3]>0.1),]
l007.2.old <- getChrmRatio.MA.old("007",2)
l007.2.red.old <- l007.2.old[which(l007.2.old[,3]>0.1),]
l008.2.old <- getChrmRatio.MA.old("008",2)
l008.2.red.old <- l008.2.old[which(l008.2.old[,3]>0.1),]
l009.2.old <- getChrmRatio.MA.old("009",2)
l009.2.red.old <- l009.2.old[which(l009.2.old[,3]>0.1),]
l011.2.old <- getChrmRatio.MA.old("011",2)
l011.2.red.old <- l011.2.old[which(l011.2.old[,3]>0.1),]
l015.2 <- getChrmRatio.MA.old("015",2)
l015.2.red <- l015.2[which(l015.2[,3]>0.1),]
l028.2.old <- getChrmRatio.MA.old("028",2)
l028.2.red.old <- l028.2.old[which(l028.2.old[,3]>0.1),]
l088.2 <- getChrmRatio.MA.old("088",2)
l088.2.red <- l088.2[which(l088.2[,3]>0.1),]
l108.2 <- getChrmRatio.MA.old("108",2)
l108.2.red <- l108.2[which(l108.2[,3]>0.1),]
l119.2 <- getChrmRatio.MA.old("119",2)
l119.2.red <- l119.2[which(l119.2[,3]>0.1),]

l001.3.old <- getChrmRatio.MA.old("001",3)
l001.3.red.old <- l001.3.old[which(l001.3.old[,3]>0.1),]
l002.3.old <- getChrmRatio.MA.old("002",3)
l002.3.red.old <- l002.3.old[which(l002.3.old[,3]>0.1),]
l003.3.old <- getChrmRatio.MA.old("003",3)
l003.3.red.old <- l003.3.old[which(l003.3.old[,3]>0.1),]
l004.3.old <- getChrmRatio.MA.old("004",3)
l004.3.red.old <- l004.3.old[which(l004.3.old[,3]>0.1),]
l005.3.old <- getChrmRatio.MA.old("005",3)
l005.3.red.old <- l005.3.old[which(l005.3.old[,3]>0.1),]
l006.3.old <- getChrmRatio.MA.old("006",3)
l006.3.red.old <- l006.3.old[which(l006.3.old[,3]>0.1),]
l007.3.old <- getChrmRatio.MA.old("007",3)
l007.3.red.old <- l007.3.old[which(l007.3.old[,3]>0.1),]
l008.3.old <- getChrmRatio.MA.old("008",3)
l008.3.red.old <- l008.3.old[which(l008.3.old[,3]>0.1),]
l009.3.old <- getChrmRatio.MA.old("009",3)
l009.3.red.old <- l009.3.old[which(l009.3.old[,3]>0.1),]
l011.3.old <- getChrmRatio.MA.old("011",3)
l011.3.red.old <- l011.3.old[which(l011.3.old[,3]>0.1),]
l015.3 <- getChrmRatio.MA.old("015",3)
l015.3.red <- l015.3[which(l015.3[,3]>0.1),]
l028.3.old <- getChrmRatio.MA.old("028",3)
l028.3.red.old <- l028.3.old[which(l028.3.old[,3]>0.1),]
l088.3 <- getChrmRatio.MA.old("088",3)
l088.3.red <- l088.3[which(l088.3[,3]>0.1),]
l108.3 <- getChrmRatio.MA.old("108",3)
l108.3.red <- l108.3[which(l108.3[,3]>0.1),]
l119.3 <- getChrmRatio.MA.old("119",3)
l119.3.red <- l119.3[which(l119.3[,3]>0.1),]

l001.4.old <- getChrmRatio.MA.old("001",4)
l001.4.red.old <- l001.4.old[which(l001.4.old[,3]>0.1),]
l002.4.old <- getChrmRatio.MA.old("002",4)
l002.4.red.old <- l002.4.old[which(l002.4.old[,3]>0.1),]
l003.4.old <- getChrmRatio.MA.old("003",4)
l003.4.red.old <- l003.4.old[which(l003.4.old[,3]>0.1),]
l004.4.old <- getChrmRatio.MA.old("004",4)
l004.4.red.old <- l004.4.old[which(l004.4.old[,3]>0.1),]
l005.4.old <- getChrmRatio.MA.old("005",4)
l005.4.red.old <- l005.4.old[which(l005.4.old[,3]>0.1),]
l006.4.old <- getChrmRatio.MA.old("006",4)
l006.4.red.old <- l006.4.old[which(l006.4.old[,3]>0.1),]
l007.4.old <- getChrmRatio.MA.old("007",4)
l007.4.red.old <- l007.4.old[which(l007.4.old[,3]>0.1),]
l008.4.old <- getChrmRatio.MA.old("008",4)
l008.4.red.old <- l008.4.old[which(l008.4.old[,3]>0.1),]
l009.4.old <- getChrmRatio.MA.old("009",4)
l009.4.red.old <- l009.4.old[which(l009.4.old[,3]>0.1),]
l011.4.old <- getChrmRatio.MA.old("011",4)
l011.4.red.old <- l011.4.old[which(l011.4.old[,3]>0.1),]
l015.4 <- getChrmRatio.MA.old("015",4)
l015.4.red <- l015.4[which(l015.4[,3]>0.1),]
l028.4.old <- getChrmRatio.MA.old("028",4)
l028.4.red.old <- l028.4.old[which(l028.4.old[,3]>0.1),]
l088.4 <- getChrmRatio.MA.old("088",4)
l088.4.red <- l088.4[which(l088.4[,3]>0.1),]
l108.4 <- getChrmRatio.MA.old("108",4)
l108.4.red <- l108.4[which(l108.4[,3]>0.1),]
l119.4 <- getChrmRatio.MA.old("119",4)
l119.4.red <- l119.4[which(l119.4[,3]>0.1),]

l001.5.old <- getChrmRatio.MA.old("001",5)
l002.5.old <- getChrmRatio.MA.old("002",5)
l003.5.old <- getChrmRatio.MA.old("003",5)
l004.5.old <- getChrmRatio.MA.old("004",5)
l005.5.old <- getChrmRatio.MA.old("005",5)
l006.5.old <- getChrmRatio.MA.old("006",5)
l007.5.old <- getChrmRatio.MA.old("007",5)
l008.5.old <- getChrmRatio.MA.old("008",5)
l009.5.old <- getChrmRatio.MA.old("009",5)
l011.5.old <- getChrmRatio.MA.old("011",5)
l015.5 <- getChrmRatio.MA.old("015",5)
l028.5.old <- getChrmRatio.MA.old("028",5)
l088.5 <- getChrmRatio.MA.old("088",5)
l108.5 <- getChrmRatio.MA.old("108",5)
l119.5 <- getChrmRatio.MA.old("119",5)
l001.5.red.old <- l001.5.old[which(l001.5.old[,3]>0.1),]
l002.5.red.old <- l002.5.old[which(l002.5.old[,3]>0.1),]
l003.5.red.old <- l003.5.old[which(l003.5.old[,3]>0.1),]
l004.5.red.old <- l004.5.old[which(l004.5.old[,3]>0.1),]
l005.5.red.old <- l005.5.old[which(l005.5.old[,3]>0.1),]
l006.5.red.old <- l006.5.old[which(l006.5.old[,3]>0.1),]
l007.5.red.old <- l007.5.old[which(l007.5.old[,3]>0.1),]
l008.5.red.old <- l008.5.old[which(l008.5.old[,3]>0.1),]
l009.5.red.old <- l009.5.old[which(l009.5.old[,3]>0.1),]
l011.5.red.old <- l011.5.old[which(l011.5.old[,3]>0.1),]
l015.5.red <- l015.5[which(l015.5[,3]>0.1),]
l028.5.red.old <- l028.5.old[which(l028.5.old[,3]>0.1),]
l088.5.red <- l088.5[which(l088.5[,3]>0.1),]
l108.5.red <- l108.5[which(l108.5[,3]>0.1),]
l119.5.red <- l119.5[which(l119.5[,3]>0.1),]

l001.6.old <- getChrmRatio.MA.old("001",6)
l002.6.old <- getChrmRatio.MA.old("002",6)
l003.6.old <- getChrmRatio.MA.old("003",6)
l004.6.old <- getChrmRatio.MA.old("004",6)
l005.6.old <- getChrmRatio.MA.old("005",6)
l006.6.old <- getChrmRatio.MA.old("006",6)
l007.6.old <- getChrmRatio.MA.old("007",6)
l008.6.old <- getChrmRatio.MA.old("008",6)
l009.6.old <- getChrmRatio.MA.old("009",6)
l011.6.old <- getChrmRatio.MA.old("011",6)
l015.6 <- getChrmRatio.MA.old("015",6)
l028.6.old <- getChrmRatio.MA.old("028",6)
l088.6 <- getChrmRatio.MA.old("088",6)
l108.6 <- getChrmRatio.MA.old("108",6)
l119.6 <- getChrmRatio.MA.old("119",6)
l001.6.red.old <- l001.6.old[which(l001.6.old[,3]>0.1),]
l002.6.red.old <- l002.6.old[which(l002.6.old[,3]>0.1),]
l003.6.red.old <- l003.6.old[which(l003.6.old[,3]>0.1),]
l004.6.red.old <- l004.6.old[which(l004.6.old[,3]>0.1),]
l005.6.red.old <- l005.6.old[which(l005.6.old[,3]>0.1),]
l006.6.red.old <- l006.6.old[which(l006.6.old[,3]>0.1),]
l007.6.red.old <- l007.6.old[which(l007.6.old[,3]>0.1),]
l008.6.red.old <- l008.6.old[which(l008.6.old[,3]>0.1),]
l009.6.red.old <- l009.6.old[which(l009.6.old[,3]>0.1),]
l011.6.red.old <- l011.6.old[which(l011.6.old[,3]>0.1),]
l015.6.red <- l015.6[which(l015.6[,3]>0.1),]
l028.6.red.old <- l028.6.old[which(l028.6.old[,3]>0.1),]
l088.6.red <- l088.6[which(l088.6[,3]>0.1),]
l108.6.red <- l108.6[which(l108.6[,3]>0.1),]
l119.6.red <- l119.6[which(l119.6[,3]>0.1),]

l001.7.old <- getChrmRatio.MA.old("001",7)
l002.7.old <- getChrmRatio.MA.old("002",7)
l003.7.old <- getChrmRatio.MA.old("003",7)
l004.7.old <- getChrmRatio.MA.old("004",7)
l005.7.old <- getChrmRatio.MA.old("005",7)
l006.7.old <- getChrmRatio.MA.old("006",7)
l007.7.old <- getChrmRatio.MA.old("007",7)
l008.7.old <- getChrmRatio.MA.old("008",7)
l009.7.old <- getChrmRatio.MA.old("009",7)
l011.7.old <- getChrmRatio.MA.old("011",7)
l015.7 <- getChrmRatio.MA.old("015",7)
l028.7.old <- getChrmRatio.MA.old("028",7)
l088.7 <- getChrmRatio.MA.old("088",7)
l108.7 <- getChrmRatio.MA.old("108",7)
l119.7 <- getChrmRatio.MA.old("119",7)
l001.7.red.old <- l001.7.old[which(l001.7.old[,3]>0.1),]
l002.7.red.old <- l002.7.old[which(l002.7.old[,3]>0.1),]
l003.7.red.old <- l003.7.old[which(l003.7.old[,3]>0.1),]
l004.7.red.old <- l004.7.old[which(l004.7.old[,3]>0.1),]
l005.7.red.old <- l005.7.old[which(l005.7.old[,3]>0.1),]
l006.7.red.old <- l006.7.old[which(l006.7.old[,3]>0.1),]
l007.7.red.old <- l007.7.old[which(l007.7.old[,3]>0.1),]
l008.7.red.old <- l008.7.old[which(l008.7.old[,3]>0.1),]
l009.7.red.old <- l009.7.old[which(l009.7.old[,3]>0.1),]
l011.7.red.old <- l011.7.old[which(l011.7.old[,3]>0.1),]
l015.7.red <- l015.7[which(l015.7[,3]>0.1),]
l028.7.red.old <- l028.7.old[which(l028.7.old[,3]>0.1),]
l088.7.red <- l088.7[which(l088.7[,3]>0.1),]
l108.7.red <- l108.7[which(l108.7[,3]>0.1),]
l119.7.red <- l119.7[which(l119.7[,3]>0.1),]

l001.8.old <- getChrmRatio.MA.old("001",8)
l002.8.old <- getChrmRatio.MA.old("002",8)
l003.8.old <- getChrmRatio.MA.old("003",8)
l004.8.old <- getChrmRatio.MA.old("004",8)
l005.8.old <- getChrmRatio.MA.old("005",8)
l006.8.old <- getChrmRatio.MA.old("006",8)
l007.8.old <- getChrmRatio.MA.old("007",8)
l008.8.old <- getChrmRatio.MA.old("008",8)
l009.8.old <- getChrmRatio.MA.old("009",8)
l011.8.old <- getChrmRatio.MA.old("011",8)
l015.8 <- getChrmRatio.MA.old("015",8)
l028.8.old <- getChrmRatio.MA.old("028",8)
l088.8 <- getChrmRatio.MA.old("088",8)
l108.8 <- getChrmRatio.MA.old("108",8)
l119.8 <- getChrmRatio.MA.old("119",8)
l001.8.red.old <- l001.8.old[which(l001.8.old[,3]>0.1),]
l002.8.red.old <- l002.8.old[which(l002.8.old[,3]>0.1),]
l003.8.red.old <- l003.8.old[which(l003.8.old[,3]>0.1),]
l004.8.red.old <- l004.8.old[which(l004.8.old[,3]>0.1),]
l005.8.red.old <- l005.8.old[which(l005.8.old[,3]>0.1),]
l006.8.red.old <- l006.8.old[which(l006.8.old[,3]>0.1),]
l007.8.red.old <- l007.8.old[which(l007.8.old[,3]>0.1),]
l008.8.red.old <- l008.8.old[which(l008.8.old[,3]>0.1),]
l009.8.red.old <- l009.8.old[which(l009.8.old[,3]>0.1),]
l011.8.red.old <- l011.8.old[which(l011.8.old[,3]>0.1),]
l015.8.red <- l015.8[which(l015.8[,3]>0.1),]
l028.8.red.old <- l028.8.old[which(l028.8.old[,3]>0.1),]
l088.8.red <- l088.8[which(l088.8[,3]>0.1),]
l108.8.red <- l108.8[which(l108.8[,3]>0.1),]
l119.8.red <- l119.8[which(l119.8[,3]>0.1),]

l001.9.old <- getChrmRatio.MA.old("001",9)
l002.9.old <- getChrmRatio.MA.old("002",9)
l003.9.old <- getChrmRatio.MA.old("003",9)
l004.9.old <- getChrmRatio.MA.old("004",9)
l005.9.old <- getChrmRatio.MA.old("005",9)
l006.9.old <- getChrmRatio.MA.old("006",9)
l007.9.old <- getChrmRatio.MA.old("007",9)
l008.9.old <- getChrmRatio.MA.old("008",9)
l009.9.old <- getChrmRatio.MA.old("009",9)
l011.9.old <- getChrmRatio.MA.old("011",9)
l015.9 <- getChrmRatio.MA.old("015",9)
l028.9.old <- getChrmRatio.MA.old("028",9)
l088.9 <- getChrmRatio.MA.old("088",9)
l108.9 <- getChrmRatio.MA.old("108",9)
l119.9 <- getChrmRatio.MA.old("119",9)
l001.9.red.old <- l001.9.old[which(l001.9.old[,3]>0.1),]
l002.9.red.old <- l002.9.old[which(l002.9.old[,3]>0.1),]
l003.9.red.old <- l003.9.old[which(l003.9.old[,3]>0.1),]
l004.9.red.old <- l004.9.old[which(l004.9.old[,3]>0.1),]
l005.9.red.old <- l005.9.old[which(l005.9.old[,3]>0.1),]
l006.9.red.old <- l006.9.old[which(l006.9.old[,3]>0.1),]
l007.9.red.old <- l007.9.old[which(l007.9.old[,3]>0.1),]
l008.9.red.old <- l008.9.old[which(l008.9.old[,3]>0.1),]
l009.9.red.old <- l009.9.old[which(l009.9.old[,3]>0.1),]
l011.9.red.old <- l011.9.old[which(l011.9.old[,3]>0.1),]
l015.9.red <- l015.9[which(l015.9[,3]>0.1),]
l028.9.red.old <- l028.9.old[which(l028.9.old[,3]>0.1),]
l088.9.red <- l088.9[which(l088.9[,3]>0.1),]
l108.9.red <- l108.9[which(l108.9[,3]>0.1),]
l119.9.red <- l119.9[which(l119.9[,3]>0.1),]

l001.10.old <- getChrmRatio.MA.old("001",10)
l002.10.old <- getChrmRatio.MA.old("002",10)
l003.10.old <- getChrmRatio.MA.old("003",10)
l004.10.old <- getChrmRatio.MA.old("004",10)
l005.10.old <- getChrmRatio.MA.old("005",10)
l006.10.old <- getChrmRatio.MA.old("006",10)
l007.10.old <- getChrmRatio.MA.old("007",10)
l008.10.old <- getChrmRatio.MA.old("008",10)
l009.10.old <- getChrmRatio.MA.old("009",10)
l011.10.old <- getChrmRatio.MA.old("011",10)
l015.10 <- getChrmRatio.MA.old("015",10)
l028.10.old <- getChrmRatio.MA.old("028",10)
l088.10 <- getChrmRatio.MA.old("088",10)
l108.10 <- getChrmRatio.MA.old("108",10)
l119.10 <- getChrmRatio.MA.old("119",10)
l001.10.red.old <- l001.10.old[which(l001.10.old[,3]>0.1),]
l002.10.red.old <- l002.10.old[which(l002.10.old[,3]>0.1),]
l003.10.red.old <- l003.10.old[which(l003.10.old[,3]>0.1),]
l004.10.red.old<- l004.10.old[which(l004.10.old[,3]>0.1),]
l005.10.red.old <- l005.10.old[which(l005.10.old[,3]>0.1),]
l006.10.red.old <- l006.10.old[which(l006.10.old[,3]>0.1),]
l007.10.red.old <- l007.10.old[which(l007.10.old[,3]>0.1),]
l008.10.red.old <- l008.10.old[which(l008.10.old[,3]>0.1),]
l009.10.red.old <- l009.10.old[which(l009.10.old[,3]>0.1),]
l011.10.red.old <- l011.10.old[which(l011.10.old[,3]>0.1),]
l015.10.red <- l015.10[which(l015.10[,3]>0.1),]
l028.10.red.old <- l028.10.old[which(l028.10.old[,3]>0.1),]
l088.10.red <- l088.10[which(l088.10[,3]>0.1),]
l108.10.red <- l108.10[which(l108.10[,3]>0.1),]
l119.10.red <- l119.10[which(l119.10[,3]>0.1),]

l001.11.old <- getChrmRatio.MA.old("001",11)
l002.11.old <- getChrmRatio.MA.old("002",11)
l003.11.old <- getChrmRatio.MA.old("003",11)
l004.11.old <- getChrmRatio.MA.old("004",11)
l005.11.old <- getChrmRatio.MA.old("005",11)
l006.11.old <- getChrmRatio.MA.old("006",11)
l007.11.old <- getChrmRatio.MA.old("007",11)
l008.11.old <- getChrmRatio.MA.old("008",11)
l009.11.old <- getChrmRatio.MA.old("009",11)
l011.11.old <- getChrmRatio.MA.old("011",11)
l015.11 <- getChrmRatio.MA.old("015",11)
l028.11.old <- getChrmRatio.MA.old("028",11)
l088.11 <- getChrmRatio.MA.old("088",11)
l108.11 <- getChrmRatio.MA.old("108",11)
l119.11 <- getChrmRatio.MA.old("119",11)
l001.11.red.old <- l001.11.old[which(l001.11.old[,3]>0.1),]
l002.11.red.old <- l002.11.old[which(l002.11.old[,3]>0.1),]
l003.11.red.old <- l003.11.old[which(l003.11.old[,3]>0.1),]
l004.11.red.old <- l004.11.old[which(l004.11.old[,3]>0.1),]
l005.11.red.old <- l005.11.old[which(l005.11.old[,3]>0.1),]
l006.11.red.old <- l006.11.old[which(l006.11.old[,3]>0.1),]
l007.11.red.old <- l007.11.old[which(l007.11.old[,3]>0.1),]
l008.11.red.old <- l008.11.old[which(l008.11.old[,3]>0.1),]
l009.11.red.old <- l009.11.old[which(l009.11.old[,3]>0.1),]
l011.11.red.old <- l011.11.old[which(l011.11.old[,3]>0.1),]
l015.11.red <- l015.11[which(l015.11[,3]>0.1),]
l028.11.red.old <- l028.11.old[which(l028.11.old[,3]>0.1),]
l088.11.red <- l088.11[which(l088.11[,3]>0.1),]
l108.11.red <- l108.11[which(l108.11[,3]>0.1),]
l119.11.red <- l119.11[which(l119.11[,3]>0.1),]

l001.12.old <- getChrmRatio.MA.old("001",12)
l002.12.old <- getChrmRatio.MA.old("002",12)
l003.12.old <- getChrmRatio.MA.old("003",12)
l004.12.old <- getChrmRatio.MA.old("004",12)
l005.12.old <- getChrmRatio.MA.old("005",12)
l006.12.old <- getChrmRatio.MA.old("006",12)
l007.12.old <- getChrmRatio.MA.old("007",12)
l008.12.old <- getChrmRatio.MA.old("008",12)
l009.12.old <- getChrmRatio.MA.old("009",12)
l011.12.old <- getChrmRatio.MA.old("011",12)
l015.12 <- getChrmRatio.MA.old("015",12)
l028.12.old <- getChrmRatio.MA.old("028",12)
l088.12 <- getChrmRatio.MA.old("088",12)
l108.12 <- getChrmRatio.MA.old("108",12)
l119.12 <- getChrmRatio.MA.old("119",12)
l001.12.red.old <- l001.12.old[which(l001.12.old[,3]>0.1),]
l002.12.red.old <- l002.12.old[which(l002.12.old[,3]>0.1),]
l003.12.red.old <- l003.12.old[which(l003.12.old[,3]>0.1),]
l004.12.red.old <- l004.12.old[which(l004.12.old[,3]>0.1),]
l005.12.red.old <- l005.12.old[which(l005.12.old[,3]>0.1),]
l006.12.red.old <- l006.12.old[which(l006.12.old[,3]>0.1),]
l007.12.red.old <- l007.12.old[which(l007.12.old[,3]>0.1),]
l008.12.red.old <- l008.12.old[which(l008.12.old[,3]>0.1),]
l009.12.red.old <- l009.12.old[which(l009.12.old[,3]>0.1),]
l011.12.red.old <- l011.12.old[which(l011.12.old[,3]>0.1),]
l015.12.red <- l015.12[which(l015.12[,3]>0.1),]
l028.12.red.old <- l028.12.old[which(l028.12.old[,3]>0.1),]
l088.12.red <- l088.12[which(l088.12[,3]>0.1),]
l108.12.red <- l108.12[which(l108.12[,3]>0.1),]
l119.12.red <- l119.12[which(l119.12[,3]>0.1),]

l001.13.old <- getChrmRatio.MA.old("001",13)
l002.13.old <- getChrmRatio.MA.old("002",13)
l003.13.old <- getChrmRatio.MA.old("003",13)
l004.13.old <- getChrmRatio.MA.old("004",13)
l005.13.old <- getChrmRatio.MA.old("005",13)
l006.13.old <- getChrmRatio.MA.old("006",13)
l007.13.old <- getChrmRatio.MA.old("007",13)
l008.13.old <- getChrmRatio.MA.old("008",13)
l009.13.old <- getChrmRatio.MA.old("009",13)
l011.13.old <- getChrmRatio.MA.old("011",13)
l015.13 <- getChrmRatio.MA.old("015",13)
l028.13.old <- getChrmRatio.MA.old("028",13)
l088.13 <- getChrmRatio.MA.old("088",13)
l108.13 <- getChrmRatio.MA.old("108",13)
l119.13 <- getChrmRatio.MA.old("119",13)
l001.13.red.old <- l001.13.old[which(l001.13.old[,3]>0.1),]
l002.13.red.old <- l002.13.old[which(l002.13.old[,3]>0.1),]
l003.13.red.old <- l003.13.old[which(l003.13.old[,3]>0.1),]
l004.13.red.old <- l004.13.old[which(l004.13.old[,3]>0.1),]
l005.13.red.old <- l005.13.old[which(l005.13.old[,3]>0.1),]
l006.13.red.old <- l006.13.old[which(l006.13.old[,3]>0.1),]
l007.13.red.old <- l007.13.old[which(l007.13.old[,3]>0.1),]
l008.13.red.old <- l008.13.old[which(l008.13.old[,3]>0.1),]
l009.13.red.old <- l009.13.old[which(l009.13.old[,3]>0.1),]
l011.13.red.old <- l011.13.old[which(l011.13.old[,3]>0.1),]
l015.13.red <- l015.13[which(l015.13[,3]>0.1),]
l028.13.red.old <- l028.13.old[which(l028.13.old[,3]>0.1),]
l088.13.red <- l088.13[which(l088.13[,3]>0.1),]
l108.13.red <- l108.13[which(l108.13[,3]>0.1),]
l119.13.red <- l119.13[which(l119.13[,3]>0.1),]

l001.14.old <- getChrmRatio.MA.old("001",14)
l002.14.old <- getChrmRatio.MA.old("002",14)
l003.14.old <- getChrmRatio.MA.old("003",14)
l004.14.old <- getChrmRatio.MA.old("004",14)
l005.14.old <- getChrmRatio.MA.old("005",14)
l006.14.old <- getChrmRatio.MA.old("006",14)
l007.14.old <- getChrmRatio.MA.old("007",14)
l008.14.old <- getChrmRatio.MA.old("008",14)
l009.14.old <- getChrmRatio.MA.old("009",14)
l011.14.old <- getChrmRatio.MA.old("011",14)
l015.14 <- getChrmRatio.MA.old("015",14)
l028.14.old <- getChrmRatio.MA.old("028",14)
l088.14 <- getChrmRatio.MA.old("088",14)
l108.14 <- getChrmRatio.MA.old("108",14)
l119.14 <- getChrmRatio.MA.old("119",14)
l001.14.red.old <- l001.14.old[which(l001.14.old[,3]>0.1),]
l002.14.red.old <- l002.14.old[which(l002.14.old[,3]>0.1),]
l003.14.red.old <- l003.14.old[which(l003.14.old[,3]>0.1),]
l004.14.red.old <- l004.14.old[which(l004.14.old[,3]>0.1),]
l005.14.red.old <- l005.14.old[which(l005.14.old[,3]>0.1),]
l006.14.red.old <- l006.14.old[which(l006.14.old[,3]>0.1),]
l007.14.red.old <- l007.14.old[which(l007.14.old[,3]>0.1),]
l008.14.red.old <- l008.14.old[which(l008.14.old[,3]>0.1),]
l009.14.red.old <- l009.14.old[which(l009.14.old[,3]>0.1),]
l011.14.red.old <- l011.14.old[which(l011.14.old[,3]>0.1),]
l015.14.red <- l015.14[which(l015.14[,3]>0.1),]
l028.14.red.old <- l028.14.old[which(l028.14.old[,3]>0.1),]
l088.14.red <- l088.14[which(l088.14[,3]>0.1),]
l108.14.red <- l108.14[which(l108.14[,3]>0.1),]
l119.14.red <- l119.14[which(l119.14[,3]>0.1),]

l001.15.old <- getChrmRatio.MA.old("001",15)
l002.15.old <- getChrmRatio.MA.old("002",15)
l003.15.old <- getChrmRatio.MA.old("003",15)
l004.15.old <- getChrmRatio.MA.old("004",15)
l005.15.old <- getChrmRatio.MA.old("005",15)
l006.15.old <- getChrmRatio.MA.old("006",15)
l007.15.old <- getChrmRatio.MA.old("007",15)
l008.15.old <- getChrmRatio.MA.old("008",15)
l009.15.old <- getChrmRatio.MA.old("009",15)
l011.15.old <- getChrmRatio.MA.old("011",15)
l015.15 <- getChrmRatio.MA.old("015",15)
l028.15.old <- getChrmRatio.MA.old("028",15)
l088.15 <- getChrmRatio.MA.old("088",15)
l108.15 <- getChrmRatio.MA.old("108",15)
l119.15 <- getChrmRatio.MA.old("119",15)
l001.15.red.old <- l001.15.old[which(l001.15.old[,3]>0.1),]
l002.15.red.old <- l002.15.old[which(l002.15.old[,3]>0.1),]
l003.15.red.old <- l003.15.old[which(l003.15.old[,3]>0.1),]
l004.15.red.old <- l004.15.old[which(l004.15.old[,3]>0.1),]
l005.15.red.old <- l005.15.old[which(l005.15.old[,3]>0.1),]
l006.15.red.old <- l006.15.old[which(l006.15.old[,3]>0.1),]
l007.15.red.old <- l007.15.old[which(l007.15.old[,3]>0.1),]
l008.15.red.old <- l008.15.old[which(l008.15.old[,3]>0.1),]
l009.15.red.old <- l009.15.old[which(l009.15.old[,3]>0.1),]
l011.15.red.old <- l011.15.old[which(l011.15.old[,3]>0.1),]
l015.15.red <- l015.15[which(l015.15[,3]>0.1),]
l028.15.red.old <- l028.15.old[which(l028.15.old[,3]>0.1),]
l088.15.red <- l088.15[which(l088.15[,3]>0.1),]
l108.15.red <- l108.15[which(l108.15[,3]>0.1),]
l119.15.red <- l119.15[which(l119.15[,3]>0.1),]

l001.16.old <- getChrmRatio.MA.old("001",16)
l002.16.old <- getChrmRatio.MA.old("002",16)
l003.16.old <- getChrmRatio.MA.old("003",16)
l004.16.old <- getChrmRatio.MA.old("004",16)
l005.16.old <- getChrmRatio.MA.old("005",16)
l006.16.old <- getChrmRatio.MA.old("006",16)
l007.16.old <- getChrmRatio.MA.old("007",16)
l008.16.old <- getChrmRatio.MA.old("008",16)
l009.16.old <- getChrmRatio.MA.old("009",16)
l011.16.old <- getChrmRatio.MA.old("011",16)
l015.16 <- getChrmRatio.MA.old("015",16)
l028.16.old <- getChrmRatio.MA.old("028",16)
l088.16 <- getChrmRatio.MA.old("088",16)
l108.16 <- getChrmRatio.MA.old("108",16)
l119.16 <- getChrmRatio.MA.old("119",16)
l001.16.red.old <- l001.16.old[which(l001.16.old[,3]>0.1),]
l002.16.red.old <- l002.16.old[which(l002.16.old[,3]>0.1),]
l003.16.red.old <- l003.16.old[which(l003.16.old[,3]>0.1),]
l004.16.red.old <- l004.16.old[which(l004.16.old[,3]>0.1),]
l005.16.red.old <- l005.16.old[which(l005.16.old[,3]>0.1),]
l006.16.red.old <- l006.16.old[which(l006.16.old[,3]>0.1),]
l007.16.red.old <- l007.16.old[which(l007.16.old[,3]>0.1),]
l008.16.red.old <- l008.16.old[which(l008.16.old[,3]>0.1),]
l009.16.red.old <- l009.16.old[which(l009.16.old[,3]>0.1),]
l011.16.red.old <- l011.16.old[which(l011.16.old[,3]>0.1),]
l015.16.red <- l015.16[which(l015.16[,3]>0.1),]
l028.16.red.old <- l028.16.old[which(l028.16.old[,3]>0.1),]
l088.16.red <- l088.16[which(l088.16[,3]>0.1),]
l108.16.red <- l108.16[which(l108.16[,3]>0.1),]
l119.16.red <- l119.16[which(l119.16[,3]>0.1),]


```

``` {r new boxplots,echo=FALSE,include=TRUE}

#New boxplot codes

#############################################################################################
####################
#CHROMOSOME 1 
exp3x <- log2(1.5)
exp3x
#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr1Data.old <- data.frame(y=c((log2(l001.1.old[,3])),(log2(l002.1.old[,3])),(log2(l003.1.old[,3])),(log2(l004.1.old[,3])),(log2(l005.1.old[,3])),
        (log2(l006.1.old[,3])),(log2(l007.1.old[,3])),(log2(l008.1.old[,3])),(log2(l009.1.old[,3])),(log2(l011.1.old[,3])),
        (log2(l015.1[,3])),(log2(l028.1.old[,3])),(log2(l088.1[,3])),(log2(l108.1[,3])),(log2(l119.1[,3]))),
Line=rep(c(1.1,"2M","3M","4M","5M","6M","7M","8","9M","11M",15,28,88,108,119),each=(length(l001.1.old$tracking_id))))
chr1Data.old$Line <-factor(chr1Data.old$Line)
#data frame with only euploids, so I can do a euploid by euploid test

aov.chr1.old <- aov(y~Line,data=chr1Data.old)
summary(aov.chr1.old)
resids.chr1.old <- resid(aov.chr1.old)
{pdf("Chrm1.resids.old.pdf")
hist(resids.chr1.old,col="turquoise",breaks=10,xlab="Residuals.Chr.1")}
shapiro.test(resids.chr1.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr1.old)
{pdf("Chrm1.Tukey.old.pdf")
plot(TukeyHSD(aov.chr1.old))}


#This code works beautifully to produce a boxplot 
{pdf("Chrm_1.old.pdf")
plot1.O <-boxplot(y~Line,data=chr1Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","cyan","cyan","cyan","cyan",
              "cyan","cyan"),main="MA Lines Chromsome 1",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
}


#fail shapiro test, so should probably use nonparametric
#kruskal.chr1 <- kruskal.test(y~Line,data=chr1Data)
#kruskal.test(y~Line,data=chr1Data)
#p-value < 2.2 e-16 (So at least one of the data points is very different from the others)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr1Data)
#multiCompsChr1 <- kruskalmc(y~Line,data=chr1Data)
#OR can do many pairwise t-tests
#pairwise.t.test(chr1Data$y,chr1Data$Line,p.adjust.method = "fdr")
#wilcox.chr1 <- data.table(pairwise.wilcox.test(chr1Data$y,chr1Data$Line,p.adjust.method = "fdr"))
#View(wilcox.chr1)
#pairwise.wilcox.test(chr1Data$y,chr1Data$Line,p.adjust.method = "fdr")


####################
#CHROMOSOME 2 

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr2Data.old <- data.frame(y=c((log2(l001.2.old[,3])),(log2(l002.2.old[,3])),(log2(l003.2.old[,3])),(log2(l004.2.old[,3])),(log2(l005.2.old[,3])),
        (log2(l006.2.old[,3])),(log2(l007.2.old[,3])),(log2(l008.2.old[,3])),(log2(l009.2.old[,3])),(log2(l011.2.old[,3])),
        (log2(l015.2[,3])),(log2(l028.2.old[,3])),(log2(l088.2[,3])),(log2(l108.2[,3])),(log2(l119.2[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,11,15,28,88,108,119),each=(length(l001.2.old$tracking_id))))
chr2Data.old$Line <-factor(chr2Data.old$Line)
#data frame with only euploids, so I can do a euploid by euploid test

aov.chr2.old <- aov(y~Line,data=chr2Data.old)
summary(aov.chr2.old)
resids.chr2.old <- resid(aov.chr2.old)
{pdf("Chrm2.resids.old.pdf")
hist(resids.chr2.old,col="turquoise",breaks=10,xlab="Residuals.Chr.2")}
#shapiro.test(resids.chr2.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr2.old)
{pdf("Chrm2.Tukey.old.pdf")
plot(TukeyHSD(aov.chr2.old))}


#This code works beautifully to produce a boxplot 
{pdf("Chrm_2.old.pdf")
boxplot(y~Line,data=chr2Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","cyan","cyan","cyan","cyan",
              "cyan","cyan"),main="MA Lines Chromsome 2",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
}


####################
#CHROMOSOME 3 

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr3Data.old <- data.frame(y=c((log2(l001.3.old[,3])),(log2(l002.3.old[,3])),(log2(l003.3.old[,3])),(log2(l004.3.old[,3])),(log2(l005.3.old[,3])),
        (log2(l006.3.old[,3])),(log2(l007.3.old[,3])),(log2(l008.3.old[,3])),(log2(l009.3.old[,3])),(log2(l011.3.old[,3])),
        (log2(l015.3[,3])),(log2(l028.3.old[,3])),(log2(l088.3[,3])),(log2(l108.3[,3])),(log2(l119.3[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,11,15,28,88,108,119),each=(length(l001.3.old$tracking_id))))
chr3Data.old$Line <-factor(chr3Data.old$Line)
#data frame with only euploids, so I can do a euploid by euploid test

aov.chr3.old <- aov(y~Line,data=chr3Data.old)
summary(aov.chr3.old)
resids.chr3.old <- resid(aov.chr3.old)
{pdf("Chrm3.resids.old.pdf")
hist(resids.chr3.old,col="turquoise",breaks=10,xlab="Residuals.Chr.3")}
#shapiro.test(resids.chr2.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr3.old)
{pdf("Chrm3.Tukey.old.pdf")
plot(TukeyHSD(aov.chr3.old))}


#This code works beautifully to produce a boxplot 
{pdf("Chrm_3.old.pdf")
boxplot(y~Line,data=chr3Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","cyan","cyan","cyan","cyan",
              "cyan","cyan"),main="MA Lines Chromsome 3",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
}


####################
#CHROMOSOME 4

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr4Data.old <- data.frame(y=c((log2(l001.4.old[,3])),(log2(l002.4.old[,3])),(log2(l003.4.old[,3])),(log2(l004.4.old[,3])),(log2(l005.4.old[,3])),
        (log2(l006.4.old[,3])),(log2(l007.4.old[,3])),(log2(l008.4.old[,3])),(log2(l009.4.old[,3])),(log2(l011.4.old[,3])),
        (log2(l015.4[,3])),(log2(l028.4.old[,3])),(log2(l088.4[,3])),(log2(l108.4[,3])),(log2(l119.4[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,11,15,28,88,108,119),each=(length(l001.4.old$tracking_id))))
chr4Data.old$Line <-factor(chr4Data.old$Line)
#data frame with only euploids, so I can do a euploid by euploid test

aov.chr4.old <- aov(y~Line,data=chr4Data.old)
summary(aov.chr4.old)
resids.chr4.old <- resid(aov.chr4.old)
{pdf("Chrm4.resids.old.pdf")
hist(resids.chr4.old,col="turquoise",breaks=10,xlab="Residuals.Chr.4")}
#shapiro.test(resids.chr2.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr4.old)
{pdf("Chrm4.Tukey.old.pdf")
plot(TukeyHSD(aov.chr4.old))}


#This code works beautifully to produce a boxplot 
{pdf("Chrm_4.old.pdf")
boxplot(y~Line,data=chr4Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","cyan","cyan","cyan","cyan",
              "cyan","cyan"),main="MA Lines Chromsome 4",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
}


####################
#CHROMOSOME 5

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr5Data.old <- data.frame(y=c((log2(l001.5.old[,3])),(log2(l002.5.old[,3])),(log2(l003.5.old[,3])),(log2(l004.5.old[,3])),(log2(l005.5.old[,3])),
        (log2(l006.5.old[,3])),(log2(l007.5.old[,3])),(log2(l008.5.old[,3])),(log2(l009.5.old[,3])),(log2(l011.5.old[,3])),
        (log2(l015.5[,3])),(log2(l028.5.old[,3])),(log2(l088.5[,3])),(log2(l108.5[,3])),(log2(l119.5[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,11,15,28,88,108,119),each=(length(l001.5.old$tracking_id))))
chr5Data.old$Line <-factor(chr5Data.old$Line)
#data frame with only euploids, so I can do a euploid by euploid test

aov.chr5.old <- aov(y~Line,data=chr5Data.old)
summary(aov.chr5.old)
resids.chr5.old <- resid(aov.chr5.old)
{pdf("Chrm5.resids.old.pdf")
hist(resids.chr5.old,col="turquoise",breaks=10,xlab="Residuals.Chr.5")}
#shapiro.test(resids.chr2.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr5.old)
{pdf("Chrm5.Tukey.old.pdf")
plot(TukeyHSD(aov.chr5.old))}


#This code works beautifully to produce a boxplot 
{pdf("Chrm_5.old.pdf")
boxplot(y~Line,data=chr5Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","cyan","cyan","cyan","cyan",
              "cyan","cyan"),main="MA Lines Chromsome 5",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
}



####################
#CHROMOSOME 6

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr6Data.old <- data.frame(y=c((log2(l001.6.old[,3])),(log2(l002.6.old[,3])),(log2(l003.6.old[,3])),(log2(l004.6.old[,3])),(log2(l005.6.old[,3])),
        (log2(l006.6.old[,3])),(log2(l007.6.old[,3])),(log2(l008.6.old[,3])),(log2(l009.6.old[,3])),(log2(l011.6.old[,3])),
        (log2(l015.6[,3])),(log2(l028.6.old[,3])),(log2(l088.6[,3])),(log2(l108.6[,3])),(log2(l119.6[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,11,15,28,88,108,119),each=(length(l001.6.old$tracking_id))))
chr6Data.old$Line <-factor(chr6Data.old$Line)
#data frame with only euploids, so I can do a euploid by euploid test

aov.chr6.old <- aov(y~Line,data=chr6Data.old)
summary(aov.chr6.old)
resids.chr6.old <- resid(aov.chr6.old)
{pdf("Chrm6.resids.old.pdf")
hist(resids.chr6.old,col="turquoise",breaks=10,xlab="Residuals.Chr.6")}
#shapiro.test(resids.chr2.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr6.old)
{pdf("Chrm6.Tukey.old.pdf")
plot(TukeyHSD(aov.chr6.old))}


#This code works beautifully to produce a boxplot 
{pdf("Chrm_6.old.pdf")
boxplot(y~Line,data=chr6Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","cyan","cyan","cyan","cyan",
              "cyan","cyan"),main="MA Lines Chromsome 6",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
}

####################
#CHROMOSOME 7

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr7Data.old <- data.frame(y=c((log2(l001.7.old[,3])),(log2(l002.7.old[,3])),(log2(l003.7.old[,3])),(log2(l004.7.old[,3])),(log2(l005.7.old[,3])),
        (log2(l006.7.old[,3])),(log2(l007.7.old[,3])),(log2(l008.7.old[,3])),(log2(l009.7.old[,3])),(log2(l011.7.old[,3])),
        (log2(l015.7[,3])),(log2(l028.7.old[,3])),(log2(l088.7[,3])),(log2(l108.7[,3])),(log2(l119.7[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,11,15,28,88,108,119),each=(length(l001.7.old$tracking_id))))
chr7Data.old$Line <-factor(chr7Data.old$Line)
#data frame with only euploids, so I can do a euploid by euploid test

aov.chr7.old <- aov(y~Line,data=chr7Data.old)
summary(aov.chr7.old)
resids.chr7.old <- resid(aov.chr7.old)
{pdf("Chrm7.resids.old.pdf")
hist(resids.chr7.old,col="turquoise",breaks=10,xlab="Residuals.Chr.7")}
#shapiro.test(resids.chr2.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr7.old)
{pdf("Chrm7.Tukey.old.pdf")
plot(TukeyHSD(aov.chr7.old))}


#This code works beautifully to produce a boxplot 
{pdf("Chrm_7.old.pdf")
boxplot(y~Line,data=chr7Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","cyan","cyan","cyan","cyan",
              "cyan","cyan"),main="MA Lines Chromsome 7",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
}

###################################
#CHROMOSOME 8 NEW CODE 

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr8Data.old <- data.frame(y=c((log2(l001.8.old[,3])),(log2(l002.8.old[,3])),(log2(l003.8.old[,3])),(log2(l004.8.old[,3])),(log2(l005.8.old[,3])),
        (log2(l006.8.old[,3])),(log2(l007.8.old[,3])),(log2(l008.8.old[,3])),(log2(l009.8.old[,3])),(log2(l011.8.old[,3])),
        (log2(l015.8[,3])),(log2(l028.8.old[,3])),(log2(l088.8[,3])),(log2(l108.8[,3])),(log2(l119.8[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,11,15,28,88,108,119),each=(length(l001.8.old$tracking_id))))
chr8Data.old$Line <-factor(chr8Data.old$Line)
#data frame with only euploids, so I can do a euploid by euploid test

#I just removed line 9,15,88,119 since it's aneuploid for not chromosome 8 
chr8Data.old.r <- data.frame(y=c((log2(l001.8.old[,3])),(log2(l002.8.old[,3])),(log2(l003.8.old[,3])),(log2(l004.8.old[,3])),(log2(l005.8.old[,3])),(log2(l006.8.old[,3])),(log2(l007.8.old[,3])),(log2(l008.8.old[,3])),(log2(l011.8.old[,3])),(log2(l028.8.old[,3])),(log2(l108.8[,3]))),Line=rep(c(1,2,3,4,5,6,7,8,11,28,108),each=(length(l001.8.old$tracking_id))))

chr8Data.old.r$Line <-factor(chr8Data.old.r$Line)

aov.chr8.old <- aov(y~Line,data=chr8Data.old.r)
summary(aov.chr8.old)
resids.chr8.old <- resid(aov.chr8.old)
{pdf("Chrm8.resids.old.pdf")
hist(resids.chr8.old,col="turquoise",breaks=10,xlab="Residuals.Chr.8")}
#shapiro.test(resids.chr8.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr8.old)
{pdf("Chrm8.Tukey.old.pdf")
plot(TukeyHSD(aov.chr8.old))}


#This code works beautifully to produce a boxplot 
{pdf("Chrm_8.old.pdf")
boxplot(y~Line,data=chr8Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","cyan","green","cyan","cyan",
              "cyan","deeppink1","cyan"),main="MA Lines Chromsome 8",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


text(x=13.7, y=1.5, "*", pos=3, cex=2, col="red")
text(x=14, y=1.5, "*", pos=3, cex=2, col="red")
text(x=14.3, y=1.5, "*", pos=3, cex=2, col="red")
}

t.test((log2(l108.8[,3])),mu=exp3x)


###############################################################################################
#CHROMOSOME 9 NEW CODE 

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr9Data.old <- data.frame(y=c((log2(l001.9.old[,3])),(log2(l002.9.old[,3])),(log2(l003.9.old[,3])),(log2(l004.9.old[,3])),(log2(l005.9.old[,3])),
        (log2(l006.9.old[,3])),(log2(l007.9.old[,3])),(log2(l008.9.old[,3])),(log2(l009.9.old[,3])),(log2(l011.9.old[,3])),
        (log2(l015.9[,3])),(log2(l028.9.old[,3])),(log2(l088.9[,3])),(log2(l108.9[,3])),(log2(l119.9[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,11,15,28,88,108,119),each=(length(l001.9.old$tracking_id))))
chr9Data.old$Line <-factor(chr9Data.old$Line)
#data frame with only euploids, so I can do a euploid by euploid test

#I just removed line 9 since it's aneuploid for not chromosome 9 
chr9Data.old.r <- data.frame(y=c((log2(l001.9.old[,3])),(log2(l002.9.old[,3])),(log2(l003.9.old[,3])),(log2(l004.9.old[,3])),(log2(l005.9.old[,3])),
        (log2(l006.9.old[,3])),(log2(l007.9.old[,3])),(log2(l008.9.old[,3])),(log2(l011.9.old[,3])),
        (log2(l015.9[,3])),(log2(l028.9.old[,3])),(log2(l088.9[,3])),(log2(l108.9[,3])),(log2(l119.9[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,11,15,28,88,108,119),each=(length(l001.9.old$tracking_id))))
chr9Data.old.r$Line <-factor(chr9Data.old.r$Line)

aov.chr9.old <- aov(y~Line,data=chr9Data.old.r)
summary(aov.chr9.old)
resids.chr9.old <- resid(aov.chr9.old)
{pdf("Chrm9.resids.old.pdf")
hist(resids.chr9.old,col="turquoise",breaks=10,xlab="Residuals.Chr.9")}
#shapiro.test(resids.chr9.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr9.old)
{pdf("Chrm9.Tukey.old.pdf")
plot(TukeyHSD(aov.chr9.old))}


#This code works beautifully to produce a boxplot 
{pdf("Chrm_9.old.pdf")
boxplot(y~Line,data=chr9Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","cyan","green","deeppink1","green",
              "deeppink1","orange","deeppink1"),main="MA Lines Chromsome 9",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

text(x=10.7, y=1.8, "*", pos=3, cex=2, col="red")
text(x=11, y=1.8, "*", pos=3, cex=2, col="red")
text(x=11.3, y=1.8, "*", pos=3, cex=2, col="red")

text(x=12.7, y=1.8, "*", pos=3, cex=2, col="red")
text(x=13, y=1.8, "*", pos=3, cex=2, col="red")
text(x=13.3, y=1.8, "*", pos=3, cex=2, col="red")

text(x=13.7, y=1.5, "*", pos=3, cex=2, col="red")
text(x=14, y=1.5, "*", pos=3, cex=2, col="red")
text(x=14.3, y=1.5, "*", pos=3, cex=2, col="red")

text(x=14.7, y=1.8, "*", pos=3, cex=2, col="red")
text(x=15, y=1.8, "*", pos=3, cex=2, col="red")
text(x=15.3, y=1.8, "*", pos=3, cex=2, col="red")
}

t.test((log2(l015.9[,3])),mu=exp3x)
t.test((log2(l088.9[,3])),mu=exp3x)
t.test((log2(l108.9[,3])),mu=exp1x)
t.test((log2(l119.9[,3])),mu=exp3x)

###############################################################################################
#CHROMOSOME 10 NEW CODE 

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr10Data.old <- data.frame(y=c((log2(l001.10.old[,3])),(log2(l002.10.old[,3])),(log2(l003.10.old[,3])),(log2(l004.10.old[,3])),(log2(l005.10.old[,3])),
        (log2(l006.10.old[,3])),(log2(l007.10.old[,3])),(log2(l008.10.old[,3])),(log2(l009.10.old[,3])),(log2(l011.10.old[,3])),
        (log2(l015.10[,3])),(log2(l028.10.old[,3])),(log2(l088.10[,3])),(log2(l108.10[,3])),(log2(l119.10[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,11,15,28,88,108,119),each=(length(l001.10.old$tracking_id))))
chr10Data.old$Line <-factor(chr10Data.old$Line)

aov.chr10.old <- aov(y~Line,data=chr10Data.old)
summary(aov.chr10.old)
resids.chr10.old <- resid(aov.chr10.old)
{pdf("Chrm10.resids.old.pdf")
hist(resids.chr10.old,col="turquoise",breaks=10,xlab="Residuals.Chr.10")}
#shapiro.test(resids.chr10.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr10.old)
{pdf("Chrm10.Tukey.old.pdf")
plot(TukeyHSD(aov.chr10.old))}

#This code works beautifully to produce a boxplot 
{pdf("Chrm_10.old.pdf")
boxplot(y~Line,data=chr10Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","cyan","green","cyan","green",
              "cyan","cyan","cyan"),main="MA Lines Chromsome 10",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
}

###############################################################################################
#CHROMOSOME 11 NEW CODE 

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr11Data.old <- data.frame(y=c((log2(l001.11.old[,3])),(log2(l002.11.old[,3])),(log2(l003.11.old[,3])),(log2(l004.11.old[,3])),(log2(l005.11.old[,3])),
        (log2(l006.11.old[,3])),(log2(l007.11.old[,3])),(log2(l008.11.old[,3])),(log2(l009.11.old[,3])),(log2(l011.11.old[,3])),
        (log2(l015.11[,3])),(log2(l028.11.old[,3])),(log2(l088.11[,3])),(log2(l108.11[,3])),(log2(l119.11[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,11,15,28,88,108,119),each=(length(l001.11.old$tracking_id))))
chr11Data.old$Line <-factor(chr11Data.old$Line)

aov.chr11.old <- aov(y~Line,data=chr11Data.old)
summary(aov.chr11.old)
resids.chr11.old <- resid(aov.chr11.old)
{pdf("Chrm11.resids.old.pdf")
hist(resids.chr11.old,col="turquoise",breaks=10,xlab="Residuals.Chr.11")}
#shapiro.test(resids.chr10.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr11.old)
{pdf("Chrm11.Tukey.old.pdf")
plot(TukeyHSD(aov.chr11.old))}

#This code works beautifully to produce a boxplot 
{pdf("Chrm_11.old.pdf")
boxplot(y~Line,data=chr11Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","cyan","green","cyan","green",
              "cyan","cyan","cyan"),main="MA Lines Chromsome 11",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
}

###############################################################################################
#CHROMOSOME 12 NEW CODE 

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr12Data.old <- data.frame(y=c((log2(l001.12.old[,3])),(log2(l002.12.old[,3])),(log2(l003.12.old[,3])),(log2(l004.12.old[,3])),(log2(l005.12.old[,3])),
        (log2(l006.12.old[,3])),(log2(l007.12.old[,3])),(log2(l008.12.old[,3])),(log2(l009.12.old[,3])),(log2(l011.12.old[,3])),
        (log2(l015.12[,3])),(log2(l028.12.old[,3])),(log2(l088.12[,3])),(log2(l108.12[,3])),(log2(l119.12[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,11,15,28,88,108,119),each=(length(l001.12.old$tracking_id))))
chr12Data.old$Line <-factor(chr12Data.old$Line)

aov.chr12.old <- aov(y~Line,data=chr12Data.old)
summary(aov.chr12.old)
resids.chr12.old <- resid(aov.chr12.old)
{pdf("Chrm12.resids.old.pdf")
hist(resids.chr12.old,col="turquoise",breaks=10,xlab="Residuals.Chr.12")}
#shapiro.test(resids.chr10.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr12.old)
{pdf("Chrm12.Tukey.old.pdf")
plot(TukeyHSD(aov.chr12.old))}

#This code works beautifully to produce a boxplot 
{pdf("Chrm_12.old.pdf")
boxplot(y~Line,data=chr12Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","cyan","green","cyan","green",
              "cyan","cyan","cyan"),main="MA Lines Chromsome 12",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
}

###############################################################################################
#CHROMOSOME 13 NEW CODE 

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr13Data.old <- data.frame(y=c((log2(l001.13.old[,3])),(log2(l002.13.old[,3])),(log2(l003.13.old[,3])),(log2(l004.13.old[,3])),(log2(l005.13.old[,3])),
        (log2(l006.13.old[,3])),(log2(l007.13.old[,3])),(log2(l008.13.old[,3])),(log2(l009.13.old[,3])),(log2(l011.13.old[,3])),
        (log2(l015.13[,3])),(log2(l028.13.old[,3])),(log2(l088.13[,3])),(log2(l108.13[,3])),(log2(l119.13[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,11,15,28,88,108,119),each=(length(l001.13.old$tracking_id))))
chr13Data.old$Line <-factor(chr13Data.old$Line)

aov.chr13.old <- aov(y~Line,data=chr13Data.old)
summary(aov.chr13.old)
resids.chr13.old <- resid(aov.chr13.old)
{pdf("Chrm13.resids.old.pdf")
hist(resids.chr13.old,col="turquoise",breaks=10,xlab="Residuals.Chr.13")}
#shapiro.test(resids.chr10.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr13.old)
{pdf("Chrm13.Tukey.old.pdf")
plot(TukeyHSD(aov.chr13.old))}

#This code works beautifully to produce a boxplot 
{pdf("Chrm_13.old.pdf")
boxplot(y~Line,data=chr13Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","cyan","green","cyan","green",
              "cyan","cyan","cyan"),main="MA Lines Chromsome 13",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
}

###############################################################################################
#CHROMOSOME 14 NEW CODE 

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr14Data.old <- data.frame(y=c((log2(l001.14.old[,3])),(log2(l002.14.old[,3])),(log2(l003.14.old[,3])),(log2(l004.14.old[,3])),(log2(l005.14.old[,3])),
        (log2(l006.14.old[,3])),(log2(l007.14.old[,3])),(log2(l008.14.old[,3])),(log2(l009.14.old[,3])),(log2(l011.14.old[,3])),
        (log2(l015.14[,3])),(log2(l028.14.old[,3])),(log2(l088.14[,3])),(log2(l108.14[,3])),(log2(l119.14[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,11,15,28,88,108,119),each=(length(l001.14.old$tracking_id))))
chr14Data.old$Line <-factor(chr14Data.old$Line)

chr14Data.old.r <- data.frame(y=c((log2(l001.14.old[,3])),(log2(l002.14.old[,3])),(log2(l003.14.old[,3])),(log2(l004.14.old[,3])),(log2(l005.14.old[,3])),
        (log2(l006.14.old[,3])),(log2(l007.14.old[,3])),(log2(l008.14.old[,3])),(log2(l009.14.old[,3])),(log2(l028.14.old[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,28),each=(length(l001.14.old$tracking_id))))
chr14Data.old.r$Line <-factor(chr14Data.old.r$Line)


aov.chr14.old <- aov(y~Line,data=chr14Data.old.r)
summary(aov.chr14.old)
resids.chr14.old <- resid(aov.chr14.old)
{pdf("Chrm14.resids.old.pdf")
hist(resids.chr14.old,col="turquoise",breaks=10,xlab="Residuals.Chr.14")}
#shapiro.test(resids.chr10.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr14.old)
{pdf("Chrm14.Tukey.old.pdf")
plot(TukeyHSD(aov.chr14.old))}

#This code works beautifully to produce a boxplot 
{pdf("Chrm_14.old.pdf")
boxplot(y~Line,data=chr14Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","deeppink1","green","cyan","green",
              "cyan","cyan","cyan"),main="MA Lines Chromsome 14",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

text(x=8.7, y=2, "*", pos=3, cex=2, col="red")
text(x=9, y=2, "*", pos=3, cex=2, col="red")
text(x=9.3, y=2, "*", pos=3, cex=2, col="red")


}

###############################################################################################
#CHROMOSOME 15 NEW CODE 

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr15Data.old <- data.frame(y=c((log2(l001.15.old[,3])),(log2(l002.15.old[,3])),(log2(l003.15.old[,3])),(log2(l004.15.old[,3])),(log2(l005.15.old[,3])),
        (log2(l006.15.old[,3])),(log2(l007.15.old[,3])),(log2(l008.15.old[,3])),(log2(l009.15.old[,3])),(log2(l011.15.old[,3])),
        (log2(l015.15[,3])),(log2(l028.15.old[,3])),(log2(l088.15[,3])),(log2(l108.15[,3])),(log2(l119.15[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,11,15,28,88,108,119),each=(length(l001.15.old$tracking_id))))
chr15Data.old$Line <-factor(chr15Data.old$Line)

aov.chr15.old <- aov(y~Line,data=chr15Data.old)
summary(aov.chr15.old)
resids.chr15.old <- resid(aov.chr15.old)
{pdf("Chrm15.resids.old.pdf")
hist(resids.chr15.old,col="turquoise",breaks=10,xlab="Residuals.Chr.15")}
#shapiro.test(resids.chr10.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr15.old)
{pdf("Chrm15.Tukey.old.pdf")
plot(TukeyHSD(aov.chr15.old))}

#This code works beautifully to produce a boxplot 
{pdf("Chrm_15.old.pdf")
boxplot(y~Line,data=chr15Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","cyan","green","cyan","green",
              "cyan","cyan","cyan"),main="MA Lines Chromsome 15",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
}

###############################################################################################
#CHROMOSOME 16 NEW CODE 

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
chr16Data.old <- data.frame(y=c((log2(l001.16.old[,3])),(log2(l002.16.old[,3])),(log2(l003.16.old[,3])),(log2(l004.16.old[,3])),(log2(l005.16.old[,3])),
        (log2(l006.16.old[,3])),(log2(l007.16.old[,3])),(log2(l008.16.old[,3])),(log2(l009.16.old[,3])),(log2(l011.16.old[,3])),
        (log2(l015.16[,3])),(log2(l028.16.old[,3])),(log2(l088.16[,3])),(log2(l108.16[,3])),(log2(l119.16[,3]))),
Line=rep(c(1,2,3,4,5,6,7,8,9,11,15,28,88,108,119),each=(length(l001.16.old$tracking_id))))
chr16Data.old$Line <-factor(chr16Data.old$Line)

aov.chr16.old <- aov(y~Line,data=chr16Data.old)
summary(aov.chr16.old)
resids.chr16.old <- resid(aov.chr16.old)
{pdf("Chrm16.resids.old.pdf")
hist(resids.chr16.old,col="turquoise",breaks=10,xlab="Residuals.Chr.16")}
#shapiro.test(resids.chr10.old)
#fail shapiro test, so should probably use nonparametric
#test pairwise comparisons
TukeyHSD(aov.chr16.old)
{pdf("Chrm16.Tukey.old.pdf")
plot(TukeyHSD(aov.chr16.old))}

#This code works beautifully to produce a boxplot 
{pdf("Chrm_16.old.pdf")
boxplot(y~Line,data=chr16Data.old,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","green","green","green","green","green","cyan","green","cyan","green",
              "cyan","cyan","cyan"),main="MA Lines Chromsome 16",las=3,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
}

```

``` {r old boxplot codes,echo=FALSE,include=TRUE}
############################################################################################
#CHROMOSOME 1 
par(mar=c(5, 4, 4, 2) + 0.1)
par(mfrow=c(1,1))
#going to do this for all lines for chromosome 1 
#try to plot them all on the same graph? 

#euRatio.test <- c(logMat2.GC.test[,7],logMat2.GC.test[,7])
{pdf("Chrm_1_old.pdf")
boxplot((log2(l001.1.old[,3])),(log2(l002.1.old[,3])),(log2(l003.1.old[,3])),(log2(l004.1.old[,3])),(log2(l005.1.old[,3])),
        (log2(l006.1.old[,3])),(log2(l007.1.old[,3])),(log2(l008.1.old[,3])),(log2(l009.1.old[,3])),(log2(l011.1.old[,3])),
        (log2(l015.1[,3])),(log2(l028.1.old[,3])),(log2(l088.1[,3])),(log2(l108.1[,3])),(log2(l119.1[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 1",las=3
        ,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

euRatio.c1.old <- c(l001.1.old[,3],l002.1.old[,3],l003.1.old[,3],l004.1.old[,3],l005.1.old[,3],l006.1.old[,3],l007.1.old[,3],l008.1.old[,3],l009.1.old[,3],l011.1.old[,3],l028.1.old[,3])
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.1.old[,3])), (log2(euRatio.c1.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.1.old[,3])), (log2(euRatio.c1.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.1.old[,3])), (log2(euRatio.c1.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.1.old[,3])), (log2(euRatio.c1.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.1.old[,3])), (log2(euRatio.c1.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.1.old[,3])), (log2(euRatio.c1.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.1.old[,3])), (log2(euRatio.c1.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.1.old[,3])), (log2(euRatio.c1.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.1.old[,3])), (log2(euRatio.c1.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.1.old[,3])), (log2(euRatio.c1.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.1[,3])), (log2(euRatio.c1.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.1.old[,3])), (log2(euRatio.c1.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.1[,3])), (log2(euRatio.c1.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.1[,3])), (log2(euRatio.c1.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.1[,3])), (log2(euRatio.c1.old)), paired=FALSE, var.equal=TRUE)
dev.off()}
#############################################################################################

#CHROMOSOME 1 Reduced
par(mar=c(5, 4, 4, 2) + 0.1)
par(mfrow=c(1,1))
#going to do this for all lines for chromosome 1 
#try to plot them all on the same graph? 

#euRatio.test <- c(logMat2.GC.test[,7],logMat2.GC.test[,7])
{pdf("Chrm_1_red_old.pdf")
boxplot((log2(l001.1.red.old[,3])),(log2(l002.1.red.old[,3])),(log2(l003.1.red.old[,3])),(log2(l004.1.red.old[,3])),(log2(l005.1.red.old[,3])),
        (log2(l006.1.red.old[,3])),(log2(l007.1.red.old[,3])),(log2(l008.1.red.old[,3])),(log2(l009.1.red.old[,3])),(log2(l011.1.red.old[,3])),
        (log2(l015.1.red[,3])),(log2(l028.1.red.old[,3])),(log2(l088.1.red[,3])),(log2(l108.1.red[,3])),(log2(l119.1.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 1 Reduced",las=3
        ,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
#p3 <- 0.0006221
#mylabel = bquote(italic(p) == .(format(p3, digits = 9)))
#text(x=2.3,y = 2.5, labels = mylabel,cex=.8)
euRatio.c1.red.old <- c(l001.1.red.old[,3],l002.1.red.old[,3],l003.1.red.old[,3],l004.1.red.old[,3],l005.1.red.old[,3],l006.1.red.old[,3],l007.1.red.old[,3],l008.1.red.old[,3],l009.1.red.old[,3],l011.1.red.old[,3],l028.1.red.old[,3])
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.1.red.old[,3])), (log2(euRatio.c1.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.1.red.old[,3])), (log2(euRatio.c1.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.1.red.old[,3])), (log2(euRatio.c1.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.1.red.old[,3])), (log2(euRatio.c1.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.1.red.old[,3])), (log2(euRatio.c1.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.1.red.old[,3])), (log2(euRatio.c1.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.1.red.old[,3])), (log2(euRatio.c1.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.1.red.old[,3])), (log2(euRatio.c1.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.1.red.old[,3])), (log2(euRatio.c1.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.1.red.old[,3])), (log2(euRatio.c1.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.1.red[,3])), (log2(euRatio.c1.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.1.red.old[,3])), (log2(euRatio.c1.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.1.red[,3])), (log2(euRatio.c1.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.1.red[,3])), (log2(euRatio.c1.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.1.red[,3])), (log2(euRatio.c1.red.old)), paired=FALSE, var.equal=TRUE)

dev.off()}

##############################################################################################

##############################################################################################
#CHROMOSOME 2
par(mar=c(5, 4, 4, 2) + 0.1)
par(mfrow=c(1,1))
#going to do this for all lines for chromosome 1 
#try to plot them all on the same graph? 

#euRatio.test <- c(logMat2.GC.test[,7],logMat2.GC.test[,7])
{pdf("Chrm_2_old.pdf")
boxplot((log2(l001.2.old[,3])),(log2(l002.2.old[,3])),(log2(l003.2.old[,3])),(log2(l004.2.old[,3])),(log2(l005.2[,3])),
        (log2(l006.2.old[,3])),(log2(l007.2.old[,3])),(log2(l008.2.old[,3])),(log2(l009.2.old[,3])),(log2(l011.2.old[,3])),
        (log2(l015.2[,3])),(log2(l028.2.old[,3])),(log2(l088.2[,3])),(log2(l108.2[,3])),(log2(l119.2[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 2",las=3
        ,ylim=c(-2,2.5))
abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

euRatio.c2.old <- c(l001.2.old[,3],l002.2.old[,3],l003.2.old[,3],l004.2.old[,3],l005.2.old[,3],l006.2.old[,3],l007.2.old[,3],l008.2.old[,3],l009.2.old[,3],l011.2.old[,3],l028.2.old[,3])
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.2.old[,3])), (log2(euRatio.c2.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.2.old[,3])), (log2(euRatio.c2.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.2.old[,3])), (log2(euRatio.c2.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.2.old[,3])), (log2(euRatio.c2.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.2.old[,3])), (log2(euRatio.c2.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.2.old[,3])), (log2(euRatio.c2.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.2.old[,3])), (log2(euRatio.c2.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.2.old[,3])), (log2(euRatio.c2.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.2.old[,3])), (log2(euRatio.c2.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.2.old[,3])), (log2(euRatio.c2.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.2[,3])), (log2(euRatio.c2.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.2.old[,3])), (log2(euRatio.c2.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.2[,3])), (log2(euRatio.c2.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.2[,3])), (log2(euRatio.c2.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.2[,3])), (log2(euRatio.c2.old)), paired=FALSE, var.equal=TRUE)
dev.off()}
##############################################################################################
#CHROMOSOME 2

#RATIOS FOR CHROMOSOME 2
{
euRatio.c2.red.old <- c(l001.2.red.old[,3],l002.2.red.old[,3],l003.2.red.old[,3],l004.2.red.old[,3],l005.2.red.old[,3],l006.2.red.old[,3],l007.2.red.old[,3],l008.2.red.old[,3],l009.2.red.old[,3],l011.2.red.old[,3],l028.2.red.old[,3])

pdf("Chrm_2_red_old.pdf")
boxplot((log2(l001.2.red.old[,3])),(log2(l002.2.red.old[,3])),(log2(l003.2.red.old[,3])),(log2(l004.2.red.old[,3])),(log2(l005.2.red.old[,3])),
        (log2(l006.2.red.old[,3])),(log2(l007.2.red.old[,3])),(log2(l008.2.red.old[,3])),(log2(l009.2.red.old[,3])),(log2(l011.2.red.old[,3])),
        (log2(l015.2.red[,3])),(log2(l028.2.red.old[,3])),(log2(l088.2.red[,3])),(log2(l108.2.red[,3])),(log2(l119.2.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 2 Reduced",las=3
        ,ylim=c(-2,2.5))

#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.2.red.old[,3])), (log2(euRatio.c2.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.2.red.old[,3])), (log2(euRatio.c2.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.2.red.old[,3])), (log2(euRatio.c2.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.2.red.old[,3])), (log2(euRatio.c2.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.2.red.old[,3])), (log2(euRatio.c2.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.2.red.old[,3])), (log2(euRatio.c2.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.2.red.old[,3])), (log2(euRatio.c2.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.2.red.old[,3])), (log2(euRatio.c2.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.2.red.old[,3])), (log2(euRatio.c2.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.2.red.old[,3])), (log2(euRatio.c2.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.2.red[,3])), (log2(euRatio.c2.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.2.red.old[,3])), (log2(euRatio.c2.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.2.red[,3])), (log2(euRatio.c2.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.2.red[,3])), (log2(euRatio.c2.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.2.red[,3])), (log2(euRatio.c2.red.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
#p3 <- 0.0006221
#mylabel = bquote(italic(p) == .(format(p3, digits = 9)))
#text(x=2.3,y = 2.5, labels = mylabel,cex=.8)
#text(x=4, y=2.5, "*", pos=3, cex=1.5, col="red")
#text(x=14, y=2.5, "*", pos=3, cex=1.5, col="red")
dev.off()}
##############################################################################################
#CHROMOSOME 3

#RATIOS FOR CHROMOSOME 3
{
euRatio.c3.old <- c(l001.3.old[,3],l002.3.old[,3],l003.3.old[,3],l004.3.old[,3],l005.3.old[,3],l006.3.old[,3],l007.3.old[,3],l008.3.old[,3],l009.3.old[,3],l011.3.old[,3],l028.3.old[,3])
pdf("Chrm_3_old.pdf")
boxplot((log2(l001.3.old[,3])),(log2(l002.3.old[,3])),(log2(l003.3.old[,3])),(log2(l004.3.old[,3])),(log2(l005.3.old[,3])),
        (log2(l006.3.old[,3])),(log2(l007.3.old[,3])),(log2(l008.3.old[,3])),(log2(l009.3.old[,3])),(log2(l011.3.old[,3])),
        (log2(l015.3[,3])),(log2(l028.3.old[,3])),(log2(l088.3[,3])),(log2(l108.3[,3])),(log2(l119.3[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 3",las=3
        ,ylim=c(-2,2.5))

#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.3.old[,3])), (log2(euRatio.c3.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.3.old[,3])), (log2(euRatio.c3.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.3.old[,3])), (log2(euRatio.c3.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.3.old[,3])), (log2(euRatio.c3.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.3.old[,3])), (log2(euRatio.c3.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.3.old[,3])), (log2(euRatio.c3.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.3.old[,3])), (log2(euRatio.c3.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.3.old[,3])), (log2(euRatio.c3.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.3.old[,3])), (log2(euRatio.c3.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.3.old[,3])), (log2(euRatio.c3.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.3[,3])), (log2(euRatio.c3.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.3.old[,3])), (log2(euRatio.c3.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.3[,3])), (log2(euRatio.c3.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.3[,3])), (log2(euRatio.c3.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.3[,3])), (log2(euRatio.c3.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
#p3 <- 0.0006221
#mylabel = bquote(italic(p) == .(format(p3, digits = 9)))
#text(x=2.3,y = 2.5, labels = mylabel,cex=.8)
#text(x=4, y=2.5, "*", pos=3, cex=1.5, col="red")
#text(x=14, y=2.5, "*", pos=3, cex=1.5, col="red")
dev.off()}
###############################################################################################
###############################################################################################
#CHROMOSOME 3

#RATIOS FOR CHROMOSOME 3
{
euRatio.c3.red.old <- c(l001.3.red.old[,3],l002.3.red.old[,3],l003.3.red.old[,3],l004.3.red.old[,3],l005.3.red.old[,3],l006.3.red.old[,3],l007.3.red.old[,3],l008.3.red.old[,3],l009.3.red.old[,3],l011.3.red.old[,3],l028.3.red.old[,3])
pdf("Chrm_3_red_old.pdf")
boxplot((log2(l001.3.red.old[,3])),(log2(l002.3.red.old[,3])),(log2(l003.3.red.old[,3])),(log2(l004.3.red.old[,3])),(log2(l005.3.red.old[,3])),
        (log2(l006.3.red.old[,3])),(log2(l007.3.red.old[,3])),(log2(l008.3.red.old[,3])),(log2(l009.3.red.old[,3])),(log2(l011.3.red.old[,3])),
        (log2(l015.3.red[,3])),(log2(l028.3.red.old[,3])),(log2(l088.3.red[,3])),(log2(l108.3.red[,3])),(log2(l119.3.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 3 Reduced",las=3
        ,ylim=c(-2,2.5))

#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.3.red.old[,3])), (log2(euRatio.c3.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.3.red.old[,3])), (log2(euRatio.c3.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.3.red.old[,3])), (log2(euRatio.c3.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.3.red.old[,3])), (log2(euRatio.c3.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.3.red.old[,3])), (log2(euRatio.c3.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.3.red.old[,3])), (log2(euRatio.c3.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.3.red.old[,3])), (log2(euRatio.c3.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.3.red.old[,3])), (log2(euRatio.c3.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.3.red.old[,3])), (log2(euRatio.c3.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.3.red.old[,3])), (log2(euRatio.c3.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.3.red[,3])), (log2(euRatio.c3.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.3.red.old[,3])), (log2(euRatio.c3.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.3.red[,3])), (log2(euRatio.c3.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.3.red[,3])), (log2(euRatio.c3.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.3.red[,3])), (log2(euRatio.c3.red.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
#p3 <- 0.0006221
#mylabel = bquote(italic(p) == .(format(p3, digits = 9)))
#text(x=2.3,y = 2.5, labels = mylabel,cex=.8)
#text(x=4, y=2.5, "*", pos=3, cex=1.5, col="red")
#text(x=14, y=2.5, "*", pos=3, cex=1.5, col="red")
dev.off()}
###############################################################################################
###############################################################################################
#CHROMOSOME 4

#RATIOS FOR CHROMOSOME 4
{
euRatio.c4.old <- c(l001.4.old[,3],l002.4.old[,3],l003.4.old[,3],l004.4.old[,3],l005.4.old[,3],l006.4.old[,3],l007.4.old[,3],l008.4.old[,3],l009.4.old[,3],l011.4.old[,3],l028.4.old[,3])
pdf("Chrm_4_old.pdf")
boxplot((log2(l001.4.old[,3])),(log2(l002.4.old[,3])),(log2(l003.4.old[,3])),(log2(l004.4.old[,3])),(log2(l005.4.old[,3])),
        (log2(l006.4.old[,3])),(log2(l007.4.old[,3])),(log2(l008.4.old[,3])),(log2(l009.4.old[,3])),(log2(l011.4.old[,3])),
        (log2(l015.4[,3])),(log2(l028.4.old[,3])),(log2(l088.4[,3])),(log2(l108.4[,3])),(log2(l119.4[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 4",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.4.old[,3])), (log2(euRatio.c4.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.4.old[,3])), (log2(euRatio.c4.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.4.old[,3])), (log2(euRatio.c4.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.4.old[,3])), (log2(euRatio.c4.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.4.old[,3])), (log2(euRatio.c4.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.4.old[,3])), (log2(euRatio.c4.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.4.old[,3])), (log2(euRatio.c4.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.4.old[,3])), (log2(euRatio.c4.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.4.old[,3])), (log2(euRatio.c4.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.4.old[,3])), (log2(euRatio.c4.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.4[,3])), (log2(euRatio.c4.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.4.old[,3])), (log2(euRatio.c4.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.4[,3])), (log2(euRatio.c4.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.4[,3])), (log2(euRatio.c4.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.4[,3])), (log2(euRatio.c4.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
#p3 <- 0.0006221
#mylabel = bquote(italic(p) == .(format(p3, digits = 9)))
#text(x=2.3,y = 2.5, labels = mylabel,cex=.8)
#text(x=4, y=2.5, "*", pos=3, cex=1.5, col="red")
#text(x=14, y=2.5, "*", pos=3, cex=1.5, col="red")
dev.off()}
###############################################################################################

#CHROMOSOME 4 reduced

#RATIOS FOR CHROMOSOME 4
{
euRatio.c4.red.old <- c(l001.4.red.old[,3],l002.4.red.old[,3],l003.4.red.old[,3],l004.4.red.old[,3],l005.4.red.old[,3],l006.4.red.old[,3],l007.4.red.old[,3],l008.4.red.old[,3],l009.4.red.old[,3],l011.4.red.old[,3],l028.4.red.old[,3])
pdf("Chrm_4_red_old.pdf")
boxplot((log2(l001.4.red.old[,3])),(log2(l002.4.red.old[,3])),(log2(l003.4.red.old[,3])),(log2(l004.4.red.old[,3])),(log2(l005.4.red.old[,3])),
        (log2(l006.4.red.old[,3])),(log2(l007.4.red.old[,3])),(log2(l008.4.red.old[,3])),(log2(l009.4.red.old[,3])),(log2(l011.4.red.old[,3])),
        (log2(l015.4.red[,3])),(log2(l028.4.red.old[,3])),(log2(l088.4.red[,3])),(log2(l108.4.red[,3])),(log2(l119.4.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 4 Reduced",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.4.red.old[,3])), (log2(euRatio.c4.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.4.red.old[,3])), (log2(euRatio.c4.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.4.red.old[,3])), (log2(euRatio.c4.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.4.red.old[,3])), (log2(euRatio.c4.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.4.red.old[,3])), (log2(euRatio.c4.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.4.red.old[,3])), (log2(euRatio.c4.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.4.red.old[,3])), (log2(euRatio.c4.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.4.red.old[,3])), (log2(euRatio.c4.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.4.red.old[,3])), (log2(euRatio.c4.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.4.red.old[,3])), (log2(euRatio.c4.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.4.red[,3])), (log2(euRatio.c4.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.4.red.old[,3])), (log2(euRatio.c4.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.4.red[,3])), (log2(euRatio.c4.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.4.red[,3])), (log2(euRatio.c4.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.4.red[,3])), (log2(euRatio.c4.red.old)), paired=FALSE, var.equal=TRUE)


abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
#p3 <- 0.0006221
#mylabel = bquote(italic(p) == .(format(p3, digits = 9)))
#text(x=2.3,y = 2.5, labels = mylabel,cex=.8)
#text(x=4, y=2.5, "*", pos=3, cex=1.5, col="red")
#text(x=14, y=2.5, "*", pos=3, cex=1.5, col="red")
dev.off()}
###############################################################################################
#CHROMOSOME 5 

#RATIOS FOR CHROMOSOME 5
{
euRatio.c5.old <- c(l001.5.old[,3],l002.5.old[,3],l003.5.old[,3],l004.5.old[,3],l005.5.old[,3],l006.5.old[,3],l007.5.old[,3],l008.5.old[,3],l009.5.old[,3],l011.5.old[,3],l028.5.old[,3])
pdf("Chrm_5_old.pdf")
boxplot((log2(l001.5.old[,3])),(log2(l002.5.old[,3])),(log2(l003.5.old[,3])),(log2(l004.5.old[,3])),(log2(l005.5.old[,3])),
        (log2(l006.5.old[,3])),(log2(l007.5.old[,3])),(log2(l008.5.old[,3])),(log2(l009.5.old[,3])),(log2(l011.5.old[,3])),
        (log2(l015.5[,3])),(log2(l028.5.old[,3])),(log2(l088.5[,3])),(log2(l108.5[,3])),(log2(l119.5[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 5",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.5.old[,3])), (log2(euRatio.c5.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.5.old[,3])), (log2(euRatio.c5.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.5.old[,3])), (log2(euRatio.c5.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.5.old[,3])), (log2(euRatio.c5.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.5.old[,3])), (log2(euRatio.c5.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.5.old[,3])), (log2(euRatio.c5.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.5.old[,3])), (log2(euRatio.c5.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.5.old[,3])), (log2(euRatio.c5.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.5.old[,3])), (log2(euRatio.c5.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.5.old[,3])), (log2(euRatio.c5.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.5[,3])), (log2(euRatio.c5.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.5.old[,3])), (log2(euRatio.c5.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.5[,3])), (log2(euRatio.c5.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.5[,3])), (log2(euRatio.c5.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.5[,3])), (log2(euRatio.c5.old)), paired=FALSE, var.equal=TRUE)


abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


dev.off()
}
########################################################################################
#CHROMOSOME 5 reduced

#RATIOS FOR CHROMOSOME 5
{
euRatio.c5.red.old <- c(l001.5.red.old[,3],l002.5.red.old[,3],l003.5.red.old[,3],l004.5.red.old[,3],l005.5.red.old[,3],l006.5.red.old[,3],l007.5.red.old[,3],l008.5.red.old[,3],l009.5.red.old[,3],l011.5.red.old[,3],l028.5.red.old[,3])
pdf("Chrm_5_red_old.pdf")
boxplot((log2(l001.5.red.old[,3])),(log2(l002.5.red.old[,3])),(log2(l003.5.red.old[,3])),(log2(l004.5.red.old[,3])),(log2(l005.5.red.old[,3])),
        (log2(l006.5.red.old[,3])),(log2(l007.5.red.old[,3])),(log2(l008.5.red.old[,3])),(log2(l009.5.red.old[,3])),(log2(l011.5.red.old[,3])),
        (log2(l015.5.red[,3])),(log2(l028.5.red.old[,3])),(log2(l088.5.red[,3])),(log2(l108.5.red[,3])),(log2(l119.5.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 5 Reduced",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.5.red.old[,3])), (log2(euRatio.c5.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.5.red.old[,3])), (log2(euRatio.c5.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.5.red.old[,3])), (log2(euRatio.c5.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.5.red.old[,3])), (log2(euRatio.c5.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.5.red.old[,3])), (log2(euRatio.c5.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.5.red.old[,3])), (log2(euRatio.c5.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.5.red.old[,3])), (log2(euRatio.c5.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.5.red.old[,3])), (log2(euRatio.c5.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.5.red.old[,3])), (log2(euRatio.c5.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.5.red.old[,3])), (log2(euRatio.c5.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.5.red[,3])), (log2(euRatio.c5.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.5.red.old[,3])), (log2(euRatio.c5.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.5.red[,3])), (log2(euRatio.c5.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.5.red[,3])), (log2(euRatio.c5.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.5.red[,3])), (log2(euRatio.c5.red.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
#p3 <- 0.0006221
#mylabel = bquote(italic(p) == .(format(p3, digits = 9)))
#text(x=2.3,y = 2.5, labels = mylabel,cex=.8)

dev.off()
}
########################################################################################
#CHROMOSOME 6 

#RATIOS FOR CHROMOSOME 6
{
euRatio.c6.old <- c(l001.6.old[,3],l002.6.old[,3],l003.6.old[,3],l004.6.old[,3],l005.6.old[,3],l006.6.old[,3],l007.6.old[,3],l008.6.old[,3],l009.6.old[,3],l011.6.old[,3],l028.6.old[,3])
pdf("Chrm_6_old.pdf")
boxplot((log2(l001.6.old[,3])),(log2(l002.6.old[,3])),(log2(l003.6.old[,3])),(log2(l004.6.old[,3])),(log2(l005.6.old[,3])),
        (log2(l006.6.old[,3])),(log2(l007.6.old[,3])),(log2(l008.6.old[,3])),(log2(l009.6.old[,3])),(log2(l011.6.old[,3])),
        (log2(l015.6[,3])),(log2(l028.6.old[,3])),(log2(l088.6[,3])),(log2(l108.6[,3])),(log2(l119.6[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 6",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.6.old[,3])), (log2(euRatio.c6.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.6.old[,3])), (log2(euRatio.c6.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.6.old[,3])), (log2(euRatio.c6.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.6.old[,3])), (log2(euRatio.c6.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.6.old[,3])), (log2(euRatio.c6.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.6.old[,3])), (log2(euRatio.c6.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.6.old[,3])), (log2(euRatio.c6.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.6.old[,3])), (log2(euRatio.c6.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.6.old[,3])), (log2(euRatio.c6.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.6.old[,3])), (log2(euRatio.c6.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.6[,3])), (log2(euRatio.c6.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.6.old[,3])), (log2(euRatio.c6.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.6[,3])), (log2(euRatio.c6.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.6[,3])), (log2(euRatio.c6.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.6[,3])), (log2(euRatio.c6.old)), paired=FALSE, var.equal=TRUE)


abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
#p3 <- 0.0006221
#mylabel = bquote(italic(p) == .(format(p3, digits = 9)))
#text(x=2.3,y = 2.5, labels = mylabel,cex=.8)
#text(x=4, y=2.5, "*", pos=3, cex=1.5, col="red")
#text(x=14, y=2.5, "*", pos=3, cex=1.5, col="red")
dev.off()}
################################################################################

#CHROMOSOME 6 reduced

#RATIOS FOR CHROMOSOME 6
{
euRatio.c6.red.old <- c(l001.6.red.old[,3],l002.6.red.old[,3],l003.6.red.old[,3],l004.6.red.old[,3],l005.6.red.old[,3],l006.6.red.old[,3],l007.6.red.old[,3],l008.6.red.old[,3],l009.6.red.old[,3],l011.6.red.old[,3],l028.6.red.old[,3])
pdf("Chrm_6_red_old.pdf")
boxplot((log2(l001.6.red.old[,3])),(log2(l002.6.red.old[,3])),(log2(l003.6.red.old[,3])),(log2(l004.6.red.old[,3])),(log2(l005.6.red.old[,3])),
        (log2(l006.6.red.old[,3])),(log2(l007.6.red.old[,3])),(log2(l008.6.red.old[,3])),(log2(l009.6.red.old[,3])),(log2(l011.6.red.old[,3])),
        (log2(l015.6.red[,3])),(log2(l028.6.red.old[,3])),(log2(l088.6.red[,3])),(log2(l108.6.red[,3])),(log2(l119.6.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 6 Reduced",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.6.red.old[,3])), (log2(euRatio.c6.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.6.red.old[,3])), (log2(euRatio.c6.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.6.red.old[,3])), (log2(euRatio.c6.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.6.red.old[,3])), (log2(euRatio.c6.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.6.red.old[,3])), (log2(euRatio.c6.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.6.red.old[,3])), (log2(euRatio.c6.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.6.red.old[,3])), (log2(euRatio.c6.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.6.red.old[,3])), (log2(euRatio.c6.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.6.red.old[,3])), (log2(euRatio.c6.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.6.red.old[,3])), (log2(euRatio.c6.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.6.red[,3])), (log2(euRatio.c6.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.6.red.old[,3])), (log2(euRatio.c6.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.6.red[,3])), (log2(euRatio.c6.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.6.red[,3])), (log2(euRatio.c6.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.6.red[,3])), (log2(euRatio.c6.red.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
#p3 <- 0.0006221
#mylabel = bquote(italic(p) == .(format(p3, digits = 9)))
#text(x=2.3,y = 2.5, labels = mylabel,cex=.8)
#text(x=4, y=2.5, "*", pos=3, cex=1.5, col="red")
#text(x=14, y=2.5, "*", pos=3, cex=1.5, col="red")
dev.off()}
################################################################################

#CHROMOSOME 7 

{#RATIOS FOR CHROMOSOME 7
euRatio.c7.old <- c(l001.7.old[,3],l002.7.old[,3],l003.7.old[,3],l004.7.old[,3],l005.7.old[,3],l006.7.old[,3],l007.7.old[,3],l008.7.old[,3],l009.7.old[,3],l011.7.old[,3],l028.7.old[,3])
pdf("Chrm_7_old.pdf")
boxplot((log2(l001.7.old[,3])),(log2(l002.7.old[,3])),(log2(l003.7.old[,3])),(log2(l004.7.old[,3])),(log2(l005.7.old[,3])),
        (log2(l006.7.old[,3])),(log2(l007.7.old[,3])),(log2(l008.7.old[,3])),(log2(l009.7.old[,3])),(log2(l011.7.old[,3])),
        (log2(l015.7[,3])),(log2(l028.7.old[,3])),(log2(l088.7[,3])),(log2(l108.7[,3])),(log2(l119.7[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 7",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.7.old[,3])), (log2(euRatio.c7.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.7.old[,3])), (log2(euRatio.c7.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.7.old[,3])), (log2(euRatio.c7.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.7.old[,3])), (log2(euRatio.c7.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.7.old[,3])), (log2(euRatio.c7.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.7.old[,3])), (log2(euRatio.c7.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.7.old[,3])), (log2(euRatio.c7.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.7.old[,3])), (log2(euRatio.c7.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.7.old[,3])), (log2(euRatio.c7.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.7.old[,3])), (log2(euRatio.c7.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.7[,3])), (log2(euRatio.c7.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.7.old[,3])), (log2(euRatio.c7.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.7[,3])), (log2(euRatio.c7.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.7[,3])), (log2(euRatio.c7.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.7[,3])), (log2(euRatio.c7.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

dev.off()}


################################################################################

#CHROMOSOME 7 

{#RATIOS FOR CHROMOSOME 7
euRatio.c7.red.old <- c(l001.7.red.old[,3],l002.7.red.old[,3],l003.7.red.old[,3],l004.7.red.old[,3],l005.7.red.old[,3],l006.7.red.old[,3],l007.7.red.old[,3],l008.7.red.old[,3],l009.7.red.old[,3],l011.7.red.old[,3],l028.7.red.old[,3])
pdf("Chrm_7_red_old.pdf")
boxplot((log2(l001.7.red.old[,3])),(log2(l002.7.red.old[,3])),(log2(l003.7.red.old[,3])),(log2(l004.7.red.old[,3])),(log2(l005.7.red.old[,3])),
        (log2(l006.7.red.old[,3])),(log2(l007.7.red.old[,3])),(log2(l008.7.red.old[,3])),(log2(l009.7.red.old[,3])),(log2(l011.7.red.old[,3])),
        (log2(l015.7.red[,3])),(log2(l028.7.red.old[,3])),(log2(l088.7.red[,3])),(log2(l108.7.red[,3])),(log2(l119.7.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 7 Reduced",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.7.red.old[,3])), (log2(euRatio.c7.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.7.red.old[,3])), (log2(euRatio.c7.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.7.red.old[,3])), (log2(euRatio.c7.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.7.red.old[,3])), (log2(euRatio.c7.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.7.red.old[,3])), (log2(euRatio.c7.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.7.red.old[,3])), (log2(euRatio.c7.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.7.red.old[,3])), (log2(euRatio.c7.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.7.red.old[,3])), (log2(euRatio.c7.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.7.red.old[,3])), (log2(euRatio.c7.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.7.red.old[,3])), (log2(euRatio.c7.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.7.red[,3])), (log2(euRatio.c7.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.7.red.old[,3])), (log2(euRatio.c7.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.7.red[,3])), (log2(euRatio.c7.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.7.red[,3])), (log2(euRatio.c7.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.7.red[,3])), (log2(euRatio.c7.red.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

dev.off()}
###############################################################################################

#CHROMOSOME 8 

#RATIOS FOR CHROMOSOME 8

{
  euRatio.c8.old <- c(l001.8.old[,3],l002.8.old[,3],l003.8.old[,3],l004.8.old[,3],l005.8.old[,3],l006.8.old[,3],l007.8.old[,3],l008.8.old[,3],l009.8.old[,3],l011.8.old[,3],l028.8.old[,3])
pdf("Chrm_8_old.pdf")
boxplot((log2(l001.8.old[,3])),(log2(l002.8.old[,3])),(log2(l003.8.old[,3])),(log2(l004.8.old[,3])),(log2(l005.8.old[,3])),
        (log2(l006.8.old[,3])),(log2(l007.8.old[,3])),(log2(l008.8.old[,3])),(log2(l009.8.old[,3])),(log2(l011.8.old[,3])),
        (log2(l015.8[,3])),(log2(l028.8.old[,3])),(log2(l088.8[,3])),(log2(l108.8[,3])),(log2(l119.8[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 8",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.8.old[,3])), (log2(euRatio.c8.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.8.old[,3])), (log2(euRatio.c8.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.8.old[,3])), (log2(euRatio.c8.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.8.old[,3])), (log2(euRatio.c8.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.8.old[,3])), (log2(euRatio.c8.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.8.old[,3])), (log2(euRatio.c8.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.8.old[,3])), (log2(euRatio.c8.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.8.old[,3])), (log2(euRatio.c8.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.8.old[,3])), (log2(euRatio.c8.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.8.old[,3])), (log2(euRatio.c8.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.8[,3])), (log2(euRatio.c8.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.8.old[,3])), (log2(euRatio.c8.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.8[,3])), (log2(euRatio.c8.old)), paired=FALSE, var.equal=TRUE)
wilcox.test((log2(l108.8[,3])), (log2(euRatio.c8.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.8[,3])), (log2(euRatio.c8.old)), paired=FALSE, var.equal=TRUE)

text(x=13.7, y=2, "*", pos=3, cex=2, col="red")
text(x=14, y=2, "*", pos=3, cex=2, col="red")
text(x=14.3, y=2, "*", pos=3, cex=2, col="red")

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
dev.off()}
###############################################################################################

#CHROMOSOME 8 reduced

#RATIOS FOR CHROMOSOME 8

{euRatio.c8.red.old <- c(l001.8.red.old[,3],l002.8.red.old[,3],l003.8.red.old[,3],l004.8.red.old[,3],l005.8.red.old[,3],l006.8.red.old[,3],l007.8.red.old[,3],l008.8.red.old[,3],l009.8.red.old[,3],l011.8.red.old[,3],l028.8.red.old[,3])
pdf("Chrm_8_red_old.pdf")
boxplot((log2(l001.8.red.old[,3])),(log2(l002.8.red.old[,3])),(log2(l003.8.red.old[,3])),(log2(l004.8.red.old[,3])),(log2(l005.8.red.old[,3])),
        (log2(l006.8.red.old[,3])),(log2(l007.8.red.old[,3])),(log2(l008.8.red.old[,3])),(log2(l009.8.red.old[,3])),(log2(l011.8.red.old[,3])),
        (log2(l015.8.red[,3])),(log2(l028.8.red.old[,3])),(log2(l088.8.red[,3])),(log2(l108.8.red[,3])),(log2(l119.8.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 7 Reduced",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.8.red.old[,3])), (log2(euRatio.c8.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.8.red.old[,3])), (log2(euRatio.c8.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.8.red.old[,3])), (log2(euRatio.c8.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.8.red.old[,3])), (log2(euRatio.c8.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.8.red.old[,3])), (log2(euRatio.c8.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.8.red.old[,3])), (log2(euRatio.c8.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.8.red.old[,3])), (log2(euRatio.c8.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.8.red.old[,3])), (log2(euRatio.c8.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.8.red.old[,3])), (log2(euRatio.c8.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.8.red.old[,3])), (log2(euRatio.c8.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.8.red[,3])), (log2(euRatio.c8.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.8.red.old[,3])), (log2(euRatio.c8.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.8.red[,3])), (log2(euRatio.c8.red.old)), paired=FALSE, var.equal=TRUE)
wilcox.test((log2(l108.8.red[,3])), (log2(euRatio.c8.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.8.red[,3])), (log2(euRatio.c8.red.old)), paired=FALSE, var.equal=TRUE)

text(x=13.7, y=2, "*", pos=3, cex=2, col="red")
text(x=14, y=2, "*", pos=3, cex=2, col="red")
text(x=14.3, y=2, "*", pos=3, cex=2, col="red")

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")
dev.off()}



###############################################################################################
#CHROMOSOME 9

#RATIOS FOR CHROMOSOME 9

{ 
  euRatio.c9.old <- c(l001.9.old[,3],l002.9.old[,3],l003.9.old[,3],l004.9.old[,3],l005.9.old[,3],l006.9.old[,3],l007.9.old[,3],l008.9.old[,3],l009.9.old[,3],l011.9.old[,3],l028.9.old[,3])
pdf("Chrm_9_old.pdf")
boxplot((log2(l001.9.old[,3])),(log2(l002.9.old[,3])),(log2(l003.9.old[,3])),(log2(l004.9.old[,3])),(log2(l005.9.old[,3])),
        (log2(l006.9.old[,3])),(log2(l007.9.old[,3])),(log2(l008.9.old[,3])),(log2(l009.9.old[,3])),(log2(l011.9.old[,3])),
        (log2(l015.9[,3])),(log2(l028.9.old[,3])),(log2(l088.9[,3])),(log2(l108.9[,3])),(log2(l119.9[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 8",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.9.old[,3])), (log2(euRatio.c9.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.9.old[,3])), (log2(euRatio.c9.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.9.old[,3])), (log2(euRatio.c9.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.9.old[,3])), (log2(euRatio.c9.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.9.old[,3])), (log2(euRatio.c9.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.9.old[,3])), (log2(euRatio.c9.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.9.old[,3])), (log2(euRatio.c9.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.9.old[,3])), (log2(euRatio.c9.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.9.old[,3])), (log2(euRatio.c9.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.9.old[,3])), (log2(euRatio.c9.old)), paired=FALSE, var.equal=TRUE)
wilcox.test((log2(l015.9[,3])), (log2(euRatio.c9.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.9.old[,3])), (log2(euRatio.c9.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.9[,3])), (log2(euRatio.c9.old)), paired=FALSE, var.equal=TRUE)
wilcox.test((log2(l108.9[,3])), (log2(euRatio.c9.old)), paired=FALSE, var.equal=TRUE)
wilcox.test((log2(l119.9[,3])), (log2(euRatio.c9.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

text(x=10.7, y=2, "*", pos=3, cex=2, col="red")
text(x=11, y=2, "*", pos=3, cex=2, col="red")
text(x=12.3, y=2, "*", pos=3, cex=2, col="red")

text(x=12.7, y=2, "*", pos=3, cex=2, col="red")
text(x=13, y=2, "*", pos=3, cex=2, col="red")
text(x=13.3, y=2, "*", pos=3, cex=2, col="red")

text(x=14.7, y=2, "*", pos=3, cex=2, col="red")
text(x=15, y=2, "*", pos=3, cex=2, col="red")
text(x=15.3, y=2, "*", pos=3, cex=2, col="red")

dev.off()}



#CHROMOSOME 9 reduced

#RATIOS FOR CHROMOSOME 9

{
  euRatio.c9.red.old <- c(l001.9.red.old[,3],l002.9.red.old[,3],l003.9.red.old[,3],l004.9.red.old[,3],l005.9.red.old[,3],l006.9.red.old[,3],l007.9.red.old[,3],l008.9.red.old[,3],l009.9.red.old[,3],l011.9.red.old[,3],l028.9.red.old[,3])
pdf("Chrm_9_red_old.pdf")
boxplot((log2(l001.9.red.old[,3])),(log2(l002.9.red.old[,3])),(log2(l003.9.red.old[,3])),(log2(l004.9.red.old[,3])),(log2(l005.9.red.old[,3])),
        (log2(l006.9.red.old[,3])),(log2(l007.9.red.old[,3])),(log2(l008.9.red.old[,3])),(log2(l009.9.red.old[,3])),(log2(l011.9.red.old[,3])),
        (log2(l015.9.red[,3])),(log2(l028.9.red.old[,3])),(log2(l088.9.red[,3])),(log2(l108.9.red[,3])),(log2(l119.9.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 7 Reduced",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.9.red.old[,3])), (log2(euRatio.c9.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.9.red.old[,3])), (log2(euRatio.c9.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.9.red.old[,3])), (log2(euRatio.c9.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.9.red.old[,3])), (log2(euRatio.c9.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.9.red.old[,3])), (log2(euRatio.c9.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.9.red.old[,3])), (log2(euRatio.c9.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.9.red.old[,3])), (log2(euRatio.c9.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.9.red.old[,3])), (log2(euRatio.c9.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.9.red.old[,3])), (log2(euRatio.c9.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.9.red.old[,3])), (log2(euRatio.c9.red.old)), paired=FALSE, var.equal=TRUE)
wilcox.test((log2(l015.9.red[,3])), (log2(euRatio.c9.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.9.red.old[,3])), (log2(euRatio.c9.red.old)), paired=FALSE, var.equal=TRUE)
wilcox.test((log2(l088.9.red[,3])), (log2(euRatio.c9.red.old)), paired=FALSE, var.equal=TRUE)
wilcox.test((log2(l108.9.red[,3])), (log2(euRatio.c9.red.old)), paired=FALSE, var.equal=TRUE)
wilcox.test((log2(l119.9.red[,3])), (log2(euRatio.c9.red.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

text(x=10.7, y=1.8, "*", pos=3, cex=2, col="red")
text(x=11, y=1.8, "*", pos=3, cex=2, col="red")
text(x=11.3, y=1.8, "*", pos=3, cex=2, col="red")

text(x=12.7, y=1.8, "*", pos=3, cex=2, col="red")
text(x=13, y=1.8, "*", pos=3, cex=2, col="red")
text(x=13.3, y=1.8, "*", pos=3, cex=2, col="red")

text(x=14.7, y=1.8, "*", pos=3, cex=2, col="red")
text(x=15, y=1.8, "*", pos=3, cex=2, col="red")
text(x=15.3, y=1.8, "*", pos=3, cex=2, col="red")

dev.off()}

###############################################################################################
#CHROMOSOME 10 

#RATIOS FOR CHROMOSOME 10

{ 
  euRatio.c10.old <- c(l001.10.old[,3],l002.10.old[,3],l003.10.old[,3],l004.10.old[,3],l005.10.old[,3],l006.10.old[,3],l007.10.old[,3],l008.10.old[,3],l009.10.old[,3],l011.10.old[,3],l028.10.old[,3])
pdf("Chrm_10_old.pdf")
boxplot((log2(l001.10.old[,3])),(log2(l002.10.old[,3])),(log2(l003.10.old[,3])),(log2(l004.10.old[,3])),(log2(l005.10.old[,3])),
        (log2(l006.10.old[,3])),(log2(l007.10.old[,3])),(log2(l008.10.old[,3])),(log2(l009.10.old[,3])),(log2(l011.10.old[,3])),
        (log2(l015.10[,3])),(log2(l028.10.old[,3])),(log2(l088.10[,3])),(log2(l108.10[,3])),(log2(l119.10[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 8",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.10.old[,3])), (log2(euRatio.c10.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.10.old[,3])), (log2(euRatio.c10.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.10.old[,3])), (log2(euRatio.c10.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.10.old[,3])), (log2(euRatio.c10.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.10.old[,3])), (log2(euRatio.c10.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.10.old[,3])), (log2(euRatio.c10.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.10.old[,3])), (log2(euRatio.c10.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.10.old[,3])), (log2(euRatio.c10.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.10.old[,3])), (log2(euRatio.c10.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.10.old[,3])), (log2(euRatio.c10.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.10[,3])), (log2(euRatio.c10.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.10.old[,3])), (log2(euRatio.c10.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.10[,3])), (log2(euRatio.c10.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.10[,3])), (log2(euRatio.c10.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.10[,3])), (log2(euRatio.c10.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


dev.off()}
###############################################################################################

#CHROMOSOME 10 

#RATIOS FOR CHROMOSOME 10

{
  euRatio.c10.red.old <- c(l001.10.red.old[,3],l002.10.red.old[,3],l003.10.red.old[,3],l004.10.red.old[,3],l005.10.red.old[,3],l006.10.red.old[,3],l007.10.red.old[,3],l008.10.red.old[,3],l009.10.red.old[,3],l011.10.red.old[,3],l028.10.red.old[,3])
pdf("Chrm_10_red_old.pdf")
boxplot((log2(l001.10.red.old[,3])),(log2(l002.10.red.old[,3])),(log2(l003.10.red.old[,3])),(log2(l004.10.red.old[,3])),(log2(l005.10.red.old[,3])),
        (log2(l006.10.red.old[,3])),(log2(l007.10.red.old[,3])),(log2(l008.10.red.old[,3])),(log2(l009.10.red.old[,3])),(log2(l011.10.red.old[,3])),
        (log2(l015.10.red[,3])),(log2(l028.10.red.old[,3])),(log2(l088.10.red[,3])),(log2(l108.10.red[,3])),(log2(l119.10.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 7 Reduced",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.10.red.old[,3])), (log2(euRatio.c10.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.10.red.old[,3])), (log2(euRatio.c10.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.10.red.old[,3])), (log2(euRatio.c10.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.10.red.old[,3])), (log2(euRatio.c10.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.10.red.old[,3])), (log2(euRatio.c10.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.10.red.old[,3])), (log2(euRatio.c10.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.10.red.old[,3])), (log2(euRatio.c10.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.10.red.old[,3])), (log2(euRatio.c10.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.10.red.old[,3])), (log2(euRatio.c10.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.10.red.old[,3])), (log2(euRatio.c10.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.10.red[,3])), (log2(euRatio.c10.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.10.red.old[,3])), (log2(euRatio.c10.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.10.red[,3])), (log2(euRatio.c10.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.10.red[,3])), (log2(euRatio.c10.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.10.red[,3])), (log2(euRatio.c10.red.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

dev.off()}
###############################################################################################
#CHROMOSOME 11

#RATIOS FOR CHROMOSOME 11

{ 
  euRatio.c11.red.old <- c(l001.11.red.old[,3],l002.11.red.old[,3],l003.11.red.old[,3],l004.11.red.old[,3],l005.11.red.old[,3],l006.11.red.old[,3],l007.11.red.old[,3],l008.11.red.old[,3],l009.11.red.old[,3],l011.11.red.old[,3],l028.11.red.old[,3])
pdf("Chrm_11_red_old.pdf")
boxplot((log2(l001.11.red.old[,3])),(log2(l002.11.red.old[,3])),(log2(l003.11.red.old[,3])),(log2(l004.11.red.old[,3])),(log2(l005.11.red.old[,3])),
        (log2(l006.11.red.old[,3])),(log2(l007.11.red.old[,3])),(log2(l008.11.red.old[,3])),(log2(l009.11.red.old[,3])),(log2(l011.11.red.old[,3])),
        (log2(l015.11.red[,3])),(log2(l028.11.red.old[,3])),(log2(l088.11.red[,3])),(log2(l108.11.red[,3])),(log2(l119.11.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 7 Reduced",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.11.red.old[,3])), (log2(euRatio.c11.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.11.red.old[,3])), (log2(euRatio.c11.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.11.red.old[,3])), (log2(euRatio.c11.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.11.red.old[,3])), (log2(euRatio.c11.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.11.red.old[,3])), (log2(euRatio.c11.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.11.red.old[,3])), (log2(euRatio.c11.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.11.red.old[,3])), (log2(euRatio.c11.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.11.red.old[,3])), (log2(euRatio.c11.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.11.red.old[,3])), (log2(euRatio.c11.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.11.red.old[,3])), (log2(euRatio.c11.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.11.red[,3])), (log2(euRatio.c11.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.11.red.old[,3])), (log2(euRatio.c11.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.11.red[,3])), (log2(euRatio.c11.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.11.red[,3])), (log2(euRatio.c11.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.11.red[,3])), (log2(euRatio.c11.red.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

#text(x=19, y=2.2, "*", pos=3, cex=1.5, col="red")
dev.off()}
###############################################################################################

#CHROMOSOME 11

#RATIOS FOR CHROMOSOME 11

{  
  euRatio.c11.old <- c(l001.11.old[,3],l002.11.old[,3],l003.11.old[,3],l004.11.old[,3],l005.11.old[,3],l006.11.old[,3],l007.11.old[,3],l008.11.old[,3],l009.11.old[,3],l011.11.old[,3],l028.11.old[,3])
pdf("Chrm_11_old.pdf")
boxplot((log2(l001.11.old[,3])),(log2(l002.11.old[,3])),(log2(l003.11.old[,3])),(log2(l004.11.old[,3])),(log2(l005.11.old[,3])),
        (log2(l006.11.old[,3])),(log2(l007.11.old[,3])),(log2(l008.11.old[,3])),(log2(l009.11.old[,3])),(log2(l011.11.old[,3])),
        (log2(l015.11[,3])),(log2(l028.11.old[,3])),(log2(l088.11[,3])),(log2(l108.11[,3])),(log2(l119.11[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 8",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.11.old[,3])), (log2(euRatio.c11.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.11.old[,3])), (log2(euRatio.c11.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.11.old[,3])), (log2(euRatio.c11.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.11.old[,3])), (log2(euRatio.c11.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.11.old[,3])), (log2(euRatio.c11.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.11.old[,3])), (log2(euRatio.c11.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.11.old[,3])), (log2(euRatio.c11.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.11.old[,3])), (log2(euRatio.c11.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.11.old[,3])), (log2(euRatio.c11.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.11.old[,3])), (log2(euRatio.c11.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.11[,3])), (log2(euRatio.c11.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.11.old[,3])), (log2(euRatio.c11.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.11[,3])), (log2(euRatio.c11.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.11[,3])), (log2(euRatio.c11.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.11[,3])), (log2(euRatio.c11.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

#text(x=19, y=2.2, "*", pos=3, cex=1.5, col="red")
dev.off()}
###############################################################################################
#CHROMOSOME 12

#RATIOS FOR CHROMOSOME 12

{  
  euRatio.c12.old <- c(l001.12.old[,3],l002.12.old[,3],l003.12.old[,3],l004.12.old[,3],l005.12.old[,3],l006.12.old[,3],l007.12.old[,3],l008.12.old[,3],l009.12.old[,3],l011.12.old[,3],l028.12.old[,3])
pdf("Chrm_12_old.pdf")
boxplot((log2(l001.12.old[,3])),(log2(l002.12.old[,3])),(log2(l003.12.old[,3])),(log2(l004.12.old[,3])),(log2(l005.12.old[,3])),
        (log2(l006.12.old[,3])),(log2(l007.12.old[,3])),(log2(l008.12.old[,3])),(log2(l009.12.old[,3])),(log2(l011.12.old[,3])),
        (log2(l015.12[,3])),(log2(l028.12.old[,3])),(log2(l088.12[,3])),(log2(l108.12[,3])),(log2(l119.12[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 8",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.12.old[,3])), (log2(euRatio.c12.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.12.old[,3])), (log2(euRatio.c12.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.12.old[,3])), (log2(euRatio.c12.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.12.old[,3])), (log2(euRatio.c12.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.12.old[,3])), (log2(euRatio.c12.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.12.old[,3])), (log2(euRatio.c12.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.12.old[,3])), (log2(euRatio.c12.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.12.old[,3])), (log2(euRatio.c12.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.12.old[,3])), (log2(euRatio.c12.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.12.old[,3])), (log2(euRatio.c12.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.12[,3])), (log2(euRatio.c12.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.12.old[,3])), (log2(euRatio.c12.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.12[,3])), (log2(euRatio.c12.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.12[,3])), (log2(euRatio.c12.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.12[,3])), (log2(euRatio.c12.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


dev.off()}

###############################################################################################

#CHROMOSOME 12 reduced 

#RATIOS FOR CHROMOSOME 12

{
  euRatio.c12.red.old <- c(l001.12.red.old[,3],l002.12.red.old[,3],l003.12.red.old[,3],l004.12.red.old[,3],l005.12.red.old[,3],l006.12.red.old[,3],l007.12.red.old[,3],l008.12.red.old[,3],l009.12.red.old[,3],l011.12.red.old[,3],l028.12.red.old[,3])
pdf("Chrm_12_red_old.pdf")
boxplot((log2(l001.12.red.old[,3])),(log2(l002.12.red.old[,3])),(log2(l003.12.red.old[,3])),(log2(l004.12.red.old[,3])),(log2(l005.12.red.old[,3])),
        (log2(l006.12.red.old[,3])),(log2(l007.12.red.old[,3])),(log2(l008.12.red.old[,3])),(log2(l009.12.red.old[,3])),(log2(l011.12.red.old[,3])),
        (log2(l015.12.red[,3])),(log2(l028.12.red.old[,3])),(log2(l088.12.red[,3])),(log2(l108.12.red[,3])),(log2(l119.12.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 7 Reduced",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.12.red.old[,3])), (log2(euRatio.c12.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.12.red.old[,3])), (log2(euRatio.c12.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.12.red.old[,3])), (log2(euRatio.c12.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.12.red.old[,3])), (log2(euRatio.c12.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.12.red.old[,3])), (log2(euRatio.c12.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.12.red.old[,3])), (log2(euRatio.c12.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.12.red.old[,3])), (log2(euRatio.c12.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.12.red.old[,3])), (log2(euRatio.c12.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.12.red.old[,3])), (log2(euRatio.c12.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.12.red.old[,3])), (log2(euRatio.c12.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.12.red[,3])), (log2(euRatio.c12.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.12.red.old[,3])), (log2(euRatio.c12.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.12.red[,3])), (log2(euRatio.c12.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.12.red[,3])), (log2(euRatio.c12.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.12.red[,3])), (log2(euRatio.c12.red.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


dev.off()}

###############################################################################################
#CHROMOSOME 13

#RATIOS FOR CHROMOSOME 13

{  
euRatio.c13.old <- c(l001.13.old[,3],l002.13.old[,3],l003.13.old[,3],l004.13.old[,3],l005.13.old[,3],l006.13.old[,3],l007.13.old[,3],l008.13.old[,3],l009.13.old[,3],l011.13.old[,3],l028.13.old[,3])
pdf("Chrm_13_old.pdf")
boxplot((log2(l001.13.old[,3])),(log2(l002.13.old[,3])),(log2(l003.13.old[,3])),(log2(l004.13.old[,3])),(log2(l005.13.old[,3])),
        (log2(l006.13.old[,3])),(log2(l007.13.old[,3])),(log2(l008.13.old[,3])),(log2(l009.13.old[,3])),(log2(l011.13.old[,3])),
        (log2(l015.13[,3])),(log2(l028.13.old[,3])),(log2(l088.13[,3])),(log2(l108.13[,3])),(log2(l119.13[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 8",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.13.old[,3])), (log2(euRatio.c13.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.13.old[,3])), (log2(euRatio.c13.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.13.old[,3])), (log2(euRatio.c13.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.13.old[,3])), (log2(euRatio.c13.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.13.old[,3])), (log2(euRatio.c13.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.13.old[,3])), (log2(euRatio.c13.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.13.old[,3])), (log2(euRatio.c13.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.13.old[,3])), (log2(euRatio.c13.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.13.old[,3])), (log2(euRatio.c13.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.13.old[,3])), (log2(euRatio.c13.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.13[,3])), (log2(euRatio.c13.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.13.old[,3])), (log2(euRatio.c13.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.13[,3])), (log2(euRatio.c13.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.13[,3])), (log2(euRatio.c13.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.13[,3])), (log2(euRatio.c13.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

#text(x=11, y=3.9, "*", pos=3, cex=1.5, col="red")
#text(x=20, y=3.9, "*", pos=3, cex=1.5, col="red")
dev.off()}
###############################################################################################
#CHROMOSOME 13

#RATIOS FOR CHROMOSOME 13

{  
  euRatio.c13.red.old <- c(l001.13.red.old[,3],l002.13.red.old[,3],l003.13.red.old[,3],l004.13.red.old[,3],l005.13.red.old[,3],l006.13.red.old[,3],l007.13.red.old[,3],l008.13.red.old[,3],l009.13.red.old[,3],l011.13.red.old[,3],l028.13.red.old[,3])
pdf("Chrm_13_red_old.pdf")
boxplot((log2(l001.13.red.old[,3])),(log2(l002.13.red.old[,3])),(log2(l003.13.red.old[,3])),(log2(l004.13.red.old[,3])),(log2(l005.13.red.old[,3])),
        (log2(l006.13.red.old[,3])),(log2(l007.13.red.old[,3])),(log2(l008.13.red.old[,3])),(log2(l009.13.red.old[,3])),(log2(l011.13.red.old[,3])),
        (log2(l015.13.red[,3])),(log2(l028.13.red.old[,3])),(log2(l088.13.red[,3])),(log2(l108.13.red[,3])),(log2(l119.13.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 7 Reduced",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.13.red.old[,3])), (log2(euRatio.c13.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.13.red.old[,3])), (log2(euRatio.c13.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.13.red.old[,3])), (log2(euRatio.c13.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.13.red.old[,3])), (log2(euRatio.c13.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.13.red.old[,3])), (log2(euRatio.c13.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.13.red.old[,3])), (log2(euRatio.c13.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.13.red.old[,3])), (log2(euRatio.c13.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.13.red.old[,3])), (log2(euRatio.c13.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.13.red.old[,3])), (log2(euRatio.c13.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.13.red.old[,3])), (log2(euRatio.c13.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.13.red[,3])), (log2(euRatio.c13.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.13.red.old[,3])), (log2(euRatio.c13.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.13.red[,3])), (log2(euRatio.c13.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.13.red[,3])), (log2(euRatio.c13.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.13.red[,3])), (log2(euRatio.c13.red.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

#text(x=11, y=3.9, "*", pos=3, cex=1.5, col="red")
#text(x=20, y=3.9, "*", pos=3, cex=1.5, col="red")
dev.off()}
###############################################################################################

#CHROMOSOME 14

#RATIOS FOR CHROMOSOME 14

{
  euRatio.c14.old <- c(l001.14.old[,3],l002.14.old[,3],l003.14.old[,3],l004.14.old[,3],l005.14.old[,3],l006.14.old[,3],l007.14.old[,3],l008.14.old[,3],l009.14.old[,3],l011.14.old[,3],l028.14.old[,3])
pdf("Chrm_14_old.pdf")
boxplot((log2(l001.14.old[,3])),(log2(l002.14.old[,3])),(log2(l003.14.old[,3])),(log2(l004.14.old[,3])),(log2(l005.14.old[,3])),
        (log2(l006.14.old[,3])),(log2(l007.14.old[,3])),(log2(l008.14.old[,3])),(log2(l009.14.old[,3])),(log2(l011.14.old[,3])),
        (log2(l015.14[,3])),(log2(l028.14.old[,3])),(log2(l088.14[,3])),(log2(l108.14[,3])),(log2(l119.14[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 8",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.14.old[,3])), (log2(euRatio.c14.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.14.old[,3])), (log2(euRatio.c14.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.14.old[,3])), (log2(euRatio.c14.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.14.old[,3])), (log2(euRatio.c14.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.14.old[,3])), (log2(euRatio.c14.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.14.old[,3])), (log2(euRatio.c14.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.14.old[,3])), (log2(euRatio.c14.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.14.old[,3])), (log2(euRatio.c14.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.14.old[,3])), (log2(euRatio.c14.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.14.old[,3])), (log2(euRatio.c14.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.14[,3])), (log2(euRatio.c14.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.14.old[,3])), (log2(euRatio.c14.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.14[,3])), (log2(euRatio.c14.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.14[,3])), (log2(euRatio.c14.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.14[,3])), (log2(euRatio.c14.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


dev.off()}
###############################################################################################

#CHROMOSOME 14

#RATIOS FOR CHROMOSOME 14

{  euRatio.c14.red.old <- c(l001.14.red.old[,3],l002.14.red.old[,3],l003.14.red.old[,3],l004.14.red.old[,3],l005.14.red.old[,3],l006.14.red.old[,3],l007.14.red.old[,3],l008.14.red.old[,3],l009.14.red.old[,3],l011.14.red.old[,3],l028.14.red.old[,3])
pdf("Chrm_14_red_old.pdf")
boxplot((log2(l001.14.red.old[,3])),(log2(l002.14.red.old[,3])),(log2(l003.14.red.old[,3])),(log2(l004.14.red.old[,3])),(log2(l005.14.red.old[,3])),
        (log2(l006.14.red.old[,3])),(log2(l007.14.red.old[,3])),(log2(l008.14.red.old[,3])),(log2(l009.14.red.old[,3])),(log2(l011.14.red.old[,3])),
        (log2(l015.14.red[,3])),(log2(l028.14.red.old[,3])),(log2(l088.14.red[,3])),(log2(l108.14.red[,3])),(log2(l119.14.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 7 Reduced",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.14.red.old[,3])), (log2(euRatio.c14.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.14.red.old[,3])), (log2(euRatio.c14.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.14.red.old[,3])), (log2(euRatio.c14.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.14.red.old[,3])), (log2(euRatio.c14.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.14.red.old[,3])), (log2(euRatio.c14.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.14.red.old[,3])), (log2(euRatio.c14.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.14.red.old[,3])), (log2(euRatio.c14.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.14.red.old[,3])), (log2(euRatio.c14.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.14.red.old[,3])), (log2(euRatio.c14.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.14.red.old[,3])), (log2(euRatio.c14.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.14.red[,3])), (log2(euRatio.c14.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.14.red.old[,3])), (log2(euRatio.c14.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.14.red[,3])), (log2(euRatio.c14.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.14.red[,3])), (log2(euRatio.c14.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.14.red[,3])), (log2(euRatio.c14.red.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


dev.off()}
###############################################################################################
#CHROMOSOME 15

#RATIOS FOR CHROMOSOME 15

{  
  euRatio.c15.old <- c(l001.15.old[,3],l002.15.old[,3],l003.15.old[,3],l004.15.old[,3],l005.15.old[,3],l006.15.old[,3],l007.15.old[,3],l008.15.old[,3],l009.15.old[,3],l011.15.old[,3],l028.15.old[,3])
pdf("Chrm_15_old.pdf")
boxplot((log2(l001.15.old[,3])),(log2(l002.15.old[,3])),(log2(l003.15.old[,3])),(log2(l004.15.old[,3])),(log2(l005.15.old[,3])),
        (log2(l006.15.old[,3])),(log2(l007.15.old[,3])),(log2(l008.15.old[,3])),(log2(l009.15.old[,3])),(log2(l011.15.old[,3])),
        (log2(l015.15[,3])),(log2(l028.15.old[,3])),(log2(l088.15[,3])),(log2(l108.15[,3])),(log2(l119.15[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 8",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.15.old[,3])), (log2(euRatio.c15.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.15.old[,3])), (log2(euRatio.c15.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.15.old[,3])), (log2(euRatio.c15.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.15.old[,3])), (log2(euRatio.c15.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.15.old[,3])), (log2(euRatio.c15.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.15.old[,3])), (log2(euRatio.c15.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.15.old[,3])), (log2(euRatio.c15.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.15.old[,3])), (log2(euRatio.c15.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.15.old[,3])), (log2(euRatio.c15.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.15.old[,3])), (log2(euRatio.c15.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.15[,3])), (log2(euRatio.c15.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.15.old[,3])), (log2(euRatio.c15.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.15[,3])), (log2(euRatio.c15.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.15[,3])), (log2(euRatio.c15.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.15[,3])), (log2(euRatio.c15.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

dev.off()}
###############################################################################################

#CHROMOSOME 15

#RATIOS FOR CHROMOSOME 15

{ 
  euRatio.c15.red.old <- c(l001.15.red.old[,3],l002.15.red.old[,3],l003.15.red.old[,3],l004.15.red.old[,3],l005.15.red.old[,3],l006.15.red.old[,3],l007.15.red.old[,3],l008.15.red.old[,3],l009.15.red.old[,3],l011.15.red.old[,3],l028.15.red.old[,3])
pdf("Chrm_15_red_old.pdf")
boxplot((log2(l001.15.red.old[,3])),(log2(l002.15.red.old[,3])),(log2(l003.15.red.old[,3])),(log2(l004.15.red.old[,3])),(log2(l005.15.red.old[,3])),
        (log2(l006.15.red.old[,3])),(log2(l007.15.red.old[,3])),(log2(l008.15.red.old[,3])),(log2(l009.15.red.old[,3])),(log2(l011.15.red.old[,3])),
        (log2(l015.15.red[,3])),(log2(l028.15.red.old[,3])),(log2(l088.15.red[,3])),(log2(l108.15.red[,3])),(log2(l119.15.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 7 Reduced",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.15.red.old[,3])), (log2(euRatio.c15.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.15.red.old[,3])), (log2(euRatio.c15.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.15.red.old[,3])), (log2(euRatio.c15.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.15.red.old[,3])), (log2(euRatio.c15.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.15.red.old[,3])), (log2(euRatio.c15.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.15.red.old[,3])), (log2(euRatio.c15.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.15.red.old[,3])), (log2(euRatio.c15.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.15.red.old[,3])), (log2(euRatio.c15.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.15.red.old[,3])), (log2(euRatio.c15.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.15.red.old[,3])), (log2(euRatio.c15.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.15.red[,3])), (log2(euRatio.c15.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.15.red.old[,3])), (log2(euRatio.c15.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.15.red[,3])), (log2(euRatio.c15.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.15.red[,3])), (log2(euRatio.c15.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.15.red[,3])), (log2(euRatio.c15.red.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

dev.off()}
###############################################################################################


#CHROMOSOME 16

#RATIOS FOR CHROMOSOME 16

{ 
  euRatio.c16.old <- c(l001.16.old[,3],l002.16.old[,3],l003.16.old[,3],l004.16.old[,3],l005.16.old[,3],l006.16.old[,3],l007.16.old[,3],l008.16.old[,3],l009.16.old[,3],l011.16.old[,3],l028.16.old[,3])
pdf("Chrm_16_old.pdf")
boxplot((log2(l001.16.old[,3])),(log2(l002.16.old[,3])),(log2(l003.16.old[,3])),(log2(l004.16.old[,3])),(log2(l005.16.old[,3])),
        (log2(l006.16.old[,3])),(log2(l007.16.old[,3])),(log2(l008.16.old[,3])),(log2(l009.16.old[,3])),(log2(l011.16.old[,3])),
        (log2(l015.16[,3])),(log2(l028.16.old[,3])),(log2(l088.16[,3])),(log2(l108.16[,3])),(log2(l119.16[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 8",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.16.old[,3])), (log2(euRatio.c16.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.16.old[,3])), (log2(euRatio.c16.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.16.old[,3])), (log2(euRatio.c16.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.16.old[,3])), (log2(euRatio.c16.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.16.old[,3])), (log2(euRatio.c16.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.16.old[,3])), (log2(euRatio.c16.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.16.old[,3])), (log2(euRatio.c16.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.16.old[,3])), (log2(euRatio.c16.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.16.old[,3])), (log2(euRatio.c16.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.16.old[,3])), (log2(euRatio.c16.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.16[,3])), (log2(euRatio.c16.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.16.old[,3])), (log2(euRatio.c16.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.16[,3])), (log2(euRatio.c16.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.16[,3])), (log2(euRatio.c16.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.16[,3])), (log2(euRatio.c16.old)), paired=FALSE, var.equal=TRUE)


abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


dev.off()}
###############################################################################################


#CHROMOSOME 16 reduced 

#RATIOS FOR CHROMOSOME 16

{ 
  euRatio.c16.red.old <- c(l001.16.red.old[,3],l002.16.red.old[,3],l003.16.red.old[,3],l004.16.red.old[,3],l005.16.red.old[,3],l006.16.red.old[,3],l007.16.red.old[,3],l008.16.red.old[,3],l009.16.red.old[,3],l011.16.red.old[,3],l028.16.red.old[,3])
pdf("Chrm_16_red_old.pdf")
boxplot((log2(l001.16.red.old[,3])),(log2(l002.16.red.old[,3])),(log2(l003.16.red.old[,3])),(log2(l004.16.red.old[,3])),(log2(l005.16.red.old[,3])),
        (log2(l006.16.red.old[,3])),(log2(l007.16.red.old[,3])),(log2(l008.16.red.old[,3])),(log2(l009.16.red.old[,3])),(log2(l011.16.red.old[,3])),
        (log2(l015.16.red[,3])),(log2(l028.16.red.old[,3])),(log2(l088.16.red[,3])),(log2(l108.16.red[,3])),(log2(l119.16.red[,3])),names=c( "1","2","3","4","5","6","7","8","9","11","15","28","88","108","119"),
        ylab="log2(fold change)", col=c("green", "green","green","green","green","green","green","green",
                                        "green",
                        "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Old Lines Chromsome 7 Reduced",las=3
        ,ylim=c(-2,2.5))
#use non-parametric because these are not normally distributed
#run same test on all lines, even those that are not "supposed" to be aneuploid
t.test((log2(l001.16.red.old[,3])), (log2(euRatio.c16.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l002.16.red.old[,3])), (log2(euRatio.c16.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l003.16.red.old[,3])), (log2(euRatio.c16.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l004.16.red.old[,3])), (log2(euRatio.c16.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l005.16.red.old[,3])), (log2(euRatio.c16.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l006.16.red.old[,3])), (log2(euRatio.c16.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l007.16.red.old[,3])), (log2(euRatio.c16.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l008.16.red.old[,3])), (log2(euRatio.c16.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l009.16.red.old[,3])), (log2(euRatio.c16.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l011.16.red.old[,3])), (log2(euRatio.c16.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l015.16.red[,3])), (log2(euRatio.c16.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l028.16.red.old[,3])), (log2(euRatio.c16.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l088.16.red[,3])), (log2(euRatio.c16.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l108.16.red[,3])), (log2(euRatio.c16.red.old)), paired=FALSE, var.equal=TRUE)
t.test((log2(l119.16.red[,3])), (log2(euRatio.c16.red.old)), paired=FALSE, var.equal=TRUE)

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


dev.off()}

```

``` {r lowly-expr genes,echo=FALSE,include=TRUE}
#Need to export the lowly-expressed genes from each ratio file produced from code chunk above 
#Want to do this in a loop

#ratioFiles<- list("l001.1","l001.10","l001.11","l001.12","l001.13","l001.14","l001.15","l001.16","l001.2","l001.3","l001.4","l001.5","l001.6","l001.7","l001.8","l001.9","l002.1","l002.10","l002.11","l002.12","l002.13","l002.14","l002.15","l002.16","l002.2","l002.3","l002.4","l002.5","l002.6","l002.7","l002.8","l002.9","l003.1","l003.10","l003.11","l003.12","l003.13","l003.14","l003.15","l003.16","l003.2","l003.3","l003.4","l003.5","l003.6","l003.7","l003.8","l003.9","l004.1","l004.10","l004.11","l004.12","l004.13","l004.14","l004.15","l004.16","l004.2","l004.3","l004.4","l004.5","l004.6","l004.7","l004.8","l004.9")
#ratioFiles<- c("l001.1","l001.10")
#setwd="/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/ratioFiles"

#for (i in ratioFiles) {
 # sample <- data.table(ratioFiles[i])
#  sample <- sample[which(sample[,3]<0.1),]
 # write.csv(unique(ratioFiles[i]),file=(paste(unique(ratioFiles[i]),".csv",sep="")))
#}



#setNames(list(l001.2,l002.2), c("l001.2","l002.2"))
ratioFiles <- list(l001.1,  l001.10,  l001.11,  l001.12,  l001.13,  l001.14,  l001.15,  l001.16,  l001.2,  l001.3,  l001.4,  l001.5,  l001.6,  l001.7,  l001.8,  l001.9,  l002.1,  l002.10,  l002.11,  l002.12,  l002.13,  l002.14,  l002.15,  l002.16,  l002.2,  l002.3,  l002.4,  l002.5,  l002.6,  l002.7,  l002.8,  l002.9,  l003.1,  l003.10,  l003.11,  l003.12,  l003.13,  l003.14,  l003.15,  l003.16,  l003.2,  l003.3, l003.4, l003.5,  l003.6,  l003.7, l003.8,  l003.9,  l004.1,  l004.10,  l004.11,  l004.12,  l004.13, l004.14, l004.15, l004.16, l004.2,  l004.3,  l004.4,  l004.5,  l004.6,  l004.7,  l004.8,  l004.9, l005.1, l005.10, l005.11, l005.12, l005.13, l005.14, l005.15, l005.16, l005.2, l005.3, l005.4, l005.6, l005.7, l005.8, l005.9, l006.1, l006.10, l006.11, l006.12, l006.13, l006.14, l006.15, l006.16, l006.2, l006.3, l006.4, l006.5, l006.6, l006.7, l006.8, l006.9, l007.1, l007.10, l007.11, l007.12, l007.13, l007.14, l007.15, l007.16, l007.2, l007.3, l007.4, l007.5, l007.6, l007.8, l007.9, l008.1, l008.10, l008.11, l001.12, l008.13, l008.14, l008.15, l008.16, l008.2, l008.3, l008.4, l008.5, l008.6, l008.7, l008.8, l008.9, l009.1, l009.10, l009.11, l009.12, l009.13, l009.14, l009.15, l009.16, l009.2, l009.3, l009.4, l009.5, l009.6, l009.7, l009.8, l009.9, l011.1, l011.10, l011.11, l011.12, l011.13, l011.14, l011.15, l011.16, l011.2, l011.3, l011.4, l011.5, l011.6, l011.7, l011.8, l011.9, l015.1, l015.10, l015.11, l015.12, l015.13, l015.14, l015.15, l015.16, l015.2, l015.3, l015.4, l015.5, l015.7, l015.6, l015.8, l015.9, l028.1, l028.10, l028.11, l028.12, l028.13, l028.14, l028.15, l028.16, l028.2, l028.3, l028.4, l028.5, l028.6, l028.7, l028.8, l028.9, l088.1, l088.10, l088.11, l088.12, l088.13, l088.14, l088.15, l088.16, l088.2, l088.3, l088.4, l088.5, l088.6, l088.7, l088.8, l088.9, l108.1, l108.10, l108.11, l108.12, l108.13, l108.15, l108.16, l108.2, l108.3, l108.4, l108.5, l108.6, l108.7, l108.8, l108.9, l119.1, l119.10, l119.11, l119.12, l119.13, l119.14, l119.15, l119.16, l119.2, l119.3, l119.4, l119.5, l119.6, l119.7, l119.8, l119.9)
names(ratioFiles) <-  c("l001.1","l001.10","l001.11","l001.12","l001.13","l001.14","l001.15","l001.16","l001.2","l001.3","l001.4","l001.5","l001.6","l001.7","l001.8","l001.9","l002.1","l002.10","l002.11","l002.12","l002.13","l002.14","l002.15","l002.16","l002.2","l002.3","l002.4","l002.5","l002.6","l002.7","l002.8","l002.9","l003.1","l003.10","l003.11","l003.12","l003.13","l003.14","l003.15","l003.16","l003.2","l003.3","l003.4","l003.5","l003.6","l003.7","l003.8","l003.9","l004.1","l004.10","l004.11","l004.12","l004.13","l004.14","l004.15","l004.16","l004.2","l004.3","l004.4","l004.5","l004.6","l004.7","l004.8","l004.9","l005.1","l005.10","l005.11","l005.12","l005.13","l005.14","l005.15","l005.16","l005.2","l005.3","l005.4","l005.6","l005.7","l005.8","l005.9","l006.1","l006.10","l006.11","l006.12","l006.13","l006.14","l006.15","l006.16","l006.2","l006.3","l006.4","l006.5","l006.6","l006.7","l006.8","l006.9","l007.1","l007.10","l007.11","l007.12","l007.13","l007.14","l007.15","l007.16","l007.2","l007.3","l007.4","l007.5","l007.6","l007.8","l007.9","l008.1","l008.10","l008.11","l001.12","l008.13","l008.14","l008.15","l008.16","l008.2","l008.3","l008.4","l008.5","l008.6","l008.7","l008.8","l008.9","l009.1","l009.10","l009.11","l009.12","l009.13","l009.14","l009.15","l009.16","l009.2","l009.3","l009.4","l009.5","l009.6","l009.7","l009.8","l009.9","l011.1","l011.10","l011.11","l011.12","l011.13","l011.14","l011.15","l011.16","l011.2","l011.3","l011.4","l011.5","l011.6","l011.7","l011.8","l011.9","l015.1","l015.10","l015.11","l015.12","l015.13","l015.14","l015.15","l015.16","l015.2","l015.3","l015.4","l015.5","l015.7","l015.6","l015.8","l015.9","l028.1","l028.10","l028.11","l028.12","l028.13","l028.14","l028.15","l028.16","l028.2","l028.3","l028.4","l028.5","l028.6","l028.7","l028.8","l028.9","l088.1","l088.10","l088.11","l088.12","l088.13","l088.14","l088.15","l088.16","l088.2","l088.3","l088.4","l088.5","l088.6","l088.7","l088.8","l088.9","l108.1","l108.10","l108.11","l108.12","l108.13","l108.15","l108.16","l108.2","l108.3","l108.4","l108.5","l108.6","l108.7","l108.8","l108.9","l119.1","l119.10","l119.11","l119.12","l119.13","l119.14","l119.15","l119.16","l119.2","l119.3","l119.4","l119.5","l119.6","l119.7","l119.8","l119.9")
ratioFiles
for (i in seq_along(ratioFiles)){
  file <- data.table(ratioFiles[[i]])
 clean.file <- ratioFiles[[i]][(ratioFiles[[i]][3] < 0.1), ]
  write.csv(clean.file, file=paste(names(ratioFiles)[i], "low_exp_genes_old.csv", sep="_"), row.names = F)
}

#low_expr_genes <- function(filename) {
#    file <- data.table(filename)
# clean.file <- filename[which(filename[,3]<0.1),]
#  write.csv(clean.file, file=paste(unique(filename), "low_exp_genes.csv", sep="_"), row.names = F)
#}

#low_expr_genes(l002.2)

#for(i in 1:5){
#  f()
#}
#for (i in  seq_along(ratioFiles)) {
#  low_expr_genes()
#}

#complete <- function(low_exp, id = 1:2) {
 #       ratioFiles<- list(l001.2,l002.2)
  #      dat <- data.table()
   #     tmp <- data.table()
    #            for (i in id) {
     #                   dat <- rbind(dat, ratioFiles[i])
      #                  clean.dat <- filename[which(filename[,3]<0.1),]
       #                 id <- i
        #                tmp <- rbind(tmp,data.frame(id,clean.dat))
         #       }
      #  write.csv(tmp, file=paste(unique(ratioFiles[i]), "low_exp_genes.csv", sep="_"), row.names = F)
#}

#complete <- function(directory, id = 1:332) {
 #       files_list <- list.files(path = directory, full.names = TRUE)
  #      dat <- data.frame()
   #     tmp <- data.frame()
    #            for (i in id) {
     #                   dat <- rbind(dat, read.csv(files_list[i]))
      #                  nobs <- sum(complete.cases(dat))
       #                 id <- i
        #                tmp <- rbind(tmp,data.frame(id,nobs))
         #       }


    #    tmp
#}

#complete
#file <- data.table(ratioFiles[[i]])
# clean.file <- ratioFiles[[i]][(ratioFiles[[i]]$ratio < 0.1), ]
#  write.csv(clean.file, file=paste(names(ratioFiles)[i], "low_exp_genes.csv", sep="_"), row.names = F)

#names(ratioFiles)
 
#write.csv(l004.9,file=(paste("Lowly_expr_genes","l004.9",".csv",sep="")))
 # sample <- read.table(ratioFiles[i],header=TRUE)
  #sample <- sample[which(sample[,3]<0.1),]
  #sample <- data.table(sample)  # make it a data table
  #dev.off()

```

``` {r fpkm vs p-value,echo=FALSE,include=TRUE}
#first need to read in the Cuffdiff data 
#first do this for one line, then make a loop to do it for each line
#setwd = "/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Redo/Cuffdiff_files/Diff/Tables"
samples <- c(1,2,3,4,5,6,7,8,9,11,15,28,88,119)

#sample1 <- read.table("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Redo/Cuffdiff_files/gene_exp_S1.diff",header=TRUE)

#plot(sample1$q_value,sample1$value_2,ylab="ancestor FPKM",xlab="FDR-adj p-value",main="Sample1")

list_of_titles <- c("Sample1","Sample108","Sample11","Sample119","Sample15","Sample2","Sample28","Sample3","Sample4","Sample5", "Sample6", "Sample7","Sample8","Sample88","Sample9")


# Plot separate ggplot figures in a loop.
library(ggplot2)

# Make list of variable names to loop over.
myFiles <- list.files(path="/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/Cuffdiff_MA_old",pattern = "*.diff", full.names = TRUE)
str(myFiles)
length(myFiles)
print(myFiles)
#View(myFiles)
# Make plots.
plot_list = list()

#plot each sample in myFiles, then save to a .pdf in the working directory 
for (i in 1:length(myFiles)) {
  setwd = "/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Redo/Cuffdiff_files/Diff/Tables"
  pdf(paste("FPKM_vs_p_val_",list_of_titles[i],"_old.pdf",sep=""))
  sample <- read.table(myFiles[i],header=TRUE)
  sample <- data.frame(sample)
  sample <- cSplit(sample,"locus",":") #split the locus column into "chr" and "locus"
  names <- c("test_id","gene_id","gene","sample_1","sample_2","status","value_1","value_2","log2.fold_change","test_stat","p_value","q_value","significant","chr","locus")
  colnames(sample) <- names
  sample <-  subset(sample, !grepl("MT", sample$chr))
  qval <- sample$q_value
  val2 <- sample$value_2
  plot(qval,val2,ylab="ancestor FPKM",xlab="FDR-adj p-value", 
       xlim=c(0,1),ylim=c(0,10000),main=list_of_titles[i])
  dev.off() 
}


```

``` {r finding_depth,echo=FALSE,include=TRUE}
#bam <- scanBam("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/bam_files/Holly-66B-redo-136232_S14_R1_001_tophat_out.sorted.bam")[[1]] # the result comes in nested lists
bamcoverage <- function (bamfile) {
  # read in the bam file
  bam <- scanBam(bamfile)[[1]]
  # filter reads without match position
  ind <- ! is.na(bam$pos)
  ## remove non-matches, they are not relevant to us
  bam <- lapply(bam, function(x) x[ind])
  ranges <- IRanges(start=bam$pos, width=bam$qwidth, names=make.names(bam$qname, unique=TRUE))
  ## names of the bam data frame:
  ## "qname"  "flag"   "rname"  "strand" "pos"    "qwidth"
  ## "mapq"   "cigar"  "mrnm"   "mpos"   "isize"  "seq"    "qual"
  ## construc: genomic ranges object containing all reads
  ranges <- GRanges(seqnames=Rle(bam$rname), ranges=ranges, strand=Rle(bam$strand), flag=bam$flag, readid=bam$rname )
  ## returns a coverage for each reference sequence (aka. chromosome) in the bam file
  return (mean(coverage(ranges)))      
}

myBamFiles <- list.files(path="/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/bam_files/",pattern="*.bam",full.names=TRUE)

print(myBamFiles)
list_of_titles <- c("11B","11C","18B","18C","1B","21B","2A","2B","2C","31B","31C","3A","3B","3C","49B","49C","4B","4C","59B","59C","5A","5B","5C","61B","61C","66C","69B","69C","6C","76B","76C","77B","77C","7C","8B","9A","9C","1","59","66","69","76","77","66B","11","18","21","4","49","61","7","8")
setwd = "/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/bam_files/"

for (i in 1:length(myBamFiles)) {
sample <- bamcoverage(myBamFiles[i])
sample <- data.frame(sample)
write.csv(sample,file=(paste("Depth_of_coverage_",list_of_titles[i],".csv",sep="")))
}

#sample <- bamcoverage("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/bam_files/Holly_66C_S21_R1_001_tophat_out.sorted.bam")
#sample <- data.frame(sample)

#write.csv(sample,file="Depth_of_coverage_66C.csv")
```

``` {r DE vs total genes,echo=FALSE,include=TRUE}
########################################################################################################
#Cuffdiff: Number of differentially expressed genes versus the number of genes on given chromosome
########################################################################################################

#read in the data
myFiles <- list.files(path="/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/Cuffdiff_MA_old",pattern = "*.diff", full.names = TRUE)

print(myFiles)
getwd()
#setwd("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Redo/Cuffdiff_files/Diff/Tables")

list_of_titles <- c("Sample1","Sample108","Sample11","Sample119","Sample15","Sample2","Sample28","Sample3","Sample4","Sample5","Sample6","Sample7","Sample8","Sample88","Sample9")

for (i in 1:length(myFiles)) {
  pdf(paste("DE_by_total_old_",list_of_titles[i],".pdf",sep=""))
  sample <- read.table(myFiles[i],header=TRUE)
  sample <- data.table(sample)  # make it a data table
  sample <- cSplit(sample,"locus",":") #split the locus column into "chr" and "locus"
  names <- c("test_id","gene_id","gene","sample_1","sample_2","status","value_1","value_2","log2.fold_change","test_stat","p_value","q_value","significant","chr","locus")
  colnames(sample) <- names
   #remove the mitochondria
  sample <-  subset(sample, !grepl("MT", sample$chr))
  chrms <- c("I","II","III","IV","V","VI","VII","VIII","IX","X","XI","XII","XIII","XIV","XV","XVI") #need to count number of rows where sample1$chr == "1" 
#and so on for each chromosome
  DE <- subset(sample,significant=="yes") #subset data into the "yeses"
  N.DE <- DE[,.N,by=chr] #count the number of rows of each chromosome that are DE (this is the number of genes on each chromosome)
  N.genes <- sample[,.N,by=chr] #count the number of rows of each chromosome (this is the number of genes on each chromosome)
  #tmp <- N.genes[which(N.genes[,1]==N.DE[,1]),]
 test <- merge(N.genes,N.DE,by="chr")
 name <- c("chr","N.genes","N.DE")
 colnames(test) <- name
 plot(N.DE~N.genes,data=test , xlab="total genes", ylab="differentially expressed genes",
    pch=1,xlim=c(0,1000),ylim=c(0,15),main=list_of_titles[i])
 abline(fit <- lm(N.DE ~ N.genes, data = test), col="blue")
 legend("topright", bty="n", legend=paste("R^2 =", format(summary(fit)$adj.r.squared, digits=4)))
coeff=data.frame(coefficients(fit))
 legend(700,14, bty="n", legend=paste("slope =", coef(fit)[2]))
  dev.off()
}

#next, put point on x-axis for the chromosomes that have 0 DE genes
#to do this, modify the data table to 


#rates = read.csv("file.txt")
#for(i in unique(rates[,1])){
#  pdf(file = paste0(i, ".pdf"))
#  match <- rates[rates[,1] == i, ]
#  plot(freq ~ temp, data = match)#,xlab="freq",ylab="temp")
#  abline(lm(freq ~ temp, data = match), col="red")
#  dev.off()
#}

#all(is.na(All$all))
#all(is.na(y))
#all(is.na(y^trans))


#  sample <- read.table("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Redo/Cuffdiff_files/Diff/Tables/gene_exp_S1.diff",header=TRUE)
 # sample <- data.table(sample)  # make it a data table
  #sample <- cSplit(sample,"locus",":") #split the locus column into "chr" and "locus"
#  names <- c("test_id","gene_id","gene","sample_1","sample_2","status","value_1","value_2","log2.fold_change","test_stat","p_value","q_value","significant","chr","locus")
 # colnames(sample) <- names
#  chrms <- c("I","II","III","IV","V","VI","VII","VIII","IX","X","XI","XII","XIII","XIV","XV","XVI") #need to count number of rows where sample1$chr == "1" 
#and so on for each chromosome
#  DE <- subset(sample,significant=="yes") #subset data into the "yeses"
#  N.DE <- DE[,.N,by=chr] #count the number of rows of each chromosome that are DE (this is the number of genes on each chromosome)
#  N.genes <- sample[,.N,by=chr] #count the number of rows of each chromosome (this is the number of genes on each chromosome)
#  All <- merge(N.DE, N.genes, by="chr")
#  names(All) <- c("chr","DE","all")
  #plot number of DE genes by number of total genes
 # plot(all ~ DE, data=All,xlab="total genes",ylab="differentially expressed genes",
  #     pch=1,xlim=c(0,1000),ylim=c(0,15),main=list_of_titles[i])
  #abline(lm(All$all ~ All$DE)) #doesn't seem to work: error: "Error in lm.fit(x, y, offset = offset, singular.ok = singular.ok, ...) : 0 (non-NA) cases"

#tmp <- N.genes[which(N.genes[,1]==N.DE[,1]),]
#test <- merge(N.genes,N.DE,by="chr")
#name <- c("chr","N.genes","N.DE")
#colnames(test) <- name
#plot(N.DE~N.genes,data=test , xlab="total genes", ylab="differentially expressed genes",
 #   pch=1,xlim=c(0,1000),ylim=c(0,15),main="test")

#abline(fit <- lm(N.DE ~ N.genes, data = test), col="red")
#legend("topright", bty="n", legend=paste("R^2 =", format(summary(fit)$adj.r.squared, digits=4)))
#coeff=data.frame(coefficients(fit))
#legend(700,14, bty="n", legend=paste("slope =", coef(fit)[2]))
#dev.list()

```
## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
