---
title: "Dosage Compensation Workflow"
author: "Holly McQueary"
date: "11/30/2017"
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE,echo=FALSE}
knitr::opts_chunk$set(echo = TRUE)

```

First load required packages for pipeline 
```{r load in required pakages, include = FALSE,echo=FALSE}
library(moments)
library("Hmisc")
library(pBrackets)
library(data.table)
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
library(ggplot2)
# 
# BiocManager::install("Rsamtools")
library(Rsamtools)
library(splitstackshape)
library(ggplot2)
library(pgirmess)
library(FSA)
library(dunn.test)
library(rcompanion)
library(grid)
library(dplyr)
library(tidyr)
library("knitr")
library("xtable")
library("stargazer")
```

Load data into matrices 

```{r load in data,make matrices, include=TRUE,echo=FALSE}

sc_counts <- read.csv("/Users/hollymcqueary/Dropbox/Github/Dosage_Compensation/data/Cuffnorm_Output/CuffnormJune19/genes_fpkm_table.csv",header=T,stringsAsFactors = FALSE)

MA_anc <- sc_counts[,122:124]
plot(MA_anc$MA_Anc_1,MA_anc$MA_Anc_0)
plot(MA_anc$MA_Anc_1,MA_anc$MA_Anc_2)
plot(MA_anc$MA_Anc_0,MA_anc$MA_Anc_2)

gene_attr <- read.csv("/Users/hollymcqueary/Dropbox/Github/Dosage_Compensation/data/Cuffnorm_Output/CuffnormJune19/genes_attr_table.csv",stringsAsFactors = FALSE)

class(sc_counts)
str(sc_counts)
# View(sc_counts)
# View(gene_attr)
str(gene_attr)
# colnames(sc_counts)[1] <- "chr"; colnames(sc_counts)[2] <- "tracking_id"
# colnames(gene_attr)[1] <- "tracking_id" ; colnames(gene_attr)[7] <- "locus"

myvars <- c("tracking_id","locus")
loci <- gene_attr[myvars]
# View(loci)
dim(loci)
sc_loci <- sc_counts[,1]
# View(sc_loci)
# str(loci)
# str(sc_counts)
# rownames(sc_counts) <- sc_counts[,1]
# rownames(loci) <- loci[,1]
typeof(loci)
typeof(sc_counts)
dim(sc_counts)
# sc_counts <- as.matrix(sc_counts)
loci <- as.data.frame(loci)
# colnames(loci) <- "tracking_id"
sc_counts.gen <- merge(sc_counts, loci, by.x="tracking_id",by.y="tracking_id")
# View(sc_counts.gen)
sc_counts.gen <- sc_counts.gen[,c(which(colnames(sc_counts.gen)=="locus"),which(colnames(sc_counts.gen)!="locus"))]
str(sc_counts.gen)
sc_counts.z <- sc_counts.gen
dim(sc_counts.z)
sc_counts.z[,3:125] <- sc_counts.z[,3:125]+.5 #add .5 to dataset so the zeroes dont mess up
#we want to exclude genes that are very lowly expressed

gene.means <- rowMeans(sc_counts.z[,3:125]) #average across the samples
table(gene.means<5)
lessThan5 <- gene.means<5
moreThan5 <- gene.means>5
# View(sc_counts.z)
anc.means.MA <- rowMeans(sc_counts.z[,123:125]) #average in the ancestor
table(anc.means.MA<5)
anc.means.MAold <- rowMeans(sc_counts.z[,3:5])
table(anc.means.MAold<5)
anc.means.GC <- rowMeans(sc_counts.z[120:122]) #average in the GC ancestor
table(anc.means.GC<5)
# View(sc_counts.z)
table(sc_counts.z$locus=="MT")
notMito <- sc_counts.z$locus!="MT"
#if the average fpkm is less than 5 in the ancestor or less than 5 in the overall, remove the gene

# rm(sc_counts.final)
sc_counts.1 <- sc_counts.z[moreThan5,]
notMito <- sc_counts.1$locus!="MT"
sc_counts.2 <- sc_counts.1[notMito,]
noSn <- grepl("sn", sc_counts.2$tracking_id)
sc_counts.3 <- sc_counts.2[!noSn,]
noT <- grepl("t", sc_counts.3$tracking_id)
sc_counts.4 <- sc_counts.3[!noT,]
sc_counts.final <- sc_counts.4
# View(sc_counts.final)
#######################################################
# write.csv(sc_counts.final,file="sc_counts.final.csv")
#final matrix for ancestor and descendans
colnames(sc_counts.final)
sc_ancGC <- sc_counts.final[,c(1,2,120:122)]
# View(sc_ancGC)
sc_ancMAnew <- sc_counts.final[,c(1,2,124:125)]
# View(sc_ancMAnew)
sc_ancMAold <- sc_counts.final[,c(1:5)]
# View(sc_ancMAold)
#average for each GC line
#going to analyze 21 and 59 separately since I need to remove the replicates that don't match to them and I don't know what they are yet. 
#this one is for line 11 to have the replicate that originally belonged to line 21 
sc_des.GC <- sc_counts.final[,c(1:2,60:62,69:71,72:74,78:80,81:83,84:86,87:89,93:95,96:98,99:101,102:104,105:107,108:110,111:113,114:116,117:119)]
str(sc_des.GC)
# View(sc_des.GC)
sc_des.GC$locus[sc_des.GC$locus == "I"] = 1
sc_des.GC$locus[sc_des.GC$locus == "II"] = 2
sc_des.GC$locus[sc_des.GC$locus == "III"] = 3
sc_des.GC$locus[sc_des.GC$locus == "IV"] = 4
sc_des.GC$locus[sc_des.GC$locus == "V"] = 5
sc_des.GC$locus[sc_des.GC$locus == "VI"] = 6
sc_des.GC$locus[sc_des.GC$locus == "VII"] = 7
sc_des.GC$locus[sc_des.GC$locus == "VIII"] = 8
sc_des.GC$locus[sc_des.GC$locus == "IX"] = 9
sc_des.GC$locus[sc_des.GC$locus == "X"] = 10
sc_des.GC$locus[sc_des.GC$locus == "XI"] = 11
sc_des.GC$locus[sc_des.GC$locus == "XII"] = 12
sc_des.GC$locus[sc_des.GC$locus == "XIII"] = 13
sc_des.GC$locus[sc_des.GC$locus == "XIV"] = 14
sc_des.GC$locus[sc_des.GC$locus == "XV"] = 15
sc_des.GC$locus[sc_des.GC$locus == "XVI"] = 16

# sc_des.GC <- within(sc_des.GC, locus <- factor(locus, labels = c(1,2,3,4,9,5,6,7,8,10,11,12,13,14,15,16)))
# new$MY_NEW_COLUMN <-as.numeric(as.character(new[,2]))
# sc_des.GC$locus <- as.numeric(as.character(sc_des.GC[,2]))
# as.numeric(levels(sc_des.GC$locus)[sc_des.GC$locus])
# View(sc_des.GC)
str(sc_des.GC)
colnames(sc_des.GC)
sc_des.MA_old <- sc_counts.final[,c(1:2,6:8,9:11,12:14,15:17,18:20,21:23,24:26,27:29,30:32,33:35,36:38,39:41,42:44,45:47,48:50)]
sc_des.MA_old$locus[sc_des.MA_old$locus == "I"] = 1
sc_des.MA_old$locus[sc_des.MA_old$locus == "II"] = 2
sc_des.MA_old$locus[sc_des.MA_old$locus == "III"] = 3
sc_des.MA_old$locus[sc_des.MA_old$locus == "IV"] = 4
sc_des.MA_old$locus[sc_des.MA_old$locus == "V"] = 5
sc_des.MA_old$locus[sc_des.MA_old$locus == "VI"] = 6
sc_des.MA_old$locus[sc_des.MA_old$locus == "VII"] = 7
sc_des.MA_old$locus[sc_des.MA_old$locus == "VIII"] = 8
sc_des.MA_old$locus[sc_des.MA_old$locus == "IX"] = 9
sc_des.MA_old$locus[sc_des.MA_old$locus == "X"] = 10
sc_des.MA_old$locus[sc_des.MA_old$locus == "XI"] = 11
sc_des.MA_old$locus[sc_des.MA_old$locus == "XII"] = 12
sc_des.MA_old$locus[sc_des.MA_old$locus == "XIII"] = 13
sc_des.MA_old$locus[sc_des.MA_old$locus == "XIV"] = 14
sc_des.MA_old$locus[sc_des.MA_old$locus == "XV"] = 15
sc_des.MA_old$locus[sc_des.MA_old$locus == "XVI"] = 16

# sc_des.MA_old <- within(sc_des.MA_old, locus <- factor(locus, labels = c(1,2,3,4,9,5,6,7,8,10,11,12,13,14,15,16)))
# as.integer(sc_des.MA_old$locus)
# View(sc_des.MA_old)
colnames(sc_des.MA_old)
sc_des.MAnew<- sc_counts.final[,c(1:2,75:77,90:92,51:53,54:56,57:59,63:65,66:68)]
sc_des.MAnew$locus[sc_des.MAnew$locus == "I"] = 1
sc_des.MAnew$locus[sc_des.MAnew$locus == "II"] = 2
sc_des.MAnew$locus[sc_des.MAnew$locus == "III"] = 3
sc_des.MAnew$locus[sc_des.MAnew$locus == "IV"] = 4
sc_des.MAnew$locus[sc_des.MAnew$locus == "V"] = 5
sc_des.MAnew$locus[sc_des.MAnew$locus == "VI"] = 6
sc_des.MAnew$locus[sc_des.MAnew$locus == "VII"] = 7
sc_des.MAnew$locus[sc_des.MAnew$locus == "VIII"] = 8
sc_des.MAnew$locus[sc_des.MAnew$locus == "IX"] = 9
sc_des.MAnew$locus[sc_des.MAnew$locus == "X"] = 10
sc_des.MAnew$locus[sc_des.MAnew$locus == "XI"] = 11
sc_des.MAnew$locus[sc_des.MAnew$locus == "XII"] = 12
sc_des.MAnew$locus[sc_des.MAnew$locus == "XIII"] = 13
sc_des.MAnew$locus[sc_des.MAnew$locus == "XIV"] = 14
sc_des.MAnew$locus[sc_des.MAnew$locus == "XV"] = 15
sc_des.MAnew$locus[sc_des.MAnew$locus == "XVI"] = 16
# sc_des.MAnew <- within(sc_des.MAnew, locus <- factor(locus, labels = c(1,2,3,4,9,5,6,7,8,10,11,12,13,14,15,16)))
# as.numeric(sc_des.MAnew$locus)
# View(sc_des.MAnew)
#View(sc_des.GC)
#View(sc_des.GC.r)
#for lines 21 and 59 
#need 12:14, 30:32
# colnames(sc_des.GC)
# sc_des.GC.r1 <- sc_des.GC[,c(1:2,13:14,30:31,39:40)]
# colnames(sc_des.GC.r1)[3] <- "X21_0"
# colnames(sc_des.GC.r1)[4] <- "X21_1"
# sc_des.GC$X66_0 <- c()
# sc_des.GC$X66_1 <- c()
# sc_des.GC$X66_2 <- c()
# sc_des.GC$X21_0 <- c()
# sc_des.GC$X21_2 <- c()
# sc_des.GC$X59_0 <- c()
# sc_des.GC$X59_1 <- c()
# sc_des.GC$X59_2 <- c()
# sc_des.GC[4] <- c()
# colnames(sc_des.GC)
# colnames(sc_des.GC)[11] <- "X11_0"
# colnames(sc_des.GC)
#21, 59, and 66 by themselves


#View(sc_des.GC)
#View(sc_des.GC.r1)

#View(sc_des.GC.r1)
#str(sc_des.GC.r1)

sc_des.MA <- sc_counts.final[,c(1:11,15:23,33:35,51:53)] #average for each MA line
# write.csv(sc_des.MA,file="sc_des.MA.csv")
# colnames(sc_des.MA)
# colnames(sc_counts.final)
# sc_des.MA_old <- sc_counts.final[,c(1:2,93:140)]
# colnames(sc_des.MA_old)

# index1 <- which(sc_anc[,1]==1)
# index2 <- which(sc_anc[,1]==2)
# index3 <- which(sc_anc[,1]==3)
# index4 <- which(sc_anc[,1]==4)
# index5 <- which(sc_anc[,1]==5)
# index6 <- which(sc_anc[,1]==6)
# index7 <- which(sc_anc[,1]==7)
# index8 <- which(sc_anc[,1]==8)
# index9 <- which(sc_anc[,1]==9)
# index10 <- which(sc_anc[,1]==10)
# index11 <- which(sc_anc[,1]==11)
# index12 <- which(sc_anc[,1]==12)
# index13 <- which(sc_anc[,1]==13)
# index14 <- which(sc_anc[,1]==14)
# index15 <- which(sc_anc[,1]==15)
# index16 <- which(sc_anc[,1]==16)



# FOLD CHANGES: average of line over average of ancestor
###ancestor average
#PROBLEM HERE! I was using the old code and not taking into account that I had two ancestors! now I have 3...
ancAvg.GC <- rowMeans(sc_ancGC[,3:5],na.rm=T)
ancAvg.GC <- cbind(sc_des.GC[,1:2],ancAvg.GC)
# as.numeric(ancAvg.GC$locus)
str(ancAvg.GC)
ancAvg.GC[,1]
ancAvg.MA <- rowMeans(sc_ancMAnew[,3:4],na.rm=T)
ancAvg.MA <- cbind(sc_des.MAnew[,1:2],ancAvg.MA)
str(ancAvg.MA)
ancAvg.MA.old <- rowMeans(sc_ancMAold[,3:5],na.rm=T)
ancAvg.MA.old <- cbind(sc_des.MA_old[,1:2],ancAvg.MA.old)

str(ancAvg.MA.old)

ancAvgs <- cbind(ancAvg.MA,ancAvg.MA.old$ancAvg.MA.old)
colnames(ancAvgs) <- c("locus","tracking_id","new","old")
library(reshape2)
ancAvgsMelted <- melt(ancAvgs, value.name = "run")
str(ancAvgsMelted)
ancAvgs2 <- ancAvgsMelted[,3:4]
levels(ancAvgs2$variable)
results.aov <- aov(run~variable,data=ancAvgs2)
summary(results.aov)

library(ggplot2)
library(ggpubr)
ancAvgsPlot <- ggplot(ancAvgs, aes(x=ancAvgs$new,y=ancAvgs$old))
ancAvgsPlot + geom_point(aes(x=ancAvgs$new,y=ancAvgs$old)) +scale_x_continuous(trans = 'log10') +
  scale_y_continuous(trans = 'log10') + ylab("Old Seq Run") + xlab("New Seq Run") + stat_regline_equation(
  mapping = aes(x=ancAvgs$new,y=ancAvgs$old),
  data = ancAvgs,
  formula = ancAvgs$new ~ ancAvgs$old,
  label.x.npc = "left",
  label.y.npc = "top",
  label.x = NULL,
  label.y = NULL,
  output.type = "expression",
  geom = "text",
  position = "identity",
  na.rm = FALSE,
  show.legend = NA,
  inherit.aes = TRUE)

pdf("scatterPlotAncAvgs.pdf")
ggscatter(ancAvgs, x = "old", y = "new", add = "reg.line") +
  stat_cor(label.x = 5, label.y = 15) +
  stat_regline_equation(label.x = 10, label.y = 15) + scale_x_continuous(trans = 'log2') +
  scale_y_continuous(trans = 'log2')
dev.off()
```



```{r variance, include=TRUE,echo=FALSE}

## Variance comparison between aneuploid and euploid lines 
## 152 and 1 

line1.152 <- sc_counts.final[,c(1:2,6:8,66:68)]
# View(line1.152)
# subset into chr 1 and chr 7
l1.152.1 <- line1.152[ which(line1.152$locus=='I'),]
# View(l1.152.1)
l1.152.1$average1 <- rowMeans(l1.152.1[,3:5])
l1.152.1$average152 <- rowMeans(l1.152.1[,6:8])
var.dfI <- cbind(l1.152.1$average1,l1.152.1$average152)
# View(var.dfI)
colnames(var.dfI) <- c("L1","L152")
library(reshape2)
var.dfI <- melt(var.dfI)
colnames(var.dfI) <- c("na","line","fpkm")
# View(var.dfI)
str(var.dfI)
var.dfI$line <- as.character(var.dfI$line)
library(car)
leveneTest(fpkm ~ line, var.dfI)
str(l1.152.1)

library(tidyverse)
library(reshape2)
sc_counts.final.t <- as_tibble(sc_counts.final)
sc_counts.final.t
sample1="152"
sample2="001"
chr1="I"
chr2="VII"
anc="MA_Anc"
anc2 = "SCA"

varianceLevenesTest2Chr <- function(sample1,sample2,anc,anc2,chr1,chr2) {
  line <- sc_counts.final %>% select(locus,tracking_id,contains(sample1),contains(sample2),contains(anc),contains(anc2))
  line1 <- line[ which(line$locus==chr1),]
  line1$average1 <- rowMeans(line1[,3:5])
  line1$average2 <- rowMeans(line1[,6:8])
  line1$average3 <- rowMeans(line1[,9:11])
  line1$average4 <- rowMeans(line1[,12:14])
  line1$ratio1 <- line1$average1 / line1$average3
  line1$ratio2 <- line1$average2/line1$average4
  df1 <- cbind(line1$ratio1,line1$ratio2)
  colnames(df1) <- c("L1","L2")
  df1 <- melt(df1)
  colnames(df1) <- c("na","line","ratio")
  leveneAn1 <- leveneTest(ratio ~ line, df1)
    ### second aneuploid chromosome
  line2 <- line[ which(line$locus==chr2),]
  line2$average1 <- rowMeans(line2[,3:5])
  line2$average2 <- rowMeans(line2[,6:8])
  line2$average3 <- rowMeans(line2[,9:11])
  line2$average4 <- rowMeans(line2[,12:14])
  line2$ratio1 <- line2$average1 / line2$average3
  line2$ratio2 <- line2$average2/line2$average4
  df2 <- cbind(line2$ratio1,line2$ratio2)
  colnames(df2) <- c("L1","L2")
  df2 <- melt(df2)
  colnames(df2) <- c("na","line","ratio")
  leveneAn2 <- leveneTest(ratio ~ line, df2)
  ### for euploid chromosomes 
  line3 <- line[ which(line$locus!=chr1 & which(line$locus!=chr2)),]
  line3$average1 <- rowMeans(line3[,3:5])
  line3$average2 <- rowMeans(line3[,6:8])
  line3$average3 <- rowMeans(line3[,9:11])
  line3$average4 <- rowMeans(line3[,12:14])
  line3$ratio1 <- line3$average1 / line3$average3
  line3$ratio2 <- line3$average2/line3$average4
  df3 <- cbind(line3$ratio1,line3$ratio2)
  colnames(df3) <- c("L1","L2")
  df3 <- melt(df3)
  colnames(df3) <- c("na","line","ratio")
  leveneAn3 <- leveneTest(ratio ~ line, df3)
  res <- data.frame(matrix(data=NA,nrow=6,ncol=2))
  colnames(res) <- c("chr","pVal")
  res$chr <- c(chr1,"NA",chr2,"NA","Eup","NA")
  res$pVal <- c(leveneAn1$`Pr(>F)`,leveneAn2$`Pr(>F)`,leveneAn3$`Pr(>F)`)
  return(res)
}
c(sample1,sample2,anc,anc2,chr1,chr2)
levene1.152 <- varianceLevenesTest2Chr("152","001","MA_Anc","SCA", "I","VII")
levene1.152

levene2.117 <- varianceLevenesTest2Chr("117","002","MA_Anc","SCA", "V","V")
levene2.117

levene3.123 <- varianceLevenesTest2Chr("123","003","MA_Anc","SCA", "V","V")
levene3.123


levene4.108 <- varianceLevenesTest2Chr("108","004","SCA","SCA", "VII","IX")
levene4.108

levene5.15 <- varianceLevenesTest2Chr("15","005","SCA","SCA", "IX","IX")
levene5.15

levene6.29 <- varianceLevenesTest2Chr("29","006","MA_Anc","SCA", "IX","IX")
levene6.29

levene7.88 <- varianceLevenesTest2Chr("88","007","SCA","SCA", "IX","IX")
levene7.88

levene8.119 <- varianceLevenesTest2Chr("119","8","SCA","SCA", "IX","IX")
levene8.119

levene9.11 <- varianceLevenesTest2Chr("011","009","SCA","SCA", "XIV","XIV")
levene9.11

levene28.112 <- varianceLevenesTest2Chr("112","28","MA_Anc","SCA", "XVI","XVI")
levene28.112

levene7.1 <- varianceLevenesTest2Chr("X7","X1","GC_Anc","GC_Anc", "I","I")
levene7.1

levene11.2 <- varianceLevenesTest2Chr("X11","X2","GC_Anc","GC_Anc", "I","XV")
levene11.2

levene18.3 <- varianceLevenesTest2Chr("X18","X3","GC_Anc","GC_Anc", "I","I")
levene18.3

levene4.5 <- varianceLevenesTest2Chr("X4","X5","GC_Anc","GC_Anc", "V","V")
levene4.5

levene49.9 <- varianceLevenesTest2Chr("X49","X9","GC_Anc","GC_Anc", "V","V")
levene49.9

levene59.69 <- varianceLevenesTest2Chr("X59","X69","GC_Anc","GC_Anc", "VII","VII")
levene59.69

levene61.1 <- varianceLevenesTest2Chr("X61","X1","GC_Anc","GC_Anc", "VII","VII")
levene61.1

levene76.2 <- varianceLevenesTest2Chr("X76","X2","GC_Anc","GC_Anc", "IX","XIV")
levene76.2


levene77.3 <- varianceLevenesTest2Chr("X77","X3","GC_Anc","GC_Anc", "XII","XII")
levene77.3

levene8.5 <- varianceLevenesTest2Chr("X8","X5","GC_Anc","GC_Anc", "XVI","XVI")
levene8.5

```

Automated functions to make matrices and perform computations on the data in them 
```{r automated functions, include = TRUE,echo=FALSE}
#first write some functions that help automate the process
##########
#this gets the average FPKM for a given line from the sc_des matrix
getAvgFPKM.MAnew <- function(line) { #for line as "###" string
  r1 <- paste(c("X",line,"_0"),collapse="") #triplicates are named X[LINE]_[rep]
  r2 <- paste(c("X",line,"_1"),collapse="")
  r3 <- paste(c("X",line,"_2"),collapse="")
  reps <- sc_des.MAnew[,c("locus","tracking_id",r1,r2,r3)] #get those 4 columns out of the descendant dataset
  reps[reps==0] <- NA #replace 0 values with NA as before
  desAvg.MA <- rowMeans(reps[,c(-1,-2)],na.rm=T)
  desAvg.MA <- cbind(sc_des.MAnew[,c(1,2)],desAvg.MA)
  #colnames(desAvg) <- c("chr","gene",paste(c(line,"_Avg"),collapse=""))
  return(desAvg.MA) #returned is a matrix equivalent to ancAvg: chrm, gene, and line average FPKM 
}


getAvgFPKM.MA.old <- function(line) { #for line as "###" string
  r1 <- paste(c("SC",line,"_0"),collapse="") #triplicates are named X[LINE]_[rep]
  r2 <- paste(c("SC",line,"_1"),collapse="")
  r3 <- paste(c("SC",line,"_2"),collapse="")
  reps <- sc_des.MA_old[,c("locus","tracking_id",r1,r2,r3)] #get those 4 columns out of the descendant dataset
  reps[reps==0] <- NA #replace 0 values with NA as before
  desAvg.MA.old <- rowMeans(reps[,c(-1,-2)],na.rm=T)
  desAvg.MA.old <- cbind(sc_des.MA_old[,c(1,2)],desAvg.MA.old)
  #colnames(desAvg) <- c("chr","gene",paste(c(line,"_Avg"),collapse=""))
  return(desAvg.MA.old) #returned is a matrix equivalent to ancAvg: chrm, gene, and line average FPKM 
}


getAvgFPKM.GC <- function(line) { #for line as "###" string
  r1 <- paste(c("X",line,"_0"),collapse="") #triplicates are named X[LINE]_[rep]
  r2 <- paste(c("X",line,"_1"),collapse="")
  r3 <- paste(c("X",line,"_2"),collapse="")
  reps <- sc_des.GC[,c("locus","tracking_id",r1,r2,r3)] #get those 4 columns out of the descendant dataset
  reps[reps==0] <- NA #replace 0 values with NA as before
  desAvg.GC <- rowMeans(reps[,c(-1,-2)],na.rm=T)
  desAvg.GC <- cbind(sc_des.GC[,c(1,2)],desAvg.GC)
  colnames(desAvg.GC) <- c("chr","gene",paste(c(line,"_Avg"),collapse=""))
  return(desAvg.GC) #returned is a matrix equivalent to ancAvg: chrm, gene, and line average FPKM 
}

##### Function to get FPKM ratio 
getFPKMRatio.MAnew <- function(line) { #line must be a 3 digit character
  desAvg.MA <- getAvgFPKM.MAnew(line) #first call avg fpkm script, which returns a 3 column matrix
  ratioByGene.MA <- desAvg.MA[,3]/ancAvg.MA[,3]
  rat.MA <- cbind(sc_des.MAnew[,1:2],ratioByGene.MA) #add gene and chrm values
  colnames(rat.MA) <- c("chr","tracking_id","ratio")
  return(rat.MA)
}
# Matrix returned is "rat.MA" - ratio MA 

getFPKMRatio.MA.old <- function(line) { #line must be a 3 digit character
  desAvg.MA.old <- getAvgFPKM.MA.old(line) #first call avg fpkm script, which returns a 3 column matrix
  ratioByGene.MA.old <- desAvg.MA.old[,3]/ancAvg.MA.old[,3]
  rat.MA.old <- cbind(sc_des.MA_old[,1:2],ratioByGene.MA.old) #add gene and chrm values
  colnames(rat.MA.old) <- c("chr","tracking_id","ratio")
  return(rat.MA.old)
}

getFPKMRatio.GC <- function(line) { #line must be a 3 digit character
  desAvg.GC <- getAvgFPKM.GC(line) #first call avg fpkm script, which returns a 3 column matrix
  ratioByGene.GC <- desAvg.GC[,3]/ancAvg.GC[,3]
  rat.GC <- cbind(sc_des.GC[,1:2],ratioByGene.GC) #add gene and chrm values
  colnames(rat.GC) <- c("chr","tracking_id","ratio")
  return(rat.GC)
}

####### Function to get chromosome ratios 
getChrmRatio.MAnew <- function(line, chr){ #input line as "XXX", chr as a number
  rat.MA <- getFPKMRatio.MAnew(line) #get list of ratios which includes chrm and gene
  rat.c.MA <- rat.MA[(rat.MA[,1]==chr),] #get just the rows for the appropriate chrom
  rat.cs.MA <- rat.c.MA[order(rat.c.MA[,3]),]#order by column 3
  return(rat.cs.MA)
}

getChrmRatio.MA.old <- function(line, chr){ #input line as "XXX", chr as a number
  rat.MA.old <- getFPKMRatio.MA.old(line) #get list of ratios which includes chrm and gene
  rat.c.MA.old <- rat.MA.old[(rat.MA.old[,1]==chr),] #get just the rows for the appropriate chrom
  rat.cs.MA.old <- rat.c.MA.old[order(rat.c.MA.old[,3]),]#order by column 3
  return(rat.cs.MA.old)
}

getChrmRatio.GC <- function(line, chr){ #input line as "XXX", chr as a number
  rat.GC <- getFPKMRatio.GC(line) #get list of ratios which includes chrm and gene
  rat.c.GC <- rat.GC[(rat.GC[,1]==chr),] #get just the rows for the appropriate chrom
  rat.cs.GC <- rat.c.GC[order(rat.c.GC[,3]),]#order by column 3
  return(rat.cs.GC)
}

# returns matrix of ratios of chromosomes (rat.cs.GC/MA/MAold)

############### Functions to make matrix of median ratios without those ~1 

y.MAnew=NULL
rep.MAnew <- c("112","115","117","123","152","29","50")
for(line in rep.MAnew){
  for(chr in 1:16){
    y.MAnew <- c(y.MAnew,median(getChrmRatio.MAnew(line,chr)[,3],na.rm=T))
  }
}

ratMatMed.MA <- matrix(y.MAnew,nrow=16,ncol=7)
colnames(ratMatMed.MA) <- rep.MAnew
ratMatMed.MA <- cbind(c(1:16),ratMatMed.MA)
colnames(ratMatMed.MA)[1] <- "chr"
#remove the ratios that are about 1
ratMatMedRed.MA <- ratMatMed.MA
ratMatMedRed.MA[ratMatMedRed.MA<1.35] <- NA

# returns ratio matrix median reduced: ratMatMedRed 
  # reduced because removed ratios that were around one 


y.MA.old=NULL
rep.MA.old <- c("001","002","003","004","005","006","007","008","009","011","015","028","088","108","119")
#View(rep.MA)
for(line in rep.MA.old){
  for(chr in 1:16){
    y.MA.old <- c(y.MA.old,median(getChrmRatio.MA.old(line,chr)[,3],na.rm=T))
  }
}



ratMatMed.MA.old <- matrix(y.MA.old,nrow=16,ncol=15)
colnames(ratMatMed.MA.old) <- rep.MA.old
ratMatMed.MA.old <- cbind(c(1:16),ratMatMed.MA.old)
colnames(ratMatMed.MA.old)[1] <- "chr"

#remove the ratios that are about 1
ratMatMedRed.MA.old <- ratMatMed.MA.old
ratMatMedRed.MA.old[ratMatMedRed.MA.old<1.35] <- NA


str(sc_des.MAnew)
str(sc_des.GC)
y.GC=NULL
colnames(sc_des.GC)
rep.GC <- c("11","18","1","2","3","49","4","59","5", "61","69","76","77","7","8","9")
# line = "11"
# chr = "1"
for(line in rep.GC){
  for(chr in 1:16){
    y.GC <- c(y.GC,median(getChrmRatio.GC(line,chr)[,3],na.rm=T))
  }
}
View(y.GC)

ratMatMed.GC <- matrix(y.GC,nrow=16,ncol=16)
# View(ratMatMed.GC)
colnames(ratMatMed.GC) <- rep.GC
ratMatMed.GC <- cbind(c(1:16),ratMatMed.GC)
colnames(ratMatMed.GC)[1] <- "chr"
#remove the ratios that are about 1
ratMatMedRed.GC <- ratMatMed.GC
ratMatMedRed.GC[ratMatMedRed.GC<1.35] <- NA


####################
#make a matrix that is the LOG2 RATIO

yl.MA=NULL
for(line in rep.MAnew){
  for(chrm in 1:16){
    yl.MA <- c(yl.MA,median(log2(getChrmRatio.MAnew(line,chrm)[,3])))
  }
}

warnings()
logMat.MAnew <- matrix(yl.MA,nrow=16,ncol=7)
colnames(logMat.MAnew) <- rep.MAnew
logMat.MAnew <- data.frame(logMat.MAnew)
chrms <- c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16)
logMat2.MAnew <-cbind(chrms,logMat.MAnew)
# write this output to a file 
write.csv(logMat2.MAnew, file="logMat.MAnew.csv")

yl.MA.old=NULL
for(line in rep.MA.old){
  for(chrm in 1:16){
    yl.MA.old <- c(yl.MA.old,median(log2(getChrmRatio.MA.old(line,chrm)[,3])))
  }
}

length(rep.MA.old)
warnings()
logMat.MA.old <- matrix(yl.MA.old,nrow=16,ncol=15)
colnames(logMat.MA.old) <- rep.MA.old
logMat.MA.old <- data.frame(logMat.MA.old)
chrms <- c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16)
logMat2.MA.old <-cbind(chrms,logMat.MA.old)
# write this output to a file 
write.csv(logMat.MA.old,file="logMat.MAold.csv")

yl.GC=NULL
for(line in rep.GC){
  for(chrm in 1:16){
    yl.GC <- c(yl.GC,median(log2(getChrmRatio.GC(line,chrm)[,3])))
  }
}
warnings()
length(rep.GC)
logMat.GC <- matrix(yl.GC,nrow=16,ncol=16)
colnames(logMat.GC) <- rep.GC
logMat.GC <- data.frame(logMat.GC)
chrms <- c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16)
logMat2.GC <-cbind(chrms,logMat.GC)
# write this output to a file 
write.csv(logMat2.GC,file="logMat.GC.csv")

############################################
##make a matrix of the standard deviations for each line and each chromosome
ysd.MA=NULL
for(line in rep.MAnew){
  for(chrm in 1:16){
    ysd.MA <- c(ysd.MA,sd(log2(getChrmRatio.MAnew(line,chrm)[,3])))
  }
}

logMatsd.MAnew <- matrix(ysd.MA,nrow=16,ncol=7)
colnames(logMatsd.MAnew) <- rep.MAnew
logMatsd.MAnew <- data.frame(logMatsd.MAnew)
chrms <- c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16)
logMatsd2.MAnew <-cbind(chrms,logMatsd.MAnew)

ysd.MA.old=NULL
for(line in rep.MA.old){
  for(chrm in 1:16){
    ysd.MA.old <- c(ysd.MA.old,sd(log2(getChrmRatio.MA.old(line,chrm)[,3])))
  }
}

logMatsd.MA.old <- matrix(ysd.MA.old,nrow=16,ncol=15)
colnames(logMatsd.MA.old) <- rep.MA.old
logMatsd.MA.old <- data.frame(logMatsd.MA.old)
chrms <- c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16)
logMatsd2.MA.old <-cbind(chrms,logMatsd.MA.old)

ysd.GC=NULL
for(line in rep.GC){
  for(chrm in 1:16){
    ysd.GC <- c(ysd.GC,sd(log2(getChrmRatio.GC(line,chrm)[,3])))
  }
}
logMatsd.GC <- matrix(ysd.GC,nrow=16,ncol=16)
colnames(logMatsd.GC) <- rep.GC
logMatsd.GC <- data.frame(logMatsd.GC)
chrms <- c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16)
logMatsd2.GC <-cbind(chrms,logMatsd.GC)


#######################################################################################
#log2 ratios for individual genes 

getFPKMRatio.MAnew <- function(line) { #line must be a 3 digit character
  desAvg.MA <- getAvgFPKM.MAnew(line) #first call avg fpkm script, which returns a 3 column matrix
  ratioByGene.MA <- desAvg.MA[,3]/ancAvg.MA[,3]
  rat.MA <- cbind(sc_des.MAnew[,1:2],ratioByGene.MA) #add gene and chrm values
  colnames(rat.MA) <- c("chr","tracking_id","ratio")
  return(rat.MA)
}

getFPKMRatio.MA.old <- function(line) { #line must be a 3 digit character
  desAvg.MA.old <- getAvgFPKM.MA.old(line) #first call avg fpkm script, which returns a 3 column matrix
  ratioByGene.MA.old <- desAvg.MA.old[,3]/ancAvg.MA.old[,3]
  rat.MA.old <- cbind(sc_des.MA_old[,1:2],ratioByGene.MA.old) #add gene and chrm values
  colnames(rat.MA.old) <- c("chr","tracking_id","ratio")
  return(rat.MA.old)
}


getFPKMRatio.GC <- function(line) { #line must be a 3 digit character
  desAvg.GC <- getAvgFPKM.GC(line) #first call avg fpkm script, which returns a 3 column matrix
  ratioByGene.GC <- desAvg.GC[,3]/ancAvg.GC[,3]
  rat.GC <- cbind(sc_des.GC[,1:2],ratioByGene.GC) #add gene and chrm values
  colnames(rat.GC) <- c("chr","tracking_id","ratio")
  return(rat.GC)
}


```


Functions for graphs of log2 ratio per chromosome in boxplot format 
```{r graph:line log2 ratio per chr,echo=FALSE, include = TRUE}

######################################################################################
#do this for the ancestor too since I have no idea if it is actually euploid...what if it isn't???
par(mar=c(5, 4, 4, 2) + 0.1)
par(mfrow=c(1,1))
#line 112
X112 <- subset(logMat.MAnew, select=c("X112"))
#get the standard deviations for each chromosome for line 112 only
X112.sd <- subset(logMatsd2.MAnew, select=c("X112"))
#merge the stdev with the mean and the chromosome number
X112.all <- cbind(chrms,X112,X112.sd)
#change the column names appropriately
colnames(X112.all)[1] <- "chr"; colnames(X112.all)[2] <- "mean";colnames(X112.all)[3] <- "sd"

#plots the chromosomes with error bars for each one 
{pdf("X112_chroms.pdf")
with (
  data = X112.all
  , expr = errbar(main="MA Line 112",chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a title to the graph
title(main="Lab Line 112")
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()
}


par(mar=c(5, 4, 4, 2) + 0.1)
#get the standard deviations for each chromosome for line 112 only
logMat.MA.old
X108 <- subset(logMat.MA.old, select=c("X108"))
X108.sd <- subset(logMatsd2.MA.old, select=c("X108"))
#merge the stdev with the mean and the chromosome number
X108.all <- cbind(chrms,X108,X108.sd)
#change the column names appropriately
colnames(X108.all)[1] <- "chr"; colnames(X108.all)[2] <- "mean";colnames(X108.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X108_chroms.pdf")
with (
  data = X108.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 108")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

X088 <- subset(logMat.MA.old, select=c("X088"))
X088.sd <- subset(logMatsd2.MA.old, select=c("X088"))
#merge the stdev with the mean and the chromosome number
X088.all <- cbind(chrms,X088,X088.sd)
#change the column names appropriately
colnames(X088.all)[1] <- "chr"; colnames(X088.all)[2] <- "mean";colnames(X088.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X088_chroms.pdf")
with (
  data = X088.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 88")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#MA old line 1
X001 <- subset(logMat.MA.old, select=c("X001"))
X001.sd <- subset(logMatsd2.MA.old, select=c("X001"))
#merge the stdev with the mean and the chromosome number
X001.all <- cbind(chrms,X001,X001.sd)
#change the column names appropriately
colnames(X001.all)[1] <- "chr"; colnames(X001.all)[2] <- "mean";colnames(X001.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X001_chroms.pdf")
with (
  data = X001.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 1")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#MA old line 2
X002 <- subset(logMat.MA.old, select=c("X002"))
X002.sd <- subset(logMatsd2.MA.old, select=c("X002"))
#merge the stdev with the mean and the chromosome number
X002.all <- cbind(chrms,X002,X002.sd)
#change the column names appropriately
colnames(X002.all)[1] <- "chr"; colnames(X002.all)[2] <- "mean";colnames(X002.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X002_chroms.pdf")
with (
  data = X002.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 2")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#MA old line 3
X003 <- subset(logMat.MA.old, select=c("X003"))
X003.sd <- subset(logMatsd2.MA.old, select=c("X003"))
#merge the stdev with the mean and the chromosome number
X003.all <- cbind(chrms,X003,X003.sd)
#change the column names appropriately
colnames(X003.all)[1] <- "chr"; colnames(X003.all)[2] <- "mean";colnames(X003.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X003_chroms.pdf")
with (
  data = X003.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 3")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}


#MA old line 4
X004 <- subset(logMat.MA.old, select=c("X004"))
X004.sd <- subset(logMatsd2.MA.old, select=c("X004"))
#merge the stdev with the mean and the chromosome number
X004.all <- cbind(chrms,X004,X004.sd)
#change the column names appropriately
colnames(X004.all)[1] <- "chr"; colnames(X004.all)[2] <- "mean";colnames(X004.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X004_chroms.pdf")
with (
  data = X004.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 4")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}


#MA old line 5
X005 <- subset(logMat.MA.old, select=c("X005"))
X005.sd <- subset(logMatsd2.MA.old, select=c("X005"))
#merge the stdev with the mean and the chromosome number
X005.all <- cbind(chrms,X005,X005.sd)
#change the column names appropriately
colnames(X005.all)[1] <- "chr"; colnames(X005.all)[2] <- "mean";colnames(X005.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X005_chroms.pdf")
with (
  data = X005.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 5")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}


#MA old line 6
X006 <- subset(logMat.MA.old, select=c("X006"))
X006.sd <- subset(logMatsd2.MA.old, select=c("X006"))
#merge the stdev with the mean and the chromosome number
X006.all <- cbind(chrms,X006,X006.sd)
#change the column names appropriately
colnames(X006.all)[1] <- "chr"; colnames(X006.all)[2] <- "mean";colnames(X006.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X006_chroms.pdf")
with (
  data = X006.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 6")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#MA old line 7
X007 <- subset(logMat.MA.old, select=c("X007"))
X007.sd <- subset(logMatsd2.MA.old, select=c("X007"))
#merge the stdev with the mean and the chromosome number
X007.all <- cbind(chrms,X007,X007.sd)
#change the column names appropriately
colnames(X007.all)[1] <- "chr"; colnames(X007.all)[2] <- "mean";colnames(X007.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X007_chroms.pdf")
with (
  data = X007.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 7")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#MA old line 8

X008 <- subset(logMat.MA.old, select=c("X008"))
X008.sd <- subset(logMatsd2.MA.old, select=c("X008"))
#merge the stdev with the mean and the chromosome number
X008.all <- cbind(chrms,X008,X008.sd)
#change the column names appropriately
colnames(X008.all)[1] <- "chr"; colnames(X008.all)[2] <- "mean";colnames(X008.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X008_chroms.pdf")
with (
  data = X008.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 8")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#MA old line 1
X009 <- subset(logMat.MA.old, select=c("X009"))
X009.sd <- subset(logMatsd2.MA.old, select=c("X009"))
#merge the stdev with the mean and the chromosome number
X009.all <- cbind(chrms,X009,X009.sd)
#change the column names appropriately
colnames(X009.all)[1] <- "chr"; colnames(X009.all)[2] <- "mean";colnames(X009.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X009_chroms.pdf")
with (
  data = X009.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 9")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}


#MA old line 11
X011 <- subset(logMat.MA.old, select=c("X011"))
X011.sd <- subset(logMatsd2.MA.old, select=c("X011"))
#merge the stdev with the mean and the chromosome number
X011.all <- cbind(chrms,X011,X011.sd)
#change the column names appropriately
colnames(X011.all)[1] <- "chr"; colnames(X011.all)[2] <- "mean";colnames(X011.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X011_chroms.pdf")
with (
  data = X011.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 11")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}


#MA old line 15
X015 <- subset(logMat.MA.old, select=c("X015"))
X015.sd <- subset(logMatsd2.MA.old, select=c("X015"))
#merge the stdev with the mean and the chromosome number
X015.all <- cbind(chrms,X015,X015.sd)
#change the column names appropriately
colnames(X015.all)[1] <- "chr"; colnames(X015.all)[2] <- "mean";colnames(X015.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X015_chroms.pdf")
with (
  data = X015.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 15")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}


#MA old line 28
X028 <- subset(logMat.MA.old, select=c("X028"))
X028.sd <- subset(logMatsd2.MA.old, select=c("X028"))
#merge the stdev with the mean and the chromosome number
X028.all <- cbind(chrms,X028,X028.sd)
#change the column names appropriately
colnames(X028.all)[1] <- "chr"; colnames(X028.all)[2] <- "mean";colnames(X028.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X028_chroms.pdf")
with (
  data = X028.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 28")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#MA old line 1
X119 <- subset(logMat.MA.old, select=c("X119"))
X119.sd <- subset(logMatsd2.MA.old, select=c("X119"))
#merge the stdev with the mean and the chromosome number
X119.all <- cbind(chrms,X119,X119.sd)
#change the column names appropriately
colnames(X119.all)[1] <- "chr"; colnames(X119.all)[2] <- "mean";colnames(X119.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X119_chroms.pdf")
with (
  data = X119.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 119")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 115
par(mar=c(5, 4, 4, 2) + 0.1)
#get the standard deviations for each chromosome for line 112 only
X115 <- subset(logMat.MAnew, select=c("X115"))
X115.sd <- subset(logMatsd2.MAnew, select=c("X115"))
#merge the stdev with the mean and the chromosome number
X115.all <- cbind(chrms,X115,X115.sd)
#change the column names appropriately
colnames(X115.all)[1] <- "chr"; colnames(X115.all)[2] <- "mean";colnames(X115.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X115_chroms.pdf")
with (
  data = X115.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 115")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}
#line 117
#get the standard deviations for each chromosome for line 112 only
X117 <- subset(logMat.MAnew, select=c("X117"))
X117.sd <- subset(logMatsd2.MAnew, select=c("X117"))
#merge the stdev with the mean and the chromosome number
X117.all <- cbind(chrms,X117,X117.sd)
#change the column names appropriately
colnames(X117.all)[1] <- "chr"; colnames(X117.all)[2] <- "mean";colnames(X117.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X117_chroms.pdf")
with (
  data = X117.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 117")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 11 
##### LINE 21 AND 11 MIGHT BE MIXED UP STILL FROM LAST TIMEEEEE ##### *****
View(logMat.GC)
X11 <- subset(logMat.GC, select=c("X11"))
X11.sd <- subset(logMatsd2.GC, select=c("X11"))
#merge the stdev with the mean and the chromosome number
X11.all <- cbind(chrms,X11,X11.sd)
#change the column names appropriately
colnames(X11.all)[1] <- "chr"; colnames(X11.all)[2] <- "mean";colnames(X11.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X11_chroms.pdf")
with (
  data = X11.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 11")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 123
X123 <- subset(logMat.MAnew, select=c("X123"))
X123.sd <- subset(logMatsd2.MAnew, select=c("X123"))
#merge the stdev with the mean and the chromosome number
X123.all <- cbind(chrms,X123,X123.sd)
#change the column names appropriately
colnames(X123.all)[1] <- "chr"; colnames(X123.all)[2] <- "mean";colnames(X123.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X123_chroms.pdf")
with (
  data = X123.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 123")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}


#line 152
X152 <- subset(logMat.MAnew, select=c("X152"))
X152.sd <- subset(logMatsd2.MAnew, select=c("X152"))
#merge the stdev with the mean and the chromosome number
X152.all <- cbind(chrms,X152,X152.sd)
#change the column names appropriately
colnames(X152.all)[1] <- "chr"; colnames(X152.all)[2] <- "mean";colnames(X152.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X152_chroms.pdf")
with (
  data = X152.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 152")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 18
X18 <- subset(logMat.GC, select=c("X18"))
X18.sd <- subset(logMatsd2.GC, select=c("X18"))
#merge the stdev with the mean and the chromosome number
X18.all <- cbind(chrms,X18,X18.sd)
#change the column names appropriately
colnames(X18.all)[1] <- "chr"; colnames(X18.all)[2] <- "mean";colnames(X18.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X18_chroms.pdf")
with (
  data = X18.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 18")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 1
X1 <- subset(logMat.GC, select=c("X1"))
X1.sd <- subset(logMatsd2.GC, select=c("X1"))
#merge the stdev with the mean and the chromosome number
X1.all <- cbind(chrms,X1,X1.sd)
#change the column names appropriately
colnames(X1.all)[1] <- "chr"; colnames(X1.all)[2] <- "mean";colnames(X1.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X1_chroms.pdf")
with (
  data = X1.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 1")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}


#line 29
X29 <- subset(logMat.MAnew, select=c("X29"))
X29.sd <- subset(logMatsd2.MAnew, select=c("X29"))
#merge the stdev with the mean and the chromosome number
X29.all <- cbind(chrms,X29,X29.sd)
#change the column names appropriately
colnames(X29.all)[1] <- "chr"; colnames(X29.all)[2] <- "mean";colnames(X29.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X29_chroms.pdf")
with (
  data = X29.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Lab Line 29")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 2
X2 <- subset(logMat.GC, select=c("X2"))
X2.sd <- subset(logMatsd2.GC, select=c("X2"))
#merge the stdev with the mean and the chromosome number
X2.all <- cbind(chrms,X2,X2.sd)
#change the column names appropriately
colnames(X2.all)[1] <- "chr"; colnames(X2.all)[2] <- "mean";colnames(X2.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X2_chroms.pdf")
with (
  data = X2.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 2")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}


#line 3
X3 <- subset(logMat.GC, select=c("X3"))
X3.sd <- subset(logMatsd2.GC, select=c("X3"))
#merge the stdev with the mean and the chromosome number
X3.all <- cbind(chrms,X3,X3.sd)
#change the column names appropriately
colnames(X3.all)[1] <- "chr"; colnames(X3.all)[2] <- "mean";colnames(X3.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X3_chroms.pdf")
with (
  data = X3.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 3")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 49
X49 <- subset(logMat.GC, select=c("X49"))
X49.sd <- subset(logMatsd2.GC, select=c("X49"))
#merge the stdev with the mean and the chromosome number
X49.all <- cbind(chrms,X49,X49.sd)
#change the column names appropriately
colnames(X49.all)[1] <- "chr"; colnames(X49.all)[2] <- "mean";colnames(X49.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X49_chroms.pdf")
with (
  data = X49.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 49")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 4
X4 <- subset(logMat.GC, select=c("X4"))
X4.sd <- subset(logMatsd2.GC, select=c("X4"))
#merge the stdev with the mean and the chromosome number
X4.all <- cbind(chrms,X4,X4.sd)
#change the column names appropriately
colnames(X4.all)[1] <- "chr"; colnames(X4.all)[2] <- "mean";colnames(X4.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X4_chroms.pdf")
with (
  data = X4.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 4")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 50
X50 <- subset(logMat.MAnew, select=c("X50"))
X50.sd <- subset(logMatsd2.MAnew, select=c("X50"))
#merge the stdev with the mean and the chromosome number
X50.all <- cbind(chrms,X50,X50.sd)
#change the column names appropriately
colnames(X50.all)[1] <- "chr"; colnames(X50.all)[2] <- "mean";colnames(X50.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X50_chroms.pdf")
with (
  data = X50.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 50")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 59
X59 <- subset(logMat.GC, select=c("X59"))
X59.sd <- subset(logMatsd2.GC, select=c("X59"))
#merge the stdev with the mean and the chromosome number
X59.all <- cbind(chrms,X59,X59.sd)
#change the column names appropriately
colnames(X59.all)[1] <- "chr"; colnames(X59.all)[2] <- "mean";colnames(X59.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X59_chroms.pdf")
with (
  data = X59.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 59")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 5
X5 <- subset(logMat.GC, select=c("X5"))
X5.sd <- subset(logMatsd2.GC, select=c("X5"))
#merge the stdev with the mean and the chromosome number
X5.all <- cbind(chrms,X5,X5.sd)
#change the column names appropriately
colnames(X5.all)[1] <- "chr"; colnames(X5.all)[2] <- "mean";colnames(X5.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X5_chroms.pdf")
with (
  data = X5.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 5")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 61
X61 <- subset(logMat.GC, select=c("X61"))
X61.sd <- subset(logMatsd2.GC, select=c("X61"))
#merge the stdev with the mean and the chromosome number
X61.all <- cbind(chrms,X61,X61.sd)
#change the column names appropriately
colnames(X61.all)[1] <- "chr"; colnames(X61.all)[2] <- "mean";colnames(X61.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X61_chroms.pdf")
with (
  data = X61.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)

#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 61")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}


#line 69
X69 <- subset(logMat.GC, select=c("X69"))
X69.sd <- subset(logMatsd2.GC, select=c("X69"))
#merge the stdev with the mean and the chromosome number
X69.all <- cbind(chrms,X69,X69.sd)
#change the column names appropriately
colnames(X69.all)[1] <- "chr"; colnames(X69.all)[2] <- "mean";colnames(X69.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X68_chroms.pdf")
with (
  data = X69.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)
#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 68")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 76
X76 <- subset(logMat.GC, select=c("X76"))
X76.sd <- subset(logMatsd2.GC, select=c("X76"))
#merge the stdev with the mean and the chromosome number
X76.all <- cbind(chrms,X76,X76.sd)
#change the column names appropriately
colnames(X76.all)[1] <- "chr"; colnames(X76.all)[2] <- "mean";colnames(X76.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X76_chroms.pdf")
with (
  data = X76.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)
#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 76")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 77
X77 <- subset(logMat.GC, select=c("X77"))
X77.sd <- subset(logMatsd2.GC, select=c("X77"))
#merge the stdev with the mean and the chromosome number
X77.all <- cbind(chrms,X77,X77.sd)
#change the column names appropriately
colnames(X77.all)[1] <- "chr"; colnames(X77.all)[2] <- "mean";colnames(X77.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X77_chroms.pdf")
with (
  data = X77.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)
#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 77")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 7
X7 <- subset(logMat.GC, select=c("X7"))
X7.sd <- subset(logMatsd2.GC, select=c("X7"))
#merge the stdev with the mean and the chromosome number
X7.all <- cbind(chrms,X7,X7.sd)
#change the column names appropriately
colnames(X7.all)[1] <- "chr"; colnames(X7.all)[2] <- "mean";colnames(X7.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X7_chroms.pdf")
with (
  data = X7.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)
#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 7")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 8
X8 <- subset(logMat.GC, select=c("X8"))
X8.sd <- subset(logMatsd2.GC, select=c("X8"))
#merge the stdev with the mean and the chromosome number
X8.all <- cbind(chrms,X8,X8.sd)
#change the column names appropriately
colnames(X8.all)[1] <- "chr"; colnames(X8.all)[2] <- "mean";colnames(X8.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X8_chroms.pdf")
with (
  data = X8.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)
#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 8")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

#line 9
X9 <- subset(logMat.GC, select=c("X9"))
X9.sd <- subset(logMatsd2.GC, select=c("X9"))
#merge the stdev with the mean and the chromosome number
X9.all <- cbind(chrms,X9,X9.sd)
#change the column names appropriately
colnames(X9.all)[1] <- "chr"; colnames(X9.all)[2] <- "mean";colnames(X9.all)[3] <- "sd"
#plots the chromosomes with error bars for each one 
{pdf("X9_chroms.pdf")
with (
  data = X9.all
  , expr = errbar(chr, mean, mean+sd, mean-sd, add=F, pch=16, cap=.015, errbar.col = "mediumvioletred",col="mediumvioletred",xlab="Chromosome",ylab="log2(fold change)",cex.axis=1,las=3,xaxt="n",
                  ylim=c(-2,2))
)
#adds a line at y=0 
abline(h=0,lty=2)
abline(h=0.585,lty=3)
abline(h=-1,lty=3)
#adds a title to the graph
title(main="Hybrid Line 9")
##makes it so that the x-axis is all of the chromosomes and none are missing 
labs <-c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,1:16)
axis(side=1, at=1:16, labels=labs[0:16],cex.axis=0.6)
dev.off()}

```

Codes to get ratio files for each individual line and chromosome
```{r getRatioFiles, echo=FALSE,include=TRUE}

l002.1 <- getChrmRatio.GC("2",1)
l003.1 <- getChrmRatio.GC("3",1)
l005.1 <- getChrmRatio.GC("5",1)
l008.1 <- getChrmRatio.GC("8",1)
l009.1 <- getChrmRatio.GC("9",1)
l011.1 <- getChrmRatio.GC("11",1)
l018.1 <- getChrmRatio.GC("18",1)
l001.1 <- getChrmRatio.GC("1",1)
l049.1 <- getChrmRatio.GC("49",1)
l004.1 <- getChrmRatio.GC("4",1)
l059.1 <- getChrmRatio.GC("59",1)
l061.1 <- getChrmRatio.GC("61",1)
l069.1 <- getChrmRatio.GC("69",1)
l076.1 <- getChrmRatio.GC("76",1)
l077.1 <- getChrmRatio.GC("77",1)
l007.1 <- getChrmRatio.GC("7",1)


l001.1.old <- getChrmRatio.MA.old("001",1)
l002.1.old <- getChrmRatio.MA.old("002",1)
l003.1.old <- getChrmRatio.MA.old("003",1)
l004.1.old <- getChrmRatio.MA.old("004",1)
l005.1.old <- getChrmRatio.MA.old("005",1)
l006.1.old <- getChrmRatio.MA.old("006",1)
l007.1.old <- getChrmRatio.MA.old("007",1)
l008.1.old <- getChrmRatio.MA.old("008",1)
l009.1.old <- getChrmRatio.MA.old("009",1)
l011.1.old <- getChrmRatio.MA.old("011",1)
l015.1.old <- getChrmRatio.MA.old("015",1)
l028.1.old <- getChrmRatio.MA.old("028",1)
l088.1.old <- getChrmRatio.MA.old("088",1)
l108.1.old <- getChrmRatio.MA.old("108",1)
l119.1.old <- getChrmRatio.MA.old("119",1)

l001.2.old <- getChrmRatio.MA.old("001",2)
l002.2.old <- getChrmRatio.MA.old("002",2)
l003.2.old <- getChrmRatio.MA.old("003",2)
l004.2.old <- getChrmRatio.MA.old("004",2)
l005.2.old <- getChrmRatio.MA.old("005",2)
l006.2.old <- getChrmRatio.MA.old("006",2)
l007.2.old <- getChrmRatio.MA.old("007",2)
l008.2.old <- getChrmRatio.MA.old("008",2)
l009.2.old <- getChrmRatio.MA.old("009",2)
l011.2.old <- getChrmRatio.MA.old("011",2)
l015.2.old <- getChrmRatio.MA.old("015",2)
l028.2.old <- getChrmRatio.MA.old("028",2)
l088.2.old <- getChrmRatio.MA.old("088",2)
l108.2.old <- getChrmRatio.MA.old("108",2)
l119.2.old <- getChrmRatio.MA.old("119",2)

l001.3.old <- getChrmRatio.MA.old("001",3)
l002.3.old <- getChrmRatio.MA.old("002",3)
l003.3.old <- getChrmRatio.MA.old("003",3)
l004.3.old <- getChrmRatio.MA.old("004",3)
l005.3.old <- getChrmRatio.MA.old("005",3)
l006.3.old <- getChrmRatio.MA.old("006",3)
l007.3.old <- getChrmRatio.MA.old("007",3)
l008.3.old <- getChrmRatio.MA.old("008",3)
l009.3.old <- getChrmRatio.MA.old("009",3)
l011.3.old <- getChrmRatio.MA.old("011",3)
l015.3.old <- getChrmRatio.MA.old("015",3)
l028.3.old <- getChrmRatio.MA.old("028",3)
l088.3.old <- getChrmRatio.MA.old("088",3)
l108.3.old <- getChrmRatio.MA.old("108",3)
l119.3.old <- getChrmRatio.MA.old("119",3)

l001.4.old <- getChrmRatio.MA.old("001",4)
l002.4.old <- getChrmRatio.MA.old("002",4)
l003.4.old <- getChrmRatio.MA.old("003",4)
l004.4.old <- getChrmRatio.MA.old("004",4)
l005.4.old <- getChrmRatio.MA.old("005",4)
l006.4.old <- getChrmRatio.MA.old("006",4)
l007.4.old <- getChrmRatio.MA.old("007",4)
l008.4.old <- getChrmRatio.MA.old("008",4)
l009.4.old <- getChrmRatio.MA.old("009",4)
l011.4.old <- getChrmRatio.MA.old("011",4)
l015.4.old <- getChrmRatio.MA.old("015",4)
l028.4.old <- getChrmRatio.MA.old("028",4)
l088.4.old <- getChrmRatio.MA.old("088",4)
l108.4.old <- getChrmRatio.MA.old("108",4)
l119.4.old <- getChrmRatio.MA.old("119",4)

l001.5.old <- getChrmRatio.MA.old("001",5)
l002.5.old <- getChrmRatio.MA.old("002",5)
l003.5.old <- getChrmRatio.MA.old("003",5)
l004.5.old <- getChrmRatio.MA.old("004",5)
l005.5.old <- getChrmRatio.MA.old("005",5)
l006.5.old <- getChrmRatio.MA.old("006",5)
l007.5.old <- getChrmRatio.MA.old("007",5)
l008.5.old <- getChrmRatio.MA.old("008",5)
l009.5.old <- getChrmRatio.MA.old("009",5)
l011.5.old <- getChrmRatio.MA.old("011",5)
l015.5.old <- getChrmRatio.MA.old("015",5)
l028.5.old <- getChrmRatio.MA.old("028",5)
l088.5.old <- getChrmRatio.MA.old("088",5)
l108.5.old <- getChrmRatio.MA.old("108",5)
l119.5.old <- getChrmRatio.MA.old("119",5)


l001.6.old <- getChrmRatio.MA.old("001",6)
l002.6.old <- getChrmRatio.MA.old("002",6)
l003.6.old <- getChrmRatio.MA.old("003",6)
l004.6.old <- getChrmRatio.MA.old("004",6)
l005.6.old <- getChrmRatio.MA.old("005",6)
l006.6.old <- getChrmRatio.MA.old("006",6)
l007.6.old <- getChrmRatio.MA.old("007",6)
l008.6.old <- getChrmRatio.MA.old("008",6)
l009.6.old <- getChrmRatio.MA.old("009",6)
l011.6.old <- getChrmRatio.MA.old("011",6)
l015.6.old <- getChrmRatio.MA.old("015",6)
l028.6.old <- getChrmRatio.MA.old("028",6)
l088.6.old <- getChrmRatio.MA.old("088",6)
l108.6.old <- getChrmRatio.MA.old("108",6)
l119.6.old <- getChrmRatio.MA.old("119",6)

l001.7.old <- getChrmRatio.MA.old("001",7)
l002.7.old <- getChrmRatio.MA.old("002",7)
l003.7.old <- getChrmRatio.MA.old("003",7)
l004.7.old <- getChrmRatio.MA.old("004",7)
l005.7.old <- getChrmRatio.MA.old("005",7)
l006.7.old <- getChrmRatio.MA.old("006",7)
l007.7.old <- getChrmRatio.MA.old("007",7)
l008.7.old <- getChrmRatio.MA.old("008",7)
l009.7.old <- getChrmRatio.MA.old("009",7)
l011.7.old <- getChrmRatio.MA.old("011",7)
l015.7.old <- getChrmRatio.MA.old("015",7)
l028.7.old <- getChrmRatio.MA.old("028",7)
l088.7.old <- getChrmRatio.MA.old("088",7)
l108.7.old <- getChrmRatio.MA.old("108",7)
l119.7.old <- getChrmRatio.MA.old("119",7)

l001.8.old <- getChrmRatio.MA.old("001",8)
l002.8.old <- getChrmRatio.MA.old("002",8)
l003.8.old <- getChrmRatio.MA.old("003",8)
l004.8.old <- getChrmRatio.MA.old("004",8)
l005.8.old <- getChrmRatio.MA.old("005",8)
l006.8.old <- getChrmRatio.MA.old("006",8)
l007.8.old <- getChrmRatio.MA.old("007",8)
l008.8.old <- getChrmRatio.MA.old("008",8)
l009.8.old <- getChrmRatio.MA.old("009",8)
l011.8.old <- getChrmRatio.MA.old("011",8)
l015.8.old <- getChrmRatio.MA.old("015",8)
l028.8.old <- getChrmRatio.MA.old("028",8)
l088.8.old <- getChrmRatio.MA.old("088",8)
l108.8.old <- getChrmRatio.MA.old("108",8)
l119.8.old <- getChrmRatio.MA.old("119",8)

l001.9.old <- getChrmRatio.MA.old("001",9)
l002.9.old <- getChrmRatio.MA.old("002",9)
l003.9.old <- getChrmRatio.MA.old("003",9)
l004.9.old <- getChrmRatio.MA.old("004",9)
l005.9.old <- getChrmRatio.MA.old("005",9)
l006.9.old <- getChrmRatio.MA.old("006",9)
l007.9.old <- getChrmRatio.MA.old("007",9)
l008.9.old <- getChrmRatio.MA.old("008",9)
l009.9.old <- getChrmRatio.MA.old("009",9)
l011.9.old <- getChrmRatio.MA.old("011",9)
l015.9.old <- getChrmRatio.MA.old("015",9)
l028.9.old <- getChrmRatio.MA.old("028",9)
l088.9.old <- getChrmRatio.MA.old("088",9)
l108.9.old <- getChrmRatio.MA.old("108",9)
l119.9.old <- getChrmRatio.MA.old("119",9)

l001.10.old <- getChrmRatio.MA.old("001",10)
l002.10.old <- getChrmRatio.MA.old("002",10)
l003.10.old <- getChrmRatio.MA.old("003",10)
l004.10.old <- getChrmRatio.MA.old("004",10)
l005.10.old <- getChrmRatio.MA.old("005",10)
l006.10.old <- getChrmRatio.MA.old("006",10)
l007.10.old <- getChrmRatio.MA.old("007",10)
l008.10.old <- getChrmRatio.MA.old("008",10)
l009.10.old <- getChrmRatio.MA.old("009",10)
l011.10.old <- getChrmRatio.MA.old("011",10)
l015.10.old <- getChrmRatio.MA.old("015",10)
l028.10.old <- getChrmRatio.MA.old("028",10)
l088.10.old <- getChrmRatio.MA.old("088",10)
l108.10.old <- getChrmRatio.MA.old("108",10)
l119.10.old <- getChrmRatio.MA.old("119",10)

l001.11.old <- getChrmRatio.MA.old("001",11)
l002.11.old <- getChrmRatio.MA.old("002",11)
l003.11.old <- getChrmRatio.MA.old("003",11)
l004.11.old <- getChrmRatio.MA.old("004",11)
l005.11.old <- getChrmRatio.MA.old("005",11)
l006.11.old <- getChrmRatio.MA.old("006",11)
l007.11.old <- getChrmRatio.MA.old("007",11)
l008.11.old <- getChrmRatio.MA.old("008",11)
l009.11.old <- getChrmRatio.MA.old("009",11)
l011.11.old <- getChrmRatio.MA.old("011",11)
l015.11.old <- getChrmRatio.MA.old("015",11)
l028.11.old <- getChrmRatio.MA.old("028",11)
l088.11.old <- getChrmRatio.MA.old("088",11)
l108.11.old <- getChrmRatio.MA.old("108",11)
l119.11.old <- getChrmRatio.MA.old("119",11)

l001.12.old <- getChrmRatio.MA.old("001",12)
l002.12.old <- getChrmRatio.MA.old("002",12)
l003.12.old <- getChrmRatio.MA.old("003",12)
l004.12.old <- getChrmRatio.MA.old("004",12)
l005.12.old <- getChrmRatio.MA.old("005",12)
l006.12.old <- getChrmRatio.MA.old("006",12)
l007.12.old <- getChrmRatio.MA.old("007",12)
l008.12.old <- getChrmRatio.MA.old("008",12)
l009.12.old <- getChrmRatio.MA.old("009",12)
l011.12.old <- getChrmRatio.MA.old("011",12)
l015.12.old <- getChrmRatio.MA.old("015",12)
l028.12.old <- getChrmRatio.MA.old("028",12)
l088.12.old <- getChrmRatio.MA.old("088",12)
l108.12.old <- getChrmRatio.MA.old("108",12)
l119.12.old <- getChrmRatio.MA.old("119",12)

l001.13.old <- getChrmRatio.MA.old("001",13)
l002.13.old <- getChrmRatio.MA.old("002",13)
l003.13.old <- getChrmRatio.MA.old("003",13)
l004.13.old <- getChrmRatio.MA.old("004",13)
l005.13.old <- getChrmRatio.MA.old("005",13)
l006.13.old <- getChrmRatio.MA.old("006",13)
l007.13.old <- getChrmRatio.MA.old("007",13)
l008.13.old <- getChrmRatio.MA.old("008",13)
l009.13.old <- getChrmRatio.MA.old("009",13)
l011.13.old <- getChrmRatio.MA.old("011",13)
l015.13.old <- getChrmRatio.MA.old("015",13)
l028.13.old <- getChrmRatio.MA.old("028",13)
l088.13.old <- getChrmRatio.MA.old("088",13)
l108.13.old <- getChrmRatio.MA.old("108",13)
l119.13.old <- getChrmRatio.MA.old("119",13)

l001.14.old <- getChrmRatio.MA.old("001",14)
l002.14.old <- getChrmRatio.MA.old("002",14)
l003.14.old <- getChrmRatio.MA.old("003",14)
l004.14.old <- getChrmRatio.MA.old("004",14)
l005.14.old <- getChrmRatio.MA.old("005",14)
l006.14.old <- getChrmRatio.MA.old("006",14)
l007.14.old <- getChrmRatio.MA.old("007",14)
l008.14.old <- getChrmRatio.MA.old("008",14)
l009.14.old <- getChrmRatio.MA.old("009",14)
l011.14.old <- getChrmRatio.MA.old("011",14)
l015.14.old <- getChrmRatio.MA.old("015",14)
l028.14.old <- getChrmRatio.MA.old("028",14)
l088.14.old <- getChrmRatio.MA.old("088",14)
l108.14.old <- getChrmRatio.MA.old("108",14)
l119.14.old <- getChrmRatio.MA.old("119",14)

l001.15.old <- getChrmRatio.MA.old("001",15)
l002.15.old <- getChrmRatio.MA.old("002",15)
l003.15.old <- getChrmRatio.MA.old("003",15)
l004.15.old <- getChrmRatio.MA.old("004",15)
l005.15.old <- getChrmRatio.MA.old("005",15)
l006.15.old <- getChrmRatio.MA.old("006",15)
l007.15.old <- getChrmRatio.MA.old("007",15)
l008.15.old <- getChrmRatio.MA.old("008",15)
l009.15.old <- getChrmRatio.MA.old("009",15)
l011.15.old <- getChrmRatio.MA.old("011",15)
l015.15.old <- getChrmRatio.MA.old("015",15)
l028.15.old <- getChrmRatio.MA.old("028",15)
l088.15.old <- getChrmRatio.MA.old("088",15)
l108.15.old <- getChrmRatio.MA.old("108",15)
l119.15.old <- getChrmRatio.MA.old("119",15)

l001.16.old <- getChrmRatio.MA.old("001",16)
l002.16.old <- getChrmRatio.MA.old("002",16)
l003.16.old <- getChrmRatio.MA.old("003",16)
l004.16.old <- getChrmRatio.MA.old("004",16)
l005.16.old <- getChrmRatio.MA.old("005",16)
l006.16.old <- getChrmRatio.MA.old("006",16)
l007.16.old <- getChrmRatio.MA.old("007",16)
l008.16.old <- getChrmRatio.MA.old("008",16)
l009.16.old <- getChrmRatio.MA.old("009",16)
l011.16.old <- getChrmRatio.MA.old("011",16)
l015.16.old <- getChrmRatio.MA.old("015",16)
l028.16.old <- getChrmRatio.MA.old("028",16)
l088.16.old <- getChrmRatio.MA.old("088",16)
l108.16.old <- getChrmRatio.MA.old("108",16)
l119.16.old <- getChrmRatio.MA.old("119",16)



l002.1 <- getChrmRatio.GC("2",1)
l003.1 <- getChrmRatio.GC("3",1)
l005.1 <- getChrmRatio.GC("5",1)
l008.1 <- getChrmRatio.GC("8",1)
l009.1 <- getChrmRatio.GC("9",1)
l011.1 <- getChrmRatio.GC("11",1)
l018.1 <- getChrmRatio.GC("18",1)
l001.1 <- getChrmRatio.GC("1",1)
l049.1 <- getChrmRatio.GC("49",1)
l004.1 <- getChrmRatio.GC("4",1)
l059.1 <- getChrmRatio.GC("59",1)
l061.1 <- getChrmRatio.GC("61",1)
l069.1 <- getChrmRatio.GC("69",1)
l076.1 <- getChrmRatio.GC("76",1)
l077.1 <- getChrmRatio.GC("77",1)
l007.1 <- getChrmRatio.GC("7",1)


l029.1 <- getChrmRatio.MAnew("29",1)
l050.1 <- getChrmRatio.MAnew("50",1)
l112.1 <- getChrmRatio.MAnew("112",1)
l115.1 <- getChrmRatio.MAnew("115",1)
l117.1 <- getChrmRatio.MAnew("117",1)
l123.1 <- getChrmRatio.MAnew("123",1)
l152.1 <- getChrmRatio.MAnew("152",1)

l002.2 <- getChrmRatio.GC("2",2)
l003.2 <- getChrmRatio.GC("3",2)
l005.2 <- getChrmRatio.GC("5",2)
l008.2 <- getChrmRatio.GC("8",2)
l009.2 <- getChrmRatio.GC("9",2)
l011.2 <- getChrmRatio.GC("11",2)
l018.2 <- getChrmRatio.GC("18",2)
l001.2 <- getChrmRatio.GC("1",2)
l049.2 <- getChrmRatio.GC("49",2)
l004.2 <- getChrmRatio.GC("4",2)
l059.2 <- getChrmRatio.GC("59",2)
l061.2 <- getChrmRatio.GC("61",2)
l069.2 <- getChrmRatio.GC("69",2)
l076.2 <- getChrmRatio.GC("76",2)
l077.2 <- getChrmRatio.GC("77",2)
l007.2 <- getChrmRatio.GC("7",2)

l029.2 <- getChrmRatio.MAnew("29",2)
l050.2 <- getChrmRatio.MAnew("50",2)
l112.2 <- getChrmRatio.MAnew("112",2)
l115.2 <- getChrmRatio.MAnew("115",2)
l117.2 <- getChrmRatio.MAnew("117",2)
l123.2 <- getChrmRatio.MAnew("123",2)
l152.2 <- getChrmRatio.MAnew("152",2)

l002.3 <- getChrmRatio.GC("2",3)
l003.3 <- getChrmRatio.GC("3",3)
l005.3 <- getChrmRatio.GC("5",3)
l008.3 <- getChrmRatio.GC("8",3)
l009.3 <- getChrmRatio.GC("9",3)
l011.3 <- getChrmRatio.GC("11",3)
l018.3 <- getChrmRatio.GC("18",3)
l001.3 <- getChrmRatio.GC("1",3)
l049.3 <- getChrmRatio.GC("49",3)
l004.3 <- getChrmRatio.GC("4",3)
l059.3 <- getChrmRatio.GC("59",3)
l061.3 <- getChrmRatio.GC("61",3)
l069.3 <- getChrmRatio.GC("69",3)
l076.3 <- getChrmRatio.GC("76",3)
l077.3 <- getChrmRatio.GC("77",3)
l007.3 <- getChrmRatio.GC("7",3)

l029.3 <- getChrmRatio.MAnew("29",3)
l050.3 <- getChrmRatio.MAnew("50",3)
l112.3 <- getChrmRatio.MAnew("112",3)
l115.3 <- getChrmRatio.MAnew("115",3)
l117.3 <- getChrmRatio.MAnew("117",3)
l123.3 <- getChrmRatio.MAnew("123",3)
l152.3 <- getChrmRatio.MAnew("152",3)

l002.4 <- getChrmRatio.GC("2",4)
l003.4 <- getChrmRatio.GC("3",4)
l005.4 <- getChrmRatio.GC("5",4)
l008.4 <- getChrmRatio.GC("8",4)
l009.4 <- getChrmRatio.GC("9",4)
l011.4 <- getChrmRatio.GC("11",4)
l018.4 <- getChrmRatio.GC("18",4)
l001.4 <- getChrmRatio.GC("1",4)
l049.4 <- getChrmRatio.GC("49",4)
l004.4 <- getChrmRatio.GC("4",4)
l059.4 <- getChrmRatio.GC("59",4)
l061.4 <- getChrmRatio.GC("61",4)
l069.4 <- getChrmRatio.GC("69",4)
l076.4 <- getChrmRatio.GC("76",4)
l077.4 <- getChrmRatio.GC("77",4)
l007.4 <- getChrmRatio.GC("7",4)

l029.4 <- getChrmRatio.MAnew("29",4)
l050.4 <- getChrmRatio.MAnew("50",4)
l112.4 <- getChrmRatio.MAnew("112",4)
l115.4 <- getChrmRatio.MAnew("115",4)
l117.4 <- getChrmRatio.MAnew("117",4)
l123.4 <- getChrmRatio.MAnew("123",4)
l152.4 <- getChrmRatio.MAnew("152",4)

l002.5 <- getChrmRatio.GC("2",5)
l003.5 <- getChrmRatio.GC("3",5)
l005.5 <- getChrmRatio.GC("5",5)
l008.5 <- getChrmRatio.GC("8",5)
l009.5 <- getChrmRatio.GC("9",5)
l011.5 <- getChrmRatio.GC("11",5)
l018.5 <- getChrmRatio.GC("18",5)
l001.5 <- getChrmRatio.GC("1",5)
l049.5 <- getChrmRatio.GC("49",5)
l004.5 <- getChrmRatio.GC("4",5)
l059.5 <- getChrmRatio.GC("59",5)
l061.5 <- getChrmRatio.GC("61",5)
l069.5 <- getChrmRatio.GC("69",5)
l076.5 <- getChrmRatio.GC("76",5)
l077.5 <- getChrmRatio.GC("77",5)
l007.5 <- getChrmRatio.GC("7",5)

l029.5 <- getChrmRatio.MAnew("29",5)
l050.5 <- getChrmRatio.MAnew("50",5)
l112.5 <- getChrmRatio.MAnew("112",5)
l115.5 <- getChrmRatio.MAnew("115",5)
l117.5 <- getChrmRatio.MAnew("117",5)
l123.5 <- getChrmRatio.MAnew("123",5)
l152.5 <- getChrmRatio.MAnew("152",5)

l002.6 <- getChrmRatio.GC("2",6)
l003.6 <- getChrmRatio.GC("3",6)
l005.6 <- getChrmRatio.GC("5",6)
l008.6 <- getChrmRatio.GC("8",6)
l009.6 <- getChrmRatio.GC("9",6)
l011.6 <- getChrmRatio.GC("11",6)
l018.6 <- getChrmRatio.GC("18",6)
l001.6 <- getChrmRatio.GC("1",6)
l049.6 <- getChrmRatio.GC("49",6)
l004.6 <- getChrmRatio.GC("4",6)
l059.6 <- getChrmRatio.GC("59",6)
l061.6 <- getChrmRatio.GC("61",6)
l069.6 <- getChrmRatio.GC("69",6)
l076.6 <- getChrmRatio.GC("76",6)
l077.6 <- getChrmRatio.GC("77",6)
l007.6 <- getChrmRatio.GC("7",6)

l029.6 <- getChrmRatio.MAnew("29",6)
l050.6 <- getChrmRatio.MAnew("50",6)
l112.6 <- getChrmRatio.MAnew("112",6)
l115.6 <- getChrmRatio.MAnew("115",6)
l117.6 <- getChrmRatio.MAnew("117",6)
l123.6 <- getChrmRatio.MAnew("123",6)
l152.6 <- getChrmRatio.MAnew("152",6)

l002.7 <- getChrmRatio.GC("2",7)
l003.7 <- getChrmRatio.GC("3",7)
l005.7 <- getChrmRatio.GC("5",7)
l008.7 <- getChrmRatio.GC("8",7)
l009.7 <- getChrmRatio.GC("9",7)
l011.7 <- getChrmRatio.GC("11",7)
l018.7 <- getChrmRatio.GC("18",7)
l001.7 <- getChrmRatio.GC("1",7)
l049.7 <- getChrmRatio.GC("49",7)
l004.7 <- getChrmRatio.GC("4",7)
l059.7 <- getChrmRatio.GC("59",7)
l061.7 <- getChrmRatio.GC("61",7)
l069.7 <- getChrmRatio.GC("69",7)
l076.7 <- getChrmRatio.GC("76",7)
l077.7 <- getChrmRatio.GC("77",7)
l007.7 <- getChrmRatio.GC("7",7)

l029.7 <- getChrmRatio.MAnew("29",7)
l050.7 <- getChrmRatio.MAnew("50",7)
l112.7 <- getChrmRatio.MAnew("112",7)
l115.7 <- getChrmRatio.MAnew("115",7)
l117.7 <- getChrmRatio.MAnew("117",7)
l123.7 <- getChrmRatio.MAnew("123",7)
l152.7 <- getChrmRatio.MAnew("152",7)

l002.8 <- getChrmRatio.GC("2",8)
l003.8 <- getChrmRatio.GC("3",8)
l005.8 <- getChrmRatio.GC("5",8)
l008.8 <- getChrmRatio.GC("8",8)
l009.8 <- getChrmRatio.GC("9",8)
l011.8 <- getChrmRatio.GC("11",8)
l018.8 <- getChrmRatio.GC("18",8)
l001.8 <- getChrmRatio.GC("1",8)
l049.8 <- getChrmRatio.GC("49",8)
l004.8 <- getChrmRatio.GC("4",8)
l059.8 <- getChrmRatio.GC("59",8)
l061.8 <- getChrmRatio.GC("61",8)
l069.8 <- getChrmRatio.GC("69",8)
l076.8 <- getChrmRatio.GC("76",8)
l077.8 <- getChrmRatio.GC("77",8)
l007.8 <- getChrmRatio.GC("7",8)

l029.8 <- getChrmRatio.MAnew("29",8)
l050.8 <- getChrmRatio.MAnew("50",8)
l112.8 <- getChrmRatio.MAnew("112",8)
l115.8 <- getChrmRatio.MAnew("115",8)
l117.8 <- getChrmRatio.MAnew("117",8)
l123.8 <- getChrmRatio.MAnew("123",8)
l152.8 <- getChrmRatio.MAnew("152",8)

l002.9 <- getChrmRatio.GC("2",9)
l003.9 <- getChrmRatio.GC("3",9)
l005.9 <- getChrmRatio.GC("5",9)
l008.9 <- getChrmRatio.GC("8",9)
l009.9 <- getChrmRatio.GC("9",9)
l011.9 <- getChrmRatio.GC("11",9)
l018.9 <- getChrmRatio.GC("18",9)
l001.9 <- getChrmRatio.GC("1",9)
l049.9 <- getChrmRatio.GC("49",9)
l004.9 <- getChrmRatio.GC("4",9)
l059.9 <- getChrmRatio.GC("59",9)
l061.9 <- getChrmRatio.GC("61",9)
l069.9 <- getChrmRatio.GC("69",9)
l076.9 <- getChrmRatio.GC("76",9)
l077.9 <- getChrmRatio.GC("77",9)
l007.9 <- getChrmRatio.GC("7",9)

l029.9 <- getChrmRatio.MAnew("29",9)
l050.9 <- getChrmRatio.MAnew("50",9)
l112.9 <- getChrmRatio.MAnew("112",9)
l115.9 <- getChrmRatio.MAnew("115",9)
l117.9 <- getChrmRatio.MAnew("117",9)
l123.9 <- getChrmRatio.MAnew("123",9)
l152.9 <- getChrmRatio.MAnew("152",9)

l002.10 <- getChrmRatio.GC("2",10)
l003.10 <- getChrmRatio.GC("3",10)
l005.10 <- getChrmRatio.GC("5",10)
l008.10 <- getChrmRatio.GC("8",10)
l009.10 <- getChrmRatio.GC("9",10)
l011.10 <- getChrmRatio.GC("11",10)
l018.10 <- getChrmRatio.GC("18",10)
l001.10 <- getChrmRatio.GC("1",10)
l049.10 <- getChrmRatio.GC("49",10)
l004.10 <- getChrmRatio.GC("4",10)
l059.10 <- getChrmRatio.GC("59",10)
l061.10 <- getChrmRatio.GC("61",10)
l069.10 <- getChrmRatio.GC("69",10)
l076.10 <- getChrmRatio.GC("76",10)
l077.10 <- getChrmRatio.GC("77",10)
l007.10 <- getChrmRatio.GC("7",10)

l029.10 <- getChrmRatio.MAnew("29",10)
l050.10 <- getChrmRatio.MAnew("50",10)
l112.10 <- getChrmRatio.MAnew("112",10)
l115.10 <- getChrmRatio.MAnew("115",10)
l117.10 <- getChrmRatio.MAnew("117",10)
l123.10 <- getChrmRatio.MAnew("123",10)
l152.10 <- getChrmRatio.MAnew("152",10)

l002.11 <- getChrmRatio.GC("2",11)
l003.11 <- getChrmRatio.GC("3",11)
l005.11 <- getChrmRatio.GC("5",11)
l008.11 <- getChrmRatio.GC("8",11)
l009.11 <- getChrmRatio.GC("9",11)
l011.11 <- getChrmRatio.GC("11",11)
l018.11 <- getChrmRatio.GC("18",11)
l001.11 <- getChrmRatio.GC("1",11)
l049.11 <- getChrmRatio.GC("49",11)
l004.11 <- getChrmRatio.GC("4",11)
l059.11 <- getChrmRatio.GC("59",11)
l061.11 <- getChrmRatio.GC("61",11)
l069.11 <- getChrmRatio.GC("69",11)
l076.11 <- getChrmRatio.GC("76",11)
l077.11 <- getChrmRatio.GC("77",11)
l007.11 <- getChrmRatio.GC("7",11)

l029.11 <- getChrmRatio.MAnew("29",11)
l050.11 <- getChrmRatio.MAnew("50",11)
l112.11 <- getChrmRatio.MAnew("112",11)
l115.11 <- getChrmRatio.MAnew("115",11)
l117.11 <- getChrmRatio.MAnew("117",11)
l123.11 <- getChrmRatio.MAnew("123",11)
l152.11 <- getChrmRatio.MAnew("152",11)

l002.12 <- getChrmRatio.GC("2",12)
l003.12 <- getChrmRatio.GC("3",12)
l005.12 <- getChrmRatio.GC("5",12)
l008.12 <- getChrmRatio.GC("8",12)
l009.12 <- getChrmRatio.GC("9",12)
l011.12 <- getChrmRatio.GC("11",12)
l018.12 <- getChrmRatio.GC("18",12)
l001.12 <- getChrmRatio.GC("1",12)
l049.12 <- getChrmRatio.GC("49",12)
l004.12 <- getChrmRatio.GC("4",12)
l059.12 <- getChrmRatio.GC("59",12)
l061.12 <- getChrmRatio.GC("61",12)
l069.12 <- getChrmRatio.GC("69",12)
l076.12 <- getChrmRatio.GC("76",12)
l077.12 <- getChrmRatio.GC("77",12)
l007.12 <- getChrmRatio.GC("7",12)

l029.12 <- getChrmRatio.MAnew("29",12)
l050.12 <- getChrmRatio.MAnew("50",12)
l112.12 <- getChrmRatio.MAnew("112",12)
l115.12 <- getChrmRatio.MAnew("115",12)
l117.12 <- getChrmRatio.MAnew("117",12)
l123.12 <- getChrmRatio.MAnew("123",12)
l152.12 <- getChrmRatio.MAnew("152",12)

l002.13 <- getChrmRatio.GC("2",13)
l003.13 <- getChrmRatio.GC("3",13)
l005.13 <- getChrmRatio.GC("5",13)
l008.13 <- getChrmRatio.GC("8",13)
l009.13 <- getChrmRatio.GC("9",13)
l011.13 <- getChrmRatio.GC("11",13)
l018.13 <- getChrmRatio.GC("18",13)
l001.13 <- getChrmRatio.GC("1",13)
l049.13 <- getChrmRatio.GC("49",13)
l004.13 <- getChrmRatio.GC("4",13)
l059.13 <- getChrmRatio.GC("59",13)
l061.13 <- getChrmRatio.GC("61",13)
l069.13 <- getChrmRatio.GC("69",13)
l076.13 <- getChrmRatio.GC("76",13)
l077.13 <- getChrmRatio.GC("77",13)
l007.13 <- getChrmRatio.GC("7",13)

l029.13 <- getChrmRatio.MAnew("29",13)
l050.13 <- getChrmRatio.MAnew("50",13)
l112.13 <- getChrmRatio.MAnew("112",13)
l115.13 <- getChrmRatio.MAnew("115",13)
l117.13 <- getChrmRatio.MAnew("117",13)
l123.13 <- getChrmRatio.MAnew("123",13)
l152.13 <- getChrmRatio.MAnew("152",13)

l002.14 <- getChrmRatio.GC("2",14)
l003.14 <- getChrmRatio.GC("3",14)
l005.14 <- getChrmRatio.GC("5",14)
l008.14 <- getChrmRatio.GC("8",14)
l009.14 <- getChrmRatio.GC("9",14)
l011.14 <- getChrmRatio.GC("11",14)
l018.14 <- getChrmRatio.GC("18",14)
l001.14 <- getChrmRatio.GC("1",14)
l049.14 <- getChrmRatio.GC("49",14)
l004.14 <- getChrmRatio.GC("4",14)
l059.14 <- getChrmRatio.GC("59",14)
l061.14 <- getChrmRatio.GC("61",14)
l069.14 <- getChrmRatio.GC("69",14)
l076.14 <- getChrmRatio.GC("76",14)
l077.14 <- getChrmRatio.GC("77",14)
l007.14 <- getChrmRatio.GC("7",14)

l029.14 <- getChrmRatio.MAnew("29",14)
l050.14 <- getChrmRatio.MAnew("50",14)
l112.14 <- getChrmRatio.MAnew("112",14)
l115.14 <- getChrmRatio.MAnew("115",14)
l117.14 <- getChrmRatio.MAnew("117",14)
l123.14 <- getChrmRatio.MAnew("123",14)
l152.14 <- getChrmRatio.MAnew("152",14)

l002.15 <- getChrmRatio.GC("2",15)
l003.15 <- getChrmRatio.GC("3",15)
l005.15 <- getChrmRatio.GC("5",15)
l008.15 <- getChrmRatio.GC("8",15)
l009.15 <- getChrmRatio.GC("9",15)
l011.15 <- getChrmRatio.GC("11",15)
l018.15 <- getChrmRatio.GC("18",15)
l001.15 <- getChrmRatio.GC("1",15)
l049.15 <- getChrmRatio.GC("49",15)
l004.15 <- getChrmRatio.GC("4",15)
l059.15 <- getChrmRatio.GC("59",15)
l061.15 <- getChrmRatio.GC("61",15)
l069.15 <- getChrmRatio.GC("69",15)
l076.15 <- getChrmRatio.GC("76",15)
l077.15 <- getChrmRatio.GC("77",15)
l007.15 <- getChrmRatio.GC("7",15)

l029.15 <- getChrmRatio.MAnew("29",15)
l050.15 <- getChrmRatio.MAnew("50",15)
l112.15 <- getChrmRatio.MAnew("112",15)
l115.15 <- getChrmRatio.MAnew("115",15)
l117.15 <- getChrmRatio.MAnew("117",15)
l123.15 <- getChrmRatio.MAnew("123",15)
l152.15 <- getChrmRatio.MAnew("152",15)

l002.16 <- getChrmRatio.GC("2",16)
l003.16 <- getChrmRatio.GC("3",16)
l005.16 <- getChrmRatio.GC("5",16)
l008.16 <- getChrmRatio.GC("8",16)
l009.16 <- getChrmRatio.GC("9",16)
l011.16 <- getChrmRatio.GC("11",16)
l018.16 <- getChrmRatio.GC("18",16)
l001.16 <- getChrmRatio.GC("1",16)
l049.16 <- getChrmRatio.GC("49",16)
l004.16 <- getChrmRatio.GC("4",16)
l059.16 <- getChrmRatio.GC("59",16)
l061.16 <- getChrmRatio.GC("61",16)
l069.16 <- getChrmRatio.GC("69",16)
l076.16 <- getChrmRatio.GC("76",16)
l077.16 <- getChrmRatio.GC("77",16)
l007.16 <- getChrmRatio.GC("7",16)

l029.16 <- getChrmRatio.MAnew("29",16)
l050.16 <- getChrmRatio.MAnew("50",16)
l112.16 <- getChrmRatio.MAnew("112",16)
l115.16 <- getChrmRatio.MAnew("115",16)
l117.16 <- getChrmRatio.MAnew("117",16)
l123.16 <- getChrmRatio.MAnew("123",16)
l152.16 <- getChrmRatio.MAnew("152",16)

#############################################################
#concatenate each line with its other chromosomes 
#then paste each line with all chromosome ratios together into one matrix
str(l001.1)

l1All <- rbind(l001.1,l001.2,l001.3,l001.4,l001.5,l001.6,l001.7,l001.8,l001.9,l001.10,l001.11,l001.12,l001.13,l001.14,l001.15,l001.16)
l2All <- rbind(l002.1,l002.2,l002.3,l002.4,l002.5,l002.6,l002.7,l002.8,l002.9,l002.10,l002.11,l002.12,l002.13,l002.14,l002.15,l002.16)
l3All <- rbind(l003.1,l003.2,l003.3,l003.4,l003.5,l003.6,l003.7,l003.8,l003.9,l003.10,l003.11,l003.12,l003.13,l003.14,l003.15,l003.16)
l4All <- rbind(l004.1,l004.2,l004.3,l004.4,l004.5,l004.6,l004.7,l004.8,l004.9,l004.10,l004.11,l004.12,l004.13,l004.14,l004.15,l004.16)
l5All <- rbind(l005.1,l005.2,l005.3,l005.4,l005.5,l005.6,l005.7,l005.8,l005.9,l005.10,l005.11,l005.12,l005.13,l005.14,l005.15,l005.16)
l7All <- rbind(l007.1,l007.2,l007.3,l007.4,l007.5,l007.6,l007.7,l007.8,l007.9,l007.10,l007.11,l007.12,l007.13,l007.14,l007.15,l007.16)
l8All <- rbind(l008.1,l008.2,l008.3,l008.4,l008.5,l008.6,l008.7,l008.8,l008.9,l008.10,l008.11,l008.12,l008.13,l008.14,l008.15,l008.16)
l9All <- rbind(l009.1,l009.2,l009.3,l009.4,l009.5,l009.6,l009.7,l009.8,l009.9,l009.10,l009.11,l009.12,l009.13,l009.14,l009.15,l009.16)
l11All <- rbind(l011.1,l011.2,l011.3,l011.4,l011.5,l011.6,l011.7,l011.8,l011.9,l011.10,l011.11,l011.12,l011.13,l011.14,l011.15,l011.16)
l18All <- rbind(l018.1,l018.2,l018.3,l018.4,l018.5,l018.6,l018.7,l018.8,l018.9,l018.10,l018.11,l018.12,l018.13,l018.14,l018.15,l018.16)
l49All <- rbind(l049.1,l049.2,l049.3,l049.4,l049.5,l049.6,l049.7,l049.8,l049.9,l049.10,l049.11,l049.12,l049.13,l049.14,l049.15,l049.16)
l59All <- rbind(l059.1,l059.2,l059.3,l059.4,l059.5,l059.6,l059.7,l059.8,l059.9,l059.10,l059.11,l059.12,l059.13,l059.14,l059.15,l059.16)
l61All <- rbind(l061.1,l061.2,l061.3,l061.4,l061.5,l061.6,l061.7,l061.8,l061.9,l061.10,l061.11,l061.12,l061.13,l061.14,l061.15,l061.16)
l69All <- rbind(l069.1,l069.2,l069.3,l069.4,l069.5,l069.6,l069.7,l069.8,l069.9,l069.10,l069.11,l069.12,l069.13,l069.14,l069.15,l069.16)
l76All <- rbind(l076.1,l076.2,l076.3,l076.4,l076.5,l076.6,l076.7,l076.8,l076.9,l076.10,l076.11,l076.12,l076.13,l076.14,l076.15,l076.16)
l77All <- rbind(l077.1,l077.2,l077.3,l077.4,l077.5,l077.6,l077.7,l077.8,l077.9,l077.10,l077.11,l077.12,l077.13,l077.14,l077.15,l077.16)

#MA new

l29All <- rbind(l029.1,l029.2,l029.3,l029.4,l029.5,l029.6,l029.7,l029.8,l029.9,l029.10,l029.11,l029.12,l029.13,l029.14,l029.15,l029.16)
l50All <- rbind(l050.1,l050.2,l050.3,l050.4,l050.5,l050.6,l050.7,l050.8,l050.9,l050.10,l050.11,l050.12,l050.13,l050.14,l050.15,l050.16)
l112All <- rbind(l112.1,l112.2,l112.3,l112.4,l112.5,l112.6,l112.7,l112.8,l112.9,l112.10,l112.11,l112.12,l112.13,l112.14,l112.15,l112.16)
l115All <- rbind(l115.1,l115.2,l115.3,l115.4,l115.5,l115.6,l115.7,l115.8,l115.9,l115.10,l115.11,l115.12,l115.13,l115.14,l115.15,l115.16)
l117All <- rbind(l117.1,l117.2,l117.3,l117.4,l117.5,l117.6,l117.7,l117.8,l117.9,l117.10,l117.11,l117.12,l117.13,l117.14,l117.15,l117.16)
l123All <- rbind(l123.1,l123.2,l123.3,l123.4,l123.5,l123.6,l123.7,l123.8,l123.9,l123.10,l123.11,l123.12,l123.13,l123.14,l123.15,l123.16)
l152All <- rbind(l152.1,l152.2,l152.3,l152.4,l152.5,l152.6,l152.7,l152.8,l152.9,l152.10,l152.11,l152.12,l152.13,l152.14,l152.15,l152.16)

### MA old 
l1Allold <- rbind(l001.1.old,l001.2.old,l001.3.old,l001.4.old,l001.5.old,l001.6.old,l001.7.old,l001.8.old,l001.9.old,l001.10.old,l001.11.old,l001.12.old,l001.13.old,l001.14.old,l001.15.old,l001.16.old)
l2Allold <- rbind(l002.1.old,l002.2.old,l002.3.old,l002.4.old,l002.5.old,l002.6.old,l002.7.old,l002.8.old,l002.9.old,l002.10.old,l002.11.old,l002.12.old,l002.13.old,l002.14.old,l002.15.old,l002.16.old)
l3Allold <- rbind(l003.1.old,l003.2.old,l003.3.old,l003.4.old,l003.5.old,l003.6.old,l003.7.old,l003.8.old,l003.9.old,l003.10.old,l003.11.old,l003.12.old,l003.13.old,l003.14.old,l003.15.old,l003.16.old)
l4Allold <- rbind(l004.1.old,l004.2.old,l004.3.old,l004.4.old,l004.5.old,l004.6.old,l004.7.old,l004.8.old,l004.9.old,l004.10.old,l004.11.old,l004.12.old,l004.13.old,l004.14.old,l004.15.old,l004.16.old)
l5Allold <- rbind(l005.1.old,l005.2.old,l005.3.old,l005.4.old,l005.5.old,l005.6.old,l005.7.old,l005.8.old,l005.9.old,l005.10.old,l005.11.old,l005.12.old,l005.13.old,l005.14.old,l005.15.old,l005.16.old)
l6Allold <- rbind(l006.1.old,l006.2.old,l006.3.old,l006.4.old,l006.5.old,l006.6.old,l006.7.old,l006.8.old,l006.9.old,l006.10.old,l006.11.old,l006.12.old,l006.13.old,l006.14.old,l006.15.old,l006.16.old)
l7Allold <- rbind(l007.1.old,l007.2.old,l007.3.old,l007.4.old,l007.5.old,l007.6.old,l007.7.old,l007.8.old,l007.9.old,l007.10.old,l007.11.old,l007.12.old,l007.13.old,l007.14.old,l007.15.old,l007.16.old)
l8Allold <- rbind(l008.1.old,l008.2.old,l008.3.old,l008.4.old,l008.5.old,l008.6.old,l008.7.old,l008.8.old,l008.9.old,l008.10.old,l008.11.old,l008.12.old,l008.13.old,l008.14.old,l008.15.old,l008.16.old)
l9Allold <- rbind(l009.1.old,l009.2.old,l009.3.old,l009.4.old,l009.5.old,l009.6.old,l009.7.old,l009.8.old,l009.9.old,l009.10.old,l009.11.old,l009.12.old,l009.13.old,l009.14.old,l009.15.old,l009.16.old)
l11Allold <- rbind(l011.1.old,l011.2.old,l011.3.old,l011.4.old,l011.5.old,l011.6.old,l011.7.old,l011.8.old,l011.9.old,l011.10.old,l011.11.old,l011.12.old,l011.13.old,l011.14.old,l011.15.old,l011.16.old)
l15Allold <- rbind(l015.1.old,l015.2.old,l015.3.old,l015.4.old,l015.5.old,l015.6.old,l015.7.old,l015.8.old,l015.9.old,l015.10.old,l015.11.old,l015.12.old,l015.13.old,l015.14.old,l015.15.old,l015.16.old)
l28Allold <- rbind(l028.1.old,l028.2.old,l028.3.old,l028.4.old,l028.5.old,l028.6.old,l028.7.old,l028.8.old,l028.9.old,l028.10.old,l028.11.old,l028.12.old,l028.13.old,l028.14.old,l028.15.old,l028.16.old)
l88Allold <- rbind(l088.1.old,l088.2.old,l088.3.old,l088.4.old,l088.5.old,l088.6.old,l088.7.old,l088.8.old,l088.9.old,l088.10.old,l088.11.old,l088.12.old,l088.13.old,l088.14.old,l088.15.old,l088.16.old)
l108Allold <- rbind(l108.1.old,l108.2.old,l108.3.old,l108.4.old,l108.5.old,l108.6.old,l108.7.old,l108.8.old,l108.9.old,l108.10.old,l108.11.old,l108.12.old,l108.13.old,l108.14.old,l108.15.old,l108.16.old)
l119Allold <- rbind(l119.1.old,l119.2.old,l119.3.old,l119.4.old,l119.5.old,l119.6.old,l119.7.old,l119.8.old,l119.9.old,l119.10.old,l119.11.old,l119.12.old,l119.13.old,l119.14.old,l119.15.old,l119.16.old)

View(l119Allold)
#cbind all of these together into one matrix

Ratios.GC <- cbind(l1All,l2All[,3],l3All[,3],l4All[,3],l5All[,3],l7All[,3],l8All[,3],l9All[,3],l11All[,3],l18All[,3],l49All[,3],l59All[,3],l61All[,3],l69All[,3],l76All[,3],l77All[,3])
dim(Ratios.GC)
Ratios.GC[3:18] <- apply(Ratios.GC[3:18], 1, function(x) log2(x) ) 
View(Ratios.GC)
write.csv(Ratios.GC, file = "log2Ratios.GC.csv")


chr1 <- subset(Ratios.GC, chr==1)

```


Codes for boxplots 
```{r boxplots, include=FALSE, echo=FALSE} 

# Load in data into data frame
# old MA lines
chr1Data.old <- data.frame(y=c((log2(l001.1.old[,3])),(log2(l002.1.old[,3])),(log2(l003.1.old[,3])),(log2(l004.1.old[,3])),(log2(l005.1.old[,3])),
       (log2(l006.1.old[,3])),(log2(l007.1.old[,3])),(log2(l008.1.old[,3])),(log2(l009.1.old[,3])),(log2(l011.1.old[,3])),
      (log2(l015.1.old[,3])),(log2(l028.1.old[,3])),(log2(l088.1.old[,3])),(log2(l108.1.old[,3])),(log2(l119.1.old[,3]))),
Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=(length(l001.1.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2")))

# new GC and MA lines
chr1Data <- data.frame(y=c((log2(l001.1[,3])),(log2(l002.1[,3])),(log2(l003.1[,3])),(log2(l004.1[,3])),(log2(l005.1[,3])),
      (log2(l007.1[,3])),(log2(l008.1[,3])),(log2(l009.1[,3])),(log2(l011.1[,3])),
       (log2(l018.1[,3])),(log2(l049.1[,3])),(log2(l059.1[,3])),
       (log2(l061.1[,3])),(log2(l069.1[,3])),(log2(l076.1[,3])),(log2(l077.1[,3])),(log2(l029.1[,3])),(log2(l050.1[,3])),(log2(l112.1[,3])),(log2(l115.1[,3])),(log2(l117.1[,3])),
        (log2(l123.1[,3])),(log2(l152.1[,3]))),
Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=length(l001.1$tracking_id)),copy_number=rep(c("2","2","2","2","2","3","2","2","1","3","2","2","2","2","2","2","2","2","2","2","2","2","3"),each=length(l001.1$tracking_id)))

# combine the old and the new 
chr1DataCombo <- rbind(chr1Data,chr1Data.old)
# make "Line" a factor
chr1DataCombo$Line <- as.factor(chr1DataCombo$Line)
# make "copy number" a factor
chr1DataCombo$copy_number <- as.factor(chr1DataCombo$copy_number)
# chr 1 has 87 genes on it that I looked at
View(chr1DataCombo)
# set labels for lines
chr1DataCombo$Line=factor(chr1DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))

colors <- c("1" = "blue", "2" = "gray", "2.5" = "pink", "3" = "red", "4" = "darkred")


myTitleTheme <- theme(plot.title = element_text(family = "Helvetica", face = "bold", size = (10)))

myXaxisTheme <- theme(axis.text.x = element_text(family = "Helvetica", face="bold", size = (8),angle = 45, hjust = 1))

myXlabelTheme <- theme(axis.title.x.bottom = element_text(family = "Helvetica", size = (8)))

myYaxisTheme <- theme(axis.text.y = element_text(family = "Helvetica", face="bold", size = (9)))

myYlabelTheme <- theme(axis.title.y.left = element_text(family = "Helvetica", size = (8)))

myLegendTheme <- theme(legend.text = element_text(family = "Helvetica", size = (8)), legend.title = element_text(family = "Helvetica", size = (8)), legend.key.size = unit(0.3,"line"))

# ggplot code to make a boxplot
 p <- ggplot(chr1DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() +scale_color_manual(values = colors) +ggtitle("Chromosome 1") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + geom_hline(yintercept = 0, col="darkgray") +geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme + myYlabelTheme + myLegendTheme

 
 # scale_color_manual(values=c("blue","gray","pink","red","darkred"))
# graph the boxplot and save it to a pdf  
pdf("chr1ggplot.pdf")
p
dev.off()


###############
#Chromosome 2 
chr2Data <- data.frame(y=c((log2(l001.2[,3])),(log2(l002.2[,3])),(log2(l003.2[,3])),(log2(l004.2[,3])),(log2(l005.2[,3])),
      (log2(l007.2[,3])),(log2(l008.2[,3])),(log2(l009.2[,3])),(log2(l011.2[,3])),
  (log2(l018.2[,3])),(log2(l049.2[,3])),(log2(l059.2[,3])),
       (log2(l061.2[,3])),(log2(l069.2[,3])),(log2(l076.2[,3])),(log2(l077.2[,3])),(log2(l029.2[,3])),(log2(l050.2[,3])),(log2(l112.2[,3])),(log2(l115.2[,3])),(log2(l117.2[,3])),
        (log2(l123.2[,3])),(log2(l152.2[,3]))),
Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=length(l001.2$tracking_id)),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=length(l001.2$tracking_id)))

chr2Data.old <- data.frame(y=c((log2(l001.2.old[,3])),(log2(l002.2.old[,3])),(log2(l003.2.old[,3])),(log2(l004.2.old[,3])),(log2(l005.2.old[,3])),
        (log2(l006.2.old[,3])),(log2(l007.2.old[,3])),(log2(l008.2.old[,3])),(log2(l009.2.old[,3])),(log2(l011.2.old[,3])),
       (log2(l015.2.old[,3])),(log2(l028.2.old[,3])),(log2(l088.2.old[,3])),(log2(l108.2.old[,3])),(log2(l119.2.old[,3]))),
Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=length(l001.2.old$tracking_id)),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=length(l001.2.old$tracking_id)))

chr2DataCombo <- rbind(chr2Data,chr2Data.old)
chr2DataCombo$Line=factor(chr2DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))

 p2 <- ggplot(chr2DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors)  +ggtitle("Chromosome 2") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + geom_hline(yintercept = 0, col="darkgray") +geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme



pdf("chr2.ggplot.pdf")
p2
dev.off()


############
#Chrom 3 

chr3Data <- data.frame(y=c((log2(l001.3[,3])),(log2(l002.3[,3])),(log2(l003.3[,3])),(log2(l004.3[,3])),(log2(l005.3[,3])),
       (log2(l007.3[,3])),(log2(l008.3[,3])),(log2(l009.3[,3])),(log2(l011.3[,3])),
      (log2(l018.3[,3])),(log2(l049.3[,3])),(log2(l059.3[,3])),
      (log2(l061.3[,3])),(log2(l069.3[,3])),(log2(l076.3[,3])),(log2(l077.3[,3])),(log2(l029.3[,3])),(log2(l050.3[,3])),(log2(l112.3[,3])),(log2(l115.3[,3])),(log2(l117.3[,3])),
       (log2(l123.3[,3])),(log2(l152.3[,3]))),
Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=(length(l001.3$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=(length(l001.3$tracking_id))))
 
chr3Data.old <- data.frame(y=c((log2(l001.3.old[,3])),(log2(l002.3.old[,3])),(log2(l003.3.old[,3])),(log2(l004.3.old[,3])),(log2(l005.3.old[,3])),
      (log2(l006.3.old[,3])),(log2(l007.3.old[,3])),(log2(l008.3.old[,3])),(log2(l009.3.old[,3])),(log2(l011.3.old[,3])),
     (log2(l015.3.old[,3])),(log2(l028.3.old[,3])),(log2(l088.3.old[,3])),(log2(l108.3.old[,3])),(log2(l119.3.old[,3]))),
Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=(length(l001.3.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=length(l001.3.old$tracking_id)))

chr3DataCombo <- rbind(chr3Data,chr3Data.old)
chr3DataCombo$Line=factor(chr3DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))

 p3 <- ggplot(chr3DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors) +ggtitle("Chromosome 3") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + geom_hline(yintercept = 0, col="darkgray") +geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme


pdf("chr3.ggplot.pdf")
p3
 dev.off()


###########
#Chrm 4 

chr4Data <- data.frame(y=c((log2(l001.4[,3])),(log2(l002.4[,3])),(log2(l003.4[,3])),(log2(l004.4[,3])),(log2(l005.4[,3])),
     (log2(l007.4[,3])),(log2(l008.4[,3])),(log2(l009.4[,3])),(log2(l011.4[,3])),
    (log2(l018.4[,3])),(log2(l049.4[,3])),(log2(l059.4[,3])),
   (log2(l061.4[,3])),(log2(l069.4[,3])),(log2(l076.4[,3])),(log2(l077.4[,3])),(log2(l029.4[,3])),(log2(l050.4[,3])),(log2(l112.4[,3])),(log2(l115.4[,3])),(log2(l117.4[,3])),
   (log2(l123.4[,3])),(log2(l152.4[,3]))),
Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=(length(l001.4$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=(length(l001.4$tracking_id))))
 
 
chr4Data.old <- data.frame(y=c((log2(l001.4.old[,3])),(log2(l002.4.old[,3])),(log2(l003.4.old[,3])),(log2(l004.4.old[,3])),(log2(l005.4.old[,3])),
       (log2(l006.4.old[,3])),(log2(l007.4.old[,3])),(log2(l008.4.old[,3])),(log2(l009.4.old[,3])),(log2(l011.4.old[,3])),
       (log2(l015.4.old[,3])),(log2(l028.4.old[,3])),(log2(l088.4.old[,3])),(log2(l108.4.old[,3])),(log2(l119.4.old[,3]))),
Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=(length(l001.4.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=length(l001.4.old$tracking_id)))

chr4DataCombo <- rbind(chr4Data,chr4Data.old)
chr4DataCombo$Line=factor(chr4DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))

 p4 <- ggplot(chr4DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors)  +ggtitle("Chromosome 4") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + geom_hline(yintercept = 0, col="darkgray") +geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme


pdf("chr4.ggplot.pdf")
p4
 dev.off()



#######
# chrom 5 
chr5Data <- data.frame(y=c((log2(l001.5[,3])),(log2(l002.5[,3])),(log2(l003.5[,3])),(log2(l004.5[,3])),(log2(l005.5[,3])),
      (log2(l007.5[,3])),(log2(l008.5[,3])),(log2(l009.5[,3])),(log2(l011.5[,3])),
    (log2(l018.5[,3])),(log2(l049.5[,3])),(log2(l059.5[,3])),
   (log2(l061.5[,3])),(log2(l069.5[,3])),(log2(l076.5[,3])),(log2(l077.5[,3])),(log2(l029.5[,3])),(log2(l050.5[,3])),(log2(l112.5[,3])),(log2(l115.5[,3])),(log2(l117.5[,3])),
       (log2(l123.5[,3])),(log2(l152.5[,3]))),
Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=(length(l001.5$tracking_id))),copy_number=rep(c("2","2","2","3","2","2","2","2","2","2","3","2","2","2","2","2","2","2","2","2","3","3","2"),each=(length(l001.5$tracking_id))))
 
chr5Data.old <- data.frame(y=c((log2(l001.5.old[,3])),(log2(l002.5.old[,3])),(log2(l003.5.old[,3])),(log2(l004.5.old[,3])),(log2(l005.5.old[,3])),        (log2(l006.5.old[,3])),(log2(l007.5.old[,3])),(log2(l008.5.old[,3])),(log2(l009.5.old[,3])),(log2(l011.5.old[,3])),        (log2(l015.5.old[,3])),(log2(l028.5.old[,3])),(log2(l088.5.old[,3])),(log2(l108.5.old[,3])),(log2(l119.5.old[,3]))),Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=(length(l001.5.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=length(l001.5.old$tracking_id)))

chr5DataCombo <- rbind(chr5Data,chr5Data.old)

chr5DataCombo$Line=factor(chr5DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))

 p5 <- ggplot(chr5DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors)  +ggtitle("Chromosome 5") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + geom_hline(yintercept = 0, col="darkgray") +geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme


pdf("chr5.ggplot.pdf")
p5
 dev.off()

#######
#chrm 6 
chr6Data <- data.frame(y=c((log2(l001.6[,3])),(log2(l002.6[,3])),(log2(l003.6[,3])),(log2(l004.6[,3])),(log2(l005.6[,3])),       (log2(l007.6[,3])),(log2(l008.6[,3])),(log2(l009.6[,3])),(log2(l011.6[,3])),(log2(l018.6[,3])),
    (log2(l049.6[,3])),(log2(l059.6[,3])),(log2(l061.6[,3])),(log2(l069.6[,3])),
    (log2(l076.6[,3])),(log2(l077.6[,3])),(log2(l029.6[,3])),(log2(l050.6[,3])),(log2(l112.6[,3])),
    (log2(l115.6[,3])),(log2(l117.6[,3])),(log2(l123.6[,3])),(log2(l152.6[,3]))),Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=(length(l001.6$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=(length(l001.6$tracking_id))))
 
chr6Data.old <- data.frame(y=c((log2(l001.6.old[,3])),(log2(l002.6.old[,3])),(log2(l003.6.old[,3])),(log2(l004.6.old[,3])),(log2(l005.6.old[,3])),        (log2(l006.6.old[,3])),(log2(l007.6.old[,3])),(log2(l008.6.old[,3])),(log2(l009.6.old[,3])),(log2(l011.6.old[,3])),        (log2(l015.6.old[,3])),(log2(l028.6.old[,3])),(log2(l088.6.old[,3])),(log2(l108.6.old[,3])),(log2(l119.6.old[,3]))),Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=(length(l001.6.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=length(l001.6.old$tracking_id)))

chr6DataCombo <- rbind(chr6Data,chr6Data.old)

chr6DataCombo$Line=factor(chr6DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))

 p6 <- ggplot(chr6DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors)  +ggtitle("Chromosome 6") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + geom_hline(yintercept = 0, col="darkgray") +geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme


pdf("chr6.ggplot.pdf")
p6
 dev.off()


#######
# chrm 7

chr7Data <- data.frame(y=c((log2(l001.7[,3])),(log2(l002.7[,3])),(log2(l003.7[,3])),(log2(l004.7[,3])),(log2(l005.7[,3])),
      (log2(l007.7[,3])),(log2(l008.7[,3])),(log2(l009.7[,3])),(log2(l011.7[,3])),
    (log2(l018.7[,3])),(log2(l049.7[,3])),(log2(l059.7[,3])),
   (log2(l061.7[,3])),(log2(l069.7[,3])),(log2(l076.7[,3])),(log2(l077.7[,3])),(log2(l029.7[,3])),(log2(l050.7[,3])),(log2(l112.7[,3])),(log2(l115.7[,3])),(log2(l117.7[,3])),
   (log2(l123.7[,3])),(log2(l152.7[,3]))),
Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=(length(l001.7$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","3","3","2","2","2","2","2","2","2","2","2","2"),each=(length(l001.7$tracking_id))))
 
chr7Data.old <- data.frame(y=c((log2(l001.7.old[,3])),(log2(l002.7.old[,3])),(log2(l003.7.old[,3])),(log2(l004.7.old[,3])),(log2(l005.7.old[,3])),
      (log2(l006.7.old[,3])),(log2(l007.7.old[,3])),(log2(l008.7.old[,3])),(log2(l009.7.old[,3])),(log2(l011.7.old[,3])),
     (log2(l015.7.old[,3])),(log2(l028.7.old[,3])),(log2(l088.7.old[,3])),(log2(l108.7.old[,3])),(log2(l119.7.old[,3]))),
Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=(length(l001.7.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=length(l001.7.old$tracking_id)))

chr7DataCombo <- rbind(chr7Data,chr7Data.old)

chr7DataCombo$Line=factor(chr7DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))
 
 
 p7 <- ggplot(chr7DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors)  +ggtitle("Chromosome 7") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + geom_hline(yintercept = 0, col="darkgray") +geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme


pdf("chr7.ggplot.pdf")
p7
 dev.off()


#######
# chrm 8
chr8Data <- data.frame(y=c((log2(l001.8[,3])),(log2(l002.8[,3])),(log2(l003.8[,3])),(log2(l004.8[,3])),(log2(l005.8[,3])),
      (log2(l007.8[,3])),(log2(l008.8[,3])),(log2(l009.8[,3])),(log2(l011.8[,3])),
    (log2(l018.8[,3])),(log2(l049.8[,3])),(log2(l059.8[,3])),
   (log2(l061.8[,3])),(log2(l069.8[,3])),(log2(l076.8[,3])),(log2(l077.8[,3])),(log2(l029.8[,3])),(log2(l050.8[,3])),(log2(l112.8[,3])),(log2(l115.8[,3])),(log2(l117.8[,3])),
   (log2(l123.8[,3])),(log2(l152.8[,3]))),
Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=(length(l001.8$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","3"),each=(length(l001.8$tracking_id))))
 
chr8Data.old <- data.frame(y=c((log2(l001.8.old[,3])),(log2(l002.8.old[,3])),(log2(l003.8.old[,3])),(log2(l004.8.old[,3])),(log2(l005.8.old[,3])),
       (log2(l006.8.old[,3])),(log2(l007.8.old[,3])),(log2(l008.8.old[,3])),(log2(l009.8.old[,3])),(log2(l011.8.old[,3])),
      (log2(l015.8.old[,3])),(log2(l028.8.old[,3])),(log2(l088.8.old[,3])),(log2(l108.8.old[,3])),(log2(l119.8.old[,3]))),
Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=(length(l001.8.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","3","2"),each=length(l001.8.old$tracking_id)))

chr8DataCombo <- rbind(chr8Data,chr8Data.old)
chr8DataCombo$Line <- factor(chr8DataCombo$Line)
levels(chr8DataCombo$Line)
chr8DataCombo$Line=factor(chr8DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))

 p8 <- ggplot(chr8DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors)  +ggtitle("Chromosome 8") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + geom_hline(yintercept = 0, col="darkgray") +geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme


pdf("chr8.ggplot.pdf")
p8
 dev.off()

#######
# chrm 9 
chr9Data <- data.frame(y=c((log2(l001.9[,3])),(log2(l002.9[,3])),(log2(l003.9[,3])),(log2(l004.9[,3])),(log2(l005.9[,3])),
      (log2(l007.9[,3])),(log2(l008.9[,3])),(log2(l009.9[,3])),(log2(l011.9[,3])),
    (log2(l018.9[,3])),(log2(l049.9[,3])),(log2(l059.9[,3])),
   (log2(l061.9[,3])),(log2(l069.9[,3])),(log2(l076.9[,3])),(log2(l077.9[,3])),(log2(l029.9[,3])),(log2(l050.9[,3])),(log2(l112.9[,3])),(log2(l115.9[,3])),(log2(l117.9[,3])),
    (log2(l123.9[,3])),(log2(l152.9[,3]))),
Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=(length(l001.9$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","3","2","1","2","2","2","2","2","2"),each=(length(l001.9$tracking_id))))
 
chr9Data.old <- data.frame(y=c((log2(l001.9.old[,3])),(log2(l002.9.old[,3])),(log2(l003.9.old[,3])),(log2(l004.9.old[,3])),(log2(l005.9.old[,3])),
      (log2(l006.9.old[,3])),(log2(l007.9.old[,3])),(log2(l008.9.old[,3])),(log2(l009.9.old[,3])),(log2(l011.9.old[,3])),
     (log2(l015.9.old[,3])),(log2(l028.9.old[,3])),(log2(l088.9.old[,3])),(log2(l108.9.old[,3])),(log2(l119.9.old[,3]))),
Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=(length(l001.9.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","3","2","3","1","3"),each=length(l001.9.old$tracking_id)))

chr9DataCombo <- rbind(chr9Data,chr9Data.old)

chr9DataCombo$Line=factor(chr9DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))

 p9 <- ggplot(chr9DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors)  +ggtitle("Chromosome 9") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + geom_hline(yintercept = 0, col="darkgray") +geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme


pdf("chr9.ggplot.pdf")
p9
 dev.off()

#######
# chrm 10 
chr10Data <- data.frame(y=c((log2(l001.10[,3])),(log2(l002.10[,3])),(log2(l003.10[,3])),(log2(l004.10[,3])),(log2(l005.10[,3])),
      (log2(l007.10[,3])),(log2(l008.10[,3])),(log2(l009.10[,3])),(log2(l011.10[,3])),
    (log2(l018.10[,3])),(log2(l049.10[,3])),(log2(l059.10[,3])),
   (log2(l061.10[,3])),(log2(l069.10[,3])),(log2(l076.10[,3])),(log2(l077.10[,3])),(log2(l029.10[,3])),(log2(l050.10[,3])),(log2(l112.10[,3])),(log2(l115.10[,3])),(log2(l117.10[,3])),
    (log2(l123.10[,3])),(log2(l152.10[,3]))),
Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=(length(l001.10$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2.5","2","2","2","2","2","2","2","2"),each=(length(l001.10$tracking_id))))
 
chr10Data.old <- data.frame(y=c((log2(l001.10.old[,3])),(log2(l002.10.old[,3])),(log2(l003.10.old[,3])),(log2(l004.10.old[,3])),(log2(l005.10.old[,3])),
      (log2(l006.10.old[,3])),(log2(l007.10.old[,3])),(log2(l008.10.old[,3])),(log2(l009.10.old[,3])),(log2(l011.10.old[,3])),
     (log2(l015.10.old[,3])),(log2(l028.10.old[,3])),(log2(l088.10.old[,3])),(log2(l108.10.old[,3])),(log2(l119.10.old[,3]))),
Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=(length(l001.10.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=length(l001.10.old$tracking_id)))

chr10DataCombo <- rbind(chr10Data,chr10Data.old)

chr10DataCombo$Line=factor(chr10DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))

 p10 <- ggplot(chr10DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors) +ggtitle("Chromosome 10") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + geom_hline(yintercept = 0, col="darkgray") +geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme


pdf("chr10.ggplot.pdf")
p10
 dev.off()

#######
# chrm 11 
chr11Data <- data.frame(y=c((log2(l001.11[,3])),(log2(l002.11[,3])),(log2(l003.11[,3])),(log2(l004.11[,3])),(log2(l005.11[,3])),
      (log2(l007.11[,3])),(log2(l008.11[,3])),(log2(l009.11[,3])),(log2(l011.11[,3])),
    (log2(l018.11[,3])),(log2(l049.11[,3])),(log2(l059.11[,3])),
   (log2(l061.11[,3])),(log2(l069.11[,3])),(log2(l076.11[,3])),(log2(l077.11[,3])),(log2(l029.11[,3])),(log2(l050.11[,3])),(log2(l112.11[,3])),(log2(l115.11[,3])),(log2(l117.11[,3])),
    (log2(l123.11[,3])),(log2(l152.11[,3]))),
Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=(length(l001.11$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=(length(l001.11$tracking_id))))
 
chr11Data.old <- data.frame(y=c((log2(l001.11.old[,3])),(log2(l002.11.old[,3])),(log2(l003.11.old[,3])),(log2(l004.11.old[,3])),(log2(l005.11.old[,3])),
      (log2(l006.11.old[,3])),(log2(l007.11.old[,3])),(log2(l008.11.old[,3])),(log2(l009.11.old[,3])),(log2(l011.11.old[,3])),
     (log2(l015.11.old[,3])),(log2(l028.11.old[,3])),(log2(l088.11.old[,3])),(log2(l108.11.old[,3])),(log2(l119.11.old[,3]))),
Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=(length(l001.11.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=length(l001.11.old$tracking_id)))

chr11DataCombo <- rbind(chr11Data,chr11Data.old)

chr11DataCombo$Line=factor(chr11DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))

 p11 <- ggplot(chr11DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors)  +ggtitle("Chromosome 11") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + geom_hline(yintercept = 0, col="darkgray") +geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme


pdf("chr11.ggplot.pdf")
p11
 dev.off()
 

#######
# chrm 12 
  chr12Data <- data.frame(y=c((log2(l001.12[,3])),(log2(l002.12[,3])),(log2(l003.12[,3])),(log2(l004.12[,3])),(log2(l005.12[,3])),
      (log2(l007.12[,3])),(log2(l008.12[,3])),(log2(l009.12[,3])),(log2(l011.12[,3])),
    (log2(l018.12[,3])),(log2(l049.12[,3])),(log2(l059.12[,3])),
   (log2(l061.12[,3])),(log2(l069.12[,3])),(log2(l076.12[,3])),(log2(l077.12[,3])),(log2(l029.12[,3])),(log2(l050.12[,3])),(log2(l112.12[,3])),(log2(l115.12[,3])),(log2(l117.12[,3])),
    (log2(l123.12[,3])),(log2(l152.12[,3]))),
Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=(length(l001.12$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","3","2","2","2","2","2","3","2","2","2","2","2","2","2"),each=(length(l001.12$tracking_id))))
 
chr12Data.old <- data.frame(y=c((log2(l001.12.old[,3])),(log2(l002.12.old[,3])),(log2(l003.12.old[,3])),(log2(l004.12.old[,3])),(log2(l005.12.old[,3])),
      (log2(l006.12.old[,3])),(log2(l007.12.old[,3])),(log2(l008.12.old[,3])),(log2(l009.12.old[,3])),(log2(l011.12.old[,3])),
     (log2(l015.12.old[,3])),(log2(l028.12.old[,3])),(log2(l088.12.old[,3])),(log2(l108.12.old[,3])),(log2(l119.12.old[,3]))),
Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=(length(l001.12.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=length(l001.12.old$tracking_id)))

chr12DataCombo <- rbind(chr12Data,chr12Data.old)

chr12DataCombo$Line=factor(chr12DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))

 p12 <- ggplot(chr12DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors)  +ggtitle("Chromosome 12") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + geom_hline(yintercept = 0, col="darkgray") +geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme


pdf("chr12.ggplot.pdf")
p12
 dev.off()

#######
## Chromosome 13
chr13Data <- data.frame(y=c((log2(l001.13[,3])),(log2(l002.13[,3])),(log2(l003.13[,3])),(log2(l004.13[,3])),(log2(l005.13[,3])),
      (log2(l007.13[,3])),(log2(l008.13[,3])),(log2(l009.13[,3])),(log2(l011.13[,3])),
    (log2(l018.13[,3])),(log2(l049.13[,3])),(log2(l059.13[,3])),
   (log2(l061.13[,3])),(log2(l069.13[,3])),(log2(l076.13[,3])),(log2(l077.13[,3])),(log2(l029.13[,3])),(log2(l050.13[,3])),(log2(l112.13[,3])),(log2(l115.13[,3])),(log2(l117.13[,3])),
    (log2(l123.13[,3])),(log2(l152.13[,3]))),
Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=(length(l001.13$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=(length(l001.13$tracking_id))))
 
chr13Data.old <- data.frame(y=c((log2(l001.13.old[,3])),(log2(l002.13.old[,3])),(log2(l003.13.old[,3])),(log2(l004.13.old[,3])),(log2(l005.13.old[,3])),
      (log2(l006.13.old[,3])),(log2(l007.13.old[,3])),(log2(l008.13.old[,3])),(log2(l009.13.old[,3])),(log2(l011.13.old[,3])),
     (log2(l015.13.old[,3])),(log2(l028.13.old[,3])),(log2(l088.13.old[,3])),(log2(l108.13.old[,3])),(log2(l119.13.old[,3]))),
Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=(length(l001.13.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=length(l001.13.old$tracking_id)))

chr13DataCombo <- rbind(chr13Data,chr13Data.old)

chr13DataCombo$Line=factor(chr13DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))

 p13 <- ggplot(chr13DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors)  +ggtitle("Chromosome 13") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + geom_hline(yintercept = 0, col="darkgray") +geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme


pdf("chr13.ggplot.pdf")
p13
 dev.off()

##########
# Chromosome 14

   chr14Data <- data.frame(y=c((log2(l001.14[,3])),(log2(l002.14[,3])),(log2(l003.14[,3])),(log2(l004.14[,3])),(log2(l005.14[,3])),
      (log2(l007.14[,3])),(log2(l008.14[,3])),(log2(l009.14[,3])),(log2(l011.14[,3])),
    (log2(l018.14[,3])),(log2(l049.14[,3])),(log2(l059.14[,3])),
   (log2(l061.14[,3])),(log2(l069.14[,3])),(log2(l076.14[,3])),(log2(l077.14[,3])),(log2(l029.14[,3])),(log2(l050.14[,3])),(log2(l112.14[,3])),(log2(l115.14[,3])),(log2(l117.14[,3])),
    (log2(l123.14[,3])),(log2(l152.14[,3]))),
Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=(length(l001.14$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","3","2","2","2","2","2","2","2","2"),each=(length(l001.14$tracking_id))))
 
chr14Data.old <- data.frame(y=c((log2(l001.14.old[,3])),(log2(l002.14.old[,3])),(log2(l003.14.old[,3])),(log2(l004.14.old[,3])),(log2(l005.14.old[,3])),
      (log2(l006.14.old[,3])),(log2(l007.14.old[,3])),(log2(l008.14.old[,3])),(log2(l009.14.old[,3])),(log2(l011.14.old[,3])),
     (log2(l015.14.old[,3])),(log2(l028.14.old[,3])),(log2(l088.14.old[,3])),(log2(l108.14.old[,3])),(log2(l119.14.old[,3]))),
Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=(length(l001.14.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","3","2","2","2","2","2","2"),each=length(l001.14.old$tracking_id)))

chr14DataCombo <- rbind(chr14Data,chr14Data.old)

chr14DataCombo$Line=factor(chr14DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))

 p14 <- ggplot(chr14DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors)  +ggtitle("Chromosome 14") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + geom_hline(yintercept = 0, col="darkgray") +geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme


pdf("chr14.ggplot.pdf")
p14
 dev.off()

##########
# chrm 15 


  chr15Data <- data.frame(y=c((log2(l001.15[,3])),(log2(l002.15[,3])),(log2(l003.15[,3])),(log2(l004.15[,3])),(log2(l005.15[,3])),
      (log2(l007.15[,3])),(log2(l008.15[,3])),(log2(l009.15[,3])),(log2(l011.15[,3])),
    (log2(l018.15[,3])),(log2(l049.15[,3])),(log2(l059.15[,3])),
   (log2(l061.15[,3])),(log2(l069.15[,3])),(log2(l076.15[,3])),(log2(l077.15[,3])),(log2(l029.15[,3])),(log2(l050.15[,3])),(log2(l112.15[,3])),(log2(l115.15[,3])),(log2(l117.15[,3])),
    (log2(l123.15[,3])),(log2(l152.15[,3]))),
Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=(length(l001.15$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","3","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=(length(l001.15$tracking_id))))
 
chr15Data.old <- data.frame(y=c((log2(l001.15.old[,3])),(log2(l002.15.old[,3])),(log2(l003.15.old[,3])),(log2(l004.15.old[,3])),(log2(l005.15.old[,3])),
      (log2(l006.15.old[,3])),(log2(l007.15.old[,3])),(log2(l008.15.old[,3])),(log2(l009.15.old[,3])),(log2(l011.15.old[,3])),
     (log2(l015.15.old[,3])),(log2(l028.15.old[,3])),(log2(l088.15.old[,3])),(log2(l108.15.old[,3])),(log2(l119.15.old[,3]))),
Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=(length(l001.15.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=length(l001.15.old$tracking_id)))

chr15DataCombo <- rbind(chr15Data,chr15Data.old)

chr15DataCombo$Line=factor(chr15DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))

 p15 <- ggplot(chr15DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors) +ggtitle("Chromosome 15") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + geom_hline(yintercept = 0, col="darkgray") +geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2)+ myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme


pdf("chr15.ggplot.pdf")
p15
 dev.off()
##########
# chrm 16 
 chr16Data <- data.frame(y=c((log2(l001.16[,3])),(log2(l002.16[,3])),(log2(l003.16[,3])),(log2(l004.16[,3])),(log2(l005.16[,3])),
      (log2(l007.16[,3])),(log2(l008.16[,3])),(log2(l009.16[,3])),(log2(l011.16[,3])),
    (log2(l018.16[,3])),(log2(l049.16[,3])),(log2(l059.16[,3])),
   (log2(l061.16[,3])),(log2(l069.16[,3])),(log2(l076.16[,3])),(log2(l077.16[,3])),(log2(l029.16[,3])),(log2(l050.16[,3])),(log2(l112.16[,3])),(log2(l115.16[,3])),(log2(l117.16[,3])),
    (log2(l123.16[,3])),(log2(l152.16[,3]))),
Line=rep(c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom29","Hom50","Hom112","Hom115","Hom117","Hom123","Hom152"),each=(length(l001.16$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","4","2","2","2","2","2","2","2","2","2","2","2","3","2","2","2","2"),each=(length(l001.16$tracking_id))))
 
chr16Data.old <- data.frame(y=c((log2(l001.16.old[,3])),(log2(l002.16.old[,3])),(log2(l003.16.old[,3])),(log2(l004.16.old[,3])),(log2(l005.16.old[,3])),
      (log2(l006.16.old[,3])),(log2(l007.16.old[,3])),(log2(l008.16.old[,3])),(log2(l009.16.old[,3])),(log2(l011.16.old[,3])),
     (log2(l015.16.old[,3])),(log2(l028.16.old[,3])),(log2(l088.16.old[,3])),(log2(l108.16.old[,3])),(log2(l119.16.old[,3]))),
Line=rep(c("Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom88","Hom108","Hom119"),each=(length(l001.16.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=length(l001.16.old$tracking_id)))

chr16DataCombo <- rbind(chr16Data,chr16Data.old)

chr16DataCombo$Line=factor(chr16DataCombo$Line, c("Het1","Het2","Het3","Het4","Het5","Het7","Het8","Het9","Het11","Het18","Het49","Het59","Het61","Het69","Het76","Het77","Hom1","Hom2","Hom3","Hom4","Hom5","Hom6","Hom7","Hom8","Hom9","Hom11","Hom15","Hom28","Hom29","Hom50","Hom88","Hom108","Hom112","Hom115","Hom117","Hom119","Hom123","Hom152"))

 p16 <- ggplot(chr16DataCombo, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors)  + geom_hline(yintercept = 0, col="darkgray") + geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) +ggtitle("Chromosome 16") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme

pdf("chr16.ggplot.pdf")
p16
 dev.off()
 
 
par(mfrow= c(4,4))
pdf("allBoxplotsDC.pdf")

install.packages("ggpubr")
library(ggpubr)
library(cowplot)


plot_grid(p, p2, labels = c('1', '2'))
# extract the legend from one of the plots


prow <- plot_grid(
  p + theme(legend.position="none"),
  p2 + theme(legend.position="none"),
  p3 + theme(legend.position="none"),
  p4 + theme(legend.position="none"),
  align = 'hv',
  nrow = 2,
  ncol = 2
)
prow

prow2 <- plot_grid(
  p5 + theme(legend.position="none"),
  p6 + theme(legend.position="none"),
  p7 + theme(legend.position="none"),
  p8 + theme(legend.position="none"),
  align = 'hv',
  nrow = 2,
  ncol = 2
)  

prow3 <- plot_grid(
  p9 + theme(legend.position="none"),
  p10 + theme(legend.position="none"),
  p11 + theme(legend.position="none"),
  p12 + theme(legend.position="none"),
  align = 'hv',
  nrow = 2,
  ncol = 2
)  
prow4 <- plot_grid(
  p13 + theme(legend.position="none"),
  p14 + theme(legend.position="none"),
  p15 + theme(legend.position="none"),
  p16 + theme(legend.position="none"),
  align = 'hv',
  nrow = 2,
  ncol = 2
) 


## need to make a plot with a fake legend to get the legend out 
## fake plot
   fakeLegendData1 <- data.frame(y=c((log2(l001.16[,3])),(log2(l002.16[,3])),(log2(l003.16[,3])),(log2(l004.16[,3])),(log2(l005.16[,3])),
      (log2(l007.16[,3])),(log2(l008.16[,3])),(log2(l009.16[,3])),(log2(l011.16[,3])),
    (log2(l018.16[,3])),(log2(l049.16[,3])),(log2(l059.16[,3])),
   (log2(l061.16[,3])),(log2(l069.16[,3])),(log2(l076.16[,3])),(log2(l077.16[,3])),(log2(l029.16[,3])),(log2(l050.16[,3])),(log2(l112.16[,3])),(log2(l115.16[,3])),(log2(l117.16[,3])),
    (log2(l123.16[,3])),(log2(l152.16[,3]))),
Line=rep(c("1","2","3","4","5","7","8","9","11","18","49","59","61","69","76","77","29","50","112","115","117","123","152"),each=(length(l001.16$tracking_id))),copy_number=rep(c("1","2","2","2","2","2","4","2","2","2","2","2","2","2","2","2","2.5","2","3","2","2","4","2"),each=(length(l001.16$tracking_id))))
 
fakeLegendData2 <- data.frame(y=c((log2(l001.16.old[,3])),(log2(l002.16.old[,3])),(log2(l003.16.old[,3])),(log2(l004.16.old[,3])),(log2(l005.16.old[,3])),
      (log2(l006.16.old[,3])),(log2(l007.16.old[,3])),(log2(l008.16.old[,3])),(log2(l009.16.old[,3])),(log2(l011.16.old[,3])),
     (log2(l015.16.old[,3])),(log2(l028.16.old[,3])),(log2(l088.16.old[,3])),(log2(l108.16.old[,3])),(log2(l119.16.old[,3]))),
Line=rep(c("1.","2.","3.","4.","5.","6.","7.","8.","9.","11.","15","28","88","108","119"),each=(length(l001.16.old$tracking_id))),copy_number=rep(c("2","2","2","2","2","2","2","2","2","2","2","2","2","2","2"),each=length(l001.16.old$tracking_id)))

fakeLegendData <- rbind(fakeLegendData1,fakeLegendData2)

fakeLegendData$Line=factor(fakeLegendData$Line, c("1","2","3","4","5","7","8","9","11","18","49","59","61","69","76","77","1.","2.","3.","4.","5.","6.","7.","8.","9.","11.","15","28","29","50","88","108","112","115","117","119","123","152"))


fakeLegend <- ggplot(fakeLegendData, aes(x=Line, y=y,color=copy_number)) + 
  geom_boxplot() + theme_classic() + scale_color_manual(values = colors)  + geom_hline(yintercept = 0, col="darkgray") + geom_hline(yintercept = 0.583, col="red", linetype=2) + geom_hline(yintercept = -1, col="blue", linetype=2) +ggtitle("Chromosome 16") +xlab("Sample") + ylab("log2(FPKMsample/FPKManc)") + myTitleTheme + myXaxisTheme + myXlabelTheme + myYaxisTheme +myYlabelTheme + myLegendTheme

legend <- get_legend(
  # create some space to the left of the legend
  fakeLegend + theme(legend.box.margin = margin(0, 0, 0, 8))
)

plot_grid(prow, legend, rel_widths = c(3, .4))

pdf("chr1_4_DC.pdf",height = 12, width = 12, useDingbats = FALSE)
plot_grid(prow, legend,rel_widths = c(3, .4))
dev.off()

pdf("chr5_8_DC.pdf",height = 12, width = 12, useDingbats = FALSE)
plot_grid(prow2, legend,rel_widths = c(3, .4))
dev.off()

pdf("chr9_12_DC.pdf",height = 12, width = 12, useDingbats = FALSE)
plot_grid(prow3, legend,rel_widths = c(3, .4))
dev.off()

pdf("chr13_16_DC.pdf",height = 12, width = 12, useDingbats = FALSE)
plot_grid(prow4, legend,rel_widths = c(3, .4))
dev.off()
          # , common.legend = TRUE, legend = "bottom")




```

Codes to run ANOVAs and Tukey's tests and plot results 
```{r ANOVAs Tukeys, include=TRUE, echo=FALSE}
####################
#CHROMOSOME 1 
#data frame with only euploids, so I can do a euploid by euploid test
# chr1DataofInterest <- data.frame(y=c((log2(l001.1[,3])),(log2(l002.1[,3])),(log2(l003.1[,3])),(log2(l005.1[,3])),
#         (log2(l007.1[,3])),(log2(l009.1[,3])),(log2(l011.1[,3])),
#         (log2(l018.1[,3])),(log2(l069.1[,3]))),
# Line=rep(c(1,2,3,5,7,9,11,18,69),each=(length(l001.1$tracking_id))))
# chr1DataofInterest$Line <-factor(chr1DataofInterest$Line)

chr1DataGC <- data.frame(y=c((log2(l001.1[,3])),(log2(l002.1[,3])),(log2(l003.1[,3])),(log2(l004.1[,3])),(log2(l005.1[,3])),
        (log2(l007.1[,3])),(log2(l008.1[,3])),(log2(l009.1[,3])),(log2(l011.1[,3])),
        (log2(l018.1[,3])),(log2(l049.1[,3])),(log2(l059.1[,3])),(log2(l061.1[,3])),(log2(l069.1[,3])),(log2(l076.1[,3])),(log2(l077.1[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.1$tracking_id))))
chr1DataGC$Line <-factor(chr1DataGC$Line)
chr1DataMA <- data.frame(y=c((log2(l001.1.old[,3])),(log2(l002.1.old[,3])),(log2(l003.1.old[,3])),(log2(l004.1.old[,3])),(log2(l005.1.old[,3])),
        (log2(l007.1.old[,3])),(log2(l008.1.old[,3])),(log2(l009.1.old[,3])),(log2(l011.1.old[,3])),
        (log2(l015.1.old[,3])),(log2(l028.1.old[,3])),(log2(l029.1[,3])),(log2(l050.1[,3])),(log2(l088.1.old[,3])),(log2(l108.1.old[,3])),(log2(l112.1[,3])),(log2(l115.1[,3])),(log2(l117.1[,3])),(log2(l119.1.old[,3])),(log2(l123.1[,3])),(log2(l152.1[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.1.old$tracking_id))))
chr1DataMA$Line <-factor(chr1DataMA$Line)



#test pairwise comparisons
# TukeyHSD(aov.chr1ofInt)
# Tukey.chr1.ofInt <- TukeyHSD(aov.chr1ofInt)


aovChr1GC <- lm(y~Line,data=chr1DataGC)
summary(aovChr1GC)


aovChr1MA <- lm(y~Line,data=chr1DataMA)
summary(aovChr1MA)

stargazer(aovChr1MA,type="html",out="anovaChr1MA.doc",title="Chromosome I Lab Strain")

#Tukey.chr1.ofInt <- data.table(TukeyHSD(aov.chr1ofInt))
#write.csv(Tukey.chr1.ofInt,file="Tukey.chr1.ofInt.csv")
#{pdf("Chrm1.Tukey.ofInt.GC.pdf")
#plot(TukeyHSD(aov.chr1ofInt))}



##############################################################################################
#CHROMOSOME 2
chr2DataGC <- data.frame(y=c((log2(l001.2[,3])),(log2(l002.2[,3])),(log2(l003.2[,3])),(log2(l004.2[,3])),(log2(l005.2[,3])),
        (log2(l007.2[,3])),(log2(l008.2[,3])),(log2(l009.2[,3])),(log2(l011.2[,3])),
        (log2(l018.2[,3])),(log2(l049.2[,3])),(log2(l059.2[,3])),(log2(l061.2[,3])),(log2(l069.2[,3])),(log2(l076.2[,3])),(log2(l077.2[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.2$tracking_id))))
chr2DataGC$Line <-factor(chr2DataGC$Line)
chr2DataMA <- data.frame(y=c((log2(l001.2.old[,3])),(log2(l002.2.old[,3])),(log2(l003.2.old[,3])),(log2(l004.2.old[,3])),(log2(l005.2.old[,3])),
        (log2(l007.2.old[,3])),(log2(l008.2.old[,3])),(log2(l009.2.old[,3])),(log2(l011.2.old[,3])),
        (log2(l015.2.old[,3])),(log2(l028.2.old[,3])),(log2(l029.2[,3])),(log2(l050.2[,3])),(log2(l088.2.old[,3])),(log2(l108.2.old[,3])),(log2(l112.2[,3])),(log2(l115.2[,3])),(log2(l117.2[,3])),(log2(l119.2.old[,3])),(log2(l123.2[,3])),(log2(l152.2[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.2.old$tracking_id))))
chr2DataMA$Line <-factor(chr2DataMA$Line)

aovchr2GC <- lm(y~Line,data=chr2DataGC)
summary(aovchr2GC)
stargazer(aovchr2GC,type="html",out="anovachr2GC.doc",title="Chromosome II Hybrid Strain")

aovchr2MA <- lm(y~Line,data=chr2DataMA)
summary(aovchr2MA)

stargazer(aovchr2MA,type="html",out="anovachr2MA.doc",title="Chromosome II Lab Strain")




# aov.chr2 <- aov(y~Line,data=chr2DataCombo)
# chr2Data
# summary(aov.chr2)
# resids.chr2 <- resid(aov.chr2)
# length(resids.chr2)
# #View(resids.chr2)
# {pdf("Chrm2.resids.pdf")
# hist(resids.chr2,col="turquoise",breaks=10,xlab="Residuals.Chr.2")
# }
# #shapiro.test(resids.chr2)
# TukeyChr2 <- data.table(TukeyHSD(aov.chr2))
# #write.csv(TukeyChr2,file="Tukey.chr2.csv")
# TukeyHSD(aov.chr2)
# {pdf("Chr2.tukey.pdf")
# plot(TukeyHSD(aov.chr2))}
# #fail shapiro test, so should probably use nonparametric
# kruskal.chr2 <- kruskal.test(y~Line,data=chr2Data)
# kruskal.test(y~Line,data=chr2Data)
# summary(kruskal.chr2)
#test pairwise comparisons
#Tukeys doesnt work with kruskal
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
# kruskalmc(y~Line,data=chr2Data)
# multiCompsChr2 <- kruskalmc(y~Line,data=chr2Data)
# #many pairwise t-tests
# pairwise.wilcox.test(chr2Data$y,chr2Data$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 3

chr3DataGC <- data.frame(y=c((log2(l001.3[,3])),(log2(l002.3[,3])),(log2(l003.3[,3])),(log2(l004.3[,3])),(log2(l005.3[,3])),
        (log2(l007.3[,3])),(log2(l008.3[,3])),(log2(l009.3[,3])),(log2(l011.3[,3])),
        (log2(l018.3[,3])),(log2(l049.3[,3])),(log2(l059.3[,3])),(log2(l061.3[,3])),(log2(l069.3[,3])),(log2(l076.3[,3])),(log2(l077.3[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.3$tracking_id))))
chr3DataGC$Line <-factor(chr3DataGC$Line)
chr3DataMA <- data.frame(y=c((log2(l001.3.old[,3])),(log2(l002.3.old[,3])),(log2(l003.3.old[,3])),(log2(l004.3.old[,3])),(log2(l005.3.old[,3])),
        (log2(l007.3.old[,3])),(log2(l008.3.old[,3])),(log2(l009.3.old[,3])),(log2(l011.3.old[,3])),
        (log2(l015.3.old[,3])),(log2(l028.3.old[,3])),(log2(l029.3[,3])),(log2(l050.3[,3])),(log2(l088.3.old[,3])),(log2(l108.3.old[,3])),(log2(l112.3[,3])),(log2(l115.3[,3])),(log2(l117.3[,3])),(log2(l119.3.old[,3])),(log2(l123.3[,3])),(log2(l152.3[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.3.old$tracking_id))))
chr3DataMA$Line <-factor(chr3DataMA$Line)
aovchr3GC <- lm(y~Line,data=chr3DataGC)
stargazer(aovchr3GC,type="html",out="anovachr3GC.doc",title="Chromosome III Hybrid Strain")
aovchr3MA <- lm(y~Line,data=chr3DataMA)
stargazer(aovchr3MA,type="html",out="anovachr3MA.doc",title="Chromosome III Lab Strain")


#RATIOS FOR CHROMOSOME 3
#euRatio.c3 <- c(l002.3[,3],l003.3[,3],l005.3[,3],l006.3[,3],l008.3[,3],l009.3[,3],l069.3[,3])

# #this code works beautifully to produce a table with all of the log2 values for chromosome 1 
# chr3Data <- data.frame(y=c((log2(l001.3[,3])),(log2(l002.3[,3])),(log2(l003.3[,3])),(log2(l004.3[,3])),
#         (log2(l005.3[,3])),(log2(l007.3[,3])),(log2(l008.3[,3])),
#         (log2(l009.3[,3])),(log2(l011.3[,3])),(log2(l018.3[,3])),(log2(l049.3[,3])),(log2(l059.3[,3])),(log2(l061.3[,3])),
#         (log2(l069.3[,3])),(log2(l076.3[,3])),(log2(l077.3[,3]))),
# Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.3$tracking_id))))
# chr3Data$Line <-factor(chr3Data$Line)
#This code works beautifully to produce a boxplot 
# {pdf("Chrm_3.pdf")
# boxplot(y~Line,data=chr3Data,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","cyan","green","green","cyan","cyan","green","darkcyan","cyan","cyan",
#               "green","cyan","cyan","cyan","cyan","green","cyan","cyan"),main="GC Lines Chromsome 3",las=3,ylim=c(-2,2.5))
# abline(h=0,lty=2)
# abline(h=0.585,lty=2,col="forestgreen")
# abline(h=-1,lty=2,col="darkviolet")
# }


# aov.chr3 <- aov(y~Line,data=chr3Data)
# summary(aov.chr3)
# resids.chr3 <- resid(aov.chr3)
# {pdf("Chrm3.resids.pdf")
# hist(resids.chr3,col="turquoise",breaks=10,xlab="Residuals.Chr.3")
# }
# #shapiro.test(resids.chr3)
# Tukey.chr3 <- data.table(TukeyHSD(aov.chr3))
# TukeyHSD(aov.chr3)
# {pdf("Tukey.chr3.pdf")
#  plot(TukeyHSD(aov.chr3)) }
# #fail shapiro test, so should probably use nonparametric
# kruskal.chr3 <- kruskal.test(y~Line,data=chr3Data)
# kruskal.test(y~Line,data=chr3Data)
# summary(kruskal.chr3)
#test pairwise comparisons
#Tukeys doesnt work with kruskal
#but there is an equivalent for Tukeys with kruskal
# #Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
# kruskalmc(y~Line,data=chr3Data)
# multiCompsChr3 <- kruskalmc(y~Line,data=chr3Data)
# #many pairwise t-tests
# pairwise.t.test(chr3Data$y,chr3Data$Line,p.adjust.method = "fdr")
# wilcox.chr3 <- pairwise.wilcox.test(chr3Data$y,chr3Data$Line,p.adjust.method = "fdr")
# pairwise.wilcox.test(chr3Data$y,chr3Data$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 4
chr4DataGC <- data.frame(y=c((log2(l001.4[,3])),(log2(l002.4[,3])),(log2(l003.4[,3])),(log2(l004.4[,3])),(log2(l005.4[,3])),
        (log2(l007.4[,3])),(log2(l008.4[,3])),(log2(l009.4[,3])),(log2(l011.4[,3])),
        (log2(l018.4[,3])),(log2(l049.4[,3])),(log2(l059.4[,3])),(log2(l061.4[,3])),(log2(l069.4[,3])),(log2(l076.4[,3])),(log2(l077.4[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.4$tracking_id))))
chr4DataGC$Line <-factor(chr4DataGC$Line)
chr4DataMA <- data.frame(y=c((log2(l001.4.old[,3])),(log2(l002.4.old[,3])),(log2(l003.4.old[,3])),(log2(l004.4.old[,3])),(log2(l005.4.old[,3])),
        (log2(l007.4.old[,3])),(log2(l008.4.old[,3])),(log2(l009.4.old[,3])),(log2(l011.4.old[,3])),
        (log2(l015.4.old[,3])),(log2(l028.4.old[,3])),(log2(l029.4[,3])),(log2(l050.4[,3])),(log2(l088.4.old[,3])),(log2(l108.4.old[,3])),(log2(l112.4[,3])),(log2(l115.4[,3])),(log2(l117.4[,3])),(log2(l119.4.old[,3])),(log2(l123.4[,3])),(log2(l152.4[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.4.old$tracking_id))))
chr4DataMA$Line <-factor(chr4DataMA$Line)
aovchr4GC <- lm(y~Line,data=chr4DataGC)
stargazer(aovchr4GC,type="html",out="anovachr4GC.doc",title="Chromosome IV Hybrid Strain")
aovchr4MA <- lm(y~Line,data=chr4DataMA)
stargazer(aovchr4MA,type="html",out="anovachr4MA.doc",title="Chromosome IV Lab Strain")


#RATIOS FOR CHROMOSOME 4
#euRatio.c4 <- c(l002.4[,3],l003.4[,3],l005.4[,3],l006.4[,3],l008.4[,3],l009.4[,3],l069.4[,3])

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
# chr4Data <- data.frame(y=c((log2(l001.4[,3])),(log2(l002.4[,3])),(log2(l003.4[,3])),(log2(l004.4[,3])),
#         (log2(l005.4[,3])),(log2(l007.4[,3])),(log2(l008.4[,3])),
#         (log2(l009.4[,3])),(log2(l011.4[,3])),(log2(l018.4[,3])),
#         (log2(l049.4[,3])),(log2(l059.4[,3])),(log2(l061.4[,3])),
#         (log2(l069.4[,3])),(log2(l076.4[,3])),(log2(l077.4[,3]))),
# Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.4$tracking_id))))
# chr4Data$Line <-factor(chr4Data$Line)
#This code works beautifully to produce a boxplot 
# {pdf("Chrm_4.pdf")
# boxplot(y~Line,data=chr4Data,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","cyan","green","green","cyan","cyan","green","darkcyan","cyan","cyan",
#               "green","cyan","cyan","cyan","cyan","green","cyan","cyan"),main="GC Lines Chromsome 4",las=3,ylim=c(-2,2.5))
# abline(h=0,lty=2)
# abline(h=0.585,lty=2,col="forestgreen")
# abline(h=-1,lty=2,col="darkviolet")
# }

# aov.chr4 <- aov(y~Line,data=chr4Data)
# summary(aov.chr4)
# resids.chr4 <- resid(aov.chr4)
# {pdf("Chrm4.resids.pdf")
# hist(resids.chr4,col="turquoise",breaks=10,xlab="Residuals.Chr.4")}
# #shapiro.test(resids.chr4)
# TukeyHSD(aov.chr4)
# {pdf("tukey.chr4.pdf")
#  plot(TukeyHSD(aov.chr4)) }

#fail shapiro test, so should probably use nonparametric
# kruskal.chr4 <- kruskal.test(y~Line,data=chr4Data)
# kruskal.test(y~Line,data=chr4Data)
# summary(kruskal.chr4)
#test pairwise comparisons
#Tukeys doesnt work with kruskal
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
# kruskalmc(y~Line,data=chr4Data)
# multiCompsChr4 <- kruskalmc(y~Line,data=chr4Data)
# #many pairwise t-tests
# pairwise.t.test(chr4Data$y,chr4Data$Line,p.adjust.method = "fdr")
# wilcox.chr4 <- pairwise.wilcox.test(chr4Data$y,chr4Data$Line,p.adjust.method = "fdr")
# pairwise.wilcox.test(chr4Data$y,chr4Data$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 5
chr5DataGC <- data.frame(y=c((log2(l001.5[,3])),(log2(l002.5[,3])),(log2(l003.5[,3])),(log2(l004.5[,3])),(log2(l005.5[,3])),
        (log2(l007.5[,3])),(log2(l008.5[,3])),(log2(l009.5[,3])),(log2(l011.5[,3])),
        (log2(l018.5[,3])),(log2(l049.5[,3])),(log2(l059.5[,3])),(log2(l061.5[,3])),(log2(l069.5[,3])),(log2(l076.5[,3])),(log2(l077.5[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.5$tracking_id))))
chr5DataGC$Line <-factor(chr5DataGC$Line)
chr5DataMA <- data.frame(y=c((log2(l001.5.old[,3])),(log2(l002.5.old[,3])),(log2(l003.5.old[,3])),(log2(l004.5.old[,3])),(log2(l005.5.old[,3])),
        (log2(l007.5.old[,3])),(log2(l008.5.old[,3])),(log2(l009.5.old[,3])),(log2(l011.5.old[,3])),
        (log2(l015.5.old[,3])),(log2(l028.5.old[,3])),(log2(l029.5[,3])),(log2(l050.5[,3])),(log2(l088.5.old[,3])),(log2(l108.5.old[,3])),(log2(l112.5[,3])),(log2(l115.5[,3])),(log2(l117.5[,3])),(log2(l119.5.old[,3])),(log2(l123.5[,3])),(log2(l152.5[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.5.old$tracking_id))))
chr5DataMA$Line <-factor(chr5DataMA$Line)
aovchr5GC <- lm(y~Line,data=chr5DataGC)
stargazer(aovchr5GC,type="html",out="anovachr5GC.doc",title="Chromosome V Hybrid Strain")
aovchr5MA <- lm(y~Line,data=chr5DataMA)
stargazer(aovchr5MA,type="html",out="anovachr5MA.doc",title="Chromosome V Lab Strain")

#RATIOS FOR CHROMOSOME 5
#euRatio.c5 <- c(l002.5[,3],l003.5[,3],l005.5[,3],l006.5[,3],l008.5[,3],l009.5[,3],l069.5[,3])

#this code works beautifully to produce a table with all of the log2 values for chromosome 1 
# chr5Data <- data.frame(y=c((log2(l001.5[,3])),(log2(l002.5[,3])),(log2(l003.5[,3])),(log2(l004.5[,3])),
#         (log2(l005.5[,3])),(log2(l007.5[,3])),(log2(l008.5[,3])),
#         (log2(l009.5[,3])),(log2(l011.5[,3])),(log2(l018.5[,3])),
#         (log2(l049.5[,3])),(log2(l059.5[,3])),(log2(l061.5[,3])),
#         (log2(l069.5[,3])),(log2(l076.5[,3])),(log2(l077.5[,3])),(log2(l029.5[,3])),(log2(l050.5[,3])),(log2(l112.5[,3])),(log2(l115.5[,3])),(log2(l117.5[,3])),
#         (log2(l123.5[,3])),(log2(l152.5[,3]))),
# Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77,29,50,112,115,117,123,152),each=(length(l001.5$tracking_id))))
# chr5Data$Line <-factor(chr5Data$Line)
# 


#This code works beautifully to produce a boxplot
#want to make this chunk into a loop
#but I can't because I want to add significance stars, which I would have to go back and do anyway, so that is pointless
# chr5DataofInt <- data.frame(y=c((log2(l001.5[,3])),(log2(l002.5[,3])),(log2(l003.5[,3])),(log2(l004.5[,3])),
#         (log2(l005.5[,3])),(log2(l009.5[,3])),(log2(l049.5[,3])),(log2(l069.5[,3]))),Line=rep(c(1,2,3,4,5,9,49,69),each=(length(l001.5$tracking_id))))
# chr5DataofInt$Line <-factor(chr5DataofInt$Line)

# {pdf("Chrm_5.GC.MA.pdf")
# boxplot(y~Line,data=chr5Data,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("darkturquoise"),main="Chromosome 1",las=3,ylim=c(-2,2.5))
# abline(h=0,lty=2)
# abline(h=0.585,lty=2,col="forestgreen")
# abline(h=-1,lty=2,col="darkviolet")
# }

# aov.chr5 <- aov(y~Line,data=chr5DataofInt)
# summary(aov.chr5)
# resids.chr5 <- resid(aov.chr5)
# {pdf("Chrm5.resids.pdf")
# hist(resids.chr5,col="turquoise",breaks=10,xlab="Residuals.Chr.5")}
# #shapiro.test(resids.chr5)
# TukeyHSD(aov.chr5)
# {pdf("chr5.tukey.pdf")
#   plot(TukeyHSD(aov.chr5))}
# 
# t.test((log2(l004.5[,3])),mu=exp3x)
# t.test((log2(l049.5[,3])),mu=exp3x)



#fail shapiro test, so should probably use nonparametric
# kruskal.chr5 <- kruskal.test(y~Line,data=chr5Data)
# kruskal.test(y~Line,data=chr5Data)
# summary(kruskal.chr5)
# #test pairwise comparisons
# #Tukeys doesnt work with kruskal
# #but there is an equivalent for Tukeys with kruskal
# #Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
# kruskalmc(y~Line,data=chr5Data)
# multiCompsChr5 <- kruskalmc(y~Line,data=chr5Data)
# #many pairwise t-tests
# pairwise.t.test(chr5Data$y,chr5Data$Line,p.adjust.method = "fdr")
# wilcox.chr5 <- pairwise.wilcox.test(chr5Data$y,chr5Data$Line,p.adjust.method = "fdr")
# pairwise.wilcox.test(chr5Data$y,chr5Data$Line,p.adjust.method = "fdr")


##############################################################################################
#CHROMOSOME 6
chr6DataGC <- data.frame(y=c((log2(l001.6[,3])),(log2(l002.6[,3])),(log2(l003.6[,3])),(log2(l004.6[,3])),(log2(l005.6[,3])),
        (log2(l007.6[,3])),(log2(l008.6[,3])),(log2(l009.6[,3])),(log2(l011.6[,3])),
        (log2(l018.6[,3])),(log2(l049.6[,3])),(log2(l059.6[,3])),(log2(l061.6[,3])),(log2(l069.6[,3])),(log2(l076.6[,3])),(log2(l077.6[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.6$tracking_id))))
chr6DataGC$Line <-factor(chr6DataGC$Line)
chr6DataMA <- data.frame(y=c((log2(l001.6.old[,3])),(log2(l002.6.old[,3])),(log2(l003.6.old[,3])),(log2(l004.6.old[,3])),(log2(l005.6.old[,3])),
        (log2(l007.6.old[,3])),(log2(l008.6.old[,3])),(log2(l009.6.old[,3])),(log2(l011.6.old[,3])),
        (log2(l015.6.old[,3])),(log2(l028.6.old[,3])),(log2(l029.6[,3])),(log2(l050.6[,3])),(log2(l088.6.old[,3])),(log2(l108.6.old[,3])),(log2(l112.6[,3])),(log2(l115.6[,3])),(log2(l117.6[,3])),(log2(l119.6.old[,3])),(log2(l123.6[,3])),(log2(l152.6[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.6.old$tracking_id))))
chr6DataMA$Line <-factor(chr6DataMA$Line)
aovchr6GC <- lm(y~Line,data=chr6DataGC)
stargazer(aovchr6GC,type="html",out="anovachr6GC.doc",title="Chromosome VI Hybrid Strain")
aovchr6MA <- lm(y~Line,data=chr6DataMA)
stargazer(aovchr6MA,type="html",out="anovachr6MA.doc",title="Chromosome VI Lab Strain")


#RATIOS FOR CHROMOSOME 6
#euRatio.c6 <- c(l002.6[,3],l003.6[,3],l005.6[,3],l006.6[,3],l008.6[,3],l009.6[,3],l069.6[,3])

#this code works beautifully to produce a table with all of the log2 values for chromosome 6 
# chr6Data <- data.frame(y=c((log2(l001.6[,3])),(log2(l002.6[,3])),(log2(l003.6[,3])),(log2(l004.6[,3])),
#         (log2(l005.6[,3])),(log2(l006.6[,3])),(log2(l007.6[,3])),(log2(l008.6[,3])),
#         (log2(l009.6[,3])),(log2(l011.6[,3])),(log2(l018.6[,3])),(log2(l021.6[,3])),
#         (log2(l031.6[,3])),(log2(l049.6[,3])),(log2(l059.6[,3])),(log2(l061.6[,3])),
#         (log2(l066.6[,3])),(log2(l069.6[,3])),(log2(l076.6[,3])),(log2(l077.6[,3]))),
# Line=rep(c(1,2,3,4,5,6,7,8,9,11,18,21,31,49,59,61,66,69,76,77),each=(length(l001.6$tracking_id))))
# chr6Data$Line <-factor(chr6Data$Line)
#This code works beautifully to produce a boxplot
#want to make this chunk into a loop
#but I can't because I want to add significance stars, which I would have to go back and do anyway, so that is pointless
# {pdf("Chrm_6.pdf")
# boxplot(y~Line,data=chr6Data,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","cyan","green","green","cyan","cyan","green","darkcyan","cyan","cyan",
#               "green","cyan","cyan","cyan","cyan","green","cyan","cyan"),main="GC Lines Chromsome 6",las=3,ylim=c(-2,2.5))
# abline(h=0,lty=2)
# abline(h=0.585,lty=2,col="forestgreen")
# abline(h=-1,lty=2,col="darkviolet")}
# 
# aov.chr6 <- aov(y~Line,data=chr6Data)
# summary(aov.chr6)
# resids.chr6 <- resid(aov.chr6)
# {pdf("Chrm6.resids.pdf")
# hist(resids.chr6,col="turquoise",breaks=10,xlab="Residuals.Chr.6")}
# #shapiro.test(resids.chr6)
# TukeyHSD(aov.chr6)
# {pdf("tukey.chr6.pdf")
#  plot(TukeyHSD(aov.chr6)) }

#fail shapiro test, so should probably use nonparametric
# kruskal.chr6 <- kruskal.test(y~Line,data=chr6Data)
# kruskal.test(y~Line,data=chr6Data)
# summary(kruskal.chr6)
# #test pairwise comparisons
# #Tukeys doesnt work with kruskal
# #but there is an equivalent for Tukeys with kruskal
# #Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
# kruskalmc(y~Line,data=chr6Data)
# multiCompsChr6 <- kruskalmc(y~Line,data=chr6Data)
# #many pairwise t-tests
# pairwise.t.test(chr6Data$y,chr6Data$Line,p.adjust.method = "fdr")
# wilcox.chr6 <- pairwise.wilcox.test(chr6Data$y,chr6Data$Line,p.adjust.method = "fdr")
# pairwise.wilcox.test(chr6Data$y,chr6Data$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 7
chr7DataGC <- data.frame(y=c((log2(l001.7[,3])),(log2(l002.7[,3])),(log2(l003.7[,3])),(log2(l004.7[,3])),(log2(l005.7[,3])),
        (log2(l007.7[,3])),(log2(l008.7[,3])),(log2(l009.7[,3])),(log2(l011.7[,3])),
        (log2(l018.7[,3])),(log2(l049.7[,3])),(log2(l059.7[,3])),(log2(l061.7[,3])),(log2(l069.7[,3])),(log2(l076.7[,3])),(log2(l077.7[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.7$tracking_id))))
chr7DataGC$Line <-factor(chr7DataGC$Line)
chr7DataMA <- data.frame(y=c((log2(l001.7.old[,3])),(log2(l002.7.old[,3])),(log2(l003.7.old[,3])),(log2(l004.7.old[,3])),(log2(l005.7.old[,3])),
        (log2(l007.7.old[,3])),(log2(l008.7.old[,3])),(log2(l009.7.old[,3])),(log2(l011.7.old[,3])),
        (log2(l015.7.old[,3])),(log2(l028.7.old[,3])),(log2(l029.7[,3])),(log2(l050.7[,3])),(log2(l088.7.old[,3])),(log2(l108.7.old[,3])),(log2(l112.7[,3])),(log2(l115.7[,3])),(log2(l117.7[,3])),(log2(l119.7.old[,3])),(log2(l123.7[,3])),(log2(l152.7[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.7.old$tracking_id))))
chr7DataMA$Line <-factor(chr7DataMA$Line)
aovchr7GC <- lm(y~Line,data=chr7DataGC)
stargazer(aovchr7GC,type="html",out="anovachr7GC.doc",title="Chromosome VII Hybrid Strain")
aovchr7MA <- lm(y~Line,data=chr7DataMA)
stargazer(aovchr7MA,type="html",out="anovachr7MA.doc",title="Chromosome VII Lab Strain")
#RATIOS FOR CHROMOSOME 7
#euRatio.c7 <- c(l002.7[,3],l003.7[,3],l005.7[,3],l006.7[,3],l008.7[,3],l009.7[,3],l069.7[,3])

#this code works beautifully to produce a table with all of the log2 values for chromosome 6 
# chr7Data <- data.frame(y=c((log2(l001.7[,3])),(log2(l002.7[,3])),(log2(l003.7[,3])),(log2(l004.7[,3])),
#         (log2(l005.7[,3])),(log2(l006.7[,3])),(log2(l007.7[,3])),(log2(l008.7[,3])),
#         (log2(l009.7[,3])),(log2(l011.7[,3])),(log2(l018.7[,3])),(log2(l021.7[,3])),
#         (log2(l031.7[,3])),(log2(l049.7[,3])),(log2(l059.7[,3])),(log2(l061.7[,3])),
#         (log2(l066.7[,3])),(log2(l069.7[,3])),(log2(l076.7[,3])),(log2(l077.7[,3]))),
# Line=rep(c(1,2,3,4,5,6,7,8,9,11,18,21,31,49,59,61,66,69,76,77),each=(length(l001.7$tracking_id))))
# chr7Data$Line <-factor(chr7Data$Line)
# #This code works beautifully to produce a boxplot
# #want to make this chunk into a loop
# #but I can't because I want to add significance stars, which I would have to go back and do anyway, so that is pointless
# {pdf("Chrm_7.pdf")
# boxplot(y~Line,data=chr7Data,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","cyan","green","green","cyan","cyan","green","darkcyan","cyan","cyan",
#               "green","cyan","deeppink1","deeppink1","deeppink1","green","cyan","cyan"),main="GC Lines Chromsome 7",las=3,ylim=c(-2,2.5))
# abline(h=0,lty=2)
# abline(h=0.585,lty=2,col="forestgreen")
# abline(h=-1,lty=2,col="darkviolet")
#need stars for 59,61,66 and comparisons between them 
#p3 <- 0.0006221
#mylabel = bquote(italic(p) == .(format(p3, digits = 9)))
#text(x=2.3,y = 2.5, labels = mylabel,cex=.8)
# text(x=14.7, y=.8, "*", pos=3, cex=2, col="red")
# text(x=15, y=.8, "*", pos=3, cex=2, col="red")
# text(x=15.3, y=.8, "*", pos=3, cex=2, col="red")
# text(x=15.7, y=.9, "*", pos=3, cex=2, col="red")
# text(x=16, y=.9, "*", pos=3, cex=2, col="red")
# text(x=16.3, y=.9, "*", pos=3, cex=2, col="red")
# text(x=16.7, y=1, "*", pos=3, cex=2, col="red")
# text(x=17, y=1, "*", pos=3, cex=2, col="red")
# text(x=17.3, y=1, "*", pos=3, cex=2, col="red")



# brackets(16, 1.9, 17, 1.9, h = .3,  ticks = 0.5, curvature = 0.5, type = 1,
# col = 1, lwd = 1, lty = 1, xpd = FALSE)
# 
# text(x=16.2, y=2, "*", pos=3, cex=2, col="red")
# text(x=16.5, y=2, "*", pos=3, cex=2, col="red")
# text(x=16.8, y=2, "*", pos=3, cex=2, col="red")
# 
# brackets(15, 2.1, 17, 2.1, h = .3,  ticks = 0.5, curvature = 0.5, type = 1,
# col = 1, lwd = 1, lty = 1, xpd = FALSE)
# 
# text(x=15.7, y=2.2, "*", pos=3, cex=2, col="red")
# text(x=16, y=2.2, "*", pos=3, cex=2, col="red")
# text(x=16.3, y=2.2, "*", pos=3, cex=2, col="red")
# }
# 
# chr7Data.red <- data.frame(y=c((log2(l001.7[,3])),(log2(l002.7[,3])),(log2(l003.7[,3])),
#         (log2(l005.7[,3])),(log2(l006.7[,3])),
#         (log2(l009.7[,3])),
#         (log2(l031.7[,3])),(log2(l059.7[,3])),(log2(l061.7[,3])),
#         (log2(l066.7[,3])),(log2(l069.7[,3]))),
# Line=rep(c(1,2,3,5,6,9,31,59,61,66,69),each=(length(l001.7$tracking_id))))
# chr7Data.red$Line <-factor(chr7Data.red$Line)
# 
# aov.chr7 <- aov(y~Line,data=chr7Data.red)
# summary(aov.chr7)
# resids.chr7 <- resid(aov.chr7)
# {pdf("Chrm7.resids.pdf")
# hist(resids.chr7,col="turquoise",breaks=10,xlab="Residuals.Chr.7")}
# 
# TukeyHSD(aov.chr7)
# {pdf("Tukey.chr7.pdf")
# plot(TukeyHSD(aov.chr7))}
# 
# t.test((log2(l059.7[,3])),mu=exp3x)
# t.test((log2(l061.7[,3])),mu=exp3x)
# # t.test((log2(l066.7[,3])),mu=exp2.5x)
# exp2.5x <- log2(1.25)
# exp2.5x
# #fail shapiro test, so should probably use nonparametric
# kruskal.chr7 <- kruskal.test(y~Line,data=chr7Data.red)
# kruskal.test(y~Line,data=chr7Data.red)
# summary(kruskal.chr7)
#test pairwise comparisons
#Tukeys doesnt work with kruskal
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
# kruskalmc(y~Line,data=chr7Data.red)
# multiCompsChr7 <- kruskalmc(y~Line,data=chr7Data.red)
# #many pairwise t-tests
# pairwise.t.test(chr7Data.red$y,chr7Data.red$Line,p.adjust.method = "fdr")
# wilcox.chr7 <- pairwise.wilcox.test(chr7Data.red$y,chr7Data.red$Line,p.adjust.method = "fdr")
# pairwise.wilcox.test(chr7Data.red$y,chr7Data.red$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 8
chr8DataGC <- data.frame(y=c((log2(l001.8[,3])),(log2(l002.8[,3])),(log2(l003.8[,3])),(log2(l004.8[,3])),(log2(l005.8[,3])),
        (log2(l007.8[,3])),(log2(l008.8[,3])),(log2(l009.8[,3])),(log2(l011.8[,3])),
        (log2(l018.8[,3])),(log2(l049.8[,3])),(log2(l059.8[,3])),(log2(l061.8[,3])),(log2(l069.8[,3])),(log2(l076.8[,3])),(log2(l077.8[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.8$tracking_id))))
chr8DataGC$Line <-factor(chr8DataGC$Line)
chr8DataMA <- data.frame(y=c((log2(l001.8.old[,3])),(log2(l002.8.old[,3])),(log2(l003.8.old[,3])),(log2(l004.8.old[,3])),(log2(l005.8.old[,3])),
        (log2(l007.8.old[,3])),(log2(l008.8.old[,3])),(log2(l009.8.old[,3])),(log2(l011.8.old[,3])),
        (log2(l015.8.old[,3])),(log2(l028.8.old[,3])),(log2(l029.8[,3])),(log2(l050.8[,3])),(log2(l088.8.old[,3])),(log2(l108.8.old[,3])),(log2(l112.8[,3])),(log2(l115.8[,3])),(log2(l117.8[,3])),(log2(l119.8.old[,3])),(log2(l123.8[,3])),(log2(l152.8[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.8.old$tracking_id))))
chr8DataMA$Line <-factor(chr8DataMA$Line)
aovchr8GC <- lm(y~Line,data=chr8DataGC)
stargazer(aovchr8GC,type="html",out="anovachr8GC.doc",title="Chromosome VIII Hybrid Strain")
aovchr8MA <- lm(y~Line,data=chr8DataMA)
stargazer(aovchr8MA,type="html",out="anovachr8MA.doc",title="Chromosome VIII Lab Strain")

#RATIOS FOR CHROMOSOME 8
#euRatio.c8 <- c(l002.8[,3],l003.8[,3],l005.8[,3],l006.8[,3],l008.8[,3],l009.8[,3],l069.8[,3])

#this code works beautifully to produce a table with all of the log2 values for chromosome 6 
# chr8Data <- data.frame(y=c((log2(l001.8[,3])),(log2(l002.8[,3])),(log2(l003.8[,3])),(log2(l004.8[,3])),
#         (log2(l005.8[,3])),(log2(l006.8[,3])),(log2(l007.8[,3])),(log2(l008.8[,3])),
#         (log2(l009.8[,3])),(log2(l011.8[,3])),(log2(l018.8[,3])),(log2(l021.8[,3])),
#         (log2(l031.8[,3])),(log2(l049.8[,3])),(log2(l059.8[,3])),(log2(l061.8[,3])),
#         (log2(l066.8[,3])),(log2(l069.8[,3])),(log2(l076.8[,3])),(log2(l077.8[,3]))),
# Line=rep(c(1,2,3,4,5,6,7,8,9,11,18,21,31,49,59,61,66,69,76,77),each=(length(l001.8$tracking_id))))
# chr8Data$Line <-factor(chr8Data$Line)
#This code works beautifully to produce a boxplot
#want to make this chunk into a loop
#but I can't because I want to add significance stars, which I would have to go back and do anyway, so that is pointless
# {
#   pdf("Chrm_8.pdf")
# boxplot(y~Line,data=chr8Data,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","cyan","green","green","cyan","cyan","green","darkcyan","cyan","cyan",
#               "green","cyan","cyan","cyan","cyan","green","cyan","cyan"),main="GC Lines Chromsome 8",las=3,ylim=c(-2,2.5))
# abline(h=0,lty=2)
# abline(h=0.585,lty=2,col="forestgreen")
# abline(h=-1,lty=2,col="darkviolet")
# }
# 
# aov.chr8 <- aov(y~Line,data=chr8Data)
# summary(aov.chr8)
# # resids.chr8 <- resid(aov.chr8)
# {
#   pdf("Chrm8.resids.pdf")
# hist(resids.chr8,col="turquoise",breaks=10,xlab="Residuals.Chr.8")
# }
# 
# TukeyHSD(aov.chr8)
# {pdf("tukey.chr8.pdf")
#   plot(TukeyHSD(aov.chr8))}
# 
# #fail shapiro test, so should probably use nonparametric
# kruskal.chr8 <- kruskal.test(y~Line,data=chr8Data)
# kruskal.test(y~Line,data=chr8Data)
# summary(kruskal.chr8)
#test pairwise comparisons
#Tukeys doesnt work with kruskal
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
# kruskalmc(y~Line,data=chr8Data)
# multiCompsChr8 <- kruskalmc(y~Line,data=chr8Data)
# #many pairwise t-tests
# pairwise.t.test(chr8Data$y,chr8Data$Line,p.adjust.method = "fdr")
# wilcox.chr8 <- pairwise.wilcox.test(chr8Data$y,chr8Data$Line,p.adjust.method = "fdr")
# pairwise.wilcox.test(chr8Data$y,chr8Data$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 9
chr9DataGC <- data.frame(y=c((log2(l001.9[,3])),(log2(l002.9[,3])),(log2(l003.9[,3])),(log2(l004.9[,3])),(log2(l005.9[,3])),
        (log2(l007.9[,3])),(log2(l008.9[,3])),(log2(l009.9[,3])),(log2(l011.9[,3])),
        (log2(l018.9[,3])),(log2(l049.9[,3])),(log2(l059.9[,3])),(log2(l061.9[,3])),(log2(l069.9[,3])),(log2(l076.9[,3])),(log2(l077.9[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.9$tracking_id))))
chr9DataGC$Line <-factor(chr9DataGC$Line)
chr9DataMA <- data.frame(y=c((log2(l001.9.old[,3])),(log2(l002.9.old[,3])),(log2(l003.9.old[,3])),(log2(l004.9.old[,3])),(log2(l005.9.old[,3])),
        (log2(l007.9.old[,3])),(log2(l008.9.old[,3])),(log2(l009.9.old[,3])),(log2(l011.9.old[,3])),
        (log2(l015.9.old[,3])),(log2(l028.9.old[,3])),(log2(l029.9[,3])),(log2(l050.9[,3])),(log2(l088.9.old[,3])),(log2(l108.9.old[,3])),(log2(l112.9[,3])),(log2(l115.9[,3])),(log2(l117.9[,3])),(log2(l119.9.old[,3])),(log2(l123.9[,3])),(log2(l152.9[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.9.old$tracking_id))))
chr9DataMA$Line <-factor(chr9DataMA$Line)
aovchr9GC <- lm(y~Line,data=chr9DataGC)
stargazer(aovchr9GC,type="html",out="anovachr9GC.doc",title="Chromosome IX Hybrid Strain")
aovchr9MA <- lm(y~Line,data=chr9DataMA)
stargazer(aovchr9MA,type="html",out="anovachr9MA.doc",title="Chromosome IX Lab Strain")

#RATIOS FOR CHROMOSOME 9
#euRatio.c9 <- c(l002.9[,3],l003.9[,3],l005.9[,3],l006.9[,3],l008.9[,3],l009.9[,3],l069.9[,3])

#this code works beautifully to produce a table with all of the log2 values for chromosome 6 
# chr9Data <- data.frame(y=c((log2(l001.9[,3])),(log2(l002.9[,3])),(log2(l003.9[,3])),(log2(l004.9[,3])),
#         (log2(l005.9[,3])),(log2(l006.9[,3])),(log2(l007.9[,3])),(log2(l008.9[,3])),
#         (log2(l009.9[,3])),(log2(l011.9[,3])),(log2(l018.9[,3])),(log2(l021.9[,3])),
#         (log2(l031.9[,3])),(log2(l049.9[,3])),(log2(l059.9[,3])),(log2(l061.9[,3])),
#         (log2(l066.9[,3])),(log2(l069.9[,3])),(log2(l076.9[,3])),(log2(l077.9[,3]))),
# Line=rep(c(1,2,3,4,5,6,7,8,9,11,18,21,31,49,59,61,66,69,76,77),each=(length(l001.9$tracking_id))))
# chr9Data$Line <-factor(chr9Data$Line)
#This code works beautifully to produce a boxplot
#want to make this chunk into a loop
#but I can't because I want to add significance stars, which I would have to go back and do anyway, so that is pointless
# {
#   pdf("Chrm_9.pdf")
# boxplot(y~Line,data=chr9Data,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","cyan","green","green","cyan","cyan","green","darkcyan","cyan","cyan",
#               "green","cyan","cyan","cyan","cyan","green","deeppink1","cyan"),main="GC Lines Chromsome 9",las=3,ylim=c(-2,2.5))
# abline(h=0,lty=2)
# abline(h=0.585,lty=2,col="forestgreen")
# abline(h=-1,lty=2,col="darkviolet")
# text(x=18.7, y=1.9, "*", pos=3, cex=2, col="red")
# text(x=19, y=1.9, "*", pos=3, cex=2, col="red")
# text(x=19.3, y=1.9, "*", pos=3, cex=2, col="red")
# }
# chr9Data.r <- data.frame(y=c((log2(l001.9[,3])),(log2(l002.9[,3])),(log2(l003.9[,3])),
#         (log2(l005.9[,3])),(log2(l006.9[,3])),
#         (log2(l009.9[,3])),
#         (log2(l031.9[,3])),
#         (log2(l069.9[,3])),(log2(l076.9[,3]))),
# Line=rep(c(1,2,3,5,6,9,31,69,76),each=(length(l001.9$tracking_id))))
# chr9Data.r$Line <-factor(chr9Data.r$Line)


# aov.chr9 <- aov(y~Line,data=chr9Data.r)
# summary(aov.chr9)
# resids.chr9 <- resid(aov.chr9)
# {
#   pdf("Chrm9.resids.pdf")
# hist(resids.chr9,col="turquoise",breaks=10,xlab="Residuals.Chr.9")
# }
# 
# TukeyHSD(aov.chr9)
# {pdf("tukey.chr9.pdf")
#   plot(TukeyHSD(aov.chr9))}
# 
# 
# t.test((log2(l076.9[,3])),mu=exp3x)


#fail shapiro test, so should probably use nonparametric
# kruskal.chr9 <- kruskal.test(y~Line,data=chr9Data)
# kruskal.test(y~Line,data=chr9Data)
# summary(kruskal.chr9)
# #test pairwise comparisons
# #Tukeys doesnt work with kruskal
# #but there is an equivalent for Tukeys with kruskal
# #Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
# kruskalmc(y~Line,data=chr9Data)
# multiCompsChr9 <- kruskalmc(y~Line,data=chr9Data)
# #many pairwise t-tests
# pairwise.t.test(chr9Data$y,chr9Data$Line,p.adjust.method = "fdr")
# wilcox.chr9 <- pairwise.wilcox.test(chr9Data$y,chr9Data$Line,p.adjust.method = "fdr")
# pairwise.wilcox.test(chr9Data$y,chr9Data$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 10
chr10DataGC <- data.frame(y=c((log2(l001.10[,3])),(log2(l002.10[,3])),(log2(l003.10[,3])),(log2(l004.10[,3])),(log2(l005.10[,3])),
        (log2(l007.10[,3])),(log2(l008.10[,3])),(log2(l009.10[,3])),(log2(l011.10[,3])),
        (log2(l018.10[,3])),(log2(l049.10[,3])),(log2(l059.10[,3])),(log2(l061.10[,3])),(log2(l069.10[,3])),(log2(l076.10[,3])),(log2(l077.10[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.10$tracking_id))))
chr10DataGC$Line <-factor(chr10DataGC$Line)
chr10DataMA <- data.frame(y=c((log2(l001.10.old[,3])),(log2(l002.10.old[,3])),(log2(l003.10.old[,3])),(log2(l004.10.old[,3])),(log2(l005.10.old[,3])),
        (log2(l007.10.old[,3])),(log2(l008.10.old[,3])),(log2(l009.10.old[,3])),(log2(l011.10.old[,3])),
        (log2(l015.10.old[,3])),(log2(l028.10.old[,3])),(log2(l029.10[,3])),(log2(l050.10[,3])),(log2(l088.10.old[,3])),(log2(l108.10.old[,3])),(log2(l112.10[,3])),(log2(l115.10[,3])),(log2(l117.10[,3])),(log2(l119.10.old[,3])),(log2(l123.10[,3])),(log2(l152.10[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.10.old$tracking_id))))
chr10DataMA$Line <-factor(chr10DataMA$Line)
aovchr10GC <- lm(y~Line,data=chr10DataGC)
stargazer(aovchr10GC,type="html",out="anovachr10GC.doc",title="Chromosome X Hybrid Strain")
aovchr10MA <- lm(y~Line,data=chr10DataMA)
stargazer(aovchr10MA,type="html",out="anovachr10MA.doc",title="Chromosome X Lab Strain")

#RATIOS FOR CHROMOSOME 10
#euRatio.c10 <- c(l002.10[,3],l003.10[,3],l005.10[,3],l006.10[,3],l008.10[,3],l009.10[,3],l069.10[,3])

#this code works beautifully to produce a table with all of the log2 values for chromosome 6 
# chr10Data <- data.frame(y=c((log2(l001.10[,3])),(log2(l002.10[,3])),(log2(l003.10[,3])),(log2(l004.10[,3])),
#         (log2(l005.10[,3])),(log2(l006.10[,3])),(log2(l007.10[,3])),(log2(l008.10[,3])),
#         (log2(l009.10[,3])),(log2(l011.10[,3])),(log2(l018.10[,3])),(log2(l021.10[,3])),
#         (log2(l031.10[,3])),(log2(l049.10[,3])),(log2(l059.10[,3])),(log2(l061.10[,3])),
#         (log2(l066.10[,3])),(log2(l069.10[,3])),(log2(l076.10[,3])),(log2(l077.10[,3]))),
# Line=rep(c(1,2,3,4,5,6,7,8,9,11,18,21,31,49,59,61,66,69,76,77),each=(length(l001.10$tracking_id))))
# chr10Data$Line <-factor(chr10Data$Line)
#This code works beautifully to produce a boxplot
#want to make this chunk into a loop
#but I can't because I want to add significance stars, which I would have to go back and do anyway, so that is pointless
# {
#   pdf("Chrm_10.pdf")
# boxplot(y~Line,data=chr10Data,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","cyan","green","green","cyan","cyan","green","darkcyan","cyan","cyan",
#               "green","cyan","cyan","cyan","cyan","green","deeppink1","cyan"),main="GC Lines Chromsome 10",las=3,ylim=c(-2,2.5))
# abline(h=0,lty=2)
# abline(h=0.585,lty=2,col="forestgreen")
# abline(h=-1,lty=2,col="darkviolet")
# 
# text(x=18.7, y=1.9, "*", pos=3, cex=2, col="red")
# text(x=19, y=1.9, "*", pos=3, cex=2, col="red")
# text(x=19.3, y=1.9, "*", pos=3, cex=2, col="red")
# }
# chr10Data.r <- data.frame(y=c((log2(l001.10[,3])),(log2(l002.10[,3])),(log2(l003.10[,3])),
#         (log2(l005.10[,3])),(log2(l006.10[,3])),
#         (log2(l009.10[,3])),
#         (log2(l031.10[,3])),
#         (log2(l069.10[,3])),(log2(l076.10[,3]))),
# Line=rep(c(1,2,3,5,6,9,31,69,76),each=(length(l001.10$tracking_id))))
# chr10Data.r$Line <-factor(chr10Data.r$Line)
# 
# 
# aov.chr10 <- aov(y~Line,data=chr10Data.r)
# summary(aov.chr10)
# resids.chr10 <- resid(aov.chr10)
# {
#   pdf("Chrm10.resids.pdf")
# hist(resids.chr10,col="turquoise",breaks=10,xlab="Residuals.Chr.10")
# }
# 
# t.test((log2(l076.10[,3])),mu=exp2.5x)
# 

# TukeyHSD(aov.chr10)
# {pdf("tukey.chr10.pdf")
# plot(TukeyHSD(aov.chr10))}
# #fail shapiro test, so should probably use nonparametric
# kruskal.chr10 <- kruskal.test(y~Line,data=chr10Data)
# kruskal.test(y~Line,data=chr10Data)
# summary(kruskal.chr10)
# #test pairwise comparisons
# #Tukeys doesnt work with kruskal
# #but there is an equivalent for Tukeys with kruskal
# #Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
# kruskalmc(y~Line,data=chr10Data)
# multiCompsChr10 <- kruskalmc(y~Line,data=chr10Data)
# #many pairwise t-tests
# pairwise.t.test(chr10Data$y,chr10Data$Line,p.adjust.method = "fdr")
# wilcox.chr10 <- pairwise.wilcox.test(chr10Data$y,chr10Data$Line,p.adjust.method = "fdr")
# pairwise.wilcox.test(chr10Data$y,chr10Data$Line,p.adjust.method = "fdr")


##############################################################################################
#CHROMOSOME 11
chr11DataGC <- data.frame(y=c((log2(l001.11[,3])),(log2(l002.11[,3])),(log2(l003.11[,3])),(log2(l004.11[,3])),(log2(l005.11[,3])),
        (log2(l007.11[,3])),(log2(l008.11[,3])),(log2(l009.11[,3])),(log2(l011.11[,3])),
        (log2(l018.11[,3])),(log2(l049.11[,3])),(log2(l059.11[,3])),(log2(l061.11[,3])),(log2(l069.11[,3])),(log2(l076.11[,3])),(log2(l077.11[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.11$tracking_id))))
chr11DataGC$Line <-factor(chr11DataGC$Line)
chr11DataMA <- data.frame(y=c((log2(l001.11.old[,3])),(log2(l002.11.old[,3])),(log2(l003.11.old[,3])),(log2(l004.11.old[,3])),(log2(l005.11.old[,3])),
        (log2(l007.11.old[,3])),(log2(l008.11.old[,3])),(log2(l009.11.old[,3])),(log2(l011.11.old[,3])),
        (log2(l015.11.old[,3])),(log2(l028.11.old[,3])),(log2(l029.11[,3])),(log2(l050.11[,3])),(log2(l088.11.old[,3])),(log2(l108.11.old[,3])),(log2(l112.11[,3])),(log2(l115.11[,3])),(log2(l117.11[,3])),(log2(l119.11.old[,3])),(log2(l123.11[,3])),(log2(l152.11[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.11.old$tracking_id))))
chr11DataMA$Line <-factor(chr11DataMA$Line)
aovchr11GC <- lm(y~Line,data=chr11DataGC)
stargazer(aovchr11GC,type="html",out="anovachr11GC.doc",title="Chromosome XI Hybrid Strain")
aovchr11MA <- lm(y~Line,data=chr11DataMA)
stargazer(aovchr11MA,type="html",out="anovachr11MA.doc",title="Chromosome XI Lab Strain")

#RATIOS FOR CHROMOSOME 11
#euRatio.c11 <- c(l002.11[,3],l003.11[,3],l005.11[,3],l006.11[,3],l008.11[,3],l009.11[,3],l069.11[,3])

#this code works beautifully to produce a table with all of the log2 values for chromosome 6 
# chr11Data <- data.frame(y=c((log2(l001.11[,3])),(log2(l002.11[,3])),(log2(l003.11[,3])),(log2(l004.11[,3])),
#         (log2(l005.11[,3])),(log2(l006.11[,3])),(log2(l007.11[,3])),(log2(l008.11[,3])),
#         (log2(l009.11[,3])),(log2(l011.11[,3])),(log2(l018.11[,3])),(log2(l021.11[,3])),
#         (log2(l031.11[,3])),(log2(l049.11[,3])),(log2(l059.11[,3])),(log2(l061.11[,3])),
#         (log2(l066.11[,3])),(log2(l069.11[,3])),(log2(l076.11[,3])),(log2(l077.11[,3]))),
# Line=rep(c(1,2,3,4,5,6,7,8,9,11,18,21,31,49,59,61,66,69,76,77),each=(length(l001.11$tracking_id))))
# chr11Data$Line <-factor(chr11Data$Line)
# #This code works beautifully to produce a boxplot
# #want to make this chunk into a loop
# #but I can't because I want to add significance stars, which I would have to go back and do anyway, so that is pointless
# {
#   pdf("Chrm_11.pdf")
# boxplot(y~Line,data=chr11Data,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","cyan","green","green","cyan","cyan","green","darkcyan","cyan","cyan",
#               "green","cyan","cyan","cyan","cyan","green","cyan","cyan"),main="GC Lines Chromsome 11",las=3,ylim=c(-2,2.5))
# abline(h=0,lty=2)
# abline(h=0.585,lty=2,col="forestgreen")
# abline(h=-1,lty=2,col="darkviolet")
# 
# }
# 
# 
# aov.chr11 <- aov(y~Line,data=chr11Data)
# summary(aov.chr11)
# resids.chr11 <- resid(aov.chr11)
# {
#   pdf("Chrm11.resids.pdf")
# hist(resids.chr11,col="turquoise",breaks=10,xlab="Residuals.Chr.11")
# }
# 
# 
# 
# TukeyHSD(aov.chr11)
# {pdf("tukey.chr11.pdf")
#   plot(TukeyHSD(aov.chr11))}

#fail shapiro test, so should probably use nonparametric
#kruskal.chr11 <- kruskal.test(y~Line,data=chr11Data)
#kruskal.test(y~Line,data=chr11Data)
#summary(kruskal.chr11)
#test pairwise comparisons
#Tukeys doesnt work with kruskal
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr11Data)
#multiCompsChr11 <- kruskalmc(y~Line,data=chr11Data)
#many pairwise t-tests
#pairwise.t.test(chr11Data$y,chr11Data$Line,p.adjust.method = "fdr")
#wilcox.chr11 <- pairwise.wilcox.test(chr11Data$y,chr11Data$Line,p.adjust.method = "fdr")
#pairwise.wilcox.test(chr11Data$y,chr11Data$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 12
chr12DataGC <- data.frame(y=c((log2(l001.12[,3])),(log2(l002.12[,3])),(log2(l003.12[,3])),(log2(l004.12[,3])),(log2(l005.12[,3])),
        (log2(l007.12[,3])),(log2(l008.12[,3])),(log2(l009.12[,3])),(log2(l011.12[,3])),
        (log2(l018.12[,3])),(log2(l049.12[,3])),(log2(l059.12[,3])),(log2(l061.12[,3])),(log2(l069.12[,3])),(log2(l076.12[,3])),(log2(l077.12[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.12$tracking_id))))
chr12DataGC$Line <-factor(chr12DataGC$Line)
chr12DataMA <- data.frame(y=c((log2(l001.12.old[,3])),(log2(l002.12.old[,3])),(log2(l003.12.old[,3])),(log2(l004.12.old[,3])),(log2(l005.12.old[,3])),
        (log2(l007.12.old[,3])),(log2(l008.12.old[,3])),(log2(l009.12.old[,3])),(log2(l011.12.old[,3])),
        (log2(l015.12.old[,3])),(log2(l028.12.old[,3])),(log2(l029.12[,3])),(log2(l050.12[,3])),(log2(l088.12.old[,3])),(log2(l108.12.old[,3])),(log2(l112.12[,3])),(log2(l115.12[,3])),(log2(l117.12[,3])),(log2(l119.12.old[,3])),(log2(l123.12[,3])),(log2(l152.12[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.12.old$tracking_id))))
chr12DataMA$Line <-factor(chr12DataMA$Line)
aovchr12GC <- lm(y~Line,data=chr12DataGC)
stargazer(aovchr12GC,type="html",out="anovachr12GC.doc",title="Chromosome XII Hybrid Strain")
aovchr12MA <- lm(y~Line,data=chr12DataMA)
stargazer(aovchr12MA,type="html",out="anovachr12MA.doc",title="Chromosome XII Lab Strain")

#RATIOS FOR CHROMOSOME 12
#euRatio.c12 <- c(l002.12[,3],l003.12[,3],l005.12[,3],l006.12[,3],l008.12[,3],l009.12[,3],l069.12[,3])

#this code works beautifully to produce a table with all of the log2 values for chromosome 6 
# chr12Data <- data.frame(y=c((log2(l001.12[,3])),(log2(l002.12[,3])),(log2(l003.12[,3])),(log2(l004.12[,3])),
#         (log2(l005.12[,3])),(log2(l006.12[,3])),(log2(l007.12[,3])),(log2(l008.12[,3])),
#         (log2(l009.12[,3])),(log2(l011.12[,3])),(log2(l018.12[,3])),(log2(l021.12[,3])),
#         (log2(l031.12[,3])),(log2(l049.12[,3])),(log2(l059.12[,3])),(log2(l061.12[,3])),
#         (log2(l066.12[,3])),(log2(l069.12[,3])),(log2(l076.12[,3])),(log2(l077.12[,3]))),
# Line=rep(c(1,2,3,4,5,6,7,8,9,11,18,21,31,49,59,61,66,69,76,77),each=(length(l001.12$tracking_id))))
# chr12Data$Line <-factor(chr12Data$Line)
#This code works beautifully to produce a boxplot
#want to make this chunk into a loop
#but I can't because I want to add significance stars, which I would have to go back and do anyway, so that is pointless
# {
#   pdf("Chrm_12.pdf")
# boxplot(y~Line,data=chr12Data,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","cyan","green","green","cyan","cyan","green","darkcyan","deeppink1","cyan",
#               "green","cyan","cyan","cyan","cyan","green","cyan","deeppink1"),main="GC Lines Chromsome 12",las=3,ylim=c(-2,2.5))
# abline(h=0,lty=2)
# abline(h=0.585,lty=2,col="forestgreen")
# abline(h=-1,lty=2,col="darkviolet")
# 
# text(x=10.7, y=1.8, "*", pos=3, cex=2, col="red")
# text(x=11, y=1.8, "*", pos=3, cex=2, col="red")
# text(x=11.3, y=1.8, "*", pos=3, cex=2, col="red")
# 
# text(x=19.7, y=1.9, "*", pos=3, cex=2, col="red")
# text(x=20, y=1.9, "*", pos=3, cex=2, col="red")
# text(x=20.3, y=1.9, "*", pos=3, cex=2, col="red")
# 
# }
# 
# chr12Data.r <- data.frame(y=c((log2(l001.12[,3])),(log2(l002.12[,3])),(log2(l003.12[,3])),
#         (log2(l005.12[,3])),(log2(l006.12[,3])),
#         (log2(l009.12[,3])),(log2(l018.12[,3])),
#         (log2(l031.12[,3])),
#         (log2(l069.12[,3])),(log2(l077.12[,3]))),
# Line=rep(c(1,2,3,5,6,9,18,31,69,77),each=(length(l001.12$tracking_id))))
# chr12Data.r$Line <-factor(chr12Data.r$Line)
# 
# aov.chr12 <- aov(y~Line,data=chr12Data.r)
# summary(aov.chr12)
# resids.chr12 <- resid(aov.chr12)
# {
#   pdf("Chrm12.resids.pdf")
# hist(resids.chr12,col="turquoise",breaks=10,xlab="Residuals.Chr.12")
# }

# t.test((log2(l018.12[,3])),mu=exp3x)
# t.test((log2(l077.12[,3])),mu=exp3x)
# 
# TukeyHSD(aov.chr12)
# {pdf("tukey.chr12.pdf")
#   plot(TukeyHSD(aov.chr12))}
# #fail shapiro test, so should probably use nonparametric
# kruskal.chr12 <- kruskal.test(y~Line,data=chr12Data)
# kruskal.test(y~Line,data=chr12Data)
# summary(kruskal.chr12)
# #test pairwise comparisons
# #Tukeys doesnt work with kruskal
# #but there is an equivalent for Tukeys with kruskal
# #Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
# kruskalmc(y~Line,data=chr12Data)
# multiCompsChr12 <- kruskalmc(y~Line,data=chr12Data)
# #many pairwise t-tests
# pairwise.t.test(chr12Data$y,chr12Data$Line,p.adjust.method = "fdr")
# wilcox.chr12 <- pairwise.wilcox.test(chr12Data$y,chr12Data$Line,p.adjust.method = "fdr")
# pairwise.wilcox.test(chr12Data$y,chr12Data$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 13
chr13DataGC <- data.frame(y=c((log2(l001.13[,3])),(log2(l002.13[,3])),(log2(l003.13[,3])),(log2(l004.13[,3])),(log2(l005.13[,3])),
        (log2(l007.13[,3])),(log2(l008.13[,3])),(log2(l009.13[,3])),(log2(l011.13[,3])),
        (log2(l018.13[,3])),(log2(l049.13[,3])),(log2(l059.13[,3])),(log2(l061.13[,3])),(log2(l069.13[,3])),(log2(l076.13[,3])),(log2(l077.13[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.13$tracking_id))))
chr13DataGC$Line <-factor(chr13DataGC$Line)
chr13DataMA <- data.frame(y=c((log2(l001.13.old[,3])),(log2(l002.13.old[,3])),(log2(l003.13.old[,3])),(log2(l004.13.old[,3])),(log2(l005.13.old[,3])),
        (log2(l007.13.old[,3])),(log2(l008.13.old[,3])),(log2(l009.13.old[,3])),(log2(l011.13.old[,3])),
        (log2(l015.13.old[,3])),(log2(l028.13.old[,3])),(log2(l029.13[,3])),(log2(l050.13[,3])),(log2(l088.13.old[,3])),(log2(l108.13.old[,3])),(log2(l112.13[,3])),(log2(l115.13[,3])),(log2(l117.13[,3])),(log2(l119.13.old[,3])),(log2(l123.13[,3])),(log2(l152.13[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.13.old$tracking_id))))
chr13DataMA$Line <-factor(chr13DataMA$Line)

aovchr13GC <- lm(y~Line,data=chr13DataGC)
stargazer(aovchr13GC,type="html",out="anovachr13GC.doc",title="Chromosome XIII Hybrid Strain")

aovchr13MA <- lm(y~Line,data=chr13DataMA)
stargazer(aovchr13MA,type="html",out="anovachr13MA.doc",title="Chromosome XIII Lab Strain")

#RATIOS FOR CHROMOSOME 13
#euRatio.c13 <- c(l002.13[,3],l003.13[,3],l005.13[,3],l006.13[,3],l008.13[,3],l009.13[,3],l069.13[,3])

#this code works beautifully to produce a table with all of the log2 values for chromosome 6 
# chr13Data <- data.frame(y=c((log2(l001.13[,3])),(log2(l002.13[,3])),(log2(l003.13[,3])),(log2(l004.13[,3])),
#         (log2(l005.13[,3])),(log2(l006.13[,3])),(log2(l007.13[,3])),(log2(l008.13[,3])),
#         (log2(l009.13[,3])),(log2(l011.13[,3])),(log2(l018.13[,3])),(log2(l021.13[,3])),
#         (log2(l031.13[,3])),(log2(l049.13[,3])),(log2(l059.13[,3])),(log2(l061.13[,3])),
#         (log2(l066.13[,3])),(log2(l069.13[,3])),(log2(l076.13[,3])),(log2(l077.13[,3]))),
# Line=rep(c(1,2,3,4,5,6,7,8,9,11,18,21,31,49,59,61,66,69,76,77),each=(length(l001.13$tracking_id))))
# chr13Data$Line <-factor(chr13Data$Line)
# #This code works beautifully to produce a boxplot
# #want to make this chunk into a loop
# #but I can't because I want to add significance stars, which I would have to go back and do anyway, so that is pointless
# {
#   pdf("Chrm_13.pdf")
# boxplot(y~Line,data=chr13Data,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","cyan","green","green","cyan","cyan","green","darkcyan","cyan","cyan",
#               "green","cyan","cyan","cyan","cyan","green","cyan","cyan"),main="GC Lines Chromsome 13",las=3,ylim=c(-2,2.5))
# abline(h=0,lty=2)
# abline(h=0.585,lty=2,col="forestgreen")
# abline(h=-1,lty=2,col="darkviolet")
# }

# 
# aov.chr13 <- aov(y~Line,data=chr13Data)
# summary(aov.chr13)
# resids.chr13 <- resid(aov.chr13)
# {
#   pdf("Chrm13.resids.pdf")
# hist(resids.chr13,col="turquoise",breaks=10,xlab="Residuals.Chr.13")
# }
# 
# 
# TukeyHSD(aov.chr13)
# {pdf("tukey.chr13.pdf")
#   plot(TukeyHSD(aov.chr13))}
# #
# #fail shapiro test, so should probably use nonparametric
# kruskal.chr13 <- kruskal.test(y~Line,data=chr13Data)
# kruskal.test(y~Line,data=chr13Data)
# summary(kruskal.chr13)
#test pairwise comparisons
#Tukeys doesnt work with kruskal
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
# kruskalmc(y~Line,data=chr13Data)
# multiCompsChr13 <- kruskalmc(y~Line,data=chr13Data)
# #many pairwise t-tests
# pairwise.t.test(chr13Data$y,chr13Data$Line,p.adjust.method = "fdr")
# wilcox.chr13 <- pairwise.wilcox.test(chr13Data$y,chr13Data$Line,p.adjust.method = "fdr")
# pairwise.wilcox.test(chr13Data$y,chr13Data$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 14
chr14DataGC <- data.frame(y=c((log2(l001.14[,3])),(log2(l002.14[,3])),(log2(l003.14[,3])),(log2(l004.14[,3])),(log2(l005.14[,3])),
        (log2(l007.14[,3])),(log2(l008.14[,3])),(log2(l009.14[,3])),(log2(l011.14[,3])),
        (log2(l018.14[,3])),(log2(l049.14[,3])),(log2(l059.14[,3])),(log2(l061.14[,3])),(log2(l069.14[,3])),(log2(l076.14[,3])),(log2(l077.14[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.14$tracking_id))))
chr14DataGC$Line <-factor(chr14DataGC$Line)
chr14DataMA <- data.frame(y=c((log2(l001.14.old[,3])),(log2(l002.14.old[,3])),(log2(l003.14.old[,3])),(log2(l004.14.old[,3])),(log2(l005.14.old[,3])),
        (log2(l007.14.old[,3])),(log2(l008.14.old[,3])),(log2(l009.14.old[,3])),(log2(l011.14.old[,3])),
        (log2(l015.14.old[,3])),(log2(l028.14.old[,3])),(log2(l029.14[,3])),(log2(l050.14[,3])),(log2(l088.14.old[,3])),(log2(l108.14.old[,3])),(log2(l112.14[,3])),(log2(l115.14[,3])),(log2(l117.14[,3])),(log2(l119.14.old[,3])),(log2(l123.14[,3])),(log2(l152.14[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.14.old$tracking_id))))
chr14DataMA$Line <-factor(chr14DataMA$Line)

aovchr14GC <- lm(y~Line,data=chr14DataGC)
stargazer(aovchr14GC,type="html",out="anovachr14GC.doc",title="Chromosome XIV Hybrid Strain")

aovchr14MA <- lm(y~Line,data=chr14DataMA)
stargazer(aovchr14MA,type="html",out="anovachr14MA.doc",title="Chromosome XIV Lab Strain")

#RATIOS FOR CHROMOSOME 14
#euRatio.c14 <- c(l002.14[,3],l003.14[,3],l005.14[,3],l006.14[,3],l008.14[,3],l009.14[,3],l069.14[,3])

#this code works beautifully to produce a table with all of the log2 values for chromosome 6 
# chr14Data <- data.frame(y=c((log2(l001.14[,3])),(log2(l002.14[,3])),(log2(l003.14[,3])),(log2(l004.14[,3])),
#         (log2(l005.14[,3])),(log2(l006.14[,3])),(log2(l007.14[,3])),(log2(l008.14[,3])),
#         (log2(l009.14[,3])),(log2(l011.14[,3])),(log2(l018.14[,3])),(log2(l021.14[,3])),
#         (log2(l031.14[,3])),(log2(l049.14[,3])),(log2(l059.14[,3])),(log2(l061.14[,3])),
#         (log2(l066.14[,3])),(log2(l069.14[,3])),(log2(l076.14[,3])),(log2(l077.14[,3]))),
# Line=rep(c(1,2,3,4,5,6,7,8,9,11,18,21,31,49,59,61,66,69,76,77),each=(length(l001.14$tracking_id))))
# chr14Data$Line <-factor(chr14Data$Line)
# #This code works beautifully to produce a boxplotl001.14
# #want to make this chunk into a loop
# #but I can't because I want to add significance stars, which I would have to go back and do anyway, so that is pointless
# {
#   pdf("Chrm_14.pdf")
# boxplot(y~Line,data=chr14Data,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","cyan","green","green","cyan","cyan","green","darkcyan","cyan","cyan",
#               "green","cyan","cyan","cyan","cyan","green","deeppink1","cyan"),main="GC Lines Chromsome 14",las=3,ylim=c(-2,2.5))
# abline(h=0,lty=2)
# abline(h=0.585,lty=2,col="forestgreen")
# abline(h=-1,lty=2,col="darkviolet")
# text(x=18.7, y=2.2, "*", pos=3, cex=2, col="red")
# text(x=19, y=2.2, "*", pos=3, cex=2, col="red")
# text(x=19.3, y=2.2, "*", pos=3, cex=2, col="red")
# }

# chr14Data.r <- data.frame(y=c((log2(l001.14[,3])),(log2(l002.14[,3])),(log2(l003.14[,3])),
#         (log2(l005.14[,3])),(log2(l006.14[,3])),
#         (log2(l009.14[,3])),
#         (log2(l031.14[,3])),
#      (log2(l069.14[,3])),(log2(l076.14[,3]))),
# Line=rep(c(1,2,3,5,6,9,31,69,76),each=(length(l001.14$tracking_id))))
# chr14Data.r$Line <-factor(chr14Data.r$Line)
# 
# aov.chr14 <- aov(y~Line,data=chr14Data.r)
# summary(aov.chr14)
# resids.chr14 <- resid(aov.chr14)
# {
#   pdf("Chrm14.resids.pdf")
# hist(resids.chr14,col="turquoise",breaks=10,xlab="Residuals.Chr.14")
# }

# t.test((log2(l076.14[,3])),mu=exp3x)
# 
# TukeyHSD(aov.chr14)
# {pdf("tukey.chr14.pdf")
#  plot(TukeyHSD(aov.chr14)) }
# 
# #fail shapiro test, so should probably use nonparametric
# kruskal.chr14 <- kruskal.test(y~Line,data=chr14Data)
# kruskal.test(y~Line,data=chr14Data)
# summary(kruskal.chr14)
#test pairwise comparisons
#Tukeys doesnt work with kruskal
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
# kruskalmc(y~Line,data=chr14Data)
# multiCompsChr14 <- kruskalmc(y~Line,data=chr14Data)
# #many pairwise t-tests
# pairwise.t.test(chr14Data$y,chr14Data$Line,p.adjust.method = "fdr")
# wilcox.chr14 <- pairwise.wilcox.test(chr14Data$y,chr14Data$Line,p.adjust.method = "fdr")
# pairwise.wilcox.test(chr14Data$y,chr14Data$Line,p.adjust.method = "fdr")
# 
# ##############################################################################################
#CHROMOSOME 15
chr15DataGC <- data.frame(y=c((log2(l001.15[,3])),(log2(l002.15[,3])),(log2(l003.15[,3])),(log2(l004.15[,3])),(log2(l005.15[,3])),
        (log2(l007.15[,3])),(log2(l008.15[,3])),(log2(l009.15[,3])),(log2(l011.15[,3])),
        (log2(l018.15[,3])),(log2(l049.15[,3])),(log2(l059.15[,3])),(log2(l061.15[,3])),(log2(l069.15[,3])),(log2(l076.15[,3])),(log2(l077.15[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.15$tracking_id))))
chr15DataGC$Line <-factor(chr15DataGC$Line)
chr15DataMA <- data.frame(y=c((log2(l001.15.old[,3])),(log2(l002.15.old[,3])),(log2(l003.15.old[,3])),(log2(l004.15.old[,3])),(log2(l005.15.old[,3])),
        (log2(l007.15.old[,3])),(log2(l008.15.old[,3])),(log2(l009.15.old[,3])),(log2(l011.15.old[,3])),
        (log2(l015.15.old[,3])),(log2(l028.15.old[,3])),(log2(l029.15[,3])),(log2(l050.15[,3])),(log2(l088.15.old[,3])),(log2(l108.15.old[,3])),(log2(l112.15[,3])),(log2(l115.15[,3])),(log2(l117.15[,3])),(log2(l119.15.old[,3])),(log2(l123.15[,3])),(log2(l152.15[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.15.old$tracking_id))))
chr15DataMA$Line <-factor(chr15DataMA$Line)

aovchr15GC <- lm(y~Line,data=chr15DataGC)
stargazer(aovchr15GC,type="html",out="anovachr15GC.doc",title="Chromosome XV Hybrid Strain")

aovchr15MA <- lm(y~Line,data=chr15DataMA)
stargazer(aovchr15MA,type="html",out="anovachr15MA.doc",title="Chromosome XV Lab Strain")

#RATIOS FOR CHROMOSOME 15
#euRatio.c15 <- c(l002.15[,3],l003.15[,3],l005.15[,3],l006.15[,3],l008.15[,3],l009.15[,3],l069.15[,3])

#this code works beautifully to produce a table with all of the log2 values for chromosome 6 
# chr15Data <- data.frame(y=c((log2(l001.15[,3])),(log2(l002.15[,3])),(log2(l003.15[,3])),(log2(l004.15[,3])),
#         (log2(l005.15[,3])),(log2(l006.15[,3])),(log2(l007.15[,3])),(log2(l008.15[,3])),
#         (log2(l009.15[,3])),(log2(l011.15[,3])),(log2(l018.15[,3])),(log2(l021.15[,3])),
#         (log2(l031.15[,3])),(log2(l049.15[,3])),(log2(l059.15[,3])),(log2(l061.15[,3])),
#         (log2(l066.15[,3])),(log2(l069.15[,3])),(log2(l076.15[,3])),(log2(l077.15[,3]))),
# Line=rep(c(1,2,3,4,5,6,7,8,9,11,18,21,31,49,59,61,66,69,76,77),each=(length(l001.15$tracking_id))))
# chr15Data$Line <-factor(chr15Data$Line)
#This code works beautifully to produce a boxplotl001.14
#want to make this chunk into a loop
#but I can't because I want to add significance stars, which I would have to go back and do anyway, so that is pointless
# {
#   pdf("Chrm_15.pdf")
# boxplot(y~Line,data=chr15Data,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","cyan","green","green","cyan","cyan","green","deeppink1","cyan","cyan",
#               "green","cyan","cyan","cyan","cyan","green","cyan","cyan"),main="GC Lines Chromsome 15",las=3,ylim=c(-2,2.5))
# abline(h=0,lty=2)
# abline(h=0.585,lty=2,col="forestgreen")
# abline(h=-1,lty=2,col="darkviolet")
# 
# 
# text(x=9.7, y=1.5, "*", pos=3, cex=2, col="red")
# text(x=10, y=1.5, "*", pos=3, cex=2, col="red")
# text(x=10.3, y=1.5, "*", pos=3, cex=2, col="red")
# 
# }
# 
# chr15Data.r <- data.frame(y=c((log2(l001.15[,3])),(log2(l002.15[,3])),(log2(l003.15[,3])),
#         (log2(l005.15[,3])),(log2(l006.15[,3])),(log2(l009.15[,3])),(log2(l011.15[,3])),(log2(l031.15[,3])),(log2(l069.15[,3]))),
# Line=rep(c(1,2,3,5,6,9,11,31,69),each=(length(l001.15$tracking_id))))
# chr15Data.r$Line <-factor(chr15Data.r$Line)
# 
# aov.chr15 <- aov(y~Line,data=chr15Data.r)
# summary(aov.chr15)
# resids.chr15 <- resid(aov.chr15)
# {
#   pdf("Chrm15.resids.pdf")
# hist(resids.chr15,col="turquoise",breaks=10,xlab="Residuals.Chr.15")
# }

# TukeyHSD(aov.chr15)
# {pdf("tukey.chr15.pdf")
#  plot(TukeyHSD(aov.chr15)) }
# 
# t.test((log2(l011.15[,3])),mu=exp3x)

#TukeyHSD(aov.chr9)
#plot(TukeyHSD(aov.chr9))
#fail shapiro test, so should probably use nonparametric


##############################################################################################
#CHROMOSOME 16
chr16DataGC <- data.frame(y=c((log2(l001.16[,3])),(log2(l002.16[,3])),(log2(l003.16[,3])),(log2(l004.16[,3])),(log2(l005.16[,3])),
        (log2(l007.16[,3])),(log2(l008.16[,3])),(log2(l009.16[,3])),(log2(l011.16[,3])),
        (log2(l018.16[,3])),(log2(l049.16[,3])),(log2(l059.16[,3])),(log2(l061.16[,3])),(log2(l069.16[,3])),(log2(l076.16[,3])),(log2(l077.16[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,18,49,59,61,69,76,77),each=(length(l001.16$tracking_id))))
chr16DataGC$Line <-factor(chr16DataGC$Line)
chr16DataMA <- data.frame(y=c((log2(l001.16.old[,3])),(log2(l002.16.old[,3])),(log2(l003.16.old[,3])),(log2(l004.16.old[,3])),(log2(l005.16.old[,3])),
        (log2(l007.16.old[,3])),(log2(l008.16.old[,3])),(log2(l009.16.old[,3])),(log2(l011.16.old[,3])),
        (log2(l015.16.old[,3])),(log2(l028.16.old[,3])),(log2(l029.16[,3])),(log2(l050.16[,3])),(log2(l088.16.old[,3])),(log2(l108.16.old[,3])),(log2(l112.16[,3])),(log2(l115.16[,3])),(log2(l117.16[,3])),(log2(l119.16.old[,3])),(log2(l123.16[,3])),(log2(l152.16[,3]))),
Line=rep(c(1,2,3,4,5,7,8,9,11,15,28,29,50,88,108,112,115,117,119,123,152),each=(length(l001.16.old$tracking_id))))
chr16DataMA$Line <-factor(chr16DataMA$Line)

aovchr16GC <- lm(y~Line,data=chr16DataGC)
stargazer(aovchr16GC,type="html",out="anovachr16GC.doc",title="Chromosome XVI Hybrid Strain")

aovchr16MA <- lm(y~Line,data=chr16DataMA)
stargazer(aovchr16MA,type="html",out="anovachr16MA.doc",title="Chromosome XVI Lab Strain")

#RATIOS FOR CHROMOSOME 16
#euRatio.c16 <- c(l002.16[,3],l003.16[,3],l005.16[,3],l006.16[,3],l008.16[,3],l009.16[,3],l069.16[,3])

#this code works beautifully to produce a table with all of the log2 values for chromosome 6 
# chr16Data <- data.frame(y=c((log2(l001.16[,3])),(log2(l002.16[,3])),(log2(l003.16[,3])),(log2(l004.16[,3])),
#         (log2(l005.16[,3])),(log2(l006.16[,3])),(log2(l007.16[,3])),(log2(l008.16[,3])),
#         (log2(l009.16[,3])),(log2(l011.16[,3])),(log2(l018.16[,3])),(log2(l021.16[,3])),
#         (log2(l031.16[,3])),(log2(l049.16[,3])),(log2(l059.16[,3])),(log2(l061.16[,3])),
#         (log2(l066.16[,3])),(log2(l069.16[,3])),(log2(l076.16[,3])),(log2(l077.16[,3]))),
# Line=rep(c(1,2,3,4,5,6,7,8,9,11,18,21,31,49,59,61,66,69,76,77),each=(length(l001.16$tracking_id))))
# chr16Data$Line <-factor(chr16Data$Line)
# #This code works beautifully to produce a boxplotl001.14
# #want to make this chunk into a loop
# #but I can't because I want to add significance stars, which I would have to go back and do anyway, so that is pointless
# {
#   pdf("Chrm_16.pdf")
# boxplot(y~Line,data=chr16Data,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("green", "green","green","cyan","green","green","cyan","deeppink1","green","cyan","cyan","cyan",
#               "green","cyan","cyan","cyan","cyan","green","cyan","cyan"),main="GC Lines Chromsome 16",las=3,ylim=c(-2,2.5))
# abline(h=0,lty=2)
# abline(h=0.585,lty=2,col="forestgreen")
# abline(h=-1,lty=2,col="darkviolet")
# abline(h=1,lty=2,col="darkblue")
# text(x=7.7, y=1.5, "*", pos=3, cex=2, col="red")
# text(x=8, y=1.5, "*", pos=3, cex=2, col="red")
# text(x=8.3, y=1.5, "*", pos=3, cex=2, col="red")
# 
# }

# exp4x <- log2(2)
# t.test((log2(l008.16[,3])),mu=exp4x)
# 
# 
# #mean((log2(l008.16[,3])))
# #average of line 8 chromosome 16 is 0.91
# #awfully close to 1 
# 
# chr16Data.r <- data.frame(y=c((log2(l001.16[,3])),(log2(l002.16[,3])),(log2(l003.16[,3])),
#         (log2(l005.16[,3])),(log2(l006.16[,3])),(log2(l008.16[,3])),
#         (log2(l009.16[,3])),
#         (log2(l031.16[,3])),
#        (log2(l069.16[,3]))),
# Line=rep(c(1,2,3,5,6,8,9,31,69),each=(length(l001.16$tracking_id))))
# chr16Data.r$Line <-factor(chr16Data.r$Line)


# aov.chr16 <- aov(y~Line,data=chr16Data.r)
# summary(aov.chr16)
# resids.chr16 <- resid(aov.chr16)
# {
#   pdf("Chrm16.resids.pdf")
# hist(resids.chr16,col="turquoise",breaks=10,xlab="Residuals.Chr.16")
# }
# 
# 
# TukeyHSD(aov.chr16)

# {pdf("tukey.chr16.pdf")
#   plot(TukeyHSD(aov.chr16))}
#fail shapiro test, so should probably use nonparametric
#kruskal.chr16 <- kruskal.test(y~Line,data=chr16Data)
#kruskal.test(y~Line,data=chr16Data)
#summary(kruskal.chr16)
#test pairwise comparisons
#Tukeys doesnt work with kruskal
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr16Data)
#multiCompsChr16 <- kruskalmc(y~Line,data=chr16Data)
#many pairwise t-tests
#pairwise.t.test(chr16Data$y,chr16Data$Line,p.adjust.method = "fdr")
#wilcox.chr16 <- pairwise.wilcox.test(chr16Data$y,chr16Data$Line,p.adjust.method = "fdr")
#pairwise.wilcox.test(chr16Data$y,chr16Data$Line,p.adjust.method = "fdr")
```

Codes for boxplots for MA lines 
```{r boxplots_MA_lines, include=TRUE, echo=FALSE}
####################
#CHROMOSOME 1 


chr1DataMA <- data.frame(y=c((log2(l029.1[,3])),(log2(l050.1[,3])),(log2(l112.1[,3])),(log2(l115.1[,3])),(log2(l117.1[,3])),
        (log2(l123.1[,3])),(log2(l152.1[,3]))), Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.1$tracking_id))))
chr1DataMA$Line <-factor(chr1DataMA$Line)

#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_1_MA.pdf")
boxplot(y~Line,data=chr1DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("cyan", "green","cyan","green","cyan","cyan","deeppink1"),main="MA Lines Chromsome 1",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

text(x=7.7, y=1.5, "*", pos=3, cex=2, col="red")
text(x=8, y=1.5, "*", pos=3, cex=2, col="red")
text(x=8.3, y=1.5, "*", pos=3, cex=2, col="red")
}

# chr1DataMA.r <- data.frame(y=c((log2(l050.1[,3])),(log2(l115.1[,3])),
#         (log2(l152.1[,3]))), Line=rep(c(50,115,152),each=(length(l029.1$tracking_id))))
# chr1DataMA.r$Line <-factor(chr1DataMA.r$Line)
# 
# aov.chr1MA <- aov(y~Line,data=chr1DataMA.r)
# summary(aov.chr1MA)
# resids.chr1MA <- resid(aov.chr1MA)
# {
#   pdf("Chrm1.residsMA.pdf")
# hist(resids.chr1MA,col="turquoise",breaks=10,xlab="Residuals.Chr.1MA")
# }
# 
# TukeyHSD(aov.chr1MA)
# {pdf("Tukey.chr1.MA.pdf")
# plot(TukeyHSD(aov.chr1MA))}
# 
# shapiro.test(resids.chr1MA)
# 
# t.test((log2(l152.1[,3])),mu=exp3x)

#fail shapiro test, so should probably use nonparametric
#kruskal.chr1MA <- kruskal.test(y~Line,data=chr1DataMA)
#kruskal.test(y~Line,data=chr1DataMA)
#p-value < 2.2 e-16 (So at least one of the data points is very different from the others)
#test pairwise comparisons
#Tukeys doesnt work with kruskal
#TukeyHSD(aov.chr1MA)
#plot(TukeyHSD(aov.chr1MA))
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr1DataMA)
#multiCompsChr1MA <- kruskalmc(y~Line,data=chr1DataMA)
#OR can do many pairwise t-tests
#pairwise.t.test(chr1DataMA$y,chr1DataMA$Line,p.adjust.method = "fdr")
#wilcox.chr1MA <- pairwise.wilcox.test(chr1DataMA$y,chr1DataMA$Line,p.adjust.method = "fdr")

#pairwise.wilcox.test(chr1DataMA$y,chr1DataMA$Line,p.adjust.method = "fdr")


##############################################################################################
#CHROMOSOME 2

{

chr2DataMA <- data.frame(y=c((log2(l029.2[,3])),(log2(l050.2[,3])),(log2(l112.2[,3])),(log2(l115.2[,3])),(log2(l117.2[,3])),
        (log2(l123.2[,3])),(log2(l152.2[,3]))), Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.2$tracking_id))))
}
chr2DataMA$Line <-factor(chr2DataMA$Line)


#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_2_MA.pdf")
boxplot(y~Line,data=chr2DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("cyan", "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Lines Chromsome 2",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


}

#ANOVA/NonParametric Equivalent

aov.chr2MA <- aov(y~Line,data=chr2DataMA)
summary(aov.chr2MA)
resids.chr2MA <- resid(aov.chr2MA)
{
  pdf("Chrm2.residsMA.pdf")
hist(resids.chr2MA,col="turquoise",breaks=10,xlab="Residuals.Chr.2MA")
}

TukeyHSD(aov.chr2MA)
{pdf("Tukey.chr2.MA.pdf")
plot(TukeyHSD(aov.chr2MA))}

#shapiro.test(resids.chr2MA)
#fail shapiro test, so should probably use nonparametric

#kruskal.test(y~Line,data=chr2DataMA)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr2DataMA)
#OR can do many pairwise t-tests
#pairwise.wilcox.test(chr2DataMA$y,chr2DataMA$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 3

{

chr3DataMA <- data.frame(y=c((log2(l029.3[,3])),(log2(l050.3[,3])),(log2(l112.3[,3])),(log2(l115.3[,3])),(log2(l117.3[,3])),
        (log2(l123.3[,3])),(log2(l152.3[,3]))), Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.3$tracking_id))))
}
chr3DataMA$Line <-factor(chr3DataMA$Line)


#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_3_MA.pdf")
boxplot(y~Line,data=chr3DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("cyan", "green","cyan","green","cyan","cyan","cyan"),main="MA Lines Chromsome 3",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


}
#text(x=7.7, y=1.5, "*", pos=3, cex=2, col="red")
#text(x=8, y=1.5, "*", pos=3, cex=2, col="red")
#text(x=8.3, y=1.5, "*", pos=3, cex=2, col="red")

#ANOVA/NonParametric Equivalent
aov.chr3MA <- aov(y~Line,data=chr3DataMA)
summary(aov.chr3MA)
resids.chr3MA <- resid(aov.chr3MA)
{
  # pdf("Chrm3.residsMA.pdf")
hist(resids.chr3MA,col="turquoise",breaks=10,xlab="Residuals.Chr.3MA")
}

TukeyHSD(aov.chr3MA)
{
  # pdf("Tukey.chr3.MA.pdf")
plot(TukeyHSD(aov.chr3MA))}

#shapiro.test(resids.chr3MA)
#fail shapiro test, so should probably use nonparametric

#kruskal.test(y~Line,data=chr3DataMA)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr3DataMA)
#OR can do many pairwise t-tests
#pairwise.wilcox.test(chr3DataMA$y,chr3DataMA$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 4

{

chr4DataMA <- data.frame(y=c((log2(l029.4[,3])),(log2(l050.4[,3])),(log2(l112.4[,3])),(log2(l115.4[,3])),(log2(l117.4[,3])),
        (log2(l123.4[,3])),(log2(l152.4[,3]))), Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.4$tracking_id))))
}
chr4DataMA$Line <-factor(chr4DataMA$Line)

#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_4_MA.pdf")
boxplot(y~Line,data=chr4DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("cyan", "green","cyan","green","cyan","cyan","cyan","cyan"),main="MA Lines Chromsome 4",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


}
#text(x=7.7, y=1.5, "*", pos=3, cex=2, col="red")
#text(x=8, y=1.5, "*", pos=3, cex=2, col="red")
#text(x=8.3, y=1.5, "*", pos=3, cex=2, col="red")

#ANOVA/NonParametric Equivalent
aov.chr4MA <- aov(y~Line,data=chr4DataMA)
summary(aov.chr4MA)
resids.chr4MA <- resid(aov.chr4MA)
{
  # pdf("Chrm4.residsMA.pdf")
hist(resids.chr4MA,col="turquoise",breaks=10,xlab="Residuals.Chr.4MA")
}
TukeyHSD(aov.chr4MA)
{
  # pdf("Tukey.chr4.MA.pdf")
plot(TukeyHSD(aov.chr4MA))}
#shapiro.test(resids.chr4MA)
#fail shapiro test, so should probably use nonparametric

kruskal.test(y~Line,data=chr4DataMA)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
kruskalmc(y~Line,data=chr4DataMA)
#OR can do many pairwise t-tests
pairwise.wilcox.test(chr4DataMA$y,chr4DataMA$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 5

{

chr5DataMA <- data.frame(y=c((log2(l029.5[,3])),(log2(l050.5[,3])),(log2(l112.5[,3])),(log2(l115.5[,3])),(log2(l117.5[,3])),
        (log2(l123.5[,3])),(log2(l152.5[,3]))), Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.5$tracking_id))))
}
chr5DataMA$Line <-factor(chr5DataMA$Line)

#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_5_MA.pdf")
boxplot(y~Line,data=chr5DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("cyan", "green","cyan","green","deeppink1","deeppink1","cyan"),main="MA Lines Chromsome 5",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

text(x=4.8, y=2.1, "*", pos=3, cex=2, col="red")
text(x=5, y=2.1, "*", pos=3, cex=2, col="red")
text(x=5.2, y=2.1, "*", pos=3, cex=2, col="red")

text(x=5.8, y=1.8, "*", pos=3, cex=2, col="red")
text(x=6, y=1.8, "*", pos=3, cex=2, col="red")
text(x=6.2, y=1.8, "*", pos=3, cex=2, col="red")

}

{

chr5DataMA.r <- data.frame(y=c((log2(l050.5[,3])),(log2(l115.5[,3])),(log2(l117.5[,3])),(log2(l123.5[,3]))), Line=rep(c(50,115,117,123),each=(length(l050.5$tracking_id))))
}
chr5DataMA.r$Line <-factor(chr5DataMA.r$Line)
#ANOVA/NonParametric Equivalent

aov.chr5MA <- aov(y~Line,data=chr5DataMA.r)
summary(aov.chr5MA)
resids.chr5MA <- resid(aov.chr5MA)
{
  # pdf("Chrm5.residsMA.pdf")
hist(resids.chr5MA,col="turquoise",breaks=10,xlab="Residuals.Chr.5MA")
}
TukeyHSD(aov.chr5MA)
{
  # pdf("Tukey.chr5.MA.pdf")
plot(TukeyHSD(aov.chr5MA))}

# t.test((log2(l117.5[,3])),mu=exp3x)
# t.test((log2(l123.5[,3])),mu=exp3x)

#shapiro.test(resids.chr5MA)
#fail shapiro test, so should probably use nonparametric


#kruskal.test(y~Line,data=chr5DataMA)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr5DataMA)
#OR can do many pairwise t-tests
#pairwise.wilcox.test(chr5DataMA$y,chr5DataMA$Line,p.adjust.method = "fdr")


##############################################################################################
#CHROMOSOME 6

{

chr6DataMA <- data.frame(y=c((log2(l029.6[,3])),(log2(l050.6[,3])),(log2(l112.6[,3])),(log2(l115.6[,3])),(log2(l117.6[,3])),
        (log2(l123.6[,3])),(log2(l152.6[,3]))), Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.6$tracking_id))))
}
chr6DataMA$Line <-factor(chr6DataMA$Line)

#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_6_MA.pdf")
boxplot(y~Line,data=chr6DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("cyan", "green","cyan","green","cyan","cyan","cyan"),main="MA Lines Chromsome 6",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

}


#ANOVA/NonParametric Equivalent
aov.chr6MA <- aov(y~Line,data=chr6DataMA)
summary(aov.chr6MA)
resids.chr6MA <- resid(aov.chr6MA)
{
  # pdf("Chrm6.residsMA.pdf")
hist(resids.chr6MA,col="turquoise",breaks=10,xlab="Residuals.Chr.6MA")
}

TukeyHSD(aov.chr6MA)
{
  # pdf("Tukey.chr6.MA.pdf")
plot(TukeyHSD(aov.chr6MA))}

shapiro.test(resids.chr6MA)
#fail shapiro test, so should probably use nonparametric

#kruskal.test(y~Line,data=chr6DataMA)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr6DataMA)
#OR can do many pairwise t-tests
#pairwise.wilcox.test(chr6DataMA$y,chr6DataMA$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 7
{

chr7DataMA <- data.frame(y=c((log2(l029.7[,3])),(log2(l050.7[,3])),(log2(l112.7[,3])),(log2(l115.7[,3])),(log2(l117.7[,3])),
        (log2(l123.7[,3])),(log2(l152.7[,3]))), Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.7$tracking_id))))
}
chr7DataMA$Line <-factor(chr7DataMA$Line)


#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_7_MA.pdf")
boxplot(y~Line,data=chr7DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("cyan", "green","cyan","green","cyan","cyan","cyan"),main="MA Lines Chromsome 7",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

}


#ANOVA/NonParametric Equivalent
aov.chr7MA <- aov(y~Line,data=chr7DataMA)
summary(aov.chr7MA)
resids.chr7MA <- resid(aov.chr7MA)
{
  # pdf("Chrm7.residsMA.pdf")
hist(resids.chr7MA,col="turquoise",breaks=10,xlab="Residuals.Chr.7MA")
}

TukeyHSD(aov.chr7MA)
{
  # pdf("Tukey.chr7.MA.pdf")
plot(TukeyHSD(aov.chr7MA))}
#shapiro.test(resids.chr7MA)
#fail shapiro test, so should probably use nonparametric

#kruskal.test(y~Line,data=chr7DataMA)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr7DataMA)
#OR can do many pairwise t-tests
#pairwise.wilcox.test(chr7DataMA$y,chr7DataMA$Line,p.adjust.method = "fdr")


##############################################################################################
#CHROMOSOME 8

{

chr8DataMA <- data.frame(y=c((log2(l029.8[,3])),(log2(l050.8[,3])),(log2(l112.8[,3])),(log2(l115.8[,3])),(log2(l117.8[,3])),
        (log2(l123.8[,3])),(log2(l152.8[,3]))), Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.8$tracking_id))))
}
chr8DataMA$Line <-factor(chr8DataMA$Line)

#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_8_MA.pdf")
boxplot(y~Line,data=chr8DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("cyan", "green","cyan","green","cyan","cyan","deeppink1"),main="MA Lines Chromsome 8",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

text(x=7.8, y=1.8, "*", pos=3, cex=2, col="red")
text(x=8, y=1.8, "*", pos=3, cex=2, col="red")
text(x=8.2, y=1.8, "*", pos=3, cex=2, col="red")

}


{

chr8DataMA.r <- data.frame(y=c((log2(l050.8[,3])),(log2(l115.8[,3])),
        (log2(l152.8[,3]))), Line=rep(c(50,115,152),each=(length(l050.8$tracking_id))))
}
chr8DataMA.r$Line <-factor(chr8DataMA.r$Line)

#ANOVA/NonParametric Equivalent
aov.chr8MA <- aov(y~Line,data=chr8DataMA.r)
summary(aov.chr8MA)
resids.chr8MA <- resid(aov.chr8MA)
{
  # pdf("Chrm8.residsMA.pdf")
hist(resids.chr8MA,col="turquoise",breaks=10,xlab="Residuals.Chr.8MA")
}

TukeyHSD(aov.chr8MA)
{
  # pdf("Tukey.chr8.MA.pdf")
plot(TukeyHSD(aov.chr8MA))}

# t.test((log2(l152.8[,3])),mu=exp3x)


#shapiro.test(resids.chr8MA)
#fail shapiro test, so should probably use nonparametric

#kruskal.test(y~Line,data=chr8DataMA)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr8DataMA)
#OR can do many pairwise t-tests
#pairwise.wilcox.test(chr8DataMA$y,chr8DataMA$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 9

{

chr9DataMA <- data.frame(y=c((log2(l029.9[,3])),(log2(l050.9[,3])),(log2(l112.9[,3])),(log2(l115.9[,3])),(log2(l117.9[,3])),
        (log2(l123.9[,3])),(log2(l152.9[,3]))), Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.9$tracking_id))))
}
chr9DataMA$Line <-factor(chr9DataMA$Line)

#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_9_MA.pdf")
boxplot(y~Line,data=chr9DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("deeppink1", "green","cyan","green","cyan","cyan","cyan"),main="MA Lines Chromsome 9",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")

text(x=0.8, y=1.8, "*", pos=3, cex=2, col="red")
text(x=1, y=1.8, "*", pos=3, cex=2, col="red")
text(x=1.2, y=1.8, "*", pos=3, cex=2, col="red")

}

{

chr9DataMA.r <- data.frame(y=c((log2(l029.9[,3])),(log2(l050.9[,3])),(log2(l115.9[,3]))), Line=rep(c(29,50,115),each=(length(l029.9$tracking_id))))
}
chr9DataMA.r$Line <-factor(chr9DataMA.r$Line)

#ANOVA/NonParametric Equivalent
aov.chr9MA <- aov(y~Line,data=chr9DataMA.r)
summary(aov.chr9MA)
resids.chr9MA <- resid(aov.chr9MA)
{
  # pdf("Chrm9.residsMA.pdf")
hist(resids.chr9MA,col="turquoise",breaks=10,xlab="Residuals.Chr.9MA")
}

TukeyHSD(aov.chr9MA)
{
  # pdf("Tukey.chr9.MA.pdf")
plot(TukeyHSD(aov.chr9MA))}

# t.test((log2(l029.9[,3])),mu=exp1x)

#shapiro.test(resids.chr9MA)
#fail shapiro test, so should probably use nonparametric

#kruskal.test(y~Line,data=chr9DataMA)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr9DataMA)
#OR can do many pairwise t-tests
#pairwise.wilcox.test(chr9DataMA$y,chr9DataMA$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 10

{

chr10DataMA <- data.frame(y=c((log2(l029.10[,3])),(log2(l050.10[,3])),(log2(l112.10[,3])),(log2(l115.10[,3])),(log2(l117.10[,3])),
        (log2(l123.10[,3])),(log2(l152.10[,3]))),Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.10$tracking_id))))
}
chr10DataMA$Line <-factor(chr10DataMA$Line)

#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_10_MA.pdf")
boxplot(y~Line,data=chr10DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("cyan", "green","cyan","green","cyan","cyan","cyan"),main="MA Lines Chromsome 10",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


}

#text(x=0.8, y=1.8, "*", pos=3, cex=2, col="red")
#text(x=1, y=1.8, "*", pos=3, cex=2, col="red")
#text(x=1.2, y=1.8, "*", pos=3, cex=2, col="red")

#ANOVA/NonParametric Equivalent
aov.chr10MA <- aov(y~Line,data=chr10DataMA)
summary(aov.chr10MA)
resids.chr10MA <- resid(aov.chr10MA)
{
  # pdf("Chrm10.residsMA.pdf")
hist(resids.chr10MA,col="turquoise",breaks=10,xlab="Residuals.Chr.10MA")
}

TukeyHSD(aov.chr10MA)
{
  # pdf("Tukey.chr10.MA.pdf")
plot(TukeyHSD(aov.chr10MA))}
#shapiro.test(resids.chr10MA)
#fail shapiro test, so should probably use nonparametric

#kruskal.test(y~Line,data=chr10DataMA)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr10DataMA)
#OR can do many pairwise t-tests
#pairwise.wilcox.test(chr10DataMA$y,chr10DataMA$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 11

{

chr11DataMA <- data.frame(y=c((log2(l029.11[,3])),(log2(l050.11[,3])),(log2(l112.11[,3])),(log2(l115.11[,3])),(log2(l117.11[,3])),
        (log2(l123.11[,3])),(log2(l152.11[,3]))), Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.11$tracking_id))))
}
chr11DataMA$Line <-factor(chr11DataMA$Line)
#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_11_MA.pdf")
boxplot(y~Line,data=chr11DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("cyan", "green","cyan","green","cyan","cyan","cyan"),main="MA Lines Chromsome 11",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


}

#text(x=0.8, y=1.8, "*", pos=3, cex=2, col="red")
#text(x=1, y=1.8, "*", pos=3, cex=2, col="red")
#text(x=1.2, y=1.8, "*", pos=3, cex=2, col="red")

#ANOVA/NonParametric Equivalent
aov.chr11MA <- aov(y~Line,data=chr11DataMA)
summary(aov.chr11MA)
resids.chr11MA <- resid(aov.chr11MA)
{
  # pdf("Chrm11.residsMA.pdf")
hist(resids.chr11MA,col="turquoise",breaks=10,xlab="Residuals.Chr.11MA")
}
TukeyHSD(aov.chr11MA)
{
  # pdf("Tukey.chr11.MA.pdf")
plot(TukeyHSD(aov.chr11MA))}
#shapiro.test(resids.chr11MA)
#fail shapiro test, so should probably use nonparametric

#kruskal.test(y~Line,data=chr11DataMA)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr11DataMA)
#OR can do many pairwise t-tests
#pairwise.wilcox.test(chr11DataMA$y,chr11DataMA$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 12

{

chr12DataMA <- data.frame(y=c((log2(l029.12[,3])),(log2(l050.12[,3])),(log2(l112.12[,3])),(log2(l115.12[,3])),(log2(l117.12[,3])),
        (log2(l123.12[,3])),(log2(l152.12[,3]))), Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.12$tracking_id))))
}
chr12DataMA$Line <-factor(chr12DataMA$Line)
#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_12_MA.pdf")
boxplot(y~Line,data=chr12DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("cyan", "green","cyan","green","cyan","cyan","cyan"),main="MA Lines Chromsome 12",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


}

#text(x=0.8, y=1.8, "*", pos=3, cex=2, col="red")
#text(x=1, y=1.8, "*", pos=3, cex=2, col="red")
#text(x=1.2, y=1.8, "*", pos=3, cex=2, col="red")


#ANOVA/NonParametric Equivalent
aov.chr12MA <- aov(y~Line,data=chr12DataMA)
summary(aov.chr12MA)
resids.chr12MA <- resid(aov.chr12MA)
{
  # pdf("Chrm12.residsMA.pdf")
hist(resids.chr12MA,col="turquoise",breaks=10,xlab="Residuals.Chr.12MA")
}
TukeyHSD(aov.chr12MA)
{
  # pdf("Tukey.chr12.MA.pdf")
plot(TukeyHSD(aov.chr12MA))}
#shapiro.test(resids.chr12MA)
#fail shapiro test, so should probably use nonparametric

#kruskal.test(y~Line,data=chr12DataMA)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr12DataMA)
#OR can do many pairwise t-tests
#pairwise.wilcox.test(chr12DataMA$y,chr12DataMA$Line,p.adjust.method = "fdr")
##############################################################################################
#CHROMOSOME 13

{

chr13DataMA <- data.frame(y=c((log2(l029.13[,3])),(log2(l050.13[,3])),(log2(l112.13[,3])),(log2(l115.13[,3])),(log2(l117.13[,3])),
        (log2(l123.13[,3])),(log2(l152.13[,3]))), Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.13$tracking_id))))
}
chr13DataMA$Line <-factor(chr13DataMA$Line)
#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_13_MA.pdf")
boxplot(y~Line,data=chr13DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("cyan", "green","cyan","green","cyan","cyan","cyan"),main="MA Lines Chromsome 13",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


}

#text(x=0.8, y=1.8, "*", pos=3, cex=2, col="red")
#text(x=1, y=1.8, "*", pos=3, cex=2, col="red")
#text(x=1.2, y=1.8, "*", pos=3, cex=2, col="red")

#ANOVA/NonParametric Equivalent
aov.chr13MA <- aov(y~Line,data=chr13DataMA)
summary(aov.chr13MA)
resids.chr13MA <- resid(aov.chr13MA)
{
  # pdf("Chrm13.residsMA.pdf")
hist(resids.chr13MA,col="turquoise",breaks=10,xlab="Residuals.Chr.13MA")
}


TukeyHSD(aov.chr13MA)
{
  # pdf("Tukey.chr13.MA.pdf")
plot(TukeyHSD(aov.chr13MA))}
#shapiro.test(resids.chr13MA)
#fail shapiro test, so should probably use nonparametric

#kruskal.test(y~Line,data=chr13DataMA)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr13DataMA)
#OR can do many pairwise t-tests
#pairwise.wilcox.test(chr13DataMA$y,chr13DataMA$Line,p.adjust.method = "fdr")
##############################################################################################
#CHROMOSOME 14


{

chr14DataMA <- data.frame(y=c((log2(l029.14[,3])),(log2(l050.14[,3])),(log2(l112.14[,3])),(log2(l115.14[,3])),(log2(l117.14[,3])),
        (log2(l123.14[,3])),(log2(l152.14[,3]))), Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.14$tracking_id))))
}
chr14DataMA$Line <-factor(chr14DataMA$Line)
#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_14_MA.pdf")
boxplot(y~Line,data=chr14DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("cyan", "green","cyan","green","cyan","cyan","cyan"),main="MA Lines Chromsome 14",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


}

#text(x=0.8, y=1.8, "*", pos=3, cex=2, col="red")
#text(x=1, y=1.8, "*", pos=3, cex=2, col="red")
#text(x=1.2, y=1.8, "*", pos=3, cex=2, col="red")

#ANOVA/NonParametric Equivalent
aov.chr14MA <- aov(y~Line,data=chr14DataMA)
summary(aov.chr14MA)
resids.chr14MA <- resid(aov.chr14MA)
{
  # pdf("Chrm14.residsMA.pdf")
hist(resids.chr14MA,col="turquoise",breaks=10,xlab="Residuals.Chr.14MA")
}
TukeyHSD(aov.chr14MA)
{
  # pdf("Tukey.chr14.MA.pdf")
plot(TukeyHSD(aov.chr14MA))}
#shapiro.test(resids.chr14MA)
#fail shapiro test, so should probably use nonparametric

#kruskal.test(y~Line,data=chr14DataMA)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr14DataMA)
#OR can do many pairwise t-tests
#pairwise.wilcox.test(chr14DataMA$y,chr14DataMA$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 15

{

chr15DataMA <- data.frame(y=c((log2(l029.15[,3])),(log2(l050.15[,3])),(log2(l112.15[,3])),(log2(l115.15[,3])),(log2(l117.15[,3])),
        (log2(l123.15[,3])),(log2(l152.15[,3]))), Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.15$tracking_id))))
}
chr15DataMA$Line <-factor(chr15DataMA$Line)
#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_15_MA.pdf")
boxplot(y~Line,data=chr15DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("cyan", "green","cyan","green","cyan","cyan","cyan"),main="MA Lines Chromsome 15",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


}

#text(x=0.8, y=1.8, "*", pos=3, cex=2, col="red")
#text(x=1, y=1.8, "*", pos=3, cex=2, col="red")
#text(x=1.2, y=1.8, "*", pos=3, cex=2, col="red")

#ANOVA/NonParametric Equivalent
aov.chr15MA <- aov(y~Line,data=chr15DataMA)
summary(aov.chr15MA)
resids.chr15MA <- resid(aov.chr15MA)
{
  # pdf("Chrm15.residsMA.pdf")
hist(resids.chr15MA,col="turquoise",breaks=10,xlab="Residuals.Chr.15MA")
}

TukeyHSD(aov.chr15MA)
{
  # pdf("Tukey.chr15.MA.pdf")
plot(TukeyHSD(aov.chr15MA))}
#shapiro.test(resids.chr15MA)
#fail shapiro test, so should probably use nonparametric

#kruskal.test(y~Line,data=chr15DataMA)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr15DataMA)
#OR can do many pairwise t-tests
#pairwise.wilcox.test(chr15DataMA$y,chr15DataMA$Line,p.adjust.method = "fdr")

##############################################################################################
#CHROMOSOME 16

{

chr16DataMA <- data.frame(y=c((log2(l029.16[,3])),(log2(l050.16[,3])),(log2(l112.16[,3])),(log2(l115.16[,3])),(log2(l117.16[,3])),
        (log2(l123.16[,3])),(log2(l152.16[,3]))), Line=rep(c(29,50,112,115,117,123,152),each=(length(l029.16$tracking_id))))
}
chr16DataMA$Line <-factor(chr16DataMA$Line)
#This code works beautifully to produce a boxplot 
{
# pdf("Chrm_16_MA.pdf")
boxplot(y~Line,data=chr16DataMA,xlab="Line",ylab="log2 Ratio", cex.lab=1.3,col=c("cyan", "green","deeppink1","green","cyan","cyan","cyan"),main="MA Lines Chromsome 16",las=3,ylim=c(-2,2.5))

abline(h=0,lty=2)
abline(h=0.585,lty=2,col="forestgreen")
abline(h=-1,lty=2,col="darkviolet")


text(x=2.8, y=1.8, "*", pos=3, cex=2, col="red")
text(x=3, y=1.8, "*", pos=3, cex=2, col="red")
text(x=3.2, y=1.8, "*", pos=3, cex=2, col="red")

text(x=6.8, y=1.8, "*", pos=3, cex=2, col="red")
text(x=7, y=1.8, "*", pos=3, cex=2, col="red")
text(x=7.2, y=1.8, "*", pos=3, cex=2, col="red")

brackets(3, 2, 7, 2, h = NULL,  ticks = 0.5, curvature = 0.5, type = 1,
col = 1, lwd = 1, lty = 1, xpd = FALSE)

text(x=4.8, y=2.1, "*", pos=3, cex=2, col="red")
text(x=5, y=2.1, "*", pos=3, cex=2, col="red")
text(x=5.2, y=2.1, "*", pos=3, cex=2, col="red")

}

#


## Export ANOVAs

capture.output(summary(aovchr3MA),file="chr3MA.doc")
capture.output(summary(aovchr4MA),file="chr4MA.doc")
capture.output(summary(aovchr5MA),file="chr5MA.doc")
capture.output(summary(aovchr6MA),file="chr6MA.doc")
capture.output(summary(aovchr7MA),file="chr7MA.doc")
capture.output(summary(aovchr8MA),file="chr8MA.doc")
capture.output(summary(aovchr9MA),file="chr9MA.doc")
capture.output(summary(aovchr10MA),file="chr10MA.doc")
capture.output(summary(aovchr11MA),file="chr11MA.doc")
capture.output(summary(aovchr12MA),file="chr12MA.doc")
capture.output(summary(aovchr13MA),file="chr13MA.doc")
capture.output(summary(aovchr14MA),file="chr14MA.doc")
capture.output(summary(aovchr15MA),file="chr15MA.doc")
capture.output(summary(aovchr16MA),file="chr16MA.doc")

capture.output(summary(aovChr1GC),file="chr1GC.doc")
capture.output(summary(aovchr2GC),file="chr2GC.doc")
capture.output(summary(aovchr3GC),file="chr3GC.doc")
capture.output(summary(aovchr4GC),file="chr4GC.doc")
capture.output(summary(aovchr5GC),file="chr5GC.doc")
capture.output(summary(aovchr6GC),file="chr6GC.doc")
capture.output(summary(aovchr7GC),file="chr7GC.doc")
capture.output(summary(aovchr8GC),file="chr8GC.doc")
capture.output(summary(aovchr9GC),file="chr9GC.doc")
capture.output(summary(aovchr10GC),file="chr10GC.doc")
capture.output(summary(aovchr11GC),file="chr11GC.doc")
capture.output(summary(aovchr12GC),file="chr12GC.doc")
capture.output(summary(aovchr13GC),file="chr13GC.doc")
capture.output(summary(aovchr14GC),file="chr14GC.doc")
capture.output(summary(aovchr15GC),file="chr15GC.doc")
capture.output(summary(aovchr16GC),file="chr16GC.doc")
#kruskal.test(y~Line,data=chr16DataMA)
#test pairwise comparisons
#but there is an equivalent for Tukeys with kruskal
#Citation Siegel and Castellan (1988) Non parametric statistics for the behavioural sciences. MacGraw Hill Int., New York. pp 213-214
#kruskalmc(y~Line,data=chr16DataMA)
#OR can do many pairwise t-tests
#pairwise.wilcox.test(chr16DataMA$y,chr16DataMA$Line,p.adjust.method = "fdr")
```

Codes to isolate the lowly-expressed genes 
```{r lowly-expr genes, echo=FALSE, include = TRUE}
#Need to export the lowly-expressed genes from each ratio file produced from code chunk above 
#Want to do this in a loop

#ratioFiles<- list("l001.1","l001.10","l001.11","l001.12","l001.13","l001.14","l001.15","l001.16","l001.2","l001.3","l001.4","l001.5","l001.6","l001.7","l001.8","l001.9","l002.1","l002.10","l002.11","l002.12","l002.13","l002.14","l002.15","l002.16","l002.2","l002.3","l002.4","l002.5","l002.6","l002.7","l002.8","l002.9","l003.1","l003.10","l003.11","l003.12","l003.13","l003.14","l003.15","l003.16","l003.2","l003.3","l003.4","l003.5","l003.6","l003.7","l003.8","l003.9","l004.1","l004.10","l004.11","l004.12","l004.13","l004.14","l004.15","l004.16","l004.2","l004.3","l004.4","l004.5","l004.6","l004.7","l004.8","l004.9")
#ratioFiles<- c("l001.1","l001.10")
# setwd="/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/ratioFiles"

#for (i in ratioFiles) {
 # sample <- data.table(ratioFiles[i])
#  sample <- sample[which(sample[,3]<0.1),]
 # write.csv(unique(ratioFiles[i]),file=(paste(unique(ratioFiles[i]),".csv",sep="")))
#}



#setNames(list(l001.2,l002.2), c("l001.2","l002.2"))
ratioFiles <- list(l001.1,  l001.10,  l001.11,  l001.12,  l001.13,  l001.14,  l001.15,  l001.16,  l001.2,  l001.3,  l001.4,  l001.5,  l001.6,  l001.7,  l001.8,  l001.9,  l002.1,  l002.10,  l002.11,  l002.12,  l002.13,  l002.14,  l002.15,  l002.16,  l002.2,  l002.3,  l002.4,  l002.5,  l002.6,  l002.7,  l002.8,  l002.9,  l003.1,  l003.10,  l003.11,  l003.12,  l003.13,  l003.14,  l003.15,  l003.16,  l003.2,  l003.3, l003.4, l003.5,  l003.6,  l003.7, l003.8,  l003.9,  l004.1,  l004.10,  l004.11,  l004.12,  l004.13, l004.14, l004.15, l004.16, l004.2,  l004.3,  l004.4,  l004.5,  l004.6,  l004.7,  l004.8,  l004.9, l005.1, l005.10, l005.11, l005.12, l005.13, l005.14, l005.15, l005.16, l005.2, l005.3, l005.4, l005.6, l005.7, l005.8, l005.9, l007.1, l007.10, l007.11, l007.12, l007.13, l007.14, l007.15, l007.16, l007.2, l007.3, l007.4, l007.5, l007.6, l007.8, l007.9, l008.1, l008.10, l008.11, l001.12, l008.13, l008.14, l008.15, l008.16, l008.2, l008.3, l008.4, l008.5, l008.6, l008.7, l008.8, l008.9, l009.1, l009.10, l009.11, l009.12, l009.13, l009.14, l009.15, l009.16, l009.2, l009.3, l009.4, l009.5, l009.6, l009.7, l009.8, l009.9, l011.1, l011.10, l011.11, l011.12, l011.13, l011.14, l011.15, l011.16, l011.2, l011.3, l011.4, l011.5, l011.6, l011.7, l011.8, l011.9, l018.1, l018.10, l018.11, l018.12, l018.13, l018.14, l018.15, l018.16, l018.2, l018.3, l018.4, l018.5, l018.7, l018.6, l018.8, l018.9, l049.1, l049.10, l049.11, l049.12, l049.13, l049.15, l049.16, l049.2, l049.3, l049.4, l049.5, l049.6, l049.7, l049.8, l049.9, l059.1, l059.10, l059.11, l059.12, l059.13, l059.14, l059.15, l059.16, l059.2, l059.3, l059.4, l059.5, l059.6, l059.7, l059.8, l059.9, l061.1, l061.10, l061.11, l061.12, l061.13, l061.14, l061.15, l061.16, l061.2, l061.3, l061.4, l061.5, l061.6, l061.7, l061.8, l061.9, l069.1, l069.10, l069.11, l069.12, l069.13, l069.14, l069.15, l069.16, l069.2, l069.3, l069.4, l069.5, l069.6, l069.7, l069.8, l069.9, l076.1, l076.10, l076.11, l076.12, l076.13, l076.14, l076.15, l076.16, l076.2, l076.3, l076.4, l076.5, l076.6, l076.7, l076.8, l076.9, l077.1, l077.10, l077.11, l077.12, l077.13, l077.14, l077.15, l077.16, l077.2, l077.3, l077.4, l077.5, l077.6, l077.7, l077.8, l077.9)
names(ratioFiles) <-  c("l001.1","l001.10","l001.11","l001.12","l001.13","l001.14","l001.15","l001.16","l001.2","l001.3","l001.4","l001.5","l001.6","l001.7","l001.8","l001.9","l002.1","l002.10","l002.11","l002.12","l002.13","l002.14","l002.15","l002.16","l002.2","l002.3","l002.4","l002.5","l002.6","l002.7","l002.8","l002.9","l003.1","l003.10","l003.11","l003.12","l003.13","l003.14","l003.15","l003.16","l003.2","l003.3","l003.4","l003.5","l003.6","l003.7","l003.8","l003.9","l004.1","l004.10","l004.11","l004.12","l004.13","l004.14","l004.15","l004.16","l004.2","l004.3","l004.4","l004.5","l004.6","l004.7","l004.8","l004.9","l005.1","l005.10","l005.11","l005.12","l005.13","l005.14","l005.15","l005.16","l005.2","l005.3","l005.4","l005.6","l005.7","l005.8","l005.9","l007.1","l007.10","l007.11","l007.12","l007.13","l007.14","l007.15","l007.16","l007.2","l007.3","l007.4","l007.5","l007.6","l007.8","l007.9","l008.1","l008.10","l008.11","l001.12","l008.13","l008.14","l008.15","l008.16","l008.2","l008.3","l008.4","l008.5","l008.6","l008.7","l008.8","l008.9","l009.1","l009.10","l009.11","l009.12","l009.13","l009.14","l009.15","l009.16","l009.2","l009.3","l009.4","l009.5","l009.6","l009.7","l009.8","l009.9","l011.1","l011.10","l011.11","l011.12","l011.13","l011.14","l011.15","l011.16","l011.2","l011.3","l011.4","l011.5","l011.6","l011.7","l011.8","l011.9","l018.1","l018.10","l018.11","l018.12","l018.13","l018.14","l018.15","l018.16","l018.2","l018.3","l018.4","l018.5","l018.7","l018.6","l018.8","l018.9","l049.1","l049.10","l049.11","l049.12","l049.13","l049.15","l049.16","l049.2","l049.3","l049.4","l049.5","l049.6","l049.7","l049.8","l049.9","l059.1","l059.10","l059.11","l059.12","l059.13","l059.14","l059.15","l059.16","l059.2","l059.3","l059.4","l059.5","l059.6","l059.7","l059.8","l059.9","l061.1","l061.10","l061.11","l061.12","l061.13","l061.14","l061.15","l061.16","l061.2","l061.3","l061.4","l061.5","l061.6","l061.7","l061.8","l061.9","l069.1","l069.10","l069.11","l069.12","l069.13","l069.14","l069.15","l069.16","l069.2","l069.3","l069.4","l069.5","l069.6","l069.7","l069.8","l069.9","l076.1","l076.10","l076.11","l076.12","l076.13","l076.14","l076.15","l076.16","l076.2","l076.3","l076.4","l076.5","l076.6","l076.7","l076.8","l076.9","l077.1","l077.10","l077.11","l077.12","l077.13","l077.14","l077.15","l077.16","l077.2","l077.3","l077.4","l077.5","l077.6","l077.7","l077.8","l077.9")

ratioFiles
for (i in seq_along(ratioFiles)){
  file <- data.table(ratioFiles[[i]])
 clean.file <- ratioFiles[[i]][(ratioFiles[[i]][3] < 0.1), ]
  write.csv(clean.file, file=paste(names(ratioFiles)[i], "low_exp_genes.csv", sep="_"), row.names = F)
}


# ratioFiles
for (i in seq_along(ratioFiles)){
  file <- data.table(ratioFiles[[i]])
 clean.file <- ratioFiles[[i]][(ratioFiles[[i]][3] > 2), ]
  write.csv(clean.file, file=paste(names(ratioFiles)[i], "high_exp_genes.csv", sep="_"), row.names = F)
}

#low_expr_genes <- function(filename) {
#    file <- data.table(filename)
# clean.file <- filename[which(filename[,3]<0.1),]
#  write.csv(clean.file, file=paste(unique(filename), "low_exp_genes.csv", sep="_"), row.names = F)
#}

#low_expr_genes(l002.2)

#for(i in 1:5){
#  f()
#}
#for (i in  seq_along(ratioFiles)) {
#  low_expr_genes()
#}

#complete <- function(low_exp, id = 1:2) {
 #       ratioFiles<- list(l001.2,l002.2)
  #      dat <- data.table()
   #     tmp <- data.table()
    #            for (i in id) {
     #                   dat <- rbind(dat, ratioFiles[i])
      #                  clean.dat <- filename[which(filename[,3]<0.1),]
       #                 id <- i
        #                tmp <- rbind(tmp,data.frame(id,clean.dat))
         #       }
      #  write.csv(tmp, file=paste(unique(ratioFiles[i]), "low_exp_genes.csv", sep="_"), row.names = F)
#}

#complete <- function(directory, id = 1:332) {
 #       files_list <- list.files(path = directory, full.names = TRUE)
  #      dat <- data.frame()
   #     tmp <- data.frame()
    #            for (i in id) {
     #                   dat <- rbind(dat, read.csv(files_list[i]))
      #                  nobs <- sum(complete.cases(dat))
       #                 id <- i
        #                tmp <- rbind(tmp,data.frame(id,nobs))
         #       }


    #    tmp
#}

#complete
#file <- data.table(ratioFiles[[i]])
# clean.file <- ratioFiles[[i]][(ratioFiles[[i]]$ratio < 0.1), ]
#  write.csv(clean.file, file=paste(names(ratioFiles)[i], "low_exp_genes.csv", sep="_"), row.names = F)

#names(ratioFiles)
 
#write.csv(l004.9,file=(paste("Lowly_expr_genes","l004.9",".csv",sep="")))
 # sample <- read.table(ratioFiles[i],header=TRUE)
  #sample <- sample[which(sample[,3]<0.1),]
  #sample <- data.table(sample)  # make it a data table
  #dev.off()

```


```{r fpkm vs p-value, echo=FALSE,include=FALSE}
#first need to read in the Cuffdiff data 
#first do this for one line, then make a loop to do it for each line
# setwd = "/Users/hollymcqueary/Dropbox/DosageCompensation/WholeChromAnalyses2017/Cuffdiff_files/Diff/Tables"
# samples <- c(1,2,3,4,5,6,7,8,9,11,18,29,49,50,59,61,69,76,77,112,115,123,152)

#sample1 <- read.table("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Redo/Cuffdiff_files/gene_exp_S1.diff",header=TRUE)

#plot(sample1$q_value,sample1$value_2,ylab="ancestor FPKM",xlab="FDR-adj p-value",main="Sample1")

# list_of_titles <- c("Sample1","Sample2","Sample3","Sample4","Sample5","Sample6","Sample7","Sample8",
#                     "Sample9","Sample11","Sample18","Sample29","Sample49","Sample50",
#                     "Sample59","Sample61","Sample69","Sample76","Sample77","Sample112","Sample115",
#                     "Sample117","Sample123","Sample152")
# 
# 
# # Plot separate ggplot figures in a loop.
# library(ggplot2)
# 
# # Make list of variable names to loop over.
# myFiles <- list.files(path="/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Redo/Cuffdiff_files/Diff/Tables",pattern = "*.diff", full.names = TRUE)
# str(myFiles)
# length(myFiles)
# print(myFiles)
# #View(myFiles)
# # Make plots.
# plot_list = list()
# 
# #plot each sample in myFiles, then save to a .pdf in the working directory 
# for (i in 1:length(myFiles)) {
#   setwd = "/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Redo/Cuffdiff_files/Diff/Tables"
#   pdf(paste("FPKM_vs_p_val_",list_of_titles[i],".pdf",sep=""))
#   sample <- read.table(myFiles[i],header=TRUE)
#   sample <- data.frame(sample)
#   sample <- cSplit(sample,"locus",":") #split the locus column into "chr" and "locus"
#   names <- c("test_id","gene_id","gene","sample_1","sample_2","status","value_1","value_2","log2.fold_change","test_stat","p_value","q_value","significant","chr","locus")
#   colnames(sample) <- names
#   sample <-  subset(sample, !grepl("MT", sample$chr))
#   qval <- sample$q_value
#   val2 <- sample$value_2
#   plot(qval,val2,ylab="ancestor FPKM",xlab="FDR-adj p-value", 
#        xlim=c(0,1),ylim=c(0,10000),main=list_of_titles[i])
#   dev.off() 
# }


```

```{r finding_depth, echo=FALSE,include=FALSE}
#bam <- scanBam("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/bam_files/Holly-66B-redo-136232_S14_R1_001_tophat_out.sorted.bam")[[1]] # the result comes in nested lists
# bamcoverage <- function (bamfile) {
#   # read in the bam file
#   bam <- scanBam(bamfile)[[1]]
#   # filter reads without match position
#   ind <- ! is.na(bam$pos)
#   ## remove non-matches, they are not relevant to us
#   bam <- lapply(bam, function(x) x[ind])
#   ranges <- IRanges(start=bam$pos, width=bam$qwidth, names=make.names(bam$qname, unique=TRUE))
#   ## names of the bam data frame:
#   ## "qname"  "flag"   "rname"  "strand" "pos"    "qwidth"
#   ## "mapq"   "cigar"  "mrnm"   "mpos"   "isize"  "seq"    "qual"
#   ## construc: genomic ranges object containing all reads
#   ranges <- GRanges(seqnames=Rle(bam$rname), ranges=ranges, strand=Rle(bam$strand), flag=bam$flag, readid=bam$rname )
#   ## returns a coverage for each reference sequence (aka. chromosome) in the bam file
#   return (mean(coverage(ranges)))      
# }
# 
# myBamFiles <- list.files(path="/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/bam_files/",pattern="*.bam",full.names=TRUE)
# 
# print(myBamFiles)
# list_of_titles <- c("11B","11C","18B","18C","1B","21B","2A","2B","2C","31B","31C","3A","3B","3C","49B","49C","4B","4C","59B","59C","5A","5B","5C","61B","61C","66C","69B","69C","6C","76B","76C","77B","77C","7C","8B","9A","9C","1","59","66","69","76","77","66B","11","18","21","4","49","61","7","8")
# setwd = "/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/bam_files/"
# 
# for (i in 1:length(myBamFiles)) {
# sample <- bamcoverage(myBamFiles[i])
# sample <- data.frame(sample)
# write.csv(sample,file=(paste("Depth_of_coverage_",list_of_titles[i],".csv",sep="")))
# }
# 
# {sample <- bamcoverage("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/bam_files/Sample76.sorted.fixed.marked.realigned.bam")
# sample <- data.frame(sample)
# write.csv(sample,file="Depth_of_coverage_Line76_DNA.csv")}
# 
# {sample <- bamcoverage("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/bam_files/Sample66.sorted.fixed.marked.realigned.bam")
# sample <- data.frame(sample)
# write.csv(sample,file="Depth_of_coverage_Line66_DNA.csv")}
# 
# {sample <- bamcoverage("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/bam_files/Sample8.sorted.fixed.marked.realigned.bam")
# sample <- data.frame(sample)
# write.csv(sample,file="Depth_of_coverage_Line8_DNA.csv")}
# 
# {sample <- bamcoverage("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/November2017/bam_files/11A_accepted_hits.bam")
# sample <- data.frame(sample)
# write.csv(sample,file="Depth_of_coverage_Line11_RNA.csv")}
# 
# {sample <- bamcoverage("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Final_Chrom_wide_Analyses/bam_files/new_ones/accepted_hits_GCAnc_B.bam")
# sample <- data.frame(sample)
# write.csv(sample,file="Depth_of_coverage_GCAnc_B_RNA.csv")}
# 
# 
# {sample <- bamcoverage("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Final_Chrom_wide_Analyses/bam_files/new_ones/accepted_hits_GCAnc_A.bam")
# sample <- data.frame(sample)
# write.csv(sample,file="Depth_of_coverage_GCAnc_A_RNA.csv")}
# 
# 
# {sample <- bamcoverage("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Final_Chrom_wide_Analyses/bam_files/new_ones/accepted_hits_GCAnc_C.bam")
# sample <- data.frame(sample)
# write.csv(sample,file="Depth_of_coverage_GCAnc_C_RNA.csv")}


```

```{r DE vs total genes, echo=FALSE,include=FALSE}
########################################################################################################
#Cuffdiff: Number of differentially expressed genes versus the number of genes on given chromosome
########################################################################################################

#read in the data
# myFiles <- list.files(path="/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Redo/Cuffdiff_files/Diff/Tables",pattern = "*.diff")
# 
# print(myFiles)
# getwd()
# setwd("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Redo/Cuffdiff_files/Diff/Tables")
# 
# list_of_titles <- c("Sample1","Sample2","Sample3","Sample4","Sample5","Sample6","Sample7","Sample8",
#                     "Sample9","Sample11","Sample18","Sample21","Sample29","Sample31","Sample49","Sample50",
#                     "Sample59","Sample61","Sample66","Sample69","Sample76","Sample77","Sample112","Sample115","Sample117","Sample123","Sample141","Sample152")
# 
# for (i in 1:length(myFiles)) {
#   pdf(paste("DE_by_total_",list_of_titles[i],".pdf",sep=""))
#   sample <- read.table(myFiles[i],header=TRUE)
#   sample <- data.table(sample)  # make it a data table
#   sample <- cSplit(sample,"locus",":") #split the locus column into "chr" and "locus"
#   names <- c("test_id","gene_id","gene","sample_1","sample_2","status","value_1","value_2","log2.fold_change","test_stat","p_value","q_value","significant","chr","locus")
#   colnames(sample) <- names
#    #remove the mitochondria
#   sample <-  subset(sample, !grepl("MT", sample$chr))
#   chrms <- c("I","II","III","IV","V","VI","VII","VIII","IX","X","XI","XII","XIII","XIV","XV","XVI") #need to count number of rows where sample1$chr == "1" 
# #and so on for each chromosome
#   DE <- subset(sample,significant=="yes") #subset data into the "yeses"
#   N.DE <- DE[,.N,by=chr] #count the number of rows of each chromosome that are DE (this is the number of genes on each chromosome)
#   N.genes <- sample[,.N,by=chr] #count the number of rows of each chromosome (this is the number of genes on each chromosome)
#   #tmp <- N.genes[which(N.genes[,1]==N.DE[,1]),]
#  test <- merge(N.genes,N.DE,by="chr")
#  name <- c("chr","N.genes","N.DE")
#  colnames(test) <- name
#  plot(N.DE~N.genes,data=test , xlab="total genes", ylab="differentially expressed genes",
#     pch=1,xlim=c(0,1000),ylim=c(0,15),main=list_of_titles[i])
#  abline(fit <- lm(N.DE ~ N.genes, data = test), col="blue")
#  legend("topright", bty="n", legend=paste("R^2 =", format(summary(fit)$adj.r.squared, digits=4)))
# coeff=data.frame(coefficients(fit))
#  legend(700,14, bty="n", legend=paste("slope =", coef(fit)[2]))
#   dev.off()
# }
# 
# #next, put point on x-axis for the chromosomes that have 0 DE genes
# #to do this, modify the data table to 
# 
# 
# #rates = read.csv("file.txt")
# #for(i in unique(rates[,1])){
# #  pdf(file = paste0(i, ".pdf"))
# #  match <- rates[rates[,1] == i, ]
# #  plot(freq ~ temp, data = match)#,xlab="freq",ylab="temp")
# #  abline(lm(freq ~ temp, data = match), col="red")
# #  dev.off()
# #}
# 
# #all(is.na(All$all))
# #all(is.na(y))
# #all(is.na(y^trans))
# 
# 
# #  sample <- read.table("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Redo/Cuffdiff_files/Diff/Tables/gene_exp_S1.diff",header=TRUE)
#  # sample <- data.table(sample)  # make it a data table
#   #sample <- cSplit(sample,"locus",":") #split the locus column into "chr" and "locus"
# #  names <- c("test_id","gene_id","gene","sample_1","sample_2","status","value_1","value_2","log2.fold_change","test_stat","p_value","q_value","significant","chr","locus")
#  # colnames(sample) <- names
# #  chrms <- c("I","II","III","IV","V","VI","VII","VIII","IX","X","XI","XII","XIII","XIV","XV","XVI") #need to count number of rows where sample1$chr == "1" 
# #and so on for each chromosome
# #  DE <- subset(sample,significant=="yes") #subset data into the "yeses"
# #  N.DE <- DE[,.N,by=chr] #count the number of rows of each chromosome that are DE (this is the number of genes on each chromosome)
# #  N.genes <- sample[,.N,by=chr] #count the number of rows of each chromosome (this is the number of genes on each chromosome)
# #  All <- merge(N.DE, N.genes, by="chr")
# #  names(All) <- c("chr","DE","all")
#   #plot number of DE genes by number of total genes
#  # plot(all ~ DE, data=All,xlab="total genes",ylab="differentially expressed genes",
#   #     pch=1,xlim=c(0,1000),ylim=c(0,15),main=list_of_titles[i])
#   #abline(lm(All$all ~ All$DE)) #doesn't seem to work: error: "Error in lm.fit(x, y, offset = offset, singular.ok = singular.ok, ...) : 0 (non-NA) cases"
# 
# #tmp <- N.genes[which(N.genes[,1]==N.DE[,1]),]
# #test <- merge(N.genes,N.DE,by="chr")
# #name <- c("chr","N.genes","N.DE")
# #colnames(test) <- name
# #plot(N.DE~N.genes,data=test , xlab="total genes", ylab="differentially expressed genes",
#  #   pch=1,xlim=c(0,1000),ylim=c(0,15),main="test")
# 
# #abline(fit <- lm(N.DE ~ N.genes, data = test), col="red")
# #legend("topright", bty="n", legend=paste("R^2 =", format(summary(fit)$adj.r.squared, digits=4)))
# #coeff=data.frame(coefficients(fit))
# #legend(700,14, bty="n", legend=paste("slope =", coef(fit)[2]))
# #dev.list()
# 
# ```
# 
# ``` {r fitness, include=TRUE}
# #read in fitness data in .txt format
# fitness <- read.csv("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Fitness/fitness_data_HM.csv", header=TRUE, sep=",")
# 
# { 
#   pdf("fitness.by.aneuploidy.pdf")
#   plot(data=fitness, fitness$Aneuploidy.Event, fitness$Slope.Avg.anc.on.given.plate,cex.axis=0.5,las=2,pch=19)
#   
# }
```

Codes to find aneuploidy rate and determine statistical significance

```{r aneuploidy rate, include=TRUE, echo=FALSE}

library(xtable)
library(sjPlot)
library(stargazer)
df.total <- data.frame(matrix(ncol = 2, nrow = 2))
x <- c("hybrid","lab")
colnames(df.total) <- x
y <- c("aneuploid","euploid")
rownames(df.total) <- y 

df.total[1,1] <- 29
df.total[1,2] <- 31
df.total[2,1] <- 46
df.total[2,2] <- 121 

test <- fisher.test(df.total )#, alternative="two.sided", conf.int=TRUE)


# iris$LongPetal <- ifelse(iris$Petal.Length > 4.5, 1, 0)
# 
# sjt.xtab(df.total$hybrid, #rows
#          df.total$lab, #columns
#          file = "testContigency.doc")

stargazer(df.total, type="html",out="test.star.doc",summary=FALSE)

# library(arsenal)
# tab.test <- tableby(hybrid ~ lab, cat.test="fe", data=df.total)
# tests(tab.test)
# 
# star <- stargazer(df.total, type = "html", 
#                   title = "Stargazer with Summary = FALSE", 
#                   summary = FALSE, out="test.star.doc",)
# 
# stargazer(df.total,
#           #note you have to specify type
# type = "html",
# #note that the argument is "out" not "file" 
# out="test.star.doc")

x <- xtable(df.total)
print.xtable(x, type="html")

write.csv(df.total,file="totalRate.csv")

df.tri <- data.frame(matrix(ncol = 2, nrow = 2))
x <- c("aneuploid","euploid")
colnames(df.tri) <- x
y <- c("hybrid","lab")
rownames(df.tri) <- y 

df.tri[1,1] <- 28
df.tri[1,2] <- 47
df.tri[2,1] <- 29
df.tri[2,2] <- 116

fisher.test(df.tri, alternative="two.sided", conf.int=TRUE)

write.csv(df.tri,file="triRate.csv")



df.mon <- data.frame(matrix(ncol = 2, nrow = 2))
x <- c("aneuploid","euploid")
colnames(df.mon) <- x
y <- c("hybrid","lab")
rownames(df.mon) <- y 

df.mon[1,1] <- 1
df.mon[1,2] <- 74
df.mon[2,1] <- 2
df.mon[2,2] <- 145

fisher.test(df.mon, alternative="two.sided", conf.int=TRUE)

write.csv(df.mon,file="monRate.csv")


## rate




```


```{r coverage_graphs, include=FALSE,echo=FALSE}
# sample2 <- list.files(path = "/Users/hollymcqueary/Dropbox/mentoring/Ariella_Tsfoni/new_to_graph/s2_new", pattern="*.txt",full.names = TRUE)
# sample2


# file <- read.table("/Users/hollymcqueary/Dropbox/mentoring/Ariella_Tsfoni/new_to_graph/Sample29.Coverage10K.head.txt", sep= "\t", header=TRUE)
# plot(file$Coverage, ylab="Coverage",ylim=c(0,30000), xlab="Bins (10kb)",main="Sample29",xlim=0,1300,col=as.factor(file$Chromosome))


# file.names <- list.files(path = "/Users/hollymcqueary/Dropbox/mentoring/Ariella_Tsfoni/new_to_graph", pattern="*.txt",full.names = TRUE)
# file.names
# i=3
# for(i in 1:length(file.names)){
#   files <- read.table(file.names[i], sep= "\t", header=TRUE)
#   print(file.names[i])
#   pdf(paste(file.names[i],".CovGraph.pdf", sep=""))
#   plot(files$Coverage, col=as.factor(files$Chromosome), ylab="Coverage",ylim=c(0,30000),xlim=c(0,1300), xlab="Bins (10kb)",main=tools::file_path_sans_ext(basename(file.names))[i])
#   dev.off()
# }
 

```



Packages used in this pipeline 
```{r packages, include = TRUE, echo = TRUE }
library(moments)
library("Hmisc")
library(pBrackets)
library(data.table)
# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# 
# BiocManager::install("Rsamtools")
library(Rsamtools)
library(splitstackshape)
library(ggplot2)
library(pgirmess)
library(FSA)
library(dunn.test)
library(rcompanion)
library(grid)
library(dplyr)
library(tidyr)
library("knitr")
library(xtable)
library(sjPlot)
library(stargazer)
sessionInfo()

```
