---
title: "DCMethods"
output: html_document
---
```{r setup, eval=TRUE,include=FALSE,echo=FALSE}
library(knitr)
knitr::opts_chunk$set(tidy=FALSE, cache=TRUE,
                      dev="png",
                      message=FALSE, error=FALSE, warning=TRUE)
# source("https://bioconductor.org/biocLite.R")
#        biocLite("apeglm")
library(apeglm)
#source("https://bioconductor.org/biocLite.R")
 #       biocLite("parathyroidSE")
#library("parathyroidSE")
#       if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
#
# BiocManager::install("DESeq2")
library(DESeq2)
```

```{r graphs, eval=TRUE,include=TRUE,echo=FALSE}
library(ggplot2)

size_v_numAneus <- read.delim("/Users/hollymcqueary/Dropbox/McQueary/DosageCompensationG3/size_vs_numAneus.txt")

View(size_v_numAneus)

## Heterozygous Ancestor
sizeNumAneus <- ggplot(size_v_numAneus,aes(x=length,y=X.eventsGC))
sizeNumAneus + geom_point(color="turquoise") + theme_classic()  + geom_smooth(method = "lm", se=FALSE, color="turquoise") + labs(y="Aneuploidy Events", x="Chromosome Length", title="Heterozygous Ancestor") 

library(ggpmisc)
my.formula <- y ~ x
p <- ggplot(size_v_numAneus,aes(x=length,y=X.eventsGC))
p2 <- p + geom_point(color="turquoise") + theme_classic()  + geom_smooth(method = "lm", se=FALSE, color="turquoise",formula=my.formula) + stat_poly_eq(formula = my.formula, 
                aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), 
                parse = TRUE) + labs(y="Aneuploidy Events", x="Chromosome Length", title="Heterozygous Ancestor") 

pdf("het_anc_size_v_aneus.pdf")
p2
dev.off()
  

## Homozygous Ancestor
sizeNumAneus1 <- ggplot(size_v_numAneus,aes(x=length,y=X.eventsMA))
sizeNumAneus1 + geom_point(color="deeppink1") + theme_classic()  + geom_smooth(method = "lm", se=FALSE, color="deeppink1") + labs(y="Aneuploidy Events", x="Chromosome Length", title="Homozygous Ancestor") 

library(ggpmisc)
my.formula <- y ~ x
p1 <- ggplot(size_v_numAneus,aes(x=length,y=X.eventsMA))
p3 <- p + geom_point(color="deeppink1") + theme_classic()  + geom_smooth(method = "lm", se=FALSE, color="deeppink1",formula=my.formula) + stat_poly_eq(formula = my.formula, 
                aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), 
                parse = TRUE) + labs(y="Aneuploidy Events", x="Chromosome Length", title="Homozygous Ancestor") 

pdf("hom_anc_size_v_aneus.pdf")
p3
dev.off()






```



```{r htseqDirI, eval=TRUE,include=FALSE,echo=FALSE}
directoryGC_All <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC"

##individual lines
directoryGC_1 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_1vsAnc"
directoryGC_2 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_2vsAnc"
directoryGC_3 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_3vsAnc"
directoryGC_4 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_4vsAnc"
directoryGC_5 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_5vsAnc"
directoryGC_6 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_6vsAnc"
directoryGC_7 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_7vsAnc"
directoryGC_8 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_8vsAnc"
directoryGC_9 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_9vsAnc"
directoryGC_11 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_11vsAnc"
directoryGC_18 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_18vsAnc"
directoryGC_31 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_31vsAnc"
directoryGC_49 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_49vsAnc"
directoryGC_59 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_59vsAnc"
directoryGC_61 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_61vsAnc"
directoryGC_69 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_69vsAnc"
directoryGC_76 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_76vsAnc"
directoryGC_77 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_77vsAnc"
#########################################################
directoryMAold <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD"

directoryMAold_1 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/1vsAnc"
directoryMAold_2 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/2vsAnc"
directoryMAold_3 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/3vsAnc"
directoryMAold_4 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/4vsAnc"
directoryMAold_5 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/5vsAnc"
# directoryMAold_6 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/6vsAnc"
directoryMAold_7 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/7vsAnc"
directoryMAold_8 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/8vsAnc"
directoryMAold_9 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/9vsAnc"
directoryMAold_11 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/11vsAnc"
directoryMAold_15 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/15vsAnc"
directoryMAold_28 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/28vsAnc"
directoryMAold_88 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/88vsAnc"
directoryMAold_108 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/108vsAnc"
directoryMAold_119 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/119vsAnc"
########################################################
directoryMAnew <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW"

directoryMAnewS29 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/29vsAnc"

directoryMAnewS50 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/50vsAnc"

directoryMAnewS112 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/112vsAnc"

directoryMAnewS115 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/115vsAnc"

directoryMAnewS117 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/117vsAnc"

directoryMAnewS123 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/123vsAnc"

directoryMAnewS141 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/141vsAnc"

directoryMAnewS152 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/152vsAnc"
```

```{r htseqInput,eval=TRUE,include=FALSE,echo=FALSE}
sampleFilesGC_All <- list.files(directoryGC_All)
sampleInfoGC_All <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/sample_data_GC.csv",header=TRUE)
sampleTableGC_All <- data.frame(sampleName = sampleInfoGC_All$sample,
                          fileName = sampleInfoGC_All$filename,
                          run = sampleInfoGC_All$seq_run ,
                           condition = sampleInfoGC_All$group)

sampleFilesGC_1 <- list.files(directoryGC_1)
sampleInfoGC_1 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S1vsAnc.csv",header=TRUE)
sampleTableGC_1 <- data.frame(sampleName = sampleInfoGC_1$sample,
                          fileName = sampleInfoGC_1$filename,
                          run = sampleInfoGC_1$seq_run ,
                           condition = sampleInfoGC_1$group)

sampleFilesGC_2 <- list.files(directoryGC_2)
sampleInfoGC_2 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S2vsAnc.csv",header=TRUE)
sampleTableGC_2 <- data.frame(sampleName = sampleInfoGC_2$sample,
                          fileName = sampleInfoGC_2$filename,
                          run = sampleInfoGC_2$seq_run ,
                           condition = sampleInfoGC_2$group)

sampleFilesGC_3 <- list.files(directoryGC_3)
sampleInfoGC_3 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S3vsAnc.csv",header=TRUE)
sampleTableGC_3 <- data.frame(sampleName = sampleInfoGC_3$sample,
                          fileName = sampleInfoGC_3$filename,
                          run = sampleInfoGC_3$seq_run ,
                           condition = sampleInfoGC_3$group)

sampleFilesGC_4 <- list.files(directoryGC_4)
sampleInfoGC_4 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S4vsAnc.csv",header=TRUE)
sampleTableGC_4 <- data.frame(sampleName = sampleInfoGC_4$sample,
                          fileName = sampleInfoGC_4$filename,
                          run = sampleInfoGC_4$seq_run ,
                           condition = sampleInfoGC_4$group)

sampleFilesGC_5 <- list.files(directoryGC_5)
sampleInfoGC_5 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S5vsAnc.csv",header=TRUE)
sampleTableGC_5 <- data.frame(sampleName = sampleInfoGC_5$sample,
                          fileName = sampleInfoGC_5$filename,
                          run = sampleInfoGC_5$seq_run ,
                           condition = sampleInfoGC_5$group)

sampleFilesGC_6 <- list.files(directoryGC_6)
sampleInfoGC_6 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S6vsAnc.csv",header=TRUE)
sampleTableGC_6 <- data.frame(sampleName = sampleInfoGC_6$sample,
                          fileName = sampleInfoGC_6$filename,
                          run = sampleInfoGC_6$seq_run ,
                           condition = sampleInfoGC_6$group)

sampleFilesGC_7 <- list.files(directoryGC_7)
sampleInfoGC_7 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S7vsAnc.csv",header=TRUE)
sampleTableGC_7 <- data.frame(sampleName = sampleInfoGC_7$sample,
                          fileName = sampleInfoGC_7$filename,
                          run = sampleInfoGC_7$seq_run ,
                           condition = sampleInfoGC_7$group)

sampleFilesGC_8 <- list.files(directoryGC_8)
sampleInfoGC_8 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S8vsAnc.csv",header=TRUE)
sampleTableGC_8 <- data.frame(sampleName = sampleInfoGC_8$sample,
                          fileName = sampleInfoGC_8$filename,
                          run = sampleInfoGC_8$seq_run ,
                           condition = sampleInfoGC_8$group)

sampleFilesGC_9 <- list.files(directoryGC_9)
sampleInfoGC_9 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S9vsAnc.csv",header=TRUE)
sampleTableGC_9 <- data.frame(sampleName = sampleInfoGC_9$sample,
                          fileName = sampleInfoGC_9$filename,
                          run = sampleInfoGC_9$seq_run ,
                           condition = sampleInfoGC_9$group)

sampleFilesGC_11 <- list.files(directoryGC_11)
sampleInfoGC_11 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S11vsAnc.csv",header=TRUE)
sampleTableGC_11 <- data.frame(sampleName = sampleInfoGC_11$sample,
                          fileName = sampleInfoGC_11$filename,
                          run = sampleInfoGC_11$seq_run ,
                           condition = sampleInfoGC_11$group)

sampleFilesGC_18 <- list.files(directoryGC_18)
sampleInfoGC_18 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S18vsAnc.csv",header=TRUE)
sampleTableGC_18 <- data.frame(sampleName = sampleInfoGC_18$sample,
                          fileName = sampleInfoGC_18$filename,
                          run = sampleInfoGC_18$seq_run ,
                           condition = sampleInfoGC_18$group)

sampleFilesGC_31 <- list.files(directoryGC_31)
sampleInfoGC_31 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S31vsAnc.csv",header=TRUE)
sampleTableGC_31 <- data.frame(sampleName = sampleInfoGC_31$sample,
                          fileName = sampleInfoGC_31$filename,
                          run = sampleInfoGC_31$seq_run ,
                           condition = sampleInfoGC_31$group)

sampleFilesGC_49 <- list.files(directoryGC_49)
sampleInfoGC_49 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S49vsAnc.csv",header=TRUE)
sampleTableGC_49 <- data.frame(sampleName = sampleInfoGC_49$sample,
                          fileName = sampleInfoGC_49$filename,
                          run = sampleInfoGC_49$seq_run ,
                           condition = sampleInfoGC_49$group)

sampleFilesGC_59 <- list.files(directoryGC_59)
sampleInfoGC_59 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S59vsAnc.csv",header=TRUE)
sampleTableGC_59 <- data.frame(sampleName = sampleInfoGC_59$sample,
                          fileName = sampleInfoGC_59$filename,
                          run = sampleInfoGC_59$seq_run ,
                           condition = sampleInfoGC_59$group)

sampleFilesGC_61 <- list.files(directoryGC_61)
sampleInfoGC_61 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S61vsAnc.csv",header=TRUE)
sampleTableGC_61 <- data.frame(sampleName = sampleInfoGC_61$sample,
                          fileName = sampleInfoGC_61$filename,
                          run = sampleInfoGC_61$seq_run ,
                           condition = sampleInfoGC_61$group)

sampleFilesGC_69 <- list.files(directoryGC_69)
sampleInfoGC_69 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S69vsAnc.csv",header=TRUE)
sampleTableGC_69 <- data.frame(sampleName = sampleInfoGC_69$sample,
                          fileName = sampleInfoGC_69$filename,
                          run = sampleInfoGC_69$seq_run ,
                           condition = sampleInfoGC_69$group)

sampleFilesGC_76 <- list.files(directoryGC_76)
sampleInfoGC_76 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S76vsAnc.csv",header=TRUE)
sampleTableGC_76 <- data.frame(sampleName = sampleInfoGC_76$sample,
                          fileName = sampleInfoGC_76$filename,
                          run = sampleInfoGC_76$seq_run ,
                           condition = sampleInfoGC_76$group)

sampleFilesGC_77 <- list.files(directoryGC_77)
sampleInfoGC_77 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S77vaAnc.csv",header=TRUE)
sampleTableGC_77 <- data.frame(sampleName = sampleInfoGC_77$sample,
                          fileName = sampleInfoGC_77$filename,
                          run = sampleInfoGC_77$seq_run ,
                           condition = sampleInfoGC_77$group)




##############################################################################
sampleFilesMAold <- list.files(directoryMAold)
sampleInfoMAold <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_sample_data.csv",header=TRUE)
sampleTableMAold <- data.frame(sampleName = sampleInfoMAold$sample,
                          fileName = sampleInfoMAold$filename,
                          condition = sampleInfoMAold$group)

sampleFilesMAold_1 <- list.files(directoryMAold_1)
sampleInfoMAold_1 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S1vsAnc.csv",header=TRUE)
sampleTableMAold_1 <- data.frame(sampleName = sampleInfoMAold_1$sample,
                          fileName = sampleInfoMAold_1$filename,
                          condition = sampleInfoMAold_1$group)

sampleFilesMAold_2 <- list.files(directoryMAold_2)
sampleInfoMAold_2 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S2vsAnc.csv",header=TRUE)
sampleTableMAold_2 <- data.frame(sampleName = sampleInfoMAold_2$sample,
                          fileName = sampleInfoMAold_2$filename,
                          condition = sampleInfoMAold_2$group)

sampleFilesMAold_3 <- list.files(directoryMAold_3)
sampleInfoMAold_3 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S3vsAnc.csv",header=TRUE)
sampleTableMAold_3 <- data.frame(sampleName = sampleInfoMAold_3$sample,
                          fileName = sampleInfoMAold_3$filename,
                          condition = sampleInfoMAold_3$group)

sampleFilesMAold_4 <- list.files(directoryMAold_4)
sampleInfoMAold_4 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S4vsAnc.csv",header=TRUE)
sampleTableMAold_4 <- data.frame(sampleName = sampleInfoMAold_4$sample,
                          fileName = sampleInfoMAold_4$filename,
                          condition = sampleInfoMAold_4$group)

sampleFilesMAold_5 <- list.files(directoryMAold_5)
sampleInfoMAold_5 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S5vsAnc.csv",header=TRUE)
sampleTableMAold_5 <- data.frame(sampleName = sampleInfoMAold_5$sample,
                          fileName = sampleInfoMAold_5$filename,
                          condition = sampleInfoMAold_5$group)

# sampleFilesMAold_6 <- list.files(directoryMAold_6)
# sampleInfoMAold_6 <- read.csv("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Paper_Results/MA_old_data_S6vsAnc.csv",header=TRUE)
# sampleTableMAold_6 <- data.frame(sampleName = sampleInfoMAold_6$sample,
#                           fileName = sampleInfoMAold_6$filename,
#                           condition = sampleInfoMAold_6$group)

sampleFilesMAold_7 <- list.files(directoryMAold_7)
sampleInfoMAold_7 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S7vsAnc.csv",header=TRUE)
sampleTableMAold_7 <- data.frame(sampleName = sampleInfoMAold_7$sample,
                          fileName = sampleInfoMAold_7$filename,
                          condition = sampleInfoMAold_7$group)

sampleFilesMAold_8 <- list.files(directoryMAold_8)
sampleInfoMAold_8 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S8vsAnc.csv",header=TRUE)
sampleTableMAold_8 <- data.frame(sampleName = sampleInfoMAold_8$sample,
                          fileName = sampleInfoMAold_8$filename,
                          condition = sampleInfoMAold_8$group)

sampleFilesMAold_9 <- list.files(directoryMAold_9)
sampleInfoMAold_9 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S9vsAnc.csv",header=TRUE)
sampleTableMAold_9 <- data.frame(sampleName = sampleInfoMAold_9$sample,
                          fileName = sampleInfoMAold_9$filename,
                          condition = sampleInfoMAold_9$group)

sampleFilesMAold_11 <- list.files(directoryMAold_11)
sampleInfoMAold_11 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S11vsAnc.csv",header=TRUE)
sampleTableMAold_11 <- data.frame(sampleName = sampleInfoMAold_11$sample,
                          fileName = sampleInfoMAold_11$filename,
                          condition = sampleInfoMAold_11$group)

sampleFilesMAold_15 <- list.files(directoryMAold_15)
sampleInfoMAold_15 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S15vsAnc.csv",header=TRUE)
sampleTableMAold_15 <- data.frame(sampleName = sampleInfoMAold_15$sample,
                          fileName = sampleInfoMAold_15$filename,
                          condition = sampleInfoMAold_15$group)

sampleFilesMAold_28 <- list.files(directoryMAold_28)
sampleInfoMAold_28 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S28vsAnc.csv",header=TRUE)
sampleTableMAold_28 <- data.frame(sampleName = sampleInfoMAold_28$sample,
                          fileName = sampleInfoMAold_28$filename,
                          condition = sampleInfoMAold_28$group)

sampleFilesMAold_88 <- list.files(directoryMAold_88)
sampleInfoMAold_88 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S88vsAnc.csv",header=TRUE)
sampleTableMAold_88 <- data.frame(sampleName = sampleInfoMAold_88$sample,
                          fileName = sampleInfoMAold_88$filename,
                          condition = sampleInfoMAold_88$group)

sampleFilesMAold_108 <- list.files(directoryMAold_108)
sampleInfoMAold_108 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S108vsAnc.csv",header=TRUE)
sampleTableMAold_108 <- data.frame(sampleName = sampleInfoMAold_108$sample,
                          fileName = sampleInfoMAold_108$filename,
                          condition = sampleInfoMAold_108$group)

sampleFilesMAold_119 <- list.files(directoryMAold_119)
sampleInfoMAold_119 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S119vsAnc.csv",header=TRUE)
sampleTableMAold_119 <- data.frame(sampleName = sampleInfoMAold_119$sample,
                          fileName = sampleInfoMAold_119$filename,
                          condition = sampleInfoMAold_119$group)

###############################################################################
sampleFilesMAnew <- list.files(directoryMAnew)
sampleInfoMAnew <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_sample_data.csv",header=TRUE)
sampleTableMAnew <- data.frame(sampleName = sampleInfoMAnew$sample,
                          fileName = sampleInfoMAnew$filename,
                          run = sampleInfoMAnew$seq_run,
                           condition = sampleInfoMAnew$group)

sampleFilesMAnewS29 <- list.files(directoryMAnewS29)
sampleInfoMAnewS29 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S29.csv",header=TRUE)
sampleTableMAnewS29 <- data.frame(sampleName = sampleInfoMAnewS29$sample,
                          fileName = sampleInfoMAnewS29$filename,
                          run = sampleInfoMAnewS29$seq_run,
                           condition = sampleInfoMAnewS29$group)

sampleFilesMAnewS50 <- list.files(directoryMAnewS50)
sampleInfoMAnewS50 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S50.csv",header=TRUE)
sampleTableMAnewS50 <- data.frame(sampleName = sampleInfoMAnewS50$sample,
                          fileName = sampleInfoMAnewS50$filename,
                          run = sampleInfoMAnewS50$seq_run,
                           condition = sampleInfoMAnewS50$group)

sampleFilesMAnewS112 <- list.files(directoryMAnewS112)
sampleInfoMAnewS112 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S112.csv",header=TRUE)
sampleTableMAnewS112 <- data.frame(sampleName = sampleInfoMAnewS112$sample,
                          fileName = sampleInfoMAnewS112$filename,
                          run = sampleInfoMAnewS112$seq_run,
                           condition = sampleInfoMAnewS112$group)

sampleFilesMAnewS115 <- list.files(directoryMAnewS115)
sampleInfoMAnewS115 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S115.csv",header=TRUE)
sampleTableMAnewS115 <- data.frame(sampleName = sampleInfoMAnewS115$sample,
                          fileName = sampleInfoMAnewS115$filename,
                          run = sampleInfoMAnewS115$seq_run,
                           condition = sampleInfoMAnewS115$group)

sampleFilesMAnewS117 <- list.files(directoryMAnewS117)
sampleInfoMAnewS117 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S117.csv",header=TRUE)
sampleTableMAnewS117 <- data.frame(sampleName = sampleInfoMAnewS117$sample,
                          fileName = sampleInfoMAnewS117$filename,
                          run = sampleInfoMAnewS117$seq_run,
                           condition = sampleInfoMAnewS117$group)

sampleFilesMAnewS123 <- list.files(directoryMAnewS123)
sampleInfoMAnewS123 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S123.csv",header=TRUE)
sampleTableMAnewS123 <- data.frame(sampleName = sampleInfoMAnewS123$sample,
                          fileName = sampleInfoMAnewS123$filename,
                          run = sampleInfoMAnewS123$seq_run,
                           condition = sampleInfoMAnewS123$group)

# sampleFilesMAnewS141 <- list.files(directoryMAnewS141)
# sampleInfoMAnewS141 <- read.csv("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Paper_Results/MA_new_data_S141.csv",header=TRUE)
# sampleTableMAnewS141 <- data.frame(sampleName = sampleInfoMAnewS141$sample,
#                           fileName = sampleInfoMAnewS141$filename,
#                           run = sampleInfoMAnewS141$seq_run,
#                            condition = sampleInfoMAnewS141$group)

sampleFilesMAnewS152 <- list.files(directoryMAnewS152)
sampleInfoMAnewS152 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S152.csv",header=TRUE)
sampleTableMAnewS152 <- data.frame(sampleName = sampleInfoMAnewS152$sample,
                          fileName = sampleInfoMAnewS152$filename,
                          run = sampleInfoMAnewS152$seq_run,
                           condition = sampleInfoMAnewS152$group)

```

```{r other files,eval=TRUE, include=TRUE, echo=FALSE}
esrGenes <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/ESR_genes/ESR_genes_Gasch2000.csv",header=FALSE)
as.data.frame(esrGenes)
names(esrGenes)[1] <- "GENEID"
length(esrGenes)
dosageSensitiveGenes <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/DENonDEGenes/dosage_sensitive_genes.csv")
as.data.frame(dosageSensitiveGenes)
names(dosageSensitiveGenes)[1] <- "GENEID"
length(dosageSensitiveGenes)
ASRgenes <- read.table("/Users/hollymcqueary/Dropbox/DosageCompensation/ESR_genes/ASRGenes.txt",header=TRUE)
as.data.frame(ASRgenes)
length(ASRgenes)

```
Then we build the *DESeqDataSet* using the following function:

```{r hsteqDds,eval=TRUE,include=FALSE,echo=FALSE}
# library("DESeq2")
sampleTableGC_All$run <- as.factor(sampleTableGC_All$run)
sampleTableGC_All$sampleName <- as.factor(sampleTableGC_All$sampleName)
sampleTableGC_All$condition <- as.factor(sampleTableGC_All$condition)
ddsHTSeqGC_All <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_All,
                                       directory = directoryGC_All,
                                       design= ~ condition + run)
ddsHTSeqGC_All

write.csv(counts(ddsHTSeqGC_All),file="counts.DEseqGC.csv")

############################# Lines Individually ########################################
sampleTableGC_1$run <- as.factor(sampleTableGC_1$run)
sampleTableGC_1$sampleName <- as.factor(sampleTableGC_1$sampleName)
sampleTableGC_1$condition <- as.factor(sampleTableGC_1$condition)
ddsHTSeqGC_1 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_1,
                                       directory = directoryGC_1,
                                       design= ~ condition + run)
ddsHTSeqGC_1

# write.csv(counts(ddsHTSeqGC_1),file="counts.DEseqGC_S1.csv")
vsdGC_1 <- vst(ddsHTSeqGC_1,blind=FALSE)

########## line 2
sampleTableGC_2$run <- as.factor(sampleTableGC_2$run)
sampleTableGC_2$sampleName <- as.factor(sampleTableGC_2$sampleName)
sampleTableGC_2$condition <- as.factor(sampleTableGC_2$condition)
ddsHTSeqGC_2 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_2,
                                       directory = directoryGC_2,
                                       design= ~ condition + run)
ddsHTSeqGC_2


#### line 3
sampleTableGC_3$run <- as.factor(sampleTableGC_3$run)
sampleTableGC_3$sampleName <- as.factor(sampleTableGC_3$sampleName)
sampleTableGC_3$condition <- as.factor(sampleTableGC_3$condition)
ddsHTSeqGC_3 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_3,
                                       directory = directoryGC_3,
                                       design= ~ condition + run)
ddsHTSeqGC_3

# write.csv(counts(ddsHTSeqGC_3),file="counts.DEseqGC_S1.csv")

######## line 4
sampleTableGC_4$run <- as.factor(sampleTableGC_4$run)
sampleTableGC_4$sampleName <- as.factor(sampleTableGC_4$sampleName)
sampleTableGC_4$condition <- as.factor(sampleTableGC_4$condition)
ddsHTSeqGC_4 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_4,
                                       directory = directoryGC_4,
                                       design= ~ condition + run)
ddsHTSeqGC_4


################## line 5
  sampleTableGC_5$run <- as.factor(sampleTableGC_5$run)
sampleTableGC_5$sampleName <- as.factor(sampleTableGC_5$sampleName)
sampleTableGC_5$condition <- as.factor(sampleTableGC_5$condition)
ddsHTSeqGC_5 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_5,
                                       directory = directoryGC_5,
                                       design= ~ condition)# + run)
ddsHTSeqGC_5

# write.csv(counts(ddsHTSeqGC_5),file="counts.DEseqGC_S1.csv")

################## line 6
#   sampleTableGC_6$run <- as.factor(sampleTableGC_6$run)
# sampleTableGC_6$sampleName <- as.factor(sampleTableGC_6$sampleName)
# sampleTableGC_6$condition <- as.factor(sampleTableGC_6$condition)
# ddsHTSeqGC_6 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_6,
#                                        directory = directoryGC_6,
#                                        design= ~ condition)# + run)
# ddsHTSeqGC_6
#
# write.csv(counts(ddsHTSeqGC_6),file="counts.DEseqGC_S6.csv")


################## line 7
sampleTableGC_7$run <- as.factor(sampleTableGC_7$run)
sampleTableGC_7$sampleName <- as.factor(sampleTableGC_7$sampleName)
sampleTableGC_7$condition <- as.factor(sampleTableGC_7$condition)
ddsHTSeqGC_7 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_7,
                                       directory = directoryGC_7,
                                       design= ~ condition + run)
ddsHTSeqGC_7


#################### line 8
  sampleTableGC_8$run <- as.factor(sampleTableGC_8$run)
sampleTableGC_8$sampleName <- as.factor(sampleTableGC_8$sampleName)
sampleTableGC_8$condition <- as.factor(sampleTableGC_8$condition)
ddsHTSeqGC_8 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_8,
                                       directory = directoryGC_8,
                                       design= ~ condition + run)
ddsHTSeqGC_8


################# line 9
  sampleTableGC_9$run <- as.factor(sampleTableGC_9$run)
sampleTableGC_9$sampleName <- as.factor(sampleTableGC_9$sampleName)
sampleTableGC_9$condition <- as.factor(sampleTableGC_9$condition)
ddsHTSeqGC_9 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_9,
                                       directory = directoryGC_9,
                                       design= ~ condition + run)
ddsHTSeqGC_9


################## line 11
  sampleTableGC_11$run <- as.factor(sampleTableGC_11$run)
sampleTableGC_11$sampleName <- as.factor(sampleTableGC_11$sampleName)
sampleTableGC_11$condition <- as.factor(sampleTableGC_11$condition)
ddsHTSeqGC_11 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_11,
                                       directory = directoryGC_11,
                                       design= ~ condition + run)
ddsHTSeqGC_11


##################### line 18
sampleTableGC_18$run <- as.factor(sampleTableGC_18$run)
sampleTableGC_18$sampleName <- as.factor(sampleTableGC_18$sampleName)
sampleTableGC_18$condition <- as.factor(sampleTableGC_18$condition)
ddsHTSeqGC_18 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_18,
                                       directory = directoryGC_18,
                                       design= ~ condition + run)
ddsHTSeqGC_18


  ##################### line 31
sampleTableGC_31$run <- as.factor(sampleTableGC_31$run)
sampleTableGC_31$sampleName <- as.factor(sampleTableGC_31$sampleName)
sampleTableGC_31$condition <- as.factor(sampleTableGC_31$condition)
ddsHTSeqGC_31 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_31,
                                       directory = directoryGC_31,
                                       design= ~ condition + run)
ddsHTSeqGC_31


############## line 49
  sampleTableGC_49$run <- as.factor(sampleTableGC_49$run)
sampleTableGC_49$sampleName <- as.factor(sampleTableGC_49$sampleName)
sampleTableGC_49$condition <- as.factor(sampleTableGC_49$condition)
ddsHTSeqGC_49 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_49,
                                       directory = directoryGC_49,
                                       design= ~ condition + run)
ddsHTSeqGC_49


#################### line 59
sampleTableGC_59$run <- as.factor(sampleTableGC_59$run)
sampleTableGC_59$sampleName <- as.factor(sampleTableGC_59$sampleName)
sampleTableGC_59$condition <- as.factor(sampleTableGC_59$condition)
ddsHTSeqGC_59 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_59,
                                       directory = directoryGC_59,
                                       design= ~ condition + run)
ddsHTSeqGC_59


####################### line 61
  sampleTableGC_61$run <- as.factor(sampleTableGC_61$run)
sampleTableGC_61$sampleName <- as.factor(sampleTableGC_61$sampleName)
sampleTableGC_61$condition <- as.factor(sampleTableGC_61$condition)
ddsHTSeqGC_61 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_61,
                                       directory = directoryGC_61,
                                       design= ~ condition + run)
ddsHTSeqGC_61


  ####################### line 69
  sampleTableGC_69$run <- as.factor(sampleTableGC_69$run)
sampleTableGC_69$sampleName <- as.factor(sampleTableGC_69$sampleName)
sampleTableGC_69$condition <- as.factor(sampleTableGC_69$condition)
ddsHTSeqGC_69 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_69,
                                       directory = directoryGC_69,
                                       design= ~ condition + run)
ddsHTSeqGC_69

################### line 76
  sampleTableGC_76$run <- as.factor(sampleTableGC_76$run)
sampleTableGC_76$sampleName <- as.factor(sampleTableGC_76$sampleName)
sampleTableGC_76$condition <- as.factor(sampleTableGC_76$condition)
ddsHTSeqGC_76 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_76,
                                       directory = directoryGC_76,
                                       design= ~ condition + run)
ddsHTSeqGC_76


################## line 77
sampleTableGC_77$run <- as.factor(sampleTableGC_77$run)
sampleTableGC_77$sampleName <- as.factor(sampleTableGC_77$sampleName)
sampleTableGC_77$condition <- as.factor(sampleTableGC_77$condition)
ddsHTSeqGC_77 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_77,
                                       directory = directoryGC_77,
                                       design= ~ condition + run)
ddsHTSeqGC_77

# write.csv(counts(ddsHTSeqGC_77),file="counts.DEseqGC_S1.csv")
 


############################################################
  #### MA Old Samples ####
sampleTableMAold$sampleName <- as.factor(sampleTableMAold$sampleName)
sampleTableMAold$condition <- as.factor(sampleTableMAold$condition)
ddsHTSeqMAold <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold,
                                       directory = directoryMAold,
                                       design= ~ condition)

ddsHTSeqMAold
#
# write.csv(counts(ddsHTSeqMAold),file="counts.DEseqMAold.csv")



##################################### Lines Individually ########################
  ######## Line 1
sampleTableMAold_1$sampleName <- as.factor(sampleTableMAold_1$sampleName)
sampleTableMAold_1$condition <- as.factor(sampleTableMAold_1$condition)
ddsHTSeqMAold_1 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_1,
                                       directory = directoryMAold_1,
                                       design= ~ condition)

ddsHTSeqMAold_1
  ######## Line 2
sampleTableMAold_2$sampleName <- as.factor(sampleTableMAold_2$sampleName)
sampleTableMAold_2$condition <- as.factor(sampleTableMAold_2$condition)
ddsHTSeqMAold_2 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_2,
                                       directory = directoryMAold_2,
                                       design= ~ condition)

ddsHTSeqMAold_2
  ######## Line 3
sampleTableMAold_3$sampleName <- as.factor(sampleTableMAold_3$sampleName)
sampleTableMAold_3$condition <- as.factor(sampleTableMAold_3$condition)
ddsHTSeqMAold_3 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_3,
                                       directory = directoryMAold_3,
                                       design= ~ condition)

ddsHTSeqMAold_3
  ######## Line 4
sampleTableMAold_4$sampleName <- as.factor(sampleTableMAold_4$sampleName)
sampleTableMAold_4$condition <- as.factor(sampleTableMAold_4$condition)
ddsHTSeqMAold_4 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_4,
                                       directory = directoryMAold_4,
                                       design= ~ condition)

ddsHTSeqMAold_4

  ######## Line 5
sampleTableMAold_5$sampleName <- as.factor(sampleTableMAold_5$sampleName)
sampleTableMAold_5$condition <- as.factor(sampleTableMAold_5$condition)
ddsHTSeqMAold_5 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_5,
                                       directory = directoryMAold_5,
                                       design= ~ condition)

ddsHTSeqMAold_5

  ######## Line 6
# sampleTableMAold_6$sampleName <- as.factor(sampleTableMAold_6$sampleName)
# sampleTableMAold_6$condition <- as.factor(sampleTableMAold_6$condition)
# ddsHTSeqMAold_6 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_6,
#                                        directory = directoryMAold_6,
#                                        design= ~ condition)
#
# ddsHTSeqMAold_6

  ######## Line 7
sampleTableMAold_7$sampleName <- as.factor(sampleTableMAold_7$sampleName)
sampleTableMAold_7$condition <- as.factor(sampleTableMAold_7$condition)
ddsHTSeqMAold_7 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_7,
                                       directory = directoryMAold_7,
                                       design= ~ condition)

ddsHTSeqMAold_7

  ######## Line 8
sampleTableMAold_8$sampleName <- as.factor(sampleTableMAold_8$sampleName)
sampleTableMAold_8$condition <- as.factor(sampleTableMAold_8$condition)
ddsHTSeqMAold_8 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_8,
                                       directory = directoryMAold_8,
                                       design= ~ condition)

ddsHTSeqMAold_8

  ######## Line 9
sampleTableMAold_9$sampleName <- as.factor(sampleTableMAold_9$sampleName)
sampleTableMAold_9$condition <- as.factor(sampleTableMAold_9$condition)
ddsHTSeqMAold_9 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_9,
                                       directory = directoryMAold_9,
                                       design= ~ condition)

ddsHTSeqMAold_9

  ######## Line 11
sampleTableMAold_11$sampleName <- as.factor(sampleTableMAold_11$sampleName)
sampleTableMAold_11$condition <- as.factor(sampleTableMAold_11$condition)
ddsHTSeqMAold_11 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_11,
                                       directory = directoryMAold_11,
                                       design= ~ condition)

ddsHTSeqMAold_11

  ######## Line 15
sampleTableMAold_15$sampleName <- as.factor(sampleTableMAold_15$sampleName)
sampleTableMAold_15$condition <- as.factor(sampleTableMAold_15$condition)
ddsHTSeqMAold_15 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_15,
                                       directory = directoryMAold_15,
                                       design= ~ condition)

ddsHTSeqMAold_15

  ######## Line 28
sampleTableMAold_28$sampleName <- as.factor(sampleTableMAold_28$sampleName)
sampleTableMAold_28$condition <- as.factor(sampleTableMAold_28$condition)
ddsHTSeqMAold_28 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_28,
                                       directory = directoryMAold_28,
                                       design= ~ condition)

ddsHTSeqMAold_28

  ######## Line 88
sampleTableMAold_88$sampleName <- as.factor(sampleTableMAold_88$sampleName)
sampleTableMAold_88$condition <- as.factor(sampleTableMAold_88$condition)
ddsHTSeqMAold_88 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_88,
                                       directory = directoryMAold_88,
                                       design= ~ condition)

ddsHTSeqMAold_88

  ######## Line 108
sampleTableMAold_108$sampleName <- as.factor(sampleTableMAold_108$sampleName)
sampleTableMAold_108$condition <- as.factor(sampleTableMAold_108$condition)
ddsHTSeqMAold_108 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_108,
                                       directory = directoryMAold_108,
                                       design= ~ condition)

ddsHTSeqMAold_108

  ######## Line 119
sampleTableMAold_119$sampleName <- as.factor(sampleTableMAold_119$sampleName)
sampleTableMAold_119$condition <- as.factor(sampleTableMAold_119$condition)
ddsHTSeqMAold_119 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_119,
                                       directory = directoryMAold_119,
                                       design= ~ condition)

ddsHTSeqMAold_119

############################################################
ddsHTSeqMAnew <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnew,
                                       directory = directoryMAnew,
                                       design= ~ condition)

ddsHTSeqMAnew

# #### PCA
sampleTableMAnewS152$sampleName <- as.factor(sampleTableMAnewS152$sampleName)
sampleTableMAnewS152$condition <- as.factor(sampleTableMAnewS152$condition)
ddsHTSeqMAnewS152 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS152,
                                       directory = directoryMAnewS152,
                                       design= ~ condition + run)
#### subset the DESeq dataset into individual samples
sampleTableMAnewS29$sampleName <- as.factor(sampleTableMAnewS29$sampleName)
sampleTableMAnewS29$condition <- as.factor(sampleTableMAnewS29$condition)
ddsHTSeqMAnewS29 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS29,
                                       directory = directoryMAnewS29,
                                       design= ~ condition + run)

sampleTableMAnewS50$sampleName <- as.factor(sampleTableMAnewS50$sampleName)
sampleTableMAnewS50$condition <- as.factor(sampleTableMAnewS50$condition)
ddsHTSeqMAnewS50 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS50,
                                       directory = directoryMAnewS50,
                                       design= ~ condition + run)

sampleTableMAnewS112$sampleName <- as.factor(sampleTableMAnewS112$sampleName)
sampleTableMAnewS112$condition <- as.factor(sampleTableMAnewS112$condition)
ddsHTSeqMAnewS112 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS112,
                                       directory = directoryMAnewS112,
                                       design= ~ condition + run)

sampleTableMAnewS115$sampleName <- as.factor(sampleTableMAnewS115$sampleName)
sampleTableMAnewS115$condition <- as.factor(sampleTableMAnewS115$condition)
ddsHTSeqMAnewS115 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS115,
                                       directory = directoryMAnewS115,
                                       design= ~ condition + run)

sampleTableMAnewS117$sampleName <- as.factor(sampleTableMAnewS117$sampleName)
sampleTableMAnewS117$condition <- as.factor(sampleTableMAnewS117$condition)
ddsHTSeqMAnewS117 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS117,
                                       directory = directoryMAnewS117,
                                       design= ~ condition + run)

sampleTableMAnewS123$sampleName <- as.factor(sampleTableMAnewS123$sampleName)
sampleTableMAnewS123$condition <- as.factor(sampleTableMAnewS123$condition)
ddsHTSeqMAnewS123 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS123,
                                       directory = directoryMAnewS123,
                                       design= ~ condition)

sampleTableMAnewS152$sampleName <- as.factor(sampleTableMAnewS152$sampleName)
sampleTableMAnewS152$condition <- as.factor(sampleTableMAnewS152$condition)
ddsHTSeqMAnewS152 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS152,
                                       directory = directoryMAnewS152,
                                       design= ~ condition)

```

Relevel with the ancestor as the reference level
```{r relevel,eval=TRUE,include=FALSE,echo=FALSE}
ddsHTSeqGCcondition <- relevel(ddsHTSeqGC_All$condition, ref = "Holly_Anc")
# ## individual samples
ddsHTSeqGC_1$condition <- relevel(ddsHTSeqGC_1$condition, ref = "Holly_Anc")
ddsHTSeqGC_2$condition <- relevel(ddsHTSeqGC_2$condition, ref = "Holly_Anc")
ddsHTSeqGC_3$condition <- relevel(ddsHTSeqGC_3$condition, ref = "Holly_Anc")
ddsHTSeqGC_4$condition <- relevel(ddsHTSeqGC_4$condition, ref = "Holly_Anc")
ddsHTSeqGC_5$condition <- relevel(ddsHTSeqGC_5$condition, ref = "Holly_Anc")
# ddsHTSeqGC_6$condition <- relevel(ddsHTSeqGC_6$condition, ref = "Holly_Anc")
ddsHTSeqGC_7$condition <- relevel(ddsHTSeqGC_7$condition, ref = "Holly_Anc")
ddsHTSeqGC_8$condition <- relevel(ddsHTSeqGC_8$condition, ref = "Holly_Anc")
ddsHTSeqGC_9$condition <- relevel(ddsHTSeqGC_9$condition, ref = "Holly_Anc")
ddsHTSeqGC_11$condition <- relevel(ddsHTSeqGC_11$condition, ref = "Holly_Anc")
ddsHTSeqGC_18$condition <- relevel(ddsHTSeqGC_18$condition, ref = "Holly_Anc")
ddsHTSeqGC_31$condition <- relevel(ddsHTSeqGC_31$condition, ref = "Holly_Anc")
ddsHTSeqGC_49$condition <- relevel(ddsHTSeqGC_49$condition, ref = "Holly_Anc")
ddsHTSeqGC_59$condition <- relevel(ddsHTSeqGC_59$condition, ref = "Holly_Anc")
ddsHTSeqGC_61$condition <- relevel(ddsHTSeqGC_61$condition, ref = "Holly_Anc")
ddsHTSeqGC_69$condition <- relevel(ddsHTSeqGC_69$condition, ref = "Holly_Anc")
ddsHTSeqGC_76$condition <- relevel(ddsHTSeqGC_76$condition, ref = "Holly_Anc")
ddsHTSeqGC_77$condition <- relevel(ddsHTSeqGC_77$condition, ref = "Holly_Anc")

###############################################################################
ddsHTSeqMAold$condition <- relevel(ddsHTSeqMAold$condition, ref = "SCA")
# ddsHTSeqMAold.short$condition <- relevel(ddsHTSeqMAold.short$condition, ref = "SCA")
ddsHTSeqMAold_1$condition <- relevel(ddsHTSeqMAold_1$condition, ref = "SCA")
ddsHTSeqMAold_2$condition <- relevel(ddsHTSeqMAold_2$condition, ref = "SCA")
ddsHTSeqMAold_3$condition <- relevel(ddsHTSeqMAold_3$condition, ref = "SCA")
ddsHTSeqMAold_4$condition <- relevel(ddsHTSeqMAold_4$condition, ref = "SCA")
ddsHTSeqMAold_5$condition <- relevel(ddsHTSeqMAold_5$condition, ref = "SCA")
# ddsHTSeqMAold_6$condition <- relevel(ddsHTSeqMAold_6$condition, ref = "SCA")
ddsHTSeqMAold_7$condition <- relevel(ddsHTSeqMAold_7$condition, ref = "SCA")
ddsHTSeqMAold_8$condition <- relevel(ddsHTSeqMAold_8$condition, ref = "SCA")
ddsHTSeqMAold_9$condition <- relevel(ddsHTSeqMAold_9$condition, ref = "SCA")
ddsHTSeqMAold_11$condition <- relevel(ddsHTSeqMAold_11$condition, ref = "SCA")
ddsHTSeqMAold_15$condition <- relevel(ddsHTSeqMAold_15$condition, ref = "SCA")
ddsHTSeqMAold_28$condition <- relevel(ddsHTSeqMAold_28$condition, ref = "SCA")
ddsHTSeqMAold_88$condition <- relevel(ddsHTSeqMAold_88$condition, ref = "SCA")
ddsHTSeqMAold_108$condition <- relevel(ddsHTSeqMAold_108$condition, ref = "SCA")
ddsHTSeqMAold_119$condition <- relevel(ddsHTSeqMAold_119$condition, ref = "SCA")
###############################################################################
ddsHTSeqMAnew$condition <- relevel(ddsHTSeqMAnew$condition, ref = "Anc")
# ddsHTSeqMAnew.short$condition <- relevel(ddsHTSeqMAnew.short$condition, ref = "Anc")
ddsHTSeqMAnewS29$condition <- relevel(ddsHTSeqMAnewS29$condition, ref = "Anc")
ddsHTSeqMAnewS50$condition <- relevel(ddsHTSeqMAnewS50$condition, ref = "Anc")
ddsHTSeqMAnewS112$condition <- relevel(ddsHTSeqMAnewS112$condition, ref = "Anc")
ddsHTSeqMAnewS115$condition <- relevel(ddsHTSeqMAnewS115$condition, ref = "Anc")
ddsHTSeqMAnewS117$condition <- relevel(ddsHTSeqMAnewS117$condition, ref = "Anc")
ddsHTSeqMAnewS123$condition <- relevel(ddsHTSeqMAnewS123$condition, ref = "Anc")
# ddsHTSeqMAnewS141$condition <- relevel(ddsHTSeqMAnewS141$condition, ref = "Anc")
ddsHTSeqMAnewS152$condition <- relevel(ddsHTSeqMAnewS152$condition, ref = "Anc")

```

Prefilter based on ancestor counts 
```{r prefilter,eval=TRUE,include=FALSE,echo=FALSE}

## separate lines
ddsHTSeqGC_1
counts(ddsHTSeqGC_1)
dim(counts(ddsHTSeqGC_1))
ancGC <-  counts(ddsHTSeqGC_1)[,4:6]
#subset it based on cpm > 1 in all 3 replicates
table(rowSums(ancGC>10)==3)
keepGC <- rowSums(ancGC>10)==3

# need to make sure the samples don't have 0s either
ddsHTSeqGC_1 <- ddsHTSeqGC_1[keepGC,]

ddsHTSeqGC_11 <- ddsHTSeqGC_11[keepGC,]
ddsHTSeqGC_11
counts(ddsHTSeqGC_11)
dim(counts(ddsHTSeqGC_11))

ddsHTSeqGC_2 <- ddsHTSeqGC_2[keepGC,]

ddsHTSeqGC_3 <- ddsHTSeqGC_3[keepGC,]

ddsHTSeqGC_4 <- ddsHTSeqGC_4[keepGC,]

ddsHTSeqGC_5 <- ddsHTSeqGC_5[keepGC,]

# ddsHTSeqGC_6 <- ddsHTSeqGC_6[keepGC,]
# dim(ddsHTSeqGC_6)
# ddsHTSeqGC_6$condition
ddsHTSeqGC_7 <- ddsHTSeqGC_7[keepGC,]

ddsHTSeqGC_8 <- ddsHTSeqGC_8[keepGC,]

ddsHTSeqGC_9 <- ddsHTSeqGC_9[keepGC,]

ddsHTSeqGC_18 <- ddsHTSeqGC_18[keepGC,]


ddsHTSeqGC_49 <- ddsHTSeqGC_49[keepGC,]

ddsHTSeqGC_59 <- ddsHTSeqGC_59[keepGC,]

ddsHTSeqGC_61 <- ddsHTSeqGC_61[keepGC,]

# ddsHTSeqGC_69 <- ddsHTSeqGC_69[keepGC,]
# dim(ddsHTSeqGC_69)
# ddsHTSeqGC_69$condition
ddsHTSeqGC_76 <- ddsHTSeqGC_76[keepGC,]

ddsHTSeqGC_77 <- ddsHTSeqGC_77[keepGC,]

###############################################################################
keepMAold <- rowSums(counts(ddsHTSeqMAold)) >= 20
ddsHTSeqMAold <- ddsHTSeqMAold[keepMAold,]
ddsHTSeqMAold
counts(ddsHTSeqMAold)
dim(counts(ddsHTSeqMAold))

#make a separate data frame with just the ancestor replicates
ancMAold <-  counts(ddsHTSeqMAold)[,43:45]
#subset it based on cpm > 1 in all 3 replicates
table(rowSums(ancMAold>10)==3)
keepMAold <- rowSums(ancMAold>10)==3
# try it in the full count set
# make a test file first
ddsHTSeqMAoldtest <- ddsHTSeqMAold
ddsHTSeqMAoldtest <- ddsHTSeqMAoldtest[keepMAold,]
ddsHTSeqMAoldtest
counts(ddsHTSeqMAoldtest)
dim(counts(ddsHTSeqMAoldtest))
View(counts(ddsHTSeqMAoldtest))
# # test file was fine so go ahead and make it the real thing
ddsHTSeqMAold <- ddsHTSeqMAoldtest
## get rid of any NAs
ddsHTSeqMAold <- na.omit(ddsHTSeqMAold)
dim(ddsHTSeqMAold)

ddsHTSeqMAold_1 <- ddsHTSeqMAold_1[keepGC,]

ddsHTSeqMAold_2 <- ddsHTSeqMAold_2[keepGC,]

ddsHTSeqMAold_2$condition

ddsHTSeqMAold_3 <- ddsHTSeqMAold_3[keepGC,]

ddsHTSeqMAold_4 <- ddsHTSeqMAold_4[keepGC,]

ddsHTSeqMAold_5 <- ddsHTSeqMAold_5[keepGC,]

ddsHTSeqMAold_7 <- ddsHTSeqMAold_7[keepGC,]

ddsHTSeqMAold_8 <- ddsHTSeqMAold_8[keepGC,]

ddsHTSeqMAold_9 <- ddsHTSeqMAold_9[keepGC,]

ddsHTSeqMAold_11 <- ddsHTSeqMAold_11[keepGC,]

ddsHTSeqMAold_15 <- ddsHTSeqMAold_15[keepGC,]

ddsHTSeqMAold_28 <- ddsHTSeqMAold_28[keepGC,]

ddsHTSeqMAold_88 <- ddsHTSeqMAold_88[keepGC,]

ddsHTSeqMAold_108 <- ddsHTSeqMAold_108[keepGC,]

ddsHTSeqMAold_119 <- ddsHTSeqMAold_119[keepGC,]

###############################################################################
keepMA <- rownames(ddsHTSeqMAnewS29)
keepGC1 <- rownames(ddsHTSeqGC_1)
keep1 <- keepMA %in% keepGC1
summary(keep1)
length(keep1)
keep <- keep1

ddsHTSeqMAnewS29 <- ddsHTSeqMAnewS29[keep,]
length(ddsHTSeqMAnewS29)

ddsHTSeqMAnewS50 <- ddsHTSeqMAnewS50[keep,]
length(ddsHTSeqMAnewS50)

ddsHTSeqMAnewS112 <- ddsHTSeqMAnewS112[keep,]

ddsHTSeqMAnewS115 <- ddsHTSeqMAnewS115[keep,]

ddsHTSeqMAnewS117 <- ddsHTSeqMAnewS117[keep,]

ddsHTSeqMAnewS123 <- ddsHTSeqMAnewS123[keep,]

ddsHTSeqMAnewS152 <- ddsHTSeqMAnewS152[keep,]

#now need to make sure all the GC and old MA samples match the new MA samples since got rid of a couple
keepMA.1 <- rownames(ddsHTSeqMAnewS152)
keepGC1.1 <- rownames(ddsHTSeqGC_77)
keep2 <- keepGC1.1 %in% keepMA.1
summary(keep2)
length(keep2)

ddsHTSeqGC_1 <- ddsHTSeqGC_1[keep2,]
ddsHTSeqGC_2 <- ddsHTSeqGC_2[keep2,]
ddsHTSeqGC_3 <- ddsHTSeqGC_3[keep2,]
ddsHTSeqGC_4 <- ddsHTSeqGC_4[keep2,]
ddsHTSeqGC_5 <- ddsHTSeqGC_5[keep2,]
ddsHTSeqGC_7 <- ddsHTSeqGC_7[keep2,]
ddsHTSeqGC_8 <- ddsHTSeqGC_8[keep2,]
ddsHTSeqGC_9 <- ddsHTSeqGC_9[keep2,]
ddsHTSeqGC_11 <- ddsHTSeqGC_11[keep2,]
ddsHTSeqGC_18 <- ddsHTSeqGC_18[keep2,]
ddsHTSeqGC_49 <- ddsHTSeqGC_49[keep2,]
ddsHTSeqGC_59 <- ddsHTSeqGC_59[keep2,]
ddsHTSeqGC_61 <- ddsHTSeqGC_61[keep2,]
ddsHTSeqGC_76 <- ddsHTSeqGC_76[keep2,]
ddsHTSeqGC_77 <- ddsHTSeqGC_77[keep2,]

##################################################################################
### Left off here, trying to get MA old genes to match GC and MA new genes..
keepMA.1 <- rownames(ddsHTSeqMAnewS152)
keepGC1.2 <- rownames(ddsHTSeqMAold_2)
keep3 <- keepGC1.2 %in% keepMA.1
summary(keep3)
length(keep3)

ddsHTSeqMAold_1 <- ddsHTSeqMAold_1[keep3,]
ddsHTSeqMAold_2 <- ddsHTSeqMAold_2[keep3,]
ddsHTSeqMAold_3 <- ddsHTSeqMAold_3[keep3,]
ddsHTSeqMAold_4 <- ddsHTSeqMAold_4[keep3,]
ddsHTSeqMAold_5 <- ddsHTSeqMAold_5[keep3,]
ddsHTSeqMAold_7 <- ddsHTSeqMAold_7[keep3,]
ddsHTSeqMAold_8 <- ddsHTSeqMAold_8[keep3,]
ddsHTSeqMAold_9 <- ddsHTSeqMAold_9[keep3,]
ddsHTSeqMAold_11 <- ddsHTSeqMAold_11[keep3,]
ddsHTSeqMAold_15 <- ddsHTSeqMAold_15[keep3,]
ddsHTSeqMAold_28 <- ddsHTSeqMAold_28[keep3,]
ddsHTSeqMAold_88 <- ddsHTSeqMAold_88[keep3,]
ddsHTSeqMAold_108 <- ddsHTSeqMAold_108[keep3,]
ddsHTSeqMAold_119 <- ddsHTSeqMAold_119[keep3,]
```

Create datasets for each sample
```{r full dataset, eval=TRUE,include=FALSE,echo=FALSE}
######### individual samples
## add annotations
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
## AS function
ddsDataSet <- function(sample) {
  geneID <- rownames(sample)
  genes <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes)
  genes <- as.data.frame(genes)
  remove <- "chrM"
  all(rownames(sample) == genes$GENEID)
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched <- intersect(rownames(sample), genes$GENEID)
  keep <- rownames(sample) %in% matched
  sample2 <- sample[keep,]
  length(sample2)
  all(rownames(sample2) == genes$GENEID)
  mcols(sample2) <- cbind(mcols(sample2), genes)
  return(sample2)
}

GC_Samples <- c("1","2","3","4","5","6","7","8","9","11","18","31","49","59","61","76","77")

ddsHTSeqGC_11 <- ddsDataSet(ddsHTSeqGC_11)
ddsHTSeqGC_1 <- ddsDataSet(ddsHTSeqGC_1)
ddsHTSeqGC_2 <- ddsDataSet(ddsHTSeqGC_2)
ddsHTSeqGC_3 <- ddsDataSet(ddsHTSeqGC_3)
# ddsHTSeqGC_5 <- ddsDataSet(ddsHTSeqGC_5)
ddsHTSeqGC_7 <- ddsDataSet(ddsHTSeqGC_7)


geneIDgC_5 <- rownames(ddsHTSeqGC_5)
genesGC_5 <- ensembldb::select(txdb, keys=geneIDgC_5, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_5 <- na.omit(genesGC_5)
genesGC_5 <- as.data.frame(genesGC_5)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove4,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_5) == genesGC_5$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS5 <- intersect(rownames(ddsHTSeqGC_5), genesGC_5$GENEID)
keep3S5 <- rownames(ddsHTSeqGC_5) %in% matchedS5
ddsHTSeqGC_5 <- ddsHTSeqGC_5[keep3S5,]
length(ddsHTSeqGC_5)
all(rownames(ddsHTSeqGC_5) == genesGC_5$GENEID)
mcols(ddsHTSeqGC_5) <- cbind(mcols(ddsHTSeqGC_5), genesGC_5)

geneIDgC_7 <- rownames(ddsHTSeqGC_7)
genesGC_7 <- ensembldb::select(txdb, keys=geneIDgC_7, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_7 <- na.omit(genesGC_7)
genesGC_7 <- as.data.frame(genesGC_7)
remove2 <- "chrM"
all(rownames(ddsHTSeqGC_7) == genesGC_7$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS7 <- intersect(rownames(ddsHTSeqGC_7), genesGC_7$GENEID)
keep3S7 <- rownames(ddsHTSeqGC_7) %in% matchedS7
ddsHTSeqGC_7 <- ddsHTSeqGC_7[keep3S7,]
length(ddsHTSeqGC_7)
all(rownames(ddsHTSeqGC_7) == genesGC_7$GENEID)
mcols(ddsHTSeqGC_7) <- cbind(mcols(ddsHTSeqGC_7), genesGC_7)

geneIDgC_8 <- rownames(ddsHTSeqGC_8)
genesGC_8 <- ensembldb::select(txdb, keys=geneIDgC_8, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_8 <- na.omit(genesGC_8)
genesGC_8 <- as.data.frame(genesGC_8)
remove2 <- "chrM"
all(rownames(ddsHTSeqGC_8) == genesGC_8$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS8 <- intersect(rownames(ddsHTSeqGC_8), genesGC_8$GENEID)
keep3S8 <- rownames(ddsHTSeqGC_8) %in% matchedS8
ddsHTSeqGC_8 <- ddsHTSeqGC_8[keep3S8,]
length(ddsHTSeqGC_8)
all(rownames(ddsHTSeqGC_8) == genesGC_8$GENEID)
mcols(ddsHTSeqGC_8) <- cbind(mcols(ddsHTSeqGC_8), genesGC_8)

geneIDgC_9 <- rownames(ddsHTSeqGC_9)
genesGC_9 <- ensembldb::select(txdb, keys=geneIDgC_9, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_9 <- na.omit(genesGC_9)
genesGC_9 <- as.data.frame(genesGC_9)
remove2 <- "chrM"
all(rownames(ddsHTSeqGC_9) == genesGC_9$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS9 <- intersect(rownames(ddsHTSeqGC_9), genesGC_9$GENEID)
keep3S9 <- rownames(ddsHTSeqGC_9) %in% matchedS9
ddsHTSeqGC_9 <- ddsHTSeqGC_9[keep3S9,]
length(ddsHTSeqGC_9)
all(rownames(ddsHTSeqGC_9) == genesGC_9$GENEID)
mcols(ddsHTSeqGC_9) <- cbind(mcols(ddsHTSeqGC_9), genesGC_9)

geneIDgC_11 <- rownames(ddsHTSeqGC_11)
genesGC_11 <- ensembldb::select(txdb, keys=geneIDgC_11, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_11 <- na.omit(genesGC_11)
genesGC_11 <- as.data.frame(genesGC_11)
remove2 <- "chrM"
all(rownames(ddsHTSeqGC_11) == genesGC_11$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS11 <- intersect(rownames(ddsHTSeqGC_11), genesGC_11$GENEID)
keep3S11 <- rownames(ddsHTSeqGC_11) %in% matchedS11
ddsHTSeqGC_11 <- ddsHTSeqGC_11[keep3S11,]
length(ddsHTSeqGC_11)
all(rownames(ddsHTSeqGC_11) == genesGC_11$GENEID)
mcols(ddsHTSeqGC_11) <- cbind(mcols(ddsHTSeqGC_11), genesGC_11)

geneIDgC_18 <- rownames(ddsHTSeqGC_18)
genesGC_18 <- ensembldb::select(txdb, keys=geneIDgC_18, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_18 <- na.omit(genesGC_18)
genesGC_18 <- as.data.frame(genesGC_18)
remove2 <- "chrM"
all(rownames(ddsHTSeqGC_18) == genesGC_18$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS18 <- intersect(rownames(ddsHTSeqGC_18), genesGC_18$GENEID)
keep3S18 <- rownames(ddsHTSeqGC_18) %in% matchedS18
ddsHTSeqGC_18 <- ddsHTSeqGC_18[keep3S18,]
length(ddsHTSeqGC_18)
all(rownames(ddsHTSeqGC_18) == genesGC_18$GENEID)
mcols(ddsHTSeqGC_18) <- cbind(mcols(ddsHTSeqGC_18), genesGC_18)

geneIDgC_31 <- rownames(ddsHTSeqGC_31)
genesGC_31 <- ensembldb::select(txdb, keys=geneIDgC_31, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_31 <- na.omit(genesGC_31)
genesGC_31 <- as.data.frame(genesGC_31)
remove2 <- "chrM"
all(rownames(ddsHTSeqGC_31) == genesGC_31$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS31 <- intersect(rownames(ddsHTSeqGC_31), genesGC_31$GENEID)
keep3S31 <- rownames(ddsHTSeqGC_31) %in% matchedS31
ddsHTSeqGC_31 <- ddsHTSeqGC_31[keep3S31,]
length(ddsHTSeqGC_31)
all(rownames(ddsHTSeqGC_31) == genesGC_31$GENEID)
mcols(ddsHTSeqGC_31) <- cbind(mcols(ddsHTSeqGC_31), genesGC_31)


# mcols(ddsHTSeqGC_31) <- cbind(mcols(ddsHTSeqGC_31), genesGC_2)
geneIDgC_49 <- rownames(ddsHTSeqGC_49)
genesGC_49 <- ensembldb::select(txdb, keys=geneIDgC_49, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_49 <- na.omit(genesGC_49)
genesGC_49 <- as.data.frame(genesGC_49)
remove2 <- "chrM"
all(rownames(ddsHTSeqGC_49) == genesGC_49$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS49 <- intersect(rownames(ddsHTSeqGC_49), genesGC_49$GENEID)
keep3S49 <- rownames(ddsHTSeqGC_49) %in% matchedS49
ddsHTSeqGC_49 <- ddsHTSeqGC_49[keep3S49,]
length(ddsHTSeqGC_49)
all(rownames(ddsHTSeqGC_49) == genesGC_49$GENEID)
mcols(ddsHTSeqGC_49) <- cbind(mcols(ddsHTSeqGC_49), genesGC_49)

geneIDgC_59 <- rownames(ddsHTSeqGC_59)
genesGC_59 <- ensembldb::select(txdb, keys=geneIDgC_59, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_59 <- na.omit(genesGC_59)
genesGC_59 <- as.data.frame(genesGC_59)
remove2 <- "chrM"
all(rownames(ddsHTSeqGC_59) == genesGC_59$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS59 <- intersect(rownames(ddsHTSeqGC_59), genesGC_59$GENEID)
keep3S59 <- rownames(ddsHTSeqGC_59) %in% matchedS59
ddsHTSeqGC_59 <- ddsHTSeqGC_59[keep3S59,]
length(ddsHTSeqGC_59)
all(rownames(ddsHTSeqGC_59) == genesGC_59$GENEID)
mcols(ddsHTSeqGC_59) <- cbind(mcols(ddsHTSeqGC_59), genesGC_59)

geneIDgC_61 <- rownames(ddsHTSeqGC_61)
genesGC_61 <- ensembldb::select(txdb, keys=geneIDgC_61, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_61 <- na.omit(genesGC_61)
genesGC_61 <- as.data.frame(genesGC_61)
remove2 <- "chrM"
all(rownames(ddsHTSeqGC_61) == genesGC_61$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS61 <- intersect(rownames(ddsHTSeqGC_61), genesGC_61$GENEID)
keep3S61 <- rownames(ddsHTSeqGC_61) %in% matchedS61
ddsHTSeqGC_61 <- ddsHTSeqGC_61[keep3S61,]
length(ddsHTSeqGC_61)
all(rownames(ddsHTSeqGC_61) == genesGC_61$GENEID)
mcols(ddsHTSeqGC_61) <- cbind(mcols(ddsHTSeqGC_61), genesGC_61)

geneIDgC_76 <- rownames(ddsHTSeqGC_76)
genesGC_76 <- ensembldb::select(txdb, keys=geneIDgC_76, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_76 <- na.omit(genesGC_76)
genesGC_76 <- as.data.frame(genesGC_76)
remove2 <- "chrM"
all(rownames(ddsHTSeqGC_76) == genesGC_76$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS76 <- intersect(rownames(ddsHTSeqGC_76), genesGC_76$GENEID)
keep3S76 <- rownames(ddsHTSeqGC_76) %in% matchedS76
ddsHTSeqGC_76 <- ddsHTSeqGC_76[keep3S76,]
length(ddsHTSeqGC_76)
all(rownames(ddsHTSeqGC_76) == genesGC_76$GENEID)
mcols(ddsHTSeqGC_76) <- cbind(mcols(ddsHTSeqGC_76), genesGC_76)

geneIDgC_77 <- rownames(ddsHTSeqGC_77)
genesGC_77 <- ensembldb::select(txdb, keys=geneIDgC_77, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_77 <- na.omit(genesGC_77)
genesGC_77 <- as.data.frame(genesGC_77)
remove2 <- "chrM"
all(rownames(ddsHTSeqGC_77) == genesGC_77$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS77 <- intersect(rownames(ddsHTSeqGC_77), genesGC_77$GENEID)
keep3S77 <- rownames(ddsHTSeqGC_77) %in% matchedS77
ddsHTSeqGC_77 <- ddsHTSeqGC_77[keep3S77,]
length(ddsHTSeqGC_77)
all(rownames(ddsHTSeqGC_77) == genesGC_77$GENEID)
mcols(ddsHTSeqGC_77) <- cbind(mcols(ddsHTSeqGC_77), genesGC_77)

###### individual samples
geneIDMAold_1 <- rownames(ddsHTSeqMAold_1)
genesMAold_1 <- ensembldb::select(txdb, keys=geneIDMAold_1, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_1 <- na.omit(genesMAold_1)
genesMAold_1 <- as.data.frame(genesMAold_1)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAold_1) == genesMAold_1$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_1), genesMAold_1$GENEID)
keep4 <- rownames(ddsHTSeqMAold_1) %in% matched_2
ddsHTSeqMAold_1 <- ddsHTSeqMAold_1[keep4,]
all(rownames(ddsHTSeqMAold_1) == genesMAold_1$GENEID)
mcols(ddsHTSeqMAold_1) <- cbind(mcols(ddsHTSeqMAold_1), genesMAold_1)

geneIDMAold_2 <- rownames(ddsHTSeqMAold_2)
genesMAold_2 <- ensembldb::select(txdb, keys=geneIDMAold_2, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_2 <- na.omit(genesMAold_2)
genesMAold_2 <- as.data.frame(genesMAold_2)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAold_2) == genesMAold_2$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_2), genesMAold_2$GENEID)
keep4 <- rownames(ddsHTSeqMAold_2) %in% matched_2
ddsHTSeqMAold_2 <- ddsHTSeqMAold_2[keep4,]
all(rownames(ddsHTSeqMAold_2) == genesMAold_2$GENEID)
mcols(ddsHTSeqMAold_2) <- cbind(mcols(ddsHTSeqMAold_2), genesMAold_2)

geneIDMAold_3 <- rownames(ddsHTSeqMAold_3)
genesMAold_3 <- ensembldb::select(txdb, keys=geneIDMAold_3, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_3 <- na.omit(genesMAold_3)
genesMAold_3 <- as.data.frame(genesMAold_3)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAold_3) == genesMAold_3$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_3), genesMAold_3$GENEID)
keep4 <- rownames(ddsHTSeqMAold_3) %in% matched_2
ddsHTSeqMAold_3 <- ddsHTSeqMAold_3[keep4,]
all(rownames(ddsHTSeqMAold_3) == genesMAold_3$GENEID)
mcols(ddsHTSeqMAold_3) <- cbind(mcols(ddsHTSeqMAold_3), genesMAold_3)

geneIDMAold_4 <- rownames(ddsHTSeqMAold_4)
genesMAold_4 <- ensembldb::select(txdb, keys=geneIDMAold_4, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_4 <- na.omit(genesMAold_4)
genesMAold_4 <- as.data.frame(genesMAold_4)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAold_4) == genesMAold_4$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_4), genesMAold_4$GENEID)
keep4 <- rownames(ddsHTSeqMAold_4) %in% matched_2
ddsHTSeqMAold_4 <- ddsHTSeqMAold_4[keep4,]
all(rownames(ddsHTSeqMAold_4) == genesMAold_4$GENEID)
mcols(ddsHTSeqMAold_4) <- cbind(mcols(ddsHTSeqMAold_4), genesMAold_4)

geneIDMAold_5 <- rownames(ddsHTSeqMAold_5)
genesMAold_5 <- ensembldb::select(txdb, keys=geneIDMAold_5, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_5 <- na.omit(genesMAold_5)
genesMAold_5 <- as.data.frame(genesMAold_5)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAold_5) == genesMAold_5$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_5), genesMAold_5$GENEID)
keep4 <- rownames(ddsHTSeqMAold_5) %in% matched_2
ddsHTSeqMAold_5 <- ddsHTSeqMAold_5[keep4,]
all(rownames(ddsHTSeqMAold_5) == genesMAold_5$GENEID)
mcols(ddsHTSeqMAold_5) <- cbind(mcols(ddsHTSeqMAold_5), genesMAold_5)

geneIDMAold_7 <- rownames(ddsHTSeqMAold_7)
genesMAold_7 <- ensembldb::select(txdb, keys=geneIDMAold_7, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_7 <- na.omit(genesMAold_7)
genesMAold_7 <- as.data.frame(genesMAold_7)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAold_7) == genesMAold_7$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_7), genesMAold_7$GENEID)
keep4 <- rownames(ddsHTSeqMAold_7) %in% matched_2
ddsHTSeqMAold_7 <- ddsHTSeqMAold_7[keep4,]
all(rownames(ddsHTSeqMAold_7) == genesMAold_7$GENEID)
mcols(ddsHTSeqMAold_7) <- cbind(mcols(ddsHTSeqMAold_7), genesMAold_7)

geneIDMAold_8 <- rownames(ddsHTSeqMAold_8)
genesMAold_8 <- ensembldb::select(txdb, keys=geneIDMAold_8, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_8 <- na.omit(genesMAold_8)
genesMAold_8 <- as.data.frame(genesMAold_8)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAold_8) == genesMAold_8$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_8), genesMAold_8$GENEID)
keep4 <- rownames(ddsHTSeqMAold_8) %in% matched_2
ddsHTSeqMAold_8 <- ddsHTSeqMAold_8[keep4,]
all(rownames(ddsHTSeqMAold_8) == genesMAold_8$GENEID)
mcols(ddsHTSeqMAold_8) <- cbind(mcols(ddsHTSeqMAold_8), genesMAold_8)

geneIDMAold_9 <- rownames(ddsHTSeqMAold_9)
genesMAold_9 <- ensembldb::select(txdb, keys=geneIDMAold_9, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_9 <- na.omit(genesMAold_9)
genesMAold_9 <- as.data.frame(genesMAold_9)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAold_9) == genesMAold_9$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_9), genesMAold_9$GENEID)
keep4 <- rownames(ddsHTSeqMAold_9) %in% matched_2
ddsHTSeqMAold_9 <- ddsHTSeqMAold_9[keep4,]
all(rownames(ddsHTSeqMAold_9) == genesMAold_9$GENEID)
mcols(ddsHTSeqMAold_9) <- cbind(mcols(ddsHTSeqMAold_9), genesMAold_9)

geneIDMAold_11 <- rownames(ddsHTSeqMAold_11)
genesMAold_11 <- ensembldb::select(txdb, keys=geneIDMAold_11, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_11 <- na.omit(genesMAold_11)
genesMAold_11 <- as.data.frame(genesMAold_11)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAold_11) == genesMAold_11$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_11), genesMAold_11$GENEID)
keep4 <- rownames(ddsHTSeqMAold_11) %in% matched_2
ddsHTSeqMAold_11 <- ddsHTSeqMAold_11[keep4,]
all(rownames(ddsHTSeqMAold_11) == genesMAold_11$GENEID)
mcols(ddsHTSeqMAold_11) <- cbind(mcols(ddsHTSeqMAold_11), genesMAold_11)

geneIDMAold_15 <- rownames(ddsHTSeqMAold_15)
genesMAold_15 <- ensembldb::select(txdb, keys=geneIDMAold_15, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_15 <- na.omit(genesMAold_15)
genesMAold_15 <- as.data.frame(genesMAold_15)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAold_15) == genesMAold_15$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_15), genesMAold_15$GENEID)
keep4 <- rownames(ddsHTSeqMAold_15) %in% matched_2
ddsHTSeqMAold_15 <- ddsHTSeqMAold_15[keep4,]
all(rownames(ddsHTSeqMAold_15) == genesMAold_15$GENEID)
mcols(ddsHTSeqMAold_15) <- cbind(mcols(ddsHTSeqMAold_15), genesMAold_15)

geneIDMAold_28 <- rownames(ddsHTSeqMAold_28)
genesMAold_28 <- ensembldb::select(txdb, keys=geneIDMAold_28, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_28 <- na.omit(genesMAold_28)
genesMAold_28 <- as.data.frame(genesMAold_28)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAold_28) == genesMAold_28$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_28), genesMAold_28$GENEID)
keep4 <- rownames(ddsHTSeqMAold_28) %in% matched_2
ddsHTSeqMAold_28 <- ddsHTSeqMAold_28[keep4,]
all(rownames(ddsHTSeqMAold_28) == genesMAold_28$GENEID)
mcols(ddsHTSeqMAold_28) <- cbind(mcols(ddsHTSeqMAold_28), genesMAold_28)

geneIDMAold_88 <- rownames(ddsHTSeqMAold_88)
genesMAold_88 <- ensembldb::select(txdb, keys=geneIDMAold_88, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_88 <- na.omit(genesMAold_88)
genesMAold_88 <- as.data.frame(genesMAold_88)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAold_88) == genesMAold_88$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_88), genesMAold_88$GENEID)
keep4 <- rownames(ddsHTSeqMAold_88) %in% matched_2
ddsHTSeqMAold_88 <- ddsHTSeqMAold_88[keep4,]
all(rownames(ddsHTSeqMAold_88) == genesMAold_88$GENEID)
mcols(ddsHTSeqMAold_88) <- cbind(mcols(ddsHTSeqMAold_88), genesMAold_88)

geneIDMAold_108 <- rownames(ddsHTSeqMAold_108)
genesMAold_108 <- ensembldb::select(txdb, keys=geneIDMAold_108, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_108 <- na.omit(genesMAold_108)
genesMAold_108 <- as.data.frame(genesMAold_108)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAold_108) == genesMAold_108$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_108), genesMAold_108$GENEID)
keep4 <- rownames(ddsHTSeqMAold_108) %in% matched_2
ddsHTSeqMAold_108 <- ddsHTSeqMAold_108[keep4,]
all(rownames(ddsHTSeqMAold_108) == genesMAold_108$GENEID)
mcols(ddsHTSeqMAold_108) <- cbind(mcols(ddsHTSeqMAold_108), genesMAold_108)

geneIDMAold_119 <- rownames(ddsHTSeqMAold_119)
genesMAold_119 <- ensembldb::select(txdb, keys=geneIDMAold_119, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_119 <- na.omit(genesMAold_119)
genesMAold_119 <- as.data.frame(genesMAold_119)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAold_119) == genesMAold_119$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_119), genesMAold_119$GENEID)
keep4 <- rownames(ddsHTSeqMAold_119) %in% matched_2
ddsHTSeqMAold_119 <- ddsHTSeqMAold_119[keep4,]
all(rownames(ddsHTSeqMAold_119) == genesMAold_119$GENEID)
mcols(ddsHTSeqMAold_119) <- cbind(mcols(ddsHTSeqMAold_119), genesMAold_119)

#### individual samples
geneIDMAnewS29 <- rownames(ddsHTSeqMAnewS29)
genesMAnewS29 <- ensembldb::select(txdb, keys=geneIDMAnewS29, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS29 <- na.omit(genesMAnewS29)
genesMAnewS29 <- as.data.frame(genesMAnewS29)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAnewS29) == genesMAnewS29$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS29), genesMAnewS29$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS29) %in% matchedS2
ddsHTSeqMAnewS29 <- ddsHTSeqMAnewS29[keep4,]
all(rownames(ddsHTSeqMAnewS29) == genesMAnewS29$GENEID)
mcols(ddsHTSeqMAnewS29) <- cbind(mcols(ddsHTSeqMAnewS29), genesMAnewS29)

geneIDMAnewS50 <- rownames(ddsHTSeqMAnewS50)
genesMAnewS50 <- ensembldb::select(txdb, keys=geneIDMAnewS50, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS50 <- na.omit(genesMAnewS50)
genesMAnewS50 <- as.data.frame(genesMAnewS50)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAnewS50) == genesMAnewS50$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS50), genesMAnewS50$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS50) %in% matchedS2
ddsHTSeqMAnewS50 <- ddsHTSeqMAnewS50[keep4,]
all(rownames(ddsHTSeqMAnewS50) == genesMAnewS50$GENEID)
mcols(ddsHTSeqMAnewS50) <- cbind(mcols(ddsHTSeqMAnewS50), genesMAnewS50)

geneIDMAnewS112 <- rownames(ddsHTSeqMAnewS112)
genesMAnewS112 <- ensembldb::select(txdb, keys=geneIDMAnewS112, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS112 <- na.omit(genesMAnewS112)
genesMAnewS112 <- as.data.frame(genesMAnewS112)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAnewS112) == genesMAnewS112$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS112), genesMAnewS112$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS112) %in% matchedS2
ddsHTSeqMAnewS112 <- ddsHTSeqMAnewS112[keep4,]
all(rownames(ddsHTSeqMAnewS112) == genesMAnewS112$GENEID)
mcols(ddsHTSeqMAnewS112) <- cbind(mcols(ddsHTSeqMAnewS112), genesMAnewS112)
#
geneIDMAnewS115 <- rownames(ddsHTSeqMAnewS115)
genesMAnewS115 <- ensembldb::select(txdb, keys=geneIDMAnewS115, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS115 <- na.omit(genesMAnewS115)
genesMAnewS115 <- as.data.frame(genesMAnewS115)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAnewS115) == genesMAnewS115$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS115), genesMAnewS115$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS115) %in% matchedS2
ddsHTSeqMAnewS115 <- ddsHTSeqMAnewS115[keep4,]
all(rownames(ddsHTSeqMAnewS115) == genesMAnewS115$GENEID)
mcols(ddsHTSeqMAnewS115) <- cbind(mcols(ddsHTSeqMAnewS115), genesMAnewS115)

geneIDMAnewS117 <- rownames(ddsHTSeqMAnewS117)
genesMAnewS117 <- ensembldb::select(txdb, keys=geneIDMAnewS117, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS117 <- na.omit(genesMAnewS117)
genesMAnewS117 <- as.data.frame(genesMAnewS117)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAnewS117) == genesMAnewS117$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS117), genesMAnewS117$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS117) %in% matchedS2
ddsHTSeqMAnewS117 <- ddsHTSeqMAnewS117[keep4,]
all(rownames(ddsHTSeqMAnewS117) == genesMAnewS117$GENEID)
mcols(ddsHTSeqMAnewS117) <- cbind(mcols(ddsHTSeqMAnewS117), genesMAnewS117)

geneIDMAnewS123 <- rownames(ddsHTSeqMAnewS123)
genesMAnewS123 <- ensembldb::select(txdb, keys=geneIDMAnewS123, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS123 <- na.omit(genesMAnewS123)
genesMAnewS123 <- as.data.frame(genesMAnewS123)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAnewS123) == genesMAnewS123$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS123), genesMAnewS123$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS123) %in% matchedS2
ddsHTSeqMAnewS123 <- ddsHTSeqMAnewS123[keep4,]
all(rownames(ddsHTSeqMAnewS123) == genesMAnewS123$GENEID)
mcols(ddsHTSeqMAnewS123) <- cbind(mcols(ddsHTSeqMAnewS123), genesMAnewS123)

geneIDMAnewS152 <- rownames(ddsHTSeqMAnewS152)
genesMAnewS152 <- ensembldb::select(txdb, keys=geneIDMAnewS152, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS152 <- na.omit(genesMAnewS152)
genesMAnewS152 <- as.data.frame(genesMAnewS152)
remove1 <- "chrM"
all(rownames(ddsHTSeqMAnewS152) == genesMAnewS152$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS152), genesMAnewS152$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS152) %in% matchedS2
ddsHTSeqMAnewS152 <- ddsHTSeqMAnewS152[keep4,]
all(rownames(ddsHTSeqMAnewS152) == genesMAnewS152$GENEID)
mcols(ddsHTSeqMAnewS152) <- cbind(mcols(ddsHTSeqMAnewS152), genesMAnewS152)

ddsDefGC_1 <- DESeq(ddsHTSeqGC_1)
ddsDefGC_1
ddsDefGC_2 <- DESeq(ddsHTSeqGC_2)
ddsDefGC_2
ddsDefGC_3 <- DESeq(ddsHTSeqGC_3)
ddsDefGC_3
ddsDefGC_4 <- DESeq(ddsHTSeqGC_4)
ddsDefGC_4
ddsDefGC_5 <- DESeq(ddsHTSeqGC_5)
ddsDefGC_5
ddsDefGC_7 <- DESeq(ddsHTSeqGC_7)
ddsDefGC_7
ddsDefGC_8 <- DESeq(ddsHTSeqGC_8)
ddsDefGC_8
ddsDefGC_9 <- DESeq(ddsHTSeqGC_9)
ddsDefGC_9
ddsDefGC_11 <- DESeq(ddsHTSeqGC_11)
ddsDefGC_11
ddsDefGC_18 <- DESeq(ddsHTSeqGC_18)
ddsDefGC_18
ddsDefGC_49 <- DESeq(ddsHTSeqGC_49)
ddsDefGC_49
ddsDefGC_59 <- DESeq(ddsHTSeqGC_59)
ddsDefGC_59
ddsDefGC_61 <- DESeq(ddsHTSeqGC_61)
ddsDefGC_61
ddsDefGC_76 <- DESeq(ddsHTSeqGC_76)
ddsDefGC_76
ddsDefGC_77 <- DESeq(ddsHTSeqGC_77)
ddsDefGC_77
#################################################
#default DESeq run
### individual lines
ddsDefMAold_1 <- DESeq(ddsHTSeqMAold_1)
ddsDefMAold_1
ddsDefMAold_2 <- DESeq(ddsHTSeqMAold_2)
ddsDefMAold_2
ddsDefMAold_3 <- DESeq(ddsHTSeqMAold_3)
ddsDefMAold_3
ddsDefMAold_4 <- DESeq(ddsHTSeqMAold_4)
ddsDefMAold_4
ddsDefMAold_5 <- DESeq(ddsHTSeqMAold_5)
ddsDefMAold_5
ddsDefMAold_7 <- DESeq(ddsHTSeqMAold_7)
ddsDefMAold_7
ddsDefMAold_8 <- DESeq(ddsHTSeqMAold_8)
ddsDefMAold_8
ddsDefMAold_9 <- DESeq(ddsHTSeqMAold_9)
ddsDefMAold_9
ddsDefMAold_11 <- DESeq(ddsHTSeqMAold_11)
ddsDefMAold_11
ddsDefMAold_15 <- DESeq(ddsHTSeqMAold_15)
ddsDefMAold_15
ddsDefMAold_28 <- DESeq(ddsHTSeqMAold_28)
ddsDefMAold_28
ddsDefMAold_88 <- DESeq(ddsHTSeqMAold_88)
ddsDefMAold_88
ddsDefMAold_108 <- DESeq(ddsHTSeqMAold_108)
ddsDefMAold_108
ddsDefMAold_119 <- DESeq(ddsHTSeqMAold_119)
ddsDefMAold_119
#################################################
##### individual samples
ddsDefMAnewS29 <- DESeq(ddsHTSeqMAnewS29)
ddsDefMAnewS50 <- DESeq(ddsHTSeqMAnewS50)
ddsDefMAnewS112 <- DESeq(ddsHTSeqMAnewS112)
ddsDefMAnewS115 <- DESeq(ddsHTSeqMAnewS115)
ddsDefMAnewS117 <- DESeq(ddsHTSeqMAnewS117)
ddsDefMAnewS123 <- DESeq(ddsHTSeqMAnewS123)
ddsDefMAnewS152 <- DESeq(ddsHTSeqMAnewS152)

#######################################################
# with individual samples
resultsNames(ddsDefGC_1)
resultsS1GC <- results(ddsDefGC_1,contrast=c("condition","Holly_1","Holly_Anc"),alpha=0.05)
summary(resultsS1GC)
resultsS2GC <- results(ddsDefGC_2,contrast=c("condition","Holly_2","Holly_Anc"))
summary(resultsS2GC)
resultsS2GC <- results(ddsDefGC_2,contrast=c("condition","Holly_2","Holly_Anc"),alpha=0.05)
summary(resultsS2GC)
resultsS3GC <- results(ddsDefGC_3,contrast=c("condition","Holly_3","Holly_Anc"),alpha=0.05)
summary(resultsS3GC)
resultsS4GC <- results(ddsDefGC_4,contrast=c("condition","Holly_4","Holly_Anc"),alpha=0.05)
summary(resultsS4GC)
resultsS49GC <- results(ddsDefGC_49,contrast=c("condition","Holly_49","Holly_Anc"),alpha=0.05)
summary(resultsS49GC)
resultsS5GC <- results(ddsDefGC_5,contrast=c("condition","Holly_5","Holly_Anc"),alpha=0.05)
summary(resultsS5GC)
resultsS7GC <- results(ddsDefGC_7,contrast=c("condition","Holly_7","Holly_Anc"),alpha=0.05)
summary(resultsS7GC)
resultsS8GC <- results(ddsDefGC_8,contrast=c("condition","Holly_8","Holly_Anc"),alpha=0.05)
summary(resultsS8GC)
resultsS9GC <- results(ddsDefGC_9,contrast=c("condition","Holly_9","Holly_Anc"),alpha=0.05)
summary(resultsS9GC)

normCountsGC_9 <- counts(ddsDefGC_9,normalized=TRUE)
write.csv(normCountsGC_9,file="normCountsGC_9.csv")

resultsS11GC <- results(ddsDefGC_11,contrast=c("condition","Holly_11","Holly_Anc"),alpha=0.05)
summary(resultsS11GC)

normCountsGC_11 <- counts(ddsDefGC_11,normalized=TRUE)

resultsS18GC <- results(ddsDefGC_18,contrast=c("condition","Holly_18","Holly_Anc"),alpha=0.05)
summary(resultsS18GC)
resultsS59GC <- results(ddsDefGC_59,contrast=c("condition","Holly_59","Holly_Anc"),alpha=0.05)#,independentFiltering = FALSE)
summary(resultsS59GC)
normCountsGC_59 <- counts(ddsDefGC_59,normalized=TRUE)
write.csv(normCountsGC_59,file="normCountsGC_59.csv")

resultsS61GC <- results(ddsDefGC_61,contrast=c("condition","Holly_61","Holly_Anc"),alpha=0.05)
summary(resultsS61GC)
resultsS76GC <- results(ddsDefGC_76,contrast=c("condition","Holly_76","Holly_Anc"),alpha=0.05)
summary(resultsS76GC)
resultsS77GC <- results(ddsDefGC_77,contrast=c("condition","Holly_77","Holly_Anc"),alpha=0.05)
summary(resultsS77GC)


###### individual samples
resultsNames(ddsDefMAold_1)
resultsS1MAold <- results(ddsDefMAold_1,contrast=c("condition","SC001","SCA"))
class(resultsS1MAold)
summary(resultsS1MAold)

resultsNames(ddsDefMAold_2)
resultsS2MAold <- results(ddsDefMAold_2,contrast=c("condition","SC002","SCA"))
class(resultsS2MAold)
summary(resultsS2MAold)

resultsNames(ddsDefMAold_3)
resultsS3MAold <- results(ddsDefMAold_3,contrast=c("condition","SC003","SCA"))
class(resultsS3MAold)
summary(resultsS3MAold)

resultsNames(ddsDefMAold_4)
resultsS4MAold <- results(ddsDefMAold_4,contrast=c("condition","SC004","SCA"))
class(resultsS4MAold)
summary(resultsS4MAold)

resultsNames(ddsDefMAold_5)
resultsS5MAold <- results(ddsDefMAold_5,contrast=c("condition","SC005","SCA"))
class(resultsS5MAold)
summary(resultsS5MAold)

resultsNames(ddsDefMAold_7)
resultsS7MAold <- results(ddsDefMAold_7,contrast=c("condition","SC007","SCA"))
class(resultsS7MAold)
summary(resultsS7MAold)

resultsNames(ddsDefMAold_8)
resultsS8MAold <- results(ddsDefMAold_8,contrast=c("condition","SC008","SCA"))
class(resultsS8MAold)
summary(resultsS8MAold)

resultsNames(ddsDefMAold_9)
resultsS9MAold <- results(ddsDefMAold_9,contrast=c("condition","SC009","SCA"))
class(resultsS9MAold)
summary(resultsS9MAold)

resultsNames(ddsDefMAold_11)
resultsS11MAold <- results(ddsDefMAold_11,contrast=c("condition","SC011","SCA"))
class(resultsS11MAold)
summary(resultsS11MAold)

resultsNames(ddsDefMAold_15)
resultsS15MAold <- results(ddsDefMAold_15,contrast=c("condition","SC015","SCA"))
class(resultsS15MAold)
summary(resultsS15MAold)

resultsNames(ddsDefMAold_28)
resultsS28MAold <- results(ddsDefMAold_28,contrast=c("condition","SC028","SCA"))
class(resultsS28MAold)
summary(resultsS28MAold)

resultsNames(ddsDefMAold_88)
resultsS88MAold <- results(ddsDefMAold_88,contrast=c("condition","SC088","SCA"))
class(resultsS88MAold)
summary(resultsS88MAold)

resultsNames(ddsDefMAold_108)
resultsS108MAold <- results(ddsDefMAold_108,contrast=c("condition","SC108","SCA"))
class(resultsS108MAold)
summary(resultsS108MAold)

resultsNames(ddsDefMAold_119)
resultsS119MAold <- results(ddsDefMAold_119,contrast=c("condition","SC119","SCA"))
class(resultsS119MAold)
summary(resultsS119MAold)

```

Create ratio distribution plots for each sample 
```{r ratioDistributions,eval=TRUE,include=TRUE,echo=FALSE}

############ Euploid distributions
ratioDistributionsEuploids <- function(dds, genes, sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)
library(ggplot2)

allGenes <- qplot(ratios$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios$sampleOverAnc)), main=paste(sample), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios$sampleOverAnc, na.rm = T)), colour = "red", linetype ="longdash", size = .5 ) +  geom_vline(aes(xintercept = median(ratios$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5 )

# {pdf(paste("euploid",sample,".pdf",sep=""))
  print(allGenes)
# dev.off()
# write.csv(ratios,file=paste("euploidRatios",sample,".csv",sep=""))
return(ratios$sampleOverAnc)
}


ratioDistributionsEuploids(ddsDefGC_1,genesGC_1,"Sample_1_GC")
ratioDistributionsEuploids(ddsDefGC_2,genesGC_2,"Sample_2_GC")
ratioDistributionsEuploids(ddsDefGC_3,genesGC_3,"Sample_3_GC")
ratioDistributionsEuploids(ddsDefGC_5,genesGC_5,"Sample_5_GC")
ratioDistributionsEuploids(ddsDefGC_9,genesGC_9,"Sample_9_GC")

ratioDistGC2 <- ratioDistributionsEuploids(ddsDefGC_2,genesGC_2,"Sample_2_GC")
ratioDistGC3 <- ratioDistributionsEuploids(ddsDefGC_3,genesGC_3,"Sample_3_GC")

mean(ratioDistGC2)
var(ratioDistGC2)
median(ratioDistGC2)

library(car)

 ddsDefGC_11


ks.test(ratioDistGC2,ratioDistGC3,alternative="two.sided")

ratioDistributionsEuploids(ddsDefMAold_1,genesMAold_1,"Sample_1_MA")

ratioDistributionsEuploids(ddsDefMAold_2,genesMAold_2,"Sample_2_MA")
ratioDistributionsEuploids(ddsDefMAold_3,genesMAold_3,"Sample_3_MA")
ratioDistributionsEuploids(ddsDefMAold_4,genesMAold_4,"Sample_4_MA")
ratioDistributionsEuploids(ddsDefMAold_5,genesMAold_5,"Sample_5_MA")
ratioDistributionsEuploids(ddsDefMAold_7,genesMAold_7,"Sample_7_MA")
ratioDistributionsEuploids(ddsDefMAold_8,genesMAold_8,"Sample_8_MA")

ratioDistributionsEuploidsMANew <- function(dds, genes, sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:5)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)

allGenes <- qplot(ratios$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios$sampleOverAnc)), main=paste(sample), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1)+ coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
# {pdf(paste("euploid",sample,".pdf",sep=""))
  print(allGenes)
# dev.off()}
# write.csv(ratios,file=paste("euploid",sample,".csv",sep=""))
}

ratioDistributionsEuploidsMANew(ddsDefMAnewS50,genesMAnewS50,"Sample_50_MA")
ratioDistributionsEuploidsMANew(ddsDefMAnewS115,genesMAnewS115,"Sample_115_MA")
#####################################################################
#####################################################################

dds=ddsDefGC_4
genes=genesGC_4
anChr="chrV"
sample="Sample_4_GC"

ratioDistributionsTrisomy <- function(dds, genes, anChr,sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]

### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Trisomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
# {pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
# dev.off()}

trans <- qplot(ratios.Trans$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)), main=paste("Trisomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept = 0.98,color="blue",linetype="longdash") + coord_cartesian(xlim=c(0,3)) + theme_classic()
# {pdf(paste("transGenes",sample,".pdf",sep=""))
  print(trans)
# dev.off()}
# write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
return(ratios.Trans$sampleOverAnc)

}

### GC lines

#euploid lines
# ratioDistributionsTrisomy(ddsDefGC_1,genesGC_1,"chrI","Sample_1_GC_ChrI")

ratioDistributionsTrisomy(ddsDefGC_7,genesGC_7,"chrI","Sample_7_GC_ChrI")
ratioDistGC7 <- ratioDistributionsTrisomy(ddsDefGC_7,genesGC_7,"chrI","Sample_7_GC_ChrI")
meanGC7 <- mean(ratioDistGC7)
medianGC7 <- median(ratioDistGC7)
ratioDistributionsTrisomy(ddsDefGC_4,genesGC_4,"chrV","Sample_4_GC_ChrV")
ratioDistGC4 <- ratioDistributionsTrisomy(ddsDefGC_4,genesGC_4,"chrV","Sample_4_GC_ChrV")
meanGC4 <- mean(ratioDistGC4)
medianGC4 <- median(ratioDistGC4)
stdevGC4 <- sd(ratioDistGC4)
ratioDistGC1 <- ratioDistributionsTrisomy(ddsDefGC_1,genesGC_1,"chrI","Sample_1_GC_ChrI")
ratioDistGC7 <- ratioDistributionsTrisomy(ddsDefGC_7,genesGC_7,"chrI","Sample_7_GC_ChrI")
ks.test(ratioDistGC1,ratioDistGC7)


ratioDistGC49 <- ratioDistributionsTrisomy(ddsDefGC_49,genesGC_49,"chrV","Sample_49_GC_ChrV")
meanGC49 <- mean(ratioDistGC49)
medianGC49 <- median(ratioDistGC49)
stdevGC49 <- sd(ratioDistGC49)
ratioDistGC59 <- ratioDistributionsTrisomy(ddsDefGC_59,genesGC_59,"chrVII","Sample_59_GC_ChrVII")
meanGC59 <- mean(ratioDistGC59)
medianGC59 <- median(ratioDistGC59)
stdevGC59 <- sd(ratioDistGC59)
ratioDistGC61 <- ratioDistributionsTrisomy(ddsDefGC_61,genesGC_61,"chrVII","Sample_61_GC_ChrVII")
meanGC61 <- mean(ratioDistGC61)
medianGC61 <- median(ratioDistGC61)
stdevGC61 <- sd(ratioDistGC61)
ratioDistGC77 <- ratioDistributionsTrisomy(ddsDefGC_77,genesGC_77,"chrXII","Sample_77_GC_ChrXII")
meanGC77 <- mean(ratioDistGC77)
medianGC77 <- median(ratioDistGC77)
stdevGC77 <- sd(ratioDistGC77)

### MA old lines trisomic
ratioDistMA15 <- ratioDistributionsTrisomy(ddsDefMAold_15,genesMAold_15,"chrIX","Sample_15_MA_ChrIX")
meanMA15 <- mean(ratioDistMA15)
medianMA15 <- median(ratioDistMA15)
stdevMA15 <- sd(ratioDistMA15)
ratioDistMA88 <- ratioDistributionsTrisomy(ddsDefMAold_88,genesMAold_88,"chrIX","Sample_88_MA_ChrIX")
meanMA88 <- mean(ratioDistMA88)
medianMA88 <- median(ratioDistMA88)
stdevMA88 <- sd(ratioDistMA88)
ratioDistMA119 <- ratioDistributionsTrisomy(ddsDefMAold_119,genesMAold_119,"chrIX","Sample_119_MA_ChrIX")
meanMA119 <- mean(ratioDistMA119)
medianMA119 <- median(ratioDistMA119)
stdevMA119 <- sd(ratioDistMA119)
ratioDistMA9 <- ratioDistributionsTrisomy(ddsDefMAold_9,genesMAold_9,"chrXIV","Sample_9_MA_ChrXIV")
meanMA9 <- mean(ratioDistMA9)
medianMA9 <- median(ratioDistMA9)
stdevMA9 <- sd(ratioDistMA9)


#### for those lines with multiple aneuploidies
ratioDistributionsTrisomyMulti <- function(dds, genes, anChr, anChr2, sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]

ratios.keepTrans2 <- ratios$chr!=anChr2
ratios.Trans2 <- ratios.Trans[ratios.keepTrans2,]

### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Trisomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
# {pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
# dev.off()}


trans <- qplot(ratios.Trans2$sampleOverAnc,geom="histogram",breaks=c(seq(0,6,by=0.05),max(ratios.Trans2$sampleOverAnc)), main=paste("Trisomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1)  +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans2$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans2$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
# {pdf(paste("transGenes",sample,".pdf",sep=""))
  print(trans)
# dev.off()}
# write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
return(ratios.Trans2$sampleOverAnc)
}

ratioDistGC18 <- ratioDistributionsTrisomyMulti(ddsDefGC_18,genesGC_18,"chrI","chrXII", "Sample_18_GC_ChrI")
meanGC18 <- mean(ratioDistGC18)
medianGC18 <- median(ratioDistGC18)
stdevGC18 <- sd(ratioDistGC18)
ratioDistributionsTrisomyMulti(ddsDefGC_11,genesGC_11,"chrXV","chrI", "Sample_11_GC_ChrXV")


ratioDistributionsTrisomyMulti(ddsDefMAold_108,genesMAold_108,"chrVIII","chrIX", "Sample_108_MA_ChrVIII")

### the chromosomes reversed
ratioDistributionsTrisomyMulti(ddsDefGC_18,genesGC_18,"chrXII","chrI", "Sample_18_GC_ChrXII")


### line 76 has 2 fully duplicated chromosomes and one partial duplication


ratioDistributionsTrisomyMulti2 <- function(dds, genes, anChr, anChr2, anChr3, sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]

ratios.keepTrans2 <- ratios$chr!=anChr2
ratios.Trans2 <- ratios.Trans[ratios.keepTrans2,]

ratios.keepTrans3 <- ratios$chr!=anChr3
ratios.Trans3 <- ratios.Trans2[ratios.keepTrans3,]


### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Trisomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
# {pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
# dev.off()}


trans <- qplot(ratios.Trans3$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans3$sampleOverAnc)), main=paste("Trisomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="magenta") + geom_vline(xintercept=0.67,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans3$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans3$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
# {pdf(paste("transGenes",sample,".pdf",sep=""))
  print(trans)
# dev.off()}
# write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
}
ratioDistributionsTrisomyMulti2(ddsDefGC_76,genesGC_76,"chrIX","chrXIV","chrX", "Sample_76_GC_ChIX")
ratioDistributionsTrisomyMulti2(ddsDefGC_76,genesGC_76,"chrXIV","chrIX","chrX", "Sample_76_GC_ChrXIV")
ratioDistributionsTrisomyMulti2(ddsDefGC_76,genesGC_76,"chrX","chrXIV","chrIX",  "Sample_76_GC_ChrX")


ratioDistributionsTrisomyMultiMANew <- function(dds, genes, anChr,anChri2,sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:5)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]


### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Trisomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
# {pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
# dev.off()}


trans <- qplot(ratios.Trans$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)), main=paste("Trisomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green")+
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
# {pdf(paste("transGenes",sample,".pdf",sep=""))
  print(trans)
# dev.off()}
# write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
return(ratios.cis$sampleOverAnc)
return(ratios.Trans$sampleOverAnc)
}


ratioDistributionsTrisomyMultiMANew(ddsDefMAnewS152,genesMAnewS152,"chrI","chrVIII", "Sample_152_MA_ChrI")
# other chromosome

ratioDistributionsTrisomyMultiMANew(ddsDefMAnewS152,genesMAnewS152,"chrVIII","chrI", "Sample_152_MA_ChrVIII")

### need a function for monosomy
ratioDistributionsMonosomy <- function(dds, genes, anChr,anChr2, sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]
ratios.keepTrans2 <- ratios$chr!=anChr2
ratios.Trans2 <- ratios.Trans[ratios.keepTrans2,]


### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Monosomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=0.5, col="darkmagenta") + geom_vline(xintercept=2,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
# {pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
# dev.off()}


trans <- qplot(ratios.Trans2$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)), main=paste("Monosomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=0.5, col="darkmagenta") + geom_vline(xintercept=2,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans2$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans2$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
# {pdf(paste("transGenes",sample,".pdf",sep=""))
  print(trans)
# dev.off()}
# write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
return(ratios.Trans2$sampleOverAnc)
}


# for GC line 11

ratioDistributionsMonosomy(ddsDefGC_11,genesGC_11,"chrI","chrXV", "Sample_11_GC_I")


# for MA old line 108
ratioDistributionsMonosomy(ddsDefMAold_108,genesMAold_108,"chrIX","chrVIII" ,"Sample_108_MA_IX")


### need to do this for tetrasomy not trisomy
ratioDistributionsTetrasomy <- function(dds, genes, anChr,sample) {

counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]


### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Tetrasomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=2, col="darkmagenta") + geom_vline(xintercept=0.5,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
# {pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
  mean(cis)
# dev.off()}


trans <- qplot(ratios.Trans$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)), main=paste("Tetrasomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=2, col="darkmagenta") + geom_vline(xintercept=0.5,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
# {pdf(paste("transGenes",sample,".pdf",sep=""))
  print(trans)
# dev.off()}
# write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
return(ratios.Trans$sampleOverAnc)
}
ratioDistGC8 <- ratioDistributionsTetrasomy(ddsDefGC_8,genesGC_8,"chrXVI","Sample_8_GC_XVI")
meanGC8 <- mean(ratioDistGC8)
medianGC8 <- median(ratioDistGC8)
stdevGC8 <- sd(ratioDistGC8)

### will need to do MA New lines separately because only 2 ancestor replicates
ratioDistributionsTrisomyMANew <- function(dds, genes, anChr,sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:5)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]


### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Trisomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
# {pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
# dev.off()}


trans <- qplot(ratios.Trans$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)), main=paste("Trisomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green")+
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
# {pdf(paste("transGenes",sample,".pdf",sep=""))
  print(trans)
# dev.off()}
# write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
return(ratios.Trans$sampleOverAnc)
}

ratioDistMA112 <- ratioDistributionsTrisomyMANew(ddsDefMAnewS112,genesMAnewS112,"chrXVI","Sample_112_MA_ChrXVI")
meanMA112 <- mean(ratioDistMA112)
medianMA112 <- median(ratioDistMA112)
stdevMA112 <- sd(ratioDistMA112)
ratioDistMA117 <- ratioDistributionsTrisomyMANew(ddsDefMAnewS117,genesMAnewS117,"chrV","Sample_117_MA_ChrV")
meanMA117 <- mean(ratioDistMA117)
medianMA117 <- median(ratioDistMA117)
stdevMA117 <- sd(ratioDistMA117)
ratioDistMA123 <- ratioDistributionsTrisomyMANew(ddsDefMAnewS123,genesMAnewS123,"chrV","Sample_123_MA_ChrV")
meanMA123 <- mean(ratioDistMA123)
medianMA123 <- median(ratioDistMA123)
stdevMA123 <- sd(ratioDistMA123)

## need monosomy one for MA new 29
ratioDistributionsMonosomyMANew <- function(dds, genes, anChr,sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:5)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]

### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram",breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Monosomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=0.5, col="green") + geom_vline(xintercept=1.5,col="darkmagenta") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)

# {pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
# dev.off()}


trans <- qplot(ratios.Trans$sampleOverAnc,geom="histogram",breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)),main=paste("Monosomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise")) + geom_vline(xintercept = 1) + geom_vline(xintercept=0.5, col="green") + geom_vline(xintercept=1.5,col="darkmagenta") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
# {pdf(paste("transGenes",sample,".pdf",sep=""))
  print(trans)
# dev.off()}
# write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
return(ratios.Trans$sampleOverAnc)
}

ratioDistMA29 <- ratioDistributionsMonosomyMANew(ddsDefMAnewS29,genesMAnewS29,"chrIX","Sample_29_MA_ChrIX")
meanMA29 <- mean(ratioDistMA29)
medianMA29 <- median(ratioDistMA29)
stdevMA29 <- sd(ratioDistMA29)

```

Find differentially expressed genes for each sample
```{r results, eval=TRUE,include=FALSE,echo=FALSE}
###########################################
#without aneuploid chromosomes
### DESeq ran on samples individually
resMAnewS29 <- results(ddsDefMAnewS29)
resultsNames(ddsDefMAnewS29)
#euploid line
resMAnew29 <- results(ddsDefMAnewS29, contrast=c("condition","29","Anc"))
summary(resMAnew29)

resMAnew50 <- results(ddsDefMAnewS50, contrast=c("condition","50","Anc"))
summary(resMAnew50)

resMAnew112 <- results(ddsDefMAnewS112, contrast=c("condition","112","Anc"))
summary(resMAnew112)
resMAnew112$pvalue

resMAnew115 <- results(ddsDefMAnewS115, contrast=c("condition","115","Anc"))
summary(resMAnew115)

resMAnew117 <- results(ddsDefMAnewS117, contrast=c("condition","117","Anc"))
summary(resMAnew117)

resMAnew123 <- results(ddsDefMAnewS123, contrast=c("condition","123","Anc"))
summary(resMAnew123)

resMAnew152 <- results(ddsDefMAnewS152, contrast=c("condition","152","Anc"))
summary(resMAnew152)

```

Find what genes are commonly sig DE between groups
```{r common sig genes, eval=TRUE,include=FALSE,echo=FALSE}
## Big Problem: How many genes are differentially expressed in ALL samples?
# need to get results from DESeq into a more easily workable format (ie data frame)
  # make results from DESeq a data frame to work with
  resGC1Dat <- as.data.frame(resultsS1GC)
  resGC2Dat <- as.data.frame(resultsS2GC)
  resGC3Dat <- as.data.frame(resultsS3GC)
  resGC5Dat <- as.data.frame(resultsS5GC)
  resGC9Dat <- as.data.frame(resultsS9GC)
  resGC4Dat <- as.data.frame(resultsS4GC)
  resGC7Dat <- as.data.frame(resultsS7GC)
  resGC8Dat <- as.data.frame(resultsS8GC)
  resGC11Dat <- as.data.frame(resultsS11GC)
  resGC18Dat <- as.data.frame(resultsS18GC)
  resGC49Dat <- as.data.frame(resultsS49GC)
  resGC59Dat <- as.data.frame(resultsS59GC)
  resGC61Dat <- as.data.frame(resultsS61GC)
  resGC76Dat <- as.data.frame(resultsS76GC)
  resGC77Dat <- as.data.frame(resultsS77GC)
  # add gene names as first column ( which is currently rownames)
  resGC1Dat <- cbind(GENEID = rownames(resGC1Dat), resGC1Dat)
  resGC2Dat <- cbind(GENEID = rownames(resGC2Dat), resGC2Dat)
  resGC3Dat <- cbind(GENEID = rownames(resGC3Dat), resGC3Dat)
  resGC4Dat <- cbind(GENEID = rownames(resGC4Dat), resGC4Dat)
  resGC5Dat <- cbind(GENEID = rownames(resGC5Dat), resGC5Dat)
  resGC7Dat <- cbind(GENEID = rownames(resGC7Dat), resGC7Dat)
  resGC8Dat <- cbind(GENEID = rownames(resGC8Dat), resGC8Dat)
  resGC9Dat <- cbind(GENEID = rownames(resGC9Dat), resGC9Dat)
  resGC11Dat <- cbind(GENEID = rownames(resGC11Dat), resGC11Dat)
  resGC18Dat <- cbind(GENEID = rownames(resGC18Dat), resGC18Dat)
  resGC59Dat <- cbind(GENEID = rownames(resGC59Dat), resGC59Dat)
    resGC49Dat <- cbind(GENEID = rownames(resGC49Dat), resGC49Dat)
  resGC61Dat <- cbind(GENEID = rownames(resGC61Dat), resGC61Dat)
  resGC76Dat <- cbind(GENEID = rownames(resGC76Dat), resGC76Dat)
  resGC77Dat <- cbind(GENEID = rownames(resGC77Dat), resGC77Dat)

  # need to make a column identifying the sample
    resGC1Dat$Sample <- "1"
    resGC2Dat$Sample <- "2"
    resGC3Dat$Sample <- "3"
    resGC4Dat$Sample <- "4"
    resGC5Dat$Sample <- "5"
    resGC7Dat$Sample <- "7"
    resGC8Dat$Sample <- "8"
    resGC9Dat$Sample <- "9"
    resGC11Dat$Sample <- "11"
    resGC18Dat$Sample <- "18"
        resGC49Dat$Sample <- "49"
    resGC59Dat$Sample <- "59"
    resGC61Dat$Sample <- "61"
    resGC76Dat$Sample <- "76"
    resGC77Dat$Sample <- "77"

View(resGC7Dat)


# put all sample expression data in the same data frame
allGCSamplesExpr <- rbind(resGC1Dat, resGC2Dat, resGC3Dat, resGC4Dat, resGC5Dat, resGC7Dat, resGC8Dat, resGC9Dat, resGC11Dat, resGC18Dat, resGC59Dat, resGC61Dat, resGC76Dat, resGC77Dat)
      str(allGCSamplesExpr)
      View(allGCSamplesExpr)
    # # # add chromosome numbers to this file
        matchGC <- intersect(allGCSamplesExpr$GENEID, genesGC_1$GENEID)
        str(matchGC)
    # #   # tell R which genes to keep
       keepGC <- genesGC_1$GENEID %in% matchGC
       str(keepGC)
    # #   # subset list of genes in analysis to include only the ones matching those found in          sample file
       genesGC <- genesGC_1[keepGC,]
       str(genesGC)
       View(genesGC)
       View(genesGC_1)
       dim(allGCSamplesExpr)
       dim(genesGC)
       str(genesGC_1)
       str(genesGC)
    # #   # add the correct chromosomes in
        allGCSamplesExpr <- cbind(genesGC_1$GENEID, genesGC_1$TXCHROM, allGCSamplesExpr)
        str(allGCSamplesExpr)
        View(allGCSamplesExpr)
       chr1.all <- subset(allGCSamplesExpr, allGCSamplesExpr$`genesNoAneu$GENEID`=="chrI")
       sample1 <- subset(allGCSamplesExpr, Sample==1)
       View(sample1)
       sample7 <- subset(allGCSamplesExpr, Sample==7)
         
# want to find how many genes are differentially expressed in 2 samples, 3 samples, 4 samples, etc
#
  #     cbind(names = rownames(df), df)
  # # first need to filter allSamplesExpr based on p-adj level
    allGCSamplesDiffExpr <- subset(allGCSamplesExpr, padj<0.1)
    View(allGCSamplesDiffExpr)
    write.csv(allGCSamplesDiffExpr,file="allGCDiffExpr.csv")
  #   # then ask how many times each GENEID is found in allSamplesExpr
    library(plyr)
    geneGCFreq <- plyr::count(allGCSamplesDiffExpr, c("GENEID"))
      # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
      numGCGenesDE <- plyr::count(geneGCFreq,"freq")
        # rename columns so that I know what is going on: column 1 is how many samples a gene is DE in, column 2 is how many           genes are in that category
        colnames(numGCGenesDE) <- c("numSamples", "numGenes")
          # export as .csv to use in excel or whatever
          # write.csv(numGCGenesDE, file="numb_comm_DE_genesALL_GC.csv")

##########################
          #### Same thing but for MA New Lines #####

  resMA29Dat <- as.data.frame(resMAnew29)
  resMA50Dat <- as.data.frame(resMAnew50)
  resMA112Dat <- as.data.frame(resMAnew112)
  resMA115Dat <- as.data.frame(resMAnew115)
  resMA117Dat <- as.data.frame(resMAnew117)
  resMA123Dat <- as.data.frame(resMAnew123)
  resMA152Dat <- as.data.frame(resMAnew152)

  # add gene names as first column ( which is currently rownames)
  resMA29Dat <- cbind(GENEID = rownames(resMA29Dat), resMA29Dat)
  resMA50Dat <- cbind(GENEID = rownames(resMA50Dat), resMA50Dat)
  resMA112Dat <- cbind(GENEID = rownames(resMA112Dat), resMA112Dat)
  resMA115Dat <- cbind(GENEID = rownames(resMA115Dat), resMA115Dat)
  resMA117Dat <- cbind(GENEID = rownames(resMA117Dat), resMA117Dat)
  resMA123Dat <- cbind(GENEID = rownames(resMA123Dat), resMA123Dat)
  resMA152Dat <- cbind(GENEID = rownames(resMA152Dat), resMA152Dat)
# need to make a column identifying the sample
    resMA29Dat$Sample <- "29"
    resMA50Dat$Sample <- "50"
    resMA112Dat$Sample <- "112"
    resMA115Dat$Sample <- "115"
    resMA117Dat$Sample <- "117"
    resMA123Dat$Sample <- "123"
    resMA152Dat$Sample <- "152"


View(resMA29Dat)

##### LEFT OFF HERE
## ** struggling to cbind genes with allMAnewSamplesExpr
## worked for GC samples and nothing has changed so why is it not working for these samples??


   allMAnewSamplesExpr <- rbind(resMA29Dat, resMA50Dat, resMA112Dat, resMA115Dat, resMA117Dat, resMA123Dat, resMA152Dat)
      str(allGCSamplesExpr)
    # # # add chromosome numbers to this file
        matchMAnew<- intersect(allMAnewSamplesExpr$GENEID, genesGC_1$GENEID)
        str(matchMAnew)
    # #   # tell R which genes to keep
       keepMAnew <- genesGC_1$GENEID %in% matchMAnew
    str(keepMAnew)
       # #   # subset list of genes in analysis to include only the ones matching those found in          sample file
       genesMAnew <- genesGC_1[keepMAnew,]
       str(genesMAnew)
       View(genesMAnew)
       View(genesGC_1)
       dim(allMAnewSamplesExpr)
       dim(genesMAnew)
       str(genesGC_1)
       str(genesGC)
    # #   # add the correct chromosomes in
        allMAnewSamplesExpr <- cbind(genesGC_1$GENEID, genesGC_1$TXCHROM, allMAnewSamplesExpr)
        str(allMAnewSamplesExpr)
        View(allMAnewSamplesExpr)
       chr1.all <- subset(allMAnewSamplesExpr, allMAnewSamplesExpr$`genesNoAneu$GENEID`=="chrI")
       sample29MAnew <- subset(allMAnewSamplesExpr, Sample==29)
       View(sample29MAnew)
       sample50MAnew <- subset(allMAnewSamplesExpr, Sample==50)
         
# want to find how many genes are differentially expressed in 2 samples, 3 samples, 4 samples, etc
#
  #     cbind(names = rownames(df), df)
  # # first need to filter allSamplesExpr based on p-adj level
    allMAnewSamplesDiffExpr <- subset(allMAnewSamplesExpr, padj<0.1)
    write.csv(allMAnewSamplesDiffExpr,file="allMAnewDiffExpr.csv")
  #   # then ask how many times each GENEID is found in allSamplesExpr
    library(plyr)
    geneMAnewFreq <- plyr::count(allMAnewSamplesDiffExpr, c("GENEID"))
      # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
      numMAnewGenesDE <- plyr::count(geneMAnewFreq,"freq")
        # rename columns so that I know what is going on: column 1 is how many samples a gene is DE in, column 2 is how many           genes are in that category
        colnames(numMAnewGenesDE) <- c("numSamples", "numGenes")
          # export as .csv to use in excel or whatever
          # write.csv(numMAnewGenesDE, file="numb_comm_DE_genesALL_MAnew.csv")
          ################## MA old Lines
  ########################################################################
  # make results from DESeq a data frame to work with
  resMAold1Dat <- as.data.frame(resultsS1MAold)
  resMAold2Dat <- as.data.frame(resultsS2MAold)
  resMAold3Dat <- as.data.frame(resultsS3MAold)
  resMAold4Dat <- as.data.frame(resultsS4MAold)
  resMAold5Dat <- as.data.frame(resultsS5MAold)
  resMAold7Dat <- as.data.frame(resultsS7MAold)
  resMAold8Dat <- as.data.frame(resultsS8MAold)
  resMAold9Dat <- as.data.frame(resultsS9MAold)
  resMAold15Dat <- as.data.frame(resultsS15MAold)
  resMAold28Dat <- as.data.frame(resultsS28MAold)
  resMAold88Dat <- as.data.frame(resultsS88MAold)
  resMAold108Dat <- as.data.frame(resultsS108MAold)
  resMAold119Dat <- as.data.frame(resultsS119MAold)
  # add gene names as first column ( which is currently rownames)
  resMAold1Dat <- cbind(GENEID = rownames(resMAold1Dat), resMAold1Dat)
  resMAold2Dat <- cbind(GENEID = rownames(resMAold2Dat), resMAold2Dat)
  resMAold3Dat <- cbind(GENEID = rownames(resMAold3Dat), resMAold3Dat)
  resMAold4Dat <- cbind(GENEID = rownames(resMAold4Dat), resMAold4Dat)
  resMAold5Dat <- cbind(GENEID = rownames(resMAold5Dat), resMAold5Dat)
  resMAold7Dat <- cbind(GENEID = rownames(resMAold7Dat), resMAold7Dat)
  resMAold8Dat <- cbind(GENEID = rownames(resMAold8Dat), resMAold8Dat)
  resMAold9Dat <- cbind(GENEID = rownames(resMAold9Dat), resMAold9Dat)
  resMAold15Dat <- cbind(GENEID = rownames(resMAold15Dat), resMAold15Dat)
  resMAold28Dat <- cbind(GENEID = rownames(resMAold28Dat), resMAold28Dat)
  resMAold88Dat <- cbind(GENEID = rownames(resMAold88Dat), resMAold88Dat)
  resMAold108Dat <- cbind(GENEID = rownames(resMAold108Dat), resMAold108Dat)
  resMAold119Dat <- cbind(GENEID = rownames(resMAold119Dat), resMAold119Dat)


  # need to make a column identifying the sample
    resMAold1Dat$Sample <- "1"
    resMAold2Dat$Sample <- "2"
    resMAold3Dat$Sample <- "3"
    resMAold4Dat$Sample <- "4"
    resMAold5Dat$Sample <- "5"
    resMAold7Dat$Sample <- "7"
    resMAold8Dat$Sample <- "8"
    resMAold9Dat$Sample <- "9"
    resMAold15Dat$Sample <- "15"
    resMAold28Dat$Sample <- "28"
    resMAold88Dat$Sample <- "88"
    resMAold108Dat$Sample <- "108"
    resMAold119Dat$Sample <- "119"

      # put all sample expression data in the same data frame
allMAoldSamplesExpr <- rbind(resMAold1Dat, resMAold2Dat, resMAold3Dat, resMAold4Dat, resMAold5Dat, resMAold7Dat, resMAold8Dat, resMAold9Dat, resMAold15Dat, resMAold28Dat, resMAold88Dat, resMAold108Dat, resMAold119Dat)
      str(allMAoldSamplesExpr)
      View(allMAoldSamplesExpr)
    # # # add chromosome numbers to this file
        matchMAold <- intersect(allMAoldSamplesExpr$GENEID, genesGC_1$GENEID)
        str(matchMAold)

    # #   # tell R which genes to keep
       keepMAold <- genesGC_1$GENEID %in% matchMAold
       str(keepMAold)

    # #   # subset list of genes in analysis to include only the ones matching those found in          sample file
       genesMAold <- genesGC_1[keepMAold,]
       str(genesMAold)
       View(genesMAold)
       View(genesGC_1)
       dim(allMAoldSamplesExpr)
       dim(genesMAold)
       str(genesGC_1)
       str(genesMAold)
    # #   # add the correct chromosomes in
        allMAoldSamplesExpr <- cbind(genesMAold$GENEID, genesMAold$TXCHROM, allMAoldSamplesExpr)

        str(allGCSamplesExpr)
        View(allGCSamplesExpr)
       chr1.all <- subset(allGCSamplesExpr, allGCSamplesExpr$`genesNoAneu$GENEID`=="chrI")
       sample1 <- subset(allGCSamplesExpr, Sample==1)
       View(sample1)
       sample7 <- subset(allGCSamplesExpr, Sample==7)
        
# want to find how many genes are differentially expressed in 2 samples, 3 samples, 4 samples, etc
#
  #     cbind(names = rownames(df), df)
  # # first need to filter allSamplesExpr based on p-adj level
    allGCSamplesDiffExpr <- subset(allGCSamplesExpr, padj<0.1)
    write.csv(allGCSamplesDiffExpr,file="allGCDiffExpr.csv")
  #   # then ask how many times each GENEID is found in allSamplesExpr
    library(plyr)
    geneGCFreq <- plyr::count(allGCSamplesDiffExpr, c("GENEID"))
      # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
      numGCGenesDE <- plyr::count(geneGCFreq,"freq")
        # rename columns so that I know what is going on: column 1 is how many samples a gene is DE in, column 2 is how many           genes are in that category
        colnames(numGCGenesDE) <- c("numSamples", "numGenes")
          # export as .csv to use in excel or whatever
          # write.csv(numGCGenesDE, file="numb_comm_DE_genesALL_GC.csv")          


##############################################################################################          
# want to know how many genes are differentially expressed commonly in 1,2,3 etc EUPLOID lines
#
  # put all EUPLOID samples in same data frame
    eupGCSamplesExpr <- rbind(resGC1Dat, resGC2Dat, resGC3Dat, resGC5Dat, resGC9Dat)
      # first need to filter allSamplesExpr based on p-adj level
      eupGCSamplesDiffExpr <- subset(eupGCSamplesExpr, padj<0.1)
      #

        # then ask how many times each GENEID  is found in eupSamplesExpr
        library(plyr)
        geneGCFreqEu <- plyr::count(eupGCSamplesDiffExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numGCEupGenesDE <- plyr::count(geneGCFreqEu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numGCEupGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numGCEupGenesDE, file="numb_comm_DE__GC_genesEUP.csv")
##############################################################################################          
# want to know how many genes are differentially expressed commonly in 1,2,3 etc EUPLOID lines
#
  # put all EUPLOID samples in same data frame
    eupMAnewSamplesExpr <- rbind(resMA50Dat, resMA115Dat)
      # first need to filter allSamplesExpr based on p-adj level
      eupMAnewSamplesDiffExpr <- subset(eupMAnewSamplesExpr, padj<0.1)
      #

        # then ask how many times each GENEID  is found in eupSamplesExpr
        library(plyr)
        geneMAnewFreqEu <- plyr::count(eupMAnewSamplesDiffExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numMAnewEupGenesDE <- plyr::count(geneMAnewFreqEu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numMAnewEupGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numMAnewEupGenesDE, file="numb_comm_DE_MAnew_genesEUP.csv")              
##############################################################################################          
# want to know how many genes are differentially expressed commonly in 1,2,3 etc ANEUPLOID lines
#
  # put all ANEUPLOID samples in same data frame
    aneuGCSamplesExpr <- rbind(resGC4Dat, resGC7Dat, resGC8Dat,  resGC11Dat, resGC18Dat, resGC59Dat, resGC61Dat, resGC76Dat, resGC77Dat)
      # first need to filter allSamplesExpr based on p-adj level
      aneuGCSamplesDiffExpr <- subset(aneuGCSamplesExpr, padj<0.1)
      #
        # then ask how many times each GENEID  is found in eupSamplesExpr
        library(plyr)
        geneGCFreqAneu <- plyr::count(aneuGCSamplesDiffExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numGCAneuGenesDE <- plyr::count(geneGCFreqAneu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numGCAneuGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numGCAneuGenesDE, file="numb_comm_DE_genesANEU.csv")     

##############################################################################################          
# want to know how many genes are differentially expressed commonly in 1,2,3 etc ANEUPLOID lines
# with the SAME ANEUPLOID CHROMOSOME
  # put all samples in same data frame
    aneuChrIGCSamplesExpr <- rbind(resGC7Dat,  resGC11Dat, resGC18Dat)
      aneuChrItriGCExpr <- rbind(resGC7Dat, resGC18Dat)
      aneuChrVtriGCExpr <- rbind(resGC4Dat)
      aneuChrVIItriGCExpr <- rbind(resGC59Dat, resGC61Dat)
      aneuChrXIItriGCExpr <- rbind(resGC18Dat, resGC77Dat)
      aneuLine76GC <- resGC76Dat
      aneuLine76GC <- merge(genesGC,resGC76Dat,by="GENEID")
      write.csv(aneuLine76GC,file="aneuLine76GC.csv")
      aneuLine76GCdiffExpr <- subset(resGC76Dat,padj<0.1)
      aneuLine76GCdiffExpr <- merge(genesGC,aneuLine76GCdiffExpr,by.x="GENEID",by.y="GENEID")
      write.csv(aneuLine76GCdiffExpr,file="aneuLine76GCdiffExpr.csv")

      # first need to filter allSamplesExpr based on p-adj level
      aneuChrIGCSamplesDiffExpr <- subset(aneuChrIGCSamplesExpr, padj<0.1)
        triChrIGCSamplesDiffExpr <- subset(aneuChrItriGCExpr, padj<0.1)
        triChrVGCSamplesDiffExpr <- subset(aneuChrVtriGCExpr, padj<0.1)
        triChrVIIGCSamplesDiffExpr <- subset(aneuChrVIItriGCExpr, padj<0.1)
        triChrXIIGCSamplesDiffExpr <- subset(aneuChrXIItriGCExpr, padj<0.1)
      #
        # then ask how many times each GENEID  is found in eupSamplesExpr
        library(plyr)
        geneFreqGCAneuChrI <- plyr::count(aneuChrIGCSamplesDiffExpr, c("GENEID"))
          geneFreqGCtriChrI <- plyr::count(triChrIGCSamplesDiffExpr, c("GENEID"))
          geneFreqGCtriChrV <- plyr::count(triChrVGCSamplesDiffExpr, c("GENEID"))
          geneFreqGCtriChrVII <- plyr::count(triChrVIIGCSamplesDiffExpr, c("GENEID"))
          geneFreqtriGCChrXII <- plyr::count(triChrXIIGCSamplesDiffExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numGCAneuChrIGenesDE <- plyr::count(geneFreqGCAneuChrI,"freq")
            numtriGCChrIGenesDE <- plyr::count(geneFreqGCtriChrI,"freq")
            numtriGCChrVGenesDE <- plyr::count(geneFreqGCtriChrV,"freq")
            numtriGCChrVIIGenesDE <- plyr::count(geneFreqGCtriChrVII,"freq")
            numtriGCChrXIIGenesDE <- plyr::count(geneFreqtriGCChrXII,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numGCAneuChrIGenesDE) <- c("numSamples", "numGenes")
              colnames(numtriGCChrIGenesDE) <- c("numSamples", "numGenes")
              colnames(numtriGCChrVGenesDE) <- c("numSamples", "numGenes")
              colnames(numtriGCChrVIIGenesDE) <- c("numSamples", "numGenes")
              colnames(numtriGCChrXIIGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numGCAneuChrIGenesDE, file="numb_comm_DE_GC_genesANEUchrI.csv")  
                write.csv(numtriGCChrIGenesDE, file="numb_comm_DE_GC_genesTRIchrI.csv")
                write.csv(numtriGCChrVGenesDE, file="numb_comm_DE_GC_genesTRIchrV.csv")
                write.csv(numtriGCChrVIIGenesDE, file="numb_comm_DE_GC_genesTRIchrVII.csv")
                write.csv(numtriGCChrXIIGenesDE, file="numb_comm_DE_GC_genesTRIchrXII.csv")

#####################################################################################
# make  this into a function
# want to know how many genes are NOT DE on the aneuploid chromosome in aneuploid samples

getPercNonDEgenesGC <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
samplenotDE <- subset(sample, padj>0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(samplenotDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, samplenotDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        notDEAneuChr <- subset(samp, chr==chrom)
      # count how many genes are not DE on this chrom in this sample
          countNotDeSample <- nrow(notDEAneuChr)
         # find how many genes are analyzed total on this chromosome
          chrgenes <- subset(genesGC, TXCHROM==chrom)
          countChr <- nrow(chrgenes)     
            # what percent of genes on chromosome V are not DE in sample 4?
              percNotDeS <- setNames(data.frame(matrix(ncol = 3, nrow = 1)), c("Sample", "Chromosome", "percentNotDE"))
              percNotDeS$Sample <- sample[1,8]
              percNotDeS$Chromosome <- chrom
              percNotDeS$percentNotDE <- countNotDeSample/countChr * 100
return(percNotDeS)
}

# test it out on one of the ones I did already
# try S4
percNonDEGCS4c5 <- getPercNonDEgenesGC(resGC4Dat, "chrV")
percNonDEGCS7c1 <- getPercNonDEgenesGC(resGC7Dat, "chrI")
percNonDEGCS8c16 <- getPercNonDEgenesGC(resGC8Dat, "chrXVI")
percNonDEGCS11c1 <- getPercNonDEgenesGC(resGC11Dat, "chrI")
percNonDEGCS11c15 <- getPercNonDEgenesGC(resGC11Dat, "chrXV")
percNonDEGCS18c1 <- getPercNonDEgenesGC(resGC18Dat, "chrI")
percNonDEGCS18c12 <- getPercNonDEgenesGC(resGC18Dat, "chrXII")
percNonDEGCS59c7 <- getPercNonDEgenesGC(resGC59Dat, "chrVII")
percNonDEGCS61c7 <- getPercNonDEgenesGC(resGC61Dat, "chrVII")
percNonDEGCS76c9 <- getPercNonDEgenesGC(resGC76Dat, "chrIX")
percNonDEGCS76c14 <- getPercNonDEgenesGC(resGC76Dat, "chrXIV")
percNonDEGCS77c12 <- getPercNonDEgenesGC(resGC77Dat, "chrXII")

percNonDEAllGCSamples <- rbind(percNonDEGCS4c5,percNonDEGCS7c1, percNonDEGCS8c16, percNonDEGCS11c1, percNonDEGCS11c15, percNonDEGCS18c1, percNonDEGCS18c12, percNonDEGCS59c7, percNonDEGCS61c7, percNonDEGCS76c9, percNonDEGCS76c14, percNonDEGCS77c12)

write.csv(percNonDEAllGCSamples, file="percNonDEAllgcLines.csv")

#########################################################################################

getNumNonDEGCgenes <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
samplenotDE <- subset(sample, padj>0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(samplenotDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, samplenotDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        notDEAneuChr <- subset(samp, chr==chrom)
      # count how many genes are not DE on this chrom in this sample
          countNotDeSample <- nrow(notDEAneuChr)
         # find how many genes are analyzed total on this chromosome
          chrgenes <- subset(genesGC, TXCHROM==chrom)
          countChr <- nrow(chrgenes)     
            # what percent of genes on chromosome V are not DE in sample 4?
              numNotDeS <- setNames(data.frame(matrix(ncol = 2, nrow = 1)), c("Sample", "Chromosome"))
              numNotDeS$Sample <- sample[1,8]
              numNotDeS$Chromosome <- chrom
              numNotDeS$countNotDe <- countNotDeSample
            numNotDeS$countChrom <- countChr
return(numNotDeS)
}

numNonDEGCS4c5 <- getNumNonDEGCgenes(resGC4Dat, "chrV")
numNonDEGCS7c1 <- getNumNonDEGCgenes(resGC7Dat, "chrI")
numNonDEGCS8c16 <- getNumNonDEGCgenes(resGC8Dat, "chrXVI")
numNonDEGCS11c1 <- getNumNonDEGCgenes(resGC11Dat, "chrI")
numNonDEGCS11c15 <- getNumNonDEGCgenes(resGC11Dat, "chrXV")
numNonDEGCS18c1 <- getNumNonDEGCgenes(resGC18Dat, "chrI")
numNonDEGCS18c12 <- getNumNonDEGCgenes(resGC18Dat, "chrXII")
numNonDEGCS59c7 <- getNumNonDEGCgenes(resGC59Dat, "chrVII")
numNonDEGCS61c7 <- getNumNonDEGCgenes(resGC61Dat, "chrVII")
numNonDEGCS76c9 <- getNumNonDEGCgenes(resGC76Dat, "chrIX")
numNonDEGCS76c14 <- getNumNonDEGCgenes(resGC76Dat, "chrXIV")
numNonDEGCS77c12 <- getNumNonDEGCgenes(resGC77Dat, "chrXII")

numNonDEAllGCSamples <- rbind(numNonDEGCS4c5,numNonDEGCS7c1, numNonDEGCS8c16, numNonDEGCS11c1, numNonDEGCS11c15, numNonDEGCS18c1, numNonDEGCS18c12, numNonDEGCS59c7, numNonDEGCS61c7, numNonDEGCS76c9, numNonDEGCS76c14, numNonDEGCS77c12)

write.csv(numNonDEAllGCSamples, file="numNonDEAllgcLines.csv")




```

Find which genes are differentially expressed 
```{r DE_genes,include=TRUE,echo=FALSE,eval=TRUE}
#######################################################################################
# want to know WHAT genes are NOT DE on the aneuploid chromosome in aneuploid samples


getNonDEGCgenes <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
samplenotDE <- subset(sample, padj>0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(samplenotDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, samplenotDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        notDEAneuChr <- subset(samp, chr==chrom)
return(notDEAneuChr)
}

# test it out on one of the ones I did already
# try S4
nonDEGCS4c5 <- getNonDEGCgenes(resGC4Dat, "chrV")
# write.csv(nonDEGCS4c5, file="nonDEGCS4c5.csv")
nonDEGCS7c1 <- getNonDEGCgenes(resGC7Dat, "chrI")
# write.csv(nonDEGCS7c1, file="nonDEGCS7c1.csv")
nonDEGCS8c16 <- getNonDEGCgenes(resGC8Dat, "chrXVI")
# write.csv(nonDEGCS8c16, file="nonDEGCS8c16.csv")
nonDEGCS11c1 <- getNonDEGCgenes(resGC11Dat, "chrI")
# write.csv(nonDEGCS11c1, file="nonDEGCS11c1.csv")
nonDEGCS11c15 <- getNonDEGCgenes(resGC11Dat, "chrXV")
# write.csv(nonDEGCS11c15, file="nonDEGCS11c15.csv")
nonDEGCS18c1 <- getNonDEGCgenes(resGC18Dat, "chrI")
# write.csv(nonDEGCS18c1, file="nonDEGCS18c1.csv")
nonDEGCS18c12 <- getNonDEGCgenes(resGC18Dat, "chrXII")
# write.csv(nonDEGCS18c12, file="nonDEGCS18c12.csv")
nonDEGCS59c7 <- getNonDEGCgenes(resGC59Dat, "chrVII")
# write.csv(nonDEGCS59c7, file="nonDEGCS59c7.csv")
nonDEGCS61c7 <- getNonDEGCgenes(resGC61Dat, "chrVII")
# write.csv(nonDEGCS61c7, file="nonDEGCS61c7.csv")
nonDEGCS76c9 <- getNonDEGCgenes(resGC76Dat, "chrIX")
# write.csv(nonDEGCS76c9, file="nonDEGCS76c9.csv")
nonDEGCS76c14 <- getNonDEGCgenes(resGC76Dat, "chrXIV")
# write.csv(nonDEGCS76c14, file="nonDEGCS76c14.csv")
nonDEGCS77c12 <- getNonDEGCgenes(resGC77Dat, "chrXII")
# write.csv(nonDEGCS77c12, file="nonDEGCS77c12.csv")

nonDEMA152c1 <- getNonDEGCgenes(resMA152Dat, "chrI")
nonDEMA117c5 <- getNonDEGCgenes(resMA117Dat, "chrV")
nonDEMA123c5 <- getNonDEGCgenes(resMA123Dat, "chrV")
nonDEMA108c7 <- getNonDEGCgenes(resMAold108Dat, "chrVII")
nonDEMA108c9 <- getNonDEGCgenes(resMAold108Dat, "chrIX")
nonDEMA15c9 <- getNonDEGCgenes(resMAold15Dat, "chrIX")
nonDEMA29c9 <- getNonDEGCgenes(resMA29Dat, "chrIX")
nonDEMA119c9 <- getNonDEGCgenes(resMAold119Dat, "chrIX")
nonDEMA9c14 <- getNonDEGCgenes(resMAold9Dat, "chrXIV")
nonDEMA112c16 <- getNonDEGCgenes(resMA112Dat, "chrXVI")


nonDE_upSet_listChrV <- list(MAS117=nonDEMA117c5$GENEID,
                       MAS123=nonDEMA123c5$GENEID
)


# upset(fromList(nonDE_upSet_listChrV),nsets=2,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "Non DE ChrV Genes")



#######################################################################################
# want to know WHAT genes are DE on the aneuploid chromosome in aneuploid samples

getDEGCgenes <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        DEAneuChr <- subset(samp, chr==chrom)
return(DEAneuChr)
}


# test it out on one of the ones I did already
# try S4
DEGCS4.c5 <- getDEGCgenes(resGC4Dat, "chrV")
# View(DEGCS4.c5)
# write.csv(DEGCS4.c5, file="DEGCS4c5.csv")
geneIDsGC4 <- DEGCS4.c5[,2]
str(geneIDsGC4)
geneIDsGC4 <- as.vector(geneIDsGC4)
# write.table(geneIDsGC4, file="DEgeneIDs4GC.txt")
DESGC7.c1 <- getDEGCgenes(resGC7Dat, "chrI")
# write.csv(DESGC7.c1, file="DESGC7c1.csv")
DESGC8.c16 <- getDEGCgenes(resGC8Dat, "chrXVI")
# write.csv(DESGC8.c16, file="DESGC8c16.csv")
DESGC11.c1 <- getDEGCgenes(resGC11Dat, "chrI")
# write.csv(DESGC11.c1, file="DESGC11c1.csv")
DESGC11.c15 <- getDEGCgenes(resGC11Dat, "chrXV")
# write.csv(DESGC11.c15, file="DESGC11c15.csv")
DESGC18.c1 <- getDEGCgenes(resGC18Dat, "chrI")
# write.csv(DESGC18.c1, file="DESGC18c1.csv")
DESGC18.c12 <- getDEGCgenes(resGC18Dat, "chrXII")
# write.csv(DESGC18.c12, file="DESGC18c12.csv")
DESGC59.c7 <- getDEGCgenes(resGC59Dat, "chrVII")
# write.csv(DESGC59.c7, file="DESGC59c7.csv")
DESGC61.c7 <- getDEGCgenes(resGC61Dat, "chrVII")
# write.csv(DESGC61.c7, file="DESGC61c7.csv")
DESGC76.c9 <- getDEGCgenes(resGC76Dat, "chrIX")
# write.csv(DESGC76.c9, file="DESGC76c9.csv")
DESGC76.c14 <- getDEGCgenes(resGC76Dat, "chrXIV")
# write.csv(DESGC76.c14, file="DESGC76c14.csv")
DESGC77.c12 <- getDEGCgenes(resGC77Dat, "chrXII")
# write.csv(DESGC77.c12, file="DESGC77c12.csv")



### DE genes in general (for euploid samples)
getDEgenes <- function(sample) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
return(samp)
}

GCS1 <- getDEgenes(resGC1Dat)
GCS2 <- getDEgenes(resGC2Dat)
GCS3 <- getDEgenes(resGC3Dat)
GCS5 <- getDEgenes(resGC5Dat)
GCS9 <- getDEgenes(resGC9Dat)

MAS1 <- getDEgenes(resMAold1Dat)
MAS2 <- getDEgenes(resMAold2Dat)
MAS3 <- getDEgenes(resMAold3Dat)
MAS4 <- getDEgenes(resMAold4Dat)
MAS5 <- getDEgenes(resMAold5Dat)
MAS7 <- getDEgenes(resMAold7Dat)
MAS8 <- getDEgenes(resMAold8Dat)
MAS28 <- getDEgenes(resMAold28Dat)
MAS50 <- getDEgenes(resMA50Dat)
MAS115 <- getDEgenes(resMA115Dat)


DE_upSet_listMAEups <- list(MAS1=MAS1$GENEID,
                      MAS2=MAS2$GENEID,
                      MAS3=MAS3$GENEID,
                      MAS4=MAS4$GENEID,
                      MAS5=MAS5$GENEID,
                      MAS7=MAS7$GENEID,
                      MAS8=MAS8$GENEID,
                      MAS28=MAS28$GENEID,
                      MAS115=MAS115$GENEID,
                      MAS50=MAS50$GENEID
)


library(UpSetR)
upset(fromList(DE_upSet_listMAEups),nsets=10,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "Shared DE Genes in Euploid Lines")

GCS1 <- getDEgenes(resGC1Dat)
GCS2 <- getDEgenes(resGC2Dat)
GCS3 <- getDEgenes(resGC3Dat)
GCS5 <- getDEgenes(resGC5Dat)
GCS9 <- getDEgenes(resGC9Dat)

DE_upSet_listGCEups <- list(GCS1=GCS1$GENEID,
                      GCS2=GCS2$GENEID,
                      GCS3=GCS3$GENEID,
                      GCS5=GCS5$GENEID,
                      GCS9=GCS9$GENEID
)


upset(fromList(DE_upSet_listGCEups),nsets=5,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "Shared DE Genes in Euploid Lines")


################################################################
# want to get DE genes in aneuploid lines that are NOT located on the aneuploid chromosome

getDEgenesGCElsewhere <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        DENotAneuChr <- subset(samp, chr!=chrom)
return(DENotAneuChr)
}

deGenesNotAneuGCS4 <- getDEgenesGCElsewhere(resGC4Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS7 <- getDEgenesGCElsewhere(resGC7Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS8 <- getDEgenesGCElsewhere(resGC8Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS18 <- getDEgenesGCElsewhere(resGC18Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS49 <- getDEgenesGCElsewhere(resGC49Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS59 <- getDEgenesGCElsewhere(resGC59Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS61 <- getDEgenesGCElsewhere(resGC61Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS76 <- getDEgenesGCElsewhere(resGC76Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS77 <- getDEgenesGCElsewhere(resGC77Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS11 <- getDEgenesGCElsewhere(resGC11Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))


# View(deGenesNotAneuGCS4)

DE_upSet_list <- list(GCS4=deGenesNotAneuGCS4$GENEID,
                      GCS7=deGenesNotAneuGCS7$GENEID,
                      GCS8=deGenesNotAneuGCS8$GENEID,
                      GCS11=deGenesNotAneuGCS11$GENEID,
                      GCS18=deGenesNotAneuGCS18$GENEID,
                      GCS49=deGenesNotAneuGCS49$GENEID,
                      GCS59=deGenesNotAneuGCS59$GENEID,
                      GCS61=deGenesNotAneuGCS61$GENEID,
                      GCS76=deGenesNotAneuGCS76$GENEID,
                      GCS77=deGenesNotAneuGCS77$GENEID
)


upset(fromList(DE_upSet_list),nsets=10,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "DE Trans Genes")

deGenesNotAneuMAS152 <- getDEgenesGCElsewhere(resMA152Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS112 <- getDEgenesGCElsewhere(resMA112Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS117 <- getDEgenesGCElsewhere(resMA117Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS123 <- getDEgenesGCElsewhere(resMA123Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS108 <- getDEgenesGCElsewhere(resMAold108Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS119 <- getDEgenesGCElsewhere(resMAold119Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS88 <- getDEgenesGCElsewhere(resMAold88Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS9 <- getDEgenesGCElsewhere(resMAold9Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS15 <- getDEgenesGCElsewhere(resMAold15Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS29 <- getDEgenesGCElsewhere(resMA29Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))


DE_upSet_list_MA <- list(MAS9=deGenesNotAneuMAS9$GENEID,
                      MAS15=deGenesNotAneuMAS15$GENEID,
                      MAS152=deGenesNotAneuMAS152$GENEID,
                      MAS112=deGenesNotAneuMAS112$GENEID,
                      MAS117=deGenesNotAneuMAS117$GENEID,
                      MAS108=deGenesNotAneuMAS108$GENEID,
                      MAS119=deGenesNotAneuMAS119$GENEID,
                      MAS88=deGenesNotAneuMAS88$GENEID,
                      MAS9=deGenesNotAneuMAS9$GENEID,
                      MAS19=deGenesNotAneuMAS29$GENEID
)


upset(fromList(DE_upSet_list_MA),nsets=10,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "DE Trans Genes")

#######################################################################################
# want to know WHAT genes are DE on the aneuploid chromosome in aneuploid samples

getDEgenesGCAll <- function(sample) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   

sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        DEAll  <- samp
return(DEAll)
}

# test it out on one of the ones I did already
# try S4
DEGCS1all <- getDEgenesGCAll(resGC1Dat)
DEGCS2all <- getDEgenesGCAll(resGC2Dat)
DEGCS3all <- getDEgenesGCAll(resGC3Dat)
DEGCS4all <- getDEgenesGCAll(resGC4Dat)
DEGCS5all <- getDEgenesGCAll(resGC5Dat)
DEGCS7all <- getDEgenesGCAll(resGC7Dat)
DEGCS8all <- getDEgenesGCAll(resGC8Dat)
DEGCS9all <- getDEgenesGCAll(resGC9Dat)
DEGCS11all <- getDEgenesGCAll(resGC11Dat)
DEGCS18all <- getDEgenesGCAll(resGC18Dat)
DEGCS49all <- getDEgenesGCAll(resGC49Dat)
DEGCS59all <- getDEgenesGCAll(resGC59Dat)
DEGCS61all <- getDEgenesGCAll(resGC61Dat)
DEGCS76all <- getDEgenesGCAll(resGC76Dat)
DEGCS77all <- getDEgenesGCAll(resGC77Dat)


########### Non DE genes on aneuploid chromosome 
getnonDEGCgenes <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
sampleDE <- subset(sample, padj>0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        nonDEAneuChr <- subset(samp, chr==chrom)
return(nonDEAneuChr)
}

DEnonGCS4.c5 <- getnonDEGCgenes(resGC4Dat, "chrV")
# View(DEnonGCS4.c5)
# write.csv(DEGCS4.c5, file="DEGCS4c5.csv")

# write.table(geneIDsGC4, file="DEgeneIDs4GC.txt")

nonDESGC7.c1 <- getnonDEGCgenes(resGC7Dat, "chrI")
# write.csv(DESGC7.c1, file="DESGC7c1.csv")

nonDESGC8.c16 <- getnonDEGCgenes(resGC8Dat, "chrXVI")
# write.csv(DESGC8.c16, file="DESGC8c16.csv")

nonESGC11.c1 <- getnonDEGCgenes(resGC11Dat, "chrI")
# write.csv(DESGC11.c1, file="DESGC11c1.csv")

nonDESGC11.c15 <- getnonDEGCgenes(resGC11Dat, "chrXV")
# write.csv(DESGC11.c15, file="DESGC11c15.csv")

nonDESGC18.c1 <- getnonDEGCgenes(resGC18Dat, "chrI")
# write.csv(DESGC18.c1, file="DESGC18c1.csv")

nonDESGC18.c12 <- getnonDEGCgenes(resGC18Dat, "chrXII")
# write.csv(DESGC18.c12, file="DESGC18c12.csv")

nonDESGC49.c5 <- getnonDEGCgenes(resGC49Dat, "chrV")

nonDESGC59.c7 <- getnonDEGCgenes(resGC59Dat, "chrVII")
# write.csv(DESGC59.c7, file="DESGC59c7.csv")

nonDESGC61.c7 <- getnonDEGCgenes(resGC61Dat, "chrVII")
# write.csv(DESGC61.c7, file="DESGC61c7.csv")

nonDESGC76.c9 <- getnonDEGCgenes(resGC76Dat, "chrIX")
# write.csv(DESGC76.c9, file="DESGC76c9.csv")

nonDESGC76.c14 <- getnonDEGCgenes(resGC76Dat, "chrXIV")
# write.csv(DESGC76.c14, file="DESGC76c14.csv")

nonDESGC77.c12 <- getnonDEGCgenes(resGC77Dat, "chrXII")



nonDE_upSet_listChrI <- list(GCS7=nonDESGC7.c1$GENEID,
                         GCS18=nonDESGC18.c1$GENEID,
                         GCS11=nonESGC11.c1$GENEID
                     
)

upset(fromList(nonDE_upSet_listChrI),nsets=3,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "Non-DE ChrI Genes")

nonDE_upSet_listChrXII <- list(GCS18=nonDESGC18.c12$GENEID,
                         GCS77=nonDESGC77.c12$GENEID
                     
)

upset(fromList(nonDE_upSet_listChrXII),nsets=2,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "Non-DE ChrXII Genes")

nonDE_upSet_listChrVII <- list(GCS59=nonDESGC59.c7$GENEID,
                         GCS61=nonDESGC61.c7$GENEID
                     
)

upset(fromList(nonDE_upSet_listChrVII),nsets=2,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "Non-DE ChrVII Genes")


########################################################################################
# what DE genes are shared between samples aneuploid with the same chromosome?
  # use data frames containing only DE genes in the given samples

library(dplyr)

getCommonDEGCgenes <- function(sample1, sample2) {
  # join them together by GENEID column
  chr.all <- inner_join(sample1, sample2, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matched <- intersect(chr.all$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepAll <- chr.all$GENEID %in% matched
          # only keep those genes in the data frame
          chr.all <- chr.all[keepAll,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenAll <- genesGC$GENEID %in% matched
              # put those genes in a vector
              GenAll <- genesGC[keepGenAll,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chr.all <- cbind(GenAll$TXCHROM, chr.all)
                return(chr.all)
}

chrVIIAllGC <- getCommonDEGCgenes(genesGC59DE, genesGC61DE)
# write.csv(chrVIIAllGC, file="chrVIIallDE_GC.csv")
chrXIIAllGC <- getCommonDEGCgenes(genesGC18DE, genesGC77DE)
# write.csv(chrXIIAllGC, file="chrXIIallDE_GC.csv")

DE_upSet_list <- list(GCS59=genesGC59DE$GENEID,
                      GCS61=genesGC61DE$GENEID
                     
)



upset(fromList(DE_upSet_list),nsets=2,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "DE Trans Genes")

########################################################################################
# what NON-DE genes are shared between samples aneuploid with the same chromosome?
  # use data frames containing only NON-DE genes in the given samples

getCommonNonDEGCgenes <- function(sample1, sample2) {
  # join them together by GENEID column
  chr.all <- inner_join(sample1, sample2, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matched <- intersect(chr.all$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepAll <- chr.all$GENEID %in% matched
          # only keep those genes in the data frame
          chr.all <- chr.all[keepAll,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenAll <- genesGC$GENEID %in% matched
              # put those genes in a vector
              GenAll <- genesGC[keepGenAll,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chr.all <- cbind(GenAll$TXCHROM, chr.all)
                return(chr.all)
}


chrVIIGCNon <- getCommonNonDEGCgenes(nonDEGCS59c7, nonDEGCS61c7)
# write.csv(chrVIIGCNon, file="chrVIIallNonDE_GC.csv")
chrXIIGCNon <- getCommonNonDEGCgenes(nonDEGCS18c12, nonDEGCS77c12)
# write.csv(chrXIIGCNon, file="chrXIIallNonDE_GC.csv")

## only trisomic samples
    # join them together by GENEID column
    chrItriGCNon <- inner_join(nonDEGCS7c1,nonDEGCS18c1, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matchedIGCNon <- intersect(chrItriGCNon$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepIGCNon <- chrItriGCNon$GENEID %in% matchedIGCNon
          # only keep those genes in the data frame
          chrItriGCNon <- chrItriGCNon[keepIGCNon,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenGCNon <- genesGC$GENEID %in% matchedIGCNon
              # put those genes in a vector
              GenNonGC <- genesGC[keepGenGCNon,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chrItriGCNon <- cbind(GenNonGC$TXCHROM, chrItriGCNon)
                  # save as .csv file
                  # write.csv(chrItriGCNon, file="chrITriNonDE_GC.csv")
#for chromosome I, which has 3 lines that are commonly aneuploid for it, so the function won't work properly
  # chrom I, all aneuploids (both trisomics plus the monosomic)
    chrI.NonGC <- inner_join(chrItriGCNon, nonDEGCS11c1, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matchedIallGCNon <- intersect(chrI.NonGC$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepIallGCNon <- chrI.NonGC$GENEID %in% matchedIallGCNon
          # only keep those genes in the data frame
          chrI.NonGC <- chrI.NonGC[keepIallGCNon,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenAllNonGC <- genesGC$GENEID %in% matchedIallGCNon
              # put those genes in a vector
              GenAllNonGC <- genesGC[keepGenAllNonGC,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chrI.NonGC <- cbind(GenAllNonGC$TXCHROM, chrI.NonGC)
                  # save as .csv file
                  # write.csv(chrI.NonGC, file="chrIAllNonDE_GC.csv")

eupLinesGC <- list(genesGC1DE, genesGC2DE, genesGC3DE, genesGC5DE, genesGC9DE)
# want to get list of common DE genes in all the euploid lines
commonDEeupGC <- Reduce(intersect, Map("[[", eupLinesGC, "GENEID"))
#Reduc
commonDEeupGC <- as.vector(commonDEeupGC)
commonDEeupGC <- as.data.frame(commonDEeupGC)
colnames(commonDEeupGC) <- c("names")
# write.csv(commonDEeupGC, file="commonDEeup_GC.csv")

# what genes are commonly DE in aneuploid lines, but NOT located on the aneuploid chromosomes?
# genes11DE,
aneuLinesGC <- 	list(deGenesNotAneuGCS4, deGenesNotAneuGCS7, deGenesNotAneuGCS8,deGenesNotAneuGCS11, deGenesNotAneuGCS18,
                     deGenesNotAneuGCS59, deGenesNotAneuGCS61, deGenesNotAneuGCS76, deGenesNotAneuGCS77)
# want to get list of common DE genes in all the euploid lines
commonDEaneuGC <- Reduce(intersect, Map("[[", aneuLinesGC, "GENEID"))
commonDEeupGC <- as.vector(commonDEeupGC)
# write.csv(commonDEeupGC, file="commonDEeup_GC.csv")

```

```{r lfcThresh2,eval=FALSE,include=FALSE,echo=FALSE}

############## Slightly different way of testing -- different order ######
## for euploid lines
library(AnnotationDbi)
library(ggplot2)
library(dplyr)
resultsEuploid <- function(dds,sample,anc,genes) {
# make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  all(rownames(dds) == genes$GENEID)
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes,resDiSig,by=c("GENEID"="names"))
   results <- list()
  results$Sig <- resDiSig
  results$nonSig <- resDiNonSig
  return(results)
}

sample2GC <- resultsEuploid(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2)
sample2GC$Sig
# write.csv(sample2GC$Sig,file="DEgenesSample2GC.csv")

sample3GC <- resultsEuploid(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3)
sample3GC$Sig
# write.csv(sample3GC$Sig,file="DEgenesSample3GC.csv")

sample5GC <- resultsEuploid(ddsDefGC_5,"Holly_5","Holly_Anc",genesGC_5)
sample5GC$Sig
# write.csv(sample5GC$Sig,file="DEgenesSample5GC.csv")

sample9GC <- resultsEuploid(ddsDefGC_9,"Holly_9","Holly_Anc",genesGC_9)
sample9GC$Sig
# write.csv(sample9GC$Sig,file="DEgenesSample9GC.csv")

# DE genes in aneuploid lines

sample1MA <- resultsEuploid(ddsDefMAold_1,"SC001","SCA",genesMAold_1)
sample1MA$Sig
# write.csv(sample1MA$Sig,file="DEgenesSample1MA.csv")

sample2MA <- resultsEuploid(ddsDefMAold_2,"SC002","SCA",genesMAold_2)
sample2MA$Sig
# write.csv(sample2MA$Sig,file="DEgenesSample2MA.csv")

sample3MA <- resultsEuploid(ddsDefMAold_3,"SC003","SCA",genesMAold_3)
sample3MA$Sig
# write.csv(sample3MA$Sig,file="DEgenesSample3MA.csv")

sample4MA <- resultsEuploid(ddsDefMAold_4,"SC004","SCA",genesMAold_4)
sample4MA$Sig
# write.csv(sample4MA$Sig,file="DEgenesSample4MA.csv")

sample5MA <- resultsEuploid(ddsDefMAold_5,"SC005","SCA",genesMAold_5)
sample5MA$Sig
# write.csv(sample5MA$Sig,file="DEgenesSample5MA.csv")

sample7MA <- resultsEuploid(ddsDefMAold_7,"SC007","SCA",genesMAold_7)
sample7MA$Sig
# write.csv(sample7MA$Sig,file="DEgenesSample7MA.csv")

sample8MA <- resultsEuploid(ddsDefMAold_8,"SC008","SCA",genesMAold_8)
sample8MA$Sig
# write.csv(sample8MA$Sig,file="DEgenesSample8MA.csv")

sample50MA <- resultsEuploid(ddsDefMAnewS50,"50","Anc",genesMAnew_50)
sample50MA$Sig
# write.csv(sample50MA$Sig,file="DEgenesSample50MA.csv")

sample115MA <- resultsEuploid(ddsDefMAnewS115,"115","Anc",genesMAnewS115)
sample115MA$Sig
# write.csv(sample115MA$Sig,file="DEgenesSample115MA.csv")

# rm(dds)

```

```{r trisomy, include=TRUE, echo=FALSE}
resultsTrisomy <- function(dds,sample,anc,genes,anChr) {
# make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  keep <- anChr
  genes <- genes[(grep(paste0(keep,"$"),genes$TXCHROM,perl=TRUE)),]
  all(rownames(dds) == genes$GENEID)
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  resTri <- results(dds,contrast=c("condition",sample,anc),lfcThreshold=0.585)
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (NO DC))
  resTriNonSig <- as.data.frame(subset(resTri,padj>0.1))
  resTriNonSig <- cbind(names = rownames(resTriNonSig), resTriNonSig)
  resTriNonSig <- dplyr::inner_join(genes,resTriNonSig,by=c("GENEID"="names"))
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resTriSig <- as.data.frame(subset(resTri,padj<0.1))
  resTriSig <- cbind(names = rownames(resTriSig), resTriSig)
  resTriSig <- dplyr::inner_join(genes,resTriSig,by=c("GENEID"="names"))
# put non-significantly different genes in a data table
  results <- list()
  results$resNoDC <- resTriNonSig
  results$resPotDC <- resTriSig
  #### run same analyses using new ddsCopy dataset and new parameters
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(ddsCopy,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get Significant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  ### Find genes that are Signficiantly different than Trisomic expectation and are Significantly different from Disomic expectation
  resPartOvDC <- dplyr::inner_join(resTriSig,resDiSig,by="GENEID")
  results$resPartOvDC <- resPartOvDC
  ### Find genes that are Signficiantly different than Trisomic expectation but NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resTriSig,resDiNonSig,by="GENEID")
  results$resFullDC <- resFullDC
  transGenesDE <- dplyr::anti_join(resDiSig,genes,by=c("GENEID"="TXCHROM"))
  results$transGenesDE <- transGenesDE
  return(results)
}

resultsNames(ddsDefMAnewS152)
maOld152.1 <- resultsTrisomy(ddsDefMAnewS152,"152","Anc",genesMAnewS152,"chrI")
# View(maOld152.1)


maOldTest <- resultsTrisomy(ddsDefMAold_108,"SC108","SCA",genesMAold_108,"chrVIII")

# View(maOldTest$resNoDC)
# View(maOldTest$resFullDC)
# View(maOldTest$resPartOvDC)

```

```{r ESR genes, include=FALSE, eval=TRUE, echo=FALSE}
### get trans genes in aneuploid lines that are differentially expressed
# then analyze if any of them are ESR genes

dds=ddsDefMAnewS152
sample="152"
anc="Anc"
genes=genesMAnewS152

resultsESR <- function(dds,sample,anc,genes) {
# make a new dds data set that we can use in future analyses
   geneID <- rownames(dds)
   genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
# isolate known ESR genes 
  ESR <- list()
  ESR <- dplyr::inner_join(genes1,esrGenes,by=c("GENEID"="GENEID"))
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resESRNonSig <- as.data.frame(subset(resDi, padj>0.1))
  resESRNonSig <- cbind(names = rownames(resESRNonSig), resESRNonSig)
  resESRNonSig <- dplyr::inner_join(genes1,resESRNonSig,by=c("GENEID"="names"))
  # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resESR <- list()
  resESRSig <- as.data.frame(subset(resDi,padj<0.1))
  resESRSig <- cbind(names = rownames(resESRSig), resESRSig)
  resESRSig <- dplyr::inner_join(genes1,resESRSig,by=c("GENEID"="names"))
  resESR$Sig <- resESRSig
  resESR$NonSig <- resESRNonSig
  return(resESR)
}



esrGenesMA152 <- resultsESR(ddsDefMAnewS152,"152","Anc",genesMAnewS152)#,"chrI")
esrGenesMA152Sig <- esrGenesMA152$Sig
esrGenesMA152Up <- subset(esrGenesMA152Sig,log2FoldChange>1)
# View(esrGenesMA152Up)
str(esrGenesMA152Up)
esrGenesMA152Down <- subset(esrGenesMA152Sig,log2FoldChange<1)
str(esrGenesMA152Down)


# write.csv(esrGenesMA152$Sig,file="sigESRGenesMA152.csv")
# write.csv(esrGenesMA152$NonSig,file="nonSigESRGenesMA152.csv")
esrGenesGC11 <- resultsESR(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11)#,"chrI")
esrGenesGC11Sig <- esrGenesGC11$Sig
esrGenesGC11Up <- subset(esrGenesGC11Sig,log2FoldChange>1)
View(esrGenesGC11Up)
str(esrGenesGC11Up)
esrGenesGC11Down <- subset(esrGenesGC11Sig,log2FoldChange<1)
str(esrGenesGC11Down)


#### these two need to be done separately bc they have 2 aneuploid chromosomes

esrGenesGC11 <- resultsESR(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11)#,"chrI")

esrGenesGC18chrI <- resultsESR(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18)#,"chrI")

esrGenesGC7chrI <- resultsESR(ddsDefGC_7,"Holly_7","Holly_Anc",genesGC_7)#,"chrI")
View(esrGenesGC7chrI)

esrGenesGC4chrV <- resultsESR(ddsDefGC_4,"Holly_4","Holly_Anc",genesGC_4)#,"chrV")
# write.csv(esrGenesGC4chrV$Sig,file="sigESRGenesGC4.csv")
# write.csv(esrGenesGC4chrV$NonSig,file="nonSigESRGenesGC4.csv")

esrGenesGC49chrV <- resultsESR(ddsDefGC_49,"Holly_49","Holly_Anc",genesGC_49)#,"chrV")
# write.csv(esrGenesGC49chrV$Sig,file="sigESRGenesGC49.csv")
# write.csv(esrGenesGC49chrV$NonSig,file="nonSigESRGenesGC49.csv")

esrGenesGC59chrVII <- resultsESR(ddsDefGC_59,"Holly_59","Holly_Anc",genesGC_59)#,"chrVII")
# write.csv(esrGenesGC59chrVII$Sig,file="sigESRGenesGC59.csv")
# write.csv(esrGenesGC59chrVII$NonSig,file="nonSigESRGenesGC59.csv")

esrGenesGC61chrVII <- resultsESR(ddsDefGC_61,"Holly_61","Holly_Anc",genesGC_61)#,"chrVII")
# write.csv(esrGenesGC61chrVII$Sig,file="sigESRGenesGC61.csv")
# write.csv(esrGenesGC61chrVII$NonSig,file="nonSigESRGenesGC61.csv")

esrGenesGC76chrIX <- resultsESR(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76)#,"chrIX")
# write.csv(esrGenesGC76chrIX$Sig,file="sigESRGenesGC76.csv")
# write.csv(esrGenesGC76chrIX$NonSig,file="nonSigESRGenesGC76.csv")

esrGenesGC77chrXII <- resultsESR(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_76)#,"chrIX")
# write.csv(esrGenesGC77chrXII$Sig,file="sigESRGenesGC77.csv")
# write.csv(esrGenesGC77chrXII$NonSig,file="nonSigESRGenesGC77.csv")

esrGenesGC8chrXVI <- resultsESR(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8)#,"chrXVI")
# write.csv(esrGenesGC8chrXVI$Sig,file="sigESRGenesGC8.csv")
# write.csv(esrGenesGC8chrXVI$NonSig,file="nonSigESRGenesGC8.csv")

upSetESRHetAneu <- list(GCS11=esrGenesGC11$Sig$GENEID,
                 GCS18=esrGenesGC18chrI$Sig$GENEID,
                 GCS7=esrGenesGC7chrI$Sig$GENEID,
                 GCS4=esrGenesGC4chrV$Sig$GENEID,
                 GCS49=esrGenesGC49chrV$Sig$GENEID,
                 GCS59=esrGenesGC59chrVII$Sig$GENEID,
                 GCS61=esrGenesGC61chrVII$Sig$GENEID,
                 GCS76=esrGenesGC76chrIX$Sig$GENEID,
                 GCS77=esrGenesGC77chrXII$Sig$GENEID,
                 GCS8=esrGenesGC8chrXVI$Sig$GENEID
)

upset(fromList(upSetESRHetAneu),nsets=10,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "DE ESR Genes")

####################### MA samples Aneuploids

esrGenesMA9chrXIV <- resultsESR(ddsDefMAold_9,"SC009","SCA",genesMAoldS9)#,"chrXIV")

esrGenesMA11 <- resultsESR(ddsDefMAold_11,"SC011","SCA",genesMAoldS11)#,"chrXIV")

esrGenesMA112chrXVI <- resultsESR(ddsDefMAnewS112,"112","Anc",genesMAnewS112)#,"chrXVI")
# write.csv(esrGenesMA112chrXVI$Sig,file="sigESRGenesMA112.csv")
# write.csv(esrGenesMA112chrXVI$NonSig,file="nonSigESRGenesMA112.csv")

esrGenesMA29chrIX <- resultsESR(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
# write.csv(esrGenesMA29chrIX$Sig,file="sigESRGenesMA29.csv")
# write.csv(esrGenesMA29chrIX$NonSig,file="nonSigESRGenesMA29.csv")

esrGenesMA119chrIX <- resultsESR(ddsDefMAold_119,"SC119","SCA",genesMAoldS119)#,"chrIX")
# write.csv(esrGenesMA119chrIX$Sig,file="sigESRGenesMA119.csv")
# write.csv(esrGenesMA119chrIX$NonSig,file="nonSigESRGenesMA119.csv")

esrGenesMA88chrIX <- resultsESR(ddsDefMAold_88,"SC088","SCA",genesMAoldS88)#,"chrIX")
# write.csv(esrGenesMA88chrIX$Sig,file="sigESRGenesMA88.csv")
# write.csv(esrGenesMA88chrIX$NonSig,file="nonSigESRGenesMA88.csv")

esrGenesMA108chrVIII <- resultsESR(ddsDefMAold_108,"SC108","SCA",genesMAoldS108)#,"chrIX")
# write.csv(esrGenesMA108chrVIII$Sig,file="sigESRGenesMA108.csv")
# write.csv(esrGenesMA108chrVIII$NonSig,file="nonSigESRGenesMA108.csv")

esrGenesMA15chrIX <- resultsESR(ddsDefMAold_15,"SC015","SCA",genesMAoldS15)#,"chrIX")
# write.csv(esrGenesMA15chrIX$Sig,file="sigESRGenesMA15.csv")
# write.csv(esrGenesMA15chrIX$NonSig,file="nonSigESRGenesMA15.csv")

esrGenesMA29chrIX <- resultsESR(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
# write.csv(esrGenesMA29chrIX$Sig,file="sigESRGenesMA29.csv")
# write.csv(esrGenesMA29chrIX$NonSig,file="nonSigESRGenesMA29.csv")

esrGenesMA117chrV <- resultsESR(ddsDefMAnewS117,"117","Anc",genesMAnewS117)#,"chrV")
# write.csv(esrGenesMA117chrV$Sig,file="sigESRGenesMA117.csv")
# write.csv(esrGenesMA117chrV$NonSig,file="nonSigESRGenesMA117.csv")

esrGenesMA123chrV <- resultsESR(ddsDefMAnewS123,"123","Anc",genesMAnewS123)#,"chrV")
# write.csv(esrGenesMA123chrV$Sig,file="sigESRGenesMA123.csv")
# write.csv(esrGenesMA123chrV$NonSig,file="nonSigESRGenesMA123.csv")
esrGenesMA152 <- resultsESR(ddsDefMAnewS152,"152","Anc",genesMAnewS152)#,"chrI")

upSetESRHomAneu <- list(MAS9=esrGenesMA9chrXIV$Sig$GENEID,
                 MAS112=esrGenesMA112chrXVI$Sig$GENEID,
                 MAS29=esrGenesMA29chrIX$Sig$GENEID,
                 MAS119=esrGenesMA119chrIX$Sig$GENEID,
                 MAS88=esrGenesMA88chrIX$Sig$GENEID,
                 MAS108=esrGenesMA108chrVIII$Sig$GENEID,
                 MAS15=esrGenesMA15chrIX$Sig$GENEID,
                 MAS117=esrGenesMA117chrV$Sig$GENEID,
                 MAS123=esrGenesMA123chrV$Sig$GENEID,
                 MAS152=esrGenesMA152$Sig$GENEID
)

upset(fromList(upSetESRHomAneu),nsets=10,keep.order = F,matrix.color = "lightslateblue",main.bar.color = "deeppink1",sets.bar.color = "turquoise1",order.by = "freq",sets.x.label = "DE ESR Genes")

############################################################################################################
### need to do the euploid samples next - ESR genes
esrGenesGC2 <- resultsESR(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2)
# write.csv(esrGenesGC2$Sig,file="sigESRGenesGC2.csv")
# write.csv(esrGenesGC2$NonSig,file="nonSigESRGenesGC2.csv")

esrGenesGC3 <- resultsESR(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3)
# write.csv(esrGenesGC3$Sig,file="sigESRGenesGC3.csv")
# write.csv(esrGenesGC3$NonSig,file="nonSigESRGenesGC3.csv")

esrGenesGC5 <- resultsESR(ddsDefGC_5,"Holly_5","Holly_Anc",genesGC_5)
# write.csv(esrGenesGC5$Sig,file="sigESRGenesGC5.csv")
# write.csv(esrGenesGC5$NonSig,file="nonSigESRGenesGC5.csv")

esrGenesGC9 <- resultsESR(ddsDefGC_9,"Holly_9","Holly_Anc",genesGC_9)
# write.csv(esrGenesGC9$Sig,file="sigESRGenesGC9.csv")
# write.csv(esrGenesGC9$NonSig,file="nonSigESRGenesGC9.csv")

upSetESRHetEu <- list(GCS2=esrGenesGC2$Sig$GENEID,
                 GCS3=esrGenesGC3$Sig$GENEID,
                 GCS5=esrGenesGC5$Sig$GENEID,
                 GCS9=esrGenesGC9$Sig$GENEID
              
)

upset(fromList(upSetESRHetEu),nsets=4,keep.order = F,matrix.color = "lightslateblue",main.bar.color = "deeppink1",sets.bar.color = "turquoise1",order.by = "freq",sets.x.label = "DE ESR Genes")

esrGenesMA1 <- resultsESR(ddsDefMAold_1,"SC001","SCA",genesMAold_1)
# write.csv(esrGenesMA1$Sig,file="sigESRGenesMA1.csv")
# write.csv(esrGenesMA1$NonSig,file="nonSigESRGenesMA1.csv")

esrGenesMA2 <- resultsESR(ddsDefMAold_2,"SC002","SCA",genesMAold_2)
# write.csv(esrGenesMA2$Sig,file="sigESRGenesMA2.csv")
# write.csv(esrGenesMA2$NonSig,file="nonSigESRGenesMA2.csv")

esrGenesMA3 <- resultsESR(ddsDefMAold_3,"SC003","SCA",genesMAold_3)
# write.csv(esrGenesMA3$Sig,file="sigESRGenesMA3.csv")
# write.csv(esrGenesMA3$NonSig,file="nonSigESRGenesMA3.csv")

esrGenesMA4 <- resultsESR(ddsDefMAold_4,"SC004","SCA",genesMAold_4)
# write.csv(esrGenesMA4$Sig,file="sigESRGenesMA4.csv")
# write.csv(esrGenesMA4$NonSig,file="nonSigESRGenesMA4.csv")

esrGenesMA5 <- resultsESR(ddsDefMAold_5,"SC005","SCA",genesMAold_5)
# write.csv(esrGenesMA5$Sig,file="sigESRGenesMA5.csv")
# write.csv(esrGenesMA5$NonSig,file="nonSigESRGenesMA5.csv")

esrGenesMA7 <- resultsESR(ddsDefMAold_7,"SC007","SCA",genesMAold_7)
# write.csv(esrGenesMA7$Sig,file="sigESRGenesMA7.csv")
# write.csv(esrGenesMA7$NonSig,file="nonSigESRGenesMA7.csv")

esrGenesMA8 <- resultsESR(ddsDefMAold_8,"SC008","SCA",genesMAold_8)
# write.csv(esrGenesMA8$Sig,file="sigESRGenesMA8.csv")
# write.csv(esrGenesMA8$NonSig,file="nonSigESRGenesMA8.csv")

esrGenesMA50 <- resultsESR(ddsDefMAnewS50,"50","Anc",genesMAnewS50)
# write.csv(esrGenesMA50$Sig,file="sigESRGenesMA50.csv")
# write.csv(esrGenesMA50$NonSig,file="nonSigESRGenesMA50.csv")

esrGenesMA115 <- resultsESR(ddsDefMAnewS115,"115","Anc",genesMAnewS115)
# write.csv(esrGenesMA115$Sig,file="sigESRGenesMA115.csv")
# write.csv(esrGenesMA115$NonSig,file="nonSigESRGenesMA115.csv")

upSetESRHomEu <- list(MAS1=esrGenesMA1$Sig$GENEID,
                 MAS2=esrGenesMA2$Sig$GENEID,
                 MAS3=esrGenesMA3$Sig$GENEID,
                 MAS4=esrGenesMA4$Sig$GENEID,
                 MAS5=esrGenesMA5$Sig$GENEID,
                 MAS7=esrGenesMA7$Sig$GENEID,
                 MAS8=esrGenesMA8$Sig$GENEID,
                 MAS50=esrGenesMA50$Sig$GENEID,
                 MAS115=esrGenesMA115$Sig$GENEID
              
)

upset(fromList(upSetESRHomEu),nsets=9,keep.order = F,matrix.color = "lightslateblue",main.bar.color = "deeppink1",sets.bar.color = "turquoise1",order.by = "freq",sets.x.label = "DE ESR Genes")










### see if they are in ASR
resultsASR <- function(dds,sample,anc,genes) {
# make a new dds data set that we can use in future analyses
   geneID <- rownames(dds)
   genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
# determine if any significantly differentially expressed genes match those identified by gasch et al 2000
  ASR <- list()
  ASR$Sig <- dplyr::inner_join(ASRgenes,resDiSig,by=c("GENEID"="GENEID"))
  ASR$NonSig <- dplyr::inner_join(ASRgenes,resDiNonSig,by=c("GENEID"="GENEID"))
  return(ASR)
}

ASRGenesMA152 <- resultsASR(ddsDefMAnewS152,"152","Anc",genesMAnewS152)#,"chrI")
# write.csv(ASRGenesMA152$Sig,file="sigASRGenesMA152.csv")
# write.csv(ASRGenesMA152$NonSig,file="nonSigASRGenesMA152.csv")
ASRGenesGC11 <- resultsASR(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11)#,"chrI")
asrGenesSigGC11 <- ASRGenesGC11$Sig
# write.csv(ASRGenesGC11$Sig,file="sigASRGenesGC11.csv")
# write.csv(ASRGenesGC11$NonSig,file="nonSigASRGenesGC11.csv")

ASRGenesGC18chrI <- resultsASR(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18)#,"chrI")
asrGenesSigGC18 <- ASRGenesGC18chrI$Sig
# write.csv(ASRGenesGC18chrI$Sig,file="sigASRGenesGC18.csv")
# write.csv(ASRGenesGC18chrI$NonSig,file="nonSigASRGenesGC18.csv")

ASRGenesGC7chrI <- resultsASR(ddsDefGC_7,"Holly_7","Holly_Anc",genesGC_7)#,"chrI")
asrGenesSigGC7 <- ASRGenesGC7chrI$Sig
# write.csv(ASRGenesGC7chrI$Sig,file="sigASRGenesGC7.csv")
# write.csv(ASRGenesGC7chrI$NonSig,file="nonSigASRGenesGC7.csv")

ASRGenesMA123chrV <- resultsASR(ddsDefMAnewS123,"123","Anc",genesMAnewS123)#,"chrV")
asrGenesSigMA123 <- ASRGenesMA123chrV$Sig
# write.csv(ASRGenesMA123chrV$Sig,file="sigASRGenesMA123.csv")
# write.csv(ASRGenesMA123chrV$NonSig,file="nonSigASRGenesMA123.csv")

ASRGenesGC4chrV <- resultsASR(ddsDefGC_4,"Holly_4","Holly_Anc",genesGC_4)#,"chrV")
asrGenesSigGC4 <- ASRGenesGC4chrV$Sig
# write.csv(ASRGenesGC4chrV$Sig,file="sigASRGenesGC4.csv")
# write.csv(ASRGenesGC4chrV$NonSig,file="nonSigASRGenesGC4.csv")

ASRGenesGC49chrV <- resultsASR(ddsDefGC_49,"Holly_49","Holly_Anc",genesGC_49)#,"chrV")
asrGenesSigGC49 <- ASRGenesGC49chrV$Sig
# write.csv(ASRGenesGC49chrV$Sig,file="sigASRGenesGC49.csv")
# write.csv(ASRGenesGC49chrV$NonSig,file="nonSigASRGenesGC49.csv")

ASRGenesMA117chrV <- resultsASR(ddsDefMAnewS117,"117","Anc",genesMAnewS117)#,"chrV")
asrGenesSigMA117 <- ASRGenesMA117chrV$Sig
# write.csv(ASRGenesMA117chrV$Sig,file="sigASRGenesMA117.csv")
# write.csv(ASRGenesMA117chrV$NonSig,file="nonSigASRGenesMA117.csv")

ASRGenesGC59chrVII <- resultsASR(ddsDefGC_59,"Holly_59","Holly_Anc",genesGC_59)#,"chrVII")
asrGenesSigGC59 <- ASRGenesGC59chrVII$Sig
# write.csv(ASRGenesGC59chrVII$Sig,file="sigASRGenesGC59.csv")
# write.csv(ASRGenesGC59chrVII$NonSig,file="nonSigASRGenesGC59.csv")

ASRGenesGC61chrVII <- resultsASR(ddsDefGC_61,"Holly_61","Holly_Anc",genesGC_61)#,"chrVII")
asrGenesSigGC61 <- ASRGenesGC61chrVII$Sig
# write.csv(ASRGenesGC61chrVII$Sig,file="sigASRGenesGC61.csv")
# write.csv(ASRGenesGC61chrVII$NonSig,file="nonSigASRGenesGC61.csv")

ASRGenesMA29chrIX <- resultsASR(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
asrGenesSigMA29 <- ASRGenesMA29chrIX$Sig
# write.csv(ASRGenesMA29chrIX$Sig,file="sigASRGenesMA29.csv")
# write.csv(ASRGenesMA29chrIX$NonSig,file="nonSigASRGenesMA29.csv")

ASRGenesMA119chrIX <- resultsASR(ddsDefMAold_119,"SC119","SCA",genesMAoldS119)#,"chrIX")
asrGenesSigMA119 <- ASRGenesMA119chrIX$Sig
# write.csv(ASRGenesMA119chrIX$Sig,file="sigASRGenesMA119.csv")
# write.csv(ASRGenesMA119chrIX$NonSig,file="nonSigASRGenesMA119.csv")

ASRGenesMA88chrIX <- resultsASR(ddsDefMAold_88,"SC088","SCA",genesMAoldS88)#,"chrIX")
asrGenesSigMA88 <- ASRGenesMA88chrIX$Sig
# write.csv(ASRGenesMA88chrIX$Sig,file="sigASRGenesMA88.csv")
# write.csv(ASRGenesMA88chrIX$NonSig,file="nonSigASRGenesMA88.csv")

ASRGenesMA108chrVIII <- resultsASR(ddsDefMAold_108,"SC108","SCA",genesMAoldS108)#,"chrIX")
asrGenesSigMA108 <- ASRGenesMA108chrVIII$Sig
# write.csv(ASRGenesMA108chrVIII$Sig,file="sigASRGenesMA108.csv")
# write.csv(ASRGenesMA108chrVIII$NonSig,file="nonSigASRGenesMA108.csv")


ASRGenesMA15chrIX <- resultsASR(ddsDefMAold_15,"SC015","SCA",genesMAoldS15)#,"chrIX")
asrGenesSigMA15 <- ASRGenesMA15chrIX$Sig
# write.csv(ASRGenesMA15chrIX$Sig,file="sigASRGenesMA15.csv")
# write.csv(ASRGenesMA15chrIX$NonSig,file="nonSigASRGenesMA15.csv")

ASRGenesMA29chrIX <- resultsASR(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
asrGenesSigMA29 <- ASRGenesMA29chrIX$Sig
# write.csv(ASRGenesMA29chrIX$Sig,file="sigASRGenesMA29.csv")
# write.csv(ASRGenesMA29chrIX$NonSig,file="nonSigASRGenesMA29.csv")

ASRGenesGC76chrIX <- resultsASR(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76)#,"chrIX")
asrGenesSigGC76 <- ASRGenesGC76chrIX$Sig
# write.csv(ASRGenesGC76chrIX$Sig,file="sigASRGenesGC76.csv")
# write.csv(ASRGenesGC76chrIX$NonSig,file="nonSigASRGenesGC76.csv")

ASRGenesGC77chrXII <- resultsASR(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_76)#,"chrIX")
asrGenesSigGC77 <- ASRGenesGC77chrXII$Sig
# write.csv(ASRGenesGC77chrXII$Sig,file="sigASRGenesGC77.csv")
# write.csv(ASRGenesGC77chrXII$NonSig,file="nonSigASRGenesGC77.csv")

ASRGenesMA9chrXIV <- resultsASR(ddsDefMAold_9,"SC009","SCA",genesMAoldS9)#,"chrXIV")
asrGenesSigMA9 <- ASRGenesMA9chrXIV$Sig
# write.csv(ASRGenesMA9chrXIV$Sig,file="sigASRGenesMA9.csv")
# write.csv(ASRGenesMA9chrXIV$NonSig,file="nonSigASRGenesMA9.csv")

ASRGenesMA11 <- resultsASR(ddsDefMAold_11,"SC011","SCA",genesMAoldS11)#,"chrXIV")
asrGenesSigMA11 <- ASRGenesMA11$Sig
# write.csv(ASRGenesMA11$Sig,file="sigASRGenesMA11.csv")
# write.csv(ASRGenesMA11$NonSig,file="nonSigASRGenesMA11.csv")

ASRGenesMA112chrXVI <- resultsASR(ddsDefMAnewS112,"112","Anc",genesMAnewS112)#,"chrXVI")
asrGenesSigMA112 <- ASRGenesMA112chrXVI$Sig
# write.csv(ASRGenesMA112chrXVI$Sig,file="sigASRGenesMA112.csv")
# write.csv(ASRGenesMA112chrXVI$NonSig,file="nonSigASRGenesMA112.csv")

ASRGenesGC8chrXVI <- resultsASR(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8)#,"chrXVI")
asrGenesSigGC8 <- ASRGenesGC8chrXVI$Sig
# write.csv(ASRGenesGC8chrXVI$Sig,file="sigASRGenesGC8.csv")
# write.csv(ASRGenesGC8chrXVI$NonSig,file="nonSigASRGenesGC8.csv")



upSetASRHetAneu <- list(GCS11=ASRGenesGC11$Sig$GENEID,
                 GCS18=ASRGenesGC18chrI$Sig$GENEID,
                 GCS7=ASRGenesGC7chrI$Sig$GENEID,
                 GCS4=ASRGenesGC4chrV$Sig$GENEID,
                 GCS49=ASRGenesGC49chrV$Sig$GENEID,
                 GCS59=ASRGenesGC59chrVII$Sig$GENEID,
                 GCS61=ASRGenesGC61chrVII$Sig$GENEID,
                 GCS76=ASRGenesGC76chrIX$Sig$GENEID,
                 GCS77=ASRGenesGC77chrXII$Sig$GENEID,
                 GCS8=ASRGenesGC8chrXVI$Sig$GENEID
)

upset(fromList(upSetASRHetAneu),nsets=10,keep.order = F,matrix.color = "blue",main.bar.color = "turquoise",sets.bar.color = "purple",order.by = "freq",sets.x.label = "DE ASR Genes")


upSetASRHomAneu <- list(MAS9=ASRGenesMA9chrXIV$Sig$GENEID,
                 MAS112=ASRGenesMA112chrXVI$Sig$GENEID,
                 MAS29=ASRGenesMA29chrIX$Sig$GENEID,
                 MAS119=ASRGenesMA119chrIX$Sig$GENEID,
                 MAS88=ASRGenesMA88chrIX$Sig$GENEID,
                 MAS108=ASRGenesMA108chrVIII$Sig$GENEID,
                 MAS15=ASRGenesMA15chrIX$Sig$GENEID,
                 MAS117=ASRGenesMA117chrV$Sig$GENEID,
                 MAS123=ASRGenesMA123chrV$Sig$GENEID,
                 MAS152=ASRGenesMA152$Sig$GENEID
)

upset(fromList(upSetASRHomAneu),nsets=10,keep.order = F,matrix.color = "blue",main.bar.color = "turquoise",sets.bar.color = "purple",order.by = "freq",sets.x.label = "DE ASR Genes")



### need to do the euploid samples next - ASR genes

ASRGenesGC1 <- resultsASR(ddsDefGC_1,"Holly_1","Holly_Anc",genesGC_1)
asrGenesSigGC1 <- ASRGenesGC1$Sig
# write.csv(ASRGenesGC1$Sig,file="sigASRGenesGC1.csv")
# write.csv(ASRGenesGC1$NonSig,file="nonSigASRGenesGC1.csv")

ASRGenesGC2 <- resultsASR(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2)
asrGenesSigGC2 <- ASRGenesGC2$Sig
# write.csv(ASRGenesGC2$Sig,file="sigASRGenesGC2.csv")
# write.csv(ASRGenesGC2$NonSig,file="nonSigASRGenesGC2.csv")

ASRGenesGC3 <- resultsASR(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3)
asrGenesSigGC3 <- ASRGenesGC3$Sig
# write.csv(ASRGenesGC3$Sig,file="sigASRGenesGC3.csv")
# write.csv(ASRGenesGC3$NonSig,file="nonSigASRGenesGC3.csv")

ASRGenesGC5 <- resultsASR(ddsDefGC_5,"Holly_5","Holly_Anc",genesGC_5)
asrGenesSigGC5 <- ASRGenesGC5$Sig
# write.csv(ASRGenesGC5$Sig,file="sigASRGenesGC5.csv")
# write.csv(ASRGenesGC5$NonSig,file="nonSigASRGenesGC5.csv")

ASRGenesGC9 <- resultsASR(ddsDefGC_9,"Holly_9","Holly_Anc",genesGC_9)
asrGenesSigGC9 <- ASRGenesGC9$Sig
# write.csv(ASRGenesGC9$Sig,file="sigASRGenesGC9.csv")
# write.csv(ASRGenesGC9$NonSig,file="nonSigASRGenesGC9.csv")

ASRGenesMA1 <- resultsASR(ddsDefMAold_1,"SC001","SCA",genesMAold_1)
asrGenesSigMA1 <- ASRGenesMA1$Sig
# write.csv(ASRGenesMA1$Sig,file="sigASRGenesMA1.csv")
# write.csv(ASRGenesMA1$NonSig,file="nonSigASRGenesMA1.csv")

ASRGenesMA2 <- resultsASR(ddsDefMAold_2,"SC002","SCA",genesMAold_2)
asrGenesSigMA2 <- ASRGenesMA2$Sig
# write.csv(ASRGenesMA2$Sig,file="sigASRGenesMA2.csv")
# write.csv(ASRGenesMA2$NonSig,file="nonSigASRGenesMA2.csv")

ASRGenesMA3 <- resultsASR(ddsDefMAold_3,"SC003","SCA",genesMAold_3)
asrGenesSigMA3 <- ASRGenesMA3$Sig
# write.csv(ASRGenesMA3$Sig,file="sigASRGenesMA3.csv")
# write.csv(ASRGenesMA3$NonSig,file="nonSigASRGenesMA3.csv")

ASRGenesMA4 <- resultsASR(ddsDefMAold_4,"SC004","SCA",genesMAold_4)
asrGenesSigMA4 <- ASRGenesMA4$Sig
# write.csv(ASRGenesMA4$Sig,file="sigASRGenesMA4.csv")
# write.csv(ASRGenesMA4$NonSig,file="nonSigASRGenesMA4.csv")

ASRGenesMA5 <- resultsASR(ddsDefMAold_5,"SC005","SCA",genesMAold_5)
asrGenesSigMA5 <- ASRGenesMA5$Sig
# write.csv(ASRGenesMA5$Sig,file="sigASRGenesMA5.csv")
# write.csv(ASRGenesMA5$NonSig,file="nonSigASRGenesMA5.csv")

ASRGenesMA7 <- resultsASR(ddsDefMAold_7,"SC007","SCA",genesMAold_7)
asrGenesSigMA7 <- ASRGenesMA7$Sig
# write.csv(ASRGenesMA7$Sig,file="sigASRGenesMA7.csv")
# write.csv(ASRGenesMA7$NonSig,file="nonSigASRGenesMA7.csv")

ASRGenesMA8 <- resultsASR(ddsDefMAold_8,"SC008","SCA",genesMAold_8)
asrGenesSigMA8 <- ASRGenesMA8$Sig
# write.csv(ASRGenesMA8$Sig,file="sigASRGenesMA8.csv")
# write.csv(ASRGenesMA8$NonSig,file="nonSigASRGenesMA8.csv")

ASRGenesMA50 <- resultsASR(ddsDefMAnewS50,"50","Anc",genesMAnewS50)
asrGenesSigMA50 <- ASRGenesMA50$Sig
# write.csv(ASRGenesMA50$Sig,file="sigASRGenesMA50.csv")
# write.csv(ASRGenesMA50$NonSig,file="nonSigASRGenesMA50.csv")

ASRGenesMA115 <- resultsASR(ddsDefMAnewS115,"115","Anc",genesMAnewS115)
asrGenesSigMA115 <- ASRGenesMA115$Sig
# write.csv(ASRGenesMA115$Sig,file="sigASRGenesMA115.csv")
# write.csv(ASRGenesMA115$NonSig,file="nonSigASRGenesMA115.csv")

upSetASRHetEu <- list(GCS2=ASRGenesGC2$Sig$GENEID,
                 GCS3=ASRGenesGC3$Sig$GENEID,
                 GCS5=ASRGenesGC5$Sig$GENEID,
                 GCS9=ASRGenesGC9$Sig$GENEID
              
)

upset(fromList(upSetASRHetEu),nsets=4,keep.order = F,matrix.color = "blue",main.bar.color = "turquoise",sets.bar.color = "purple",order.by = "freq",sets.x.label = "DE ASR Genes")


upSetASRHomEu <- list(MAS1=ASRGenesMA1$Sig$GENEID,
                 MAS2=ASRGenesMA2$Sig$GENEID,
                 MAS3=ASRGenesMA3$Sig$GENEID,
                 MAS4=ASRGenesMA4$Sig$GENEID,
                 MAS5=ASRGenesMA5$Sig$GENEID,
                 MAS7=ASRGenesMA7$Sig$GENEID,
                 MAS8=ASRGenesMA8$Sig$GENEID,
                 MAS50=ASRGenesMA50$Sig$GENEID,
                 MAS115=ASRGenesMA115$Sig$GENEID
              
)

upset(fromList(upSetASRHomEu),nsets=9,keep.order = F,matrix.color = "blue",main.bar.color = "turquoise",sets.bar.color = "purple",order.by = "freq",sets.x.label = "DE ASR Genes")

resultsDosSens <- function(dds,sample,anc,genes) {
# make a new dds data set that we can use in future analyses
   geneID <- rownames(dds)
   genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
# subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
# determine if any significantly differentially expressed genes match those identified by gasch et al 2000
  dosSensRes <- list()
  dosSensRes$Sig <- dplyr::inner_join(dosageSensitiveGenes,resDiSig,by=c("GENEID"="GENEID"))
  dosSensRes$NonSig <- dplyr::inner_join(dosageSensitiveGenes,resDiNonSig,by=c("GENEID"="GENEID"))
  return(dosSensRes)
}


dosSensGenesMA152 <- resultsDosSens(ddsDefMAnewS152,"152","Anc",genesMAnewS152)#,"chrI")
dosSensGenesSigMA152 <- dosSensGenesMA152$Sig
dosSensGenesNonSigMA152 <- dosSensGenesMA152$NonSig

# write.csv(dosSensGenesMA152$Sig,file="sigEdosSensGenesMA152.csv")
# write.csv(dosSensGenesMA152$NonSig,file="nonSigdosSensGenesMA152.csv")

dosSensGenesGC11 <- resultsDosSens(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11)#,"chrI")
dosSensGenesSigGC11 <- dosSensGenesGC11$Sig
# write.csv(dosSensGenesGC11$Sig,file="sigdosSensGenesGC11.csv")
# write.csv(dosSensGenesGC11$NonSig,file="nondosSensGenesGC11.csv")

dosSensGenesGC18chrI <- resultsDosSens(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18)#,"chrI")
dosSensGenesSigGC18 <- dosSensGenesGC18chrI$Sig
# write.csv(dosSensGenesGC18chrI$Sig,file="sigdosSensGenesGC18.csv")
# write.csv(dosSensGenesGC18chrI$NonSig,file="nonSigdosSensGenesGC18.csv")

dosSensGenesGC7chrI <- resultsDosSens(ddsDefGC_7,"Holly_7","Holly_Anc",genesGC_7)#,"chrI")
dosSensGenesSigGC7 <- dosSensGenesGC7chrI$Sig
# write.csv(dosSensGenesGC7chrI$Sig,file="sigdosSensGenesGC7.csv")
# write.csv(dosSensGenesGC7chrI$NonSig,file="nonSigdosSensGenesGC7.csv")

dosSensGenesMA123chrV <- resultsDosSens(ddsDefMAnewS123,"123","Anc",genesMAnewS123)#,"chrV")
dosSensGenesSigMA123 <- dosSensGenesMA123chrV$Sig
# write.csv(dosSensGenesMA123chrV$Sig,file="sigdosSensGenesMA123.csv")
# write.csv(dosSensGenesMA123chrV$NonSig,file="nonSigdosSensGenesMA123.csv")

dosSensGenesGC4chrV <- resultsDosSens(ddsDefGC_4,"Holly_4","Holly_Anc",genesGC_4)#,"chrV")
dosSensGenesSigGC4 <- dosSensGenesGC4chrV$Sig
# write.csv(dosSensGenesGC4chrV$Sig,file="sigdosSensGenesGC4.csv")
# write.csv(dosSensGenesGC4chrV$NonSig,file="nonSigdosSensGenesGC4.csv")

dosSensGenesGC49chrV <- resultsDosSens(ddsDefGC_49,"Holly_49","Holly_Anc",genesGC_49)#,"chrV")
dosSensGenesSigGC49 <- dosSensGenesGC49chrV$Sig
# write.csv(dosSensGenesGC49chrV$Sig,file="sigdosSensGenesGC49.csv")
# write.csv(dosSensGenesGC49chrV$NonSig,file="nonSigdosSensGenesGC49.csv")

dosSensGenesMA117chrV <- resultsDosSens(ddsDefMAnewS117,"117","Anc",genesMAnewS117)#,"chrV")
dosSensGenesSigMA117 <- dosSensGenesMA117chrV$Sig
# write.csv(dosSensGenesMA117chrV$Sig,file="sigdosSensGenesMA117.csv")
# write.csv(dosSensGenesMA117chrV$NonSig,file="nonSigdosSensGenesMA117.csv")

dosSensGenesGC59chrVII <- resultsDosSens(ddsDefGC_59,"Holly_59","Holly_Anc",genesGC_59)#,"chrVII")
dosSensGenesSigGC59 <- dosSensGenesGC59chrVII$Sig
# write.csv(dosSensGenesGC59chrVII$Sig,file="sigdosSensGenesGC59.csv")
# write.csv(dosSensGenesGC59chrVII$NonSig,file="nonSigdosSensGenesGC59.csv")

dosSensGenesGC61chrVII <- resultsDosSens(ddsDefGC_61,"Holly_61","Holly_Anc",genesGC_61)#,"chrVII")
dosSensGenesSigGC61 <- dosSensGenesGC61chrVII$Sig
# write.csv(dosSensGenesGC61chrVII$Sig,file="sigdosSensGenesGC61.csv")
# write.csv(dosSensGenesGC61chrVII$NonSig,file="nonSigdosSensGenesGC61.csv")

dosSensGenesMA29chrIX <- resultsDosSens(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
dosSensGenesSigMA29 <- dosSensGenesMA29chrIX$Sig
# write.csv(dosSensGenesMA29chrIX$Sig,file="sigdosSensGenesMA29.csv")
# write.csv(dosSensGenesMA29chrIX$NonSig,file="nonSigdosSensGenesMA29.csv")

dosSensGenesMA119chrIX <- resultsDosSens(ddsDefMAold_119,"SC119","SCA",genesMAoldS119)#,"chrIX")
dosSensGenesSigMA119 <- dosSensGenesMA119chrIX$Sig
# write.csv(dosSensGenesMA119chrIX$Sig,file="sigdosSensGenesMA119.csv")
# write.csv(dosSensGenesMA119chrIX$NonSig,file="nonSigdosSensGenesMA119.csv")

dosSensGenesMA88chrIX <- resultsDosSens(ddsDefMAold_88,"SC088","SCA",genesMAoldS88)#,"chrIX")
dosSensGenesSigMA88 <- dosSensGenesMA88chrIX$Sig
# write.csv(dosSensGenesMA88chrIX$Sig,file="sigdosSensGenesMA88.csv")
# write.csv(dosSensGenesMA88chrIX$NonSig,file="nonSigdosSensGenesMA88.csv")

dosSensGenesMA108chrVIII <- resultsDosSens(ddsDefMAold_108,"SC108","SCA",genesMAoldS108)#,"chrIX")
dosSensGenesSigMA108 <- dosSensGenesMA108chrVIII$Sig
# write.csv(dosSensGenesMA108chrVIII$Sig,file="sigdosSensGenesMA108.csv")
# write.csv(dosSensGenesMA108chrVIII$NonSig,file="nonSigdosSensGenesMA108.csv")

dosSensGenesMA15chrIX <- resultsDosSens(ddsDefMAold_15,"SC015","SCA",genesMAoldS15)#,"chrIX")
dosSensGenesSigMA15 <- dosSensGenesMA15chrIX$Sig
# write.csv(dosSensGenesMA15chrIX$Sig,file="sigdosSensGenesMA15.csv")
# write.csv(dosSensGenesMA15chrIX$NonSig,file="nonSigdosSensGenesMA15.csv")

dosSensGenesMA29chrIX <- resultsDosSens(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
dosSensGenesSigMA29 <- dosSensGenesMA29chrIX$Sig
# write.csv(dosSensGenesMA29chrIX$Sig,file="sigdosSensGenesMA29.csv")
# write.csv(dosSensGenesMA29chrIX$NonSig,file="nonSigdosSensGenesMA29.csv")

dosSensGenesGC76chrIX <- resultsDosSens(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76)#,"chrIX")
dosSensGenesSigGC76 <- dosSensGenesGC76chrIX$Sig
# write.csv(dosSensGenesGC76chrIX$Sig,file="sigdosSensGenesGC76.csv")
# write.csv(dosSensGenesGC76chrIX$NonSig,file="nonSigdosSensGenesGC76.csv")

dosSensGenesGC77chrXII <- resultsDosSens(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_76)#,"chrIX")
dosSensGenesSigGC77 <- dosSensGenesGC77chrXII$Sig
# write.csv(dosSensGenesGC77chrXII$Sig,file="sigdosSensGenesGC77.csv")
# write.csv(dosSensGenesGC77chrXII$NonSig,file="nonSigdosSensGenesGC77.csv")

dosSensGenesMA9chrXIV <- resultsDosSens(ddsDefMAold_9,"SC009","SCA",genesMAoldS9)#,"chrXIV")
dosSensGenesSigMA9 <- dosSensGenesMA9chrXIV$Sig
# write.csv(dosSensGenesMA9chrXIV$Sig,file="sigdosSensGenesMA9.csv")
# write.csv(dosSensGenesMA9chrXIV$NonSig,file="nonSigdosSensGenesMA9.csv")

dosSensGenesMA11 <- resultsDosSens(ddsDefMAold_11,"SC011","SCA",genesMAoldS11)#,"chrXIV")
dosSensGenesSigMA11 <- dosSensGenesMA11$Sig
# write.csv(dosSensGenesMA11$Sig,file="sigdosSensGenesMA11.csv")
# write.csv(dosSensGenesMA11$NonSig,file="nonSigdosSensGenesMA11.csv")

dosSensGenesMA112chrXVI <- resultsDosSens(ddsDefMAnewS112,"112","Anc",genesMAnewS112)#,"chrXVI")
dosSensGenesSigMA112 <- dosSensGenesMA112chrXVI$Sig
# write.csv(dosSensGenesMA112chrXVI$Sig,file="sigdosSensGenesMA112.csv")
# write.csv(dosSensGenesMA112chrXVI$NonSig,file="nonSigdosSensGenesMA112.csv")

dosSensGenesGC8chrXVI <- resultsDosSens(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8)#,"chrXVI")
dosSensGenesSigGC8 <- dosSensGenesGC8chrXVI$Sig
# write.csv(dosSensGenesGC8chrXVI$Sig,file="sigdosSensGenesGC8.csv")
# write.csv(dosSensGenesGC8chrXVI$NonSig,file="nonSigdosSensGenesGC8.csv")

### need to do the euploid samples next - dosage sensitive genes
dosSensGenesGC1 <- resultsDosSens(ddsDefGC_1,"Holly_1","Holly_Anc",genesGC_1)
dosSensGenesSigGC1 <- dosSensGenesGC1$Sig

dosSensGenesGC2 <- resultsDosSens(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2)
dosSensGenesSigGC2 <- dosSensGenesGC2$Sig
# write.csv(dosSensGenesGC2$Sig,file="sigdosSensGenesGC2.csv")
# write.csv(dosSensGenesGC2$NonSig,file="nonSigdosSensGenesGC2.csv")

dosSensGenesGC3 <- resultsDosSens(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3)
dosSensGenesSigGC3 <- dosSensGenesGC3$Sig
# write.csv(dosSensGenesGC3$Sig,file="sigdosSensGenesGC3.csv")
# write.csv(dosSensGenesGC3$NonSig,file="nonSigdosSensGenesGC3.csv")

dosSensGenesGC5 <- resultsDosSens(ddsDefGC_5,"Holly_5","Holly_Anc",genesGC_5)
dosSensGenesSigGC5 <- dosSensGenesGC5$Sig
# write.csv(dosSensGenesGC5$Sig,file="sigdosSensGenesGC5.csv")
# write.csv(dosSensGenesGC5$NonSig,file="nonSigdosSensGenesGC5.csv")


dosSensGenesGC9 <- resultsDosSens(ddsDefGC_9,"Holly_9","Holly_Anc",genesGC_9)
dosSensGenesSigGC9 <- dosSensGenesGC9$Sig
# write.csv(dosSensGenesGC9$Sig,file="sigdosSensGenesGC9.csv")
# write.csv(dosSensGenesGC9$NonSig,file="nonSigdosSensGenesGC9.csv")

dosSensGenesMA1 <- resultsDosSens(ddsDefMAold_1,"SC001","SCA",genesMAold_1)
dosSensGenesSigMA1 <- dosSensGenesMA1$Sig
# write.csv(dosSensGenesMA1$Sig,file="sigdosSensGenesMA1.csv")
# write.csv(dosSensGenesMA1$NonSig,file="nonSigdosSensGenesMA1.csv")

dosSensGenesMA2 <- resultsDosSens(ddsDefMAold_2,"SC002","SCA",genesMAold_2)
dosSensGenesSigMA2 <- dosSensGenesMA2$Sig
# write.csv(dosSensGenesMA2$Sig,file="sigdosSensGenesMA2.csv")
# write.csv(dosSensGenesMA2$NonSig,file="nonSigdosSensGenesMA2.csv")

dosSensGenesMA3 <- resultsDosSens(ddsDefMAold_3,"SC003","SCA",genesMAold_3)
dosSensGenesSigMA3 <- dosSensGenesMA3$Sig
# write.csv(dosSensGenesMA3$Sig,file="sigdosSensGenesMA3.csv")
# write.csv(dosSensGenesMA3$NonSig,file="nonSigdosSensGenesMA3.csv")

dosSensGenesMA4 <- resultsDosSens(ddsDefMAold_4,"SC004","SCA",genesMAold_4)
dosSensGenesSigMA4 <- dosSensGenesMA4$Sig
# write.csv(dosSensGenesMA4$Sig,file="sigdosSensGenesMA4.csv")
# write.csv(dosSensGenesMA4$NonSig,file="nonSigdosSensGenesMA4.csv")

dosSensGenesMA5 <- resultsDosSens(ddsDefMAold_5,"SC005","SCA",genesMAold_5)
dosSensGenesSigMA5 <- dosSensGenesMA5$Sig
# write.csv(dosSensGenesMA5$Sig,file="sigdosSensGenesMA5.csv")
# write.csv(dosSensGenesMA5$NonSig,file="nonSigdosSensGenesMA5.csv")

dosSensGenesMA7 <- resultsDosSens(ddsDefMAold_7,"SC007","SCA",genesMAold_7)
dosSensGenesSigMA7 <- dosSensGenesMA7$Sig
# write.csv(dosSensGenesMA7$Sig,file="sigdosSensGenesMA7.csv")
# write.csv(dosSensGenesMA7$NonSig,file="nonSigdosSensGenesMA7.csv")

dosSensGenesMA8 <- resultsDosSens(ddsDefMAold_8,"SC008","SCA",genesMAold_8)
dosSensGenesSigMA8 <- dosSensGenesMA8$Sig
# write.csv(dosSensGenesMA8$Sig,file="sigdosSensGenesMA8.csv")
# write.csv(dosSensGenesMA8$NonSig,file="nonSigdosSensGenesMA8.csv")

dosSensGenesMA50 <- resultsDosSens(ddsDefMAnewS50,"50","Anc",genesMAnewS50)
dosSensGenesSigMA50 <-  dosSensGenesMA50$Sig
# write.csv(dosSensGenesMA50$Sig,file="sigdosSensGenesMA50.csv")
# write.csv(dosSensGenesMA50$NonSig,file="nonSigdosSensGenesMA50.csv")

dosSensGenesMA115 <- resultsDosSens(ddsDefMAnewS115,"115","Anc",genesMAnewS115)

# write.csv(dosSensResMA115$Sig,file="sigDosSensGenesMA115.csv")
# write.csv(dosSensResMA115$NonSig,file="nonSigDosSensGenesMA115.csv")

upSetdosSensHetAneu <- list(GCS11=dosSensGenesGC11$Sig$GENEID,
                 GCS18=dosSensGenesGC18chrI$Sig$GENEID,
                 GCS7=dosSensGenesGC7chrI$Sig$GENEID,
                 GCS4=dosSensGenesGC4chrV$Sig$GENEID,
                 GCS49=dosSensGenesGC49chrV$Sig$GENEID,
                 GCS59=dosSensGenesGC59chrVII$Sig$GENEID,
                 GCS61=dosSensGenesGC61chrVII$Sig$GENEID,
                 GCS76=dosSensGenesGC76chrIX$Sig$GENEID,
                 GCS77=dosSensGenesGC77chrXII$Sig$GENEID,
                 GCS8=dosSensGenesGC8chrXVI$Sig$GENEID
)

upset(fromList(upSetdosSensHetAneu),nsets=10,keep.order = F,matrix.color = "blue",main.bar.color = "turquoise",sets.bar.color = "purple",order.by = "freq",sets.x.label = "DE Dosage Sensitive Genes")


upSetdosSensHomAneu <- list(MAS9=dosSensGenesMA9chrXIV$Sig$GENEID,
                 MAS112=dosSensGenesMA112chrXVI$Sig$GENEID,
                 MAS29=dosSensGenesMA29chrIX$Sig$GENEID,
                 MAS119=dosSensGenesMA119chrIX$Sig$GENEID,
                 MAS88=dosSensGenesMA88chrIX$Sig$GENEID,
                 MAS108=dosSensGenesMA108chrVIII$Sig$GENEID,
                 MAS15=dosSensGenesMA15chrIX$Sig$GENEID,
                 MAS117=dosSensGenesMA117chrV$Sig$GENEID,
                 MAS123=dosSensGenesMA123chrV$Sig$GENEID,
                 MAS152=dosSensGenesMA152$Sig$GENEID
)

upset(fromList(upSetdosSensHomAneu),nsets=10,keep.order = F,matrix.color = "blue",main.bar.color = "turquoise",sets.bar.color = "purple",order.by = "freq",sets.x.label = "DE dosSens Genes")


upSetdosSensHetEu <- list(GCS2=dosSensGenesGC2$Sig$GENEID,
                 GCS3=dosSensGenesGC3$Sig$GENEID,
                 GCS5=dosSensGenesGC5$Sig$GENEID,
                 GCS9=dosSensGenesGC9$Sig$GENEID
              
)

upset(fromList(upSetdosSensHetEu),nsets=4,keep.order = F,matrix.color = "blue",main.bar.color = "turquoise",sets.bar.color = "purple",order.by = "freq",sets.x.label = "DE dosSens Genes")


upSetdosSensHomEu <- list(MAS1=dosSensGenesMA1$Sig$GENEID,
                 MAS2=dosSensGenesMA2$Sig$GENEID,
                 MAS3=dosSensGenesMA3$Sig$GENEID,
                 MAS4=dosSensGenesMA4$Sig$GENEID,
                 MAS5=dosSensGenesMA5$Sig$GENEID,
                 MAS7=dosSensGenesMA7$Sig$GENEID,
                 MAS8=dosSensGenesMA8$Sig$GENEID,
                 MAS50=dosSensGenesMA50$Sig$GENEID,
                 MAS115=dosSensGenesMA115$Sig$GENEID
              
)

upset(fromList(upSetdosSensHomEu),nsets=9,keep.order = F,matrix.color = "blue",main.bar.color = "turquoise",sets.bar.color = "purple",order.by = "freq",sets.x.label = "DE dosSens Genes")








### find commonly DE dosage Sensitive genes in euploid GC samples
dosSensGenesCommGCEup <- Reduce(intersect, list(dosSensGenesSigGC1,dosSensGenesSigGC2,dosSensGenesSigGC3,dosSensGenesSigGC5,dosSensGenesSigGC9))
dosSensGenesCommGCEup
## no commonly found sig DE dosage Sensitive genes in all GC euploids

dosSensGenesCommMAoldEup <- Reduce(intersect, list(dosSensGenesSigMA1,dosSensGenesSigMA2,dosSensGenesSigMA3,dosSensGenesSigMA4,dosSensGenesSigMA5,dosSensGenesSigMA7,dosSensGenesSigMA8))
dosSensGenesCommMAoldEup

dosSensGenesCommMAnewEup <- Reduce(intersect,list(dosSensGenesSigMA50,dosSensGenesSigMA115))
# also 0

### see if there are any ASR genes commonly found in All Aneuploid GC samples
dosSensGenesCommGCAneu <- Reduce(intersect,list(dosSensGenesSigGC11,dosSensGenesSigGC18,dosSensGenesSigGC4,dosSensGenesSigGC49,dosSensGenesSigGC59,dosSensGenesSigGC61,dosSensGenesSigGC7,dosSensGenesSigGC76,dosSensGenesSigGC77,dosSensGenesSigGC8))
# no commonly sig DE ASR genes in aneuploid GC samples

##old ma samples
dosSensGenesCommMAOldAneu <- Reduce(intersect,list(dosSensGenesSigMA108,dosSensGenesSigMA119,dosSensGenesSigMA15,dosSensGenesSigMA88))

dosSensGenesCommMANewAneu <- Reduce(intersect,list(dosSensGenesSigMA112,dosSensGenesSigMA117,dosSensGenesSigMA123,dosSensGenesSigMA152,dosSensGenesSigMA29))

# 0 in MA old and new samples too

### command to find common elements in several files: 1,2,3,4, + files
# want to know how many genes are differentially expressed commonly in 1,2,3 etc EUPLOID lines
#
  # put all ANEUPLOID samples in same data frame
    dosSensGenesAneuGCSamplesExpr <- rbind(dosSensGenesSigGC11,dosSensGenesSigGC18,dosSensGenesSigGC4,dosSensGenesSigGC49,dosSensGenesSigGC59,dosSensGenesSigGC61,dosSensGenesSigGC7,dosSensGenesSigGC76,dosSensGenesSigGC77,dosSensGenesSigGC8)
      # # first need to filter allSamplesExpr based on p-adj level
      # eupMAnewSamplesDiffExpr <- subset(eupMAnewSamplesExpr, padj<0.1)
      # #
      #
        # then ask how many times each GENEID  is found in eupSamplesExpr
        # library(plyr)
        dosSensGeneGCFreqAneu <- plyr::count(dosSensGenesAneuGCSamplesExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numGCAneuDosSensGenesDE <- plyr::count(dosSensGeneGCFreqAneu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numGCAneuDosSensGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              # write.csv(numGCAneuDosSensGenesDE, file="numb_comm_DE_GC_DosSens_genesANEU.csv")   

### command to find common elements in several files: 1,2,3,4, + files
# want to know how many genes are differentially expressed commonly in 1,2,3 etc ANEUPLOID lines
#
  # put all ANEUPLOID samples in same data frame
    dosSensGenesAneuMAoldSamplesExpr <- rbind(dosSensGenesSigMA108,dosSensGenesSigMA119,dosSensGenesSigMA15,dosSensGenesSigMA88)
      # # first need to filter allSamplesExpr based on p-adj level
      # eupMAnewSamplesDiffExpr <- subset(eupMAnewSamplesExpr, padj<0.1)
      # #
      #
        # then ask how many times each GENEID  is found in eupSamplesExpr
        # library(plyr)
        dosSensGeneMAoldFreqAneu <- plyr::count(dosSensGenesAneuMAoldSamplesExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numMAoldAneudosSensGenesDE <- plyr::count(dosSensGeneMAoldFreqAneu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numMAoldAneudosSensGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              # write.csv(numMAoldAneudosSensGenesDE, file="numb_comm_DE_MAold_dosSens_genesANEU.csv")


### command to find common elements in several files: 1,2,3,4, + files
# want to know how many genes are differentially expressed commonly in 1,2,3 etc ANEUPLOID lines
#
  # put all ANEUPLOID samples in same data frame
    dosSensGenesAneuMAnewSamplesExpr <- rbind(dosSensGenesSigMA112,dosSensGenesSigMA29,dosSensGenesSigMA117,dosSensGenesSigMA123,dosSensGenesSigMA152)
      # # first need to filter allSamplesExpr based on p-adj level
      # eupMAnewSamplesDiffExpr <- subset(eupMAnewSamplesExpr, padj<0.1)
      # #
      #
        # then ask how many times each GENEID  is found in eupSamplesExpr
        # library(plyr)
        dosSensGeneMAnewFreqAneu <- plyr::count(dosSensGenesAneuMAnewSamplesExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numMAnewAneudosSensGenesDE <- plyr::count(dosSensGeneMAnewFreqAneu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numMAnewAneudosSensGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              # write.csv(numMAnewAneudosSensGenesDE, file="numb_comm_DE_MAnew_dosSens_genesANEU.csv")



# anChr = "chrXII"
# dds=ddsDefGC_18
# anc="Holly_Anc"
# genes="genesGC_18"

resultsMonosomy <- function(dds,sample,anc,genes,anChr) {
# make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  remove1 <- anChr
  genes <- genes[(grep(paste0(remove1,"$"),genes$TXCHROM,perl=TRUE)),]
  all(rownames(dds) == genes$GENEID)
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  resMon <- results(dds,contrast=c("condition",sample,anc),lfcThreshold=1, altHypothesis="less")
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (NO DC))
  resMonNonSig <- as.data.frame(subset(resMon,padj>0.1))
  resMonNonSig <- cbind(names = rownames(resMonNonSig), resMonNonSig)
  resMonNonSig <- dplyr::inner_join(genes,resMonNonSig,by=c("GENEID"="names"))
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resMonSig <- as.data.frame(subset(resMon,padj<0.1))
  resMonSig <- cbind(names = rownames(resMonSig), resMonSig)
  resMonSig <- dplyr::inner_join(genes,resMonSig,by=c("GENEID"="names"))
# put non-significantly different genes in a data table
  results <- list()
  results$resNoDC <- resMonNonSig
  results$resPotDC <- resMonSig
  resDi <- results(ddsCopy,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  ### Find genes that are Signficiantly different than Trisomic expectation and are Significantly different from Disomic expectation
  resPartOvDC <- dplyr::inner_join(resMonSig,resDiSig,by="GENEID")
  results$resPartOvDC <- resPartOvDC
  ### Find genes that are Signficiantly different than Trisomic expectation but NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resMonSig,resDiNonSig,by="GENEID")
  results$resFullDC <- resFullDC
  transGenesDE <- dplyr::anti_join(resDiSig,genes,by=c("GENEID"="TXCHROM"))
  results$transGenesDE <- transGenesDE
  return(results)
}

resultsTetrasomy <- function(dds,sample,anc,genes,anChr) {
# make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  remove1 <- anChr
  genes <- genes[(grep(paste0(remove1,"$"),genes$TXCHROM,perl=TRUE)),]
  all(rownames(dds) == genes$GENEID)
  # genes <- genes$GENEID
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  resTetra <- results(dds,contrast=c("condition",sample,anc),lfcThreshold=1, altHypothesis="greater")
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (NO DC))
  resTetraNonSig <- as.data.frame(subset(resTetra,padj>0.1))
  resTetraNonSig <- cbind(names = rownames(resTetraNonSig), resTetraNonSig)
  resTetraNonSig <- dplyr::inner_join(genes,resTetraNonSig,by=c("GENEID"="names"))
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resTetraSig <- as.data.frame(subset(resTetra,padj<0.1))
  resTetraSig <- cbind(names = rownames(resTetraSig), resTetraSig)
  resTetraSig <- dplyr::inner_join(genes,resTetraSig,by=c("GENEID"="names"))
# put non-significantly different genes in a data table
  results <- list()
  results$resNoDC <- resTetraNonSig
  results$resPotDC <- resTetraSig
  #### run same analyses using new ddsCopy dataset and new parameters
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(ddsCopy,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  ### Find genes that are Signficiantly different than Trisomic expectation and are Significantly different from Disomic expectation
  resPartOvDC <- dplyr::inner_join(resTetraSig,resDiSig,by="GENEID")
  results$resPartOvDC <- resPartOvDC
  ### Find genes that are Signficiantly different than Trisomic expectation but NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resTetraSig,resDiNonSig,by="GENEID")
  results$resFullDC <- resFullDC
  transGenesDE <- dplyr::anti_join(resDiSig,genes,by=c("GENEID"="TXCHROM"))
  results$transGenesDE <- transGenesDE
  return(results)
}


#### GC Samples
# resultsNames(ddsDefGC_11)
# sampleResults11GCXV <- resultsTrisomy(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11,"chrXV")
# write.csv(sampleResults11GCXV$resNoDC,file="sample11GCchrXVnoDCgenes.csv")
# want % of genes on this chromosome that are not dosage compensated
# write.csv(sampleResults11GCXV$resPartOvDC,file="sample11GCchrXVPotDCgenes.csv")
# write.csv(sampleResults11GCXV$resFullDC,file="sample11GCchrXVFullDCgenes.csv")
# write.csv(sampleResults11GCXV$transGenesDE,file="sample11GCchrXVtransDEgenes.csv")

# sampleResults4V <- resultsTrisomy(ddsDefGC_4,"Holly_4","Holly_Anc",genesGC_4,"chrV")
# write.csv(sampleResults4V$resNoDC,file="sample4GCchrVnoDCgenes.csv")
# # want % of genes on this chromosome that are not dosage compensated
# write.csv(sampleResults4V$resPartOvDC,file="sample4GCchrVPotDCgenes.csv")
# write.csv(sampleResults4V$resFullDC,file="sample4GCchr4VFullDCgenes.csv")
# write.csv(sampleResults4V$transGenesDE,file="sample4GCchrVtransDEgenes.csv")


# sampleResults18GCI <- resultsTrisomy(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18,"chrI")
# write.csv(sampleResults18GCI$resNoDC,file="sample18GCnoDCgenesI.csv")
# write.csv(sampleResults18GCI$resPartOvDC,file="sample18GCPartOverDCgenesI.csv")
# write.csv(sampleResults18GCI$resFullDC,file="sample18GCfullDCgenesI.csv")
# write.csv(sampleResults18GCI$transGenesDE,file="sample18GCtransDEgenesI.csv")

# sampleResults18GCXII <- resultsTrisomy(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18,"chrXII")
# write.csv(sampleResults18GCXII$resNoDC,file="sample18GCnoDCgenesXII.csv")
# write.csv(sampleResults18GCXII$resPartOvDC,file="sample18GCPartOverDCgenesXII.csv")
# write.csv(sampleResults18GCXII$resFullDC,file="sample18GCfullDCgenesXII.csv")
# write.csv(sampleResults18GCXII$transGenesDE,file="sample18GCtransDEgenesXII.csv")

# sampleResults49GC <- resultsTrisomy(ddsDefGC_49,"Holly_49","Holly_Anc",genesGC_49,"chrV")
# write.csv(sampleResults49GC$resNoDC,file="sample49GCnoDCgenesV.csv")
# write.csv(sampleResults49GC$resPartOvDC,file="sample49PartOverDCgenesV.csv")
# write.csv(sampleResults49GC$resFullDC,file="sample49FullDCgenesV.csv")
# write.csv(sampleResults49GC$transGenesDE,file="sample49GCtransDEgenesV.csv")

# sampleResults59GC <- resultsTrisomy(ddsDefGC_59,"Holly_59","Holly_Anc",genesGC_59,"chrVII")
# write.csv(sampleResults59GC$resNoDC,file="sample59GCnoDCgenesVII.csv")
# write.csv(sampleResults59GC$resPartOvDC,file="sample59GCPartOverGenesVII.csv")
# write.csv(sampleResults59GC$resFullDC,file="sample59GCFullDCGenesVII.csv")
# write.csv(sampleResults59GC$transGenesDE,file="sample59GCtransDEgenesVII.csv")

# sampleResults61GC <- resultsTrisomy(ddsDefGC_61,"Holly_61","Holly_Anc",genesGC_61,"chrVII")
# write.csv(sampleResults61GC$resNoDC,file="sample61GCnoDCgenesVII.csv")
# write.csv(sampleResults61GC$resPartOvDC, file="sample61GCpartOverDCgenesVII.csv")
# write.csv(sampleResults61GC$resFullDC,file="sample61GCfullDCgenesVII.csv")
# write.csv(sampleResults61GC$transGenesDE,file="sample61GCtransDEgenesVII.csv")

# sampleResults76GCIX <- resultsTrisomy(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76,"chrIX")
# write.csv(sampleResults76GCIX$resNoDC,file="sample76GCnoDCgenesIX.csv")
# write.csv(sampleResults76GCIX$resPartOvDC, file="sample76GCpartOverDCgenesIX.csv")
# write.csv(sampleResults76GCIX$resFullDC,file="sample76GCfullDCgenesIX.csv")
# write.csv(sampleResults76GCIX$transGenesDE,file="sample76GCtransDEgenesIX.csv")

# sampleResults76GCX <- resultsTrisomy(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76,"chrX")
# write.csv(sampleResults76GCX$resNoDC,file="sample76GCnoDCgenesX.csv")
# write.csv(sampleResults76GCX$resPartOvDC, file="sample76GCpartOverDCgenesX.csv")
# write.csv(sampleResults76GCX$resFullDC,file="sample76GCfullDCgenesX.csv")
# write.csv(sampleResults76GCX$transGenesDE,file="sample76GCtransDEgenesX.csv")

# sampleResults76GCXIV <- resultsTrisomy(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76,"chrXIV")
# write.csv(sampleResults76GCXIV$resNoDC,file="sample76GCnoDCgenesXIV.csv")
# write.csv(sampleResults76GCXIV$resPartOvDC, file="sample76GCpartOverDCgenesXIV.csv")
# write.csv(sampleResults76GCXIV$resFullDC,file="sample76GCfullDCgenesXIV.csv")
# write.csv(sampleResults76GCXIV$transGenesDE,file="sample76GCtransDEgenesXIV.csv")

# sampleResults7GC <- resultsTrisomy(ddsDefGC_7,"Holly_7","Holly_Anc",genesGC_7,"chrI")
# write.csv(sampleResults7GC$resNoDC,file="sample7GCnoDCgenesI.csv")
# write.csv(sampleResults7GC$resPartOvDC, file="sample7GCpartOverDCgenesI.csv")
# write.csv(sampleResults7GC$resFullDC,file="sample7GCfullDCgenesI.csv")
# write.csv(sampleResults7GC$transGenesDE,file="sample7GCtransDEgenesI.csv")

# sampleResults77GC <- resultsTrisomy(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_77,"chrXIV")
# write.csv(sampleResults77GC$resNoDC,file="sample77GCnoDCgenesXIV.csv")
# write.csv(sampleResults77GC$resPartOvDC, file="sample77GCpartOverDCgenesXIV.csv")
# write.csv(sampleResults77GC$resFullDC,file="sample77GCfullDCgenesXIV.csv")
# write.csv(sampleResults77GC$transGenesDE,file="sample177GCtransDEgenesXIV.csv")

# sampleResults8GC <- resultsTetrasomy(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8,"chrXVI")
# write.csv(sampleResults8GC$resNoDC,file="sample8GCnoDCgenesXVI.csv")
# write.csv(sampleResults8GC$resPartOvDC, file="sample8GCpartOverDCgenesXVI.csv")
# write.csv(sampleResults8GC$resFullDC,file="sample8GCfullDCgenesXVI.csv")
# write.csv(sampleResults8GC$transGenesDE,file="sample8GCtransDEgenesXVI.csv")

# sampleResults11GCI <- resultsMonosomy(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11,"chrI")
# write.csv(sampleResults11GCI$resNoDC,file="sample11GCnoDCgenesI.csv")
# write.csv(sampleResults11GCI$resPartOvDC, file="sample11GCpartOverDCgenesI.csv")
# write.csv(sampleResults11GCI$resFullDC,file="sample11GCfullDCgenesI.csv")
# write.csv(sampleResults11GCI$transGenesDE,file="sample11GCtransDEgenesI.csv")


#### MA Old Samples
# resultsNames(ddsDefMAold)
# sampleResults9MAoldXIV <- resultsTrisomy(ddsDefMAold_9,"SC009","SCA",genesMAold_9,"chrXIV")
# write.csv(sampleResults9MAoldXIV$resNoDC,file="sample9MAnoDCgenesXIV.csv")
# write.csv(sampleResults9MAoldXIV$resPartOvDC, file="sample9MApartOverDCgenesXIV.csv")
# write.csv(sampleResults9MAoldXIV$resFullDC,file="sample9MAfullDCgenesXIV.csv")
# write.csv(sampleResults9MAoldXIV$transGenesDE,file="sample9MAtransDEgenesXIV.csv")

# sampleResults15MAoldIX <- resultsTrisomy(ddsDefMAold_15,"SC015","SCA",genesMAold_15,"chrIX")
# write.csv(sampleResults15MAoldIX$resNoDC,file="sample15MAnoDCgenesIX.csv")
# write.csv(sampleResults15MAoldIX$resPartOvDC, file="sample15MApartOverDCgenesIX.csv")
# write.csv(sampleResults15MAoldIX$resFullDC,file="sample15MAfullDCgenesIX.csv")
# write.csv(sampleResults15MAoldIX$transGenesDE,file="sample15MAtransDEgenesIX.csv")

# sampleResults88MAoldIX <- resultsTrisomy(ddsDefMAold_88,"SC088","SCA",genesMAold_88,"chrIX")
# write.csv(sampleResults88MAoldIX$resNoDC,file="sample88MAnoDCgenesIX.csv")
# write.csv(sampleResults88MAoldIX$resPartOvDC, file="sample88MApartOverDCgenesIX.csv")
# write.csv(sampleResults88MAoldIX$resFullDC,file="sample88MAfullDCgenesIX.csv")
# write.csv(sampleResults88MAoldIX$transGenesDE,file="sample88MAtransDEgenesIX.csv")

# sampleResults108MAoldVIII <- resultsTrisomy(ddsDefMAold_108,"SC108","SCA",genesMAold_108,"chrVIII")
# write.csv(sampleResults108MAoldVIII$resNoDC,file="sample108MAnoDCgenesVIII.csv")
# write.csv(sampleResults108MAoldVIII$resPartOvDC, file="sample108MApartOverDCgenesVIII.csv")
# write.csv(sampleResults108MAoldVIII$resFullDC,file="sample108MAfullDCgenesVIII.csv")
# write.csv(sampleResults108MAoldVIII$transGenesDE,file="sample108MAtransDEgenesVIII.csv")

# sampleResults119MAoldIX <- resultsTrisomy(ddsDefMAold_119,"SC119","SCA",genesMAold_119,"chrIX")
# write.csv(sampleResults119MAoldIX$resNoDC,file="sample119MAnoDCgenesIX.csv")
# write.csv(sampleResults119MAoldIX$resPartOvDC, file="sample119MApartOverDCgenesIX.csv")
# write.csv(sampleResults119MAoldIX$resFullDC,file="sample119MAfullDCgenesIX.csv")
# write.csv(sampleResults119MAoldIX$transGenesDE,file="sample119MAtransDEgenesIX.csv")

# sampleResults108MAoldIX <- resultsMonosomy(ddsDefMAold_108,"SC108","SCA",genesMAold_108,"chrIX")
# write.csv(sampleResults108MAoldIX$resNoDC,file="sample108MAnoDCgenesIX.csv")
# write.csv(sampleResults108MAoldIX$resPartOvDC, file="sample108MApartOverDCgenesIX.csv")
# write.csv(sampleResults108MAoldIX$resFullDC,file="sample108MAfullDCgenesIX.csv")
# write.csv(sampleResults108MAoldIX$transGenesDE,file="sample108MAtransDEgenesIX.csv")

### MA New Samples
# resultsNames(ddsDefMAnewS29)
# sampleResults29MANewIX <- resultsMonosomy(ddsDefMAnewS29,"29","Anc",genesMAnewS29,"chrIX")
# write.csv(sampleResults29MANewIX$resNoDC,file="sample29MAnoDCgenesIX.csv")
# write.csv(sampleResults29MANewIX$resPartOvDC, file="sample29MApartOverDCgenesIX.csv")
# write.csv(sampleResults29MANewIX$resFullDC,file="sample29MAfullDCgenesIX.csv")
# write.csv(sampleResults29MANewIX$transGenesDE,file="sample29MAtransDEgenesIX.csv")

# sampleResults112MANewXVI <- resultsTrisomy(ddsDefMAnewS112,"112","Anc",genesMAnewS112,"chrXVI")
# write.csv(sampleResults112MANewXVI$resNoDC,file="sample112MAnoDCgenesXVI.csv")
# write.csv(sampleResults112MANewXVI$resPartOvDC, file="sample112MApartOverDCgenesXVI.csv")
# write.csv(sampleResults112MANewXVI$resFullDC,file="sample112MAfullDCgenesXVI.csv")
# write.csv(sampleResults112MANewXVI$transGenesDE,file="sample112MAtransDEgenesXVI.csv")

# sampleResults117MANewV <- resultsTrisomy(ddsDefMAnewS117,"117","Anc",genesMAnewS117,"chrV")
# write.csv(sampleResults117MANewV$resNoDC,file="sample117MAnoDCgenesV.csv")
# write.csv(sampleResults117MANewV$resPartOvDC, file="sample117MApartOverDCgenesV.csv")
# write.csv(sampleResults117MANewV$resFullDC,file="sample117MAfullDCgenesV.csv")
# write.csv(sampleResults117MANewV$transGenesDE,file="sample117MAtransDEgenesV.csv")

# sampleResults123MANewV <- resultsTrisomy(ddsDefMAnewS123,"123","Anc",genesMAnewS123,"chrV")
# write.csv(sampleResults123MANewV$resNoDC,file="sample123MAnoDCgenesV.csv")
# write.csv(sampleResults123MANewV$resPartOvDC, file="sample123MApartOverDCgenesV.csv")
# write.csv(sampleResults123MANewV$resFullDC,file="sample123MAfullDCgenesV.csv")
# write.csv(sampleResults123MANewV$transGenesDE,file="sample123MAtransDEgenesV.csv")

# sampleResults152MANewI <- resultsTrisomy(ddsDefMAnewS152,"152","Anc",genesMAnewS152,"chrI")
# write.csv(sampleResults152MANewI$resNoDC,file="sample152MAnoDCgenesI.csv")
# write.csv(sampleResults152MANewI$resPartOvDC, file="sample152MApartOverDCgenesI.csv")
# write.csv(sampleResults152MANewI$resFullDC,file="sample152MAfullDCgenesI.csv")
# write.csv(sampleResults152MANewI$transGenesDE,file="sample152MAtransDEgenesI.csv")

# sampleResults152MANewVIII <- resultsTrisomy(ddsDefMAnewS152,"152","Anc",genesMAnewS152,"chrVIII")
# write.csv(sampleResults152MANewVIII$resNoDC,file="sample152MAnoDCgenesVIII.csv")
# write.csv(sampleResults152MANewVIII$resPartOvDC, file="sample152MApartOverDCgenesVIII.csv")
# write.csv(sampleResults152MANewVIII$resFullDC,file="sample152MAfullDCgenesVII.csv")
# write.csv(sampleResults152MANewVIII$transGenesDE,file="sample152MAtransDEgenesVII.csv")

```

```{r trisomy, eval=TRUE, echo=FALSE,include=FALSE}
resultsTrisomy <- function(dds,sample,anc,genes,anChr) {
  # make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  genes
  keep <- anChr
  genes <- genes[(grep(paste0(keep,"$"),genes$TXCHROM,perl=TRUE)),]
  genes
  all(rownames(dds) == genes$GENEID)
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  ddsGenes <- intersect(rownames(dds),genes)
  as.data.frame(genes)
  str(genes)
  # ddsGenes <- genes[,ddsGenes]
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  resTri <- results(dds,contrast=c("condition",sample,anc),lfcThreshold=0.585)
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (either no DC or unknown))
  resTriNonSig <- as.data.frame(subset(resTri,padj>0.1))
  length(resTriNonSig)
  resTriNonSig <- cbind(names = rownames(resTriNonSig), resTriNonSig)
  resTriNonSig <- dplyr::inner_join(genes,resTriNonSig,by=c("GENEID"="names"))
  str(resTriNonSig)
  # View(resTriNonSig)
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resTriSig <- as.data.frame(subset(resTri,padj<0.1))
  resTriSig <- cbind(names = rownames(resTriSig), resTriSig)
  resTriSig <- dplyr::inner_join(genes,resTriSig,by=c("GENEID"="names"))
  str(resTriSig)
  #### run same analyses using new ddsCopy dataset and new parameters
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (either Full DC or unknown genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  str(resDiNonSig)
  # subset data frame to get Significant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  str(resDiSig)
  #####################################
  ### Find genes that are Signficiantly different than Trisomic expectation and 
  # are Significantly different from Disomic expectation
  results <- list()
  resSigBoth<- dplyr::inner_join(resTriSig,resDiSig,by="GENEID")
  ## over DC genes for trisomy : less than 0 (1x)
  over.4 <- resSigBoth[which(resSigBoth$log2FoldChange.x<0),]
  ## anti DC genes for trisomy: more than 0.585 (1.5x)
  anti.5 <- resSigBoth[which(resSigBoth$log2FoldChange.x > 0.585),]
  results$over.4 <- over.4
  results$anti.5 <- anti.5
  ### Find genes that are Signficiantly different than Trisomic expectation but
  # NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resTriSig,resDiNonSig,by="GENEID")
  results$full.3 <- resFullDC
  ## Find genes that are not significantly diff from trisomic exp but sig diff from disomic
  resPartNo <- dplyr::inner_join(resTriNonSig,resDiSig,by="GENEID")
  resPart <- resPartNo[which(resPartNo$log2FoldChange.x < 0.585),]
  results$Part.2 <- resPart
  resNo <- resPartNo[which(resPartNo$log2FoldChange.x >= 0.585),]
  results$No.1 <- resNo
  ## Get genes that are not significantly different from either Trisomic or Disomic expectation
  resUnknown <- dplyr::inner_join(resTriNonSig,resDiNonSig,by="GENEID")
  results$Unknown.0 <- resUnknown
  return(results)
}
```


```{r tetrasomy, eval=FALSE, echo=FALSE,include=FALSE}

dds=ddsDefGC_8
sample="Holly_8"
anc="Holly_Anc"
genes=genesGC_8
anChr="chrXVI"

resultsTetrasomy <- function(dds,sample,anc,genes,anChr) {
  # make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  genes
  keep <- anChr
  genes <- genes[(grep(paste0(keep,"$"),genes$TXCHROM,perl=TRUE)),]
  genes
  all(rownames(dds) == genes$GENEID)
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  ddsGenes <- intersect(rownames(dds),genes)
  as.data.frame(genes)
  str(genes)
  # ddsGenes <- genes[,ddsGenes]
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 1 (expectation for tetrasomic genes)
  resTet <- results(dds,contrast=c("condition",sample,anc), lfcThreshold=1 )
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (either no DC or unknown))
  resTetNonSig <- as.data.frame(subset(resTet,padj>0.1))
  length(resTetNonSig)
 resTetNonSig <- cbind(names = rownames(resTetNonSig), resTetNonSig)
  resTetNonSig <- dplyr::inner_join(genes,resTetNonSig,by=c("GENEID"="names"))
  str(resTetNonSig)
  # View(resTriNonSig)
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resTetSig <- as.data.frame(subset(resTet,padj<0.1))
  resTetSig <- cbind(names = rownames(resTetSig), resTetSig)
  resTetSig <- dplyr::inner_join(genes,resTetSig,by=c("GENEID"="names"))
  str(resTetSig)
  #### run same analyses using new ddsCopy dataset and new parameters
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (either Full DC or unknown genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  str(resDiNonSig)
  # subset data frame to get Significant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  str(resDiSig)
  #####################################
  ### Find genes that are Signficiantly different than Trisomic expectation and 
  # are Significantly different from Disomic expectation
  results <- list()
  resSigBoth<- dplyr::inner_join(resTetSig,resDiSig,by="GENEID")
  ## over DC genes for tetrasomy : less than 0 (1x)
  over.4 <- resSigBoth[which(resSigBoth$log2FoldChange.x<0),]
  ## anti DC genes for trisomy: more than 1 (2x)
  anti.5 <- resSigBoth[which(resSigBoth$log2FoldChange.x > 1),]
  results$over.4 <- over.4
  results$anti.5 <- anti.5
  ### Find genes that are Signficiantly different than Trisomic expectation but
  # NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resTetSig,resDiNonSig,by="GENEID")
  results$full.3 <- resFullDC
  ## Find genes that are not significantly diff from trisomic exp but sig diff from disomic
  resPartNo <- dplyr::inner_join(resTetNonSig,resDiSig,by="GENEID")
  resPart <- resPartNo[which(resPartNo$log2FoldChange.x < 1),]
  resPart1 <- resPart[which(resPart$log2FoldChange.x > 0),]
  results$Part.2 <- resPart1
  resNo <- resPartNo[which(resPartNo$log2FoldChange.x >= 1),]
  results$No.1 <- resNo
  ## Get genes that are not significantly different from either Trisomic or Disomic expectation
  resUnknown <- dplyr::inner_join(resTriNonSig,resDiNonSig,by="GENEID")
  results$Unknown.0 <- resUnknown
  return(results)
}

# input is dds for that sample, line #, ancestor name, genes matrix for that sample, and chromosome of interest
# output is list with full, part, over, anti, no, and unknown DC genes 
rm(dds)
rm(sample)
rm(anc)
rm(genes)
rm(anChr)

L <- resultsTrisomy(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8,"chrXVI")
View(L)




```


```{r monosomy, eval=TRUE,echo=FALSE,include=FALSE}

resultsMonosomy <- function(dds,sample,anc,genes,anChr) {
  # make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  # genes
  keep <- anChr
  genes <- genes[(grep(paste0(keep,"$"),genes$TXCHROM,perl=TRUE)),]
  # genes
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  ddsGenes <- intersect(rownames(dds),genes)
  # as.data.frame(genes)
  # str(genes)
  # ddsGenes <- genes[,ddsGenes]
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at -1 (expectation for monosomic genes)
  resMon <- results(dds,contrast=c("condition",sample,anc),lfcThreshold=1, altHypothesis="less")
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (either no DC or unknown))
  resMonNonSig <- as.data.frame(subset(resMon,padj>0.1))
  length(resMonNonSig)
  resMonNonSig <- cbind(names = rownames(resMonNonSig), resMonNonSig)
  resMonNonSig <- dplyr::inner_join(genes,resMonNonSig,by=c("GENEID"="names"))
  # str(resMonNonSig)
  # View(resTriNonSig)
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resMonSig <- as.data.frame(subset(resMon,padj<0.1))
  resMonSig <- cbind(names = rownames(resMonSig), resMonSig)
  resMonSig <- dplyr::inner_join(genes,resMonSig,by=c("GENEID"="names"))
  str(resMonSig)
  #### run same analyses using new ddsCopy dataset and new parameters
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (either Full DC or unknown genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  # str(resDiNonSig)
  # subset data frame to get Significant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  # str(resDiSig)
  #####################################
  ### Find genes that are Signficiantly different than Monosomic expectation and 
  # are Significantly different from Disomic expectation
  results <- list()
  resSigBoth <- dplyr::inner_join(resMonSig,resDiSig,by="GENEID")
  ## over DC genes for monosomy : greater than 0 (1x)
  over.4 <- resSigBoth[which(resSigBoth$log2FoldChange.x>0),]
  ## anti DC genes for monosomy: less than -1 (0.5)
  anti.5 <- resSigBoth[which(resSigBoth$log2FoldChange.x < -1),]
  results$over.4 <- over.4
  results$anti.5 <- anti.5
  ### Find genes that are Signficiantly different than Trisomic expectation but
  # NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resMonSig,resDiNonSig,by="GENEID")
  results$full.3 <- resFullDC
  ## Find genes that are not significantly diff from trisomic exp but sig diff from disomic
  resPartNo <- dplyr::inner_join(resMonNonSig,resDiSig,by="GENEID")
  resPart <- resPartNo[which(resPartNo$log2FoldChange.x > -1),]
  results$Part.2 <- resPart
  resNo <- resPartNo[which(resPartNo$log2FoldChange.x <= -1),]
  results$No.1 <- resNo
  ## Get genes that are not significantly different from either Trisomic or Disomic expectation
  resUnknown <- dplyr::inner_join(resMonNonSig,resDiNonSig,by="GENEID")
  results$Unknown.0 <- resUnknown
  return(results)
}

L <- resultsMonosomy(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11,"chrI")




```


## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
