---
title: "Dosage compensation analysis with DESeq2"
author: "Holly McQueary"
date: "`r format(Sys.Date(), '%m/%d/%Y')`"
abstract: >

output:
  rmarkdown::html_document:
    highlight: pygments
    toc: true
    fig_width: 5
bibliography: library.bib
editor_options:
  chunk_output_type: inline
---

```{r setup, eval=TRUE,include=TRUE}
library(knitr)
knitr::opts_chunk$set(tidy=FALSE, cache=TRUE,
                      dev="png",
                      message=FALSE, error=FALSE, warning=TRUE)
# source("https://bioconductor.org/biocLite.R")
#        biocLite("apeglm")
library(apeglm)
#source("https://bioconductor.org/biocLite.R")
 #       biocLite("parathyroidSE")
#library("parathyroidSE")
#       if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
#
# BiocManager::install("DESeq2")
library(DESeq2)
```

This pipleline is adapted from the DESeq2 tutorial, so some text here is quoted from that R markdown document. Other text is my own. 

### *htseq-count* input

You can use the function *DESeqDataSetFromHTSeqCount* if you
have used *htseq-count* from the
[HTSeq](http://www-huber.embl.de/users/anders/HTSeq)
python package [@Anders:2014:htseq].
For an example of using the python scripts, see the
[pasilla](http://bioconductor.org/packages/pasilla) data package. First you will want to specify a
variable which points to the directory in which the *htseq-count*
output files are located.

I had to load in each line and ancestor individually because I wanted to analyze them individually, not all together, because of the skew of normalization that aneuploidy causes. 
```{r htseqDirI, eval=TRUE,include=TRUE}
directoryGC_All <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC"
##individual lines
directoryGC_1 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_1vsAnc"
directoryGC_2 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_2vsAnc"
directoryGC_3 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_3vsAnc"
directoryGC_4 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_4vsAnc"
directoryGC_5 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_5vsAnc"
directoryGC_6 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_6vsAnc"
directoryGC_7 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_7vsAnc"
directoryGC_8 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_8vsAnc"
directoryGC_9 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_9vsAnc"
directoryGC_11 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_11vsAnc"
directoryGC_18 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_18vsAnc"
directoryGC_31 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_31vsAnc"
directoryGC_49 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_49vsAnc"
directoryGC_59 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_59vsAnc"
directoryGC_61 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_61vsAnc"
directoryGC_69 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_69vsAnc"
directoryGC_76 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_76vsAnc"
directoryGC_77 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_77vsAnc"
#########################################################
directoryMAold <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD"
directoryMAold_1 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/1vsAnc"
directoryMAold_2 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/2vsAnc"
directoryMAold_3 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/3vsAnc"
directoryMAold_4 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/4vsAnc"
directoryMAold_5 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/5vsAnc"
# directoryMAold_6 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/6vsAnc"
directoryMAold_7 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/7vsAnc"
directoryMAold_8 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/8vsAnc"
directoryMAold_9 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/9vsAnc"
directoryMAold_11 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/11vsAnc"
directoryMAold_15 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/15vsAnc"
directoryMAold_28 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/28vsAnc"
directoryMAold_88 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/88vsAnc"
directoryMAold_108 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/108vsAnc"
directoryMAold_119 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/119vsAnc"
########################################################
directoryMAnew <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW"
directoryMAnewS29 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/29vsAnc"
directoryMAnewS50 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/50vsAnc"
directoryMAnewS112 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/112vsAnc"
directoryMAnewS115 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/115vsAnc"
directoryMAnewS117 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/117vsAnc"
directoryMAnewS123 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/123vsAnc"
directoryMAnewS141 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/141vsAnc"
directoryMAnewS152 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/152vsAnc"
```

We specify which files to read in using *list.files*,
and select those files which contain the string `"treated"`
using *grep*. The *sub* function is used to
chop up the sample filename to obtain the condition status, or
you might alternatively read in a phenotypic table
using *read.table*.

```{r htseqInput,eval=TRUE,include=TRUE}
sampleFilesGC_All <- list.files(directoryGC_All)
sampleInfoGC_All <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/sample_data_GC.csv",header=TRUE)
sampleTableGC_All <- data.frame(sampleName = sampleInfoGC_All$sample,
                          fileName = sampleInfoGC_All$filename,
                          run = sampleInfoGC_All$seq_run ,
                           condition = sampleInfoGC_All$group)

sampleFilesGC_1 <- list.files(directoryGC_1)
sampleInfoGC_1 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S1vsAnc.csv",header=TRUE)
sampleTableGC_1 <- data.frame(sampleName = sampleInfoGC_1$sample,
                          fileName = sampleInfoGC_1$filename,
                          run = sampleInfoGC_1$seq_run ,
                           condition = sampleInfoGC_1$group)

sampleFilesGC_2 <- list.files(directoryGC_2)
sampleInfoGC_2 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S2vsAnc.csv",header=TRUE)
sampleTableGC_2 <- data.frame(sampleName = sampleInfoGC_2$sample,
                          fileName = sampleInfoGC_2$filename,
                          run = sampleInfoGC_2$seq_run ,
                           condition = sampleInfoGC_2$group)

sampleFilesGC_3 <- list.files(directoryGC_3)
sampleInfoGC_3 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S3vsAnc.csv",header=TRUE)
sampleTableGC_3 <- data.frame(sampleName = sampleInfoGC_3$sample,
                          fileName = sampleInfoGC_3$filename,
                          run = sampleInfoGC_3$seq_run ,
                           condition = sampleInfoGC_3$group)

sampleFilesGC_4 <- list.files(directoryGC_4)
sampleInfoGC_4 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S4vsAnc.csv",header=TRUE)
sampleTableGC_4 <- data.frame(sampleName = sampleInfoGC_4$sample,
                          fileName = sampleInfoGC_4$filename,
                          run = sampleInfoGC_4$seq_run ,
                           condition = sampleInfoGC_4$group)

sampleFilesGC_5 <- list.files(directoryGC_5)
sampleInfoGC_5 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S5vsAnc.csv",header=TRUE)
sampleTableGC_5 <- data.frame(sampleName = sampleInfoGC_5$sample,
                          fileName = sampleInfoGC_5$filename,
                          run = sampleInfoGC_5$seq_run ,
                           condition = sampleInfoGC_5$group)

sampleFilesGC_6 <- list.files(directoryGC_6)
sampleInfoGC_6 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S6vsAnc.csv",header=TRUE)
sampleTableGC_6 <- data.frame(sampleName = sampleInfoGC_6$sample,
                          fileName = sampleInfoGC_6$filename,
                          run = sampleInfoGC_6$seq_run ,
                           condition = sampleInfoGC_6$group)

sampleFilesGC_7 <- list.files(directoryGC_7)
sampleInfoGC_7 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S7vsAnc.csv",header=TRUE)
sampleTableGC_7 <- data.frame(sampleName = sampleInfoGC_7$sample,
                          fileName = sampleInfoGC_7$filename,
                          run = sampleInfoGC_7$seq_run ,
                           condition = sampleInfoGC_7$group)

sampleFilesGC_8 <- list.files(directoryGC_8)
sampleInfoGC_8 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S8vsAnc.csv",header=TRUE)
sampleTableGC_8 <- data.frame(sampleName = sampleInfoGC_8$sample,
                          fileName = sampleInfoGC_8$filename,
                          run = sampleInfoGC_8$seq_run ,
                           condition = sampleInfoGC_8$group)

sampleFilesGC_9 <- list.files(directoryGC_9)
sampleInfoGC_9 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S9vsAnc.csv",header=TRUE)
sampleTableGC_9 <- data.frame(sampleName = sampleInfoGC_9$sample,
                          fileName = sampleInfoGC_9$filename,
                          run = sampleInfoGC_9$seq_run ,
                           condition = sampleInfoGC_9$group)

sampleFilesGC_11 <- list.files(directoryGC_11)
sampleInfoGC_11 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S11vsAnc.csv",header=TRUE)
sampleTableGC_11 <- data.frame(sampleName = sampleInfoGC_11$sample,
                          fileName = sampleInfoGC_11$filename,
                          run = sampleInfoGC_11$seq_run ,
                           condition = sampleInfoGC_11$group)

sampleFilesGC_18 <- list.files(directoryGC_18)
sampleInfoGC_18 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S18vsAnc.csv",header=TRUE)
sampleTableGC_18 <- data.frame(sampleName = sampleInfoGC_18$sample,
                          fileName = sampleInfoGC_18$filename,
                          run = sampleInfoGC_18$seq_run ,
                           condition = sampleInfoGC_18$group)

sampleFilesGC_31 <- list.files(directoryGC_31)
sampleInfoGC_31 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S31vsAnc.csv",header=TRUE)
sampleTableGC_31 <- data.frame(sampleName = sampleInfoGC_31$sample,
                          fileName = sampleInfoGC_31$filename,
                          run = sampleInfoGC_31$seq_run ,
                           condition = sampleInfoGC_31$group)

sampleFilesGC_49 <- list.files(directoryGC_49)
sampleInfoGC_49 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S49vsAnc.csv",header=TRUE)
sampleTableGC_49 <- data.frame(sampleName = sampleInfoGC_49$sample,
                          fileName = sampleInfoGC_49$filename,
                          run = sampleInfoGC_49$seq_run ,
                           condition = sampleInfoGC_49$group)

sampleFilesGC_59 <- list.files(directoryGC_59)
sampleInfoGC_59 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S59vsAnc.csv",header=TRUE)
sampleTableGC_59 <- data.frame(sampleName = sampleInfoGC_59$sample,
                          fileName = sampleInfoGC_59$filename,
                          run = sampleInfoGC_59$seq_run ,
                           condition = sampleInfoGC_59$group)

sampleFilesGC_61 <- list.files(directoryGC_61)
sampleInfoGC_61 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S61vsAnc.csv",header=TRUE)
sampleTableGC_61 <- data.frame(sampleName = sampleInfoGC_61$sample,
                          fileName = sampleInfoGC_61$filename,
                          run = sampleInfoGC_61$seq_run ,
                           condition = sampleInfoGC_61$group)

sampleFilesGC_69 <- list.files(directoryGC_69)
sampleInfoGC_69 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S69vsAnc.csv",header=TRUE)
sampleTableGC_69 <- data.frame(sampleName = sampleInfoGC_69$sample,
                          fileName = sampleInfoGC_69$filename,
                          run = sampleInfoGC_69$seq_run ,
                           condition = sampleInfoGC_69$group)

sampleFilesGC_76 <- list.files(directoryGC_76)
sampleInfoGC_76 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S76vsAnc.csv",header=TRUE)
sampleTableGC_76 <- data.frame(sampleName = sampleInfoGC_76$sample,
                          fileName = sampleInfoGC_76$filename,
                          run = sampleInfoGC_76$seq_run ,
                           condition = sampleInfoGC_76$group)

sampleFilesGC_77 <- list.files(directoryGC_77)
sampleInfoGC_77 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S77vaAnc.csv",header=TRUE)
sampleTableGC_77 <- data.frame(sampleName = sampleInfoGC_77$sample,
                          fileName = sampleInfoGC_77$filename,
                          run = sampleInfoGC_77$seq_run ,
                           condition = sampleInfoGC_77$group)

##############################################################################
sampleFilesMAold <- list.files(directoryMAold)
sampleInfoMAold <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_sample_data.csv",header=TRUE)
sampleTableMAold <- data.frame(sampleName = sampleInfoMAold$sample,
                          fileName = sampleInfoMAold$filename,
                          condition = sampleInfoMAold$group)

sampleFilesMAold_1 <- list.files(directoryMAold_1)
sampleInfoMAold_1 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S1vsAnc.csv",header=TRUE)
sampleTableMAold_1 <- data.frame(sampleName = sampleInfoMAold_1$sample,
                          fileName = sampleInfoMAold_1$filename,
                          condition = sampleInfoMAold_1$group)

sampleFilesMAold_2 <- list.files(directoryMAold_2)
sampleInfoMAold_2 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S2vsAnc.csv",header=TRUE)
sampleTableMAold_2 <- data.frame(sampleName = sampleInfoMAold_2$sample,
                          fileName = sampleInfoMAold_2$filename,
                          condition = sampleInfoMAold_2$group)

sampleFilesMAold_3 <- list.files(directoryMAold_3)
sampleInfoMAold_3 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S3vsAnc.csv",header=TRUE)
sampleTableMAold_3 <- data.frame(sampleName = sampleInfoMAold_3$sample,
                          fileName = sampleInfoMAold_3$filename,
                          condition = sampleInfoMAold_3$group)

sampleFilesMAold_4 <- list.files(directoryMAold_4)
sampleInfoMAold_4 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S4vsAnc.csv",header=TRUE)
sampleTableMAold_4 <- data.frame(sampleName = sampleInfoMAold_4$sample,
                          fileName = sampleInfoMAold_4$filename,
                          condition = sampleInfoMAold_4$group)

sampleFilesMAold_5 <- list.files(directoryMAold_5)
sampleInfoMAold_5 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S5vsAnc.csv",header=TRUE)
sampleTableMAold_5 <- data.frame(sampleName = sampleInfoMAold_5$sample,
                          fileName = sampleInfoMAold_5$filename,
                          condition = sampleInfoMAold_5$group)

sampleFilesMAold_7 <- list.files(directoryMAold_7)
sampleInfoMAold_7 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S7vsAnc.csv",header=TRUE)
sampleTableMAold_7 <- data.frame(sampleName = sampleInfoMAold_7$sample,
                          fileName = sampleInfoMAold_7$filename,
                          condition = sampleInfoMAold_7$group)

sampleFilesMAold_8 <- list.files(directoryMAold_8)
sampleInfoMAold_8 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S8vsAnc.csv",header=TRUE)
sampleTableMAold_8 <- data.frame(sampleName = sampleInfoMAold_8$sample,
                          fileName = sampleInfoMAold_8$filename,
                          condition = sampleInfoMAold_8$group)

sampleFilesMAold_9 <- list.files(directoryMAold_9)
sampleInfoMAold_9 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S9vsAnc.csv",header=TRUE)
sampleTableMAold_9 <- data.frame(sampleName = sampleInfoMAold_9$sample,
                          fileName = sampleInfoMAold_9$filename,
                          condition = sampleInfoMAold_9$group)

sampleFilesMAold_11 <- list.files(directoryMAold_11)
sampleInfoMAold_11 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S11vsAnc.csv",header=TRUE)
sampleTableMAold_11 <- data.frame(sampleName = sampleInfoMAold_11$sample,
                          fileName = sampleInfoMAold_11$filename,
                          condition = sampleInfoMAold_11$group)

sampleFilesMAold_15 <- list.files(directoryMAold_15)
sampleInfoMAold_15 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S15vsAnc.csv",header=TRUE)
sampleTableMAold_15 <- data.frame(sampleName = sampleInfoMAold_15$sample,
                          fileName = sampleInfoMAold_15$filename,
                          condition = sampleInfoMAold_15$group)

sampleFilesMAold_28 <- list.files(directoryMAold_28)
sampleInfoMAold_28 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S28vsAnc.csv",header=TRUE)
sampleTableMAold_28 <- data.frame(sampleName = sampleInfoMAold_28$sample,
                          fileName = sampleInfoMAold_28$filename,
                          condition = sampleInfoMAold_28$group)

sampleFilesMAold_88 <- list.files(directoryMAold_88)
sampleInfoMAold_88 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S88vsAnc.csv",header=TRUE)
sampleTableMAold_88 <- data.frame(sampleName = sampleInfoMAold_88$sample,
                          fileName = sampleInfoMAold_88$filename,
                          condition = sampleInfoMAold_88$group)

sampleFilesMAold_108 <- list.files(directoryMAold_108)
sampleInfoMAold_108 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S108vsAnc.csv",header=TRUE)
sampleTableMAold_108 <- data.frame(sampleName = sampleInfoMAold_108$sample,
                          fileName = sampleInfoMAold_108$filename,
                          condition = sampleInfoMAold_108$group)

sampleFilesMAold_119 <- list.files(directoryMAold_119)
sampleInfoMAold_119 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S119vsAnc.csv",header=TRUE)
sampleTableMAold_119 <- data.frame(sampleName = sampleInfoMAold_119$sample,
                          fileName = sampleInfoMAold_119$filename,
                          condition = sampleInfoMAold_119$group)

###############################################################################
sampleFilesMAnew <- list.files(directoryMAnew)
sampleInfoMAnew <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_sample_data.csv",header=TRUE)
sampleTableMAnew <- data.frame(sampleName = sampleInfoMAnew$sample,
                          fileName = sampleInfoMAnew$filename,
                          run = sampleInfoMAnew$seq_run,
                           condition = sampleInfoMAnew$group)

sampleFilesMAnewS29 <- list.files(directoryMAnewS29)
sampleInfoMAnewS29 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S29.csv",header=TRUE)
sampleTableMAnewS29 <- data.frame(sampleName = sampleInfoMAnewS29$sample,
                          fileName = sampleInfoMAnewS29$filename,
                          run = sampleInfoMAnewS29$seq_run,
                           condition = sampleInfoMAnewS29$group)

sampleFilesMAnewS50 <- list.files(directoryMAnewS50)
sampleInfoMAnewS50 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S50.csv",header=TRUE)
sampleTableMAnewS50 <- data.frame(sampleName = sampleInfoMAnewS50$sample,
                          fileName = sampleInfoMAnewS50$filename,
                          run = sampleInfoMAnewS50$seq_run,
                           condition = sampleInfoMAnewS50$group)

sampleFilesMAnewS112 <- list.files(directoryMAnewS112)
sampleInfoMAnewS112 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S112.csv",header=TRUE)
sampleTableMAnewS112 <- data.frame(sampleName = sampleInfoMAnewS112$sample,
                          fileName = sampleInfoMAnewS112$filename,
                          run = sampleInfoMAnewS112$seq_run,
                           condition = sampleInfoMAnewS112$group)

sampleFilesMAnewS115 <- list.files(directoryMAnewS115)
sampleInfoMAnewS115 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S115.csv",header=TRUE)
sampleTableMAnewS115 <- data.frame(sampleName = sampleInfoMAnewS115$sample,
                          fileName = sampleInfoMAnewS115$filename,
                          run = sampleInfoMAnewS115$seq_run,
                           condition = sampleInfoMAnewS115$group)

sampleFilesMAnewS117 <- list.files(directoryMAnewS117)
sampleInfoMAnewS117 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S117.csv",header=TRUE)
sampleTableMAnewS117 <- data.frame(sampleName = sampleInfoMAnewS117$sample,
                          fileName = sampleInfoMAnewS117$filename,
                          run = sampleInfoMAnewS117$seq_run,
                           condition = sampleInfoMAnewS117$group)

sampleFilesMAnewS123 <- list.files(directoryMAnewS123)
sampleInfoMAnewS123 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S123.csv",header=TRUE)
sampleTableMAnewS123 <- data.frame(sampleName = sampleInfoMAnewS123$sample,
                          fileName = sampleInfoMAnewS123$filename,
                          run = sampleInfoMAnewS123$seq_run,
                           condition = sampleInfoMAnewS123$group)

sampleFilesMAnewS152 <- list.files(directoryMAnewS152)
sampleInfoMAnewS152 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S152.csv",header=TRUE)
sampleTableMAnewS152 <- data.frame(sampleName = sampleInfoMAnewS152$sample,
                          fileName = sampleInfoMAnewS152$filename,
                          run = sampleInfoMAnewS152$seq_run,
                           condition = sampleInfoMAnewS152$group)

```

Other files for our analysis, including ASR, ESR, and dosage-sensitive gene lists
```{r other files,eval=TRUE,include=FALSE}
esrGenes <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/ESR_genes/ESR_genes_Gasch2000.csv",header=FALSE)
as.data.frame(esrGenes)
names(esrGenes)[1] <- "GENEID"

dosageSensitiveGenes <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/DENonDEGenes/dosage_sensitive_genes.csv")
as.data.frame(dosageSensitiveGenes)
names(dosageSensitiveGenes)[1] <- "GENEID"

ASRgenes <- read.table("/Users/hollymcqueary/Dropbox/DosageCompensation/ESR_genes/ASRGenes.txt",header=TRUE)
as.data.frame(ASRgenes)

```

Then we build the *DESeqDataSets* using the following function:

```{r hsteqDds,eval=TRUE,include=TRUE}
library("DESeq2")
sampleTableGC_All$run <- as.factor(sampleTableGC_All$run)
sampleTableGC_All$sampleName <- as.factor(sampleTableGC_All$sampleName)
sampleTableGC_All$condition <- as.factor(sampleTableGC_All$condition)
ddsHTSeqGC_All <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_All,
                                       directory = directoryGC_All,
                                       design= ~ condition + run)
ddsHTSeqGC_All

# write.csv(counts(ddsHTSeqGC_All),file="counts.DEseqGC.csv")

############################# Lines Individually ########################################
sampleTableGC_1$run <- as.factor(sampleTableGC_1$run)
sampleTableGC_1$sampleName <- as.factor(sampleTableGC_1$sampleName)
sampleTableGC_1$condition <- as.factor(sampleTableGC_1$condition)
ddsHTSeqGC_1 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_1,
                                       directory = directoryGC_1,
                                       design= ~ condition + run)
ddsHTSeqGC_1

# write.csv(counts(ddsHTSeqGC_1),file="counts.DEseqGC_S1.csv")
vsdGC_1 <- vst(ddsHTSeqGC_1,blind=FALSE)

# library(ggplot2)
# pdf("PCA_GC_by_run_1.pdf")
# ggplot(pcaDataGC_1,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_1[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_1[2],"% variance"))
#   # coord_fixed()
#   dev.off()
########## line 2
sampleTableGC_2$run <- as.factor(sampleTableGC_2$run)
sampleTableGC_2$sampleName <- as.factor(sampleTableGC_2$sampleName)
sampleTableGC_2$condition <- as.factor(sampleTableGC_2$condition)
ddsHTSeqGC_2 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_2,
                                       directory = directoryGC_2,
                                       design= ~ condition + run)
ddsHTSeqGC_2


#### line 3
sampleTableGC_3$run <- as.factor(sampleTableGC_3$run)
sampleTableGC_3$sampleName <- as.factor(sampleTableGC_3$sampleName)
sampleTableGC_3$condition <- as.factor(sampleTableGC_3$condition)
ddsHTSeqGC_3 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_3,
                                       directory = directoryGC_3,
                                       design= ~ condition + run)
ddsHTSeqGC_3

# write.csv(counts(ddsHTSeqGC_3),file="counts.DEseqGC_S1.csv")

######## line 4
sampleTableGC_4$run <- as.factor(sampleTableGC_4$run)
sampleTableGC_4$sampleName <- as.factor(sampleTableGC_4$sampleName)
sampleTableGC_4$condition <- as.factor(sampleTableGC_4$condition)
ddsHTSeqGC_4 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_4,
                                       directory = directoryGC_4,
                                       design= ~ condition + run)
ddsHTSeqGC_4


################## line 5
  sampleTableGC_5$run <- as.factor(sampleTableGC_5$run)
sampleTableGC_5$sampleName <- as.factor(sampleTableGC_5$sampleName)
sampleTableGC_5$condition <- as.factor(sampleTableGC_5$condition)
ddsHTSeqGC_5 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_5,
                                       directory = directoryGC_5,
                                       design= ~ condition)# + run)
ddsHTSeqGC_5

# write.csv(counts(ddsHTSeqGC_5),file="counts.DEseqGC_S1.csv")

################## line 7
sampleTableGC_7$run <- as.factor(sampleTableGC_7$run)
sampleTableGC_7$sampleName <- as.factor(sampleTableGC_7$sampleName)
sampleTableGC_7$condition <- as.factor(sampleTableGC_7$condition)
ddsHTSeqGC_7 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_7,
                                       directory = directoryGC_7,
                                       design= ~ condition + run)
ddsHTSeqGC_7


#################### line 8
  sampleTableGC_8$run <- as.factor(sampleTableGC_8$run)
sampleTableGC_8$sampleName <- as.factor(sampleTableGC_8$sampleName)
sampleTableGC_8$condition <- as.factor(sampleTableGC_8$condition)
ddsHTSeqGC_8 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_8,
                                       directory = directoryGC_8,
                                       design= ~ condition + run)
ddsHTSeqGC_8


################# line 9
  sampleTableGC_9$run <- as.factor(sampleTableGC_9$run)
sampleTableGC_9$sampleName <- as.factor(sampleTableGC_9$sampleName)
sampleTableGC_9$condition <- as.factor(sampleTableGC_9$condition)
ddsHTSeqGC_9 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_9,
                                       directory = directoryGC_9,
                                       design= ~ condition + run)
ddsHTSeqGC_9


################## line 11
  sampleTableGC_11$run <- as.factor(sampleTableGC_11$run)
sampleTableGC_11$sampleName <- as.factor(sampleTableGC_11$sampleName)
sampleTableGC_11$condition <- as.factor(sampleTableGC_11$condition)
ddsHTSeqGC_11 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_11,
                                       directory = directoryGC_11,
                                       design= ~ condition + run)
ddsHTSeqGC_11


##################### line 18
sampleTableGC_18$run <- as.factor(sampleTableGC_18$run)
sampleTableGC_18$sampleName <- as.factor(sampleTableGC_18$sampleName)
sampleTableGC_18$condition <- as.factor(sampleTableGC_18$condition)
ddsHTSeqGC_18 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_18,
                                       directory = directoryGC_18,
                                       design= ~ condition + run)
ddsHTSeqGC_18


  ##################### line 31
sampleTableGC_31$run <- as.factor(sampleTableGC_31$run)
sampleTableGC_31$sampleName <- as.factor(sampleTableGC_31$sampleName)
sampleTableGC_31$condition <- as.factor(sampleTableGC_31$condition)
ddsHTSeqGC_31 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_31,
                                       directory = directoryGC_31,
                                       design= ~ condition + run)
ddsHTSeqGC_31


############## line 49
  sampleTableGC_49$run <- as.factor(sampleTableGC_49$run)
sampleTableGC_49$sampleName <- as.factor(sampleTableGC_49$sampleName)
sampleTableGC_49$condition <- as.factor(sampleTableGC_49$condition)
ddsHTSeqGC_49 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_49,
                                       directory = directoryGC_49,
                                       design= ~ condition + run)
ddsHTSeqGC_49


#################### line 59
sampleTableGC_59$run <- as.factor(sampleTableGC_59$run)
sampleTableGC_59$sampleName <- as.factor(sampleTableGC_59$sampleName)
sampleTableGC_59$condition <- as.factor(sampleTableGC_59$condition)
ddsHTSeqGC_59 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_59,
                                       directory = directoryGC_59,
                                       design= ~ condition + run)
ddsHTSeqGC_59


####################### line 61
  sampleTableGC_61$run <- as.factor(sampleTableGC_61$run)
sampleTableGC_61$sampleName <- as.factor(sampleTableGC_61$sampleName)
sampleTableGC_61$condition <- as.factor(sampleTableGC_61$condition)
ddsHTSeqGC_61 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_61,
                                       directory = directoryGC_61,
                                       design= ~ condition + run)
ddsHTSeqGC_61


  ####################### line 69
  sampleTableGC_69$run <- as.factor(sampleTableGC_69$run)
sampleTableGC_69$sampleName <- as.factor(sampleTableGC_69$sampleName)
sampleTableGC_69$condition <- as.factor(sampleTableGC_69$condition)
ddsHTSeqGC_69 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_69,
                                       directory = directoryGC_69,
                                       design= ~ condition + run)
ddsHTSeqGC_69

################### line 76
  sampleTableGC_76$run <- as.factor(sampleTableGC_76$run)
sampleTableGC_76$sampleName <- as.factor(sampleTableGC_76$sampleName)
sampleTableGC_76$condition <- as.factor(sampleTableGC_76$condition)
ddsHTSeqGC_76 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_76,
                                       directory = directoryGC_76,
                                       design= ~ condition + run)
ddsHTSeqGC_76


################## line 77
sampleTableGC_77$run <- as.factor(sampleTableGC_77$run)
sampleTableGC_77$sampleName <- as.factor(sampleTableGC_77$sampleName)
sampleTableGC_77$condition <- as.factor(sampleTableGC_77$condition)
ddsHTSeqGC_77 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_77,
                                       directory = directoryGC_77,
                                       design= ~ condition + run)
ddsHTSeqGC_77

# write.csv(counts(ddsHTSeqGC_77),file="counts.DEseqGC_S1.csv")
 


############################################################
  #### MA Old Samples ####
sampleTableMAold$sampleName <- as.factor(sampleTableMAold$sampleName)
sampleTableMAold$condition <- as.factor(sampleTableMAold$condition)
ddsHTSeqMAold <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold,
                                       directory = directoryMAold,
                                       design= ~ condition)

ddsHTSeqMAold
#
# write.csv(counts(ddsHTSeqMAold),file="counts.DEseqMAold.csv")



##################################### Lines Individually ########################
  ######## Line 1
sampleTableMAold_1$sampleName <- as.factor(sampleTableMAold_1$sampleName)
sampleTableMAold_1$condition <- as.factor(sampleTableMAold_1$condition)
ddsHTSeqMAold_1 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_1,
                                       directory = directoryMAold_1,
                                       design= ~ condition)

ddsHTSeqMAold_1
  ######## Line 2
sampleTableMAold_2$sampleName <- as.factor(sampleTableMAold_2$sampleName)
sampleTableMAold_2$condition <- as.factor(sampleTableMAold_2$condition)
ddsHTSeqMAold_2 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_2,
                                       directory = directoryMAold_2,
                                       design= ~ condition)

ddsHTSeqMAold_2
  ######## Line 3
sampleTableMAold_3$sampleName <- as.factor(sampleTableMAold_3$sampleName)
sampleTableMAold_3$condition <- as.factor(sampleTableMAold_3$condition)
ddsHTSeqMAold_3 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_3,
                                       directory = directoryMAold_3,
                                       design= ~ condition)

ddsHTSeqMAold_3
  ######## Line 4
sampleTableMAold_4$sampleName <- as.factor(sampleTableMAold_4$sampleName)
sampleTableMAold_4$condition <- as.factor(sampleTableMAold_4$condition)
ddsHTSeqMAold_4 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_4,
                                       directory = directoryMAold_4,
                                       design= ~ condition)

ddsHTSeqMAold_4

  ######## Line 5
sampleTableMAold_5$sampleName <- as.factor(sampleTableMAold_5$sampleName)
sampleTableMAold_5$condition <- as.factor(sampleTableMAold_5$condition)
ddsHTSeqMAold_5 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_5,
                                       directory = directoryMAold_5,
                                       design= ~ condition)

ddsHTSeqMAold_5

  ######## Line 7
sampleTableMAold_7$sampleName <- as.factor(sampleTableMAold_7$sampleName)
sampleTableMAold_7$condition <- as.factor(sampleTableMAold_7$condition)
ddsHTSeqMAold_7 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_7,
                                       directory = directoryMAold_7,
                                       design= ~ condition)

ddsHTSeqMAold_7

  ######## Line 8
sampleTableMAold_8$sampleName <- as.factor(sampleTableMAold_8$sampleName)
sampleTableMAold_8$condition <- as.factor(sampleTableMAold_8$condition)
ddsHTSeqMAold_8 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_8,
                                       directory = directoryMAold_8,
                                       design= ~ condition)

ddsHTSeqMAold_8

  ######## Line 9
sampleTableMAold_9$sampleName <- as.factor(sampleTableMAold_9$sampleName)
sampleTableMAold_9$condition <- as.factor(sampleTableMAold_9$condition)
ddsHTSeqMAold_9 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_9,
                                       directory = directoryMAold_9,
                                       design= ~ condition)

ddsHTSeqMAold_9

  ######## Line 11
sampleTableMAold_11$sampleName <- as.factor(sampleTableMAold_11$sampleName)
sampleTableMAold_11$condition <- as.factor(sampleTableMAold_11$condition)
ddsHTSeqMAold_11 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_11,
                                       directory = directoryMAold_11,
                                       design= ~ condition)

ddsHTSeqMAold_11

  ######## Line 15
sampleTableMAold_15$sampleName <- as.factor(sampleTableMAold_15$sampleName)
sampleTableMAold_15$condition <- as.factor(sampleTableMAold_15$condition)
ddsHTSeqMAold_15 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_15,
                                       directory = directoryMAold_15,
                                       design= ~ condition)

ddsHTSeqMAold_15

  ######## Line 28
sampleTableMAold_28$sampleName <- as.factor(sampleTableMAold_28$sampleName)
sampleTableMAold_28$condition <- as.factor(sampleTableMAold_28$condition)
ddsHTSeqMAold_28 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_28,
                                       directory = directoryMAold_28,
                                       design= ~ condition)

ddsHTSeqMAold_28

  ######## Line 88
sampleTableMAold_88$sampleName <- as.factor(sampleTableMAold_88$sampleName)
sampleTableMAold_88$condition <- as.factor(sampleTableMAold_88$condition)
ddsHTSeqMAold_88 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_88,
                                       directory = directoryMAold_88,
                                       design= ~ condition)

ddsHTSeqMAold_88

  ######## Line 108
sampleTableMAold_108$sampleName <- as.factor(sampleTableMAold_108$sampleName)
sampleTableMAold_108$condition <- as.factor(sampleTableMAold_108$condition)
ddsHTSeqMAold_108 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_108,
                                       directory = directoryMAold_108,
                                       design= ~ condition)

ddsHTSeqMAold_108

  ######## Line 119
sampleTableMAold_119$sampleName <- as.factor(sampleTableMAold_119$sampleName)
sampleTableMAold_119$condition <- as.factor(sampleTableMAold_119$condition)
ddsHTSeqMAold_119 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_119,
                                       directory = directoryMAold_119,
                                       design= ~ condition)

ddsHTSeqMAold_119

############################################################
# sampleTableMAnew$sampleName <- as.factor(sampleTableMAnew$sampleName)
# sampleTableMAnew$condition <- as.factor(sampleTableMAnew$condition)
# sampleTableMAnew$run <- as.factor(sampleTableMAnew$run)
# ddsHTSeqMAnew.run <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnew,
#                                        directory = directoryMAnew,
#                                        design= ~ condition + run)
#
# ddsHTSeqMAnew.run
# counts(ddsHTSeqMAnew.run)
# write.csv(counts(ddsHTSeqMAnew.run),file="counts.DEseqMAnew.run.csv")
ddsHTSeqMAnew <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnew,
                                       directory = directoryMAnew,
                                       design= ~ condition)

ddsHTSeqMAnew
# counts(ddsHTSeqMAnew)
# write.csv(counts(ddsHTSeqMAnew),file="counts.DEseqMAnew.csv")
#
#
# #### PCA
sampleTableMAnewS152$sampleName <- as.factor(sampleTableMAnewS152$sampleName)
sampleTableMAnewS152$condition <- as.factor(sampleTableMAnewS152$condition)
ddsHTSeqMAnewS152 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS152,
                                       directory = directoryMAnewS152,
                                       design= ~ condition + run)
#### subset the DESeq dataset into individual samples
sampleTableMAnewS29$sampleName <- as.factor(sampleTableMAnewS29$sampleName)
sampleTableMAnewS29$condition <- as.factor(sampleTableMAnewS29$condition)
ddsHTSeqMAnewS29 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS29,
                                       directory = directoryMAnewS29,
                                       design= ~ condition + run)

sampleTableMAnewS50$sampleName <- as.factor(sampleTableMAnewS50$sampleName)
sampleTableMAnewS50$condition <- as.factor(sampleTableMAnewS50$condition)
ddsHTSeqMAnewS50 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS50,
                                       directory = directoryMAnewS50,
                                       design= ~ condition + run)

sampleTableMAnewS112$sampleName <- as.factor(sampleTableMAnewS112$sampleName)
sampleTableMAnewS112$condition <- as.factor(sampleTableMAnewS112$condition)
ddsHTSeqMAnewS112 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS112,
                                       directory = directoryMAnewS112,
                                       design= ~ condition + run)

sampleTableMAnewS115$sampleName <- as.factor(sampleTableMAnewS115$sampleName)
sampleTableMAnewS115$condition <- as.factor(sampleTableMAnewS115$condition)
ddsHTSeqMAnewS115 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS115,
                                       directory = directoryMAnewS115,
                                       design= ~ condition + run)

sampleTableMAnewS117$sampleName <- as.factor(sampleTableMAnewS117$sampleName)
sampleTableMAnewS117$condition <- as.factor(sampleTableMAnewS117$condition)
ddsHTSeqMAnewS117 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS117,
                                       directory = directoryMAnewS117,
                                       design= ~ condition + run)

sampleTableMAnewS123$sampleName <- as.factor(sampleTableMAnewS123$sampleName)
sampleTableMAnewS123$condition <- as.factor(sampleTableMAnewS123$condition)
ddsHTSeqMAnewS123 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS123,
                                       directory = directoryMAnewS123,
                                       design= ~ condition)

sampleTableMAnewS152$sampleName <- as.factor(sampleTableMAnewS152$sampleName)
sampleTableMAnewS152$condition <- as.factor(sampleTableMAnewS152$condition)
ddsHTSeqMAnewS152 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS152,
                                       directory = directoryMAnewS152,
                                       design= ~ condition)

```


```{r PCAs,include=TRUE,echo=FALSE,eval=TRUE}
#### PCA analysis of samples
# write.csv(counts(ddsHTSeqGC_2),file="counts.DEseqGC_S1.csv")
vsdGC_2 <- vst(ddsHTSeqGC_2,blind=FALSE)
rldGC_2 <- rlog(ddsHTSeqGC_2,blind=FALSE)
#
# ddsHTSeqGC_2$condition
# levels(ddsHTSeqGC_2$condition)
# head(assay(vsdGC_2),3)
plotPCA(vsdGC_2, intgroup=c("condition"))
pcaDataGC_2 <- plotPCA(vsdGC_2,intgroup=c("condition"),returnData=TRUE)
percentVar_2 <- round(100 * attr(pcaDataGC_2,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_2.pdf")
ggplot(pcaDataGC_2,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_2[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_2[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
  plotPCA(rldGC_2, intgroup=c("condition"))
pcaDataGcRld_2 <- plotPCA(rldGC_2,intgroup=c("condition"),returnData=TRUE)
percentVarRld_2 <- round(100 * attr(pcaDataGcRld_2,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_2.pdf")
ggplot(pcaDataGcRld_2,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_2[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_2[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_2)
plotPCA(vsdGC_2, intgroup=c("run"))
pcaDataGC_2 <- plotPCA(vsdGC_2,intgroup=c("run"),returnData=TRUE)
percentVar_2 <- round(100 * attr(pcaDataGC_2,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_2.pdf")
ggplot(pcaDataGC_2,aes(PC1,PC2,color=run)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_2[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_2[2],"% variance"))
  # coord_fixed()
  # dev.off()
vsdGC_3 <- vst(ddsHTSeqGC_3,blind=FALSE)
rldGC_3 <- rlog(ddsHTSeqGC_3,blind=FALSE)

ddsHTSeqGC_3$condition
levels(ddsHTSeqGC_3$condition)
head(assay(vsdGC_3),3)
plotPCA(vsdGC_3, intgroup=c("condition"))
pcaDataGC_3 <- plotPCA(vsdGC_3,intgroup=c("condition"),returnData=TRUE)
percentVar_3 <- round(100 * attr(pcaDataGC_3,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_3.pdf")
ggplot(pcaDataGC_3,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_3[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_3[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
  plotPCA(rldGC_3, intgroup=c("condition"))
pcaDataGcRld_3 <- plotPCA(rldGC_3,intgroup=c("condition"),returnData=TRUE)
percentVarRld_3 <- round(100 * attr(pcaDataGcRld_3,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_3.pdf")
ggplot(pcaDataGcRld_3,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_3[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_3[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
colData(ddsHTSeqGC_3)
plotPCA(vsdGC_3, intgroup=c("run"))
pcaDataGC_3 <- plotPCA(vsdGC_3,intgroup=c("run"),returnData=TRUE)
percentVar_3 <- round(100 * attr(pcaDataGC_3,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_3.pdf")
ggplot(pcaDataGC_3,aes(PC1,PC2,color=run)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_3[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_3[2],"% variance"))
  # coord_fixed()
  # dev.off()
# write.csv(counts(ddsHTSeqGC_4),file="counts.DEseqGC_S1.csv")
vsdGC_4 <- vst(ddsHTSeqGC_4,blind=FALSE)
rldGC_4 <- rlog(ddsHTSeqGC_4,blind=FALSE)

ddsHTSeqGC_4$condition
levels(ddsHTSeqGC_4$condition)
head(assay(vsdGC_4),3)
plotPCA(vsdGC_4, intgroup=c("condition"))
pcaDataGC_4 <- plotPCA(vsdGC_4,intgroup=c("condition"),returnData=TRUE)
percentVar_4 <- round(100 * attr(pcaDataGC_4,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_4.pdf")
ggplot(pcaDataGC_4,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_4[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_4[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
  plotPCA(rldGC_4, intgroup=c("condition"))
pcaDataGcRld_4 <- plotPCA(rldGC_4,intgroup=c("condition"),returnData=TRUE)
percentVarRld_4 <- round(100 * attr(pcaDataGcRld_4,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_4.pdf")
ggplot(pcaDataGcRld_4,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_4[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_4[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
colData(ddsHTSeqGC_4)
plotPCA(vsdGC_4, intgroup=c("run"))
pcaDataGC_4 <- plotPCA(vsdGC_4,intgroup=c("run"),returnData=TRUE)
percentVar_4 <- round(100 * attr(pcaDataGC_4,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_4.pdf")
ggplot(pcaDataGC_4,aes(PC1,PC2,color=run)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_4[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_4[2],"% variance"))
  # coord_fixed()
  # dev.off()
vsdGC_5 <- vst(ddsHTSeqGC_5,blind=FALSE)
rldGC_5 <- rlog(ddsHTSeqGC_5,blind=FALSE)

ddsHTSeqGC_5$condition
levels(ddsHTSeqGC_5$condition)
head(assay(vsdGC_5),3)
plotPCA(vsdGC_5, intgroup=c("condition"))
pcaDataGC_5 <- plotPCA(vsdGC_5,intgroup=c("condition"),returnData=TRUE)
percentVar_5 <- round(100 * attr(pcaDataGC_5,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_5.pdf")
ggplot(pcaDataGC_5,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_5[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_5[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
  plotPCA(rldGC_5, intgroup=c("condition"))
pcaDataGcRld_5 <- plotPCA(rldGC_5,intgroup=c("condition"),returnData=TRUE)
percentVarRld_5 <- round(100 * attr(pcaDataGcRld_5,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_rld_5.pdf")
ggplot(pcaDataGcRld_5,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_5[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_5[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
colData(ddsHTSeqGC_5)
plotPCA(vsdGC_5, intgroup=c("run"))
pcaDataGC_5 <- plotPCA(vsdGC_5,intgroup=c("run"),returnData=TRUE)
percentVar_5 <- round(100 * attr(pcaDataGC_5,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_by_run_5.pdf")
ggplot(pcaDataGC_5,aes(PC1,PC2,color=run)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_5[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_5[2],"% variance"))
  # coord_fixed()
  # dev.off()


vsdGC_7 <- vst(ddsHTSeqGC_7,blind=FALSE)
rldGC_7 <- rlog(ddsHTSeqGC_7,blind=FALSE)

ddsHTSeqGC_7$condition
levels(ddsHTSeqGC_7$condition)
head(assay(vsdGC_7),3)
plotPCA(vsdGC_7, intgroup=c("condition"))
pcaDataGC_7 <- plotPCA(vsdGC_7,intgroup=c("condition"),returnData=TRUE)
percentVar_7 <- round(100 * attr(pcaDataGC_7,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_7.pdf")
ggplot(pcaDataGC_7,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_7[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_7[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
  plotPCA(rldGC_7, intgroup=c("condition"))
pcaDataGcRld_7 <- plotPCA(rldGC_7,intgroup=c("condition"),returnData=TRUE)
percentVarRld_7 <- round(100 * attr(pcaDataGcRld_7,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_7.pdf")
ggplot(pcaDataGcRld_7,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_7[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_7[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
colData(ddsHTSeqGC_7)
plotPCA(vsdGC_7, intgroup=c("run"))
pcaDataGC_7 <- plotPCA(vsdGC_7,intgroup=c("run"),returnData=TRUE)
percentVar_7 <- round(100 * attr(pcaDataGC_7,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_7.pdf")
ggplot(pcaDataGC_7,aes(PC1,PC2,color=run)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_7[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_7[2],"% variance"))
  # coord_fixed()
  # dev.off()
# write.csv(counts(ddsHTSeqGC_8),file="counts.DEseqGC_S1.csv")
vsdGC_8 <- vst(ddsHTSeqGC_8,blind=FALSE)
rldGC_8 <- rlog(ddsHTSeqGC_8,blind=FALSE)

ddsHTSeqGC_8$condition
levels(ddsHTSeqGC_8$condition)
head(assay(vsdGC_8),3)
plotPCA(vsdGC_8, intgroup=c("condition"))
pcaDataGC_8 <- plotPCA(vsdGC_8,intgroup=c("condition"),returnData=TRUE)
percentVar_8 <- round(100 * attr(pcaDataGC_8,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_8.pdf")
ggplot(pcaDataGC_8,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_8[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_8[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
  plotPCA(rldGC_8, intgroup=c("condition"))
pcaDataGcRld_8 <- plotPCA(rldGC_8,intgroup=c("condition"),returnData=TRUE)
percentVarRld_8 <- round(100 * attr(pcaDataGcRld_8,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_8.pdf")
ggplot(pcaDataGcRld_8,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_8[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_8[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
colData(ddsHTSeqGC_8)
plotPCA(vsdGC_8, intgroup=c("run"))
pcaDataGC_8 <- plotPCA(vsdGC_8,intgroup=c("run"),returnData=TRUE)
percentVar_8 <- round(100 * attr(pcaDataGC_8,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_8.pdf")
ggplot(pcaDataGC_8,aes(PC1,PC2,color=run)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_8[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_8[2],"% variance"))
  # coord_fixed()
  # dev.off()

# write.csv(counts(ddsHTSeqGC_11),file="counts.DEseqGC_S1.csv")
vsdGC_11 <- vst(ddsHTSeqGC_11,blind=FALSE)
rldGC_11 <- rlog(ddsHTSeqGC_11,blind=FALSE)

ddsHTSeqGC_11$condition
levels(ddsHTSeqGC_11$condition)
head(assay(vsdGC_11),3)
plotPCA(vsdGC_11, intgroup=c("condition"))
pcaDataGC_11 <- plotPCA(vsdGC_11,intgroup=c("condition"),returnData=TRUE)
percentVar_11 <- round(100 * attr(pcaDataGC_11,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_11.pdf")
ggplot(pcaDataGC_11,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_11[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_11[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
  plotPCA(rldGC_11, intgroup=c("condition"))
pcaDataGcRld_11 <- plotPCA(rldGC_11,intgroup=c("condition"),returnData=TRUE)
percentVarRld_11 <- round(100 * attr(pcaDataGcRld_11,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_rld_11.pdf")
ggplot(pcaDataGcRld_11,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_11[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_11[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
colData(ddsHTSeqGC_11)
plotPCA(vsdGC_11, intgroup=c("run"))
pcaDataGC_11 <- plotPCA(vsdGC_11,intgroup=c("run"),returnData=TRUE)
percentVar_11 <- round(100 * attr(pcaDataGC_11,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_by_run_11.pdf")
ggplot(pcaDataGC_11,aes(PC1,PC2,color=run)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_11[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_11[2],"% variance"))
  # coord_fixed()
  # dev.off()
# write.csv(counts(ddsHTSeqGC_18),file="counts.DEseqGC_S1.csv")
vsdGC_18 <- vst(ddsHTSeqGC_18,blind=FALSE)
rldGC_18 <- rlog(ddsHTSeqGC_18,blind=FALSE)

ddsHTSeqGC_18$condition
levels(ddsHTSeqGC_18$condition)
head(assay(vsdGC_18),3)
plotPCA(vsdGC_18, intgroup=c("condition"))
pcaDataGC_18 <- plotPCA(vsdGC_18,intgroup=c("condition"),returnData=TRUE)
percentVar_18 <- round(100 * attr(pcaDataGC_18,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_18.pdf")
ggplot(pcaDataGC_18,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_18[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_18[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
  plotPCA(rldGC_18, intgroup=c("condition"))
pcaDataGcRld_18 <- plotPCA(rldGC_18,intgroup=c("condition"),returnData=TRUE)
percentVarRld_18 <- round(100 * attr(pcaDataGcRld_18,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_rld_18.pdf")
ggplot(pcaDataGcRld_18,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_18[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_18[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
colData(ddsHTSeqGC_18)
plotPCA(vsdGC_18, intgroup=c("run"))
pcaDataGC_18 <- plotPCA(vsdGC_18,intgroup=c("run"),returnData=TRUE)
percentVar_18 <- round(100 * attr(pcaDataGC_18,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_by_run_18.pdf")
ggplot(pcaDataGC_18,aes(PC1,PC2,color=run)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_18[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_18[2],"% variance"))
  # coord_fixed()
  # dev.off()

vsdGC_49 <- vst(ddsHTSeqGC_49,blind=FALSE)
rldGC_49 <- rlog(ddsHTSeqGC_49,blind=FALSE)

ddsHTSeqGC_49$condition
levels(ddsHTSeqGC_49$condition)
head(assay(vsdGC_49),3)
plotPCA(vsdGC_49, intgroup=c("condition"))
pcaDataGC_49 <- plotPCA(vsdGC_49,intgroup=c("condition"),returnData=TRUE)
percentVar_49 <- round(100 * attr(pcaDataGC_49,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_49.pdf")
ggplot(pcaDataGC_49,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_49[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_49[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
  plotPCA(rldGC_49, intgroup=c("condition"))
pcaDataGcRld_49 <- plotPCA(rldGC_49,intgroup=c("condition"),returnData=TRUE)
percentVarRld_49 <- round(100 * attr(pcaDataGcRld_49,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_rld_49.pdf")
ggplot(pcaDataGcRld_49,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_49[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_49[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
colData(ddsHTSeqGC_49)
plotPCA(vsdGC_49, intgroup=c("run"))
pcaDataGC_49 <- plotPCA(vsdGC_49,intgroup=c("run"),returnData=TRUE)
percentVar_49 <- round(100 * attr(pcaDataGC_49,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_by_run_49.pdf")
ggplot(pcaDataGC_49,aes(PC1,PC2,color=run)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_49[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_49[2],"% variance"))
  # coord_fixed()
  # dev.off()
# write.csv(counts(ddsHTSeqGC_59),file="counts.DEseqGC_S1.csv")
vsdGC_59 <- vst(ddsHTSeqGC_59,blind=FALSE)
rldGC_59 <- rlog(ddsHTSeqGC_59,blind=FALSE)

ddsHTSeqGC_59$condition
levels(ddsHTSeqGC_59$condition)
head(assay(vsdGC_59),3)
plotPCA(vsdGC_59, intgroup=c("condition"))
pcaDataGC_59 <- plotPCA(vsdGC_59,intgroup=c("condition"),returnData=TRUE)
percentVar_59 <- round(100 * attr(pcaDataGC_59,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_59.pdf")
ggplot(pcaDataGC_59,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_59[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_59[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
  plotPCA(rldGC_59, intgroup=c("condition"))
pcaDataGcRld_59 <- plotPCA(rldGC_59,intgroup=c("condition"),returnData=TRUE)
percentVarRld_59 <- round(100 * attr(pcaDataGcRld_59,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_rld_59.pdf")
ggplot(pcaDataGcRld_59,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_59[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_59[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
colData(ddsHTSeqGC_59)
plotPCA(vsdGC_59, intgroup=c("run"))
pcaDataGC_59 <- plotPCA(vsdGC_59,intgroup=c("run"),returnData=TRUE)
percentVar_59 <- round(100 * attr(pcaDataGC_59,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_by_run_59.pdf")
ggplot(pcaDataGC_59,aes(PC1,PC2,color=run)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_59[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_59[2],"% variance"))
  # coord_fixed()
  # dev.off()
# transformation of count data
vsdGC_All <- vst(ddsHTSeqGC_All,blind=FALSE)
rldGC_All <- rlog(ddsHTSeqGC_All,blind=FALSE)

ddsHTSeqGC_All$condition
levels(ddsHTSeqGC_All$condition)
head(assay(vsdGC_All),3)
plotPCA(vsdGC_All, intgroup=c("condition"))
pcaDataGC_All <- plotPCA(vsdGC_All,intgroup=c("condition"),returnData=TRUE)
percentVar_All <- round(100 * attr(pcaDataGC_All,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_All.pdf")
ggplot(pcaDataGC_All,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_All[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_All[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()

## pca by sequencing run
colData(ddsHTSeqGC_All)
plotPCA(vsdGC_All, intgroup=c("run"))
pcaDataGC_All <- plotPCA(vsdGC_All,intgroup=c("run"),returnData=TRUE)
percentVar_All <- round(100 * attr(pcaDataGC_All,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_by_run_All.pdf")
ggplot(pcaDataGC_All,aes(PC1,PC2,color=run)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_All[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_All[2],"% variance"))
  # coord_fixed()
  # dev.off()

## PCA with rld data
  plotPCA(rldGC_All, intgroup=c("condition"))
pcaDataGcRld_All <- plotPCA(rldGC_All,intgroup=c("condition"),returnData=TRUE)
percentVarRld_All <- round(100 * attr(pcaDataGcRld_All,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_rld.pdf")
ggplot(pcaDataGcRld_All,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_All[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_All[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
# PCA with rld data by run
  # plotPCA(rldGC.run, intgroup=c("run"))
# pcaDataGcRldRun <- plotPCA(rldGC.run,intgroup=c("run"),returnData=TRUE)
# percentVarRldRun <- round(100 * attr(pcaDataGcRldRun,"percentVar"))
# library(ggplot2)
# # pdf("PCA_GC_rld_Run.pdf")
# ggplot(pcaDataGcRldRun,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRldRun[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRldRun[2],"% variance")) +
#   scale_fill_hue(l=80)
# # dev.off()

rldGC_1 <- rlog(ddsHTSeqGC_1,blind=FALSE)

ddsHTSeqGC_1$condition
levels(ddsHTSeqGC_1$condition)
head(assay(vsdGC_1),3)
plotPCA(vsdGC_1, intgroup=c("condition"))
pcaDataGC_1 <- plotPCA(vsdGC_1,intgroup=c("condition"),returnData=TRUE)
percentVar_1 <- round(100 * attr(pcaDataGC_1,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_1.pdf")
ggplot(pcaDataGC_1,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_1[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_1[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
  plotPCA(rldGC_1, intgroup=c("condition"))
pcaDataGcRld_1 <- plotPCA(rldGC_1,intgroup=c("condition"),returnData=TRUE)
percentVarRld_1 <- round(100 * attr(pcaDataGcRld_1,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_1.pdf")
ggplot(pcaDataGcRld_1,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_1[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_1[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
colData(ddsHTSeqGC_1)
plotPCA(vsdGC_1, intgroup=c("run"))
pcaDataGC_1 <- plotPCA(vsdGC_1,intgroup=c("run"),returnData=TRUE)
percentVar_1 <- round(100 * attr(pcaDataGC_1,"percentVar"))

vsdMAnew152 <- varianceStabilizingTransformation(ddsHTSeqMAnewS152,blind=FALSE)
rldMAnew152 <- rlog(ddsHTSeqMAnewS152,blind=FALSE)
head(assay(vsdMAnew152),3)
DESeq2::plotPCA(vsdMAnew152, intgroup=c("condition"))
pcaDataMAnewVsd152 <- DESeq2::plotPCA(vsdMAnew152,intgroup=c("condition"),returnData=TRUE)
pcaDataMAnewRld152 <- DESeq2::plotPCA(rldMAnew152,intgroup=c("condition"),returnData=TRUE)

percentVarMAnewVsd152 <- round(100 * attr(pcaDataMAnewVsd152,"percentVar"))
percentVarMAnewRld152 <- round(100 * attr(pcaDataMAnewRld152,"percentVar"))

# vsdMAnew.run <- vst(ddsHTSeqMAnew.run,blind=FALSE)
# rldMAnew.run <- rlog(ddsHTSeqMAnew.run,blind=FALSE)
# head(assay(vsdMAnew.run),3)

# library(ggplot2)
# pdf("MA_new_PCA_Line.pdf")
# ggplot(pcaDataMAnewRld,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarMAnew[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarMAnew[2],"% variance")) +
#   scale_fill_hue(l=80) +
#   scale_shape_manual(values=c(0,1,2,3,4,7,8,11))
# # dev.off()

# pdf("MA_new_PCA_Line_vsd152.pdf")
ggplot(pcaDataMAnewVsd152,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarMAnewVsd152[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarMAnewVsd152[2],"% variance")) +
  scale_fill_hue(l=80) +
  scale_shape_manual(values=c(0,1,2,3,4,7,8,11))
# dev.off()

# # pdf("MA_new_PCA_Line_rld.pdf")
# ggplot(pcaDataMAnewRld,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarMAnewRld[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarMAnewRld[2],"% variance")) +
#   scale_fill_hue(l=80) +
#   scale_shape_manual(values=c(0,1,2,3,4,7,8,11))
# # dev.off()

#can you do PCA on just the raw counts (no transformation)?
## apparently not
# pdf("MA_new_PCA_Line.pdf")
# ggplot(pcaDataMAnew,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarMAnew[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarMAnew[2],"% variance")) +
#   scale_fill_hue(l=80) +
#   scale_shape_manual(values=c(0,1,2,3,4,7,8,11))
# # dev.off()

# by sequencing run
# colData(ddsHTSeqMAnew.run)
# DESeq2::plotPCA(vsdMAnew.run, intgroup=c("run"))
# pcaDataMAnew.run <- DESeq2::plotPCA(vsdMAnew.run,intgroup=c("run"),returnData=TRUE)
# percentVarMAnew.run <- round(100 * attr(pcaDataMAnew.run,"percentVar"))
# library(ggplot2)
# pdf("MA_new_PCA_SeqRun.pdf")
# ggplot(pcaDataMAnew.run,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarMAnew.run[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarMAnew.run[2],"% variance")) +
#   scale_fill_hue(l=80)
# # dev.off()
# scale_shape_manual(values=c(0,1,2,3,4,7,8,11,15))

subsetMAnewS29 <- which(ddsHTSeqMAnew$condition=="29")
subsetMAnewAnc <- which(ddsHTSeqMAnew$condition=="Anc")
subsetMAnewS29 <- c(subsetMAnewS29,subsetMAnewAnc)
ddsHTSeqMAnewS29 <- ddsHTSeqMAnew[,subsetMAnewS29]
ddsHTSeqMAnewS29$condition
ddsHTSeqMAnewS29$condition <- droplevels(ddsHTSeqMAnewS29$condition)
ddsHTSeqMAnewS29$condition

subsetMAnewS112 <- which(ddsHTSeqMAnew$condition=="112")
subsetMAnewS112 <- c(subsetMAnewS112,subsetMAnewAnc)
ddsHTSeqMAnewS112 <- ddsHTSeqMAnew[,subsetMAnewS112]
ddsHTSeqMAnewS112$condition
ddsHTSeqMAnewS112$condition <- droplevels(ddsHTSeqMAnewS112$condition)
ddsHTSeqMAnewS112$condition

#
subsetMAnewS50 <- which(ddsHTSeqMAnew$condition=="50")
subsetMAnewS50 <- c(subsetMAnewS50,subsetMAnewAnc)
ddsHTSeqMAnewS50 <- ddsHTSeqMAnew[,subsetMAnewS50]
ddsHTSeqMAnewS50$condition
ddsHTSeqMAnewS50$condition <- droplevels(ddsHTSeqMAnewS50$condition)
ddsHTSeqMAnewS50$condition
subsetMAnewS115 <- which(ddsHTSeqMAnew$condition=="115")
subsetMAnewS115 <- c(subsetMAnewS115,subsetMAnewAnc)
ddsHTSeqMAnewS115 <- ddsHTSeqMAnew[,subsetMAnewS115]
ddsHTSeqMAnewS115$condition
ddsHTSeqMAnewS115$condition <- droplevels(ddsHTSeqMAnewS115$condition)
ddsHTSeqMAnewS115$condition
subsetMAnewS117 <- which(ddsHTSeqMAnew$condition=="117")
subsetMAnewS117 <- c(subsetMAnewS117,subsetMAnewAnc)
ddsHTSeqMAnewS117 <- ddsHTSeqMAnew[,subsetMAnewS117]
ddsHTSeqMAnewS117$condition
ddsHTSeqMAnewS117$condition <- droplevels(ddsHTSeqMAnewS117$condition)
ddsHTSeqMAnewS117$condition
subsetMAnewS123 <- which(ddsHTSeqMAnew$condition=="123")
subsetMAnewS123 <- c(subsetMAnewS123,subsetMAnewAnc)
ddsHTSeqMAnewS123 <- ddsHTSeqMAnew[,subsetMAnewS123]
ddsHTSeqMAnewS123$condition
ddsHTSeqMAnewS123$condition <- droplevels(ddsHTSeqMAnewS123$condition)
ddsHTSeqMAnewS123$condition
# subsetMAnewS152 <- which(ddsHTSeqMAnew$condition=="152")
# subsetMAnewS152 <- c(subsetMAnewS152,subsetMAnewAnc)
# ddsHTSeqMAnewS152 <- ddsHTSeqMAnew[,subsetMAnewS152]
ddsHTSeqMAnewS152$condition
ddsHTSeqMAnewS152$condition <- droplevels(ddsHTSeqMAnewS152$condition)
ddsHTSeqMAnewS152$condition


# subsetMAnewS29.run <- which(ddsHTSeqMAnew.run$condition=="29")
# subsetMAnewAnc.run <- which(ddsHTSeqMAnew.run$condition=="Anc")
# subsetMAnewS29.run <- c(subsetMAnewS29.run,subsetMAnewAnc.run)
# ddsHTSeqMAnewS29.run <- ddsHTSeqMAnew.run[,subsetMAnewS29.run]
# ddsHTSeqMAnewS29.run$condition
# ddsHTSeqMAnewS29.run$condition <- droplevels(ddsHTSeqMAnewS29.run$condition)
# ddsHTSeqMAnewS29.run$condition
# 
# subsetMAnewS112.run <- which(ddsHTSeqMAnew.run$condition=="112")
# subsetMAnewS112.run <- c(subsetMAnewS112.run,subsetMAnewAnc.run)
# ddsHTSeqMAnewS112.run <- ddsHTSeqMAnew.run[,subsetMAnewS112.run]
# ddsHTSeqMAnewS112.run$condition
# ddsHTSeqMAnewS112.run$condition <- droplevels(ddsHTSeqMAnewS112.run$condition)
# ddsHTSeqMAnewS112.run$condition


# subsetMAnewS50.run <- which(ddsHTSeqMAnew.run$condition=="50")
# subsetMAnewS50.run <- c(subsetMAnewS50.run,subsetMAnewAnc.run)
# ddsHTSeqMAnewS50.run <- ddsHTSeqMAnew.run[,subsetMAnewS50.run]
# ddsHTSeqMAnewS50.run$condition
# ddsHTSeqMAnewS50.run$condition <- droplevels(ddsHTSeqMAnewS50.run$condition)
# ddsHTSeqMAnewS50.run$condition
# subsetMAnewS115.run <- which(ddsHTSeqMAnew.run$condition=="115")
# subsetMAnewS115.run <- c(subsetMAnewS115.run,subsetMAnewAnc.run)
# ddsHTSeqMAnewS115.run <- ddsHTSeqMAnew.run[,subsetMAnewS115.run]
# ddsHTSeqMAnewS115.run$condition
# ddsHTSeqMAnewS115.run$condition <- droplevels(ddsHTSeqMAnewS115.run$condition)
# ddsHTSeqMAnewS115.run$condition
# subsetMAnewS117.run <- which(ddsHTSeqMAnew.run$condition=="117")
# subsetMAnewS117.run <- c(subsetMAnewS117.run,subsetMAnewAnc.run)
# ddsHTSeqMAnewS117.run <- ddsHTSeqMAnew.run[,subsetMAnewS117.run]
# ddsHTSeqMAnewS117.run$condition
# ddsHTSeqMAnewS117.run$condition <- droplevels(ddsHTSeqMAnewS117.run$condition)
# ddsHTSeqMAnewS117.run$condition
# subsetMAnewS123.run <- which(ddsHTSeqMAnew.run$condition=="123")
# subsetMAnewS123.run <- c(subsetMAnewS123.run,subsetMAnewAnc.run)
# ddsHTSeqMAnewS123.run <- ddsHTSeqMAnew.run[,subsetMAnewS123.run]
# ddsHTSeqMAnewS123.run$condition
# ddsHTSeqMAnewS123.run$condition <- droplevels(ddsHTSeqMAnewS123.run$condition)
# ddsHTSeqMAnewS123.run$condition
# subsetMAnewS152.run <- which(ddsHTSeqMAnew.run$condition=="152")
# subsetMAnewS152.run <- c(subsetMAnewS152.run,subsetMAnewAnc.run)
# ddsHTSeqMAnewS152.run <- ddsHTSeqMAnew.run[,subsetMAnewS152.run]
# ddsHTSeqMAnewS152.run$condition
# ddsHTSeqMAnewS152.run$condition <- droplevels(ddsHTSeqMAnewS152.run$condition)
# ddsHTSeqMAnewS152.run$condition
# 
# subsetMAnewAll <- c(subsetMAnewS50,subsetMAnewS115, subsetMAnewS117,subsetMAnewS123,subsetMAnewS152,subsetMAnewAnc)
# 
# ddsHTSeqMAnew.short <- ddsHTSeqMAnew[,subsetMAnewAll]
# # find the loadings for PCA
# pcaTest <- prcomp(pcaDataMAnew)
# DESeq2:::plotPCA.DESeqTransform

##### PCA analysis of samples
## transformation of count data
vsdMA_old <- vst(ddsHTSeqMAold,blind=FALSE)
rldMA_old <- rlog(ddsHTSeqMAold,blind=FALSE)
head(assay(vsdMA_old),3)
plotPCA(vsdMA_old, intgroup=c("condition"))
pcaDataMA_old <- plotPCA(vsdMA_old,intgroup=c("condition"),returnData=TRUE)
percentVarMAold <- round(100 * attr(pcaDataMA_old,"percentVar"))
# library(ggplot2)
# pdf("MA_old_PCA_Line.pdf")
ggplot(pcaDataMA_old,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarMAold[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarMAold[2],"% variance")) +
  scale_fill_hue(l=80) +
  scale_shape_manual(values=c(0,1,2,3,4,7,8,11,15,16,17,6,22,9,12,10))
# #  coord_fixed()
#   dev.off()
#   
#   ### PCA with rld data
  plotPCA(rldMA_old, intgroup=c("condition"))
pcaDataMA_oldRld <- plotPCA(rldMA_old,intgroup=c("condition"),returnData=TRUE)
percentVarMAoldRld <- round(100 * attr(pcaDataMA_oldRld,"percentVar"))
# library(ggplot2)
# pdf("MA_old_PCA_Line_Rld.pdf")
ggplot(pcaDataMA_oldRld,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarMAoldRld[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarMAoldRld[2],"% variance")) +
  scale_fill_hue(l=80) +
  scale_shape_manual(values=c(0,1,2,3,4,7,8,11,15,16,17,6,22,9,12,10))
# #  coord_fixed()
#   dev.off()
#   

vsdGC_77 <- vst(ddsHTSeqGC_77,blind=FALSE)
rldGC_77 <- rlog(ddsHTSeqGC_77,blind=FALSE)

ddsHTSeqGC_77$condition
levels(ddsHTSeqGC_77$condition)
head(assay(vsdGC_77),3)
plotPCA(vsdGC_77, intgroup=c("condition"))
pcaDataGC_77 <- plotPCA(vsdGC_77,intgroup=c("condition"),returnData=TRUE)
percentVar_77 <- round(100 * attr(pcaDataGC_77,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_77.pdf")
ggplot(pcaDataGC_77,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_77[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_77[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
  plotPCA(rldGC_77, intgroup=c("condition"))
pcaDataGcRld_77 <- plotPCA(rldGC_77,intgroup=c("condition"),returnData=TRUE)
percentVarRld_77 <- round(100 * attr(pcaDataGcRld_77,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_rld_77.pdf")
ggplot(pcaDataGcRld_77,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_77[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_77[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
colData(ddsHTSeqGC_77)
plotPCA(vsdGC_77, intgroup=c("run"))
pcaDataGC_77 <- plotPCA(vsdGC_77,intgroup=c("run"),returnData=TRUE)
percentVar_77 <- round(100 * attr(pcaDataGC_77,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_by_run_77.pdf")
ggplot(pcaDataGC_77,aes(PC1,PC2,color=run)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_77[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_77[2],"% variance"))
  # coord_fixed()
#   dev.off()  
#  

write.csv(counts(ddsHTSeqGC_76),file="counts.DEseqGC_S1.csv")
vsdGC_76 <- vst(ddsHTSeqGC_76,blind=FALSE)
rldGC_76 <- rlog(ddsHTSeqGC_76,blind=FALSE)

ddsHTSeqGC_76$condition
levels(ddsHTSeqGC_76$condition)
head(assay(vsdGC_76),3)
plotPCA(vsdGC_76, intgroup=c("condition"))
pcaDataGC_76 <- plotPCA(vsdGC_76,intgroup=c("condition"),returnData=TRUE)
percentVar_76 <- round(100 * attr(pcaDataGC_76,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_76.pdf")
ggplot(pcaDataGC_76,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_76[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_76[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
  plotPCA(rldGC_76, intgroup=c("condition"))
pcaDataGcRld_76 <- plotPCA(rldGC_76,intgroup=c("condition"),returnData=TRUE)
percentVarRld_76 <- round(100 * attr(pcaDataGcRld_76,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_rld_76.pdf")
ggplot(pcaDataGcRld_76,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_76[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_76[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
colData(ddsHTSeqGC_76)
plotPCA(vsdGC_76, intgroup=c("run"))
pcaDataGC_76 <- plotPCA(vsdGC_76,intgroup=c("run"),returnData=TRUE)
percentVar_76 <- round(100 * attr(pcaDataGC_76,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_by_run_76.pdf")
ggplot(pcaDataGC_76,aes(PC1,PC2,color=run)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_76[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_76[2],"% variance"))
  # coord_fixed()
#   dev.off()
#
# write.csv(counts(ddsHTSeqGC_69),file="counts.DEseqGC_S69.csv")
vsdGC_69 <- vst(ddsHTSeqGC_69,blind=FALSE)
rldGC_69 <- rlog(ddsHTSeqGC_69,blind=FALSE)

ddsHTSeqGC_69$condition
levels(ddsHTSeqGC_69$condition)
head(assay(vsdGC_69),3)
plotPCA(vsdGC_69, intgroup=c("condition"))
pcaDataGC_69 <- plotPCA(vsdGC_69,intgroup=c("condition"),returnData=TRUE)
percentVar_69 <- round(100 * attr(pcaDataGC_69,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_69.pdf")
ggplot(pcaDataGC_69,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_69[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_69[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
  plotPCA(rldGC_69, intgroup=c("condition"))
pcaDataGcRld_69 <- plotPCA(rldGC_69,intgroup=c("condition"),returnData=TRUE)
percentVarRld_69 <- round(100 * attr(pcaDataGcRld_69,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_rld_69.pdf")
ggplot(pcaDataGcRld_69,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_69[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_69[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
# dev.off()
colData(ddsHTSeqGC_69)
plotPCA(vsdGC_69, intgroup=c("run"))
pcaDataGC_69 <- plotPCA(vsdGC_69,intgroup=c("run"),returnData=TRUE)
percentVar_69 <- round(100 * attr(pcaDataGC_69,"percentVar"))
library(ggplot2)
# pdf("PCA_GC_by_run_69.pdf")
ggplot(pcaDataGC_69,aes(PC1,PC2,color=run)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_69[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_69[2],"% variance"))
  # coord_fixed()
# dev.off()

# write.csv(counts(ddsHTSeqGC_61),file="counts.DEseqGC_S1.csv")
vsdGC_61 <- vst(ddsHTSeqGC_61,blind=FALSE)
rldGC_61 <- rlog(ddsHTSeqGC_61,blind=FALSE)

ddsHTSeqGC_61$condition
levels(ddsHTSeqGC_61$condition)
head(assay(vsdGC_61),3)
plotPCA(vsdGC_61, intgroup=c("condition"))
pcaDataGC_61 <- plotPCA(vsdGC_61,intgroup=c("condition"),returnData=TRUE)
percentVar_61 <- round(100 * attr(pcaDataGC_61,"percentVar"))
library(ggplot2)

# pdf("PCA_GC_61.pdf")

ggplot(pcaDataGC_61,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_61[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_61[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()

# dev.off()

  plotPCA(rldGC_61, intgroup=c("condition"))
pcaDataGcRld_61 <- plotPCA(rldGC_61,intgroup=c("condition"),returnData=TRUE)
percentVarRld_61 <- round(100 * attr(pcaDataGcRld_61,"percentVar"))
library(ggplot2)

# pdf("PCA_GC_rld_61.pdf")

ggplot(pcaDataGcRld_61,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_61[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_61[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()

# dev.off()

colData(ddsHTSeqGC_61)
plotPCA(vsdGC_61, intgroup=c("run"))
pcaDataGC_61 <- plotPCA(vsdGC_61,intgroup=c("run"),returnData=TRUE)
percentVar_61 <- round(100 * attr(pcaDataGC_61,"percentVar"))
library(ggplot2)

# pdf("PCA_GC_by_run_61.pdf")

ggplot(pcaDataGC_61,aes(PC1,PC2,color=run)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_61[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_61[2],"% variance"))
  # coord_fixed()

#   dev.off()
```

### Note on factor levels

By default, R will choose a *reference level* for factors based on
alphabetical order. Then, if you never tell the DESeq2 functions which
level you want to compare against (e.g. which level represents the
control group), the comparisons will be based on the alphabetical
order of the levels. There are two solutions: you can either
explicitly tell *results* which comparison to make using the
`contrast` argument (this will be shown later), or you can explicitly
set the factors levels. You should only change the factor levels
of variables in the design before running the DESeq2 analysis, not
during or afterward.

```{r relevel,eval=TRUE,include=TRUE,echo=FALSE}
ddsHTSeqGCcondition <- relevel(ddsHTSeqGC_All$condition, ref = "Holly_Anc")
# ddsHTSeqGC.short$condition <- relevel(ddsHTSeqGC.short$condition,ref="Holly_Anc")

#filtered genes
# ddsHTSeqGC.fil$condition <- relevel(ddsHTSeqGC.fil$condition, ref = "Holly_Anc")
# ## individual samples
ddsHTSeqGC_1$condition <- relevel(ddsHTSeqGC_1$condition, ref = "Holly_Anc")
ddsHTSeqGC_2$condition <- relevel(ddsHTSeqGC_2$condition, ref = "Holly_Anc")
ddsHTSeqGC_3$condition <- relevel(ddsHTSeqGC_3$condition, ref = "Holly_Anc")
ddsHTSeqGC_4$condition <- relevel(ddsHTSeqGC_4$condition, ref = "Holly_Anc")
ddsHTSeqGC_5$condition <- relevel(ddsHTSeqGC_5$condition, ref = "Holly_Anc")
# ddsHTSeqGC_6$condition <- relevel(ddsHTSeqGC_6$condition, ref = "Holly_Anc")
ddsHTSeqGC_7$condition <- relevel(ddsHTSeqGC_7$condition, ref = "Holly_Anc")
ddsHTSeqGC_8$condition <- relevel(ddsHTSeqGC_8$condition, ref = "Holly_Anc")
ddsHTSeqGC_9$condition <- relevel(ddsHTSeqGC_9$condition, ref = "Holly_Anc")
ddsHTSeqGC_11$condition <- relevel(ddsHTSeqGC_11$condition, ref = "Holly_Anc")
ddsHTSeqGC_18$condition <- relevel(ddsHTSeqGC_18$condition, ref = "Holly_Anc")
ddsHTSeqGC_31$condition <- relevel(ddsHTSeqGC_31$condition, ref = "Holly_Anc")
ddsHTSeqGC_49$condition <- relevel(ddsHTSeqGC_49$condition, ref = "Holly_Anc")
ddsHTSeqGC_59$condition <- relevel(ddsHTSeqGC_59$condition, ref = "Holly_Anc")
ddsHTSeqGC_61$condition <- relevel(ddsHTSeqGC_61$condition, ref = "Holly_Anc")
ddsHTSeqGC_69$condition <- relevel(ddsHTSeqGC_69$condition, ref = "Holly_Anc")
ddsHTSeqGC_76$condition <- relevel(ddsHTSeqGC_76$condition, ref = "Holly_Anc")
ddsHTSeqGC_77$condition <- relevel(ddsHTSeqGC_77$condition, ref = "Holly_Anc")

###############################################################################
ddsHTSeqMAold$condition <- relevel(ddsHTSeqMAold$condition, ref = "SCA")
# ddsHTSeqMAold.short$condition <- relevel(ddsHTSeqMAold.short$condition, ref = "SCA")
ddsHTSeqMAold_1$condition <- relevel(ddsHTSeqMAold_1$condition, ref = "SCA")
ddsHTSeqMAold_2$condition <- relevel(ddsHTSeqMAold_2$condition, ref = "SCA")
ddsHTSeqMAold_3$condition <- relevel(ddsHTSeqMAold_3$condition, ref = "SCA")
ddsHTSeqMAold_4$condition <- relevel(ddsHTSeqMAold_4$condition, ref = "SCA")
ddsHTSeqMAold_5$condition <- relevel(ddsHTSeqMAold_5$condition, ref = "SCA")
# ddsHTSeqMAold_6$condition <- relevel(ddsHTSeqMAold_6$condition, ref = "SCA")
ddsHTSeqMAold_7$condition <- relevel(ddsHTSeqMAold_7$condition, ref = "SCA")
ddsHTSeqMAold_8$condition <- relevel(ddsHTSeqMAold_8$condition, ref = "SCA")
ddsHTSeqMAold_9$condition <- relevel(ddsHTSeqMAold_9$condition, ref = "SCA")
ddsHTSeqMAold_11$condition <- relevel(ddsHTSeqMAold_11$condition, ref = "SCA")
ddsHTSeqMAold_15$condition <- relevel(ddsHTSeqMAold_15$condition, ref = "SCA")
ddsHTSeqMAold_28$condition <- relevel(ddsHTSeqMAold_28$condition, ref = "SCA")
ddsHTSeqMAold_88$condition <- relevel(ddsHTSeqMAold_88$condition, ref = "SCA")
ddsHTSeqMAold_108$condition <- relevel(ddsHTSeqMAold_108$condition, ref = "SCA")
ddsHTSeqMAold_119$condition <- relevel(ddsHTSeqMAold_119$condition, ref = "SCA")
###############################################################################
ddsHTSeqMAnew$condition <- relevel(ddsHTSeqMAnew$condition, ref = "Anc")
# ddsHTSeqMAnew.short$condition <- relevel(ddsHTSeqMAnew.short$condition, ref = "Anc")
ddsHTSeqMAnewS29$condition <- relevel(ddsHTSeqMAnewS29$condition, ref = "Anc")
ddsHTSeqMAnewS50$condition <- relevel(ddsHTSeqMAnewS50$condition, ref = "Anc")
ddsHTSeqMAnewS112$condition <- relevel(ddsHTSeqMAnewS112$condition, ref = "Anc")
ddsHTSeqMAnewS115$condition <- relevel(ddsHTSeqMAnewS115$condition, ref = "Anc")
ddsHTSeqMAnewS117$condition <- relevel(ddsHTSeqMAnewS117$condition, ref = "Anc")
ddsHTSeqMAnewS123$condition <- relevel(ddsHTSeqMAnewS123$condition, ref = "Anc")
# ddsHTSeqMAnewS141$condition <- relevel(ddsHTSeqMAnewS141$condition, ref = "Anc")
ddsHTSeqMAnewS152$condition <- relevel(ddsHTSeqMAnewS152$condition, ref = "Anc")


```


### Pre-filtering
We filtered based on counts in the ancestor of GC: we kept only those genes with more than 10 counts in all 3 ancestor replicates of GC. We used the same metric for old and new MA datasets, and then found the common genes between all 3 for continuity, for a total of 5532 genes analyzed. 

```{r prefilter,eval=TRUE,include=TRUE,echo=FALSE}

## separate lines
ddsHTSeqGC_1
counts(ddsHTSeqGC_1)
dim(counts(ddsHTSeqGC_1))
ancGC <-  counts(ddsHTSeqGC_1)[,4:6]
#subset it based on cpm > 1 in all 3 replicates
table(rowSums(ancGC>10)==3)
keepGC <- rowSums(ancGC>10)==3

# need to make sure the samples don't have 0s either
ddsHTSeqGC_1 <- ddsHTSeqGC_1[keepGC,]

ddsHTSeqGC_11 <- ddsHTSeqGC_11[keepGC,]
ddsHTSeqGC_11
counts(ddsHTSeqGC_11)
dim(counts(ddsHTSeqGC_11))

ddsHTSeqGC_2 <- ddsHTSeqGC_2[keepGC,]

ddsHTSeqGC_3 <- ddsHTSeqGC_3[keepGC,]

ddsHTSeqGC_4 <- ddsHTSeqGC_4[keepGC,]

ddsHTSeqGC_5 <- ddsHTSeqGC_5[keepGC,]

# ddsHTSeqGC_6 <- ddsHTSeqGC_6[keepGC,]
# dim(ddsHTSeqGC_6)
# ddsHTSeqGC_6$condition
ddsHTSeqGC_7 <- ddsHTSeqGC_7[keepGC,]

ddsHTSeqGC_8 <- ddsHTSeqGC_8[keepGC,]

ddsHTSeqGC_9 <- ddsHTSeqGC_9[keepGC,]

ddsHTSeqGC_18 <- ddsHTSeqGC_18[keepGC,]

# ddsHTSeqGC_31 <- ddsHTSeqGC_31[keepGC,]
# dim(ddsHTSeqGC_31)
# ddsHTSeqGC_31$condition
ddsHTSeqGC_49 <- ddsHTSeqGC_49[keepGC,]

ddsHTSeqGC_59 <- ddsHTSeqGC_59[keepGC,]

ddsHTSeqGC_61 <- ddsHTSeqGC_61[keepGC,]

# ddsHTSeqGC_69 <- ddsHTSeqGC_69[keepGC,]
# dim(ddsHTSeqGC_69)
# ddsHTSeqGC_69$condition
ddsHTSeqGC_76 <- ddsHTSeqGC_76[keepGC,]

ddsHTSeqGC_77 <- ddsHTSeqGC_77[keepGC,]

###############################################################################
keepMAold <- rowSums(counts(ddsHTSeqMAold)) >= 20
ddsHTSeqMAold <- ddsHTSeqMAold[keepMAold,]
ddsHTSeqMAold
counts(ddsHTSeqMAold)
dim(counts(ddsHTSeqMAold))

#make a separate data frame with just the ancestor replicates
ancMAold <-  counts(ddsHTSeqMAold)[,43:45]
#subset it based on cpm > 1 in all 3 replicates
table(rowSums(ancMAold>10)==3)
keepMAold <- rowSums(ancMAold>10)==3
# try it in the full count set
# make a test file first
ddsHTSeqMAoldtest <- ddsHTSeqMAold
ddsHTSeqMAoldtest <- ddsHTSeqMAoldtest[keepMAold,]
ddsHTSeqMAoldtest
counts(ddsHTSeqMAoldtest)
dim(counts(ddsHTSeqMAoldtest))
View(counts(ddsHTSeqMAoldtest))
# # test file was fine so go ahead and make it the real thing
ddsHTSeqMAold <- ddsHTSeqMAoldtest
## get rid of any NAs
ddsHTSeqMAold <- na.omit(ddsHTSeqMAold)
dim(ddsHTSeqMAold)

ddsHTSeqMAold_1 <- ddsHTSeqMAold_1[keepGC,]

ddsHTSeqMAold_2 <- ddsHTSeqMAold_2[keepGC,]

ddsHTSeqMAold_2$condition

ddsHTSeqMAold_3 <- ddsHTSeqMAold_3[keepGC,]

ddsHTSeqMAold_4 <- ddsHTSeqMAold_4[keepGC,]

ddsHTSeqMAold_5 <- ddsHTSeqMAold_5[keepGC,]

ddsHTSeqMAold_7 <- ddsHTSeqMAold_7[keepGC,]

ddsHTSeqMAold_8 <- ddsHTSeqMAold_8[keepGC,]

ddsHTSeqMAold_9 <- ddsHTSeqMAold_9[keepGC,]

ddsHTSeqMAold_11 <- ddsHTSeqMAold_11[keepGC,]

ddsHTSeqMAold_15 <- ddsHTSeqMAold_15[keepGC,]

ddsHTSeqMAold_28 <- ddsHTSeqMAold_28[keepGC,]

ddsHTSeqMAold_88 <- ddsHTSeqMAold_88[keepGC,]

ddsHTSeqMAold_108 <- ddsHTSeqMAold_108[keepGC,]

ddsHTSeqMAold_119 <- ddsHTSeqMAold_119[keepGC,]

###############################################################################
keepMA <- rownames(ddsHTSeqMAnewS29)
keepGC1 <- rownames(ddsHTSeqGC_1)
keep1 <- keepMA %in% keepGC1
summary(keep1)
length(keep1)
keep <- keep1

ddsHTSeqMAnewS29 <- ddsHTSeqMAnewS29[keep,]
length(ddsHTSeqMAnewS29)

ddsHTSeqMAnewS50 <- ddsHTSeqMAnewS50[keep,]
length(ddsHTSeqMAnewS50)

ddsHTSeqMAnewS112 <- ddsHTSeqMAnewS112[keep,]

ddsHTSeqMAnewS115 <- ddsHTSeqMAnewS115[keep,]

ddsHTSeqMAnewS117 <- ddsHTSeqMAnewS117[keep,]

ddsHTSeqMAnewS123 <- ddsHTSeqMAnewS123[keep,]

# ddsHTSeqMAnewS141 <- ddsHTSeqMAnewS141[keepMAnew,]
# ddsHTSeqMAnewS141

ddsHTSeqMAnewS152 <- ddsHTSeqMAnewS152[keep,]

#now need to make sure all the GC and old MA samples match the new MA samples since got rid of a couple
keepMA.1 <- rownames(ddsHTSeqMAnewS152)
keepGC1.1 <- rownames(ddsHTSeqGC_77)
# keep <- keepGC1 %in% keepMA
keep2 <- keepGC1.1 %in% keepMA.1
summary(keep2)
length(keep2)

ddsHTSeqGC_1 <- ddsHTSeqGC_1[keep2,]
ddsHTSeqGC_2 <- ddsHTSeqGC_2[keep2,]
ddsHTSeqGC_3 <- ddsHTSeqGC_3[keep2,]
ddsHTSeqGC_4 <- ddsHTSeqGC_4[keep2,]
ddsHTSeqGC_5 <- ddsHTSeqGC_5[keep2,]
ddsHTSeqGC_7 <- ddsHTSeqGC_7[keep2,]
ddsHTSeqGC_8 <- ddsHTSeqGC_8[keep2,]
ddsHTSeqGC_9 <- ddsHTSeqGC_9[keep2,]
ddsHTSeqGC_11 <- ddsHTSeqGC_11[keep2,]
ddsHTSeqGC_18 <- ddsHTSeqGC_18[keep2,]
ddsHTSeqGC_49 <- ddsHTSeqGC_49[keep2,]
ddsHTSeqGC_59 <- ddsHTSeqGC_59[keep2,]
ddsHTSeqGC_61 <- ddsHTSeqGC_61[keep2,]
ddsHTSeqGC_76 <- ddsHTSeqGC_76[keep2,]
ddsHTSeqGC_77 <- ddsHTSeqGC_77[keep2,]

##################################################################################
### Left off here, trying to get MA old genes to match GC and MA new genes..
keepMA.1 <- rownames(ddsHTSeqMAnewS152)
keepGC1.2 <- rownames(ddsHTSeqMAold_2)
# keep <- keepGC1 %in% keepMA
keep3 <- keepGC1.2 %in% keepMA.1
summary(keep3)
length(keep3)

ddsHTSeqMAold_1 <- ddsHTSeqMAold_1[keep3,]
ddsHTSeqMAold_2 <- ddsHTSeqMAold_2[keep3,]
ddsHTSeqMAold_3 <- ddsHTSeqMAold_3[keep3,]
ddsHTSeqMAold_4 <- ddsHTSeqMAold_4[keep3,]
ddsHTSeqMAold_5 <- ddsHTSeqMAold_5[keep3,]
ddsHTSeqMAold_7 <- ddsHTSeqMAold_7[keep3,]
ddsHTSeqMAold_8 <- ddsHTSeqMAold_8[keep3,]
ddsHTSeqMAold_9 <- ddsHTSeqMAold_9[keep3,]
ddsHTSeqMAold_11 <- ddsHTSeqMAold_11[keep3,]
ddsHTSeqMAold_15 <- ddsHTSeqMAold_15[keep3,]
ddsHTSeqMAold_28 <- ddsHTSeqMAold_28[keep3,]
ddsHTSeqMAold_88 <- ddsHTSeqMAold_88[keep3,]
ddsHTSeqMAold_108 <- ddsHTSeqMAold_108[keep3,]
ddsHTSeqMAold_119 <- ddsHTSeqMAold_119[keep3,]
```

<a name="factorlevels"/>

## Differential expression analysis

The standard differential expression analysis steps are wrapped
into a single function, *DESeq*. The estimation steps performed
by this function are described [below](#theory), in the manual page for
`?DESeq` and in the Methods section of the DESeq2 publication [@Love2014].

Results tables are generated using the function *results*, which
extracts a results table with log2 fold changes, *p* values and adjusted
*p* values. With no additional arguments to *results*, the log2 fold change and
Wald test *p* value will be for the **last variable** in the design
formula, and if this is a factor, the comparison will be the **last
level** of this variable over the **reference level**
(see previous [note on factor levels](#factorlevels)).
However, the order of the variables of the design do not matter
so long as the user specifies the comparison to build a results table
for, using the `name` or `contrast` arguments of *results*.

Details about the comparison are printed to the console, directly above the
results table. The text, `condition treated vs untreated`, tells you that the
estimates are of the logarithmic fold change log2(treated/untreated).



```{r full dataset, eval=TRUE,include=TRUE}
#####################################################
#default
## add annotations
# geneIDgC <- rownames(ddsHTSeqGC.short)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesGC <- ensembldb::select(txdb, keys=geneIDgC, columnns="GENEID","TXCHROM" , keytype="GENEID")
#
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGC <- na.omit(genesGC)
# genesGC <- as.data.frame(genesGC)
# remove1 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGC.short) == genesGC$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS1 <- intersect(rownames(ddsHTSeqGC.short), genesGC$GENEID)
# keep3 <- rownames(ddsHTSeqGC.short) %in% matchedS1
# ddsHTSeqGC.short <- ddsHTSeqGC.short[keep3,]
# length(ddsHTSeqGC.short)
# all(rownames(ddsHTSeqGC.short) == genesGC$GENEID)
# mcols(ddsHTSeqGC.short) <- cbind(mcols(ddsHTSeqGC.short), genesGC)
#### filtered genes
#default
## add annotations
# geneIDgC.fil <- rownames(ddsHTSeqGC.short)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesGC.fil <- ensembldb::select(txdb, keys=geneIDgC.fil, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGC.fil <- na.omit(genesGC.fil)
# genesGC.fil <- as.data.frame(genesGC.fil)
# remove1 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGC.fil) == genesGC.fil$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS1.fil <- intersect(rownames(ddsHTSeqGC.fil), genesGC.fil$GENEID)
# keep3.fil <- rownames(ddsHTSeqGC.fil) %in% matchedS1.fil
# ddsHTSeqGC.fil <- ddsHTSeqGC.fil[keep3.fil,]
# length(ddsHTSeqGC.fil)
# all(rownames(ddsHTSeqGC.fil) == genesGC.fil$GENEID)
# mcols(ddsHTSeqGC.fil) <- cbind(mcols(ddsHTSeqGC.fil), genesGC.fil)
#
#
# geneIDgC.short <- rownames(ddsHTSeqGC.short)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesGC.short <- ensembldb::select(txdb, keys=geneIDgC.short, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGC.short <- na.omit(genesGC.short)
# genesGC.short <- as.data.frame(genesGC.short)
# remove1 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGC.short) == genesGC.short$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS1.short <- intersect(rownames(ddsHTSeqGC.short), genesGC.short$GENEID)
# keep3.short <- rownames(ddsHTSeqGC.short) %in% matchedS1.short
# ddsHTSeqGC.short <- ddsHTSeqGC.short[keep3.short,]
# length(ddsHTSeqGC.short)
# all(rownames(ddsHTSeqGC.short) == genesGC.short$GENEID)
# mcols(ddsHTSeqGC.short) <- cbind(mcols(ddsHTSeqGC.short), genesGC.short)
#

### Add annotations to ESR, ASR, DS datasets 
geneID_ASR <- ASRgenes$GENEID
geneID_ASR <- as.character(geneID_ASR)
geneID_ESR <- esrGenes$GENEID
geneID_ESR <- as.character(geneID_ESR)
geneID_DS <- dosageSensitiveGenes$GENEID
geneID_DS <- as.character(geneID_DS)
library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesASR <- ensembldb::select(txdb, keys=geneID_ASR, columnns="GENEID","TXCHROM", keytype="GENEID")
genesESR <- ensembldb::select(txdb, keys=geneID_ESR, columnns="GENEID","TXCHROM", keytype="GENEID")
genesDS <- ensembldb::select(txdb, keys=geneID_DS, columnns="GENEID","TXCHROM", keytype="GENEID")


######### individual samples
## add annotations
# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
#
# BiocManager::install("TxDb.Scerevisiae.UCSC.sacCer2.sgdGene")
geneIDgC_1 <- rownames(ddsHTSeqGC_1)
library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
#
# BiocManager::install("ensembldb")
columns(txdb)
genesGC_1 <- ensembldb::select(txdb, keys=geneIDgC_1, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_1 <- na.omit(genesGC_1)
genesGC_1 <- as.data.frame(genesGC_1)
remove1 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_1) == genesGC_1$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS1 <- intersect(rownames(ddsHTSeqGC_1), genesGC_1$GENEID)
keep3S1 <- rownames(ddsHTSeqGC_1) %in% matchedS1
ddsHTSeqGC_1 <- ddsHTSeqGC_1[keep3S1,]
length(ddsHTSeqGC_1)
all(rownames(ddsHTSeqGC_1) == genesGC_1$GENEID)
mcols(ddsHTSeqGC_1) <- cbind(mcols(ddsHTSeqGC_1), genesGC_1)

geneIDgC_11 <- rownames(ddsHTSeqGC_11)
# library(TxDb.Scerevisiae.UC_C.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UC_C.sacCer2.sgdGene
genesGC_11 <- ensembldb::select(txdb, keys=geneIDgC_11, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_11 <- na.omit(genesGC_11)
genesGC_11 <- as.data.frame(genesGC_11)
remove1 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_11) == genesGC_11$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS11 <- intersect(rownames(ddsHTSeqGC_11), genesGC_11$GENEID)
keep3S11 <- rownames(ddsHTSeqGC_11) %in% matchedS11
ddsHTSeqGC_11 <- ddsHTSeqGC_11[keep3S11,]
length(ddsHTSeqGC_11)
all(rownames(ddsHTSeqGC_11) == genesGC_11$GENEID)
mcols(ddsHTSeqGC_11) <- cbind(mcols(ddsHTSeqGC_11), genesGC_11)
geneIDgC_2 <- rownames(ddsHTSeqGC_2)
genesGC_2 <- ensembldb::select(txdb, keys=geneIDgC_2, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_2 <- na.omit(genesGC_2)
genesGC_2 <- as.data.frame(genesGC_2)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_2) == genesGC_2$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqGC_2), genesGC_2$GENEID)
keep3S2 <- rownames(ddsHTSeqGC_2) %in% matchedS2
ddsHTSeqGC_2 <- ddsHTSeqGC_2[keep3S2,]
length(ddsHTSeqGC_2)
all(rownames(ddsHTSeqGC_2) == genesGC_2$GENEID)
mcols(ddsHTSeqGC_2) <- cbind(mcols(ddsHTSeqGC_2), genesGC_2)

all(rownames(ddsHTSeqGC_3) == genesGC_2$GENEID)

geneIDgC_1 <- rownames(ddsHTSeqGC_1)
genesGC_1 <- ensembldb::select(txdb, keys=geneIDgC_1, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_1 <- na.omit(genesGC_1)
genesGC_1 <- as.data.frame(genesGC_1)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_1) == genesGC_1$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS1 <- intersect(rownames(ddsHTSeqGC_1), genesGC_1$GENEID)
keep3S1 <- rownames(ddsHTSeqGC_1) %in% matchedS1
ddsHTSeqGC_1 <- ddsHTSeqGC_1[keep3S1,]
length(ddsHTSeqGC_1)
all(rownames(ddsHTSeqGC_1) == genesGC_1$GENEID)
mcols(ddsHTSeqGC_1) <- cbind(mcols(ddsHTSeqGC_1), genesGC_1)
mcols(ddsHTSeqGC_2) <- cbind(mcols(ddsHTSeqGC_2), genesGC_2)

geneIDgC_3 <- rownames(ddsHTSeqGC_3)
genesGC_3 <- ensembldb::select(txdb, keys=geneIDgC_3, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_3 <- na.omit(genesGC_3)
genesGC_3 <- as.data.frame(genesGC_3)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove3,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_3) == genesGC_3$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS3 <- intersect(rownames(ddsHTSeqGC_3), genesGC_3$GENEID)
keep3S3 <- rownames(ddsHTSeqGC_3) %in% matchedS3
ddsHTSeqGC_3 <- ddsHTSeqGC_3[keep3S3,]
length(ddsHTSeqGC_3)
all(rownames(ddsHTSeqGC_3) == genesGC_3$GENEID)
mcols(ddsHTSeqGC_3) <- cbind(mcols(ddsHTSeqGC_3), genesGC_3)

geneIDgC_4 <- rownames(ddsHTSeqGC_4)
genesGC_4 <- ensembldb::select(txdb, keys=geneIDgC_4, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_4 <- na.omit(genesGC_4)
genesGC_4 <- as.data.frame(genesGC_4)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove4,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_4) == genesGC_4$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS4 <- intersect(rownames(ddsHTSeqGC_4), genesGC_4$GENEID)
keep3S4 <- rownames(ddsHTSeqGC_4) %in% matchedS4
ddsHTSeqGC_4 <- ddsHTSeqGC_4[keep3S4,]
length(ddsHTSeqGC_4)
all(rownames(ddsHTSeqGC_4) == genesGC_4$GENEID)
mcols(ddsHTSeqGC_4) <- cbind(mcols(ddsHTSeqGC_4), genesGC_4)
# mcols(ddsHTSeqGC_2.fil) <- cbind(mcols(ddsHTSeqGC_2.fil), genesGC_2)

{geneIDgC_5 <- rownames(ddsHTSeqGC_5)
genesGC_5 <- ensembldb::select(txdb, keys=geneIDgC_5, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_5 <- na.omit(genesGC_5)
genesGC_5 <- as.data.frame(genesGC_5)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove4,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_5) == genesGC_5$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS5 <- intersect(rownames(ddsHTSeqGC_5), genesGC_5$GENEID)
keep3S5 <- rownames(ddsHTSeqGC_5) %in% matchedS4
ddsHTSeqGC_5 <- ddsHTSeqGC_5[keep3S5,]
length(ddsHTSeqGC_5)
all(rownames(ddsHTSeqGC_5) == genesGC_5$GENEID)
mcols(ddsHTSeqGC_5) <- cbind(mcols(ddsHTSeqGC_5), genesGC_5)}
# mcols(ddsHTSeqGC_2.fil) <- cbind(mcols(ddsHTSeqGC_2.fil), genesGC_2)

# geneIDgC_6 <- rownames(ddsHTSeqGC_6)
# genesGC_6 <- ensembldb::select(txdb, keys=geneIDgC_6, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGC_6 <- na.omit(genesGC_6)
# genesGC_6 <- as.data.frame(genesGC_6)
# remove2 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove4,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGC_6) == genesGC_6$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS6 <- intersect(rownames(ddsHTSeqGC_6), genesGC_6$GENEID)
# keep3S6 <- rownames(ddsHTSeqGC_6) %in% matchedS6
# ddsHTSeqGC_6 <- ddsHTSeqGC_6[keep3S6,]
# length(ddsHTSeqGC_6)
# all(rownames(ddsHTSeqGC_6) == genesGC_6$GENEID)
# mcols(ddsHTSeqGC_6) <- cbind(mcols(ddsHTSeqGC_6), genesGC_6)


geneIDgC_7 <- rownames(ddsHTSeqGC_7)
genesGC_7 <- ensembldb::select(txdb, keys=geneIDgC_7, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_7 <- na.omit(genesGC_7)
genesGC_7 <- as.data.frame(genesGC_7)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove7,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_7) == genesGC_7$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS7 <- intersect(rownames(ddsHTSeqGC_7), genesGC_7$GENEID)
keep3S7 <- rownames(ddsHTSeqGC_7) %in% matchedS7
ddsHTSeqGC_7 <- ddsHTSeqGC_7[keep3S7,]
length(ddsHTSeqGC_7)
all(rownames(ddsHTSeqGC_7) == genesGC_7$GENEID)
mcols(ddsHTSeqGC_7) <- cbind(mcols(ddsHTSeqGC_7), genesGC_7)

geneIDgC_8 <- rownames(ddsHTSeqGC_8)
genesGC_8 <- ensembldb::select(txdb, keys=geneIDgC_8, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_8 <- na.omit(genesGC_8)
genesGC_8 <- as.data.frame(genesGC_8)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove8,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_8) == genesGC_8$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS8 <- intersect(rownames(ddsHTSeqGC_8), genesGC_8$GENEID)
keep3S8 <- rownames(ddsHTSeqGC_8) %in% matchedS8
ddsHTSeqGC_8 <- ddsHTSeqGC_8[keep3S8,]
length(ddsHTSeqGC_8)
all(rownames(ddsHTSeqGC_8) == genesGC_8$GENEID)
mcols(ddsHTSeqGC_8) <- cbind(mcols(ddsHTSeqGC_8), genesGC_8)

geneIDgC_9 <- rownames(ddsHTSeqGC_9)
genesGC_9 <- ensembldb::select(txdb, keys=geneIDgC_9, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_9 <- na.omit(genesGC_9)
genesGC_9 <- as.data.frame(genesGC_9)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove9,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_9) == genesGC_9$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS9 <- intersect(rownames(ddsHTSeqGC_9), genesGC_9$GENEID)
keep3S9 <- rownames(ddsHTSeqGC_9) %in% matchedS9
ddsHTSeqGC_9 <- ddsHTSeqGC_9[keep3S9,]
length(ddsHTSeqGC_9)
all(rownames(ddsHTSeqGC_9) == genesGC_9$GENEID)
mcols(ddsHTSeqGC_9) <- cbind(mcols(ddsHTSeqGC_9), genesGC_9)

geneIDgC_11 <- rownames(ddsHTSeqGC_11)
genesGC_11 <- ensembldb::select(txdb, keys=geneIDgC_11, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_11 <- na.omit(genesGC_11)
genesGC_11 <- as.data.frame(genesGC_11)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove11,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_11) == genesGC_11$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS11 <- intersect(rownames(ddsHTSeqGC_11), genesGC_11$GENEID)
keep3S11 <- rownames(ddsHTSeqGC_11) %in% matchedS11
ddsHTSeqGC_11 <- ddsHTSeqGC_11[keep3S11,]
length(ddsHTSeqGC_11)
all(rownames(ddsHTSeqGC_11) == genesGC_11$GENEID)
mcols(ddsHTSeqGC_11) <- cbind(mcols(ddsHTSeqGC_11), genesGC_11)

geneIDgC_18 <- rownames(ddsHTSeqGC_18)
genesGC_18 <- ensembldb::select(txdb, keys=geneIDgC_18, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_18 <- na.omit(genesGC_18)
genesGC_18 <- as.data.frame(genesGC_18)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove18,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_18) == genesGC_18$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS18 <- intersect(rownames(ddsHTSeqGC_18), genesGC_18$GENEID)
keep3S18 <- rownames(ddsHTSeqGC_18) %in% matchedS18
ddsHTSeqGC_18 <- ddsHTSeqGC_18[keep3S18,]
length(ddsHTSeqGC_18)
all(rownames(ddsHTSeqGC_18) == genesGC_18$GENEID)
mcols(ddsHTSeqGC_18) <- cbind(mcols(ddsHTSeqGC_18), genesGC_18)

geneIDgC_31 <- rownames(ddsHTSeqGC_31)
genesGC_31 <- ensembldb::select(txdb, keys=geneIDgC_31, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_31 <- na.omit(genesGC_31)
genesGC_31 <- as.data.frame(genesGC_31)
remove2 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove18,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_31) == genesGC_31$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS31 <- intersect(rownames(ddsHTSeqGC_31), genesGC_31$GENEID)
keep3S31 <- rownames(ddsHTSeqGC_31) %in% matchedS31
ddsHTSeqGC_31 <- ddsHTSeqGC_31[keep3S31,]
length(ddsHTSeqGC_31)
all(rownames(ddsHTSeqGC_31) == genesGC_31$GENEID)
mcols(ddsHTSeqGC_31) <- cbind(mcols(ddsHTSeqGC_31), genesGC_31)


# mcols(ddsHTSeqGC_31) <- cbind(mcols(ddsHTSeqGC_31), genesGC_2)
geneIDgC_49 <- rownames(ddsHTSeqGC_49)
genesGC_49 <- ensembldb::select(txdb, keys=geneIDgC_49, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_49 <- na.omit(genesGC_49)
genesGC_49 <- as.data.frame(genesGC_49)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove49,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_49) == genesGC_49$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS49 <- intersect(rownames(ddsHTSeqGC_49), genesGC_49$GENEID)
keep3S49 <- rownames(ddsHTSeqGC_49) %in% matchedS49
ddsHTSeqGC_49 <- ddsHTSeqGC_49[keep3S49,]
length(ddsHTSeqGC_49)
all(rownames(ddsHTSeqGC_49) == genesGC_49$GENEID)
mcols(ddsHTSeqGC_49) <- cbind(mcols(ddsHTSeqGC_49), genesGC_49)

geneIDgC_59 <- rownames(ddsHTSeqGC_59)
genesGC_59 <- ensembldb::select(txdb, keys=geneIDgC_59, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_59 <- na.omit(genesGC_59)
genesGC_59 <- as.data.frame(genesGC_59)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove59,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_59) == genesGC_59$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS59 <- intersect(rownames(ddsHTSeqGC_59), genesGC_59$GENEID)
keep3S59 <- rownames(ddsHTSeqGC_59) %in% matchedS59
ddsHTSeqGC_59 <- ddsHTSeqGC_59[keep3S59,]
length(ddsHTSeqGC_59)
all(rownames(ddsHTSeqGC_59) == genesGC_59$GENEID)
mcols(ddsHTSeqGC_59) <- cbind(mcols(ddsHTSeqGC_59), genesGC_59)

geneIDgC_61 <- rownames(ddsHTSeqGC_61)
genesGC_61 <- ensembldb::select(txdb, keys=geneIDgC_61, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_61 <- na.omit(genesGC_61)
genesGC_61 <- as.data.frame(genesGC_61)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove61,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_61) == genesGC_61$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS61 <- intersect(rownames(ddsHTSeqGC_61), genesGC_61$GENEID)
keep3S61 <- rownames(ddsHTSeqGC_61) %in% matchedS61
ddsHTSeqGC_61 <- ddsHTSeqGC_61[keep3S61,]
length(ddsHTSeqGC_61)
all(rownames(ddsHTSeqGC_61) == genesGC_61$GENEID)
mcols(ddsHTSeqGC_61) <- cbind(mcols(ddsHTSeqGC_61), genesGC_61)
# mcols(ddsHTSeqGC_69) <- cbind(mcols(ddsHTSeqGC_69), genesGC_2)

# geneIDgC_69 <- rownames(ddsHTSeqGC_69)
# genesGC_69 <- ensembldb::select(txdb, keys=geneIDgC_69, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGC_69 <- na.omit(genesGC_69)
# genesGC_69 <- as.data.frame(genesGC_69)
# remove2 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove61,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGC_69) == genesGC_69$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS69 <- intersect(rownames(ddsHTSeqGC_69), genesGC_69$GENEID)
# keep3S69 <- rownames(ddsHTSeqGC_69) %in% matchedS69
# ddsHTSeqGC_69 <- ddsHTSeqGC_69[keep3S69,]
# length(ddsHTSeqGC_69)
# all(rownames(ddsHTSeqGC_69) == genesGC_69$GENEID)
# mcols(ddsHTSeqGC_69) <- cbind(mcols(ddsHTSeqGC_69), genesGC_69)
# # mcols(ddsHTSeqGC_69) <- cbind(mcols(ddsHTSeqGC_69), genesGC_2)


geneIDgC_76 <- rownames(ddsHTSeqGC_76)
genesGC_76 <- ensembldb::select(txdb, keys=geneIDgC_76, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_76 <- na.omit(genesGC_76)
genesGC_76 <- as.data.frame(genesGC_76)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove76,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_76) == genesGC_76$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS76 <- intersect(rownames(ddsHTSeqGC_76), genesGC_76$GENEID)
keep3S76 <- rownames(ddsHTSeqGC_76) %in% matchedS76
ddsHTSeqGC_76 <- ddsHTSeqGC_76[keep3S76,]
length(ddsHTSeqGC_76)
all(rownames(ddsHTSeqGC_76) == genesGC_76$GENEID)
mcols(ddsHTSeqGC_76) <- cbind(mcols(ddsHTSeqGC_76), genesGC_76)

geneIDgC_77 <- rownames(ddsHTSeqGC_77)
genesGC_77 <- ensembldb::select(txdb, keys=geneIDgC_77, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_77 <- na.omit(genesGC_77)
genesGC_77 <- as.data.frame(genesGC_77)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove77,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_77) == genesGC_77$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS77 <- intersect(rownames(ddsHTSeqGC_77), genesGC_77$GENEID)
keep3S77 <- rownames(ddsHTSeqGC_77) %in% matchedS77
ddsHTSeqGC_77 <- ddsHTSeqGC_77[keep3S77,]
length(ddsHTSeqGC_77)
all(rownames(ddsHTSeqGC_77) == genesGC_77$GENEID)
mcols(ddsHTSeqGC_77) <- cbind(mcols(ddsHTSeqGC_77), genesGC_77)



### with run as factor
# geneIDgCS1.run <- rownames(ddsHTSeqGCS1.run)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesGCS1.run <- ensembldb::select(txdb, keys=geneIDgCS1.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGCS1.run <- na.omit(genesGCS1.run)
# genesGCS1.run <- as.data.frame(genesGCS1.run)
# remove1 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGCS1.run) == genesGCS1.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS1.run <- intersect(rownames(ddsHTSeqGCS1.run), genesGCS1.run$GENEID)
# keep3S1.run <- rownames(ddsHTSeqGCS1.run) %in% matchedS1.run
# ddsHTSeqGCS1.run <- ddsHTSeqGCS1.run[keep3S1.run,]
# length(ddsHTSeqGCS1.run)
# all(rownames(ddsHTSeqGCS1.run) == genesGCS1.run$GENEID)
# mcols(ddsHTSeqGCS1.run) <- cbind(mcols(ddsHTSeqGCS1.run), genesGCS1.run)
#
# geneIDgCS11.run <- rownames(ddsHTSeqGCS11.run)
# # library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# # txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesGCS11.run <- ensembldb::select(txdb, keys=geneIDgCS11.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGCS11.run <- na.omit(genesGCS11.run)
# genesGCS11.run <- as.data.frame(genesGCS11.run)
# remove1 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGCS11.run) == genesGCS11.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS11.run <- intersect(rownames(ddsHTSeqGCS11.run), genesGCS11.run$GENEID)
# keep3S11.run <- rownames(ddsHTSeqGCS11.run) %in% matchedS11.run
# ddsHTSeqGCS11.run <- ddsHTSeqGCS11.run[keep3S11.run,]
# length(ddsHTSeqGCS11.run)
# all(rownames(ddsHTSeqGCS11.run) == genesGCS11.run$GENEID)
# mcols(ddsHTSeqGCS11.run) <- cbind(mcols(ddsHTSeqGCS11.run), genesGCS11.run)
# geneIDgCS2.run <- rownames(ddsHTSeqGCS2.run)
# genesGCS2.run <- ensembldb::select(txdb, keys=geneIDgCS2.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGCS2.run <- na.omit(genesGCS2.run)
# genesGCS2.run <- as.data.frame(genesGCS2.run)
# remove2 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGCS2.run) == genesGCS2.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqGCS2.run), genesGCS2.run$GENEID)
# keep3S2.run <- rownames(ddsHTSeqGCS2.run) %in% matchedS2.run
# ddsHTSeqGCS2.run <- ddsHTSeqGCS2.run[keep3S2.run,]
# length(ddsHTSeqGCS2.run)
# all(rownames(ddsHTSeqGCS2.run) == genesGCS2.run$GENEID)
# mcols(ddsHTSeqGCS2.run) <- cbind(mcols(ddsHTSeqGCS2.run), genesGCS2.run)
#
# all(rownames(ddsHTSeqGCS3.run) == genesGCS2.run$GENEID)
# # mcols(ddsHTSeqGCS1.run) <- cbind(mcols(ddsHTSeqGCS1.run), genesGCS2.run)
# mcols(ddsHTSeqGCS3.run) <- cbind(mcols(ddsHTSeqGCS3.run), genesGCS2.run)
# mcols(ddsHTSeqGCS4.run) <- cbind(mcols(ddsHTSeqGCS4.run), genesGCS2.run)
# # mcols(ddsHTSeqGCS2.fil) <- cbind(mcols(ddsHTSeqGCS2.fil), genesGCS2)
# mcols(ddsHTSeqGCS7.run) <- cbind(mcols(ddsHTSeqGCS7.run), genesGCS2.run)
# mcols(ddsHTSeqGCS8.run) <- cbind(mcols(ddsHTSeqGCS8.run), genesGCS2.run)
# mcols(ddsHTSeqGCS9.run) <- cbind(mcols(ddsHTSeqGCS9.run), genesGCS2.run)
# # mcols(ddsHTSeqGCS11.run) <- cbind(mcols(ddsHTSeqGCS11.run), genesGCS2.run)
# mcols(ddsHTSeqGCS18.run) <- cbind(mcols(ddsHTSeqGCS18.run), genesGCS2.run)
# mcols(ddsHTSeqGCS31.run) <- cbind(mcols(ddsHTSeqGCS31.run), genesGCS2.run)
# mcols(ddsHTSeqGCS49.run) <- cbind(mcols(ddsHTSeqGCS49.run), genesGCS2.run)
# mcols(ddsHTSeqGCS59.run) <- cbind(mcols(ddsHTSeqGCS59.run), genesGCS2.run)
# mcols(ddsHTSeqGCS61.run) <- cbind(mcols(ddsHTSeqGCS61.run), genesGCS2.run)
# mcols(ddsHTSeqGCS69.run) <- cbind(mcols(ddsHTSeqGCS69.run), genesGCS2.run)
# mcols(ddsHTSeqGCS76.run) <- cbind(mcols(ddsHTSeqGCS76.run), genesGCS2.run)
# mcols(ddsHTSeqGCS77.run) <- cbind(mcols(ddsHTSeqGCS77.run), genesGCS2.run)

#### filtered genes
#default
# ## add annotations
# geneIDgC.fil <- rownames(ddsHTSeqGC.fil)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesGC.fil <- ensembldb::select(txdb, keys=geneIDgC.fil, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGC.fil <- na.omit(genesGC.fil)
# genesGC.fil <- as.data.frame(genesGC.fil)
# remove1 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGC.fil) == genesGC.fil$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS1.fil <- intersect(rownames(ddsHTSeqGC.fil), genesGC.fil$GENEID)
# keep3.fil <- rownames(ddsHTSeqGC.fil) %in% matchedS1.fil
# ddsHTSeqGC.fil <- ddsHTSeqGC.fil[keep3.fil,]
# length(ddsHTSeqGC.fil)
# all(rownames(ddsHTSeqGC.fil) == genesGC.fil$GENEID)
# mcols(ddsHTSeqGC.fil) <- cbind(mcols(ddsHTSeqGC.fil), genesGC.fil)
#####################################################
#default
## add annotations
# geneIDMAold <- rownames(ddsHTSeqMAold)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAold <- ensembldb::select(txdb, keys=geneIDMAold, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAold <- na.omit(genesMAold)
# genesMAold <- as.data.frame(genesMAold)
# remove1 <- "chrM"
# # genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAold) == genesMAold$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2 <- intersect(rownames(ddsHTSeqMAold), genesMAold$GENEID)
# keep4 <- rownames(ddsHTSeqMAold) %in% matchedS2
# ddsHTSeqMAold <- ddsHTSeqMAold[keep4,]
# # length(ddsHTSeqMAold)
# all(rownames(ddsHTSeqMAold) == genesMAold$GENEID)
# mcols(ddsHTSeqMAold) <- cbind(mcols(ddsHTSeqMAold), genesMAold)

###### individual samples
geneIDMAold_1 <- rownames(ddsHTSeqMAold_1)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_1 <- ensembldb::select(txdb, keys=geneIDMAold_1, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_1 <- na.omit(genesMAold_1)
genesMAold_1 <- as.data.frame(genesMAold_1)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_1) == genesMAold_1$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_1), genesMAold_1$GENEID)
keep4 <- rownames(ddsHTSeqMAold_1) %in% matched_2
ddsHTSeqMAold_1 <- ddsHTSeqMAold_1[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_1) == genesMAold_1$GENEID)
mcols(ddsHTSeqMAold_1) <- cbind(mcols(ddsHTSeqMAold_1), genesMAold_1)

geneIDMAold_2 <- rownames(ddsHTSeqMAold_2)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_2 <- ensembldb::select(txdb, keys=geneIDMAold_2, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_2 <- na.omit(genesMAold_2)
genesMAold_2 <- as.data.frame(genesMAold_2)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_2) == genesMAold_2$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_2), genesMAold_2$GENEID)
keep4 <- rownames(ddsHTSeqMAold_2) %in% matched_2
ddsHTSeqMAold_2 <- ddsHTSeqMAold_2[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_2) == genesMAold_2$GENEID)
mcols(ddsHTSeqMAold_2) <- cbind(mcols(ddsHTSeqMAold_2), genesMAold_2)

geneIDMAold_3 <- rownames(ddsHTSeqMAold_3)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_3 <- ensembldb::select(txdb, keys=geneIDMAold_3, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_3 <- na.omit(genesMAold_3)
genesMAold_3 <- as.data.frame(genesMAold_3)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_3) == genesMAold_3$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_3), genesMAold_3$GENEID)
keep4 <- rownames(ddsHTSeqMAold_3) %in% matched_2
ddsHTSeqMAold_3 <- ddsHTSeqMAold_3[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_3) == genesMAold_3$GENEID)
mcols(ddsHTSeqMAold_3) <- cbind(mcols(ddsHTSeqMAold_3), genesMAold_3)

geneIDMAold_4 <- rownames(ddsHTSeqMAold_4)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_4 <- ensembldb::select(txdb, keys=geneIDMAold_4, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_4 <- na.omit(genesMAold_4)
genesMAold_4 <- as.data.frame(genesMAold_4)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_4) == genesMAold_4$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_4), genesMAold_4$GENEID)
keep4 <- rownames(ddsHTSeqMAold_4) %in% matched_2
ddsHTSeqMAold_4 <- ddsHTSeqMAold_4[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_4) == genesMAold_4$GENEID)
mcols(ddsHTSeqMAold_4) <- cbind(mcols(ddsHTSeqMAold_4), genesMAold_4)

geneIDMAold_5 <- rownames(ddsHTSeqMAold_5)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_5 <- ensembldb::select(txdb, keys=geneIDMAold_5, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_5 <- na.omit(genesMAold_5)

genesMAold_5 <- as.data.frame(genesMAold_5)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_5) == genesMAold_5$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_5), genesMAold_5$GENEID)
keep4 <- rownames(ddsHTSeqMAold_5) %in% matched_2
ddsHTSeqMAold_5 <- ddsHTSeqMAold_5[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_5) == genesMAold_5$GENEID)
mcols(ddsHTSeqMAold_5) <- cbind(mcols(ddsHTSeqMAold_5), genesMAold_5)

# geneIDMAold_6 <- rownames(ddsHTSeqMAold_6)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAold_6 <- ensembldb::select(txdb, keys=geneIDMAold_6, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAold_6 <- na.omit(genesMAold_6)
# genesMAold_6 <- as.data.frame(genesMAold_6)
# remove1 <- "chrM"
# # genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAold_6) == genesMAold_6$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matched_2 <- intersect(rownames(ddsHTSeqMAold_6), genesMAold_6$GENEID)
# keep4 <- rownames(ddsHTSeqMAold_6) %in% matched_2
# ddsHTSeqMAold_6 <- ddsHTSeqMAold_6[keep4,]
# # length(ddsHTSeqMAold)
# all(rownames(ddsHTSeqMAold_6) == genesMAold_6$GENEID)
# mcols(ddsHTSeqMAold_6) <- cbind(mcols(ddsHTSeqMAold_6), genesMAold_6)

geneIDMAold_7 <- rownames(ddsHTSeqMAold_7)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_7 <- ensembldb::select(txdb, keys=geneIDMAold_7, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_7 <- na.omit(genesMAold_7)
genesMAold_7 <- as.data.frame(genesMAold_7)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_7) == genesMAold_7$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_7), genesMAold_7$GENEID)
keep4 <- rownames(ddsHTSeqMAold_7) %in% matched_2
ddsHTSeqMAold_7 <- ddsHTSeqMAold_7[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_7) == genesMAold_7$GENEID)
mcols(ddsHTSeqMAold_7) <- cbind(mcols(ddsHTSeqMAold_7), genesMAold_7)

geneIDMAold_8 <- rownames(ddsHTSeqMAold_8)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_8 <- ensembldb::select(txdb, keys=geneIDMAold_8, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_8 <- na.omit(genesMAold_8)
genesMAold_8 <- as.data.frame(genesMAold_8)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_8) == genesMAold_8$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_8), genesMAold_8$GENEID)
keep4 <- rownames(ddsHTSeqMAold_8) %in% matched_2
ddsHTSeqMAold_8 <- ddsHTSeqMAold_8[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_8) == genesMAold_8$GENEID)
mcols(ddsHTSeqMAold_8) <- cbind(mcols(ddsHTSeqMAold_8), genesMAold_8)

geneIDMAold_9 <- rownames(ddsHTSeqMAold_9)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_9 <- ensembldb::select(txdb, keys=geneIDMAold_9, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_9 <- na.omit(genesMAold_9)
genesMAold_9 <- as.data.frame(genesMAold_9)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_9) == genesMAold_9$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_9), genesMAold_9$GENEID)
keep4 <- rownames(ddsHTSeqMAold_9) %in% matched_2
ddsHTSeqMAold_9 <- ddsHTSeqMAold_9[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_9) == genesMAold_9$GENEID)
mcols(ddsHTSeqMAold_9) <- cbind(mcols(ddsHTSeqMAold_9), genesMAold_9)

geneIDMAold_11 <- rownames(ddsHTSeqMAold_11)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_11 <- ensembldb::select(txdb, keys=geneIDMAold_11, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_11 <- na.omit(genesMAold_11)
genesMAold_11 <- as.data.frame(genesMAold_11)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_11) == genesMAold_11$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_11), genesMAold_11$GENEID)
keep4 <- rownames(ddsHTSeqMAold_11) %in% matched_2
ddsHTSeqMAold_11 <- ddsHTSeqMAold_11[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_11) == genesMAold_11$GENEID)
mcols(ddsHTSeqMAold_11) <- cbind(mcols(ddsHTSeqMAold_11), genesMAold_11)

geneIDMAold_15 <- rownames(ddsHTSeqMAold_15)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_15 <- ensembldb::select(txdb, keys=geneIDMAold_15, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_15 <- na.omit(genesMAold_15)
genesMAold_15 <- as.data.frame(genesMAold_15)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_15) == genesMAold_15$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_15), genesMAold_15$GENEID)
keep4 <- rownames(ddsHTSeqMAold_15) %in% matched_2
ddsHTSeqMAold_15 <- ddsHTSeqMAold_15[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_15) == genesMAold_15$GENEID)
mcols(ddsHTSeqMAold_15) <- cbind(mcols(ddsHTSeqMAold_15), genesMAold_15)

geneIDMAold_28 <- rownames(ddsHTSeqMAold_28)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_28 <- ensembldb::select(txdb, keys=geneIDMAold_28, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_28 <- na.omit(genesMAold_28)
genesMAold_28 <- as.data.frame(genesMAold_28)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_28) == genesMAold_28$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_28), genesMAold_28$GENEID)
keep4 <- rownames(ddsHTSeqMAold_28) %in% matched_2
ddsHTSeqMAold_28 <- ddsHTSeqMAold_28[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_28) == genesMAold_28$GENEID)
mcols(ddsHTSeqMAold_28) <- cbind(mcols(ddsHTSeqMAold_28), genesMAold_28)

geneIDMAold_88 <- rownames(ddsHTSeqMAold_88)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_88 <- ensembldb::select(txdb, keys=geneIDMAold_88, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_88 <- na.omit(genesMAold_88)
genesMAold_88 <- as.data.frame(genesMAold_88)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_88) == genesMAold_88$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_88), genesMAold_88$GENEID)
keep4 <- rownames(ddsHTSeqMAold_88) %in% matched_2
ddsHTSeqMAold_88 <- ddsHTSeqMAold_88[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_88) == genesMAold_88$GENEID)
mcols(ddsHTSeqMAold_88) <- cbind(mcols(ddsHTSeqMAold_88), genesMAold_88)

geneIDMAold_108 <- rownames(ddsHTSeqMAold_108)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_108 <- ensembldb::select(txdb, keys=geneIDMAold_108, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_108 <- na.omit(genesMAold_108)
genesMAold_108 <- as.data.frame(genesMAold_108)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_108) == genesMAold_108$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_108), genesMAold_108$GENEID)
keep4 <- rownames(ddsHTSeqMAold_108) %in% matched_2
ddsHTSeqMAold_108 <- ddsHTSeqMAold_108[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_108) == genesMAold_108$GENEID)
mcols(ddsHTSeqMAold_108) <- cbind(mcols(ddsHTSeqMAold_108), genesMAold_108)

geneIDMAold_119 <- rownames(ddsHTSeqMAold_119)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_119 <- ensembldb::select(txdb, keys=geneIDMAold_119, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_119 <- na.omit(genesMAold_119)
genesMAold_119 <- as.data.frame(genesMAold_119)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_119) == genesMAold_119$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_119), genesMAold_119$GENEID)
keep4 <- rownames(ddsHTSeqMAold_119) %in% matched_2
ddsHTSeqMAold_119 <- ddsHTSeqMAold_119[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_119) == genesMAold_119$GENEID)
mcols(ddsHTSeqMAold_119) <- cbind(mcols(ddsHTSeqMAold_119), genesMAold_119)





#####################################################
#default
## add annotations
# geneIDMAnew <- rownames(ddsHTSeqMAnew)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnew <- ensembldb::select(txdb, keys=geneIDMAnew, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnew <- na.omit(genesMAnew)
# genesMAnew <- as.data.frame(genesMAnew)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnew) == genesMAnew$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedMAnew <- intersect(rownames(ddsHTSeqMAnew), genesMAnew$GENEID)
# keepMAnew<- rownames(ddsHTSeqMAnew) %in% matchedMAnew
# ddsHTSeqMAnew <- ddsHTSeqMAnew[keepMAnew,]
# length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnew) == genesMAnew$GENEID)
# mcols(ddsHTSeqMAnew) <- cbind(mcols(ddsHTSeqMAnew), genesMAnew)


#### individual samples
geneIDMAnewS29 <- rownames(ddsHTSeqMAnewS29)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAnewS29 <- ensembldb::select(txdb, keys=geneIDMAnewS29, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS29 <- na.omit(genesMAnewS29)
genesMAnewS29 <- as.data.frame(genesMAnewS29)
remove1 <- "chrM"
# genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAnewS29) == genesMAnewS29$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS29), genesMAnewS29$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS29) %in% matchedS2
ddsHTSeqMAnewS29 <- ddsHTSeqMAnewS29[keep4,]
# length(ddsHTSeqMAnew)
all(rownames(ddsHTSeqMAnewS29) == genesMAnewS29$GENEID)
mcols(ddsHTSeqMAnewS29) <- cbind(mcols(ddsHTSeqMAnewS29), genesMAnewS29)

geneIDMAnewS50 <- rownames(ddsHTSeqMAnewS50)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAnewS50 <- ensembldb::select(txdb, keys=geneIDMAnewS50, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS50 <- na.omit(genesMAnewS50)
genesMAnewS50 <- as.data.frame(genesMAnewS50)
remove1 <- "chrM"
# genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAnewS50) == genesMAnewS50$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS50), genesMAnewS50$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS50) %in% matchedS2
ddsHTSeqMAnewS50 <- ddsHTSeqMAnewS50[keep4,]
# length(ddsHTSeqMAnew)
all(rownames(ddsHTSeqMAnewS50) == genesMAnewS50$GENEID)
mcols(ddsHTSeqMAnewS50) <- cbind(mcols(ddsHTSeqMAnewS50), genesMAnewS50)

geneIDMAnewS112 <- rownames(ddsHTSeqMAnewS112)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAnewS112 <- ensembldb::select(txdb, keys=geneIDMAnewS112, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS112 <- na.omit(genesMAnewS112)
genesMAnewS112 <- as.data.frame(genesMAnewS112)
remove1 <- "chrM"
# genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAnewS112) == genesMAnewS112$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS112), genesMAnewS112$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS112) %in% matchedS2
ddsHTSeqMAnewS112 <- ddsHTSeqMAnewS112[keep4,]
# length(ddsHTSeqMAnew)
all(rownames(ddsHTSeqMAnewS112) == genesMAnewS112$GENEID)
mcols(ddsHTSeqMAnewS112) <- cbind(mcols(ddsHTSeqMAnewS112), genesMAnewS112)
#
geneIDMAnewS115 <- rownames(ddsHTSeqMAnewS115)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAnewS115 <- ensembldb::select(txdb, keys=geneIDMAnewS115, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS115 <- na.omit(genesMAnewS115)
genesMAnewS115 <- as.data.frame(genesMAnewS115)
remove1 <- "chrM"
# genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAnewS115) == genesMAnewS115$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS115), genesMAnewS115$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS115) %in% matchedS2
ddsHTSeqMAnewS115 <- ddsHTSeqMAnewS115[keep4,]
# length(ddsHTSeqMAnew)
all(rownames(ddsHTSeqMAnewS115) == genesMAnewS115$GENEID)
mcols(ddsHTSeqMAnewS115) <- cbind(mcols(ddsHTSeqMAnewS115), genesMAnewS115)

geneIDMAnewS117 <- rownames(ddsHTSeqMAnewS117)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAnewS117 <- ensembldb::select(txdb, keys=geneIDMAnewS117, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS117 <- na.omit(genesMAnewS117)
genesMAnewS117 <- as.data.frame(genesMAnewS117)
remove1 <- "chrM"
# genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAnewS117) == genesMAnewS117$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS117), genesMAnewS117$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS117) %in% matchedS2
ddsHTSeqMAnewS117 <- ddsHTSeqMAnewS117[keep4,]
# length(ddsHTSeqMAnew)
all(rownames(ddsHTSeqMAnewS117) == genesMAnewS117$GENEID)
mcols(ddsHTSeqMAnewS117) <- cbind(mcols(ddsHTSeqMAnewS117), genesMAnewS117)

geneIDMAnewS123 <- rownames(ddsHTSeqMAnewS123)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAnewS123 <- ensembldb::select(txdb, keys=geneIDMAnewS123, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS123 <- na.omit(genesMAnewS123)
genesMAnewS123 <- as.data.frame(genesMAnewS123)
remove1 <- "chrM"
# genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAnewS123) == genesMAnewS123$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS123), genesMAnewS123$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS123) %in% matchedS2
ddsHTSeqMAnewS123 <- ddsHTSeqMAnewS123[keep4,]
# length(ddsHTSeqMAnew)
all(rownames(ddsHTSeqMAnewS123) == genesMAnewS123$GENEID)
mcols(ddsHTSeqMAnewS123) <- cbind(mcols(ddsHTSeqMAnewS123), genesMAnewS123)

# geneIDMAnewS141 <- rownames(ddsHTSeqMAnewS141)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS141 <- ensembldb::select(txdb, keys=geneIDMAnewS141, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS141 <- na.omit(genesMAnewS141)
# genesMAnewS141 <- as.data.frame(genesMAnewS141)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS141) == genesMAnewS141$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2 <- intersect(rownames(ddsHTSeqMAnewS141), genesMAnewS141$GENEID)
# keep4 <- rownames(ddsHTSeqMAnewS141) %in% matchedS2
# ddsHTSeqMAnewS141 <- ddsHTSeqMAnewS141[keep4,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS141) == genesMAnewS141$GENEID)
# mcols(ddsHTSeqMAnewS141) <- cbind(mcols(ddsHTSeqMAnewS141), genesMAnewS141)

geneIDMAnewS152 <- rownames(ddsHTSeqMAnewS152)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAnewS152 <- ensembldb::select(txdb, keys=geneIDMAnewS152, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS152 <- na.omit(genesMAnewS152)
genesMAnewS152 <- as.data.frame(genesMAnewS152)
remove1 <- "chrM"
# genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAnewS152) == genesMAnewS152$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS152), genesMAnewS152$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS152) %in% matchedS2
ddsHTSeqMAnewS152 <- ddsHTSeqMAnewS152[keep4,]
# length(ddsHTSeqMAnew)
all(rownames(ddsHTSeqMAnewS152) == genesMAnewS152$GENEID)
mcols(ddsHTSeqMAnewS152) <- cbind(mcols(ddsHTSeqMAnewS152), genesMAnewS152)

### with run included as factor in design
# geneIDMAnewS29.run <- rownames(ddsHTSeqMAnewS29.run)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS29.run <- ensembldb::select(txdb, keys=geneIDMAnewS29.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS29.run <- na.omit(genesMAnewS29.run)
# genesMAnewS29.run <- as.data.frame(genesMAnewS29.run)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS29.run) == genesMAnewS29.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqMAnewS29.run), genesMAnewS29.run$GENEID)
# keep4.run <- rownames(ddsHTSeqMAnewS29.run) %in% matchedS2.run
# ddsHTSeqMAnewS29.run <- ddsHTSeqMAnewS29.run[keep4.run,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS29.run) == genesMAnewS29.run$GENEID)
# mcols(ddsHTSeqMAnewS29.run) <- cbind(mcols(ddsHTSeqMAnewS29.run), genesMAnewS29.run)
#
# geneIDMAnewS50.run <- rownames(ddsHTSeqMAnewS50.run)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS50.run <- ensembldb::select(txdb, keys=geneIDMAnewS50.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS50.run <- na.omit(genesMAnewS50.run)
# genesMAnewS50.run <- as.data.frame(genesMAnewS50.run)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS50.run) == genesMAnewS50.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqMAnewS50.run), genesMAnewS50.run$GENEID)
# keep4.run <- rownames(ddsHTSeqMAnewS50.run) %in% matchedS2.run
# ddsHTSeqMAnewS50.run <- ddsHTSeqMAnewS50.run[keep4.run,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS50.run) == genesMAnewS50.run$GENEID)
# mcols(ddsHTSeqMAnewS50.run) <- cbind(mcols(ddsHTSeqMAnewS50.run), genesMAnewS50.run)
#
# geneIDMAnewS112.run <- rownames(ddsHTSeqMAnewS112.run)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS112.run <- ensembldb::select(txdb, keys=geneIDMAnewS112.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS112.run <- na.omit(genesMAnewS112.run)
# genesMAnewS112.run <- as.data.frame(genesMAnewS112.run)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS112.run) == genesMAnewS112.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqMAnewS112.run), genesMAnewS112.run$GENEID)
# keep4.run <- rownames(ddsHTSeqMAnewS112.run) %in% matchedS2.run
# ddsHTSeqMAnewS112.run <- ddsHTSeqMAnewS112.run[keep4.run,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS112.run) == genesMAnewS112.run$GENEID)
# mcols(ddsHTSeqMAnewS112.run) <- cbind(mcols(ddsHTSeqMAnewS112.run), genesMAnewS112.run)
# #
# geneIDMAnewS115.run <- rownames(ddsHTSeqMAnewS115.run)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS115.run <- ensembldb::select(txdb, keys=geneIDMAnewS115.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS115.run <- na.omit(genesMAnewS115.run)
# genesMAnewS115.run <- as.data.frame(genesMAnewS115.run)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS115.run) == genesMAnewS115.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqMAnewS115.run), genesMAnewS115.run$GENEID)
# keep4.run <- rownames(ddsHTSeqMAnewS115.run) %in% matchedS2.run
# ddsHTSeqMAnewS115.run <- ddsHTSeqMAnewS115.run[keep4.run,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS115.run) == genesMAnewS115.run$GENEID)
# mcols(ddsHTSeqMAnewS115.run) <- cbind(mcols(ddsHTSeqMAnewS115.run), genesMAnewS115.run)
#
# geneIDMAnewS117.run <- rownames(ddsHTSeqMAnewS117.run)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS117.run <- ensembldb::select(txdb, keys=geneIDMAnewS117.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS117.run <- na.omit(genesMAnewS117.run)
# genesMAnewS117.run <- as.data.frame(genesMAnewS117.run)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS117.run) == genesMAnewS117.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqMAnewS117.run), genesMAnewS117.run$GENEID)
# keep4.run <- rownames(ddsHTSeqMAnewS117.run) %in% matchedS2.run
# ddsHTSeqMAnewS117.run <- ddsHTSeqMAnewS117.run[keep4.run,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS117.run) == genesMAnewS117.run$GENEID)
# mcols(ddsHTSeqMAnewS117.run) <- cbind(mcols(ddsHTSeqMAnewS117.run), genesMAnewS117.run)
#
# geneIDMAnewS123.run <- rownames(ddsHTSeqMAnewS123.run)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS123.run <- ensembldb::select(txdb, keys=geneIDMAnewS123.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS123.run <- na.omit(genesMAnewS123.run)
# genesMAnewS123.run <- as.data.frame(genesMAnewS123.run)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS123.run) == genesMAnewS123.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqMAnewS123.run), genesMAnewS123.run$GENEID)
# keep4.run <- rownames(ddsHTSeqMAnewS123.run) %in% matchedS2.run
# ddsHTSeqMAnewS123.run <- ddsHTSeqMAnewS123.run[keep4.run,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS123.run) == genesMAnewS123.run$GENEID)
# mcols(ddsHTSeqMAnewS123.run) <- cbind(mcols(ddsHTSeqMAnewS123.run), genesMAnewS123.run)
#
# # geneIDMAnewS141 <- rownames(ddsHTSeqMAnewS141)
# # #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# # #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# # genesMAnewS141 <- ensembldb::select(txdb, keys=geneIDMAnewS141, columnns="GENEID","TXCHROM" , keytype="GENEID")
# # #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# # genesMAnewS141 <- na.omit(genesMAnewS141)
# # genesMAnewS141 <- as.data.frame(genesMAnewS141)
# # remove1 <- "chrM"
# # # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# # all(rownames(ddsHTSeqMAnewS141) == genesMAnewS141$GENEID)
# # #need to remove the ones in rownames of dds that don't match genes$GENEID
# # matchedS2 <- intersect(rownames(ddsHTSeqMAnewS141), genesMAnewS141$GENEID)
# # keep4 <- rownames(ddsHTSeqMAnewS141) %in% matchedS2
# # ddsHTSeqMAnewS141 <- ddsHTSeqMAnewS141[keep4,]
# # # length(ddsHTSeqMAnew)
# # all(rownames(ddsHTSeqMAnewS141) == genesMAnewS141$GENEID)
# # mcols(ddsHTSeqMAnewS141) <- cbind(mcols(ddsHTSeqMAnewS141), genesMAnewS141)
# #
# geneIDMAnewS152.run <- rownames(ddsHTSeqMAnewS152.run)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS152.run <- ensembldb::select(txdb, keys=geneIDMAnewS152.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS152.run <- na.omit(genesMAnewS152.run)
# genesMAnewS152.run <- as.data.frame(genesMAnewS152.run)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS152.run) == genesMAnewS152.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqMAnewS152.run), genesMAnewS152.run$GENEID)
# keep4.run <- rownames(ddsHTSeqMAnewS152.run) %in% matchedS2.run
# ddsHTSeqMAnewS152.run <- ddsHTSeqMAnewS152.run[keep4.run,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS152.run) == genesMAnewS152.run$GENEID)
# mcols(ddsHTSeqMAnewS152.run) <- cbind(mcols(ddsHTSeqMAnewS152.run), genesMAnewS152.run)
#
#
# #################################################
# #default DESeq run
# # ddsDefGC.short <- DESeq(ddsHTSeqGC.short)
# # ddsDefGC.short
# # attr(dispersionFunction(ddsDefGC.short),"dispPriorVar")
# # norm.factors.with.aneuGC <- as.data.frame(sizeFactors(ddsDefGC.short))
# # dispersions.with.aneuGC <- dispersions(ddsDefGC.short)
# ## with filtered genes
# #default DESeq run
# # ddsDefGC.fil <- DESeq(ddsHTSeqGC.fil)
# # ddsDefGC.fil
# # attr(dispersionFunction(ddsDefGC.fil),"dispPriorVar")
# # norm.factors.with.aneuGC.fil <- as.data.frame(sizeFactors(ddsDefGC.fil))
# # dispersions.with.aneuGC.fil <- dispersions(ddsDefGC.fil)
#
# ## with filtered genes
# #default DESeq run
# # with each sample individually
# # ddsHTSeqGCS1$run <- droplevels(ddsHTSeqGCS1$run)
# ddsDefGCS1.run <- DESeq(ddsHTSeqGCS1.run)
#
# ddsDefGCS1.run
# ddsDefGCS2.run <- DESeq(ddsHTSeqGCS2.run)
# ddsDefGCS2.run
# ddsDefGCS3.run <- DESeq(ddsHTSeqGCS3.run)
# ddsDefGCS3.run
# ddsDefGCS4.run <- DESeq(ddsHTSeqGCS4.run)
# ddsDefGCS4.run
# ddsDefGCS5.run <- DESeq(ddsHTSeqGCS5.run)
# ddsDefGCS5.run
# ddsDefGCS6.run <-DESeq(ddsHTSeqGCS6.run)
# ddsDefGCS6.run
# ddsDefGCS7.run <- DESeq(ddsHTSeqGCS7.run)
# ddsDefGCS7.run
# ddsDefGCS8.run <- DESeq(ddsHTSeqGCS8.run)
# ddsDefGCS8.run
# ddsDefGCS9.run <- DESeq(ddsHTSeqGCS9.run)
# ddsDefGCS9.run
# # ddsHTSeqGCS11$run <- droplevels(ddsHTSeqGCS11$run)
# ddsDefGCS11.run <- DESeq(ddsHTSeqGCS11.run)
# ddsDefGCS11.run
# ddsDefGCS18.run <- DESeq(ddsHTSeqGCS18.run)
# ddsDefGCS18.run
# ddsDefGCS31.run <- DESeq(ddsHTSeqGCS31.run)
# ddsDefGCS31.run
# ddsDefGCS49.run <- DESeq(ddsHTSeqGCS49.run)
# ddsDefGCS49.run
# ddsDefGCS59.run <- DESeq(ddsHTSeqGCS59.run)
# ddsDefGCS59.run
# ddsDefGCS61.run <- DESeq(ddsHTSeqGCS61.run)
# ddsDefGCS61.run
# ddsDefGCS69.run <- DESeq(ddsHTSeqGCS69.run)
# ddsDefGCS69.run
# ddsDefGCS76.run <- DESeq(ddsHTSeqGCS76.run)
# ddsDefGCS76.run
# ddsDefGCS77.run <- DESeq(ddsHTSeqGCS77.run)
# ddsDefGCS77.run


### with run included as factor in design matrix

ddsDefGC_1 <- DESeq(ddsHTSeqGC_1)
ddsDefGC_1
ddsDefGC_2 <- DESeq(ddsHTSeqGC_2)
ddsDefGC_2
ddsDefGC_3 <- DESeq(ddsHTSeqGC_3)
ddsDefGC_3
ddsDefGC_4 <- DESeq(ddsHTSeqGC_4)
ddsDefGC_4
ddsDefGC_5 <- DESeq(ddsHTSeqGC_5)
ddsDefGC_5
# ddsDefGC_6 <- DESeq(ddsHTSeqGC_6)
# ddsDefGC_6
ddsDefGC_7 <- DESeq(ddsHTSeqGC_7)
ddsDefGC_7
ddsDefGC_8 <- DESeq(ddsHTSeqGC_8)
ddsDefGC_8
ddsDefGC_9 <- DESeq(ddsHTSeqGC_9)
ddsDefGC_9
# ddsHTSeqGC_11$run <- droplevels(ddsHTSeqGC_11$run)
ddsDefGC_11 <- DESeq(ddsHTSeqGC_11)
ddsDefGC_11
ddsDefGC_18 <- DESeq(ddsHTSeqGC_18)
ddsDefGC_18
# ddsDefGC_31 <- DESeq(ddsHTSeqGC_31)
# ddsDefGC_31
ddsDefGC_49 <- DESeq(ddsHTSeqGC_49)
ddsDefGC_49
ddsDefGC_59 <- DESeq(ddsHTSeqGC_59)
ddsDefGC_59
ddsDefGC_61 <- DESeq(ddsHTSeqGC_61)
ddsDefGC_61
# ddsDefGC_69 <- DESeq(ddsHTSeqGC_69)
# ddsDefGC_69
ddsDefGC_76 <- DESeq(ddsHTSeqGC_76)
ddsDefGC_76
ddsDefGC_77 <- DESeq(ddsHTSeqGC_77)
ddsDefGC_77
#################################################
#default DESeq run
# ddsDefMAold <- DESeq(ddsHTSeqMAold)
# ddsDefMAold
# attr(dispersionFunction(ddsDefMAold),"dispPriorVar")
# norm.factors.with.aneuMAold <- as.data.frame(sizeFactors(ddsDefMAold))
# dispersions.with.aneuMAold <- dispersions(ddsDefMAold)

### individual lines
ddsDefMAold_1 <- DESeq(ddsHTSeqMAold_1)
ddsDefMAold_1
ddsDefMAold_2 <- DESeq(ddsHTSeqMAold_2)
ddsDefMAold_2
ddsDefMAold_3 <- DESeq(ddsHTSeqMAold_3)
ddsDefMAold_3
ddsDefMAold_4 <- DESeq(ddsHTSeqMAold_4)
ddsDefMAold_4
ddsDefMAold_5 <- DESeq(ddsHTSeqMAold_5)
ddsDefMAold_5
# ddsDefMAold_6 <- DESeq(ddsHTSeqMAold_6)
# ddsDefMAold_6
ddsDefMAold_7 <- DESeq(ddsHTSeqMAold_7)
ddsDefMAold_7
ddsDefMAold_8 <- DESeq(ddsHTSeqMAold_8)
ddsDefMAold_8
ddsDefMAold_9 <- DESeq(ddsHTSeqMAold_9)
ddsDefMAold_9
ddsDefMAold_11 <- DESeq(ddsHTSeqMAold_11)
ddsDefMAold_11
ddsDefMAold_15 <- DESeq(ddsHTSeqMAold_15)
ddsDefMAold_15
ddsDefMAold_28 <- DESeq(ddsHTSeqMAold_28)
ddsDefMAold_28
ddsDefMAold_88 <- DESeq(ddsHTSeqMAold_88)
ddsDefMAold_88
ddsDefMAold_108 <- DESeq(ddsHTSeqMAold_108)
ddsDefMAold_108
ddsDefMAold_119 <- DESeq(ddsHTSeqMAold_119)
ddsDefMAold_119
#################################################
#default DESeq run
# ddsHTSeqMAnew.short$run <- droplevels(ddsHTSeqMAnew.short$run)
# ddsHTSeqMAnew$condition <- droplevels(ddsHTSeqMAnew$condition)

# ddsDefMAnew <- DESeq(ddsHTSeqMAnew)
# ddsDefMAnew
# attr(dispersionFunction(ddsDefMAnew),"dispPriorVar")
# norm.factors.with.aneuMAnew <- as.data.frame(sizeFactors(ddsDefMAnew))
# dispersions.with.aneuMAnew <- dispersions(ddsDefMAnew)


##### individual samples
ddsDefMAnewS29 <- DESeq(ddsHTSeqMAnewS29)
ddsDefMAnewS50 <- DESeq(ddsHTSeqMAnewS50)
ddsDefMAnewS112 <- DESeq(ddsHTSeqMAnewS112)
ddsDefMAnewS115 <- DESeq(ddsHTSeqMAnewS115)
ddsDefMAnewS117 <- DESeq(ddsHTSeqMAnewS117)
ddsDefMAnewS123 <- DESeq(ddsHTSeqMAnewS123)
# ddsDefMAnewS141 <- DESeq(ddsHTSeqMAnewS141)
ddsDefMAnewS152 <- DESeq(ddsHTSeqMAnewS152)

# ## with run included as factor
# ddsDefMAnewS29.run <- DESeq(ddsHTSeqMAnewS29.run)
# ddsDefMAnewS50.run <- DESeq(ddsHTSeqMAnewS50.run)
# ddsDefMAnewS112.run <- DESeq(ddsHTSeqMAnewS112.run)
# ddsDefMAnewS115.run <- DESeq(ddsHTSeqMAnewS115.run)
# ddsDefMAnewS117.run <- DESeq(ddsHTSeqMAnewS117.run)
# ddsDefMAnewS123.run <- DESeq(ddsHTSeqMAnewS123.run)
# # ddsDefMAnewS141 <- DESeq(ddsHTSeqMAnewS141)
# ddsDefMAnewS152.run <- DESeq(ddsHTSeqMAnewS152.run)
#

#######################################################
#### results with aneuploid chromosomes included ######
# resultsNames(ddsDefGC.short)
# resultsS1GC <- results(ddsDefGC,contrast=c("condition","Holly_1","Holly_Anc"))
# resultsS1GC
# sigGenesGC1 <- rownames(subset(resultsS1GC,padj<0.1))
# summary(resultsS1GC)

# resultsNames(ddsDefGC.short)
# resultsS1GC.sh <- results(ddsDefGC.short,contrast=c("condition","Holly_1","Holly_Anc"))
# resultsS1GC.sh
# # sigGenesGC1 <- rownames(subset(resultsS1GC,padj<0.1))
# summary(resultsS1GC.sh)

# resultsNames(ddsDefGC.short)
# resultsS2GC <- results(ddsDefGC.short,contrast=c("condition","Holly_2","Holly_Anc"))
# resultsS2GC.sh
# sigGenesGC1 <- rownames(subset(resultsS1GC,padj<0.1))
# summary(resultsS2GC)

# resultsS2GC <- results(ddsDefGC,contrast=c("condition","Holly_2","Holly_Anc"))
# resultsS2GC
# sigGenesGC2 <- rownames(subset(resultsS2GC,padj<0.1))
# summary(resultsS2GC)

# resultsS3GC <- results(ddsDefGC.short,contrast=c("condition","Holly_3","Holly_Anc"))
# resultsS3GC
# sigGenesGC3 <- rownames(subset(resultsS3GC,padj<0.1))
# summary(resultsS3GC)
#
# resultsS4GC <- results(ddsDefGC.short,contrast=c("condition","Holly_4","Holly_Anc"))
# resultsS4GC
# sigGenesGC4 <- rownames(subset(resultsS4GC,padj<0.1))
# summary(resultsS4GC)
#
# resultsS5GC <- results(ddsDefGC.short,contrast=c("condition","Holly_5","Holly_Anc"))
# resultsS5GC
# sigGenesGC5 <- rownames(subset(resultsS5GC,padj<0.1))
# summary(resultsS5GC)
#
# resultsS7GC <- results(ddsDefGC.short,contrast=c("condition","Holly_7","Holly_Anc"))
# resultsS7GC
# sigGenesGC7 <- rownames(subset(resultsS7GC,padj<0.1))
# summary(resultsS7GC)
#
# resultsS8GC <- results(ddsDefGC.short,contrast=c("condition","Holly_8","Holly_Anc"))
# resultsS8GC
# sigGenesGC8 <- rownames(subset(resultsS8GC,padj<0.1))
# summary(resultsS8GC)
#
#
# resultsS9GC <- results(ddsDefGC.short,contrast=c("condition","Holly_9","Holly_Anc"))
# resultsS9GC
# sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS9GC)
#
#   resultsS1GCDat <- as.data.frame(resultsS1GC)
#     resultsS2GCDat <- as.data.frame(resultsS2GC)
#       resultsS3GCDat <- as.data.frame(resultsS3GC)
#         resultsS5GCDat <- as.data.frame(resultsS5GC)
#           resultsS9GCDat <- as.data.frame(resultsS9GC)

  # add gene names as first column ( which is currently rownames)
  # resS1GCDat <- cbind(GENEID = rownames(resultsS1GCDat), resultsS1GCDat)
  # resS2GCDat <- cbind(GENEID = rownames(resultsS2GCDat), resultsS2GCDat)
  # resS3GCDat <- cbind(GENEID = rownames(resultsS3GCDat), resultsS3GCDat)
  # resS5GCDat <- cbind(GENEID = rownames(resultsS5GCDat), resultsS5GCDat)
  # resS9GCDat <- cbind(GENEID = rownames(resultsS9GCDat), resultsS9GCDat)

  # need to make a column identifying the sample
    # resS1GCDat$Sample <- "1"
    # resS2GCDat$Sample <- "2"
    # resS3GCDat$Sample <- "3"
    # resS5GCDat$Sample <- "5"
    # resS9GCDat$Sample <- "9"

## with filtered genes #################
    #### results with aneuploid chromosomes included ######
# resultsNames(ddsDefGC.fil)
# resultsS1GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_1","Holly_Anc"))
# resultsS1GC.fil
# sigGenesGC1 <- rownames(subset(resultsS1GC,padj<0.1))
# summary(resultsS1GC.fil)
# pdf("resultsS1GC.filMAplot.pdf")
# plotMA(resultsS1GC.fil,ylim=c(-2,2))
# dev.off()

# resultsS2GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_2","Holly_Anc"))
# resultsS2GC.fil
# sigGenesGC2 <- rownames(subset(resultsS2GC,padj<0.1))
# summary(resultsS2GC.fil)
# pdf("resultsS2GC.filMAplot.pdf")
# plotMA(resultsS2GC.fil,ylim=c(-2,2))
# dev.off()

# resultsS3GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_3","Holly_Anc"))
# resultsS3GC.fil
# # sigGenesGC3 <- rownames(subset(resultsS3GC,padj<0.1))
# summary(resultsS3GC.fil)
# pdf("resultsS3GC.filMAplot.pdf")
# plotMA(resultsS3GC.fil,ylim=c(-2,2))
# dev.off()

# resultsS5GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_5","Holly_Anc"))
# resultsS5GC.fil
# # sigGenesGC5 <- rownames(subset(resultsS5GC,padj<0.1))
# summary(resultsS5GC.fil)
# pdf("resultsS5GC.filMAplot.pdf")
# plotMA(resultsS5GC.fil)#,ylim=c(-2,2))
# dev.off()
#
# resultsS9GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_9","Holly_Anc"))
# resultsS9GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS9GC.fil)

# resultsS4GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_4","Holly_Anc"))
# resultsS4GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS4GC.fil)
#
# resultsS7GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_7","Holly_Anc"))
# resultsS7GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS7GC.fil)

# resultsS8GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_8","Holly_Anc"))
# resultsS8GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS8GC.fil)
#
# resultsS11GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_11","Holly_Anc"))
# resultsS11GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS11GC.fil)

# resultsS18GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_18","Holly_Anc"))
# resultsS18GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS18GC.fil)
#
# resultsS49GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_49","Holly_Anc"))
# resultsS49GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS49GC.fil)

# resultsS59GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_59","Holly_Anc"))
# resultsS59GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS59GC.fil)
#
# resultsS61GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_61","Holly_Anc"))
# resultsS61GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS61GC.fil)

# resultsS76GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_76","Holly_Anc"))
# resultsS76GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS76GC.fil)
#
# resultsS77GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_77","Holly_Anc"))
# resultsS77GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS77GC.fil)

# with individual samples
resultsNames(ddsDefGC_1)
resultsS1GC <- results(ddsDefGC_1,contrast=c("condition","Holly_1","Holly_Anc"),alpha=0.05)
summary(resultsS1GC)
# alpha set to 0.05, does it make a huge difference?
# resultsS1GC <- results(ddsDefGC_1,contrast=c("condition","Holly_1","Holly_Anc"))
# summary(resultsS1GC)
resultsS2GC <- results(ddsDefGC_2,contrast=c("condition","Holly_2","Holly_Anc"))
summary(resultsS2GC)
resultsS2GC <- results(ddsDefGC_2,contrast=c("condition","Holly_2","Holly_Anc"),alpha=0.05)
summary(resultsS2GC)
resultsS3GC <- results(ddsDefGC_3,contrast=c("condition","Holly_3","Holly_Anc"),alpha=0.05)
summary(resultsS3GC)
resultsS4GC <- results(ddsDefGC_4,contrast=c("condition","Holly_4","Holly_Anc"),alpha=0.05)
summary(resultsS4GC)
resultsS49GC <- results(ddsDefGC_49,contrast=c("condition","Holly_49","Holly_Anc"),alpha=0.05)
summary(resultsS49GC)
resultsS5GC <- results(ddsDefGC_5,contrast=c("condition","Holly_5","Holly_Anc"),alpha=0.05)
summary(resultsS5GC)
# resultsS6GC <- results(ddsDefGC_6,contrast=c("condition","Holly_6","Holly_Anc"),alpha=0.05)
# summary(resultsS6GC)
resultsS7GC <- results(ddsDefGC_7,contrast=c("condition","Holly_7","Holly_Anc"),alpha=0.05)
summary(resultsS7GC)
resultsS8GC <- results(ddsDefGC_8,contrast=c("condition","Holly_8","Holly_Anc"),alpha=0.05)
summary(resultsS8GC)
resultsS9GC <- results(ddsDefGC_9,contrast=c("condition","Holly_9","Holly_Anc"),alpha=0.05)
summary(resultsS9GC)

normCountsGC_9 <- counts(ddsDefGC_9,normalized=TRUE)
write.csv(normCountsGC_9,file="normCountsGC_9.csv")

resultsS11GC <- results(ddsDefGC_11,contrast=c("condition","Holly_11","Holly_Anc"),alpha=0.05)
summary(resultsS11GC)

normCountsGC_11 <- counts(ddsDefGC_11,normalized=TRUE)

resultsS18GC <- results(ddsDefGC_18,contrast=c("condition","Holly_18","Holly_Anc"),alpha=0.05)
summary(resultsS18GC)
# resultsS31GC <- results(ddsDefGC_31,contrast=c("condition","Holly_31","Holly_Anc"),alpha=0.05)
# summary(resultsS31GC)
# # resultsS49GC <- results(ddsDefGC_49,contrast=c("condition","Holly_49","Holly_Anc"),alpha=0.05)
# summary(resultsS49GC)
##without independent filtering
# resultsS49GC <- results(ddsDefGC_49,contrast=c("condition","Holly_49","Holly_Anc"),alpha=0.05)
# summary(resultsS49GC)
# resultsS49GC <- as.data.frame(resultsS49GC)
# resultsS49GC$padj
# normCountsGC_49 <- counts(ddsDefGC_49,normalized=TRUE)
# write.csv(normCountsGC_49,file="normCountsGC_49.csv")

resultsS59GC <- results(ddsDefGC_59,contrast=c("condition","Holly_59","Holly_Anc"),alpha=0.05)#,independentFiltering = FALSE)
summary(resultsS59GC)
normCountsGC_59 <- counts(ddsDefGC_59,normalized=TRUE)
write.csv(normCountsGC_59,file="normCountsGC_59.csv")

resultsS61GC <- results(ddsDefGC_61,contrast=c("condition","Holly_61","Holly_Anc"),alpha=0.05)
summary(resultsS61GC)
# resultsS69GC <- results(ddsDefGC_69,contrast=c("condition","Holly_69","Holly_Anc"))#,alpha=0.05)
# summary(resultsS69GC)
resultsS76GC <- results(ddsDefGC_76,contrast=c("condition","Holly_76","Holly_Anc"),alpha=0.05)
summary(resultsS76GC)
resultsS77GC <- results(ddsDefGC_77,contrast=c("condition","Holly_77","Holly_Anc"),alpha=0.05)
summary(resultsS77GC)

## with run included in design formula
# resultsNames(ddsDefGCS1.run)
# resultsS1GC.run <- results(ddsDefGCS1.run,contrast=c("condition","Holly_1","Holly_Anc"))
# summary(resultsS1GC.run)
# # alpha set to 0.05, does it make a huge difference?
# # resultsS1GC <- results(ddsDefGCS1.fil,contrast=c("condition","Holly_1","Holly_Anc"),alpha = 0.05)
# # summary(resultsS1GC)
# resultsS2GC.run <- results(ddsDefGCS2.run,contrast=c("condition","Holly_2","Holly_Anc"))
# summary(resultsS2GC.run)
# resultsS3GC.run <- results(ddsDefGCS3.run,contrast=c("condition","Holly_3","Holly_Anc"))#,alpha=0.05)
# summary(resultsS3GC.run)
# resultsS4GC.run <- results(ddsDefGCS4.run,contrast=c("condition","Holly_4","Holly_Anc"))#,alpha=0.05)
# summary(resultsS4GC.run)
# resultsS5GC.run <- results(ddsDefGCS5.run,contrast=c("condition","Holly_5","Holly_Anc"))#,alpha=0.05)
# summary(resultsS5GC.run)
# resultsS6GC.run <- results(ddsDefGCS6.run,contrast=c("condition","Holly_6","Holly_Anc"))#,alpha=0.05)
# summary(resultsS6GC.run)
# resultsS7GC.run <- results(ddsDefGCS7.run,contrast=c("condition","Holly_7","Holly_Anc"))#,alpha=0.05)
# summary(resultsS7GC.run)
# resultsS8GC.run <- results(ddsDefGCS8.run,contrast=c("condition","Holly_8","Holly_Anc"))#,alpha=0.05)
# summary(resultsS8GC.run)
# resultsS9GC.run <- results(ddsDefGCS9.run,contrast=c("condition","Holly_9","Holly_Anc"))#,alpha=0.05)#,independentFiltering = FALSE)
# summary(resultsS9GC.run)
# resultsS11GC.run <- results(ddsDefGCS11.run,contrast=c("condition","Holly_11","Holly_Anc"))#,alpha=0.05)
# summary(resultsS11GC.run)
#
# resultsS18GC.run <- results(ddsDefGCS18.run,contrast=c("condition","Holly_18","Holly_Anc"))#,alpha=0.05)
# summary(resultsS18GC.run)
# resultsS31GC.run <- results(ddsDefGCS31.run,contrast=c("condition","Holly_31","Holly_Anc"))#,alpha=0.05)
# summary(resultsS31GC.run)
# resultsS49GC.run <- results(ddsDefGCS49.run,contrast=c("condition","Holly_49","Holly_Anc"))#,alpha=0.05)
# summary(resultsS49GC.run)
# resultsS59GC.run <- results(ddsDefGCS59.run,contrast=c("condition","Holly_59","Holly_Anc"))#,alpha=0.05)#,independentFiltering = FALSE)
# summary(resultsS59GC.run)
# resultsS61GC.run <- results(ddsDefGCS61.run,contrast=c("condition","Holly_61","Holly_Anc"))#,alpha=0.05)
# summary(resultsS61GC.run)
# resultsS69GC.run <- results(ddsDefGCS69.run,contrast=c("condition","Holly_69","Holly_Anc"))#,alpha=0.05)
# summary(resultsS69GC.run)
# resultsS76GC.run <- results(ddsDefGCS76.run,contrast=c("condition","Holly_76","Holly_Anc"))#,alpha=0.05)
# summary(resultsS76GC.run)
# resultsS77GC.run <- results(ddsDefGCS77.run,contrast=c("condition","Holly_77","Holly_Anc"))#,alpha=0.05)
# summary(resultsS77GC.run)

 # put all sample expression data in the same data frame
# euploidGCSamples <- rbind(resS1GCDat, resS2GCDat, resS3GCDat, resS5GCDat, resS9GCDat)
# dim(euploidGCSamples)
# View(euploidGCSamples)
  # # # add chromosome numbers to this file
      # match <- intersect(euploidGCSamples$GENEID, genesGC$GENEID)
      # length(match)
  # #   # tell R which genes to keep
      # rm(keep)
      # keep <- euploidGCSamples$GENEID %in% match
      # length(keep)
      # View(keep)
      # View(euploidGCSamples$GENEID)
# #   # subset list of genes in analysis to include only the ones matching those found in          sample file
#     genesGCeu <- genesGC[keep,]
# # #   # add the correct chromosomes in
#     length(genesGCeu)
#     length(euploidGCSamples)
#     dim(genesGCeu)
#     dim(euploidGCSamples)
#     euploidGCSamples <- cbind(genesGCeu$GENEID, genesGCeu$TXCHROM, euploidGCSamples)
#     # chr1.all <- subset(euploidGCSamplesDE, euploidGCSamplesDE$`genesNoAneu$GENEID`=="chrI")
#     sample1GC <- subset(euploidGCSamples, Sample==1)
#     View(sample1GC)
#     # sample7 <- subset(allSamplesExpr, Sample==7)
# # # first need to filter allSamplesExpr based on p-adj level
#     euploidGCSamplesDiffExpr <- subset(euploidGCSamples, padj<0.1)
#     
    # genesS1GCDE <- subset(euploidGCSamplesDiffExpr, Sample=="1")
    # genesS2GCDE <- subset(euploidGCSamplesDiffExpr, Sample=="2")
    # genesS3GCDE <- subset(euploidGCSamplesDiffExpr, Sample=="3")
    # genesS5GCDE <- subset(euploidGCSamplesDiffExpr, Sample=="5")
    # genesS9GCDE <- subset(euploidGCSamplesDiffExpr, Sample=="9")

#     library(VennDiagram)
# venn.plot <- venn.diagram(
# 	x = list(
# 		A = genesS1GCDE$GENEID,
# 		B = genesS2GCDE$GENEID,
# 		C = genesS3GCDE$GENEID,
# 		D = genesS5GCDE$GENEID,
# 		E = genesS9GCDE$GENEID
# 		),
# 	filename = NULL, #"euploid_commonDE.tiff"
# 	col = "black",
# 	fill = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
# 	alpha = 0.50,
# 	cex = c(1.5, 1.5, 1.5, 1.5, 1.5, 1, 0.8, 1, 0.8, 1, 0.8, 1, 0.8,
# 	 1, 0.8, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 1, 1, 1, 1, 1.5),
# 	cat.col = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
# 	cat.cex = 1.5,
# 	cat.fontface = "bold",
# 	margin = 0.05
# 	)
# pdf("euploidGC_commonDE.pdf")
# grid.draw(venn.plot)

# eupGCLinesDE <- list(genesS1GCDE, genesS2GCDE, genesS3GCDE, genesS5GCDE, genesS9GCDE)
# # want to get list of common DE genes in all the euploid lines
# commonDEGCeup <- Reduce(intersect, Map("[[", eupGCLinesDE, "GENEID"))
# #Reduc
# commonDEGCeup <- as.vector(commonDEGCeup)
# commonDEGCeup <- as.data.frame(commonDEGCeup)
# colnames(commonDEGCeup) <- c("names")
# write.csv(commonDEGCeup, file="commonDEeup.csv")


#######################################################
#### results with aneuploid chromosomes included ######
########## MA old ######################################
# resultsNames(ddsDefMAold)
# resultsS1MAold <- results(ddsDefMAold,contrast=c("condition","SC001","SCA"))
# class(resultsS1MAold)
# sigGenesMAold1 <- rownames(subset(resultsS1MAold,padj<0.1))
# summary(resultsS1MAold)
#
# resultsS2MAold <- results(ddsDefMAold,contrast=c("condition","SC002","SCA"))
# resultsS2MAold
# sigGenesMAold2 <- rownames(subset(resultsS2MAold,padj<0.1))
# summary(resultsS2MAold)

# resultsS3MAold <- results(ddsDefMAold,contrast=c("condition","SC003","SCA"))
# resultsS3MAold
# sigGenesMAold3 <- rownames(subset(resultsS3MAold,padj<0.1))
# summary(resultsS3MAold)
#
# resultsS4MAold <- results(ddsDefMAold,contrast=c("condition","SC004","SCA"))
# resultsS4MAold
# sigGenesMAold4 <- rownames(subset(resultsS4MAold,padj<0.1))
# summary(resultsS4MAold)

# resultsS5MAold <- results(ddsDefMAold,contrast=c("condition","SC005","SCA"))
# resultsS5MAold
# sigGenesMAold5 <- rownames(subset(resultsS5MAold,padj<0.1))
# summary(resultsS5MAold)
#
# resultsS6MAold <- results(ddsDefMAold,contrast=c("condition","SC006","SCA"))
# resultsS6MAold
# sigGenesMAold6 <- rownames(subset(resultsS6MAold,padj<0.1))
# summary(resultsS6MAold)

# resultsS7MAold <- results(ddsDefMAold,contrast=c("condition","SC007","SCA"))
# resultsS7MAold
# sigGenesMAold7 <- rownames(subset(resultsS7MAold,padj<0.1))
# summary(resultsS7MAold)
#
# resultsS8MAold <- results(ddsDefMAold,contrast=c("condition","SC008","SCA"))
# resultsS8MAold
# sigGenesMAold8 <- rownames(subset(resultsS8MAold,padj<0.1))
# summary(resultsS8MAold)

# resultsS9MAold <- results(ddsDefMAold,contrast=c("condition","SC009","SCA"))
# resultsS9MAold
# sigGenesMAold9 <- rownames(subset(resultsS9MAold,padj<0.1))
# summary(resultsS9MAold)

# resultsS11MAold <- results(ddsDefMAold,contrast=c("condition","SC011","SCA"))
# resultsS11MAold
# sigGenesMAold11 <- rownames(subset(resultsS11MAold,padj<0.1))
# summary(resultsS11MAold)
#
# resultsS15MAold <- results(ddsDefMAold,contrast=c("condition","SC015","SCA"))
# resultsS15MAold
# sigGenesMAold15 <- rownames(subset(resultsS15MAold,padj<0.1))
# summary(resultsS15MAold)

# resultsS88MAold <- results(ddsDefMAold,contrast=c("condition","SC088","SCA"))
# resultsS88MAold
# sigGenesMAold88 <- rownames(subset(resultsS88MAold,padj<0.1))
# summary(resultsS88MAold)
#
# resultsS108MAold <- results(ddsDefMAold,contrast=c("condition","SC108","SCA"))
# resultsS108MAold
# sigGenesMAold108 <- rownames(subset(resultsS108MAold,padj<0.1))
# summary(resultsS108MAold)

# resultsS119MAold <- results(ddsDefMAold,contrast=c("condition","SC119","SCA"))
# resultsS119MAold
# sigGenesMAold119 <- rownames(subset(resultsS119MAold,padj<0.1))
# summary(resultsS119MAold)
#
# resultsS28MAold <- results(ddsDefMAold,contrast=c("condition","SC028","SCA"))
# resultsS28MAold
# sigGenesMAold28 <- rownames(subset(resultsS28MAold,padj<0.1))
# summary(resultsS28MAold)

  # resultsS1MAoldDat <- as.data.frame(resultsS1MAold)
  #   resultsS2MAoldDat <- as.data.frame(resultsS2MAold)
  #     resultsS3MAoldDat <- as.data.frame(resultsS3MAold)
  #     resultsS4MAoldDat <- as.data.frame(resultsS4MAold)
  #       resultsS5MAoldDat <- as.data.frame(resultsS5MAold)
  #       resultsS6MAoldDat <- as.data.frame(resultsS6MAold)
  #       resultsS7MAoldDat <- as.data.frame(resultsS7MAold)
  #       resultsS8MAoldDat <- as.data.frame(resultsS8MAold)
  #         # resultsS9MAoldDat <- as.data.frame(resultsS9MAold)
  #          resultsS28MAoldDat <- as.data.frame(resultsS28MAold)
  #         
  # add gene names as first column ( which is currently rownames)
  # resS1MAoldDat <- cbind(GENEID = rownames(resultsS1MAoldDat), resultsS1MAoldDat)
  # write.csv(resS1MAoldDat, file="resS1MAoldDat.csv")
  # resS2MAoldDat <- cbind(GENEID = rownames(resultsS2MAoldDat), resultsS2MAoldDat)
  # write.csv(resS2MAoldDat, file="resS2MAoldDat.csv")
  # resS3MAoldDat <- cbind(GENEID = rownames(resultsS3MAoldDat), resultsS3MAoldDat)
  # write.csv(resS3MAoldDat, file="resS3MAoldDat.csv")
  # resS4MAoldDat <- cbind(GENEID = rownames(resultsS4MAoldDat), resultsS4MAoldDat)
  # write.csv(resS4MAoldDat, file="resS4MAoldDat.csv")
  # resS5MAoldDat <- cbind(GENEID = rownames(resultsS5MAoldDat), resultsS5MAoldDat)
  # write.csv(resS5MAoldDat, file="resS5MAoldDat.csv")
  # resS6MAoldDat <- cbind(GENEID = rownames(resultsS6MAoldDat), resultsS6MAoldDat)
  # write.csv(resS6MAoldDat, file="resS6MAoldDat.csv")
  # resS7MAoldDat <- cbind(GENEID = rownames(resultsS7MAoldDat), resultsS7MAoldDat)
  # write.csv(resS7MAoldDat, file="resS7MAoldDat.csv")
  # resS8MAoldDat <- cbind(GENEID = rownames(resultsS8MAoldDat), resultsS8MAoldDat)
  # write.csv(resS8MAoldDat, file="resS8MAoldDat.csv")
  # # resS9MAoldDat <- cbind(GENEID = rownames(resultsS9MAoldDat), resultsS9MAoldDat)
  #   resS28MAoldDat <- cbind(GENEID = rownames(resultsS28MAoldDat), resultsS28MAoldDat)
  #   write.csv(resS28MAoldDat, file="resS28MAoldDat.csv")
  #
  # need to make a column identifying the sample
    # resS1MAoldDat$Sample <- "1"
    # resS2MAoldDat$Sample <- "2"
    # resS3MAoldDat$Sample <- "3"
    # resS4MAoldDat$Sample <- "4"
    # resS5MAoldDat$Sample <- "5"
    # resS6MAoldDat$Sample <- "6"
    # resS7MAoldDat$Sample <- "7"
    # resS8MAoldDat$Sample <- "8"
    # # resS9MAoldDat$Sample <- "9"
    # resS28MAoldDat$Sample <- "28"
    #

# put all sample expression data in the same data frame

# euploidMAoldSamples <- rbind(resS1MAoldDat, resS2MAoldDat, resS3MAoldDat, resS4MAoldDat, resS5MAoldDat,resS6MAoldDat,resS7MAoldDat, resS8MAoldDat, resS28MAoldDat)
#
#
# dim(euploidMAoldSamples)

# View(euploidMAoldSamples)
#   # # # add chromosome numbers to this file
#       match <- intersect(euploidMAoldSamples$GENEID, genesMAold$GENEID)
#       length(match)
#   # #   # tell R which genes to keep
#       rm(keep)
#       keep <- euploidMAoldSamples$GENEID %in% match
#       length(keep)
#       View(keep)
#       View(euploidMAoldSamples$GENEID)
# # #   # subset list of genes in analysis to include only the ones matching those found in          sample file
#     genesMAoldeu <- genesMAold[keep,]
# # #   # add the correct chromosomes in
#     length(genesMAoldeu)
    # length(euploidMAoldSamples)
    # dim(genesMAoldeu)
    # dim(euploidMAoldSamples)
    #
    #
    # euploidMAoldSamples <- cbind(genesMAoldeu$GENEID, genesMAoldeu$TXCHROM, euploidMAoldSamples)
    # class(euploidMAoldSamples)
    # class(euploidGCSamples)
#     # chr1.all <- subset(euploidMAoldSamplesDE, euploidMAoldSamplesDE$`genesNoAneu$GENEID`=="chrI")
#     sample1MAold <- subset(euploidMAoldSamples, Sample==1)
#     View(sample1MAold)
#     # sample7 <- subset(allSamplesExpr, Sample==7)
# # # first need to filter allSamplesExpr based on p-adj level
#     euploidMAoldSamplesDiffExpr <- subset(euploidMAoldSamples, padj<0.1)
#     

    # genesS1MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="1")
    # genesS2MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="2")
    # genesS3MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="3")
    # genesS4MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="4")
    # genesS5MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="5")
    # genesS6MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="6")
    # genesS7MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="7")
    # genesS8MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="8")
    # # line 9 is potentially aneuploid
    # # genesS9MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="9")
    # # line 28 is euploid
    # genesS28MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="28")
    # View(genesS28MAoldDE)
#     library(VennDiagram)
# venn.plot <- venn.diagram(
# 	x = list(
# 		A = genesS1GCDE$GENEID,
# 		B = genesS2GCDE$GENEID,
# 		C = genesS3GCDE$GENEID,
# 		D = genesS5GCDE$GENEID,
# 		E = genesS9GCDE$GENEID
# 		),
# 	filename = NULL, #"euploid_commonDE.tiff"
# 	col = "black",
# 	fill = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
# 	alpha = 0.50,
# 	cex = c(1.5, 1.5, 1.5, 1.5, 1.5, 1, 0.8, 1, 0.8, 1, 0.8, 1, 0.8,
# 	 1, 0.8, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 1, 1, 1, 1, 1.5),
# 	cat.col = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
# 	cat.cex = 1.5,
# 	cat.fontface = "bold",
# 	margin = 0.05
# 	)
# pdf("euploidGC_commonDE.pdf")
# grid.draw(venn.plot)

# eupMAoldLinesDE <- list(genesS1MAoldDE, genesS2MAoldDE, genesS3MAoldDE, genesS4MAoldDE,genesS5MAoldDE,genesS6MAoldDE,genesS7MAoldDE,genesS8MAoldDE, genesS28MAoldDE)
# # want to get list of common DE genes in all the euploid lines
# # class(eupMAoldLines)
# class(eupMAoldLinesDE)
# commonDEMAoldeup <- Reduce(intersect, Map("[[", eupMAoldLinesDE, "GENEID"))
# #Reduc
# commonDEMAoldeup <- as.data.frame(commonDEMAoldeup)
# colnames(commonDEMAoldeup) <- c("names")
# write.csv(commonDEMAoldeup, file="commonDEeupMAold.csv")

###### individual samples
resultsNames(ddsDefMAold_1)
resultsS1MAold <- results(ddsDefMAold_1,contrast=c("condition","SC001","SCA"))
class(resultsS1MAold)
summary(resultsS1MAold)

resultsNames(ddsDefMAold_2)
resultsS2MAold <- results(ddsDefMAold_2,contrast=c("condition","SC002","SCA"))
class(resultsS2MAold)
summary(resultsS2MAold)

resultsNames(ddsDefMAold_3)
resultsS3MAold <- results(ddsDefMAold_3,contrast=c("condition","SC003","SCA"))
class(resultsS3MAold)
summary(resultsS3MAold)

resultsNames(ddsDefMAold_4)
resultsS4MAold <- results(ddsDefMAold_4,contrast=c("condition","SC004","SCA"))
class(resultsS4MAold)
summary(resultsS4MAold)

resultsNames(ddsDefMAold_5)
resultsS5MAold <- results(ddsDefMAold_5,contrast=c("condition","SC005","SCA"))
class(resultsS5MAold)
summary(resultsS5MAold)

# resultsNames(ddsDefMAold_6)
# resultsS6MAold <- results(ddsDefMAold_6,contrast=c("condition","SC006","SCA"))
# class(resultsS6MAold)
# summary(resultsS6MAold)

# normCountsMAold_6 <- counts(ddsDefMAold_6,normalized=TRUE)
# write.csv(normCountsMAold_6,file="normCountsMAold_6.csv")

resultsNames(ddsDefMAold_7)
resultsS7MAold <- results(ddsDefMAold_7,contrast=c("condition","SC007","SCA"))
class(resultsS7MAold)
summary(resultsS7MAold)

resultsNames(ddsDefMAold_8)
resultsS8MAold <- results(ddsDefMAold_8,contrast=c("condition","SC008","SCA"))
class(resultsS8MAold)
summary(resultsS8MAold)

resultsNames(ddsDefMAold_9)
resultsS9MAold <- results(ddsDefMAold_9,contrast=c("condition","SC009","SCA"))
class(resultsS9MAold)
summary(resultsS9MAold)

resultsNames(ddsDefMAold_11)
resultsS11MAold <- results(ddsDefMAold_11,contrast=c("condition","SC011","SCA"))
class(resultsS11MAold)
summary(resultsS11MAold)

resultsNames(ddsDefMAold_15)
resultsS15MAold <- results(ddsDefMAold_15,contrast=c("condition","SC015","SCA"))
class(resultsS15MAold)
summary(resultsS15MAold)

resultsNames(ddsDefMAold_28)
resultsS28MAold <- results(ddsDefMAold_28,contrast=c("condition","SC028","SCA"))
class(resultsS28MAold)
summary(resultsS28MAold)

resultsNames(ddsDefMAold_88)
resultsS88MAold <- results(ddsDefMAold_88,contrast=c("condition","SC088","SCA"))
class(resultsS88MAold)
summary(resultsS88MAold)

resultsNames(ddsDefMAold_108)
resultsS108MAold <- results(ddsDefMAold_108,contrast=c("condition","SC108","SCA"))
class(resultsS108MAold)
summary(resultsS108MAold)

resultsNames(ddsDefMAold_119)
resultsS119MAold <- results(ddsDefMAold_119,contrast=c("condition","SC119","SCA"))
class(resultsS119MAold)
summary(resultsS119MAold)


################################################################################

#aneuploid chromosomes removed from normalization
# allGenesGC <- as.data.frame(mcols(ddsHTSeqGC))
# View(allGenesGC)
# ddsGCNoAneuInNorm <- estimateSizeFactors(ddsHTSeqGC, controlGenes=c(85:481,482:629,630:1373,1647:1761,3090:3403,3881:4340))
# genesGCnoAn <- rownames(ddsGCNoAneuInNorm)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesNoAneuGC <- ensembldb::select(txdb, keys=genesGCnoAn, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesNoAneuGC <- na.omit(genesNoAneuGC)
# genesNoAneuGC <- as.data.frame(genesNoAneuGC)
# # remove1 <- "chrM"
# # genesNoAneu <- genesNoAneu[-(grep(paste0(remove1,"$"),genesNoAneu$TXCHROM,perl=TRUE)),]
# ddsGCNoAneuInNorm <- estimateDispersions(ddsGCNoAneuInNorm)
# attr(dispersionFunction(ddsGCNoAneuInNorm),"dispPriorVar")
# ddsGCNoAneuInNorm <- nbinomWaldTest(ddsGCNoAneuInNorm,useT=TRUE,df=(nrow(ddsGCNoAneuInNorm)))
# norm.factors.no.aneuGC <- as.data.frame(sizeFactors(ddsGCNoAneuInNorm))
# design(ddsGCNoAneuInNorm)
# dispersions.no.aneuGC <- dispersions(ddsGCNoAneuInNorm)
# #add annotation to DESeq2 data set
# all(rownames(ddsGCNoAneuInNorm) == genesNoAneuGC$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedNoAneuGC <- intersect(rownames(ddsGCNoAneuInNorm), genesNoAneuGC$GENEID)
# keep3 <- rownames(ddsGCNoAneuInNorm) %in% matchedNoAneuGC
# ddsGCNoAneuInNorm <- ddsGCNoAneuInNorm[keep3,]
# length(ddsGCNoAneuInNorm)
# all(rownames(ddsGCNoAneuInNorm) == genesNoAneuGC$GENEID)
# mcols(ddsGCNoAneuInNorm)
# #then check again
# all(rownames(ddsGCNoAneuInNorm) == genesNoAneuGC$GENEID)
# #nrow(ddsNoAneuInNorm)
# #nrow(genesNoAneu)
# #mcols(ddsNoAneuInNorm) <- cbind(mcols(ddsNoAneuInNorm), genesNoAneu)
# #mcols(ddsNoAneuInNorm)
# #write.csv(counts(dds), file="counts.Rsubread.csv")
# #mcols(dds)
# # write.csv(genes,file="gene_anno.csv")
# # write.csv(counts(ddsHTSeq),file="counts.filtered.csv")

############################################################################
#aneuploid chromosomes removed
# allGenesMAold <- as.data.frame(mcols(ddsHTSeqMAold))
# View(allGenesMAold)
# ddsMAoldNoAneuInNorm <- estimateSizeFactors(ddsHTSeqMAold, controlGenes=c(1:2268,2740:5667))
# genesMAoldnoAn <- rownames(ddsMAoldNoAneuInNorm)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesNoAneuMAold <- ensembldb::select(txdb, keys=genesMAoldnoAn, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesNoAneuMAold <- na.omit(genesNoAneuMAold)
# genesNoAneuMAold <- as.data.frame(genesNoAneuMAold)
# # remove1 <- "chrM"
# # genesNoAneu <- genesNoAneu[-(grep(paste0(remove1,"$"),genesNoAneu$TXCHROM,perl=TRUE)),]
# ddsMAoldNoAneuInNorm <- estimateDispersions(ddsMAoldNoAneuInNorm)
# attr(dispersionFunction(ddsMAoldNoAneuInNorm),"dispPriorVar")
# ddsMAoldNoAneuInNorm <- nbinomWaldTest(ddsMAoldNoAneuInNorm,useT=TRUE,df=(nrow(ddsMAoldNoAneuInNorm)))
# norm.factors.no.aneuMAold <- as.data.frame(sizeFactors(ddsMAoldNoAneuInNorm))
# design(ddsMAoldNoAneuInNorm)
# dispersions.no.aneuMAold <- dispersions(ddsMAoldNoAneuInNorm)
# #add annotation to DESeq2 data set
# all(rownames(ddsMAoldNoAneuInNorm) == genesNoAneuMAold$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedNoAneuMAold <- intersect(rownames(ddsMAoldNoAneuInNorm), genesNoAneuMAold$GENEID)
# keep3 <- rownames(ddsMAoldNoAneuInNorm) %in% matchedNoAneuMAold
# ddsMAoldNoAneuInNorm <- ddsMAoldNoAneuInNorm[keep3,]
# length(ddsMAoldNoAneuInNorm)
# all(rownames(ddsMAoldNoAneuInNorm) == genesNoAneuMAold$GENEID)
# mcols(ddsMAoldNoAneuInNorm)
# #then check again
# all(rownames(ddsMAoldNoAneuInNorm) == genesNoAneuMAold$GENEID)
#
# ############################################################################
# #aneuploid chromosomes removed
# allGenesMAnew <- as.data.frame(mcols(ddsHTSeqMAnew))
# View(allGenesMAnew)
# ddsMAnewNoAneuInNorm <- estimateSizeFactors(ddsHTSeqMAnew, controlGenes=c(89:485,638:1385,1657:1769,2549:3412,3886:4337,4723:5237))
# genesMAnewnoAn <- rownames(ddsMAnewNoAneuInNorm)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesNoAneuMAnew <- ensembldb::select(txdb, keys=genesMAnewnoAn, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesNoAneuMAnew <- na.omit(genesNoAneuMAnew)
# genesNoAneuMAnew <- as.data.frame(genesNoAneuMAnew)
# # remove1 <- "chrM"
# # genesNoAneu <- genesNoAneu[-(grep(paste0(remove1,"$"),genesNoAneu$TXCHROM,perl=TRUE)),]
# ddsMAnewNoAneuInNorm <- estimateDispersions(ddsMAnewNoAneuInNorm)
# attr(dispersionFunction(ddsMAnewNoAneuInNorm),"dispPriorVar")
# ddsMAnewNoAneuInNorm <- nbinomWaldTest(ddsMAnewNoAneuInNorm,useT=TRUE,df=(nrow(ddsMAnewNoAneuInNorm)))
# norm.factors.no.aneuMAnew <- as.data.frame(sizeFactors(ddsMAnewNoAneuInNorm))
# design(ddsMAnewNoAneuInNorm)
# dispersions.no.aneuMAnew <- dispersions(ddsMAnewNoAneuInNorm)
# #add annotation to DESeq2 data set
# all(rownames(ddsMAnewNoAneuInNorm) == genesNoAneuMAnew$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedNoAneuMAnew <- intersect(rownames(ddsMAnewNoAneuInNorm), genesNoAneuMAnew$GENEID)
# keep3 <- rownames(ddsMAnewNoAneuInNorm) %in% matchedNoAneuMAnew
# ddsMAnewNoAneuInNorm <- ddsMAnewNoAneuInNorm[keep3,]
# length(ddsMAnewNoAneuInNorm)
# all(rownames(ddsMAnewNoAneuInNorm) == genesNoAneuMAnew$GENEID)
# mcols(ddsMAnewNoAneuInNorm)
# #then check again
# all(rownames(ddsMAnewNoAneuInNorm) == genesNoAneuMAnew$GENEID)

```

Now want to make distributions of ratios for cis and trans genes in aneuploid samples

```{r ratioDistributions,eval=TRUE,include=TRUE}

############ Euploid distributions
ratioDistributionsEuploids <- function(dds, genes) {
  
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)
library(ggplot2)

allGenes <- ggplot(ratios,aes(x=sampleOverAnc))
g1 <- allGenes + geom_density(fill="turquoise",alpha=0.2) + geom_vline(xintercept = 1)+ coord_cartesian(xlim=c(0,4))  + geom_vline(aes(xintercept = mean(ratios$sampleOverAnc, na.rm = T)), colour = "red", linetype ="longdash", size = .5 )  +  geom_vline(aes(xintercept = median(ratios$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5 ) + theme_classic()  + theme(axis.title = element_text(size=7),axis.text=element_text(size=7)) 
return(g1)
}

ratioDistGC1 <- ratioDistributionsEuploids(ddsDefGC_1,genesGC_1) + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistGC2 <- ratioDistributionsEuploids(ddsDefGC_2,genesGC_2) + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistGC3 <- ratioDistributionsEuploids(ddsDefGC_3,genesGC_3) + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistGC5 <- ratioDistributionsEuploids(ddsDefGC_5,genesGC_5) + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistGC9 <- ratioDistributionsEuploids(ddsDefGC_9,genesGC_9) + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )

library(cowplot)
library(ggpubr)

pdf("hetEups.pdf")
hetEups <- plot_grid(ratioDistGC1, ratioDistGC2, ratioDistGC3,ratioDistGC5, ratioDistGC9, ncol = 2, labels="AUTO",label_size = 8, hjust=1)
annotate_figure(hetEups,
               bottom = text_grob("Ratio", size = 10),
               left = text_grob("Density of Genes", rot = 90,size=10))
dev.off()

library(car)
# ks.test(ratioDistGC2,ratioDistGC3,alternative="two.sided")

ratioDistMA1 <- ratioDistributionsEuploids(ddsDefMAold_1,genesMAold_1) + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA2 <- ratioDistributionsEuploids(ddsDefMAold_2,genesMAold_2)+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA3 <- ratioDistributionsEuploids(ddsDefMAold_3,genesMAold_3)+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA4 <- ratioDistributionsEuploids(ddsDefMAold_4,genesMAold_4)+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA5 <- ratioDistributionsEuploids(ddsDefMAold_5,genesMAold_5)+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA7 <- ratioDistributionsEuploids(ddsDefMAold_7,genesMAold_7)+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA8 <- ratioDistributionsEuploids(ddsDefMAold_8,genesMAold_8)+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )

ratioDistributionsEuploidsMANew <- function(dds, genes) {
  
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:5)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)

allGenes <- ggplot(ratios,aes(x=sampleOverAnc))
g1 <- allGenes + geom_density(fill="turquoise",alpha=0.2) + geom_vline(xintercept = 1)+ coord_cartesian(xlim=c(0,4))  + geom_vline(aes(xintercept = mean(ratios$sampleOverAnc, na.rm = T)), colour = "red", linetype ="longdash", size = .5 )  +  geom_vline(aes(xintercept = median(ratios$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5 ) + theme_classic()  + theme(axis.title = element_text(size=7),axis.text=element_text(size=7)) 
print(g1)
}

ratioDistMA50 <- ratioDistributionsEuploidsMANew(ddsDefMAnewS50,genesMAnewS50)+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA115 <- ratioDistributionsEuploidsMANew(ddsDefMAnewS115,genesMAnewS115)+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )

homEups <- plot_grid(ratioDistMA1, ratioDistMA2, ratioDistMA3, ratioDistMA4, ratioDistMA5, ratioDistMA7, ratioDistMA8, ratioDistMA50, ratioDistMA115 , ncol = 2, labels= "AUTO", label_size = 7.5)

pdf("homEups.pdf")
homEups
annotate_figure(homEups,
              bottom = text_grob("Ratio", size = 10),
            left = text_grob("Density of Genes", rot = 90,size=10))
dev.off()
#####################################################################
#####################################################################


dds=ddsDefGC_4
genes=genesGC_4
anChr="chrV"

library(scales)
show_col(hue_pal()(4))

ratioDistributionsTrisomy <- function(dds, genes, anChr,sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)
ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]
ratios.cis$type <- "cis"
ratios.cis$chr <- anChr
# View(ratios.cis)
ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]
ratios.Trans$type <- "trans"
ratios.Trans$chr <- "Non-Aneu"
# View(ratios.Trans)
ratios.All <- rbind(ratios.cis, ratios.Trans)
# View(ratios.All)
allRatiosPlot <- ggplot(ratios.All, aes(x=sampleOverAnc,fill=type,linetype=chr)) +
  geom_density(alpha=0.2) + 
  coord_cartesian(xlim=c(0,4)) + 
  geom_vline(xintercept = 1) + #exp for disomy 
  geom_vline(xintercept=1.5, col="magenta") + #exp for trisomy 
  # geom_vline(xintercept=0.5,col="green") + #exp for monosomy
  # geom_vline(xintercept=2,col="orange") + #exp for tetrasomy
  geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),colour = "red", linetype ="longdash", size = .5) + #mean cis genes
  geom_vline(aes(xintercept = mean(ratios.Trans$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5) + #mean trans genes
  theme_classic() + labs(x="Ratio (Line/Ancestor)",y="Number of Genes",title=sample)
}

#black line is expected for euploid samples 
# yellow line is expected ratio for line with trisomy chr I 
ratioDistGC7 <- ratioDistributionsTrisomy(ddsDefGC_7,genesGC_7,"chrI","A") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )

ratioDistGC4 <- ratioDistributionsTrisomy(ddsDefGC_4,genesGC_4,"chrV","B")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistGC49 <- ratioDistributionsTrisomy(ddsDefGC_49,genesGC_49,"chrV","C")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )

ratioDistGC59 <- ratioDistributionsTrisomy(ddsDefGC_59,genesGC_59,"chrVII","D")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistGC61 <- ratioDistributionsTrisomy(ddsDefGC_61,genesGC_61,"chrVII","E")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )

ratioDistGC77 <- ratioDistributionsTrisomy(ddsDefGC_77,genesGC_77,"chrXII","F")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )

## paste together with ggpubr with common legend
library(ggpubr)
## for supplement
pdf("gcTris_samples_ratios.pdf")
gcTri <- ggarrange(ratioDistGC7,ratioDistGC4,ratioDistGC49,ratioDistGC59,ratioDistGC61,ratioDistGC77,ncol=2,nrow=3,label.x="Ratio (Line/Ancestor)",label.y = "Number of Genes", common.legend = TRUE)
annotate_figure(gcTri,
             bottom = text_grob("log2Ratio (Sample/Anc)", size = 10),
           left = text_grob("Density of Genes", rot = 90,size=10))
dev.off()
## for publication: labels are as follows: A-line 7 chromosome I, B-line 4 chromosome V, C- line 49 chromosome V, D-line 59 chr VII, E-line 61 chr VII, F-line 77 chr XII


### same function but for lines that have multiple aneuploidies
#### for those lines with multiple aneuploidies
ratioDistributionsTrisomyMulti <- function(dds, genes, anChr, anChr2, sample) {
counts <- counts(dds,normalized=TRUE)
ratios <- c()
as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)
ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]
ratios.cis$type="cis"
ratios.cis$chr=anChr
ratios.keepCis2 <- ratios$chr==anChr2
ratios.Cis2 <- ratios[ratios.keepCis2,]
ratios.Cis2$type <- "cis"
ratios.Cis2$chr=anChr2
ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]
ratios.keepTrans2 <- ratios$chr!=anChr2
ratios.Trans2 <- ratios.Trans[ratios.keepTrans2,]
ratios.Trans2$type="trans"
ratios.Trans2$chr="trans"
ratios.All <- rbind(ratios.cis, ratios.Cis2, ratios.Trans2)
allRatiosPlot <- ggplot(ratios.All, aes(x=sampleOverAnc,linetype=chr,fill=type)) +
  geom_density(alpha=0.2) +
  coord_cartesian(xlim=c(0,4)) + geom_vline(xintercept = 1) + 
  geom_vline(xintercept=1.5, col="magenta") + #exp for trisomy 
  # geom_vline(xintercept=0.5,col="green") + #exp for monosomy
  # geom_vline(xintercept=2,col="orange") + #exp for tetrasomy
  geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),colour = "red", linetype ="longdash", size = .5) +
  #mean cis genes
  geom_vline(aes(xintercept = median(ratios.Cis2$sampleOverAnc, na.rm = T)), colour = "purple", linetype ="longdash", size =
               .5) + #mean trans genes
  geom_vline(aes(xintercept = median(ratios.Trans2$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size =
               .5) + 
  #mean trans genes
  theme_classic() + labs(x="Ratio (Line/Ancestor)",y="Number of Genes", title = sample)
}

ratioDistGC18 <- ratioDistributionsTrisomyMulti(ddsDefGC_18,genesGC_18,"chrI","chrXII", "") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )


ratioDistributionsTriMonMulti <- function(dds, genes, anChr, anChr2, sample) {
counts <- counts(dds,normalized=TRUE)
ratios <- c()
as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)
ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]
ratios.cis$type="cis"
ratios.cis$chr=anChr
ratios.keepCis2 <- ratios$chr==anChr2
ratios.Cis2 <- ratios[ratios.keepCis2,]
ratios.Cis2$type="cis"
ratios.Cis2$chr=anChr2
ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]
ratios.keepTrans2 <- ratios$chr!=anChr2
ratios.Trans2 <- ratios.Trans[ratios.keepTrans2,]
ratios.Trans2$type="trans"
ratios.Trans2$chr="trans"
ratios.All <- rbind(ratios.cis, ratios.Cis2, ratios.Trans2)
allRatiosPlot <- ggplot(ratios.All, aes(x=sampleOverAnc,fill=type,linetype=chr)) +
  geom_density(alpha=0.2) +
  coord_cartesian(xlim=c(0,4)) + geom_vline(xintercept = 1) + 
  geom_vline(xintercept=1.5, col="magenta") + #exp for trisomy 
  geom_vline(xintercept=0.5,col="green") +   #exp for monosomy
  # geom_vline(xintercept=2,col="orange") + #exp for tetrasomy
  geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),colour = "red", linetype ="longdash", size = .5) +
  #mean cis genes
   geom_vline(aes(xintercept = median(ratios.Cis2$sampleOverAnc, na.rm = T)), colour = "purple", linetype ="longdash", size = .5) + #mean trans genes
  geom_vline(aes(xintercept = median(ratios.Trans2$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5) + 
  #mean trans genes
  theme_classic() + labs(x="Ratio (Line/Ancestor)",y="Number of Genes", title = sample)
}
ratioDistGC11 <- ratioDistributionsTriMonMulti(ddsDefGC_11,genesGC_11,"chrI","chrXV", "") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )

ratioDistMA108 <- ratioDistributionsTriMonMulti(ddsDefMAold_108,genesMAold_108,"chrVIII","chrIX", "") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )

## for publication
pdf("gcMulti_samples_ratios.pdf")
gcMulti <- ggarrange(ratioDistGC7,ratioDistGC4,ratioDistGC18,ratioDistGC11,ratioDistGC49,ratioDistGC59,ratioDistGC61,ratioDistGC76,ratioDistGC77,ratioDistGC8,ncol=2,nrow=5,label.x="log2 Ratio (Line/Ancestor)",label.y = "Density of Genes", common.legend = TRUE)
annotate_figure(gcMulti,
         bottom = text_grob("log2 Ratio(Sample/Anc)", size = 10),
       left = text_grob("Density of Genes", rot = 90,size=10))
dev.off()

pdf("maMulti_samples_ratios.pdf")
ratioDistMA108
dev.off()
## for publication: labels are as follows: A: line 18 B: line 11 het anc 

pdf("gcMulti_ratios.pdf")
gcMulti <- ggarrange(ratioDistGC11,ratioDistGC18,ncol=1,nrow=2,label.x="log2 Ratio (Line/Ancestor)",label.y = "Density of Genes", common.legend = FALSE)
annotate_figure(gcMulti,
         bottom = text_grob("log2 Ratio(Sample/Anc)", size = 10),
       left = text_grob("Density of Genes", rot = 90,size=10))
dev.off()


```

```{r line76_ratios,include=TRUE,eval=TRUE,echo=FALSE}
### line 76 has 2 fully duplicated chromosomes and one partial duplication
counts <- counts(ddsDefGC_76,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genesGC_76$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)
ratios.keepCis <- ratios$chr=="chrIX"
ratios.cis <- ratios[ratios.keepCis,]
ratios.cis$chr="chrIX"
ratios.cis$type="cis"
ratios.keepCis2 <- ratios$chr=="chrXIV"
ratios.cis2 <- ratios[ratios.keepCis2,]
ratios.cis2$type="cis"
ratios.cis2$chr <- "chrXIV"
ratios.keepCis3 <- ratios$chr=="chrX"
ratios.cis3 <- ratios[ratios.keepCis3,]
ratios.cis3$type="cis"
ratios.cis3$chr <- "chrX"
ratios.keepTrans <- ratios$chr!="chrIX"
ratios.Trans <- ratios[ratios.keepTrans,]
ratios.Trans$type="trans"
ratios.Trans$chr <- "trans"
ratios.keepTrans2 <- ratios$chr!="chrXIV"
ratios.Trans2 <- ratios.Trans[ratios.keepTrans2,]
ratios.Trans2$type <- "trans"
ratios.Trans2$chr <- "trans"
ratios.keepTrans3 <- ratios$chr!="chrX"
ratios.Trans3 <- ratios.Trans2[ratios.keepTrans3,]
ratios.Trans3$type <- "trans"
ratios.Trans3$chr <- "trans"
ratios76.All <- rbind(ratios.cis, ratios.cis2, ratios.cis3, ratios.Trans3)

ratioDistGC76 <- ggplot(ratios76.All, aes(x=sampleOverAnc,fill=type,linetype=chr)) +
  geom_density(alpha=0.2) + coord_cartesian(xlim=c(0,4)) + geom_vline(xintercept = 1) + 
  geom_vline(xintercept=1.5, col="magenta") + #exp for trisomy 
  geom_vline(xintercept=0.5,col="green") +   #exp for monosomy
  # geom_vline(xintercept=2,col="orange") + #exp for tetrasomy
  geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),colour = "red", linetype ="longdash", size = .5) +
  #mean cis genes
   geom_vline(aes(xintercept = median(ratios.cis2$sampleOverAnc, na.rm = T)), colour = "purple", linetype ="longdash", size = .5) + #mean cis2 genes
  geom_vline(aes(xintercept = median(ratios.cis3$sampleOverAnc, na.rm = T)), colour = "deeppink1", linetype ="longdash", size = .5) + #mean cis3 genes
  geom_vline(aes(xintercept = median(ratios.Trans3$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5) + 
  #mean trans genes
  theme_classic() + labs(x="Ratio (Line/Ancestor)",y="Density of Genes", title="")

ratioDistGC76 <- ratioDistGC76 + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )

pdf("gcLine76.pdf")
ratioDistGC76
dev.off()
```


```{r ratioDistCont,include=TRUE,eval=TRUE,echo=FALSE}

### MA old lines trisomic

ratioDistMA15 <- ratioDistributionsTrisomy(ddsDefMAold_15,genesMAold_15,"chrIX","A") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA88 <- ratioDistributionsTrisomy(ddsDefMAold_88,genesMAold_88,"chrIX","B")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA119 <- ratioDistributionsTrisomy(ddsDefMAold_119,genesMAold_119,"chrIX","C")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA9 <- ratioDistributionsTrisomy(ddsDefMAold_9,genesMAold_9,"chrXIV","D")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )

# for MA new lines bc only two anc replicates
ratioDistributionsTrisomyMAnew <- function(dds, genes, anChr,sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:2)
ratios$ancMean <- rowMeans2(counts,cols=3:5)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)
ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]
ratios.cis$type <- "cis"
ratios.cis$chr <- anChr
# View(ratios.cis)
ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]
ratios.Trans$type <- "trans"
ratios.Trans$chr <- "Non-Aneu"
# View(ratios.Trans)
ratios.All <- rbind(ratios.cis, ratios.Trans)
# View(ratios.All)
allRatiosPlot <- ggplot(ratios.All, aes(x=sampleOverAnc,fill=type,linetype=chr)) +
  geom_density(alpha=0.2) + coord_cartesian(xlim=c(0,4))+ geom_vline(xintercept = 1) + 
  geom_vline(xintercept=1.5, col="magenta") + #exp for trisomy 
  # geom_vline(xintercept=0.5,col="green") + #exp for monosomy
  # geom_vline(xintercept=2,col="orange") + #exp for tetrasomy
  geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),colour = "red", linetype ="longdash", size = .5) + #mean cis genes
  geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5) + #mean trans genes
  theme_classic() + labs(x="Ratio (Line/Ancestor)",y="Number of Genes", title = sample)

}

ratioDistMA112 <- ratioDistributionsTrisomyMAnew(ddsDefMAnewS112,genesMAnewS112,"chrXVI","E")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA117 <- ratioDistributionsTrisomyMAnew(ddsDefMAnewS117,genesMAnewS117,"chrV","F")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA123 <- ratioDistributionsTrisomyMAnew(ddsDefMAnewS123,genesMAnewS123,"chrV","G")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )

pdf("MA_ratios_trisomies.pdf")
MAtri <- ggarrange(ratioDistMA15,ratioDistMA88,ratioDistMA119,ratioDistMA9,ratioDistMA112,ratioDistMA117,ratioDistMA123,ncol=2,nrow=4, common.legend = TRUE)
annotate_figure(MAtri,
    bottom = text_grob("log2 Ratio(sample/Anc)", size = 10),
     left = text_grob("Density of Genes", rot = 90,size=10))
dev.off()




## MA new multi chromosome aneuploidies

ratioDistributionsTriMonMultiMAnew <- function(dds, genes, anChr, anChr2, sample) {
counts <- counts(dds,normalized=TRUE)
ratios <- c()
as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:2)
ratios$ancMean <- rowMeans2(counts,cols=3:5)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)
ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]
ratios.cis$type="cis"
ratios.cis$chr <- anChr
ratios.keepCis2 <- ratios$chr==anChr2
ratios.Cis2 <- ratios[ratios.keepCis2,]
ratios.Cis2$type="cis"
ratios.Cis2$chr <- anChr2
ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]
ratios.keepTrans2 <- ratios$chr!=anChr2
ratios.Trans2 <- ratios.Trans[ratios.keepTrans2,]
ratios.Trans2$type="trans"
ratios.Trans2$chr <- "Non-Aneu"
ratios.All <- rbind(ratios.cis, ratios.Cis2, ratios.Trans2)
allRatiosPlot <- ggplot(ratios.All, aes(x=sampleOverAnc,fill=type,linetype=chr)) +
  geom_density(alpha=0.2) + coord_cartesian(xlim=c(0,4)) + geom_vline(xintercept = 1) + 
  geom_vline(xintercept=1.5, col="magenta") + #exp for trisomy 
  geom_vline(xintercept=0.5,col="green") +   #exp for monosomy
  # geom_vline(xintercept=2,col="orange") + #exp for tetrasomy
  geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),colour = "red", linetype ="longdash", size = .5) +
  #mean cis genes
   geom_vline(aes(xintercept = median(ratios.Cis2$sampleOverAnc, na.rm = T)), colour = "purple", linetype ="longdash", size = .5) + #mean trans genes
  geom_vline(aes(xintercept = median(ratios.Trans2$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5) + 
  #mean trans genes
  theme_classic() + labs(x="Ratio (Line/Ancestor)",y="Number of Genes", title = sample)
}

ratioDistMA152.VIII <- ratioDistributionsTriMonMultiMAnew(ddsDefMAnewS152,genesMAnewS152,"chrVIII","chrI", "") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )


pdf("MA_lines_multi_aneu.pdf")
maMulti <- ggarrange(ratioDistMA108,ratioDistMA152.VIII,nrow=2,labels="AUTO",common.legend = FALSE)
annotate_figure(maMulti,
    bottom = text_grob("log2 Ratio(sample/Anc)", size = 10),
     left = text_grob("Density of Genes", rot = 90,size=10))
dev.off()

# for tetrasomy 
ratioDistributionsTetrsomy <- function(dds, genes, anChr,sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)
ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]
ratios.cis$type <- "cis"
ratios.cis$chr <- anChr
# View(ratios.cis)
ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]
ratios.Trans$type <- "trans"
ratios.Trans$chr <- "Non-Aneu"
# View(ratios.Trans)
ratios.All <- rbind(ratios.cis, ratios.Trans)
# View(ratios.All)
allRatiosPlot <- ggplot(ratios.All, aes(x=sampleOverAnc,fill=type,linetype=chr)) +
  geom_density(alpha=0.2) + coord_cartesian(xlim=c(0,4)) + geom_vline(xintercept = 1) + 
  # geom_vline(xintercept=1.5, col="magenta") + #exp for trisomy 
  # geom_vline(xintercept=0.5,col="green") + #exp for monosomy
  geom_vline(xintercept=2,col="orange") + #exp for tetrasomy
  geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),colour = "red", linetype ="longdash", size = .5) + #mean cis genes
  geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5) + #mean trans genes
  theme_classic() + labs(x="Ratio (Line/Ancestor)",y="Number of Genes", title = sample)

}
ratioDistGC8 <- ratioDistributionsTetrsomy(ddsDefGC_8,genesGC_8,"chrXVI","J")  + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )

pdf("gc_line8_ratios.pdf")
ratioDistGC8
dev.off()

## need monosomy one for MA new 29
ratioDistributionsMonosomyMAnew <- function(dds, genes, anChr,sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:2)
ratios$ancMean <- rowMeans2(counts,cols=3:5)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)
ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]
ratios.cis$type <- "cis"
ratios.cis$chr <- anChr
# View(ratios.cis)
ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]
ratios.Trans$type <- "trans"
ratios.Trans$chr <- "Non-Aneu"
# View(ratios.Trans)
ratios.All <- rbind(ratios.cis, ratios.Trans)
# View(ratios.All)
allRatiosPlot <- ggplot(ratios.All, aes(x=sampleOverAnc,fill=type,linetype=chr)) +
  geom_density(alpha=0.2) + coord_cartesian(xlim=c(0,4)) + geom_vline(xintercept = 1) + 
  # geom_vline(xintercept=1.5, col="magenta") + #exp for trisomy 
  geom_vline(xintercept=0.5,col="green") +  #exp for monosomy
  # geom_vline(xintercept=2,col="orange") +  #exp for tetrasomy
  geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),colour = "red", linetype ="longdash", size = .5) + #mean cis genes
  geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5) + #mean trans genes
  theme_classic() + labs(x="Ratio (Line/Ancestor)",y="Density of Genes", title = sample)

}
ratioDistMA29 <- ratioDistributionsMonosomyMAnew(ddsDefMAnewS29,genesMAnewS29,"chrIX","") # + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )

pdf("ratios_MA_29.pdf")
ratioDistMA29
dev.off()


## for supplement 
pdf("maTris_samples_ratios.pdf")
maMulti <- ggarrange(ratioDistMA15,ratioDistMA88,ratioDistMA119,ratioDistMA9,ncol=2,nrow=2,label.x="Ratio (Line/Ancestor)",label.y = "Density of Genes", common.legend = TRUE)
annotate_figure(maMulti,
    bottom = text_grob("log2 Ratio(sample/Anc)", size = 10),
     left = text_grob("Density of Genes", rot = 90,size=10))
dev.off()


ratioDistGC4 <- ratioDistributionsTrisomy(ddsDefGC_4,genesGC_4,"chrV","A")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA117 <- ratioDistributionsTrisomyMAnew(ddsDefMAnewS117,genesMAnewS117,"chrV","B")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistGC59 <- ratioDistributionsTrisomy(ddsDefGC_59,genesGC_59,"chrVII","C")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA15 <- ratioDistributionsTrisomy(ddsDefMAold_15,genesMAold_15,"chrIX","D") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA29 <- ratioDistributionsMonosomyMAnew(ddsDefMAnewS29,genesMAnewS29,"chrIX","E")  + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA9 <- ratioDistributionsTrisomy(ddsDefMAold_9,genesMAold_9,"chrXIV","F")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistGC8 <- ratioDistributionsTetrsomy(ddsDefGC_8,genesGC_8,"chrXVI","G")  + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA112 <- ratioDistributionsTrisomyMAnew(ddsDefMAnewS112,genesMAnewS112,"chrXVI","H")+ theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistGC11 <- ratioDistributionsTriMonMulti(ddsDefGC_11,genesGC_11,"chrI","chrXV", "I") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistGC18 <- ratioDistributionsTrisomyMulti(ddsDefGC_18,genesGC_18,"chrI","chrXII", "J") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )



## for main body of paper 
pdf("main_body_fig_trisomics.pdf")
trisomics <- ggarrange(ratioDistGC4, ratioDistMA117, ratioDistGC59, ratioDistMA15, ratioDistMA29, ratioDistMA9, ratioDistGC8, ratioDistMA112, ratioDistGC11, ratioDistGC18, ratioDistGC76,  ncol=3, nrow=4, label.x="Ratio (Line/Ancestor)",label.y = "Density of Genes", common.legend = TRUE)
annotate_figure(trisomics,
    bottom = text_grob("log2 Ratio(sample/Anc)", size = 10),
     left = text_grob("Density of Genes", rot = 90,size=10))
dev.off()


```






```{r hist, eval=TRUE,include=TRUE,}
dds=ddsDefGC_4
genes=genesGC_4
anChr="chrV"


transMeanTrisomy <- function(dds, genes, anChr,mu) {
counts <- counts(dds,normalized=TRUE)
ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)
ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]
mean(ratios.Trans$sampleOverAnc)
p <- t.test(ratios.Trans$sampleOverAnc,mu=mu,alternative = "less") # Ho: mu=3
p$p.value
return(p$p.value)
}

p.valGC_4 <- transMeanTrisomy(ddsDefGC_4,genesGC_4,"chrV",0.95)
p.valGC_4
p.valGC_7 <- transMeanTrisomy(ddsDefGC_7,genesGC_7,"chrI",0.98)
p.valGC_7
p.valGC_49 <- transMeanTrisomy(ddsDefGC_49,genesGC_49,"chrV",0.95)
p.valGC_49
p.valGC_59 <- transMeanTrisomy(ddsDefGC_59,genesGC_59,"chrVII",0.92)
p.valGC_59
p.valGC_61 <- transMeanTrisomy(ddsDefGC_61,genesGC_61,"chrVII",0.92)
p.valGC_61
p.valGC_77 <- transMeanTrisomy(ddsDefGC_77,genesGC_77,"chrXII",0.92)
p.valGC_77

transMeanTrisomyTable=NULL
rep.Trisomy <- c("ddsDefGC_7","ddsDefGC_4","ddsDefGC_49","ddsDefGC_59","ddsDefGC_61","ddsDefGC_77")
genesTris <- c("genesGC_7","genesGC_4","genesGC_49","genesGC_59","genesGC_61","genesGC_77")
anChr <- c("chrI","chrV","chrV","chrVII","chrXII")

transMeanTrisomyTable <- matrix(data=NA, nrow=3, ncol=length(rep.Trisomy))
View(transMeanTrisomyTable)
colnames(transMeanTrisomyTable) <-c("ddsDefGC_7","ddsDefGC_4","ddsDefGC_49","ddsDefGC_59","ddsDefGC_61","ddsDefGC_77") 
rownames(transMeanTrisomyTable) <- c("Expected_Mean","Observed_Mean","p-value (t-test")
transMeanTrisomyTable <- as.data.frame(transMeanTrisomyTable)

# meanTrisomyGC_7 <- transMeanTrisomy(ddsDefGC_7,genesGC_7,"chrI")

expectedMeanChrITri <- 0.98
# chisq.test(meanTrisomyGC_7,expectedMeanChrITri)

# meanTrisomyGC_4 <- transMeanTrisomy(ddsDefGC_4,)
# 
# for(line in rep.Trisomy){
#   for(genesTris in genesTris){
#     for(anChr in anChr) {
#     transMeanTrisomyTable <- c(transMeanTrisomyTable,transMeanTrisomy(rep.Trisomy, genesTris, anChr))
#     }
#   }
# }

p.valMA15 <- transMeanTrisomy(ddsDefMAold_15,genesMAold_15,"chrIX",0.96)
p.valMA15
p.valMA88 <- transMeanTrisomy(ddsDefMAold_88,genesMAold_88,"chrIX",0.96)
p.valMA88
p.valMA119 <- transMeanTrisomy(ddsDefMAold_119,genesMAold_119,"chrIX",0.96)
p.valMA119
p.valMA9 <- transMeanTrisomy(ddsDefMAold_9,genesMAold_9,"chrXIV",0.94)
p.valMA9


ratioDistributionsTrisomyMultiMANew <- function(dds, genes, anChr,anChri2,sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:5)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="magenta") + geom_vline(xintercept=0.5,col="green") + geom_vline(xintercept=2,col="orange") + coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),colour = "blue", linetype ="longdash", size = .5) + theme_classic() + labs(title=paste("Trisomy",anChr,"Cis Genes",sample,sep=" "),size = 6)

trans <- qplot(ratios.Trans$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)), main=paste("Trisomy",anChr, "Trans Genes",sample,sep=" ",size=7), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="magenta") + geom_vline(xintercept=0.5,col="green")+ geom_vline(xintercept=2,col="orange") + coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans$sampleOverAnc, na.rm = T)), colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5) + theme_classic() + labs(title=paste("Trisomy",anChr,"Trans Genes",sample,sep=" "),size = 6)
both <- plot_grid(cis, trans, nrow=1)
print (both)
}

ratioDistMA152.I <- ratioDistributionsTrisomyMultiMANew(ddsDefMAnewS152,genesMAnewS152,"chrI","chrVIII", "Hom Sample 152 Chr I")
ratioDistMA152.VIII <- ratioDistributionsTrisomyMultiMANew(ddsDefMAnewS152,genesMAnewS152,"chrVIII","chrI", "Hom Sample 152 Chr VIII")

### need a function for monosomy
ratioDistributionsMonosomy <- function(dds, genes, anChr,anChr2, sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]
ratios.keepTrans2 <- ratios$chr!=anChr2
ratios.Trans2 <- ratios.Trans[ratios.keepTrans2,]

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="magenta") + geom_vline(xintercept=0.5,col="green") + geom_vline(xintercept=2,col="orange") + coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5) + theme_classic() + labs(title=paste("Trisomy",anChr,"Cis Genes",sample,sep=" "),size = 6)

trans <- qplot(ratios.Trans2$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)), main=paste("Monosomy",anChr, "Trans Genes",sample,sep=" ",size=7), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="magenta") + geom_vline(xintercept=0.5,col="green") + geom_vline(xintercept=2,col="orange") + coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans2$sampleOverAnc, na.rm = T)), colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans2$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5) + theme_classic() + labs(title=paste("Trisomy",anChr,"Trans Genes",sample,sep=" "),size = 6)
both <- plot_grid(cis, trans, nrow=1)
print (both)
return(mean(ratios.Trans$sampleOverAnc))
}

# for GC line 11
ratioDistGC11.I <- ratioDistributionsMonosomy(ddsDefGC_11,genesGC_11,"chrI","chrXV", "Het Sample 11 Chr I")

# for MA old line 108
ratioDistMA108.IX <- ratioDistributionsMonosomy(ddsDefMAold_108,genesMAold_108,"chrIX","chrVIII" ,"Hom Sample 108 Chr IX")


### need to do this for tetrasomy not trisomy
ratioDistributionsTetrasomy <- function(dds, genes, anChr,sample) {

counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="magenta") + geom_vline(xintercept=0.5,col="green") + geom_vline(xintercept=2,col="orange") + coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),colour = "blue", linetype ="longdash", size = .5) + theme_classic() + labs(title=paste("Trisomy",anChr,"Cis Genes",sample,sep=" "),size = 6)

trans <- qplot(ratios.Trans$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="magenta") + geom_vline(xintercept=0.5,col="green") + geom_vline(xintercept=2,col="orange") + coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans$sampleOverAnc, na.rm = T)),colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5) + theme_classic() + labs(title=paste("Trisomy",anChr,"Trans Genes",sample,sep=" "),size = 6)
both <- plot_grid(cis, trans, nrow=1)
print (both)
return(mean(ratios.Trans$sampleOverAnc))
}

ratioDistGC8 <- ratioDistributionsTetrasomy(ddsDefGC_8,genesGC_8,"chrXVI","Het Sample 8 Chr XVI")

### will need to do MA New lines separately because only 2 ancestor replicates
ratioDistributionsTrisomyMANew <- function(dds, genes, anChr,sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:5)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="magenta") + geom_vline(xintercept=0.5,col="green") + geom_vline(xintercept=2,col="orange") + coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)), colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5) + theme_classic() + labs(title=paste("Trisomy",anChr,"Cis Genes",sample,sep=" "),size = 6)

trans <- qplot(ratios.Trans$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="magenta") + geom_vline(xintercept=0.5,col="green")+ geom_vline(xintercept=2,col="orange") + coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans$sampleOverAnc, na.rm = T)),colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5) + theme_classic() + labs(title=paste("Trisomy",anChr,"Trans Genes",sample,sep=" "),size = 6)
both <- plot_grid(cis, trans, nrow=1)
print (both)
return(mean(ratios.Trans$sampleOverAnc))
}

ratioDistMA112 <- ratioDistributionsTrisomyMANew(ddsDefMAnewS112,genesMAnewS112,"chrXVI","Hom Sample 112 Chr XVI")
ratioDistMA117 <- ratioDistributionsTrisomyMANew(ddsDefMAnewS117,genesMAnewS117,"chrV","Hom Sample 117 Chr V")
ratioDistMA123 <- ratioDistributionsTrisomyMANew(ddsDefMAnewS123,genesMAnewS123,"chrV","Hom Sample 123 Chr V")

## need monosomy one for MA new 29
ratioDistributionsMonosomyMANew <- function(dds, genes, anChr,sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:5)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram",breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=0.5, col="green") + geom_vline(xintercept=1.5,col="magenta") + geom_vline(xintercept=2, col="orange") + coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),colour = "blue", linetype ="longdash", size = .5) + theme_classic() + labs(title=paste("Monosomy",anChr,"Cis Genes",sample,sep=" "),size = 6)

trans <- qplot(ratios.Trans$sampleOverAnc,geom="histogram",breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise")) + geom_vline(xintercept = 1) + geom_vline(xintercept=0.5, col="green") + geom_vline(xintercept=1.5,col="magenta") + geom_vline(xintercept=2, col="orange") + coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans$sampleOverAnc, na.rm = T)),colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)),colour = "blue", linetype ="longdash", size = .5) + theme_classic() + labs(title=paste("Monosomy",anChr,"Trans Genes",sample,sep=" "),size = 6)
both <- plot_grid(cis, trans, nrow=1)
print (both)
}

ratioDistMA29 <- ratioDistributionsMonosomyMANew(ddsDefMAnewS29,genesMAnewS29,"chrIX","Hom Sample 29 Chr IX")

#### using ggplot to make figures 


library(ggplot2)

chr1 <- plot_grid(ratioDistGC7, ratioDistGC18, ratioDistMA152.I, ratioDistGC11.I, ncol = 1, nrow =4 )
pdf("chrI.pdf", width = 10, height= 10)
chr1
dev.off()

chr5 <- plot_grid(ratioDistMA117, ratioDistMA123, ratioDistGC4, ratioDistGC49, ncol = 1, nrow =4 )
pdf("chrV.pdf", width = 10, height= 10)
chr5
dev.off()

chr7 <- plot_grid(ratioDistMA152.VIII, ratioDistMA108, ratioDistGC59, ratioDistGC61, ncol = 1, nrow =4 )
pdf("chrVII.pdf", width = 10, height= 10)
chr7
dev.off()


# chr9 <- plot_grid(ratioDistMA108.IX, ratioDistMA15, ratioDistMA29, ratioDistMA88, ratioDistGC76.IX, ncol = 1, nrow =5 )
# pdf("chrIX.pdf", width = 10, height= 10)
# chr9
# dev.off()

# pdf("chrX.pdf", width=20, height=10)
# ratioDistGC76.X
# dev.off()

pdf("chrXII.pdf", width=20, height=10)
ratioDistGC77
dev.off()

# chr14 <- plot_grid(ratioDistMA9, ratioDistGC76.XIV,  ncol = 1, nrow = 2 )
# pdf("chrXIV.pdf", width = 10, height= 10)
# chr14
# dev.off()


pdf("chrXV.pdf", width=20, height=10)
ratioDistGC11
dev.off()


chr16 <- plot_grid(ratioDistMA112, ratioDistGC8,  ncol = 1, nrow = 2 )
pdf("chrXVI.pdf", width = 10, height= 10)
chr16
dev.off()


# need to make a figure for ratio expectations 
# load in tble 
ratioTable <- read.table(file="/Users/hollymcqueary/Dropbox/McQueary/DosageCompensationG3/Ratios.txt",header=TRUE)
library(ggplot2)
ratioTable

g <- ggplot(ratioTable,aes(Sample,Mean,ymin=Mean-2*standard_deviation,ymax=Mean+2*standard_deviation),main="Expected and Observed Mean Ratio of Trans Gene Expression in Aneuploid Samples") + theme_classic()  + geom_pointrange() + theme(axis.text.x = element_text(family = "Helvetica", face="bold", size = (8),angle = 45, hjust = 1)) + geom_point(y =  ratioTable$standard_deviation,col="pink") 
g <- g + labs(y = "Mean Ratio Sample/Anc" , title="Expected and Observed Mean Ratio of Trans Gene Expression")
print(g)
pdf("expRatios.pdf")
g
dev.off()

```

```{r hist_main_body, include=FALSE}

##need to remove labels from plots 

ratioDistGC4 <- ratioDistributionsTrisomy(ddsDefGC_4,genesGC_4,"chrV","A") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA117 <- ratioDistributionsTrisomyMAnew(ddsDefMAnewS117,genesMAnewS117,"chrV","B") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistGC59 <- ratioDistributionsTrisomy(ddsDefGC_59,genesGC_59,"chrVII","C") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA15 <- ratioDistributionsTrisomy(ddsDefMAold_15,genesMAold_15,"chrIX","D") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA29 <- ratioDistributionsMonosomyMAnew(ddsDefMAnewS29,genesMAnewS29,"chrIX","E") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA9 <- ratioDistributionsTrisomy(ddsDefMAold_9,genesMAold_9,"chrXIV","F") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistGC8 <- ratioDistributionsTetrsomy(ddsDefGC_8,genesGC_8,"chrXVI","G") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistMA112 <- ratioDistributionsTrisomyMAnew(ddsDefMAnewS112,genesMAnewS112,"chrXVI","H") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )




pdf("ratios_one_chr_main_body.pdf")
ratiosOneChr <- ggarrange(ratioDistGC4, ratioDistMA117, ratioDistGC59, ratioDistMA15, ratioDistMA29, ratioDistMA9, ratioDistGC8, ratioDistMA112,ncol=2,nrow=4, label.x="Ratio (Line/Ancestor)",label.y = "Number of Genes", font.label=list(size=8), common.legend = TRUE)
annotate_figure(ratiosOneChr,
                bottom = text_grob("Ratio", size = 10),
                left = text_grob("Number of Genes", rot = 90,size=10))
dev.off() 


ratioDistGC11 <- ratioDistributionsTriMonMulti(ddsDefGC_11,genesGC_11,"chrI","chrXV", "A") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistGC18 <- ratioDistributionsTrisomyMulti(ddsDefGC_18,genesGC_18,"chrI","chrXII", "B") + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )
ratioDistGC76 + theme(axis.title.y = element_blank(),  axis.title.x = element_blank() )

pdf("ratios_multi_chrom_main_body.pdf")
ratiosMultiChr <- ggarrange(ratioDistGC11, ratioDistGC18, ratioDistGC76,nrow=3, label.x="Ratio (Line/Ancestor)",label.y = "Number of Genes", font.label=list(size=8), common.legend = FALSE) 
annotate_figure(ratiosMultiChr,
                bottom = text_grob("Ratio", size = 10),
                left = text_grob("Number of Genes", rot = 90,size=10))
dev.off()




```








Note that we could have specified the coefficient or contrast we want
to build a results table for, using either of the following equivalent
commands:

```{r results, eval=TRUE,include=TRUE}
###########################################
#without aneuploid chromosomes
# resGC <- results(ddsGCNoAneuInNorm)
# resGC
# resultsNames(ddsGCNoAneuInNorm)
# #euploid line
# resGC2 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_2","Holly_Anc"))
# summary(resGC2)
# sigGenesGC2 <- rownames(subset(resGC2,padj<0.1))
# #aneuploid line
# resultsNames(ddsGCNoAneuInNorm)
# resGC7 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_7","Holly_Anc"))#,cooksCutoff=FALSE)
# summary(resGC7)
# # testing with no indepdent filtering - does it change the results
# resGC7.test <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_7","Holly_Anc"), independentFiltering = FALSE)#,cooksCutoff=FALSE)
# summary(resGC7.test) # no difference
# # what about without cooksCutoff
# resGC7.test2 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_7","Holly_Anc"), independentFiltering = FALSE ,cooksCutoff=FALSE)
# summary(resGC7.test2) # no difference
#
# sigGenesGC7 <- rownames(subset(resGC7,padj<0.1))
# avgRatioGC7 <- mean(resGC7$log2FoldChange)
# avgRatioGC7
# resGC18 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_18","Holly_Anc"), independentFiltering=FALSE) #, cooksCutoff = FALSE)
# summary(resGC18)
# sigGenesGC18 <- rownames(subset(resGC18,padj<0.1))
# avgRatioGC18 <- mean(resGC18$log2FoldChange)
# resGC11 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_11","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC11)
# sigGenesGC11 <- rownames(subset(resGC11,padj<0.1))
# resGC1 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_1","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC1)
# avgRatioGC11 <- mean(resGC11$log2FoldChange)
# sigGenesGC1 <- rownames(subset(resGC1,padj<0.1))
# resGC3 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_3","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC3)
# sigGenesGC3 <- rownames(subset(resGC3,padj<0.1))
# resGC4 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_4","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC4)
# sigGenesGC4 <- rownames(subset(resGC4,padj<0.1))
# resGC49 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_49","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC49)
# sigGenesGC49 <- rownames(subset(resGC49,padj<0.1))
# resGC59 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_59","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC59)
# sigGenesGC59 <- rownames(subset(resGC59,padj<0.1))
# resGC61 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_61","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC61)
# sigGenesGC61 <- rownames(subset(resGC61,padj<0.1))
# resGC8 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_8","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC8)
# sigGenesGC8 <- rownames(subset(resGC8,padj<0.1))
# resGC9 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_9","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC9)
# sigGenesGC9 <- rownames(subset(resGC9,padj<0.1))
# #res69 <- results(ddsNoAneuInNorm, contrast=c("condition","Holly_69","Holly_Anc"))#, cooksCutoff = FALSE)
# #summary(res69)
# resGC76 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_76","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC76)
# sigGenesGC76 <- rownames(subset(resGC76,padj<0.1))
# resGC77 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_77","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC77)
# sigGenesGC77 <- rownames(subset(resGC77,padj<0.1))
#
# resGC5 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_5","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC5)
# sigGenesGC5 <- rownames(subset(resGC5,padj<0.1))

###########################################
#with aneuploid chromosomes
# resDefGC <- results(ddsDefGC)
# resDefGC
# resultsNames(ddsDefGC)
#euploid line
# resDefGC1 <- results(ddsDefGC, contrast=c("condition","Holly_1","Holly_Anc"))
# summary(resDefGC1)
# resDefGC2 <- results(ddsDefGC, contrast=c("condition","Holly_2","Holly_Anc"))
# summary(resDefGC2)
# resDefGC3 <- results(ddsDefGC, contrast=c("condition","Holly_3","Holly_Anc"))
# summary(resDefGC3)
# resDefGC4 <- results(ddsDefGC, contrast=c("condition","Holly_4","Holly_Anc"))
# summary(resDefGC4)
# resDefGC7 <- results(ddsDefGC, contrast=c("condition","Holly_7","Holly_Anc"))
# summary(resDefGC7)
#
#
# ###########################################
# #without aneuploid chromosomes
# resMAold <- results(ddsMAoldNoAneuInNorm)
# resMAold
# resultsNames(ddsMAoldNoAneuInNorm)
# #euploid line
# resMAold1 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC001","SCA"))
# summary(resMAold1)
# resMAold2 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC002","SCA"))
# summary(resMAold2)
# resMAold3 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC003","SCA"))
# summary(resMAold3)
# resMAold4 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC004","SCA"))
# summary(resMAold4)
# resMAold5 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC005","SCA"))
# summary(resMAold5)
# resMAold6 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC006","SCA"))
# summary(resMAold6)
# resMAold7 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC007","SCA"))
# summary(resMAold7)
# resMAold8 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC008","SCA"))
# summary(resMAold8)
# resMAold9 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC009","SCA"))
# summary(resMAold9)
# resMAold11 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC011","SCA"))
# summary(resMAold11)
# resMAold15 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC015","SCA"))
# summary(resMAold15)
# resMAold28 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC028","SCA"))
# summary(resMAold28)
# resMAold88 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC088","SCA"))
# summary(resMAold88)
# resMAold108 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC108","SCA"))
# summary(resMAold108)
# resMAold119 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC119","SCA"))
# summary(resMAold119)
#
# #############################################
# ### MA New no aneuploid chromosomes in normalization ####################
#
# resMAnew <- results(ddsMAnewNoAneuInNorm)
# resMAnew
# resultsNames(ddsMAnewNoAneuInNorm)
# #euploid line
# resMAnew29 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","29","Anc"))
# summary(resMAnew29)
# resMAnew50 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","50","Anc"))
# summary(resMAnew50)
# resMAnew112 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","112","Anc"))
# summary(resMAnew112)
# resMAnew115 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","115","Anc"))
# summary(resMAnew115)
# resMAnew117 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","117","Anc"))
# summary(resMAnew117)
# resMAnew123 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","123","Anc"))
# summary(resMAnew123)
# resMAnew141 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","141","Anc"))
# summary(resMAnew141)
# resMAnew152 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","152","Anc"))
# summary(resMAnew152)
#

### DESeq ran on samples individually
resMAnewS29 <- results(ddsDefMAnewS29)
resultsNames(ddsDefMAnewS29)
#euploid line
resMAnew29 <- results(ddsDefMAnewS29, contrast=c("condition","29","Anc"))
summary(resMAnew29)

resMAnew50 <- results(ddsDefMAnewS50, contrast=c("condition","50","Anc"))
summary(resMAnew50)

resMAnew112 <- results(ddsDefMAnewS112, contrast=c("condition","112","Anc"))
summary(resMAnew112)
resMAnew112$pvalue

resMAnew115 <- results(ddsDefMAnewS115, contrast=c("condition","115","Anc"))
summary(resMAnew115)

resMAnew117 <- results(ddsDefMAnewS117, contrast=c("condition","117","Anc"))
summary(resMAnew117)

resMAnew123 <- results(ddsDefMAnewS123, contrast=c("condition","123","Anc"))
summary(resMAnew123)

# resMAnew141 <- results(ddsDefMAnewS141, contrast=c("condition","141","Anc"))
# summary(resMAnew141)

resMAnew152 <- results(ddsDefMAnewS152, contrast=c("condition","152","Anc"))
summary(resMAnew152)

```

```{r common sig genes, eval=TRUE,include=TRUE}
## Big Problem: How many genes are differentially expressed in ALL samples?
# need to get results from DESeq into a more easily workable format (ie data frame)
  # make results from DESeq a data frame to work with
  resGC1Dat <- as.data.frame(resultsS1GC)
  resGC2Dat <- as.data.frame(resultsS2GC)
  resGC3Dat <- as.data.frame(resultsS3GC)
  resGC5Dat <- as.data.frame(resultsS5GC)
  resGC9Dat <- as.data.frame(resultsS9GC)
  resGC4Dat <- as.data.frame(resultsS4GC)
  resGC49Dat <- as.data.frame(resultsS49GC)
  resGC7Dat <- as.data.frame(resultsS7GC)
  resGC8Dat <- as.data.frame(resultsS8GC)
  resGC11Dat <- as.data.frame(resultsS11GC)
  resGC18Dat <- as.data.frame(resultsS18GC)
  resGC59Dat <- as.data.frame(resultsS59GC)
  resGC61Dat <- as.data.frame(resultsS61GC)
  resGC76Dat <- as.data.frame(resultsS76GC)
  resGC77Dat <- as.data.frame(resultsS77GC)
  # add gene names as first column ( which is currently rownames)
  resGC1Dat <- cbind(GENEID = rownames(resGC1Dat), resGC1Dat)
  resGC2Dat <- cbind(GENEID = rownames(resGC2Dat), resGC2Dat)
  resGC3Dat <- cbind(GENEID = rownames(resGC3Dat), resGC3Dat)
  resGC4Dat <- cbind(GENEID = rownames(resGC4Dat), resGC4Dat)
  resGC49Dat <- cbind(GENEID = rownames(resGC49Dat), resGC49Dat)

  resGC5Dat <- cbind(GENEID = rownames(resGC5Dat), resGC5Dat)
  resGC7Dat <- cbind(GENEID = rownames(resGC7Dat), resGC7Dat)
  resGC8Dat <- cbind(GENEID = rownames(resGC8Dat), resGC8Dat)
  resGC9Dat <- cbind(GENEID = rownames(resGC9Dat), resGC9Dat)
  resGC11Dat <- cbind(GENEID = rownames(resGC11Dat), resGC11Dat)
  resGC18Dat <- cbind(GENEID = rownames(resGC18Dat), resGC18Dat)
  resGC59Dat <- cbind(GENEID = rownames(resGC59Dat), resGC59Dat)
  resGC61Dat <- cbind(GENEID = rownames(resGC61Dat), resGC61Dat)
  resGC76Dat <- cbind(GENEID = rownames(resGC76Dat), resGC76Dat)
  resGC77Dat <- cbind(GENEID = rownames(resGC77Dat), resGC77Dat)

  # need to make a column identifying the sample
    resGC1Dat$Sample <- "1"
    resGC2Dat$Sample <- "2"
    resGC3Dat$Sample <- "3"
    resGC4Dat$Sample <- "4"
    resGC5Dat$Sample <- "5"
    resGC7Dat$Sample <- "7"
    resGC8Dat$Sample <- "8"
    resGC9Dat$Sample <- "9"
    resGC11Dat$Sample <- "11"
    resGC18Dat$Sample <- "18"
    resGC49Dat$Sample <- "49"
    resGC59Dat$Sample <- "59"
    resGC61Dat$Sample <- "61"
    resGC76Dat$Sample <- "76"
    resGC77Dat$Sample <- "77"

View(resGC7Dat)


      # put all sample expression data in the same data frame
      allGCSamplesExpr <- rbind(resGC1Dat, resGC2Dat, resGC3Dat, resGC4Dat, resGC5Dat, resGC7Dat, resGC8Dat, resGC9Dat, resGC11Dat, resGC18Dat, resGC59Dat, resGC61Dat, resGC76Dat, resGC77Dat)
      str(allGCSamplesExpr)
    # # # add chromosome numbers to this file
        matchGC <- intersect(allGCSamplesExpr$GENEID, genesGC_1$GENEID)
        str(matchGC)
    # #   # tell R which genes to keep
       keepGC <- genesGC_1$GENEID %in% matchGC
       str(keepGC)
    # #   # subset list of genes in analysis to include only the ones matching those found in          sample file
       genesGC <- genesGC_1[keepGC,]
       str(genesGC)
       View(genesGC)
       View(genesGC_1)
       dim(allGCSamplesExpr)
       dim(genesGC)
       str(genesGC_1)
       str(genesGC)
    # #   # add the correct chromosomes in
        allGCSamplesExpr <- cbind(genesGC_1$GENEID, genesGC_1$TXCHROM, allGCSamplesExpr)
        str(allGCSamplesExpr)
        View(allGCSamplesExpr)
       chr1.all <- subset(allGCSamplesExpr, allGCSamplesExpr$`genesNoAneu$GENEID`=="chrI")
       sample1 <- subset(allGCSamplesExpr, Sample==1)
       View(sample1)
       sample7 <- subset(allGCSamplesExpr, Sample==7)

    #    { plot(sample1$log2FoldChange,sample7$log2FoldChange)
    #    # want to draw two lines to indicate differential expression cutoffs
    #    # one for monosomy
    #    abline(-1, 1, col='purple', lty="dashed")
    #    #one for trisomy
    #     abline(0.583, 1, col="purple", lty="dashed")
    #     # # one for tetrasomy
    #     # abline(1, 1, col="purple", lty="dashed")
    #     # tell R to color dots above or below these lines certain colors
    #     above <- sample7$log2FoldChange-sample1$log2FoldChange > 1
    #     points(sample7$log2FoldChange[above], sample1$log2FoldChange[above], col="red", cex=0.5)
    #     below <- sample1$log2FoldChange-sample7$log2FoldChange < -1
    #     points(sample1$log2FoldChange[below], sample7$log2FoldChange[below], col="blue", cex=0.5) }
    #            
# want to find how many genes are differentially expressed in 2 samples, 3 samples, 4 samples, etc
#
  #     cbind(names = rownames(df), df)
  # # first need to filter allSamplesExpr based on p-adj level
    allGCSamplesDiffExpr <- subset(allGCSamplesExpr, padj<0.1)
    write.csv(allGCSamplesDiffExpr,file="allGCDiffExpr.csv")
  #   # then ask how many times each GENEID is found in allSamplesExpr
    library(plyr)
    geneGCFreq <- plyr::count(allGCSamplesDiffExpr, c("GENEID"))
      # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
      numGCGenesDE <- plyr::count(geneGCFreq,"freq")
        # rename columns so that I know what is going on: column 1 is how many samples a gene is DE in, column 2 is how many           genes are in that category
        colnames(numGCGenesDE) <- c("numSamples", "numGenes")
          # export as .csv to use in excel or whatever
          write.csv(numGCGenesDE, file="numb_comm_DE_genesALL_GC.csv")

##########################
          #### Same thing but for MA New Lines #####

  resMA29Dat <- as.data.frame(resMAnew29)
  resMA50Dat <- as.data.frame(resMAnew50)
  resMA112Dat <- as.data.frame(resMAnew112)
  resMA115Dat <- as.data.frame(resMAnew115)
  resMA117Dat <- as.data.frame(resMAnew117)
  resMA123Dat <- as.data.frame(resMAnew123)
  resMA152Dat <- as.data.frame(resMAnew152)

  # add gene names as first column ( which is currently rownames)
  resMA29Dat <- cbind(GENEID = rownames(resMA29Dat), resMA29Dat)
  resMA50Dat <- cbind(GENEID = rownames(resMA50Dat), resMA50Dat)
  resMA112Dat <- cbind(GENEID = rownames(resMA112Dat), resMA112Dat)
  resMA115Dat <- cbind(GENEID = rownames(resMA115Dat), resMA115Dat)
  resMA117Dat <- cbind(GENEID = rownames(resMA117Dat), resMA117Dat)
  resMA123Dat <- cbind(GENEID = rownames(resMA123Dat), resMA123Dat)
  resMA152Dat <- cbind(GENEID = rownames(resMA152Dat), resMA152Dat)
# need to make a column identifying the sample
    resMA29Dat$Sample <- "29"
    resMA50Dat$Sample <- "50"
    resMA112Dat$Sample <- "112"
    resMA115Dat$Sample <- "115"
    resMA117Dat$Sample <- "117"
    resMA123Dat$Sample <- "123"
    resMA152Dat$Sample <- "152"


View(resMA29Dat)

##### LEFT OFF HERE
## ** struggling to cbind genes with allMAnewSamplesExpr
## worked for GC samples and nothing has changed so why is it not working for these samples??


   allMAnewSamplesExpr <- rbind(resMA29Dat, resMA50Dat, resMA112Dat, resMA115Dat, resMA117Dat, resMA123Dat, resMA152Dat)
      str(allGCSamplesExpr)
    # # # add chromosome numbers to this file
        matchMAnew<- intersect(allMAnewSamplesExpr$GENEID, genesGC_1$GENEID)
        str(matchMAnew)
    # #   # tell R which genes to keep
       keepMAnew <- genesGC_1$GENEID %in% matchMAnew
    str(keepMAnew)
       # #   # subset list of genes in analysis to include only the ones matching those found in          sample file
       genesMAnew <- genesGC_1[keepMAnew,]
       str(genesMAnew)
       View(genesMAnew)
       View(genesGC_1)
       dim(allMAnewSamplesExpr)
       dim(genesMAnew)
       str(genesGC_1)
       str(genesGC)
    # #   # add the correct chromosomes in
        allMAnewSamplesExpr <- cbind(genesGC_1$GENEID, genesGC_1$TXCHROM, allMAnewSamplesExpr)
        str(allMAnewSamplesExpr)
        View(allMAnewSamplesExpr)
       chr1.all <- subset(allMAnewSamplesExpr, allMAnewSamplesExpr$`genesNoAneu$GENEID`=="chrI")
       sample29MAnew <- subset(allMAnewSamplesExpr, Sample==29)
       View(sample29MAnew)
       sample50MAnew <- subset(allMAnewSamplesExpr, Sample==50)

    #    { plot(sample1$log2FoldChange,sample7$log2FoldChange)
    #    # want to draw two lines to indicate differential expression cutoffs
    #    # one for monosomy
    #    abline(-1, 1, col='purple', lty="dashed")
    #    #one for trisomy
    #     abline(0.583, 1, col="purple", lty="dashed")
    #     # # one for tetrasomy
    #     # abline(1, 1, col="purple", lty="dashed")
    #     # tell R to color dots above or below these lines certain colors
    #     above <- sample7$log2FoldChange-sample1$log2FoldChange > 1
    #     points(sample7$log2FoldChange[above], sample1$log2FoldChange[above], col="red", cex=0.5)
    #     below <- sample1$log2FoldChange-sample7$log2FoldChange < -1
    #     points(sample1$log2FoldChange[below], sample7$log2FoldChange[below], col="blue", cex=0.5) }
    #            
# want to find how many genes are differentially expressed in 2 samples, 3 samples, 4 samples, etc
#
  #     cbind(names = rownames(df), df)
  # # first need to filter allSamplesExpr based on p-adj level
    allMAnewSamplesDiffExpr <- subset(allMAnewSamplesExpr, padj<0.1)
    write.csv(allMAnewSamplesDiffExpr,file="allMAnewDiffExpr.csv")
  #   # then ask how many times each GENEID is found in allSamplesExpr
    library(plyr)
    geneMAnewFreq <- plyr::count(allMAnewSamplesDiffExpr, c("GENEID"))
      # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
      numMAnewGenesDE <- plyr::count(geneMAnewFreq,"freq")
        # rename columns so that I know what is going on: column 1 is how many samples a gene is DE in, column 2 is how many           genes are in that category
        colnames(numMAnewGenesDE) <- c("numSamples", "numGenes")
          # export as .csv to use in excel or whatever
          write.csv(numMAnewGenesDE, file="numb_comm_DE_genesALL_MAnew.csv")
          ################## MA old Lines
  ########################################################################
  # make results from DESeq a data frame to work with
  resMAold1Dat <- as.data.frame(resultsS1MAold)
  resMAold2Dat <- as.data.frame(resultsS2MAold)
  resMAold3Dat <- as.data.frame(resultsS3MAold)
  resMAold4Dat <- as.data.frame(resultsS4MAold)
  resMAold5Dat <- as.data.frame(resultsS5MAold)
  resMAold7Dat <- as.data.frame(resultsS7MAold)
  resMAold8Dat <- as.data.frame(resultsS8MAold)
  resMAold9Dat <- as.data.frame(resultsS9MAold)
  resMAold15Dat <- as.data.frame(resultsS15MAold)
  resMAold28Dat <- as.data.frame(resultsS28MAold)
  resMAold88Dat <- as.data.frame(resultsS88MAold)
  resMAold108Dat <- as.data.frame(resultsS108MAold)
  resMAold119Dat <- as.data.frame(resultsS119MAold)
  # add gene names as first column ( which is currently rownames)
  resMAold1Dat <- cbind(GENEID = rownames(resMAold1Dat), resMAold1Dat)
  resMAold2Dat <- cbind(GENEID = rownames(resMAold2Dat), resMAold2Dat)
  resMAold3Dat <- cbind(GENEID = rownames(resMAold3Dat), resMAold3Dat)
  resMAold4Dat <- cbind(GENEID = rownames(resMAold4Dat), resMAold4Dat)
  resMAold5Dat <- cbind(GENEID = rownames(resMAold5Dat), resMAold5Dat)
  resMAold7Dat <- cbind(GENEID = rownames(resMAold7Dat), resMAold7Dat)
  resMAold8Dat <- cbind(GENEID = rownames(resMAold8Dat), resMAold8Dat)
  resMAold9Dat <- cbind(GENEID = rownames(resMAold9Dat), resMAold9Dat)
  resMAold15Dat <- cbind(GENEID = rownames(resMAold15Dat), resMAold15Dat)
  resMAold28Dat <- cbind(GENEID = rownames(resMAold28Dat), resMAold28Dat)
  # res49Dat <- cbind(GENEID = rownames(res49Dat), res49Dat)
  resMAold88Dat <- cbind(GENEID = rownames(resMAold88Dat), resMAold88Dat)
  resMAold108Dat <- cbind(GENEID = rownames(resMAold108Dat), resMAold108Dat)
  resMAold119Dat <- cbind(GENEID = rownames(resMAold119Dat), resMAold119Dat)


  # need to make a column identifying the sample
    resMAold1Dat$Sample <- "1"
    resMAold2Dat$Sample <- "2"
    resMAold3Dat$Sample <- "3"
    resMAold4Dat$Sample <- "4"
    resMAold5Dat$Sample <- "5"
    resMAold7Dat$Sample <- "7"
    resMAold8Dat$Sample <- "8"
    resMAold9Dat$Sample <- "9"
    resMAold15Dat$Sample <- "15"
    resMAold28Dat$Sample <- "28"
    resMAold88Dat$Sample <- "88"
    resMAold108Dat$Sample <- "108"
    resMAold119Dat$Sample <- "119"

      # put all sample expression data in the same data frame
allMAoldSamplesExpr <- rbind(resMAold1Dat, resMAold2Dat, resMAold3Dat, resMAold4Dat, resMAold5Dat, resMAold7Dat, resMAold8Dat, resMAold9Dat, resMAold15Dat, resMAold28Dat, resMAold88Dat, resMAold108Dat, resMAold119Dat)
      str(allMAoldSamplesExpr)
      View(allMAoldSamplesExpr)
    # # # add chromosome numbers to this file
        matchMAold <- intersect(allMAoldSamplesExpr$GENEID, genesGC_1$GENEID)
        str(matchMAold)

    # #   # tell R which genes to keep
       keepMAold <- genesGC_1$GENEID %in% matchMAold
       str(keepMAold)

    # #   # subset list of genes in analysis to include only the ones matching those found in          sample file
       genesMAold <- genesGC_1[keepMAold,]
       str(genesMAold)
       View(genesMAold)
       View(genesGC_1)
       dim(allMAoldSamplesExpr)
       dim(genesMAold)
       str(genesGC_1)
       str(genesMAold)
    # #   # add the correct chromosomes in
        allMAoldSamplesExpr <- cbind(genesMAold$GENEID, genesMAold$TXCHROM, allMAoldSamplesExpr)

        str(allGCSamplesExpr)
        View(allGCSamplesExpr)
       chr1.all <- subset(allGCSamplesExpr, allGCSamplesExpr$`genesNoAneu$GENEID`=="chrI")
       sample1 <- subset(allGCSamplesExpr, Sample==1)
       View(sample1)
       sample7 <- subset(allGCSamplesExpr, Sample==7)

    #    { plot(sample1$log2FoldChange,sample7$log2FoldChange)
    #    # want to draw two lines to indicate differential expression cutoffs
    #    # one for monosomy
    #    abline(-1, 1, col='purple', lty="dashed")
    #    #one for trisomy
    #     abline(0.583, 1, col="purple", lty="dashed")
    #     # # one for tetrasomy
    #     # abline(1, 1, col="purple", lty="dashed")
    #     # tell R to color dots above or below these lines certain colors
    #     above <- sample7$log2FoldChange-sample1$log2FoldChange > 1
    #     points(sample7$log2FoldChange[above], sample1$log2FoldChange[above], col="red", cex=0.5)
    #     below <- sample1$log2FoldChange-sample7$log2FoldChange < -1
    #     points(sample1$log2FoldChange[below], sample7$log2FoldChange[below], col="blue", cex=0.5) }
    #            
# want to find how many genes are differentially expressed in 2 samples, 3 samples, 4 samples, etc
#
  #     cbind(names = rownames(df), df)
  # # first need to filter allSamplesExpr based on p-adj level
    allGCSamplesDiffExpr <- subset(allGCSamplesExpr, padj<0.1)
    write.csv(allGCSamplesDiffExpr,file="allGCDiffExpr.csv")
  #   # then ask how many times each GENEID is found in allSamplesExpr
    library(plyr)
    geneGCFreq <- plyr::count(allGCSamplesDiffExpr, c("GENEID"))
      # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
      numGCGenesDE <- plyr::count(geneGCFreq,"freq")
        # rename columns so that I know what is going on: column 1 is how many samples a gene is DE in, column 2 is how many           genes are in that category
        colnames(numGCGenesDE) <- c("numSamples", "numGenes")
          # export as .csv to use in excel or whatever
          write.csv(numGCGenesDE, file="numb_comm_DE_genesALL_GC.csv")          


##############################################################################################          
# want to know how many genes are differentially expressed commonly in 1,2,3 etc EUPLOID lines
#
  # put all EUPLOID samples in same data frame
    eupGCSamplesExpr <- rbind(resGC1Dat, resGC2Dat, resGC3Dat, resGC5Dat, resGC9Dat)
      # first need to filter allSamplesExpr based on p-adj level
      eupGCSamplesDiffExpr <- subset(eupGCSamplesExpr, padj<0.1)
      #

        # then ask how many times each GENEID  is found in eupSamplesExpr
        library(plyr)
        geneGCFreqEu <- plyr::count(eupGCSamplesDiffExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numGCEupGenesDE <- plyr::count(geneGCFreqEu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numGCEupGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numGCEupGenesDE, file="numb_comm_DE__GC_genesEUP.csv")
##############################################################################################          
# want to know how many genes are differentially expressed commonly in 1,2,3 etc EUPLOID lines
#
  # put all EUPLOID samples in same data frame
    eupMAnewSamplesExpr <- rbind(resMA50Dat, resMA115Dat)
      # first need to filter allSamplesExpr based on p-adj level
      eupMAnewSamplesDiffExpr <- subset(eupMAnewSamplesExpr, padj<0.1)
      #

        # then ask how many times each GENEID  is found in eupSamplesExpr
        library(plyr)
        geneMAnewFreqEu <- plyr::count(eupMAnewSamplesDiffExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numMAnewEupGenesDE <- plyr::count(geneMAnewFreqEu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numMAnewEupGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numMAnewEupGenesDE, file="numb_comm_DE_MAnew_genesEUP.csv")              
##############################################################################################          
# want to know how many genes are differentially expressed commonly in 1,2,3 etc ANEUPLOID lines
#
  # put all ANEUPLOID samples in same data frame
    aneuGCSamplesExpr <- rbind(resGC4Dat, resGC7Dat, resGC8Dat,  resGC11Dat, resGC18Dat, resGC59Dat, resGC61Dat, resGC76Dat, resGC77Dat)
      # first need to filter allSamplesExpr based on p-adj level
      aneuGCSamplesDiffExpr <- subset(aneuGCSamplesExpr, padj<0.1)
      #
        # then ask how many times each GENEID  is found in eupSamplesExpr
        library(plyr)
        geneGCFreqAneu <- plyr::count(aneuGCSamplesDiffExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numGCAneuGenesDE <- plyr::count(geneGCFreqAneu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numGCAneuGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numGCAneuGenesDE, file="numb_comm_DE_genesANEU.csv")     

##############################################################################################          
# want to know how many genes are differentially expressed commonly in 1,2,3 etc ANEUPLOID lines
# with the SAME ANEUPLOID CHROMOSOME
  # put all samples in same data frame
    aneuChrIGCSamplesExpr <- rbind(resGC7Dat,  resGC11Dat, resGC18Dat)
      aneuChrItriGCExpr <- rbind(resGC7Dat, resGC18Dat)
      aneuChrVtriGCExpr <- rbind(resGC4Dat)
      aneuChrVIItriGCExpr <- rbind(resGC59Dat, resGC61Dat)
      aneuChrXIItriGCExpr <- rbind(resGC18Dat, resGC77Dat)
      aneuLine76GC <- resGC76Dat
      aneuLine76GC <- merge(genesGC,resGC76Dat,by="GENEID")
      write.csv(aneuLine76GC,file="aneuLine76GC.csv")
      aneuLine76GCdiffExpr <- subset(resGC76Dat,padj<0.1)
      aneuLine76GCdiffExpr <- merge(genesGC,aneuLine76GCdiffExpr,by.x="GENEID",by.y="GENEID")
      write.csv(aneuLine76GCdiffExpr,file="aneuLine76GCdiffExpr.csv")

      # first need to filter allSamplesExpr based on p-adj level
      aneuChrIGCSamplesDiffExpr <- subset(aneuChrIGCSamplesExpr, padj<0.1)
        triChrIGCSamplesDiffExpr <- subset(aneuChrItriGCExpr, padj<0.1)
        triChrVGCSamplesDiffExpr <- subset(aneuChrVtriGCExpr, padj<0.1)
        triChrVIIGCSamplesDiffExpr <- subset(aneuChrVIItriGCExpr, padj<0.1)
        triChrXIIGCSamplesDiffExpr <- subset(aneuChrXIItriGCExpr, padj<0.1)
      #
        # then ask how many times each GENEID  is found in eupSamplesExpr
        library(plyr)
        geneFreqGCAneuChrI <- plyr::count(aneuChrIGCSamplesDiffExpr, c("GENEID"))
          geneFreqGCtriChrI <- plyr::count(triChrIGCSamplesDiffExpr, c("GENEID"))
          geneFreqGCtriChrV <- plyr::count(triChrVGCSamplesDiffExpr, c("GENEID"))
          geneFreqGCtriChrVII <- plyr::count(triChrVIIGCSamplesDiffExpr, c("GENEID"))
          geneFreqtriGCChrXII <- plyr::count(triChrXIIGCSamplesDiffExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numGCAneuChrIGenesDE <- plyr::count(geneFreqGCAneuChrI,"freq")
            numtriGCChrIGenesDE <- plyr::count(geneFreqGCtriChrI,"freq")
            numtriGCChrVGenesDE <- plyr::count(geneFreqGCtriChrV,"freq")
            numtriGCChrVIIGenesDE <- plyr::count(geneFreqGCtriChrVII,"freq")
            numtriGCChrXIIGenesDE <- plyr::count(geneFreqtriGCChrXII,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numGCAneuChrIGenesDE) <- c("numSamples", "numGenes")
              colnames(numtriGCChrIGenesDE) <- c("numSamples", "numGenes")
              colnames(numtriGCChrVGenesDE) <- c("numSamples", "numGenes")
              colnames(numtriGCChrVIIGenesDE) <- c("numSamples", "numGenes")
              colnames(numtriGCChrXIIGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numGCAneuChrIGenesDE, file="numb_comm_DE_GC_genesANEUchrI.csv")  
                write.csv(numtriGCChrIGenesDE, file="numb_comm_DE_GC_genesTRIchrI.csv")
                write.csv(numtriGCChrVGenesDE, file="numb_comm_DE_GC_genesTRIchrV.csv")
                write.csv(numtriGCChrVIIGenesDE, file="numb_comm_DE_GC_genesTRIchrVII.csv")
                write.csv(numtriGCChrXIIGenesDE, file="numb_comm_DE_GC_genesTRIchrXII.csv")

# want to know how many genes are NOT DE on the aneuploid chromosome in aneuploid samples
#
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1
  # #
  #   sample4notDE <- subset(res4Dat, padj>0.1)
  #   #
  #   sample49notDE <- subset(res49Dat, padj>0.1)  
  #   #
  #   sample7notDE <- subset(res7Dat, padj>0.1)
  #   #
  #   sample11notDE <- subset(res11Dat, padj>0.1)
  #   #
  #   sample18notDE <- subset(res18Dat, padj>0.1)
  #   #
  #   sample59notDE <- subset(res59Dat, padj>0.1)
  #   # then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
  #   #
  #   # add column with chromosome numbers matching the GENEIDs
  #     matchedS4 <- intersect(sample4notDE$GENEID, genesNoAneu$GENEID)
  #     keep4 <- genesNoAneu$GENEID %in% matchedS4
  #     s4 <- genesNoAneu[keep4,]
  #     test4 <- cbind(s4$GENEID, s4$TXCHROM, sample4notDE)
  #     sample4 <- test4[,2:10]
  #     #
  #     matchedS49 <- intersect(sample49notDE$GENEID, genesNoAneu$GENEID)
  #     keep49 <- genesNoAneu$GENEID %in% matchedS49
  #     s49 <- genesNoAneu[keep49,]
  #     test49 <- cbind(s49$GENEID, s49$TXCHROM, sample49notDE)
  #     sample49 <- test49[,2:10]
  #     #
  #     matchedS7 <- intersect(sample7notDE$GENEID, genesNoAneu$GENEID)
  #     keep7 <- genesNoAneu$GENEID %in% matchedS7
  #     s7 <- genesNoAneu[keep7,]
  #     test7 <- cbind(s7$GENEID, s7$TXCHROM, sample7notDE)
  #     sample7 <- test7[,2:10]
  #     #
  #     matchedS11 <- intersect(sample11notDE$GENEID, genesNoAneu$GENEID)
  #     keep11 <- genesNoAneu$GENEID %in% matchedS11
  #     s11 <- genesNoAneu[keep11,]
  #     test11 <- cbind(s11$GENEID, s11$TXCHROM, sample11notDE)
  #     sample11 <- test11[,2:10]
  #     #
  #     matchedS18 <- intersect(sample18notDE$GENEID, genesNoAneu$GENEID)
  #     keep18 <- genesNoAneu$GENEID %in% matchedS18
  #     s18 <- genesNoAneu[keep18,]
  #     test18 <- cbind(s18$GENEID, s18$TXCHROM, sample18notDE)
  #     sample18 <- test18[,2:10]
  #     #
  #     matchedS59 <- intersect(sample59notDE$GENEID, genesNoAneu$GENEID)
  #     keep59 <- genesNoAneu$GENEID %in% matchedS59
  #     s59 <- genesNoAneu[keep59,]
  #     test59 <- cbind(s59$GENEID, s59$TXCHROM, sample59notDE)
  #     sample59 <- test59[,2:10]
  #     # add a column name to the chromosome column
  #     colnames(sample4)[1] <- "chr"
  #     #
  #     colnames(sample49)[1] <- "chr"
  #     #
  #     colnames(sample7)[1] <- "chr"
  #     #
  #     colnames(sample11)[1] <- "chr"
  #     #
  #     colnames(sample18)[1] <- "chr"
  #     #
  #     colnames(sample59)[1] <- "chr"
  #     # take out all chromosomes except the aneuploid one(s)
  #       sample4notDE <- subset(sample4, chr=="chrV")
  #       #
  #       sample49notDE <- subset(sample49, chr=="chrV")
  #       #
  #       sample7notDE <- subset(sample7, chr=="chrI")
  #       #
  #       sample11notDE <- subset(sample11, chr=="chrI")
  #       #
  #       sample18notDE <- subset(sample18, chr=="chrI")
  #       # count how many genes are not DE on this chrom in this sample
  #         countNotDeSample4 <- nrow(sample4notDE)
  #         #
  #         countNotDeSample49 <- nrow(sample49notDE)
  #         #
  #         countNotDeSample7 <- nrow(sample7notDE)
  #         #
  #         countNotDeSample11 <- nrow(sample11notDE)
  #         #
  #         countNotDeSample18 <- nrow(sample18notDE)
  #         # find how many genes are analyzed total on this chromosome
  #         chrVgenes <- subset(genesNoAneu, TXCHROM=="chrV")
  #         countChrV <- nrow(chrVgenes)
  #         #
  #         chrIgenes <- subset(genesNoAneu, TXCHROM=="chrI")
  #         countChrI <- nrow(chrIgenes)
  #         
  #           # what percent of genes on chromosome V are not DE in sample 4?
  #             percNotDeS4 = countNotDeSample4/countChrV * 100
  #             # 49% in sample 4
  #             #
  #             percNotDeS49 = countNotDeSample49/countChrV * 100
  #             # 68% in sample 49
  #             #
  #             percNotDeS7 = countNotDeSample7/countChrI * 100
  #             # 77% in sample 7 for chr I
  #             #
  #             percNotDeS11 = countNotDeSample11/countChrI * 100
  #             # 39% not differentially expressed in chr I for line 11
  #             #
  #             percNotDeS18 = countNotDeSample18/countChrI * 100
              # 65% not DE on chr I for line 18

#####################################################################################
# make  this into a function
# want to know how many genes are NOT DE on the aneuploid chromosome in aneuploid samples
# #
# sample = res4Dat
# chrom = "chrV"

getPercNonDEgenesGC <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
samplenotDE <- subset(sample, padj>0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(samplenotDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, samplenotDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        notDEAneuChr <- subset(samp, chr==chrom)
      # count how many genes are not DE on this chrom in this sample
          countNotDeSample <- nrow(notDEAneuChr)
         # find how many genes are analyzed total on this chromosome
          chrgenes <- subset(genesGC, TXCHROM==chrom)
          countChr <- nrow(chrgenes)     
            # what percent of genes on chromosome V are not DE in sample 4?
              percNotDeS <- setNames(data.frame(matrix(ncol = 3, nrow = 1)), c("Sample", "Chromosome", "percentNotDE"))
              percNotDeS$Sample <- sample[1,8]
              percNotDeS$Chromosome <- chrom
              percNotDeS$percentNotDE <- countNotDeSample/countChr * 100
              # percNotDeS <- c(sample[1,8], chrom, percNotDeS)
              # percNotDeS =
              # # 49% in sample 4
              # percNotDeS <- as.data.frame(percNotDeS)
              # colnames(percNotDeS) <- c("Sample","Chromosome","percentNotDE")
return(percNotDeS)
}

# test it out on one of the ones I did already
# try S4
percNonDEGCS4c5 <- getPercNonDEgenesGC(resGC4Dat, "chrV")
percNonDEGCS7c1 <- getPercNonDEgenesGC(resGC7Dat, "chrI")
percNonDEGCS8c16 <- getPercNonDEgenesGC(resGC8Dat, "chrXVI")
percNonDEGCS11c1 <- getPercNonDEgenesGC(resGC11Dat, "chrI")
percNonDEGCS11c15 <- getPercNonDEgenesGC(resGC11Dat, "chrXV")
percNonDEGCS18c1 <- getPercNonDEgenesGC(resGC18Dat, "chrI")
percNonDEGCS18c12 <- getPercNonDEgenesGC(resGC18Dat, "chrXII")
# percNonDEGCS49c5 <- getPercNonDEgenesGC(res49Dat, "chrV")
percNonDEGCS59c7 <- getPercNonDEgenesGC(resGC59Dat, "chrVII")
percNonDEGCS61c7 <- getPercNonDEgenesGC(resGC61Dat, "chrVII")
percNonDEGCS76c9 <- getPercNonDEgenesGC(resGC76Dat, "chrIX")
percNonDEGCS76c14 <- getPercNonDEgenesGC(resGC76Dat, "chrXIV")
percNonDEGCS77c12 <- getPercNonDEgenesGC(resGC77Dat, "chrXII")

percNonDEAllGCSamples <- rbind(percNonDEGCS4c5,percNonDEGCS7c1, percNonDEGCS8c16, percNonDEGCS11c1, percNonDEGCS11c15, percNonDEGCS18c1, percNonDEGCS18c12, percNonDEGCS59c7, percNonDEGCS61c7, percNonDEGCS76c9, percNonDEGCS76c14, percNonDEGCS77c12)

write.csv(percNonDEAllGCSamples, file="percNonDEAllgcLines.csv")

#########################################################################################

getNumNonDEGCgenes <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
samplenotDE <- subset(sample, padj>0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(samplenotDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, samplenotDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        notDEAneuChr <- subset(samp, chr==chrom)
      # count how many genes are not DE on this chrom in this sample
          countNotDeSample <- nrow(notDEAneuChr)
         # find how many genes are analyzed total on this chromosome
          chrgenes <- subset(genesGC, TXCHROM==chrom)
          countChr <- nrow(chrgenes)     
            # what percent of genes on chromosome V are not DE in sample 4?
              numNotDeS <- setNames(data.frame(matrix(ncol = 2, nrow = 1)), c("Sample", "Chromosome"))
              numNotDeS$Sample <- sample[1,8]
              numNotDeS$Chromosome <- chrom
              numNotDeS$countNotDe <- countNotDeSample
            numNotDeS$countChrom <- countChr
              # percNotDeS <- c(sample[1,8], chrom, percNotDeS)
              # percNotDeS =
              # # 49% in sample 4
              # percNotDeS <- as.data.frame(percNotDeS)
              # colnames(percNotDeS) <- c("Sample","Chromosome","percentNotDE")
return(numNotDeS)
}

numNonDEGCS4c5 <- getNumNonDEGCgenes(resGC4Dat, "chrV")
numNonDEGCS7c1 <- getNumNonDEGCgenes(resGC7Dat, "chrI")
numNonDEGCS8c16 <- getNumNonDEGCgenes(resGC8Dat, "chrXVI")
numNonDEGCS11c1 <- getNumNonDEGCgenes(resGC11Dat, "chrI")
numNonDEGCS11c15 <- getNumNonDEGCgenes(resGC11Dat, "chrXV")
numNonDEGCS18c1 <- getNumNonDEGCgenes(resGC18Dat, "chrI")
numNonDEGCS18c12 <- getNumNonDEGCgenes(resGC18Dat, "chrXII")
# numNonDEGCS49c5 <- getNumNonDEGCgenes(resGC49Dat, "chrV")
numNonDEGCS59c7 <- getNumNonDEGCgenes(resGC59Dat, "chrVII")
numNonDEGCS61c7 <- getNumNonDEGCgenes(resGC61Dat, "chrVII")
numNonDEGCS76c9 <- getNumNonDEGCgenes(resGC76Dat, "chrIX")
numNonDEGCS76c14 <- getNumNonDEGCgenes(resGC76Dat, "chrXIV")
numNonDEGCS77c12 <- getNumNonDEGCgenes(resGC77Dat, "chrXII")

numNonDEAllGCSamples <- rbind(numNonDEGCS4c5,numNonDEGCS7c1, numNonDEGCS8c16, numNonDEGCS11c1, numNonDEGCS11c15, numNonDEGCS18c1, numNonDEGCS18c12, numNonDEGCS59c7, numNonDEGCS61c7, numNonDEGCS76c9, numNonDEGCS76c14, numNonDEGCS77c12)

write.csv(numNonDEAllGCSamples, file="numNonDEAllgcLines.csv")







#######################################################################################
# want to know WHAT genes are NOT DE on the aneuploid chromosome in aneuploid samples
#
sample = resGC4Dat
chrom = "chrV"

getNonDEGCgenes <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
samplenotDE <- subset(sample, padj>0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(samplenotDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, samplenotDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        notDEAneuChr <- subset(samp, chr==chrom)
return(notDEAneuChr)
}

# test it out on one of the ones I did already
# try S4
nonDEGCS4c5 <- getNonDEGCgenes(resGC4Dat, "chrV")
write.csv(nonDEGCS4c5, file="nonDEGCS4c5.csv")
nonDEGCS7c1 <- getNonDEGCgenes(resGC7Dat, "chrI")
write.csv(nonDEGCS7c1, file="nonDEGCS7c1.csv")
nonDEGCS8c16 <- getNonDEGCgenes(resGC8Dat, "chrXVI")
write.csv(nonDEGCS8c16, file="nonDEGCS8c16.csv")
nonDEGCS11c1 <- getNonDEGCgenes(resGC11Dat, "chrI")
write.csv(nonDEGCS11c1, file="nonDEGCS11c1.csv")
nonDEGCS11c15 <- getNonDEGCgenes(resGC11Dat, "chrXV")
write.csv(nonDEGCS11c15, file="nonDEGCS11c15.csv")
nonDEGCS18c1 <- getNonDEGCgenes(resGC18Dat, "chrI")
write.csv(nonDEGCS18c1, file="nonDEGCS18c1.csv")
nonDEGCS18c12 <- getNonDEGCgenes(resGC18Dat, "chrXII")
write.csv(nonDEGCS18c12, file="nonDEGCS18c12.csv")
# nonDEGCS49c5 <- getNonDEGCgenes(resGC49Dat, "chrV")
# write.csv(nonDES49c5, file="nonDES49c5.csv")
nonDEGCS59c7 <- getNonDEGCgenes(resGC59Dat, "chrVII")
write.csv(nonDEGCS59c7, file="nonDEGCS59c7.csv")
nonDEGCS61c7 <- getNonDEGCgenes(resGC61Dat, "chrVII")
write.csv(nonDEGCS61c7, file="nonDEGCS61c7.csv")
nonDEGCS76c9 <- getNonDEGCgenes(resGC76Dat, "chrIX")
write.csv(nonDEGCS76c9, file="nonDEGCS76c9.csv")
nonDEGCS76c14 <- getNonDEGCgenes(resGC76Dat, "chrXIV")
write.csv(nonDEGCS76c14, file="nonDEGCS76c14.csv")
nonDEGCS77c12 <- getNonDEGCgenes(resGC77Dat, "chrXII")
write.csv(nonDEGCS77c12, file="nonDEGCS77c12.csv")

############################################################################
# want to know how many genes are DE on the aneuploid chromosome in aneuploid samples
# #
# sample = res4Dat
# chrom = "chrV"

getPercDEGCgenes <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        DEAneuChr <- subset(samp, chr==chrom)
      # count how many genes are DE on this chrom in this sample
          countDeSample <- nrow(DEAneuChr)
         # find how many genes are analyzed total on this chromosome
          chrgenes <- subset(genesGC, TXCHROM==chrom)
          countChr <- nrow(chrgenes)     
            # what percent of genes on chromosome V are DE in sample 4?
              # percDeS = countDeSample/countChr * 100
              # # 49% in sample 4
              # percDeS <- c(sample[1,8], chrom, percDeS)
              # percDeS <- as.data.frame(percDeS)
              # colnames(percDeS) <- c("Sample","Chromosome","percentNotDE")
              percDeS <- setNames(data.frame(matrix(ncol = 3, nrow = 1)), c("Sample", "Chromosome", "percentNotDE"))
              percDeS$Sample <- sample[1,8]
              percDeS$Chromosome <- chrom
              percDeS$percentNotDE <- countDeSample/countChr * 100

return(percDeS)
}

# test it out on one of the ones I did already
# try S4
percDEGCS4c5 <- getPercDEGCgenes(resGC4Dat, "chrV")
percDEGCS7c1 <- getPercDEGCgenes(resGC7Dat, "chrI")
percDEGCS8c16 <- getPercDEGCgenes(resGC8Dat, "chrXVI")
percDEGCS11c1 <- getPercDEGCgenes(resGC11Dat, "chrI")
percDEGCS11c15 <- getPercDEGCgenes(resGC11Dat, "chrXV")
percDEGCS18c1 <- getPercDEGCgenes(resGC18Dat, "chrI")
percDEGCS18c12 <- getPercDEGCgenes(resGC18Dat, "chrXII")
# percDEGCS49c5 <- getPercDEGCgenes(resGC49Dat, "chrV")
percDEGCS59c7 <- getPercDEGCgenes(resGC59Dat, "chrVII")
percDEGCS61c7 <- getPercDEGCgenes(resGC61Dat, "chrVII")
percDEGCS76c9 <- getPercDEGCgenes(resGC76Dat, "chrIX")
percDEGCS76c14 <- getPercDEGCgenes(resGC76Dat, "chrXIV")
percDEGCS77c12 <- getPercDEGCgenes(resGC77Dat, "chrXII")

percDEAllGCSamples <- rbind(percDEGCS4c5,percDEGCS7c1, percDEGCS8c16, percDEGCS11c1, percDEGCS11c15, percDEGCS18c1, percDEGCS18c12, percDEGCS59c7, percDEGCS61c7, percDEGCS76c9, percDEGCS76c14, percDEGCS77c12)

write.csv(percDEAllGCSamples, file="percDEAllgcLines.csv")


###################################################################

#######################################################################################
# want to know WHAT genes are DE on the aneuploid chromosome in aneuploid samples
#
# sample = res4Dat
# chrom = "chrV"
getDEGCgenes <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        DEAneuChr <- subset(samp, chr==chrom)
return(DEAneuChr)
}


# test it out on one of the ones I did already
# try S4
DEGCS4.c5 <- getDEGCgenes(resGC4Dat, "chrV")


write.csv(DEGCS4.c5, file="DEGCS4c5.csv")
# View(geneIDs4)
geneIDsGC4 <- DEGCS4.c5[,2]
str(geneIDsGC4)
# geneIDs4 <- geneIDs4$GENEID
geneIDsGC4 <- as.vector(geneIDsGC4)
write.table(geneIDsGC4, file="DEgeneIDs4GC.txt")

DESGC7.c1 <- getDEGCgenes(resGC7Dat, "chrI")
write.csv(DESGC7.c1, file="DESGC7c1.csv")

DESGC8.c16 <- getDEGCgenes(resGC8Dat, "chrXVI")
write.csv(DESGC8.c16, file="DESGC8c16.csv")

DESGC11.c1 <- getDEGCgenes(resGC11Dat, "chrI")
write.csv(DESGC11.c1, file="DESGC11c1.csv")

DESGC11.c15 <- getDEGCgenes(resGC11Dat, "chrXV")
write.csv(DESGC11.c15, file="DESGC11c15.csv")

DESGC18.c1 <- getDEGCgenes(resGC18Dat, "chrI")
write.csv(DESGC18.c1, file="DESGC18c1.csv")

DESGC18.c12 <- getDEGCgenes(resGC18Dat, "chrXII")
write.csv(DESGC18.c12, file="DESGC18c12.csv")

# DESGC49.c5 <- getDEGCgenes(resGC49Dat, "chrV")
# write.csv(DES49.c5, file="DES18c5.csv")

DESGC59.c7 <- getDEGCgenes(resGC59Dat, "chrVII")
write.csv(DESGC59.c7, file="DESGC59c7.csv")

DESGC61.c7 <- getDEGCgenes(resGC61Dat, "chrVII")
write.csv(DESGC61.c7, file="DESGC61c7.csv")

DESGC76.c9 <- getDEGCgenes(resGC76Dat, "chrIX")
write.csv(DESGC76.c9, file="DESGC76c9.csv")

DESGC76.c14 <- getDEGCgenes(resGC76Dat, "chrXIV")
write.csv(DESGC76.c14, file="DESGC76c14.csv")

DESGC77.c12 <- getDEGCgenes(resGC77Dat, "chrXII")
write.csv(DESGC77.c12, file="DESGC77c12.csv")

################################################################
# want to get DE genes in aneuploid lines that are NOT located on the aneuploid chromosome
# sample = resGC4Dat
# chrom= "chrV"

getDEgenesGCElsewhere <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        DENotAneuChr <- subset(samp, chr!=chrom)
return(DENotAneuChr)
}

deGenesNotAneuGCS4 <- getDEgenesGCElsewhere(resGC4Dat, chrom="chrV")
deGenesNotAneuGCS7 <- getDEgenesGCElsewhere(resGC7Dat, chrom="chrI")
deGenesNotAneuGCS8 <- getDEgenesGCElsewhere(resGC8Dat, chrom="chrXVI")
deGenesNotAneuGCS18 <- getDEgenesGCElsewhere(resGC18Dat, chrom=c("chrXII","chrI"))
# deGenesNotAneuGCS49 <- getDEgenesGCElsewhere(resGC49Dat, chrom="chrV")
deGenesNotAneuGCS59 <- getDEgenesGCElsewhere(resGC59Dat, chrom="chrVII")
deGenesNotAneuGCS61 <- getDEgenesGCElsewhere(resGC61Dat, chrom="chrVII")
deGenesNotAneuGCS76 <- getDEgenesGCElsewhere(resGC76Dat, chrom=c("chrIX","XIV"))
deGenesNotAneuGCS77 <- getDEgenesGCElsewhere(resGC77Dat, chrom="chrXII")
deGenesNotAneuGCS11 <- getDEgenesGCElsewhere(resGC11Dat, chrom=c("chrI","chrXV"))

genesGC1DE <- subset(allGCSamplesDiffExpr, Sample=="1")
        write.table(genesGC1DE, file="genesGC1DE.txt")
        genesGC2DE <- subset(allGCSamplesDiffExpr, Sample=="2")
        write.table(genesGC2DE, file="genesGC2DE.txt")
        genesGC3DE <- subset(allGCSamplesDiffExpr, Sample=="3")
        write.table(genesGC3DE, file="genesGC3DE.txt")
        genesGC4DE <- subset(allGCSamplesDiffExpr, Sample=="4")
        write.table(genesGC4DE, file="genesGC4DE.txt")
        genesGC5DE <- subset(allGCSamplesDiffExpr, Sample=="5")
        write.table(genesGC5DE, file="genesGC5DE.txt")
        genesGC7DE <- subset(allGCSamplesDiffExpr, Sample=="7")
        write.table(genesGC7DE, file="genesGC7DE.txt")
        genesGC8DE <- subset(allGCSamplesDiffExpr, Sample=="8")
        write.table(genesGC8DE, file="genesGC8DE.txt")
        genesGC9DE <- subset(allGCSamplesDiffExpr, Sample=="9")
        write.table(genesGC9DE, file="genesGC9DE.txt")
        genesGC11DE <- subset(allGCSamplesDiffExpr, Sample=="11")
        write.table(genesGC11DE, file="genesGC11DE.txt")
        genesGC18DE <- subset(allGCSamplesDiffExpr, Sample=="18")
        write.table(genesGC18DE, file="genesGC18DE.txt")
        # genes49DE <- subset(allSamplesDiffExpr, Sample=="49")
        # write.table(genes49DE, file="genes49DE.txt")
        genesGC59DE <- subset(allGCSamplesDiffExpr, Sample=="59")
        write.table(genesGC59DE, file="genesGC59DE.txt")
        genesGC61DE <- subset(allGCSamplesDiffExpr, Sample=="61")
        write.table(genesGC61DE, file="genesGC61DE.txt")
        genesGC76DE <- subset(allGCSamplesDiffExpr, Sample=="76")
        write.table(genesGC76DE, file="genesGC76DE.txt")
        genesGC77DE <- subset(allGCSamplesDiffExpr, Sample=="77")
        write.table(genesGC77DE, file="genesGC77DE.txt")
      # trying different things to make a venn diagram
        library(VennDiagram)
          # pdf("venn_diagram.pdf")
          # venn.plot <- venn.diagram(list(genes1DE$GENEID, genes2DE$GENEID,genes3DE$GENEID,genes4DE$GENEID, genes5DE$GENEID, genes7DE$GENEID, genes8DE$GENEID, genes9DE$GENEID, genes11DE$GENEID, genes18DE$GENEID, genes49DE$GENEID, genes59DE$GENEID, genes61DE$GENEID, genes76DE$GENEID, genes77DE$GENEID), NULL, fill=c("red", "green","blue","pink","orange","turquoise","yellow","purple","lightblue","lightyellow","lightred","lightgreen","lightpink","lightorange"), alpha=c(0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5), cex = 2, cat.fontface=3, category.names=c("Sample1", "Sample2","Sample3", "Sample4","Sample5","Sample7","Sample8","Sample9","Sample11","Sample18","Sample49","Sample59","Sample61","Sample76","Sample77"))
          # grid.draw(venn.plot)
          # dev.off()


venn.plotGCAneu <- venn.diagram(
	x = list(
		A = deGenesNotAneuGCS4$GENEID,
		B = deGenesNotAneuGCS7$GENEID,
		C = deGenesNotAneuGCS8$GENEID,
		D = deGenesNotAneuGCS59$GENEID,
		E = deGenesNotAneuGCS77$GENEID
		),
	filename = NULL, #"euploid_commonDE.tiff"
	col = "black",
	fill = c("lightblue", "goldenrod1", "lavender", "turquoise", "lightpink"),
	category.names = c("Chr V 3n","Chr I 3n","Chr XVI 4n","Chr VII 3n","Chr XII 3n"),
	alpha = 0.50,
	cex = c(1.5, 1.5, 1.5, 1.5, 1.5, 1, 0.8, 1, 0.8, 1, 0.8, 1, 0.8,
	 1, 0.8, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 1, 1, 1, 1, 1.5),
	cat.col = c("mediumblue", "darkgoldenrod", "purple", "darkturquoise", "deeppink"),
	cat.cex = 1,
	cat.default.pos ="outer",
	cat.fontface = "bold",
	main="Common DE Genes Not on Aneuploid Chromosome in Aneuploid Samples",
	main.fontface = "bold",
	main.cex = 1,
	margin = 0.05
	)

grid.draw(venn.plotGCAneu)

# venn.plotGCChrV <- venn.diagram(
# 	x = list(
# 		A = deGenesNotAneuGCS4$GENEID,
# 		B = deGenesNotAneuGCS49$GENEID
# 		),
# 	filename = NULL, #"euploid_commonDE.tiff",
# 	fill = c("turquoise", "lightpink")
# 	)
#
# pdf("commonDE_chrV.pdf")
# grid.draw(venn.plotChrV)

pdf("aneuploid_commonDE_GC.pdf")
grid.draw(venn.plotGCAneu)
dev.off()
# aneuLines <- list(genes4DE, genes7DE, genes8DE, genes18DE, genes49DE)
# want to get list of common DE genes in all the euploid lines
#commonDEaneu <- Reduce(intersect, Map("[[", aneuLines, "GENEID"))
#Reduc
#commonDEaneu <- as.vector(commonDEaneu)
#write.csv(commonDEaneu, file="commonDEaneu.csv")

venn.plotGCChrI <- venn.diagram(
	x = list(
		A = deGenesNotAneuGCS7$GENEID,
		B = deGenesNotAneuGCS18$GENEID,
		C = deGenesNotAneuGCS11$GENEID
		),
	filename = NULL,
	fill = c("orange", "deeppink1","turquoise"),
	cex=2,
	cat.col = c("orange", "deeppink1", "darkturquoise"),
	category.names = c("Sample 7 (3n)","Sample 18 (3n)","Sample 11 (1n)"),
	cat.cex = 1,
	cat.default.pos ="outer",
	cat.fontface = "bold",
	main= "Common DE Genes Not on Chromosome 1 in Aneuploid Samples",
	main.fontface = "bold",
	main.cex = 1.5,
	margin = 0.05
	)

pdf("commonGC.DE_chrI.pdf")
grid.draw(venn.plotGCChrI)

venn.plotGCChrXII <- venn.diagram(
	x = list(
		A = deGenesNotAneuGCS18$GENEID,
		B = deGenesNotAneuGCS77$GENEID
		),
	filename = NULL,
	fill = c("orange", "deeppink1"),
	category.names = c("Sample 18 (3n)","Sample 77 (3n)"),
	cat.cex = 1,
	cat.col = c("orange","deeppink"),
	main= "Common DE Genes Not on Chrom XII in Aneuploid Samples ",
	main.fontface = "bold",
	main.cex = 1.5,
	margin=0.05
	)

pdf("commonGC.DE_chrVXII.pdf")
grid.draw(venn.plotGCChrXII)

venn.plotGCChrVII <- venn.diagram(
	x = list(
		A = deGenesNotAneuGCS59$GENEID,
		B = deGenesNotAneuGCS61$GENEID
		),
	filename = NULL,
	fill = c("turquoise", "purple"),
	category.names = c("Sample 59 (3n)","Sample 61 (3n)"),
	cat.cex = 1,
	cat.col=c("darkturquoise","purple"),
	main="Common DE Genes Not on Chrom VII in Aneuploid Samples",
	main.fontface = "bold",
	main.cex = 1.5,
	margin=0.05
	)

pdf("commonGC.DE_chrVII.pdf")
grid.draw(venn.plotGCChrVII)


############################################################################
# want to know what genes are DE in all samples
# getDEgenes <- function(sample) {              
#   # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
# allSampleDE <- subset(sample, padj<0.1)
# # then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
#     #
#     # add column with chromosome numbers matching the GENEIDs
#       # find matches of GENEID in sample file to GENEID in list of genes included in analysis
#       matched <- intersect(sampleDE$GENEID, genesNoAneu$GENEID)
#       # tell R which genes to keep
#       keep <- genesNoAneu$GENEID %in% matched
#       # subset list of genes in analysis to include only the ones matching those found in          sample file
#       s <- genesNoAneu[keep,]
#       # use c(olumn)bind to stick the chromosome #s to the sample analysis file
#       test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
#       # subset to exclude first column (duplicate)
#       samp <- test[,2:10]
#       # add a column name to the chromosome column
#       colnames(samp)[1] <- "chr"
# return(deAll)
# }



############################################################################
# want to know how many genes are DE on the aneuploid chromosome in multiple aneuploid samples sharing the same aneuploid
#
# chr I
  # lines 7,11,and 18
  # deChrI <-

#######################################################################################
# want to know WHAT genes are DE on the aneuploid chromosome in aneuploid samples
#
# sample = res4Dat
#chrom = "chrV"

getDEgenesGCAll <- function(sample) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   

sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        DEAll  <- samp
return(DEAll)
}

# test it out on one of the ones I did already
# try S4
DEGCS1all <- getDEgenesGCAll(resGC1Dat)
DEGCS2all <- getDEgenesGCAll(resGC2Dat)
DEGCS3all <- getDEgenesGCAll(resGC3Dat)
DEGCS4all <- getDEgenesGCAll(resGC4Dat)
DEGCS5all <- getDEgenesGCAll(resGC5Dat)
DEGCS7all <- getDEgenesGCAll(resGC7Dat)
DEGCS8all <- getDEgenesGCAll(resGC8Dat)
DEGCS9all <- getDEgenesGCAll(resGC9Dat)
DEGCS11all <- getDEgenesGCAll(resGC11Dat)
DEGCS18all <- getDEgenesGCAll(resGC18Dat)
DEGCS59all <- getDEgenesGCAll(resGC59Dat)
DEGCS61all <- getDEgenesGCAll(resGC61Dat)
DEGCS76all <- getDEgenesGCAll(resGC76Dat)
DEGCS77all <- getDEgenesGCAll(resGC77Dat)

########################################################################################
# what DE genes are shared between samples aneuploid with the same chromosome?
  # use data frames containing only DE genes in the given samples
# sample1 = genes4DE
# sample2 = genes49DE
library(dplyr)

getCommonDEGCgenes <- function(sample1, sample2) {
  # join them together by GENEID column
  chr.all <- inner_join(sample1, sample2, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matched <- intersect(chr.all$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepAll <- chr.all$GENEID %in% matched
          # only keep those genes in the data frame
          chr.all <- chr.all[keepAll,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenAll <- genesGC$GENEID %in% matched
              # put those genes in a vector
              GenAll <- genesGC[keepGenAll,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chr.all <- cbind(GenAll$TXCHROM, chr.all)
                return(chr.all)
}

# chrVAll <- getCommonDEGCgenes(genesGC4DE, genes49DE)
# write.csv(chrVAll, file="chrVallDE_GC.csv")
chrVIIAllGC <- getCommonDEGCgenes(genesGC59DE, genesGC61DE)
write.csv(chrVIIAllGC, file="chrVIIallDE_GC.csv")
chrXIIAllGC <- getCommonDEGCgenes(genesGC18DE, genesGC77DE)
write.csv(chrXIIAllGC, file="chrXIIallDE_GC.csv")

## only trisomic samples
    # # join them together by GENEID column
    # chrI.tri <- inner_join(genes7DE,genes18DE, by="GENEID")
    #   # match the genes to the genes in genes table that includes chromosome number
    #   matchedI <- intersect(chrI.tri$GENEID, genesNoAneu$GENEID)
    #     # tell R which genes to keep from the chrI.tri dataframe
    #     keepI <- chrI.tri$GENEID %in% matchedI
    #       # only keep those genes in the data frame
    #       chrI.tri <- chrI.tri[keepI,]
    #         # tell R which genes to keep from the genesNoAneu dataframe
    #         keepGen <- genesNoAneu$GENEID %in% matchedI
    #           # put those genes in a vector
    #           Gen <- genesNoAneu[keepGen,]
    #             # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
    #             chrI.tri <- cbind(Gen$TXCHROM, chrI.tri)
    #               # save as .csv file
    #               write.csv(chrI.tri, file="chrITriDE.csv")

#for chromosome I, which has 3 lines that are commonly aneuploid for it, so the function won't work properly
#chrom I, all aneuploids (both trisomics plus the monosomic)
# chrI.all <- inner_join(chrI.tri, genes11DE, by="GENEID")
#   # match the genes to the genes in genes table that includes chromosome number
#   matchedIall <- intersect(chrI.all$GENEID, genesNoAneu$GENEID)
#     # tell R which genes to keep from the chrI.tri dataframe
#     keepIall <- chrI.all$GENEID %in% matchedI
#       # only keep those genes in the data frame
#       chrI.all <- chrI.all[keepIall,]
#         # tell R which genes to keep from the genesNoAneu dataframe
#         keepGenAll <- genesNoAneu$GENEID %in% matchedIall
#           # put those genes in a vector
#           GenAll <- genesNoAneu[keepGenAll,]
#             # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
#             chrI.all <- cbind(GenAll$TXCHROM, chrI.all)
#               # save as .csv file
#               write.csv(chrI.all, file="chrIAllDE.csv")

########################################################################################
# what NON-DE genes are shared between samples aneuploid with the same chromosome?
  # use data frames containing only NON-DE genes in the given samples
 # sample1 = nonDES4c5
 # sample2 = nonDES49c5
 #
getCommonNonDEGCgenes <- function(sample1, sample2) {
  # join them together by GENEID column
  chr.all <- inner_join(sample1, sample2, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matched <- intersect(chr.all$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepAll <- chr.all$GENEID %in% matched
          # only keep those genes in the data frame
          chr.all <- chr.all[keepAll,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenAll <- genesGC$GENEID %in% matched
              # put those genes in a vector
              GenAll <- genesGC[keepGenAll,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chr.all <- cbind(GenAll$TXCHROM, chr.all)
                return(chr.all)
}

# chrVNon <- getCommonNonDEgenes(nonDES4c5, nonDES49c5)
# write.csv(chrVNon, file="chrVallDNonE_GC.csv")
chrVIIGCNon <- getCommonNonDEGCgenes(nonDEGCS59c7, nonDEGCS61c7)
write.csv(chrVIIGCNon, file="chrVIIallNonDE_GC.csv")
chrXIIGCNon <- getCommonNonDEGCgenes(nonDEGCS18c12, nonDEGCS77c12)
write.csv(chrXIIGCNon, file="chrXIIallNonDE_GC.csv")

## only trisomic samples
    # join them together by GENEID column
    chrItriGCNon <- inner_join(nonDEGCS7c1,nonDEGCS18c1, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matchedIGCNon <- intersect(chrItriGCNon$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepIGCNon <- chrItriGCNon$GENEID %in% matchedIGCNon
          # only keep those genes in the data frame
          chrItriGCNon <- chrItriGCNon[keepIGCNon,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenGCNon <- genesGC$GENEID %in% matchedIGCNon
              # put those genes in a vector
              GenNonGC <- genesGC[keepGenGCNon,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chrItriGCNon <- cbind(GenNonGC$TXCHROM, chrItriGCNon)
                  # save as .csv file
                  write.csv(chrItriGCNon, file="chrITriNonDE_GC.csv")
#for chromosome I, which has 3 lines that are commonly aneuploid for it, so the function won't work properly
  # chrom I, all aneuploids (both trisomics plus the monosomic)
    chrI.NonGC <- inner_join(chrItriGCNon, nonDEGCS11c1, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matchedIallGCNon <- intersect(chrI.NonGC$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepIallGCNon <- chrI.NonGC$GENEID %in% matchedIallGCNon
          # only keep those genes in the data frame
          chrI.NonGC <- chrI.NonGC[keepIallGCNon,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenAllNonGC <- genesGC$GENEID %in% matchedIallGCNon
              # put those genes in a vector
              GenAllNonGC <- genesGC[keepGenAllNonGC,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chrI.NonGC <- cbind(GenAllNonGC$TXCHROM, chrI.NonGC)
                  # save as .csv file
                  write.csv(chrI.NonGC, file="chrIAllNonDE_GC.csv")

{
  venn.plotChrINonDEGC <- venn.diagram(
	x = list(
		"L11" = nonDEGCS11c1$GENEID,
		"L18" =nonDEGCS18c1$GENEID,
		"L7" = nonDEGCS7c1$GENEID
		),
	filename = NULL,
	fill = c("orange", "deeppink1","turquoise"),
	cex=2
	)
pdf("commonNonDE_chrI_GC.pdf")
grid.draw(venn.plotChrINonDEGC)
dev.off()
}


{venn.plotChrINonDETrisGC <- venn.diagram(
	x = list(
		"L18" =nonDEGCS18c1$GENEID,
		"L7" = nonDEGCS7c1$GENEID
		),
	filename = NULL,
	fill = c( "deeppink1","turquoise"),
	cex=2
	)
pdf("commonNonDE_chrITri_GC.pdf")
grid.draw(venn.plotChrINonDETrisGC)
dev.off()}

# venn.plotChrVNonDE <- venn.diagram(
# 	x = list(
# 		"L4" =nonDES4c5$GENEID,
# 		"L49" = nonDES49c5$GENEID
# 		),
# 	filename = NULL,
# 	fill = c( "goldenrod","turquoise"),
# 	cex=2
# 	)
# pdf("commonNonDE_chrVTri.pdf")
# grid.draw(venn.plotChrVNonDE)
# dev.off()

# then compare how many of these genes are shared between samples aneuploid for the same chromosome       #
# want to know WHAT genes are not DE - GO analysis?           

# put this information into a venn diagram
  #
  # subset the allSamplesDiffExpr data frame into vectors by sample with gene names

# A <- sample(1:1000, 400, replace = FALSE);
# B <- sample(1:1000, 600, replace = FALSE);
# C <- sample(1:1000, 350, replace = FALSE);
# D <- sample(1:1000, 550, replace = FALSE);
# E <- sample(1:1000, 375, replace = FALSE);

{
  venn.plot.EuGC <- venn.diagram(
	x = list(
		A = genesGC1DE$GENEID,
		B = genesGC2DE$GENEID,
		C = genesGC3DE$GENEID,
		D = genesGC5DE$GENEID,
		E = genesGC9DE$GENEID
		),
	filename = NULL, #"euploid_commonDE.tiff"
	col = "black",
	fill = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
	alpha = 0.50,
	cex = c(1.5, 1.5, 1.5, 1.5, 1.5, 1, 0.8, 1, 0.8, 1, 0.8, 1, 0.8,
	 1, 0.8, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 1, 1, 1, 1, 1.5),
	cat.col = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
	cat.cex = 1.5,
	cat.fontface = "bold",
	margin = 0.05
	)
pdf("euploid_commonDE.pdf")
grid.draw(venn.plot.EuGC)
dev.off()
}

eupLinesGC <- list(genesGC1DE, genesGC2DE, genesGC3DE, genesGC5DE, genesGC9DE)
# want to get list of common DE genes in all the euploid lines
commonDEeupGC <- Reduce(intersect, Map("[[", eupLinesGC, "GENEID"))
#Reduc
commonDEeupGC <- as.vector(commonDEeupGC)
commonDEeupGC <- as.data.frame(commonDEeupGC)
colnames(commonDEeupGC) <- c("names")
write.csv(commonDEeupGC, file="commonDEeup_GC.csv")

# what genes are commonly DE in aneuploid lines, but NOT located on the aneuploid chromosomes?
# genes11DE,
aneuLinesGC <- 	list(deGenesNotAneuGCS4, deGenesNotAneuGCS7, deGenesNotAneuGCS8,deGenesNotAneuGCS11, deGenesNotAneuGCS18,
                     deGenesNotAneuGCS59, deGenesNotAneuGCS61, deGenesNotAneuGCS76, deGenesNotAneuGCS77)
# want to get list of common DE genes in all the euploid lines
commonDEaneuGC <- Reduce(intersect, Map("[[", aneuLinesGC, "GENEID"))
commonDEeupGC <- as.vector(commonDEeupGC)
write.csv(commonDEeupGC, file="commonDEeup_GC.csv")

# aneuLines4GC <- 	list(deGenesNotAneuGCS4, deGenesNotAneuGCS7, deGenesNotAneuGCS8,deGenesNotAneuGCS59)
# # want to get list of common DE genes in all the euploid lines
# commonDEaneu4 <- Reduce(intersect, Map("[[", aneuLines4, "GENEID"))
#
# aneuLines3 <- 	list(deGenesNotAneuS4, deGenesNotAneuS7, deGenesNotAneuS8)
# # want to get list of common DE genes in all the euploid lines
# commonDEaneu3 <- Reduce(intersect, Map("[[", aneuLines3, "GENEID"))




# later want to ask how many genes are DE in only euploid samples or only aneuploid samples
#
  # then query which samples the gene is DE in

```

```{r common DE genes Aneuploids, eval=TRUE,include=TRUE}

# setDT(res4Dat, keep.rownames = TRUE)[]
# setnames(res4Dat, 1, "GENEID")
# genes4DE <- subset(res4Dat,padj<0.1)


# setDT(res7Dat, keep.rownames = TRUE)[]
# setnames(res7Dat, 1, "GENEID")
# genes7DE <- subset(res7Dat,padj<0.1)


# setDT(res8Dat, keep.rownames = TRUE)[]
# setnames(res8Dat, 1, "GENEID")
# genes8DE <- subset(res8Dat,padj<0.1)


# setDT(res11Dat, keep.rownames = TRUE)[]
# setnames(res11Dat, 1, "GENEID")
# genes11DE <- subset(res11Dat,padj<0.1)


# setDT(res18Dat, keep.rownames = TRUE)[]
# setnames(res18Dat, 1, "GENEID")
# genes18DE <- subset(res18Dat,padj<0.1)


# setDT(res49Dat, keep.rownames = TRUE)[]
# setnames(res49Dat, 1, "GENEID")
# genes49DE <- subset(res49Dat,padj<0.1)


# setDT(res59Dat, keep.rownames = TRUE)[]
# setnames(res59Dat, 1, "GENEID")
# genes59DE <- subset(res59Dat,padj<0.1)


# setDT(res61Dat, keep.rownames = TRUE)[]
# setnames(res61Dat, 1, "GENEID")
# genes61DE <- subset(res61Dat,padj<0.1)


# setDT(res76Dat, keep.rownames = TRUE)[]
# setnames(res76Dat, 1, "GENEID")
# genes76DE <- subset(res76Dat,padj<0.1)


# setDT(res77Dat, keep.rownames = TRUE)[]
# setnames(res77Dat, 1, "GENEID")
# genes77DE <- subset(res77Dat,padj<0.1)

#remove genes that are commonly DE in euploid lines
# genes4DE.r <- anti_join(genes4DE, common1235.9,by="GENEID")
# genes7DE.r <- anti_join(genes7DE, common1235.9,by="GENEID")
# genes8DE.r <- anti_join(genes8DE, common1235.9,by="GENEID")
# genes11DE.r <- anti_join(genes11DE, common1235.9,by="GENEID")
# genes18DE.r <- anti_join(genes18DE, common1235.9,by="GENEID")
# genes49DE.r <- anti_join(genes49DE, common1235.9,by="GENEID")
# genes59DE.r <- anti_join(genes59DE, common1235.9,by="GENEID")
# genes61DE.r <- anti_join(genes61DE, common1235.9,by="GENEID")
# genes76DE.r <- anti_join(genes76DE, common1235.9,by="GENEID")
# genes77DE.r <- anti_join(genes77DE, common1235.9,by="GENEID")

#find common DE genes in aneuploids with same chromosome aneuploid
# chrI.tri <- inner_join(genes7DE,genes18DE, by="GENEID")
# matchedI <- intersect(chrI.tri$GENEID, genes$GENEID)
# keepI <- chrI.tri$GENEID %in% matchedI
# chrI.tri <- chrI.tri[keepI,]
# keepGen <- genes$GENEID %in% matchedI
# Gen <- genes[keepGen,]
# chrI.tri <- cbind(Gen$TXCHROM, chrI.tri)
#
# chrI.all <- inner_join(chrI.tri, genes11DE, by="GENEID")
# matchedI <- intersect(chrI.all$GENEID, genes$GENEID)
# keepI <- chrI.all$GENEID %in% matchedI
# chrI.all <- chrI.all[keepI,]
#
# chrV.tri <- inner_join(genes4DE, genes49DE, by="GENEID")
# matchedV <- intersect(chrV.tri$GENEID, genes$GENEID)
# keepV <- chrV.tri$GENEID %in% matchedV
# chrV.tri <- chrV.tri[keepI,]
# keepGen <- genes$GENEID %in% matchedV
# Gen <- genes[keepGen,]
# chrV.tri <- cbind(Gen$TXCHROM, chrV.tri)
#
# chrVII.tri <- inner_join(genes59DE, genes61DE, by="GENEID")
# matchedVII <- intersect(chrVII.tri$GENEID, genes$GENEID)
# keepVII <- chrVII.tri$GENEID %in% matchedVII
# chrVII.tri <- chrVII.tri[keepVII,]
# keepGen <- genes$GENEID %in% matchedVII
# Gen <- genes[keepGen,]
# chrVII.tri <- cbind(Gen$TXCHROM, chrVII.tri)
#
# chrXII.tri <- inner_join(genes18DE, genes77DE, by="GENEID")
# matchedXII <- intersect(chrXII.tri$GENEID, genes$GENEID)
# keepXII <- chrXII.tri$GENEID %in% matchedXII
# chrXII.tri <- chrXII.tri[keepXII,]
# keepGen <- genes$GENEID %in% matchedXII
# Gen <- genes[keepGen,]
# chrXII.tri <- cbind(Gen$TXCHROM, chrXII.tri)

##any commonly DE genes between 2+ aneuploids?

#that are NOT commonly DE in euploid lines
# dupRows <- dupsBetweenGroups(allSamplesDE, "Sample")

#find any genes on aneuploid chromosomes that are NOT DE
# notDE4 <- subset(res4Dat,padj>0.1)
# notDE7 <- subset(res7Dat,padj>0.1)
# notDE8 <- subset(res8Dat,padj>0.1)
# notDE11 <- subset(res11Dat,padj>0.1)
# notDE18 <- subset(res18Dat,padj>0.1)
# notDE49 <- subset(res49Dat,padj>0.1)
# notDE59 <- subset(res59Dat,padj>0.1)
# notDE61 <- subset(res61Dat,padj>0.1)
# notDE76 <- subset(res76Dat,padj>0.1)
# notDE77 <- subset(res77Dat,padj>0.1)

# notDECom <- inner_join(notDE4, notDE7, by="GENEID")
# notDECom <- inner_join(notDECom, notDE8, by="GENEID")
# notDECom <- inner_join(notDECom, notDE11, by="GENEID")
# notDECom <- inner_join(notDECom, notDE18, by="GENEID")
# notDECom <- inner_join(notDECom, notDE49, by="GENEID")
# notDECom <- inner_join(notDECom, notDE59, by="GENEID")
# notDECom <- inner_join(notDECom, notDE61, by="GENEID")
# notDECom <- inner_join(notDECom, notDE76, by="GENEID")
# notDECom <- inner_join(notDECom, notDE77, by="GENEID")

```


Scripts for upsetR plots 
```{r ESR genes, include=FALSE, eval=TRUE, echo=FALSE}
### get trans genes in aneuploid lines that are differentially expressed
# then analyze if any of them are ESR genes
library(UpSetR)
dds= ddsDefMAnewS152
sample="152"
anc="Anc"
gene=genesMAnewS152
resultsESR <- function(dds,sample,anc,genes,Anc) {
# make a new dds data set that we can use in future analyses
   geneID <- rownames(dds)
   genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
# isolate known ESR genes 
  ESR <- list()
  ESR <- dplyr::inner_join(genes1,esrGenes,by=c("GENEID"="GENEID"))
  resultsNames(dds)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resESR <- as.data.frame(resDi)
  resESR <- cbind(names = rownames(resESR), resESR)
  resESR <- dplyr::inner_join(ESR,resESR,by=c("GENEID"="names"))
  # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  # resESR <- list()
  # resESRSig <- as.data.frame(subset(resDi,padj<0.1))
  # resESRSig <- cbind(names = rownames(resESRSig), resESRSig)
  # resESRSig <- dplyr::inner_join(ESR,resESRSig,by=c("GENEID"="GENEID"))
  # resESR$Sig <- resESRSig
  # resESR$NonSig <- resESRNonSig
  resESR$type <- "ESR"
  resESR$anc <- Anc
  names(resESR)[names(resESR) == 'log2FoldChange'] <- paste("log2FoldChange",sample,sep="")
  resESRred <- resESR[,c(1,2,4,9,10)]
  # View(resESRred)
  return(resESRred)
}


library(dplyr)
esrGenesMA152 <- resultsESR(ddsDefMAnewS152,"152","Anc",genesMAnewS152,"Homozygous")
View(esrGenesMA152)
esrGenesMA152$line <- "152"
esrGenesGC11 <- resultsESR(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11,"Heterozygous")
esrGenesGC11 %>% 
  rename(
     log2FoldChange = log2FoldChangeHolly_11
    )
esrGenesGC11$line <- "11"

esrGenesTest <- rbind(esrGenesMA152,esrGenesGC11)

esrGenesGC18 <- resultsESR(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18,"Heterozygous")
esrGenesGC7 <- resultsESR(ddsDefGC_7,"Holly_7","Holly_Anc",genesGC_7,"Heterozygous")
View(esrGenesGC7)
esrGenesGC4 <- resultsESR(ddsDefGC_4,"Holly_4","Holly_Anc",genesGC_4,"Heterozygous")
esrGenesGC49 <- resultsESR(ddsDefGC_49,"Holly_49","Holly_Anc",genesGC_49,"Heterozygous")
esrGenesGC59c <- resultsESR(ddsDefGC_59,"Holly_59","Holly_Anc",genesGC_59,"Heterozygous")
esrGenesGC61 <- resultsESR(ddsDefGC_61,"Holly_61","Holly_Anc",genesGC_61,"Heterozygous")
esrGenesGC76 <- resultsESR(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76,"Heterozygous")
esrGenesGC77 <- resultsESR(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_77,"Heterozygous")
esrGenesGC8 <- resultsESR(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8,"Heterozygous")

esrGenesHet <- cbind(esrGenesGC11$GENEID, esrGenesGC11$type, esrGenesGC11$anc, esrGenesGC11$log2FoldChangeHolly_11,esrGenesGC18$log2FoldChangeHolly_18,esrGenesGC7$log2FoldChangeHolly_7,esrGenesGC4$log2FoldChangeHolly_4,esrGenesGC49$log2FoldChangeHolly_49,esrGenesGC59c$log2FoldChangeHolly_59,esrGenesGC61$log2FoldChangeHolly_61,esrGenesGC76$log2FoldChangeHolly_76,esrGenesGC77$log2FoldChangeHolly_77,esrGenesGC8$log2FoldChangeHolly_8)
View(esrGenesHet)
colnames(esrGenesHet) <- c("GENEID","type","anc","11","18","7","4","49","59","61","76","77","8")
View(esrGenesHet)






```

```{r heatmap, include=FALSE,echo=FALSE,eval=TRUE}

# Library
library(ggplot2)
library(tidyr)
library(tibble)
library(hrbrthemes)
library(dplyr)

# Volcano dataset
rm(volcano)
volcano
View(volcano)


# Heatmap 
volc <- volcano %>% as_tibble() 
View(volc)
  volc <- volc %>% rowid_to_column(var="X")
 volc <- volc  %>% gather(key="Y", value="Z", -1)
 volc <- volc  %>% mutate(Y=as.numeric(gsub("V","",Y)))  # Change Y to numeric

  # Viz
  ggplot(volc, aes(X, Y, fill= Z)) + 
    geom_tile() +
    theme_ipsum() +
    theme(legend.position="none")

### Try with my esrGenes data 
  
esrGenesHet
esrHet <- esrGenesHet %>% as_tibble()
View(esrHet)
esrHet <- esrHet[,c(1,4:13)]
esrHet <- esrHet %>% gather(key="Y", value="Z", -1)
esrHet <- esrHet %>% mutate(Z=as.numeric(Z))
  # Viz
  ggplot(esrHet, aes(Y, GENEID, fill= Z)) + 
    geom_tile() +
  scale_fill_distiller(palette = "RdPu") +
  theme_ipsum()

  
  
  ## Let's try pHeatmap now 
  
library(pheatmap)
  BiocManager::install("DESeq")
  # load package
library("DESeq")
 
# load data and subset
example_file <- system.file ("extra/TagSeqExample.tab", package="DESeq")
data <- read.delim(example_file, header=T, row.names="gene")
data_subset <- as.matrix(data[rowSums(data)>50000,])
 str(data_subset)
# create heatmap using pheatmap
pheatmap(data_subset)
  
  
pheatmap(esrGenesHet)
#NAs introduced by coercion 
# problem likely that it is trying to cluster the first column (geneIDs) so I will have to make that rownames instead
library(tidyverse)
esrGenesHet2 <- as.data.frame(esrGenesHet)
esrGenesHet2 <- esrGenesHet2 %>% remove_rownames %>% column_to_rownames(var="GENEID")
View(esrGenesHet2)
pheatmap(esrGenesHet2)
esrGenesHet3 <- esrGenesHet2[,c(3:12)]
View(esrGenesHet3)
pheatmap(esrGenesHet3)
str(esrGenesHet3)
esrGenesHet3 <- mutate_all(esrGenesHet3, function(x) as.numeric(as.character(x)))
str(esrGenesHet3)
View(esrGenesHet3)
esrGenesHet4 <- as.matrix(esrGenesHet3)
View(esrGenesHet4)
rownames(esrGenesHet4) <- rownames(esrGenesHet2)
str(esrGenesHet4)
pheatmap(esrGenesHet4)

## let's try adding annotations now 
# for the lines (by anc)
my_sample_col <- data.frame(sample = rep(c("Hh"), c(10)))
View(my_sample_col)
row.names(my_sample_col) <- colnames(esrGenesHet4)
line11chr <- c("I,XV")
line76chr <- c("IX,X,XIV")
my_chrs <- c(line11chr,"I","I","V","V","VII","VII",line76chr,"XII","XVI")
View(my_chrs)
my_sample_col$chr <- my_chrs

# for the genes 
# these are all ESR genes (for now)
my_gene_col <- esrGenesHet[,2]
View(my_gene_col)
str(my_gene_col)
my_gene_col <- as.data.frame(my_gene_col)
rownames(my_gene_col) <- rownames(esrGenesHet4)
head(my_gene_col)
# try adding the annotations to the heatmap 
pheatmap(esrGenesHet4, annotation_row = my_gene_col,annotation_col=my_sample_col)




```



```{r esrGenes, include=TRUE,echo=FALSE,eval=TRUE}
####################### MA samples Aneuploids

esrGenesMA9 <- resultsESR(ddsDefMAold_9,"SC009","SCA",genesMAoldS9,"Homozygous")

esrGenesMA<- resultsESR(ddsDefMAold_11,"SC011","SCA",genesMAoldS11,"Homozygous")

esrGenesMA112 <- resultsESR(ddsDefMAnewS112,"112","Anc",genesMAnewS112,"Homozygous")

esrGenesMA29<- resultsESR(ddsDefMAnewS29,"29","Anc",genesMAnewS29,"Homozygous")

esrGenesMA119 <- resultsESR(ddsDefMAold_119,"SC119","SCA",genesMAoldS119,"Homozygous")

esrGenesMA88 <- resultsESR(ddsDefMAold_88,"SC088","SCA",genesMAoldS88,"Homozygous")

esrGenesMA108 <- resultsESR(ddsDefMAold_108,"SC108","SCA",genesMAoldS108,"Homozygous")

esrGenesMA15<- resultsESR(ddsDefMAold_15,"SC015","SCA",genesMAoldS15,"Homozygous")

esrGenesMA29 <- resultsESR(ddsDefMAnewS29,"29","Anc",genesMAnewS29,"Homozygous")

esrGenesMA117 <- resultsESR(ddsDefMAnewS117,"117","Anc",genesMAnewS117,"Homozygous")

esrGenesMA123 <- resultsESR(ddsDefMAnewS123,"123","Anc",genesMAnewS123,"Homozygous")

############################################################################################################
### need to do the euploid samples next - ESR genes
esrGenesGC2 <- resultsESR(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2,"Heterozygous")

esrGenesGC3 <- resultsESR(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3,"Heterozygous")

esrGenesGC5 <- resultsESR(ddsDefGC_5,"Holly_5","Holly_Anc",genesGC_5,"Heterozygous")

esrGenesGC9 <- resultsESR(ddsDefGC_9,"Holly_9","Holly_Anc",genesGC_9,"Heterozygous")

### for MA euploid samples 
esrGenesMA1 <- resultsESR(ddsDefMAold_1,"SC001","SCA",genesMAold_1,"Homozygous")
# write.csv(esrGenesMA1$Sig,file="sigESRGenesMA1.csv")
# write.csv(esrGenesMA1$NonSig,file="nonSigESRGenesMA1.csv")

esrGenesMA2 <- resultsESR(ddsDefMAold_2,"SC002","SCA",genesMAold_2,"Homozygous")
# write.csv(esrGenesMA2$Sig,file="sigESRGenesMA2.csv")
# write.csv(esrGenesMA2$NonSig,file="nonSigESRGenesMA2.csv")

esrGenesMA3 <- resultsESR(ddsDefMAold_3,"SC003","SCA",genesMAold_3,"Homozygous")
# write.csv(esrGenesMA3$Sig,file="sigESRGenesMA3.csv")
# write.csv(esrGenesMA3$NonSig,file="nonSigESRGenesMA3.csv")

esrGenesMA4 <- resultsESR(ddsDefMAold_4,"SC004","SCA",genesMAold_4,"Homozygous")
# write.csv(esrGenesMA4$Sig,file="sigESRGenesMA4.csv")
# write.csv(esrGenesMA4$NonSig,file="nonSigESRGenesMA4.csv")

esrGenesMA5 <- resultsESR(ddsDefMAold_5,"SC005","SCA",genesMAold_5,"Homozygous")
# write.csv(esrGenesMA5$Sig,file="sigESRGenesMA5.csv")
# write.csv(esrGenesMA5$NonSig,file="nonSigESRGenesMA5.csv")

esrGenesMA7 <- resultsESR(ddsDefMAold_7,"SC007","SCA",genesMAold_7,"Homozygous")
# write.csv(esrGenesMA7$Sig,file="sigESRGenesMA7.csv")
# write.csv(esrGenesMA7$NonSig,file="nonSigESRGenesMA7.csv")

esrGenesMA8 <- resultsESR(ddsDefMAold_8,"SC008","SCA",genesMAold_8,"Homozygous")
# write.csv(esrGenesMA8$Sig,file="sigESRGenesMA8.csv")
# write.csv(esrGenesMA8$NonSig,file="nonSigESRGenesMA8.csv")

esrGenesMA50 <- resultsESR(ddsDefMAnewS50,"50","Anc",genesMAnewS50,"Homozygous")
# write.csv(esrGenesMA50$Sig,file="sigESRGenesMA50.csv")
# write.csv(esrGenesMA50$NonSig,file="nonSigESRGenesMA50.csv")

esrGenesMA115 <- resultsESR(ddsDefMAnewS115,"115","Anc",genesMAnewS115,"Homozygous")
# write.csv(esrGenesMA115$Sig,file="sigESRGenesMA115.csv")
# write.csv(esrGenesMA115$NonSig,file="nonSigESRGenesMA115.csv")

upSetESRHomEu <- list(MAS1=esrGenesMA1$Sig$GENEID,
                 MAS2=esrGenesMA2$Sig$GENEID,
                 MAS3=esrGenesMA3$Sig$GENEID,
                 MAS4=esrGenesMA4$Sig$GENEID,
                 MAS5=esrGenesMA5$Sig$GENEID,
                 MAS7=esrGenesMA7$Sig$GENEID,
                 MAS8=esrGenesMA8$Sig$GENEID,
                 MAS50=esrGenesMA50$Sig$GENEID,
                 MAS115=esrGenesMA115$Sig$GENEID
              
)

pdf("esrGenesHomEu.pdf",width=12,height=10)
upset(fromList(upSetESRHomEu),nsets=9,keep.order = F,matrix.color = "deeppink1",main.bar.color = "black",sets.bar.color = "black",order.by = "freq",sets.x.label = "DE ESR Genes")
dev.off()






```

```{r ASR genes, include=TRUE, eval=TRUE, echo=FALSE}


### see if they are in ASR
resultsASR <- function(dds,sample,anc,genes) {
# make a new dds data set that we can use in future analyses
   geneID <- rownames(dds)
   genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
# determine if any significantly differentially expressed genes match those identified by gasch et al 2000
  ASR <- list()
  ASR$Sig <- dplyr::inner_join(ASRgenes,resDiSig,by=c("GENEID"="GENEID"))
  ASR$NonSig <- dplyr::inner_join(ASRgenes,resDiNonSig,by=c("GENEID"="GENEID"))
  return(ASR)
}

ASRGenesMA152 <- resultsASR(ddsDefMAnewS152,"152","Anc",genesMAnewS152)#,"chrI")
# write.csv(ASRGenesMA152$Sig,file="sigASRGenesMA152.csv")
# write.csv(ASRGenesMA152$NonSig,file="nonSigASRGenesMA152.csv")
ASRGenesGC11 <- resultsASR(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11)#,"chrI")
asrGenesSigGC11 <- ASRGenesGC11$Sig
# write.csv(ASRGenesGC11$Sig,file="sigASRGenesGC11.csv")
# write.csv(ASRGenesGC11$NonSig,file="nonSigASRGenesGC11.csv")

ASRGenesGC18chrI <- resultsASR(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18)#,"chrI")
asrGenesSigGC18 <- ASRGenesGC18chrI$Sig
# write.csv(ASRGenesGC18chrI$Sig,file="sigASRGenesGC18.csv")
# write.csv(ASRGenesGC18chrI$NonSig,file="nonSigASRGenesGC18.csv")

ASRGenesGC7chrI <- resultsASR(ddsDefGC_7,"Holly_7","Holly_Anc",genesGC_7)#,"chrI")
asrGenesSigGC7 <- ASRGenesGC7chrI$Sig
# write.csv(ASRGenesGC7chrI$Sig,file="sigASRGenesGC7.csv")
# write.csv(ASRGenesGC7chrI$NonSig,file="nonSigASRGenesGC7.csv")

ASRGenesMA123chrV <- resultsASR(ddsDefMAnewS123,"123","Anc",genesMAnewS123)#,"chrV")
asrGenesSigMA123 <- ASRGenesMA123chrV$Sig
# write.csv(ASRGenesMA123chrV$Sig,file="sigASRGenesMA123.csv")
# write.csv(ASRGenesMA123chrV$NonSig,file="nonSigASRGenesMA123.csv")

ASRGenesGC4chrV <- resultsASR(ddsDefGC_4,"Holly_4","Holly_Anc",genesGC_4)#,"chrV")
asrGenesSigGC4 <- ASRGenesGC4chrV$Sig
# write.csv(ASRGenesGC4chrV$Sig,file="sigASRGenesGC4.csv")
# write.csv(ASRGenesGC4chrV$NonSig,file="nonSigASRGenesGC4.csv")

ASRGenesGC49chrV <- resultsASR(ddsDefGC_49,"Holly_49","Holly_Anc",genesGC_49)#,"chrV")
asrGenesSigGC49 <- ASRGenesGC49chrV$Sig
# write.csv(ASRGenesGC49chrV$Sig,file="sigASRGenesGC49.csv")
# write.csv(ASRGenesGC49chrV$NonSig,file="nonSigASRGenesGC49.csv")

ASRGenesMA117chrV <- resultsASR(ddsDefMAnewS117,"117","Anc",genesMAnewS117)#,"chrV")
asrGenesSigMA117 <- ASRGenesMA117chrV$Sig
# write.csv(ASRGenesMA117chrV$Sig,file="sigASRGenesMA117.csv")
# write.csv(ASRGenesMA117chrV$NonSig,file="nonSigASRGenesMA117.csv")

ASRGenesGC59chrVII <- resultsASR(ddsDefGC_59,"Holly_59","Holly_Anc",genesGC_59)#,"chrVII")
asrGenesSigGC59 <- ASRGenesGC59chrVII$Sig
# write.csv(ASRGenesGC59chrVII$Sig,file="sigASRGenesGC59.csv")
# write.csv(ASRGenesGC59chrVII$NonSig,file="nonSigASRGenesGC59.csv")

ASRGenesGC61chrVII <- resultsASR(ddsDefGC_61,"Holly_61","Holly_Anc",genesGC_61)#,"chrVII")
asrGenesSigGC61 <- ASRGenesGC61chrVII$Sig
# write.csv(ASRGenesGC61chrVII$Sig,file="sigASRGenesGC61.csv")
# write.csv(ASRGenesGC61chrVII$NonSig,file="nonSigASRGenesGC61.csv")

ASRGenesMA29chrIX <- resultsASR(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
asrGenesSigMA29 <- ASRGenesMA29chrIX$Sig
# write.csv(ASRGenesMA29chrIX$Sig,file="sigASRGenesMA29.csv")
# write.csv(ASRGenesMA29chrIX$NonSig,file="nonSigASRGenesMA29.csv")

ASRGenesMA119chrIX <- resultsASR(ddsDefMAold_119,"SC119","SCA",genesMAoldS119)#,"chrIX")
asrGenesSigMA119 <- ASRGenesMA119chrIX$Sig
# write.csv(ASRGenesMA119chrIX$Sig,file="sigASRGenesMA119.csv")
# write.csv(ASRGenesMA119chrIX$NonSig,file="nonSigASRGenesMA119.csv")

ASRGenesMA88chrIX <- resultsASR(ddsDefMAold_88,"SC088","SCA",genesMAoldS88)#,"chrIX")
asrGenesSigMA88 <- ASRGenesMA88chrIX$Sig
# write.csv(ASRGenesMA88chrIX$Sig,file="sigASRGenesMA88.csv")
# write.csv(ASRGenesMA88chrIX$NonSig,file="nonSigASRGenesMA88.csv")

ASRGenesMA108chrVIII <- resultsASR(ddsDefMAold_108,"SC108","SCA",genesMAoldS108)#,"chrIX")
asrGenesSigMA108 <- ASRGenesMA108chrVIII$Sig
# write.csv(ASRGenesMA108chrVIII$Sig,file="sigASRGenesMA108.csv")
# write.csv(ASRGenesMA108chrVIII$NonSig,file="nonSigASRGenesMA108.csv")


ASRGenesMA15chrIX <- resultsASR(ddsDefMAold_15,"SC015","SCA",genesMAoldS15)#,"chrIX")
asrGenesSigMA15 <- ASRGenesMA15chrIX$Sig
# write.csv(ASRGenesMA15chrIX$Sig,file="sigASRGenesMA15.csv")
# write.csv(ASRGenesMA15chrIX$NonSig,file="nonSigASRGenesMA15.csv")

ASRGenesMA29chrIX <- resultsASR(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
asrGenesSigMA29 <- ASRGenesMA29chrIX$Sig
# write.csv(ASRGenesMA29chrIX$Sig,file="sigASRGenesMA29.csv")
# write.csv(ASRGenesMA29chrIX$NonSig,file="nonSigASRGenesMA29.csv")

ASRGenesGC76chrIX <- resultsASR(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76)#,"chrIX")
asrGenesSigGC76 <- ASRGenesGC76chrIX$Sig
# write.csv(ASRGenesGC76chrIX$Sig,file="sigASRGenesGC76.csv")
# write.csv(ASRGenesGC76chrIX$NonSig,file="nonSigASRGenesGC76.csv")

ASRGenesGC77chrXII <- resultsASR(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_76)#,"chrIX")
asrGenesSigGC77 <- ASRGenesGC77chrXII$Sig
# write.csv(ASRGenesGC77chrXII$Sig,file="sigASRGenesGC77.csv")
# write.csv(ASRGenesGC77chrXII$NonSig,file="nonSigASRGenesGC77.csv")

ASRGenesMA9chrXIV <- resultsASR(ddsDefMAold_9,"SC009","SCA",genesMAoldS9)#,"chrXIV")
asrGenesSigMA9 <- ASRGenesMA9chrXIV$Sig
# write.csv(ASRGenesMA9chrXIV$Sig,file="sigASRGenesMA9.csv")
# write.csv(ASRGenesMA9chrXIV$NonSig,file="nonSigASRGenesMA9.csv")

ASRGenesMA11 <- resultsASR(ddsDefMAold_11,"SC011","SCA",genesMAoldS11)#,"chrXIV")
asrGenesSigMA11 <- ASRGenesMA11$Sig
# write.csv(ASRGenesMA11$Sig,file="sigASRGenesMA11.csv")
# write.csv(ASRGenesMA11$NonSig,file="nonSigASRGenesMA11.csv")

ASRGenesMA112chrXVI <- resultsASR(ddsDefMAnewS112,"112","Anc",genesMAnewS112)#,"chrXVI")
asrGenesSigMA112 <- ASRGenesMA112chrXVI$Sig
# write.csv(ASRGenesMA112chrXVI$Sig,file="sigASRGenesMA112.csv")
# write.csv(ASRGenesMA112chrXVI$NonSig,file="nonSigASRGenesMA112.csv")

ASRGenesGC8chrXVI <- resultsASR(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8)#,"chrXVI")
asrGenesSigGC8 <- ASRGenesGC8chrXVI$Sig
# write.csv(ASRGenesGC8chrXVI$Sig,file="sigASRGenesGC8.csv")
# write.csv(ASRGenesGC8chrXVI$NonSig,file="nonSigASRGenesGC8.csv")



upSetASRHetAneu <- list(GCS11=ASRGenesGC11$Sig$GENEID,
                 GCS18=ASRGenesGC18chrI$Sig$GENEID,
                 GCS7=ASRGenesGC7chrI$Sig$GENEID,
                 GCS4=ASRGenesGC4chrV$Sig$GENEID,
                 GCS49=ASRGenesGC49chrV$Sig$GENEID,
                 GCS59=ASRGenesGC59chrVII$Sig$GENEID,
                 GCS61=ASRGenesGC61chrVII$Sig$GENEID,
                 GCS76=ASRGenesGC76chrIX$Sig$GENEID,
                 GCS77=ASRGenesGC77chrXII$Sig$GENEID,
                 GCS8=ASRGenesGC8chrXVI$Sig$GENEID
)

pdf("asrGenesHetAneu.pdf",width=12,height=10)
upset(fromList(upSetASRHetAneu),nsets=10,keep.order = F,matrix.color = "turquoise",main.bar.color = "black",sets.bar.color = "black",order.by = "freq",sets.x.label = "DE ASR Genes")
dev.off()
# 
# UpSet_contentsHetAneu <- listVennRegions(plotVenn(upSetASRHetAneu))
# ASR_genes_AneuHet <- UpSet_contents$`0, 1, 1, 1, 0, 1, 1, 0, 1, 1 (GCS18, GCS7, GCS4, GCS59, GCS61, GCS77, GCS8)`
# ASR_genes_AneuHet <- as.character(ASR_genes_AneuHet)
# ASR_genes_AneuHet <- ensembldb::select(txdb, keys=ASR_genes_AneuHet, columnns="GENEID","TXCHROM" , keytype="GENEID")
# 



upSetASRHomAneu <- list(MAS9=ASRGenesMA9chrXIV$Sig$GENEID,
                 MAS112=ASRGenesMA112chrXVI$Sig$GENEID,
                 MAS29=ASRGenesMA29chrIX$Sig$GENEID,
                 MAS119=ASRGenesMA119chrIX$Sig$GENEID,
                 MAS88=ASRGenesMA88chrIX$Sig$GENEID,
                 MAS108=ASRGenesMA108chrVIII$Sig$GENEID,
                 MAS15=ASRGenesMA15chrIX$Sig$GENEID,
                 MAS117=ASRGenesMA117chrV$Sig$GENEID,
                 MAS123=ASRGenesMA123chrV$Sig$GENEID,
                 MAS152=ASRGenesMA152$Sig$GENEID
)

pdf("asrGenesHomAneu.pdf",width=12,height=10)
upset(fromList(upSetASRHomAneu),nsets=10,keep.order = F,matrix.color = "turquoise",main.bar.color = "black",sets.bar.color = "black",order.by = "freq",sets.x.label = "DE ASR Genes")
dev.off()

UpSet_contentsHomAneu <- listVennRegions(plotVenn(upSetASRHomAneu))
ASR_genes_AneuHom <- UpSet_contents$`1, 0, 0, 0, 1, 1, 1, 0, 1, 1 (MAS9, MAS88, MAS108, MAS15, MAS123, MAS152)`
ASR_genes_AneuHom <- as.character(ASR_genes_AneuHom)
ASR_genes_AneuHom <- ensembldb::select(txdb, keys=ASR_genes_AneuHom, columnns="GENEID","TXCHROM" , keytype="GENEID")





### need to do the euploid samples next - ASR genes

ASRGenesGC1 <- resultsASR(ddsDefGC_1,"Holly_1","Holly_Anc",genesGC_1)
asrGenesSigGC1 <- ASRGenesGC1$Sig
# write.csv(ASRGenesGC1$Sig,file="sigASRGenesGC1.csv")
# write.csv(ASRGenesGC1$NonSig,file="nonSigASRGenesGC1.csv")

ASRGenesGC2 <- resultsASR(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2)
asrGenesSigGC2 <- ASRGenesGC2$Sig
# write.csv(ASRGenesGC2$Sig,file="sigASRGenesGC2.csv")
# write.csv(ASRGenesGC2$NonSig,file="nonSigASRGenesGC2.csv")

ASRGenesGC3 <- resultsASR(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3)
asrGenesSigGC3 <- ASRGenesGC3$Sig
# write.csv(ASRGenesGC3$Sig,file="sigASRGenesGC3.csv")
# write.csv(ASRGenesGC3$NonSig,file="nonSigASRGenesGC3.csv")

ASRGenesGC5 <- resultsASR(ddsDefGC_5,"Holly_5","Holly_Anc",genesGC_5)
asrGenesSigGC5 <- ASRGenesGC5$Sig
# write.csv(ASRGenesGC5$Sig,file="sigASRGenesGC5.csv")
# write.csv(ASRGenesGC5$NonSig,file="nonSigASRGenesGC5.csv")

ASRGenesGC9 <- resultsASR(ddsDefGC_9,"Holly_9","Holly_Anc",genesGC_9)
asrGenesSigGC9 <- ASRGenesGC9$Sig
# write.csv(ASRGenesGC9$Sig,file="sigASRGenesGC9.csv")
# write.csv(ASRGenesGC9$NonSig,file="nonSigASRGenesGC9.csv")

ASRGenesMA1 <- resultsASR(ddsDefMAold_1,"SC001","SCA",genesMAold_1)
asrGenesSigMA1 <- ASRGenesMA1$Sig
# write.csv(ASRGenesMA1$Sig,file="sigASRGenesMA1.csv")
# write.csv(ASRGenesMA1$NonSig,file="nonSigASRGenesMA1.csv")

ASRGenesMA2 <- resultsASR(ddsDefMAold_2,"SC002","SCA",genesMAold_2)
asrGenesSigMA2 <- ASRGenesMA2$Sig
# write.csv(ASRGenesMA2$Sig,file="sigASRGenesMA2.csv")
# write.csv(ASRGenesMA2$NonSig,file="nonSigASRGenesMA2.csv")

ASRGenesMA3 <- resultsASR(ddsDefMAold_3,"SC003","SCA",genesMAold_3)
asrGenesSigMA3 <- ASRGenesMA3$Sig
# write.csv(ASRGenesMA3$Sig,file="sigASRGenesMA3.csv")
# write.csv(ASRGenesMA3$NonSig,file="nonSigASRGenesMA3.csv")

ASRGenesMA4 <- resultsASR(ddsDefMAold_4,"SC004","SCA",genesMAold_4)
asrGenesSigMA4 <- ASRGenesMA4$Sig
# write.csv(ASRGenesMA4$Sig,file="sigASRGenesMA4.csv")
# write.csv(ASRGenesMA4$NonSig,file="nonSigASRGenesMA4.csv")

ASRGenesMA5 <- resultsASR(ddsDefMAold_5,"SC005","SCA",genesMAold_5)
asrGenesSigMA5 <- ASRGenesMA5$Sig
# write.csv(ASRGenesMA5$Sig,file="sigASRGenesMA5.csv")
# write.csv(ASRGenesMA5$NonSig,file="nonSigASRGenesMA5.csv")

ASRGenesMA7 <- resultsASR(ddsDefMAold_7,"SC007","SCA",genesMAold_7)
asrGenesSigMA7 <- ASRGenesMA7$Sig
# write.csv(ASRGenesMA7$Sig,file="sigASRGenesMA7.csv")
# write.csv(ASRGenesMA7$NonSig,file="nonSigASRGenesMA7.csv")

ASRGenesMA8 <- resultsASR(ddsDefMAold_8,"SC008","SCA",genesMAold_8)
asrGenesSigMA8 <- ASRGenesMA8$Sig
# write.csv(ASRGenesMA8$Sig,file="sigASRGenesMA8.csv")
# write.csv(ASRGenesMA8$NonSig,file="nonSigASRGenesMA8.csv")

ASRGenesMA50 <- resultsASR(ddsDefMAnewS50,"50","Anc",genesMAnewS50)
asrGenesSigMA50 <- ASRGenesMA50$Sig
# write.csv(ASRGenesMA50$Sig,file="sigASRGenesMA50.csv")
# write.csv(ASRGenesMA50$NonSig,file="nonSigASRGenesMA50.csv")

ASRGenesMA115 <- resultsASR(ddsDefMAnewS115,"115","Anc",genesMAnewS115)
asrGenesSigMA115 <- ASRGenesMA115$Sig
# write.csv(ASRGenesMA115$Sig,file="sigASRGenesMA115.csv")
# write.csv(ASRGenesMA115$NonSig,file="nonSigASRGenesMA115.csv")

upSetASRHetEu <- list(GCS2=ASRGenesGC2$Sig$GENEID,
                 GCS3=ASRGenesGC3$Sig$GENEID,
                 GCS5=ASRGenesGC5$Sig$GENEID,
                 GCS9=ASRGenesGC9$Sig$GENEID
              
)

pdf("asrGenesHetEu.pdf")
upset(fromList(upSetASRHetEu),nsets=4,keep.order = F,matrix.color = "turquoise",main.bar.color = "black",sets.bar.color = "black",order.by = "freq",sets.x.label = "DE ASR Genes")
dev.off()




upSetASRHomEu <- list(MAS1=ASRGenesMA1$Sig$GENEID,
                 MAS2=ASRGenesMA2$Sig$GENEID,
                 MAS3=ASRGenesMA3$Sig$GENEID,
                 MAS4=ASRGenesMA4$Sig$GENEID,
                 MAS5=ASRGenesMA5$Sig$GENEID,
                 MAS7=ASRGenesMA7$Sig$GENEID,
                 MAS8=ASRGenesMA8$Sig$GENEID,
                 MAS50=ASRGenesMA50$Sig$GENEID,
                 MAS115=ASRGenesMA115$Sig$GENEID
              
)

pdf("asrGenesHomEu.pdf",width=12,height=10)
upset(fromList(upSetASRHomEu),nsets=9,keep.order = F,matrix.color = "turquoise",main.bar.color = "black",sets.bar.color = "black",order.by = "freq",sets.x.label = "DE ASR Genes")
dev.off()






```

```{r dosSens Genes, include=TRUE, eval=TRUE, echo=FALSE}


resultsDosSens <- function(dds,sample,anc,genes) {
# make a new dds data set that we can use in future analyses
   geneID <- rownames(dds)
   genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
# subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
# determine if any significantly differentially expressed genes match those identified by gasch et al 2000
  dosSensRes <- list()
  dosSensRes$Sig <- dplyr::inner_join(dosageSensitiveGenes,resDiSig,by=c("GENEID"="GENEID"))
  dosSensRes$NonSig <- dplyr::inner_join(dosageSensitiveGenes,resDiNonSig,by=c("GENEID"="GENEID"))
  return(dosSensRes)
}


dosSensGenesMA152 <- resultsDosSens(ddsDefMAnewS152,"152","Anc",genesMAnewS152)#,"chrI")
dosSensGenesSigMA152 <- dosSensGenesMA152$Sig
dosSensGenesNonSigMA152 <- dosSensGenesMA152$NonSig

# write.csv(dosSensGenesMA152$Sig,file="sigEdosSensGenesMA152.csv")
# write.csv(dosSensGenesMA152$NonSig,file="nonSigdosSensGenesMA152.csv")

dosSensGenesGC11 <- resultsDosSens(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11)#,"chrI")
dosSensGenesSigGC11 <- dosSensGenesGC11$Sig
# write.csv(dosSensGenesGC11$Sig,file="sigdosSensGenesGC11.csv")
# write.csv(dosSensGenesGC11$NonSig,file="nondosSensGenesGC11.csv")

dosSensGenesGC18chrI <- resultsDosSens(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18)#,"chrI")
dosSensGenesSigGC18 <- dosSensGenesGC18chrI$Sig
# write.csv(dosSensGenesGC18chrI$Sig,file="sigdosSensGenesGC18.csv")
# write.csv(dosSensGenesGC18chrI$NonSig,file="nonSigdosSensGenesGC18.csv")

dosSensGenesGC7chrI <- resultsDosSens(ddsDefGC_7,"Holly_7","Holly_Anc",genesGC_7)#,"chrI")
dosSensGenesSigGC7 <- dosSensGenesGC7chrI$Sig
# write.csv(dosSensGenesGC7chrI$Sig,file="sigdosSensGenesGC7.csv")
# write.csv(dosSensGenesGC7chrI$NonSig,file="nonSigdosSensGenesGC7.csv")

dosSensGenesMA123chrV <- resultsDosSens(ddsDefMAnewS123,"123","Anc",genesMAnewS123)#,"chrV")
dosSensGenesSigMA123 <- dosSensGenesMA123chrV$Sig
# write.csv(dosSensGenesMA123chrV$Sig,file="sigdosSensGenesMA123.csv")
# write.csv(dosSensGenesMA123chrV$NonSig,file="nonSigdosSensGenesMA123.csv")

dosSensGenesGC4chrV <- resultsDosSens(ddsDefGC_4,"Holly_4","Holly_Anc",genesGC_4)#,"chrV")
dosSensGenesSigGC4 <- dosSensGenesGC4chrV$Sig
# write.csv(dosSensGenesGC4chrV$Sig,file="sigdosSensGenesGC4.csv")
# write.csv(dosSensGenesGC4chrV$NonSig,file="nonSigdosSensGenesGC4.csv")

dosSensGenesGC49chrV <- resultsDosSens(ddsDefGC_49,"Holly_49","Holly_Anc",genesGC_49)#,"chrV")
dosSensGenesSigGC49 <- dosSensGenesGC49chrV$Sig
# write.csv(dosSensGenesGC49chrV$Sig,file="sigdosSensGenesGC49.csv")
# write.csv(dosSensGenesGC49chrV$NonSig,file="nonSigdosSensGenesGC49.csv")

dosSensGenesMA117chrV <- resultsDosSens(ddsDefMAnewS117,"117","Anc",genesMAnewS117)#,"chrV")
dosSensGenesSigMA117 <- dosSensGenesMA117chrV$Sig
# write.csv(dosSensGenesMA117chrV$Sig,file="sigdosSensGenesMA117.csv")
# write.csv(dosSensGenesMA117chrV$NonSig,file="nonSigdosSensGenesMA117.csv")

dosSensGenesGC59chrVII <- resultsDosSens(ddsDefGC_59,"Holly_59","Holly_Anc",genesGC_59)#,"chrVII")
dosSensGenesSigGC59 <- dosSensGenesGC59chrVII$Sig
# write.csv(dosSensGenesGC59chrVII$Sig,file="sigdosSensGenesGC59.csv")
# write.csv(dosSensGenesGC59chrVII$NonSig,file="nonSigdosSensGenesGC59.csv")

dosSensGenesGC61chrVII <- resultsDosSens(ddsDefGC_61,"Holly_61","Holly_Anc",genesGC_61)#,"chrVII")
dosSensGenesSigGC61 <- dosSensGenesGC61chrVII$Sig
# write.csv(dosSensGenesGC61chrVII$Sig,file="sigdosSensGenesGC61.csv")
# write.csv(dosSensGenesGC61chrVII$NonSig,file="nonSigdosSensGenesGC61.csv")

dosSensGenesMA29chrIX <- resultsDosSens(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
dosSensGenesSigMA29 <- dosSensGenesMA29chrIX$Sig
# write.csv(dosSensGenesMA29chrIX$Sig,file="sigdosSensGenesMA29.csv")
# write.csv(dosSensGenesMA29chrIX$NonSig,file="nonSigdosSensGenesMA29.csv")

dosSensGenesMA119chrIX <- resultsDosSens(ddsDefMAold_119,"SC119","SCA",genesMAoldS119)#,"chrIX")
dosSensGenesSigMA119 <- dosSensGenesMA119chrIX$Sig
# write.csv(dosSensGenesMA119chrIX$Sig,file="sigdosSensGenesMA119.csv")
# write.csv(dosSensGenesMA119chrIX$NonSig,file="nonSigdosSensGenesMA119.csv")

dosSensGenesMA88chrIX <- resultsDosSens(ddsDefMAold_88,"SC088","SCA",genesMAoldS88)#,"chrIX")
dosSensGenesSigMA88 <- dosSensGenesMA88chrIX$Sig
# write.csv(dosSensGenesMA88chrIX$Sig,file="sigdosSensGenesMA88.csv")
# write.csv(dosSensGenesMA88chrIX$NonSig,file="nonSigdosSensGenesMA88.csv")

dosSensGenesMA108chrVIII <- resultsDosSens(ddsDefMAold_108,"SC108","SCA",genesMAoldS108)#,"chrIX")
dosSensGenesSigMA108 <- dosSensGenesMA108chrVIII$Sig
# write.csv(dosSensGenesMA108chrVIII$Sig,file="sigdosSensGenesMA108.csv")
# write.csv(dosSensGenesMA108chrVIII$NonSig,file="nonSigdosSensGenesMA108.csv")

dosSensGenesMA15chrIX <- resultsDosSens(ddsDefMAold_15,"SC015","SCA",genesMAoldS15)#,"chrIX")
dosSensGenesSigMA15 <- dosSensGenesMA15chrIX$Sig
# write.csv(dosSensGenesMA15chrIX$Sig,file="sigdosSensGenesMA15.csv")
# write.csv(dosSensGenesMA15chrIX$NonSig,file="nonSigdosSensGenesMA15.csv")

dosSensGenesMA29chrIX <- resultsDosSens(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
dosSensGenesSigMA29 <- dosSensGenesMA29chrIX$Sig
# write.csv(dosSensGenesMA29chrIX$Sig,file="sigdosSensGenesMA29.csv")
# write.csv(dosSensGenesMA29chrIX$NonSig,file="nonSigdosSensGenesMA29.csv")

dosSensGenesGC76chrIX <- resultsDosSens(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76)#,"chrIX")
dosSensGenesSigGC76 <- dosSensGenesGC76chrIX$Sig
# write.csv(dosSensGenesGC76chrIX$Sig,file="sigdosSensGenesGC76.csv")
# write.csv(dosSensGenesGC76chrIX$NonSig,file="nonSigdosSensGenesGC76.csv")

dosSensGenesGC77chrXII <- resultsDosSens(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_76)#,"chrIX")
dosSensGenesSigGC77 <- dosSensGenesGC77chrXII$Sig
# write.csv(dosSensGenesGC77chrXII$Sig,file="sigdosSensGenesGC77.csv")
# write.csv(dosSensGenesGC77chrXII$NonSig,file="nonSigdosSensGenesGC77.csv")

dosSensGenesMA9chrXIV <- resultsDosSens(ddsDefMAold_9,"SC009","SCA",genesMAoldS9)#,"chrXIV")
dosSensGenesSigMA9 <- dosSensGenesMA9chrXIV$Sig
# write.csv(dosSensGenesMA9chrXIV$Sig,file="sigdosSensGenesMA9.csv")
# write.csv(dosSensGenesMA9chrXIV$NonSig,file="nonSigdosSensGenesMA9.csv")

dosSensGenesMA11 <- resultsDosSens(ddsDefMAold_11,"SC011","SCA",genesMAoldS11)#,"chrXIV")
dosSensGenesSigMA11 <- dosSensGenesMA11$Sig
# write.csv(dosSensGenesMA11$Sig,file="sigdosSensGenesMA11.csv")
# write.csv(dosSensGenesMA11$NonSig,file="nonSigdosSensGenesMA11.csv")

dosSensGenesMA112chrXVI <- resultsDosSens(ddsDefMAnewS112,"112","Anc",genesMAnewS112)#,"chrXVI")
dosSensGenesSigMA112 <- dosSensGenesMA112chrXVI$Sig
# write.csv(dosSensGenesMA112chrXVI$Sig,file="sigdosSensGenesMA112.csv")
# write.csv(dosSensGenesMA112chrXVI$NonSig,file="nonSigdosSensGenesMA112.csv")

dosSensGenesGC8chrXVI <- resultsDosSens(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8)#,"chrXVI")
dosSensGenesSigGC8 <- dosSensGenesGC8chrXVI$Sig
# write.csv(dosSensGenesGC8chrXVI$Sig,file="sigdosSensGenesGC8.csv")
# write.csv(dosSensGenesGC8chrXVI$NonSig,file="nonSigdosSensGenesGC8.csv")

### need to do the euploid samples next - dosage sensitive genes
dosSensGenesGC1 <- resultsDosSens(ddsDefGC_1,"Holly_1","Holly_Anc",genesGC_1)
dosSensGenesSigGC1 <- dosSensGenesGC1$Sig

dosSensGenesGC2 <- resultsDosSens(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2)
dosSensGenesSigGC2 <- dosSensGenesGC2$Sig
# write.csv(dosSensGenesGC2$Sig,file="sigdosSensGenesGC2.csv")
# write.csv(dosSensGenesGC2$NonSig,file="nonSigdosSensGenesGC2.csv")

dosSensGenesGC3 <- resultsDosSens(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3)
dosSensGenesSigGC3 <- dosSensGenesGC3$Sig
# write.csv(dosSensGenesGC3$Sig,file="sigdosSensGenesGC3.csv")
# write.csv(dosSensGenesGC3$NonSig,file="nonSigdosSensGenesGC3.csv")

dosSensGenesGC5 <- resultsDosSens(ddsDefGC_5,"Holly_5","Holly_Anc",genesGC_5)
dosSensGenesSigGC5 <- dosSensGenesGC5$Sig
# write.csv(dosSensGenesGC5$Sig,file="sigdosSensGenesGC5.csv")
# write.csv(dosSensGenesGC5$NonSig,file="nonSigdosSensGenesGC5.csv")


dosSensGenesGC9 <- resultsDosSens(ddsDefGC_9,"Holly_9","Holly_Anc",genesGC_9)
dosSensGenesSigGC9 <- dosSensGenesGC9$Sig
# write.csv(dosSensGenesGC9$Sig,file="sigdosSensGenesGC9.csv")
# write.csv(dosSensGenesGC9$NonSig,file="nonSigdosSensGenesGC9.csv")

dosSensGenesMA1 <- resultsDosSens(ddsDefMAold_1,"SC001","SCA",genesMAold_1)
dosSensGenesSigMA1 <- dosSensGenesMA1$Sig
# write.csv(dosSensGenesMA1$Sig,file="sigdosSensGenesMA1.csv")
# write.csv(dosSensGenesMA1$NonSig,file="nonSigdosSensGenesMA1.csv")

dosSensGenesMA2 <- resultsDosSens(ddsDefMAold_2,"SC002","SCA",genesMAold_2)
dosSensGenesSigMA2 <- dosSensGenesMA2$Sig
# write.csv(dosSensGenesMA2$Sig,file="sigdosSensGenesMA2.csv")
# write.csv(dosSensGenesMA2$NonSig,file="nonSigdosSensGenesMA2.csv")

dosSensGenesMA3 <- resultsDosSens(ddsDefMAold_3,"SC003","SCA",genesMAold_3)
dosSensGenesSigMA3 <- dosSensGenesMA3$Sig
# write.csv(dosSensGenesMA3$Sig,file="sigdosSensGenesMA3.csv")
# write.csv(dosSensGenesMA3$NonSig,file="nonSigdosSensGenesMA3.csv")

dosSensGenesMA4 <- resultsDosSens(ddsDefMAold_4,"SC004","SCA",genesMAold_4)
dosSensGenesSigMA4 <- dosSensGenesMA4$Sig
# write.csv(dosSensGenesMA4$Sig,file="sigdosSensGenesMA4.csv")
# write.csv(dosSensGenesMA4$NonSig,file="nonSigdosSensGenesMA4.csv")

dosSensGenesMA5 <- resultsDosSens(ddsDefMAold_5,"SC005","SCA",genesMAold_5)
dosSensGenesSigMA5 <- dosSensGenesMA5$Sig
# write.csv(dosSensGenesMA5$Sig,file="sigdosSensGenesMA5.csv")
# write.csv(dosSensGenesMA5$NonSig,file="nonSigdosSensGenesMA5.csv")

dosSensGenesMA7 <- resultsDosSens(ddsDefMAold_7,"SC007","SCA",genesMAold_7)
dosSensGenesSigMA7 <- dosSensGenesMA7$Sig
# write.csv(dosSensGenesMA7$Sig,file="sigdosSensGenesMA7.csv")
# write.csv(dosSensGenesMA7$NonSig,file="nonSigdosSensGenesMA7.csv")

dosSensGenesMA8 <- resultsDosSens(ddsDefMAold_8,"SC008","SCA",genesMAold_8)
dosSensGenesSigMA8 <- dosSensGenesMA8$Sig
# write.csv(dosSensGenesMA8$Sig,file="sigdosSensGenesMA8.csv")
# write.csv(dosSensGenesMA8$NonSig,file="nonSigdosSensGenesMA8.csv")

dosSensGenesMA50 <- resultsDosSens(ddsDefMAnewS50,"50","Anc",genesMAnewS50)
dosSensGenesSigMA50 <-  dosSensGenesMA50$Sig
# write.csv(dosSensGenesMA50$Sig,file="sigdosSensGenesMA50.csv")
# write.csv(dosSensGenesMA50$NonSig,file="nonSigdosSensGenesMA50.csv")

dosSensGenesMA115 <- resultsDosSens(ddsDefMAnewS115,"115","Anc",genesMAnewS115)

# write.csv(dosSensResMA115$Sig,file="sigDosSensGenesMA115.csv")
# write.csv(dosSensResMA115$NonSig,file="nonSigDosSensGenesMA115.csv")

upSetdosSensHetAneu <- list(GCS11=dosSensGenesGC11$Sig$GENEID,
                 GCS18=dosSensGenesGC18chrI$Sig$GENEID,
                 GCS7=dosSensGenesGC7chrI$Sig$GENEID,
                 GCS4=dosSensGenesGC4chrV$Sig$GENEID,
                 GCS49=dosSensGenesGC49chrV$Sig$GENEID,
                 GCS59=dosSensGenesGC59chrVII$Sig$GENEID,
                 GCS61=dosSensGenesGC61chrVII$Sig$GENEID,
                 GCS76=dosSensGenesGC76chrIX$Sig$GENEID,
                 GCS77=dosSensGenesGC77chrXII$Sig$GENEID,
                 GCS8=dosSensGenesGC8chrXVI$Sig$GENEID
)

pdf("dosSensHetAneu.pdf",width=12,height=10)
upset(fromList(upSetdosSensHetAneu),nsets=10,keep.order = F,matrix.color = "purple",main.bar.color = "black",sets.bar.color = "black",order.by = "freq",sets.x.label = "DE Dosage Sensitive Genes")
dev.off()

upSetdosSensHomAneu <- list(MAS9=dosSensGenesMA9chrXIV$Sig$GENEID,
                 MAS112=dosSensGenesMA112chrXVI$Sig$GENEID,
                 MAS29=dosSensGenesMA29chrIX$Sig$GENEID,
                 MAS119=dosSensGenesMA119chrIX$Sig$GENEID,
                 MAS88=dosSensGenesMA88chrIX$Sig$GENEID,
                 MAS108=dosSensGenesMA108chrVIII$Sig$GENEID,
                 MAS15=dosSensGenesMA15chrIX$Sig$GENEID,
                 MAS117=dosSensGenesMA117chrV$Sig$GENEID,
                 MAS123=dosSensGenesMA123chrV$Sig$GENEID,
                 MAS152=dosSensGenesMA152$Sig$GENEID
)

pdf("dosSensHomAneu.pdf",width=12,height=10)
upset(fromList(upSetdosSensHomAneu),nsets=10,keep.order = F,matrix.color = "purple",main.bar.color = "black",sets.bar.color = "black",order.by = "freq",sets.x.label = "DE dosSens Genes")
dev.off()

upSetdosSensHetEu <- list(GCS2=dosSensGenesGC2$Sig$GENEID,
                 GCS3=dosSensGenesGC3$Sig$GENEID,
                 GCS5=dosSensGenesGC5$Sig$GENEID,
                 GCS9=dosSensGenesGC9$Sig$GENEID
              
)

pdf("dosSensHetEu.pdf")
upset(fromList(upSetdosSensHetEu),nsets=4,keep.order = F,matrix.color = "purple",main.bar.color = "black",sets.bar.color = "black",order.by = "freq",sets.x.label = "DE dosSens Genes")
dev.off()

upSetdosSensHomEu <- list(MAS1=dosSensGenesMA1$Sig$GENEID,
                 MAS2=dosSensGenesMA2$Sig$GENEID,
                 MAS3=dosSensGenesMA3$Sig$GENEID,
                 MAS4=dosSensGenesMA4$Sig$GENEID,
                 MAS5=dosSensGenesMA5$Sig$GENEID,
                 MAS7=dosSensGenesMA7$Sig$GENEID,
                 MAS8=dosSensGenesMA8$Sig$GENEID,
                 MAS50=dosSensGenesMA50$Sig$GENEID,
                 MAS115=dosSensGenesMA115$Sig$GENEID
              
)

pdf("dosSensHomEu.pdf")
upset(fromList(upSetdosSensHomEu),nsets=9,keep.order = F,matrix.color = "purple",main.bar.color = "black",sets.bar.color = "black",order.by = "freq",sets.x.label = "DE dosSens Genes")
dev.off()
```



Below is a script for differentiating genes that are differentially expressed from those that are not dosage compensated, dosage compensated, or partially dosage compensated. I used the log fold change threshold function of the results function to isolate these genes. FDR p values < 0.1 were considered statistically significant.

```{r results Euploids,eval=TRUE}
############## Slightly different way of testing -- different order ######
## for euploid lines
resultsEuploid <- function(dds,sample,anc,genes) {
# make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  all(rownames(dds) == genes$GENEID)
  # genes <- genes$GENEID
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes,resDiSig,by=c("GENEID"="names"))
   results <- list()
  results$Sig <- resDiSig
  results$nonSig <- resDiNonSig
  return(results)
}

sample2GC <- resultsEuploid(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2)
sample2GC$Sig
# write.csv(sample2GC$Sig,file="DEgenesSample2GC.csv")

sample3GC <- resultsEuploid(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3)
sample3GC$Sig
# write.csv(sample3GC$Sig,file="DEgenesSample3GC.csv")

sample5GC <- resultsEuploid(ddsDefGC_5,"Holly_5","Holly_Anc",genesGC_5)
sample5GC$Sig
# write.csv(sample5GC$Sig,file="DEgenesSample5GC.csv")

sample9GC <- resultsEuploid(ddsDefGC_9,"Holly_9","Holly_Anc",genesGC_9)
sample9GC$Sig
# write.csv(sample9GC$Sig,file="DEgenesSample9GC.csv")

# DE genes in aneuploid lines

sample1MA <- resultsEuploid(ddsDefMAold_1,"SC001","SCA",genesMAold_1)
sample1MA$Sig
# write.csv(sample1MA$Sig,file="DEgenesSample1MA.csv")

sample2MA <- resultsEuploid(ddsDefMAold_2,"SC002","SCA",genesMAold_2)
sample2MA$Sig
# write.csv(sample2MA$Sig,file="DEgenesSample2MA.csv")

sample3MA <- resultsEuploid(ddsDefMAold_3,"SC003","SCA",genesMAold_3)
sample3MA$Sig
# write.csv(sample3MA$Sig,file="DEgenesSample3MA.csv")

sample4MA <- resultsEuploid(ddsDefMAold_4,"SC004","SCA",genesMAold_4)
sample4MA$Sig
# write.csv(sample4MA$Sig,file="DEgenesSample4MA.csv")

sample5MA <- resultsEuploid(ddsDefMAold_5,"SC005","SCA",genesMAold_5)
sample5MA$Sig
# write.csv(sample5MA$Sig,file="DEgenesSample5MA.csv")

sample7MA <- resultsEuploid(ddsDefMAold_7,"SC007","SCA",genesMAold_7)
sample7MA$Sig
# write.csv(sample7MA$Sig,file="DEgenesSample7MA.csv")

sample8MA <- resultsEuploid(ddsDefMAold_8,"SC008","SCA",genesMAold_8)
sample8MA$Sig
# write.csv(sample8MA$Sig,file="DEgenesSample8MA.csv")

sample50MA <- resultsEuploid(ddsDefMAnewS50,"50","Anc",genesMAnew_50)
sample50MA$Sig
# write.csv(sample50MA$Sig,file="DEgenesSample50MA.csv")

sample115MA <- resultsEuploid(ddsDefMAnewS115,"115","Anc",genesMAnewS115)
sample115MA$Sig
# write.csv(sample115MA$Sig,file="DEgenesSample115MA.csv")
```

trisomy scripts 
```{r resultsTrisomy,include=FALSE,echo=FALSE,eval=TRUE}
resultsTrisomy <- function(dds,sample,anc,genes,anChr) {
# make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  keep <- anChr
  genes <- genes[(grep(paste0(keep,"$"),genes$TXCHROM,perl=TRUE)),]
  all(rownames(dds) == genes$GENEID)
  # genes <- genes$GENEID
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  resTri <- results(dds,contrast=c("condition",sample,anc),lfcThreshold=0.585)
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (NO DC))
  resTriNonSig <- as.data.frame(subset(resTri,padj>0.1))
  resTriNonSig <- cbind(names = rownames(resTriNonSig), resTriNonSig)
  resTriNonSig <- dplyr::inner_join(genes,resTriNonSig,by=c("GENEID"="names"))
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resTriSig <- as.data.frame(subset(resTri,padj<0.1))
  resTriSig <- cbind(names = rownames(resTriSig), resTriSig)
  resTriSig <- dplyr::inner_join(genes,resTriSig,by=c("GENEID"="names"))
# put non-significantly different genes in a data table
  results <- list()
  results$resNoDC <- resTriNonSig
  results$resPotDC <- resTriSig
  #### run same analyses using new ddsCopy dataset and new parameters
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(ddsCopy,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get Significant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  ### Find genes that are Signficiantly different than Trisomic expectation and are Significantly different from Disomic expectation
  resPartOvDC <- dplyr::inner_join(resTriSig,resDiSig,by="GENEID")
  results$resPartOvDC <- resPartOvDC
  ### Find genes that are Signficiantly different than Trisomic expectation but NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resTriSig,resDiNonSig,by="GENEID")
  results$resFullDC <- resFullDC
  transGenesDE <- dplyr::anti_join(resDiSig,genes,by=c("GENEID"="TXCHROM"))
  results$transGenesDE <- transGenesDE
  return(results)
}

resultsNames(ddsDefMAnewS152)
maOld152.1 <- resultsTrisomy(ddsDefMAnewS152,"152","Anc",genesMAnewS152,"chrI")
View(maOld152.1)


maOldTest <- resultsTrisomy(ddsDefMAold_108,"SC108","SCA",genesMAold_108,"chrVIII")
```

ESR genes scripts 

```{r resultsEsr, include=FALSE,echo=FALSE,eval=TRUE}
resultsESR <- function(dds,sample,anc,genes) {
# make a new dds data set that we can use in future analyses
   geneID <- rownames(dds)
  # #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  # #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
   genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
# determine if any significantly differentially expressed genes match those identified by gasch et al 2000
  ESR <- list()
  ESR$Sig <- dplyr::inner_join(ASRgenes,resDiSig,by=c("GENEID"="GENEID"))
  ESR$NonSig <- dplyr::inner_join(ASRgenes,resDiNonSig,by=c("GENEID"="GENEID"))
  return(ESR)
}
#### these two need to be done separately bc they have 2 aneuploid chromosomes
esrGenesMA152 <- resultsESR(ddsDefMAnewS152,"152","Anc",genesMAnewS152)#,"chrI")
esrGenesMA152Sig <- esrGenesMA152$Sig
esrGenesMA152Up <- subset(esrGenesMA152Sig,log2FoldChange>1)
View(esrGenesMA152Up)
str(esrGenesMA152Up)
esrGenesMA152Down <- subset(esrGenesMA152Sig,log2FoldChange<1)
str(esrGenesMA152Down)

esrGenesGC11 <- resultsESR(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11)#,"chrI")
esrGenesGC11Sig <- esrGenesGC11$Sig
esrGenesGC11Up <- subset(esrGenesGC11Sig,log2FoldChange>1)
View(esrGenesGC11Up)
str(esrGenesGC11Up)
esrGenesGC11Down <- subset(esrGenesGC11Sig,log2FoldChange<1)
str(esrGenesGC11Down)

# write.csv(esrGenesGC11$Sig,file="sigESRGenesGC11.csv")
# write.csv(esrGenesGC11$NonSig,file="nonSigESRGenesGC11.csv")

esrGenesGC18chrI <- resultsESR(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18)#,"chrI")
# write.csv(esrGenesGC18chrI$Sig,file="sigESRGenesGC18.csv")
# write.csv(esrGenesGC18chrI$NonSig,file="nonSigESRGenesGC18.csv")

esrGenesGC7chrI <- resultsESR(ddsDefGC_7,"Holly_7","Holly_Anc",genesGC_7)#,"chrI")
# write.csv(esrGenesGC7chrI$Sig,file="sigESRGenesGC7.csv")
# write.csv(esrGenesGC7chrI$NonSig,file="nonSigESRGenesGC7.csv")

esrGenesMA123chrV <- resultsESR(ddsDefMAnewS123,"123","Anc",genesMAnewS123)#,"chrV")
# write.csv(esrGenesMA123chrV$Sig,file="sigESRGenesMA123.csv")
# write.csv(esrGenesMA123chrV$NonSig,file="nonSigESRGenesMA123.csv")

esrGenesGC4chrV <- resultsESR(ddsDefGC_4,"Holly_4","Holly_Anc",genesGC_4)#,"chrV")
# write.csv(esrGenesGC4chrV$Sig,file="sigESRGenesGC4.csv")
# write.csv(esrGenesGC4chrV$NonSig,file="nonSigESRGenesGC4.csv")

esrGenesGC49chrV <- resultsESR(ddsDefGC_49,"Holly_49","Holly_Anc",genesGC_49)#,"chrV")
# write.csv(esrGenesGC49chrV$Sig,file="sigESRGenesGC49.csv")
# write.csv(esrGenesGC49chrV$NonSig,file="nonSigESRGenesGC49.csv")

esrGenesMA117chrV <- resultsESR(ddsDefMAnewS117,"117","Anc",genesMAnewS117)#,"chrV")
# write.csv(esrGenesMA117chrV$Sig,file="sigESRGenesMA117.csv")
# write.csv(esrGenesMA117chrV$NonSig,file="nonSigESRGenesMA117.csv")

esrGenesGC59chrVII <- resultsESR(ddsDefGC_59,"Holly_59","Holly_Anc",genesGC_59)#,"chrVII")
# write.csv(esrGenesGC59chrVII$Sig,file="sigESRGenesGC59.csv")
# write.csv(esrGenesGC59chrVII$NonSig,file="nonSigESRGenesGC59.csv")

esrGenesGC61chrVII <- resultsESR(ddsDefGC_61,"Holly_61","Holly_Anc",genesGC_61)#,"chrVII")
# write.csv(esrGenesGC61chrVII$Sig,file="sigESRGenesGC61.csv")
# write.csv(esrGenesGC61chrVII$NonSig,file="nonSigESRGenesGC61.csv")

esrGenesMA29chrIX <- resultsESR(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
# write.csv(esrGenesMA29chrIX$Sig,file="sigESRGenesMA29.csv")
# write.csv(esrGenesMA29chrIX$NonSig,file="nonSigESRGenesMA29.csv")

esrGenesMA119chrIX <- resultsESR(ddsDefMAold_119,"SC119","SCA",genesMAoldS119)#,"chrIX")
# write.csv(esrGenesMA119chrIX$Sig,file="sigESRGenesMA119.csv")
# write.csv(esrGenesMA119chrIX$NonSig,file="nonSigESRGenesMA119.csv")

esrGenesMA88chrIX <- resultsESR(ddsDefMAold_88,"SC088","SCA",genesMAoldS88)#,"chrIX")
# write.csv(esrGenesMA88chrIX$Sig,file="sigESRGenesMA88.csv")
# write.csv(esrGenesMA88chrIX$NonSig,file="nonSigESRGenesMA88.csv")

esrGenesMA108chrVIII <- resultsESR(ddsDefMAold_108,"SC108","SCA",genesMAoldS108)#,"chrIX")
# write.csv(esrGenesMA108chrVIII$Sig,file="sigESRGenesMA108.csv")
# write.csv(esrGenesMA108chrVIII$NonSig,file="nonSigESRGenesMA108.csv")
# 

esrGenesMA15chrIX <- resultsESR(ddsDefMAold_15,"SC015","SCA",genesMAoldS15)#,"chrIX")
# write.csv(esrGenesMA15chrIX$Sig,file="sigESRGenesMA15.csv")
# write.csv(esrGenesMA15chrIX$NonSig,file="nonSigESRGenesMA15.csv")

esrGenesMA29chrIX <- resultsESR(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
# write.csv(esrGenesMA29chrIX$Sig,file="sigESRGenesMA29.csv")
# write.csv(esrGenesMA29chrIX$NonSig,file="nonSigESRGenesMA29.csv")

esrGenesGC76chrIX <- resultsESR(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76)#,"chrIX")
# write.csv(esrGenesGC76chrIX$Sig,file="sigESRGenesGC76.csv")
# write.csv(esrGenesGC76chrIX$NonSig,file="nonSigESRGenesGC76.csv")

esrGenesGC77chrXII <- resultsESR(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_76)#,"chrIX")
# write.csv(esrGenesGC77chrXII$Sig,file="sigESRGenesGC77.csv")
# write.csv(esrGenesGC77chrXII$NonSig,file="nonSigESRGenesGC77.csv")

esrGenesMA9chrXIV <- resultsESR(ddsDefMAold_9,"SC009","SCA",genesMAoldS9)#,"chrXIV")
# write.csv(esrGenesMA9chrXIV$Sig,file="sigESRGenesMA9.csv")
# write.csv(esrGenesMA9chrXIV$NonSig,file="nonSigESRGenesMA9.csv")

esrGenesMA11 <- resultsESR(ddsDefMAold_11,"SC011","SCA",genesMAoldS11)#,"chrXIV")
# write.csv(esrGenesMA11$Sig,file="sigESRGenesMA11.csv")
# write.csv(esrGenesMA11$NonSig,file="nonSigESRGenesMA11.csv")

esrGenesMA112chrXVI <- resultsESR(ddsDefMAnewS112,"112","Anc",genesMAnewS112)#,"chrXVI")
# write.csv(esrGenesMA112chrXVI$Sig,file="sigESRGenesMA112.csv")
# write.csv(esrGenesMA112chrXVI$NonSig,file="nonSigESRGenesMA112.csv")

esrGenesGC8chrXVI <- resultsESR(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8)#,"chrXVI")
# write.csv(esrGenesGC8chrXVI$Sig,file="sigESRGenesGC8.csv")
# write.csv(esrGenesGC8chrXVI$NonSig,file="nonSigESRGenesGC8.csv")

### need to do the euploid samples next - ESR genes
esrGenesGC2 <- resultsESR(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2)
# write.csv(esrGenesGC2$Sig,file="sigESRGenesGC2.csv")
# write.csv(esrGenesGC2$NonSig,file="nonSigESRGenesGC2.csv")

esrGenesGC3 <- resultsESR(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3)
# write.csv(esrGenesGC3$Sig,file="sigESRGenesGC3.csv")
# write.csv(esrGenesGC3$NonSig,file="nonSigESRGenesGC3.csv")

esrGenesGC5 <- resultsESR(ddsDefGC_5,"Holly_5","Holly_Anc",genesGC_5)
# write.csv(esrGenesGC5$Sig,file="sigESRGenesGC5.csv")
# write.csv(esrGenesGC5$NonSig,file="nonSigESRGenesGC5.csv")

esrGenesGC9 <- resultsESR(ddsDefGC_9,"Holly_9","Holly_Anc",genesGC_9)
# write.csv(esrGenesGC9$Sig,file="sigESRGenesGC9.csv")
# write.csv(esrGenesGC9$NonSig,file="nonSigESRGenesGC9.csv")

esrGenesMA1 <- resultsESR(ddsDefMAold_1,"SC001","SCA",genesMAold_1)
# write.csv(esrGenesMA1$Sig,file="sigESRGenesMA1.csv")
# write.csv(esrGenesMA1$NonSig,file="nonSigESRGenesMA1.csv")

esrGenesMA2 <- resultsESR(ddsDefMAold_2,"SC002","SCA",genesMAold_2)
# write.csv(esrGenesMA2$Sig,file="sigESRGenesMA2.csv")
# write.csv(esrGenesMA2$NonSig,file="nonSigESRGenesMA2.csv")

esrGenesMA3 <- resultsESR(ddsDefMAold_3,"SC003","SCA",genesMAold_3)
# write.csv(esrGenesMA3$Sig,file="sigESRGenesMA3.csv")
# write.csv(esrGenesMA3$NonSig,file="nonSigESRGenesMA3.csv")

esrGenesMA4 <- resultsESR(ddsDefMAold_4,"SC004","SCA",genesMAold_4)
# write.csv(esrGenesMA4$Sig,file="sigESRGenesMA4.csv")
# write.csv(esrGenesMA4$NonSig,file="nonSigESRGenesMA4.csv")

esrGenesMA5 <- resultsESR(ddsDefMAold_5,"SC005","SCA",genesMAold_5)
# write.csv(esrGenesMA5$Sig,file="sigESRGenesMA5.csv")
# write.csv(esrGenesMA5$NonSig,file="nonSigESRGenesMA5.csv")

esrGenesMA7 <- resultsESR(ddsDefMAold_7,"SC007","SCA",genesMAold_7)
# write.csv(esrGenesMA7$Sig,file="sigESRGenesMA7.csv")
# write.csv(esrGenesMA7$NonSig,file="nonSigESRGenesMA7.csv")

esrGenesMA8 <- resultsESR(ddsDefMAold_8,"SC008","SCA",genesMAold_8)
# write.csv(esrGenesMA8$Sig,file="sigESRGenesMA8.csv")
# write.csv(esrGenesMA8$NonSig,file="nonSigESRGenesMA8.csv")

esrGenesMA50 <- resultsESR(ddsDefMAnewS50,"50","Anc",genesMAnewS50)
# write.csv(esrGenesMA50$Sig,file="sigESRGenesMA50.csv")
# write.csv(esrGenesMA50$NonSig,file="nonSigESRGenesMA50.csv")

esrGenesMA115 <- resultsESR(ddsDefMAnewS115,"115","Anc",genesMAnewS115)
# write.csv(esrGenesMA115$Sig,file="sigESRGenesMA115.csv")
# write.csv(esrGenesMA115$NonSig,file="nonSigESRGenesMA115.csv")
```


ASR genes scripts 

```{r ASRgenes, include=FALSE,echo=FALSE,eval=TRUE}
### see if they are in ASR
resultsASR <- function(dds,sample,anc,genes) {
# make a new dds data set that we can use in future analyses
   geneID <- rownames(dds)
  # #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  # #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
   genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
# determine if any significantly differentially expressed genes match those identified by gasch et al 2000
  ASR <- list()
  ASR$Sig <- dplyr::inner_join(ASRgenes,resDiSig,by=c("GENEID"="GENEID"))
  ASR$NonSig <- dplyr::inner_join(ASRgenes,resDiNonSig,by=c("GENEID"="GENEID"))
  return(ASR)
}

ASRGenesMA152 <- resultsASR(ddsDefMAnewS152,"152","Anc",genesMAnewS152)#,"chrI")
asrGenesSigMA152 <- ASRGenesMA152$Sig
# write.csv(ASRGenesMA152$Sig,file="sigASRGenesMA152.csv")
# write.csv(ASRGenesMA152$NonSig,file="nonSigASRGenesMA152.csv")
ASRGenesGC11 <- resultsASR(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11)#,"chrI")
asrGenesSigGC11 <- ASRGenesGC11$Sig
# write.csv(ASRGenesGC11$Sig,file="sigASRGenesGC11.csv")
# write.csv(ASRGenesGC11$NonSig,file="nonSigASRGenesGC11.csv")

ASRGenesGC18chrI <- resultsASR(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18)#,"chrI")
asrGenesSigGC18 <- ASRGenesGC18chrI$Sig
# write.csv(ASRGenesGC18chrI$Sig,file="sigASRGenesGC18.csv")
# write.csv(ASRGenesGC18chrI$NonSig,file="nonSigASRGenesGC18.csv")

ASRGenesGC7chrI <- resultsASR(ddsDefGC_7,"Holly_7","Holly_Anc",genesGC_7)#,"chrI")
asrGenesSigGC7 <- ASRGenesGC7chrI$Sig
# write.csv(ASRGenesGC7chrI$Sig,file="sigASRGenesGC7.csv")
# write.csv(ASRGenesGC7chrI$NonSig,file="nonSigASRGenesGC7.csv")

ASRGenesMA123chrV <- resultsASR(ddsDefMAnewS123,"123","Anc",genesMAnewS123)#,"chrV")
asrGenesSigMA123 <- ASRGenesMA123chrV$Sig
# write.csv(ASRGenesMA123chrV$Sig,file="sigASRGenesMA123.csv")
# write.csv(ASRGenesMA123chrV$NonSig,file="nonSigASRGenesMA123.csv")

ASRGenesGC4chrV <- resultsASR(ddsDefGC_4,"Holly_4","Holly_Anc",genesGC_4)#,"chrV")
asrGenesSigGC4 <- ASRGenesGC4chrV$Sig
# write.csv(ASRGenesGC4chrV$Sig,file="sigASRGenesGC4.csv")
# write.csv(ASRGenesGC4chrV$NonSig,file="nonSigASRGenesGC4.csv")

ASRGenesGC49chrV <- resultsASR(ddsDefGC_49,"Holly_49","Holly_Anc",genesGC_49)#,"chrV")
asrGenesSigGC49 <- ASRGenesGC49chrV$Sig
# write.csv(ASRGenesGC49chrV$Sig,file="sigASRGenesGC49.csv")
# write.csv(ASRGenesGC49chrV$NonSig,file="nonSigASRGenesGC49.csv")

ASRGenesMA117chrV <- resultsASR(ddsDefMAnewS117,"117","Anc",genesMAnewS117)#,"chrV")
asrGenesSigMA117 <- ASRGenesMA117chrV$Sig
# write.csv(ASRGenesMA117chrV$Sig,file="sigASRGenesMA117.csv")
# write.csv(ASRGenesMA117chrV$NonSig,file="nonSigASRGenesMA117.csv")

ASRGenesGC59chrVII <- resultsASR(ddsDefGC_59,"Holly_59","Holly_Anc",genesGC_59)#,"chrVII")
asrGenesSigGC59 <- ASRGenesGC59chrVII$Sig
# write.csv(ASRGenesGC59chrVII$Sig,file="sigASRGenesGC59.csv")
# write.csv(ASRGenesGC59chrVII$NonSig,file="nonSigASRGenesGC59.csv")

ASRGenesGC61chrVII <- resultsASR(ddsDefGC_61,"Holly_61","Holly_Anc",genesGC_61)#,"chrVII")
asrGenesSigGC61 <- ASRGenesGC61chrVII$Sig
# write.csv(ASRGenesGC61chrVII$Sig,file="sigASRGenesGC61.csv")
# write.csv(ASRGenesGC61chrVII$NonSig,file="nonSigASRGenesGC61.csv")

ASRGenesMA29chrIX <- resultsASR(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
asrGenesSigMA29 <- ASRGenesMA29chrIX$Sig
# write.csv(ASRGenesMA29chrIX$Sig,file="sigASRGenesMA29.csv")
# write.csv(ASRGenesMA29chrIX$NonSig,file="nonSigASRGenesMA29.csv")

ASRGenesMA119chrIX <- resultsASR(ddsDefMAold_119,"SC119","SCA",genesMAoldS119)#,"chrIX")
asrGenesSigMA119 <- ASRGenesMA119chrIX$Sig
# write.csv(ASRGenesMA119chrIX$Sig,file="sigASRGenesMA119.csv")
# write.csv(ASRGenesMA119chrIX$NonSig,file="nonSigASRGenesMA119.csv")

ASRGenesMA88chrIX <- resultsASR(ddsDefMAold_88,"SC088","SCA",genesMAoldS88)#,"chrIX")
asrGenesSigMA88 <- ASRGenesMA88chrIX$Sig
# write.csv(ASRGenesMA88chrIX$Sig,file="sigASRGenesMA88.csv")
# write.csv(ASRGenesMA88chrIX$NonSig,file="nonSigASRGenesMA88.csv")

ASRGenesMA108chrVIII <- resultsASR(ddsDefMAold_108,"SC108","SCA",genesMAoldS108)#,"chrIX")
asrGenesSigMA108 <- ASRGenesMA108chrVIII$Sig
# write.csv(ASRGenesMA108chrVIII$Sig,file="sigASRGenesMA108.csv")
# write.csv(ASRGenesMA108chrVIII$NonSig,file="nonSigASRGenesMA108.csv")


ASRGenesMA15chrIX <- resultsASR(ddsDefMAold_15,"SC015","SCA",genesMAoldS15)#,"chrIX")
asrGenesSigMA15 <- ASRGenesMA15chrIX$Sig
# write.csv(ASRGenesMA15chrIX$Sig,file="sigASRGenesMA15.csv")
# write.csv(ASRGenesMA15chrIX$NonSig,file="nonSigASRGenesMA15.csv")

ASRGenesMA29chrIX <- resultsASR(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
asrGenesSigMA29 <- ASRGenesMA29chrIX$Sig
# write.csv(ASRGenesMA29chrIX$Sig,file="sigASRGenesMA29.csv")
# write.csv(ASRGenesMA29chrIX$NonSig,file="nonSigASRGenesMA29.csv")

ASRGenesGC76chrIX <- resultsASR(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76)#,"chrIX")
asrGenesSigGC76 <- ASRGenesGC76chrIX$Sig
# write.csv(ASRGenesGC76chrIX$Sig,file="sigASRGenesGC76.csv")
# write.csv(ASRGenesGC76chrIX$NonSig,file="nonSigASRGenesGC76.csv")

ASRGenesGC77chrXII <- resultsASR(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_76)#,"chrIX")
asrGenesSigGC77 <- ASRGenesGC77chrXII$Sig
# write.csv(ASRGenesGC77chrXII$Sig,file="sigASRGenesGC77.csv")
# write.csv(ASRGenesGC77chrXII$NonSig,file="nonSigASRGenesGC77.csv")

ASRGenesMA9chrXIV <- resultsASR(ddsDefMAold_9,"SC009","SCA",genesMAoldS9)#,"chrXIV")
asrGenesSigMA9 <- ASRGenesMA9chrXIV$Sig
# write.csv(ASRGenesMA9chrXIV$Sig,file="sigASRGenesMA9.csv")
# write.csv(ASRGenesMA9chrXIV$NonSig,file="nonSigASRGenesMA9.csv")

ASRGenesMA11 <- resultsASR(ddsDefMAold_11,"SC011","SCA",genesMAoldS11)#,"chrXIV")
asrGenesSigMA11 <- ASRGenesMA11$Sig
# write.csv(ASRGenesMA11$Sig,file="sigASRGenesMA11.csv")
# write.csv(ASRGenesMA11$NonSig,file="nonSigASRGenesMA11.csv")

ASRGenesMA112chrXVI <- resultsASR(ddsDefMAnewS112,"112","Anc",genesMAnewS112)#,"chrXVI")
asrGenesSigMA112 <- ASRGenesMA112chrXVI$Sig
# write.csv(ASRGenesMA112chrXVI$Sig,file="sigASRGenesMA112.csv")
# write.csv(ASRGenesMA112chrXVI$NonSig,file="nonSigASRGenesMA112.csv")

ASRGenesGC8chrXVI <- resultsASR(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8)#,"chrXVI")
asrGenesSigGC8 <- ASRGenesGC8chrXVI$Sig
# write.csv(ASRGenesGC8chrXVI$Sig,file="sigASRGenesGC8.csv")
# write.csv(ASRGenesGC8chrXVI$NonSig,file="nonSigASRGenesGC8.csv")

### need to do the euploid samples next - ASR genes

ASRGenesGC1 <- resultsASR(ddsDefGC_1,"Holly_1","Holly_Anc",genesGC_1)
asrGenesSigGC1 <- ASRGenesGC1$Sig
# write.csv(ASRGenesGC1$Sig,file="sigASRGenesGC1.csv")
# write.csv(ASRGenesGC1$NonSig,file="nonSigASRGenesGC1.csv")

ASRGenesGC2 <- resultsASR(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2)
asrGenesSigGC2 <- ASRGenesGC2$Sig
# write.csv(ASRGenesGC2$Sig,file="sigASRGenesGC2.csv")
# write.csv(ASRGenesGC2$NonSig,file="nonSigASRGenesGC2.csv")

ASRGenesGC3 <- resultsASR(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3)
asrGenesSigGC3 <- ASRGenesGC3$Sig
# write.csv(ASRGenesGC3$Sig,file="sigASRGenesGC3.csv")
# write.csv(ASRGenesGC3$NonSig,file="nonSigASRGenesGC3.csv")

ASRGenesGC5 <- resultsASR(ddsDefGC_5,"Holly_5","Holly_Anc",genesGC_5)
asrGenesSigGC5 <- ASRGenesGC5$Sig
# write.csv(ASRGenesGC5$Sig,file="sigASRGenesGC5.csv")
# write.csv(ASRGenesGC5$NonSig,file="nonSigASRGenesGC5.csv")

ASRGenesGC9 <- resultsASR(ddsDefGC_9,"Holly_9","Holly_Anc",genesGC_9)
asrGenesSigGC9 <- ASRGenesGC9$Sig
# write.csv(ASRGenesGC9$Sig,file="sigASRGenesGC9.csv")
# write.csv(ASRGenesGC9$NonSig,file="nonSigASRGenesGC9.csv")

ASRGenesMA1 <- resultsASR(ddsDefMAold_1,"SC001","SCA",genesMAold_1)
asrGenesSigMA1 <- ASRGenesMA1$Sig
# write.csv(ASRGenesMA1$Sig,file="sigASRGenesMA1.csv")
# write.csv(ASRGenesMA1$NonSig,file="nonSigASRGenesMA1.csv")

ASRGenesMA2 <- resultsASR(ddsDefMAold_2,"SC002","SCA",genesMAold_2)
asrGenesSigMA2 <- ASRGenesMA2$Sig
# write.csv(ASRGenesMA2$Sig,file="sigASRGenesMA2.csv")
# write.csv(ASRGenesMA2$NonSig,file="nonSigASRGenesMA2.csv")

ASRGenesMA3 <- resultsASR(ddsDefMAold_3,"SC003","SCA",genesMAold_3)
asrGenesSigMA3 <- ASRGenesMA3$Sig
# write.csv(ASRGenesMA3$Sig,file="sigASRGenesMA3.csv")
# write.csv(ASRGenesMA3$NonSig,file="nonSigASRGenesMA3.csv")

ASRGenesMA4 <- resultsASR(ddsDefMAold_4,"SC004","SCA",genesMAold_4)
asrGenesSigMA4 <- ASRGenesMA4$Sig
# write.csv(ASRGenesMA4$Sig,file="sigASRGenesMA4.csv")
# write.csv(ASRGenesMA4$NonSig,file="nonSigASRGenesMA4.csv")

ASRGenesMA5 <- resultsASR(ddsDefMAold_5,"SC005","SCA",genesMAold_5)
asrGenesSigMA5 <- ASRGenesMA5$Sig
# write.csv(ASRGenesMA5$Sig,file="sigASRGenesMA5.csv")
# write.csv(ASRGenesMA5$NonSig,file="nonSigASRGenesMA5.csv")

ASRGenesMA7 <- resultsASR(ddsDefMAold_7,"SC007","SCA",genesMAold_7)
asrGenesSigMA7 <- ASRGenesMA7$Sig
# write.csv(ASRGenesMA7$Sig,file="sigASRGenesMA7.csv")
# write.csv(ASRGenesMA7$NonSig,file="nonSigASRGenesMA7.csv")

ASRGenesMA8 <- resultsASR(ddsDefMAold_8,"SC008","SCA",genesMAold_8)
asrGenesSigMA8 <- ASRGenesMA8$Sig
# write.csv(ASRGenesMA8$Sig,file="sigASRGenesMA8.csv")
# write.csv(ASRGenesMA8$NonSig,file="nonSigASRGenesMA8.csv")

ASRGenesMA50 <- resultsASR(ddsDefMAnewS50,"50","Anc",genesMAnewS50)
asrGenesSigMA50 <- ASRGenesMA50$Sig
# write.csv(ASRGenesMA50$Sig,file="sigASRGenesMA50.csv")
# write.csv(ASRGenesMA50$NonSig,file="nonSigASRGenesMA50.csv")

ASRGenesMA115 <- resultsASR(ddsDefMAnewS115,"115","Anc",genesMAnewS115)
asrGenesSigMA115 <- ASRGenesMA115$Sig
# write.csv(ASRGenesMA115$Sig,file="sigASRGenesMA115.csv")
# write.csv(ASRGenesMA115$NonSig,file="nonSigASRGenesMA115.csv")


asrGenesCommGCEup <- Reduce(intersect, list(asrGenesSigGC1,asrGenesSigGC2,asrGenesSigGC3,asrGenesSigGC5,asrGenesSigGC9))
asrGenesCommGCEup
## no commonly found sig DE ASR genes in all GC euploids

asrGenesCommMAoldEup <- Reduce(intersect, list(asrGenesSigMA1,asrGenesSigMA2,asrGenesSigMA3,asrGenesSigMA4,asrGenesSigMA5,asrGenesSigMA7,asrGenesSigMA8))
asrGenesCommMAoldEup

asrGenesCommMAnewEup <- Reduce(intersect,list(asrGenesSigMA50,asrGenesSigMA115))
# also 0

### see if there are any ASR genes commonly found in All Aneuploid GC samples
asrGenesCommGCAneu <- Reduce(intersect,list(asrGenesSigGC11,asrGenesSigGC18,asrGenesSigGC4,asrGenesSigGC49,asrGenesSigGC59,asrGenesSigGC61,asrGenesSigGC7,asrGenesSigGC76,asrGenesSigGC77,asrGenesSigGC8))
# no commonly sig DE ASR genes in aneuploid GC samples

##old ma samples
asrGenesCommMAOldAneu <- Reduce(intersect,list(asrGenesSigMA108,asrGenesSigMA119,asrGenesSigMA15,asrGenesSigMA88))

asrGenesCommMANewAneu <- Reduce(intersect,list(asrGenesSigMA112,asrGenesSigMA117,asrGenesSigMA123,asrGenesSigMA152,asrGenesSigMA29))

```


Dosage sensitive genes scripts 

```{r dosSensGenes, include=FALSE,echo=FALSE,eval=TRUE}
## then do all samples - dosage Sensitive Genes
resultsDosSens <- function(dds,sample,anc,genes) {
# make a new dds data set that we can use in future analyses
   geneID <- rownames(dds)
  # #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  # #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
   genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
# subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
# determine if any significantly differentially expressed genes match those identified by gasch et al 2000
  dosSensRes <- list()
  dosSensRes$Sig <- dplyr::inner_join(dosageSensitiveGenes,resDiSig,by=c("GENEID"="GENEID"))
  dosSensRes$NonSig <- dplyr::inner_join(dosageSensitiveGenes,resDiNonSig,by=c("GENEID"="GENEID"))
  return(dosSensRes)
}


dosSensGenesMA152 <- resultsDosSens(ddsDefMAnewS152,"152","Anc",genesMAnewS152)#,"chrI")
dosSensGenesSigMA152 <- dosSensGenesMA152$Sig
dosSensGenesNonSigMA152 <- dosSensGenesMA152$NonSig

# write.csv(dosSensGenesMA152$Sig,file="sigEdosSensGenesMA152.csv")
# write.csv(dosSensGenesMA152$NonSig,file="nonSigdosSensGenesMA152.csv")

dosSensGenesGC11 <- resultsDosSens(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11)#,"chrI")
dosSensGenesSigGC11 <- dosSensGenesGC11$Sig
# write.csv(dosSensGenesGC11$Sig,file="sigdosSensGenesGC11.csv")
# write.csv(dosSensGenesGC11$NonSig,file="nondosSensGenesGC11.csv")

dosSensGenesGC18chrI <- resultsDosSens(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18)#,"chrI")
dosSensGenesSigGC18 <- dosSensGenesGC18chrI$Sig
# write.csv(dosSensGenesGC18chrI$Sig,file="sigdosSensGenesGC18.csv")
# write.csv(dosSensGenesGC18chrI$NonSig,file="nonSigdosSensGenesGC18.csv")

dosSensGenesGC7chrI <- resultsDosSens(ddsDefGC_7,"Holly_7","Holly_Anc",genesGC_7)#,"chrI")
dosSensGenesSigGC7 <- dosSensGenesGC7chrI$Sig
# write.csv(dosSensGenesGC7chrI$Sig,file="sigdosSensGenesGC7.csv")
# write.csv(dosSensGenesGC7chrI$NonSig,file="nonSigdosSensGenesGC7.csv")

dosSensGenesMA123chrV <- resultsDosSens(ddsDefMAnewS123,"123","Anc",genesMAnewS123)#,"chrV")
dosSensGenesSigMA123 <- dosSensGenesMA123chrV$Sig
# write.csv(dosSensGenesMA123chrV$Sig,file="sigdosSensGenesMA123.csv")
# write.csv(dosSensGenesMA123chrV$NonSig,file="nonSigdosSensGenesMA123.csv")

dosSensGenesGC4chrV <- resultsDosSens(ddsDefGC_4,"Holly_4","Holly_Anc",genesGC_4)#,"chrV")
dosSensGenesSigGC4 <- dosSensGenesGC4chrV$Sig
# write.csv(dosSensGenesGC4chrV$Sig,file="sigdosSensGenesGC4.csv")
# write.csv(dosSensGenesGC4chrV$NonSig,file="nonSigdosSensGenesGC4.csv")

dosSensGenesGC49chrV <- resultsDosSens(ddsDefGC_49,"Holly_49","Holly_Anc",genesGC_49)#,"chrV")
dosSensGenesSigGC49 <- dosSensGenesGC49chrV$Sig
# write.csv(dosSensGenesGC49chrV$Sig,file="sigdosSensGenesGC49.csv")
# write.csv(dosSensGenesGC49chrV$NonSig,file="nonSigdosSensGenesGC49.csv")

dosSensGenesMA117chrV <- resultsDosSens(ddsDefMAnewS117,"117","Anc",genesMAnewS117)#,"chrV")
dosSensGenesSigMA117 <- dosSensGenesMA117chrV$Sig
# write.csv(dosSensGenesMA117chrV$Sig,file="sigdosSensGenesMA117.csv")
# write.csv(dosSensGenesMA117chrV$NonSig,file="nonSigdosSensGenesMA117.csv")

dosSensGenesGC59chrVII <- resultsDosSens(ddsDefGC_59,"Holly_59","Holly_Anc",genesGC_59)#,"chrVII")
dosSensGenesSigGC59 <- dosSensGenesGC59chrVII$Sig
# write.csv(dosSensGenesGC59chrVII$Sig,file="sigdosSensGenesGC59.csv")
# write.csv(dosSensGenesGC59chrVII$NonSig,file="nonSigdosSensGenesGC59.csv")

dosSensGenesGC61chrVII <- resultsDosSens(ddsDefGC_61,"Holly_61","Holly_Anc",genesGC_61)#,"chrVII")
dosSensGenesSigGC61 <- dosSensGenesGC61chrVII$Sig
# write.csv(dosSensGenesGC61chrVII$Sig,file="sigdosSensGenesGC61.csv")
# write.csv(dosSensGenesGC61chrVII$NonSig,file="nonSigdosSensGenesGC61.csv")

dosSensGenesMA29chrIX <- resultsDosSens(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
dosSensGenesSigMA29 <- dosSensGenesMA29chrIX$Sig
# write.csv(dosSensGenesMA29chrIX$Sig,file="sigdosSensGenesMA29.csv")
# write.csv(dosSensGenesMA29chrIX$NonSig,file="nonSigdosSensGenesMA29.csv")

dosSensGenesMA119chrIX <- resultsDosSens(ddsDefMAold_119,"SC119","SCA",genesMAoldS119)#,"chrIX")
dosSensGenesSigMA119 <- dosSensGenesMA119chrIX$Sig
# write.csv(dosSensGenesMA119chrIX$Sig,file="sigdosSensGenesMA119.csv")
# write.csv(dosSensGenesMA119chrIX$NonSig,file="nonSigdosSensGenesMA119.csv")

dosSensGenesMA88chrIX <- resultsDosSens(ddsDefMAold_88,"SC088","SCA",genesMAoldS88)#,"chrIX")
dosSensGenesSigMA88 <- dosSensGenesMA88chrIX$Sig
# write.csv(dosSensGenesMA88chrIX$Sig,file="sigdosSensGenesMA88.csv")
# write.csv(dosSensGenesMA88chrIX$NonSig,file="nonSigdosSensGenesMA88.csv")

dosSensGenesMA108chrVIII <- resultsDosSens(ddsDefMAold_108,"SC108","SCA",genesMAoldS108)#,"chrIX")
dosSensGenesSigMA108 <- dosSensGenesMA108chrVIII$Sig
# write.csv(dosSensGenesMA108chrVIII$Sig,file="sigdosSensGenesMA108.csv")
# write.csv(dosSensGenesMA108chrVIII$NonSig,file="nonSigdosSensGenesMA108.csv")

dosSensGenesMA15chrIX <- resultsDosSens(ddsDefMAold_15,"SC015","SCA",genesMAoldS15)#,"chrIX")
dosSensGenesSigMA15 <- dosSensGenesMA15chrIX$Sig
# write.csv(dosSensGenesMA15chrIX$Sig,file="sigdosSensGenesMA15.csv")
# write.csv(dosSensGenesMA15chrIX$NonSig,file="nonSigdosSensGenesMA15.csv")

dosSensGenesMA29chrIX <- resultsDosSens(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
dosSensGenesSigMA29 <- dosSensGenesMA29chrIX$Sig
# write.csv(dosSensGenesMA29chrIX$Sig,file="sigdosSensGenesMA29.csv")
# write.csv(dosSensGenesMA29chrIX$NonSig,file="nonSigdosSensGenesMA29.csv")

dosSensGenesGC76chrIX <- resultsDosSens(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76)#,"chrIX")
dosSensGenesSigGC76 <- dosSensGenesGC76chrIX$Sig
# write.csv(dosSensGenesGC76chrIX$Sig,file="sigdosSensGenesGC76.csv")
# write.csv(dosSensGenesGC76chrIX$NonSig,file="nonSigdosSensGenesGC76.csv")

dosSensGenesGC77chrXII <- resultsDosSens(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_76)#,"chrIX")
dosSensGenesSigGC77 <- dosSensGenesGC77chrXII$Sig
# write.csv(dosSensGenesGC77chrXII$Sig,file="sigdosSensGenesGC77.csv")
# write.csv(dosSensGenesGC77chrXII$NonSig,file="nonSigdosSensGenesGC77.csv")

dosSensGenesMA9chrXIV <- resultsDosSens(ddsDefMAold_9,"SC009","SCA",genesMAoldS9)#,"chrXIV")
dosSensGenesSigMA9 <- dosSensGenesMA9chrXIV$Sig
# write.csv(dosSensGenesMA9chrXIV$Sig,file="sigdosSensGenesMA9.csv")
# write.csv(dosSensGenesMA9chrXIV$NonSig,file="nonSigdosSensGenesMA9.csv")

dosSensGenesMA11 <- resultsDosSens(ddsDefMAold_11,"SC011","SCA",genesMAoldS11)#,"chrXIV")
dosSensGenesSigMA11 <- dosSensGenesMA11$Sig
# write.csv(dosSensGenesMA11$Sig,file="sigdosSensGenesMA11.csv")
# write.csv(dosSensGenesMA11$NonSig,file="nonSigdosSensGenesMA11.csv")

dosSensGenesMA112chrXVI <- resultsDosSens(ddsDefMAnewS112,"112","Anc",genesMAnewS112)#,"chrXVI")
dosSensGenesSigMA112 <- dosSensGenesMA112chrXVI$Sig
# write.csv(dosSensGenesMA112chrXVI$Sig,file="sigdosSensGenesMA112.csv")
# write.csv(dosSensGenesMA112chrXVI$NonSig,file="nonSigdosSensGenesMA112.csv")

dosSensGenesGC8chrXVI <- resultsDosSens(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8)#,"chrXVI")
dosSensGenesSigGC8 <- dosSensGenesGC8chrXVI$Sig
# write.csv(dosSensGenesGC8chrXVI$Sig,file="sigdosSensGenesGC8.csv")
# write.csv(dosSensGenesGC8chrXVI$NonSig,file="nonSigdosSensGenesGC8.csv")

### need to do the euploid samples next - dosage sensitive genes
dosSensGenesGC1 <- resultsDosSens(ddsDefGC_1,"Holly_1","Holly_Anc",genesGC_1)
dosSensGenesSigGC1 <- dosSensGenesGC1$Sig

dosSensGenesGC2 <- resultsDosSens(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2)
dosSensGenesSigGC2 <- dosSensGenesGC2$Sig
# write.csv(dosSensGenesGC2$Sig,file="sigdosSensGenesGC2.csv")
# write.csv(dosSensGenesGC2$NonSig,file="nonSigdosSensGenesGC2.csv")

dosSensGenesGC3 <- resultsDosSens(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3)
dosSensGenesSigGC3 <- dosSensGenesGC3$Sig
# write.csv(dosSensGenesGC3$Sig,file="sigdosSensGenesGC3.csv")
# write.csv(dosSensGenesGC3$NonSig,file="nonSigdosSensGenesGC3.csv")

dosSensGenesGC5 <- resultsDosSens(ddsDefGC_5,"Holly_5","Holly_Anc",genesGC_5)
dosSensGenesSigGC5 <- dosSensGenesGC5$Sig
# write.csv(dosSensGenesGC5$Sig,file="sigdosSensGenesGC5.csv")
# write.csv(dosSensGenesGC5$NonSig,file="nonSigdosSensGenesGC5.csv")

dosSensGenesGC9 <- resultsDosSens(ddsDefGC_9,"Holly_9","Holly_Anc",genesGC_9)
dosSensGenesSigGC9 <- dosSensGenesGC9$Sig
# write.csv(dosSensGenesGC9$Sig,file="sigdosSensGenesGC9.csv")
# write.csv(dosSensGenesGC9$NonSig,file="nonSigdosSensGenesGC9.csv")

dosSensGenesMA1 <- resultsDosSens(ddsDefMAold_1,"SC001","SCA",genesMAold_1)
dosSensGenesSigMA1 <- dosSensGenesMA1$Sig
# write.csv(dosSensGenesMA1$Sig,file="sigdosSensGenesMA1.csv")
# write.csv(dosSensGenesMA1$NonSig,file="nonSigdosSensGenesMA1.csv")

dosSensGenesMA2 <- resultsDosSens(ddsDefMAold_2,"SC002","SCA",genesMAold_2)
dosSensGenesSigMA2 <- dosSensGenesMA2$Sig
# write.csv(dosSensGenesMA2$Sig,file="sigdosSensGenesMA2.csv")
# write.csv(dosSensGenesMA2$NonSig,file="nonSigdosSensGenesMA2.csv")

dosSensGenesMA3 <- resultsDosSens(ddsDefMAold_3,"SC003","SCA",genesMAold_3)
dosSensGenesSigMA3 <- dosSensGenesMA3$Sig
# write.csv(dosSensGenesMA3$Sig,file="sigdosSensGenesMA3.csv")
# write.csv(dosSensGenesMA3$NonSig,file="nonSigdosSensGenesMA3.csv")

dosSensGenesMA4 <- resultsDosSens(ddsDefMAold_4,"SC004","SCA",genesMAold_4)
dosSensGenesSigMA4 <- dosSensGenesMA4$Sig
# write.csv(dosSensGenesMA4$Sig,file="sigdosSensGenesMA4.csv")
# write.csv(dosSensGenesMA4$NonSig,file="nonSigdosSensGenesMA4.csv")

dosSensGenesMA5 <- resultsDosSens(ddsDefMAold_5,"SC005","SCA",genesMAold_5)
dosSensGenesSigMA5 <- dosSensGenesMA5$Sig
# write.csv(dosSensGenesMA5$Sig,file="sigdosSensGenesMA5.csv")
# write.csv(dosSensGenesMA5$NonSig,file="nonSigdosSensGenesMA5.csv")

dosSensGenesMA7 <- resultsDosSens(ddsDefMAold_7,"SC007","SCA",genesMAold_7)
dosSensGenesSigMA7 <- dosSensGenesMA7$Sig
# write.csv(dosSensGenesMA7$Sig,file="sigdosSensGenesMA7.csv")
# write.csv(dosSensGenesMA7$NonSig,file="nonSigdosSensGenesMA7.csv")

dosSensGenesMA8 <- resultsDosSens(ddsDefMAold_8,"SC008","SCA",genesMAold_8)
dosSensGenesSigMA8 <- dosSensGenesMA8$Sig
# write.csv(dosSensGenesMA8$Sig,file="sigdosSensGenesMA8.csv")
# write.csv(dosSensGenesMA8$NonSig,file="nonSigdosSensGenesMA8.csv")

dosSensGenesMA50 <- resultsDosSens(ddsDefMAnewS50,"50","Anc",genesMAnewS50)
dosSensGenesSigMA50 <-  dosSensGenesMA50$Sig
# write.csv(dosSensGenesMA50$Sig,file="sigdosSensGenesMA50.csv")
# write.csv(dosSensGenesMA50$NonSig,file="nonSigdosSensGenesMA50.csv")

dosSensResMA115 <- resultsDosSens(ddsDefMAnewS115,"115","Anc",genesMAnewS115)
dosSensGenesSigMA115 <- dosSensResMA115$Sig
# write.csv(dosSensResMA115$Sig,file="sigDosSensGenesMA115.csv")
# write.csv(dosSensResMA115$NonSig,file="nonSigDosSensGenesMA115.csv")


### find commonly DE dosage Sensitive genes in euploid GC samples
dosSensGenesCommGCEup <- Reduce(intersect, list(dosSensGenesSigGC1,dosSensGenesSigGC2,dosSensGenesSigGC3,dosSensGenesSigGC5,dosSensGenesSigGC9))
dosSensGenesCommGCEup
## no commonly found sig DE dosage Sensitive genes in all GC euploids

dosSensGenesCommMAoldEup <- Reduce(intersect, list(dosSensGenesSigMA1,dosSensGenesSigMA2,dosSensGenesSigMA3,dosSensGenesSigMA4,dosSensGenesSigMA5,dosSensGenesSigMA7,dosSensGenesSigMA8))
dosSensGenesCommMAoldEup

dosSensGenesCommMAnewEup <- Reduce(intersect,list(dosSensGenesSigMA50,dosSensGenesSigMA115))
# also 0

### see if there are any ASR genes commonly found in All Aneuploid GC samples
dosSensGenesCommGCAneu <- Reduce(intersect,list(dosSensGenesSigGC11,dosSensGenesSigGC18,dosSensGenesSigGC4,dosSensGenesSigGC49,dosSensGenesSigGC59,dosSensGenesSigGC61,dosSensGenesSigGC7,dosSensGenesSigGC76,dosSensGenesSigGC77,dosSensGenesSigGC8))
# no commonly sig DE ASR genes in aneuploid GC samples

##old ma samples
dosSensGenesCommMAOldAneu <- Reduce(intersect,list(dosSensGenesSigMA108,dosSensGenesSigMA119,dosSensGenesSigMA15,dosSensGenesSigMA88))

dosSensGenesCommMANewAneu <- Reduce(intersect,list(dosSensGenesSigMA112,dosSensGenesSigMA117,dosSensGenesSigMA123,dosSensGenesSigMA152,dosSensGenesSigMA29))

```



Below is a script for differentiating genes that are differentially expressed from those that are not dosage compensated, dosage compensated, anti DC, over DC, unknown, or partially dosage compensated. I used the log fold change threshold function of the results function to isolate these genes. FDR p values < 0.1 were considered statistically significant.
```{r trisomy, eval=TRUE, echo=FALSE}
resultsTrisomy <- function(dds,sample,anc,genes,anChr) {
  # make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  genes
  keep <- anChr
  genes <- genes[(grep(paste0(keep,"$"),genes$TXCHROM,perl=TRUE)),]
  genes
  all(rownames(dds) == genes$GENEID)
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  ddsGenes <- intersect(rownames(dds),genes)
  as.data.frame(genes)
  str(genes)
  # ddsGenes <- genes[,ddsGenes]
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  resTri <- results(dds,contrast=c("condition",sample,anc),lfcThreshold=0.585)
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (either no DC or unknown))
  resTriNonSig <- as.data.frame(subset(resTri,padj>0.1))
  length(resTriNonSig)
  resTriNonSig <- cbind(names = rownames(resTriNonSig), resTriNonSig)
  resTriNonSig <- dplyr::inner_join(genes,resTriNonSig,by=c("GENEID"="names"))
  str(resTriNonSig)
  # View(resTriNonSig)
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resTriSig <- as.data.frame(subset(resTri,padj<0.1))
  resTriSig <- cbind(names = rownames(resTriSig), resTriSig)
  resTriSig <- dplyr::inner_join(genes,resTriSig,by=c("GENEID"="names"))
  str(resTriSig)
  #### run same analyses using new ddsCopy dataset and new parameters
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (either Full DC or unknown genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  str(resDiNonSig)
  # subset data frame to get Significant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  str(resDiSig)
  #####################################
  ### Find genes that are Signficiantly different than Trisomic expectation and 
  # are Significantly different from Disomic expectation
  results <- list()
  resSigBoth<- dplyr::inner_join(resTriSig,resDiSig,by="GENEID")
  ## over DC genes for trisomy : less than 0 (1x)
  over.4 <- resSigBoth[which(resSigBoth$log2FoldChange.x<0),]
  ## anti DC genes for trisomy: more than 0.585 (1.5x)
  anti.5 <- resSigBoth[which(resSigBoth$log2FoldChange.x > 0.585),]
  results$over.4 <- over.4
  results$anti.5 <- anti.5
  ### Find genes that are Signficiantly different than Trisomic expectation but
  # NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resTriSig,resDiNonSig,by="GENEID")
  results$full.3 <- resFullDC
  ## Find genes that are not significantly diff from trisomic exp but sig diff from disomic
  resPartNo <- dplyr::inner_join(resTriNonSig,resDiSig,by="GENEID")
  resPart <- resPartNo[which(resPartNo$log2FoldChange.x < 0.585),]
  results$Part.2 <- resPart
  resNo <- resPartNo[which(resPartNo$log2FoldChange.x >= 0.585),]
  results$No.1 <- resNo
  ## Get genes that are not significantly different from either Trisomic or Disomic expectation
  resUnknown <- dplyr::inner_join(resTriNonSig,resDiNonSig,by="GENEID")
  results$Unknown.0 <- resUnknown
  return(results)
}

# input is dds for that sample, line #, ancestor name, genes matrix for that sample, and chromosome of interest
# output is list with full, part, over, anti, no, and unknown DC genes 

```

```{r results with specific chrm, includ=TRUE,eval=TRUE,echo=FALSE}
volcanoPlots <- function(dds,sample,anc,genes,anChr,FC) {
  # make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  keep <- anChr
  genes <- genes[(grep(paste0(keep,"$"),genes$TXCHROM,perl=TRUE)),]
  all(rownames(dds) == genes$GENEID)
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  ddsGenes <- intersect(rownames(dds),genes)
  as.data.frame(genes)
  # ddsGenes <- genes[,ddsGenes]
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  resTri <- results(dds,contrast=c("condition",sample,anc),lfcThreshold=0.585)
  library(EnhancedVolcano)
  plot <- EnhancedVolcano(resTri,
    lab = rownames(resTri),
    x = 'log2FoldChange',
    y = 'pvalue',
    xlim = c(-5, 5),
    ylim = c(0,10),
    axisLabSize = 10, 
    titleLabSize = 12,
    subtitle = NULL,
    captionLabSize = 10,
    legendPosition = 'right',
    legendLabSize = 8,
    legendIconSize = 5.0,
    title = paste(sample,""),
    pCutoff = 0.1,
    vline = c(-1,0,0.585,1),
    vlineCol = c("green","black","magenta","orange"),
    FCcutoff = FC)  
  print(plot)
}  

gC7 <- volcanoPlots(ddsDefGC_7,"Holly_7","Holly_Anc",genesGC_7,"chrI",1)
gC1 <- volcanoPlots(ddsDefGC_1,"Holly_1","Holly_Anc",genesGC_1,"chrI",1)


gC11 <- volcanoPlots(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11,"chrI",-1)
gC2 <- volcanoPlots(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2,"chrI",-1)

gC11.15 <- volcanoPlots(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11,"chrXV",-1)
gC2.15 <- volcanoPlots(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2,"chrXV",-1)

gC18 <- volcanoPlots(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18,"chrI",-1)
gC3 <- volcanoPlots(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3,"chrI",-1)


library(cowplot)
plot_grid(gC11,gC2)


```



```{r testing EnhancedVolcano, include=TRUE,eval=TRUE,echo=FALSE}
L <- resultsTrisomy(ddsDefMAold_9,"SC009","SCA",genesMAold_9,"chrXIV")
View(L)
L$over.4
  if (!requireNamespace('BiocManager', quietly = TRUE))
    install.packages('BiocManager')

  BiocManager::install('EnhancedVolcano')
  library(EnhancedVolcano)
  
   EnhancedVolcano(resGC11Dat,
    lab = rownames(resGC11Dat),
    x = 'log2FoldChange',
    y = 'pvalue',
    xlim = c(-5, 8))

 EnhancedVolcano(test11,
    lab = rownames(test11),
    x = 'log2FoldChange',
    y = 'pvalue',
    xlim = c(-8, 8),
    title = 'GC 11 test',
    pCutoff = 0.1,
    FCcutoff = -1)      
   
   
   
   
  EnhancedVolcano(L$anti.5,
    lab = rownames(L$anti.5),
    x = 'log2FoldChange',
    y = 'pvalue',
    xlim = c(-5, 8))
  
  
  
    
```  
  
```{r resultsTrisomy, include=TRUE,eval=TRUE,echo=FALSE}
L <- resultsTrisomy(ddsDefMAnewS112,"112","Anc",genesMAnewS112,"chrXVI")
View(L)

resultsNames(ddsDefGC_77)
L <- resultsTrisomy(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_77,"chrXII")
View(L)




```

```{r tetrasomy, eval=TRUE, echo=FALSE}

dds=ddsDefGC_8
sample="Holly_8"
anc="Holly_Anc"
genes=genesGC_8
anChr="chrXVI"

resultsTetrasomy <- function(dds,sample,anc,genes,anChr) {
  # make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  genes
  keep <- anChr
  genes <- genes[(grep(paste0(keep,"$"),genes$TXCHROM,perl=TRUE)),]
  genes
  all(rownames(dds) == genes$GENEID)
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  ddsGenes <- intersect(rownames(dds),genes)
  as.data.frame(genes)
  str(genes)
  # ddsGenes <- genes[,ddsGenes]
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 1 (expectation for tetrasomic genes)
  resTet <- results(dds,contrast=c("condition",sample,anc), lfcThreshold=1 )
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (either no DC or unknown))
  resTetNonSig <- as.data.frame(subset(resTet,padj>0.1))
  length(resTetNonSig)
 resTetNonSig <- cbind(names = rownames(resTetNonSig), resTetNonSig)
  resTetNonSig <- dplyr::inner_join(genes,resTetNonSig,by=c("GENEID"="names"))
  str(resTetNonSig)
  # View(resTriNonSig)
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resTetSig <- as.data.frame(subset(resTet,padj<0.1))
  resTetSig <- cbind(names = rownames(resTetSig), resTetSig)
  resTetSig <- dplyr::inner_join(genes,resTetSig,by=c("GENEID"="names"))
  str(resTetSig)
  #### run same analyses using new ddsCopy dataset and new parameters
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (either Full DC or unknown genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  str(resDiNonSig)
  # subset data frame to get Significant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  str(resDiSig)
  #####################################
  ### Find genes that are Signficiantly different than Trisomic expectation and 
  # are Significantly different from Disomic expectation
  results <- list()
  resSigBoth<- dplyr::inner_join(resTetSig,resDiSig,by="GENEID")
  ## over DC genes for tetrasomy : less than 0 (1x)
  over.4 <- resSigBoth[which(resSigBoth$log2FoldChange.x<0),]
  ## anti DC genes for trisomy: more than 1 (2x)
  anti.5 <- resSigBoth[which(resSigBoth$log2FoldChange.x > 1),]
  results$over.4 <- over.4
  results$anti.5 <- anti.5
  ### Find genes that are Signficiantly different than Trisomic expectation but
  # NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resTetSig,resDiNonSig,by="GENEID")
  results$full.3 <- resFullDC
  ## Find genes that are not significantly diff from trisomic exp but sig diff from disomic
  resPartNo <- dplyr::inner_join(resTetNonSig,resDiSig,by="GENEID")
  resPart <- resPartNo[which(resPartNo$log2FoldChange.x < 1),]
  resPart1 <- resPart[which(resPart$log2FoldChange.x > 0),]
  results$Part.2 <- resPart1
  resNo <- resPartNo[which(resPartNo$log2FoldChange.x >= 1),]
  results$No.1 <- resNo
  ## Get genes that are not significantly different from either Trisomic or Disomic expectation
  resUnknown <- dplyr::inner_join(resTriNonSig,resDiNonSig,by="GENEID")
  results$Unknown.0 <- resUnknown
  return(results)
}

# input is dds for that sample, line #, ancestor name, genes matrix for that sample, and chromosome of interest
# output is list with full, part, over, anti, no, and unknown DC genes 
rm(dds)
rm(sample)
rm(anc)
rm(genes)
rm(anChr)

L <- resultsTrisomy(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8,"chrXVI")
View(L)




```

```{r monosomy, eval=TRUE,echo=FALSE}


dds=ddsDefMAold_108
sample="SC108"
anc="SCA"
genes=genesMAold_108
anChr="chrIX"

resultsMonosomy <- function(dds,sample,anc,genes,anChr) {
  # make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  genes
  keep <- anChr
  genes <- genes[(grep(paste0(keep,"$"),genes$TXCHROM,perl=TRUE)),]
  genes
  all(rownames(dds) == genes$GENEID)
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  ddsGenes <- intersect(rownames(dds),genes)
  as.data.frame(genes)
  str(genes)
  # ddsGenes <- genes[,ddsGenes]
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at -1 (expectation for monosomic genes)
  resMon <- results(dds,contrast=c("condition",sample,anc),lfcThreshold=1, altHypothesis="less")
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (either no DC or unknown))
  resMonNonSig <- as.data.frame(subset(resMon,padj>0.1))
  length(resMonNonSig)
  resMonNonSig <- cbind(names = rownames(resMonNonSig), resMonNonSig)
  resMonNonSig <- dplyr::inner_join(genes,resMonNonSig,by=c("GENEID"="names"))
  str(resMonNonSig)
  # View(resTriNonSig)
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resMonSig <- as.data.frame(subset(resMon,padj<0.1))
  resMonSig <- cbind(names = rownames(resMonSig), resMonSig)
  resMonSig <- dplyr::inner_join(genes,resMonSig,by=c("GENEID"="names"))
  str(resMonSig)
  #### run same analyses using new ddsCopy dataset and new parameters
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (either Full DC or unknown genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  str(resDiNonSig)
  # subset data frame to get Significant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  str(resDiSig)
  #####################################
  ### Find genes that are Signficiantly different than Monosomic expectation and 
  # are Significantly different from Disomic expectation
  results <- list()
  resSigBoth<- dplyr::inner_join(resMonSig,resDiSig,by="GENEID")
  ## over DC genes for monosomy : greater than 0 (1x)
  over.4 <- resSigBoth[which(resSigBoth$log2FoldChange.x>0),]
  ## anti DC genes for monosomy: less than -1 (0.5)
  anti.5 <- resSigBoth[which(resSigBoth$log2FoldChange.x < -1),]
  results$over.4 <- over.4
  results$anti.5 <- anti.5
  ### Find genes that are Signficiantly different than Trisomic expectation but
  # NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resTriSig,resDiNonSig,by="GENEID")
  results$full.3 <- resFullDC
  ## Find genes that are not significantly diff from trisomic exp but sig diff from disomic
  resPartNo <- dplyr::inner_join(resTriNonSig,resDiSig,by="GENEID")
  resPart <- resPartNo[which(resPartNo$log2FoldChange.x > -1),]
  results$Part.2 <- resPart
  resNo <- resPartNo[which(resPartNo$log2FoldChange.x <= -1),]
  results$No.1 <- resNo
  ## Get genes that are not significantly different from either Trisomic or Disomic expectation
  resUnknown <- dplyr::inner_join(resMonNonSig,resDiNonSig,by="GENEID")
  results$Unknown.0 <- resUnknown
  return(results)
}

L <- resultsTrisomy(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11,"chrI")
View(L)



```



Find which genes are not differentially expressed 
```{r nonDE_genes_cis,include=TRUE,echo=FALSE,eval=TRUE}
#######################################################################################
# want to know WHAT genes are NOT DE on the aneuploid chromosome in aneuploid samples
getNonDEGCgenes <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
samplenotDE <- subset(sample, padj>0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(samplenotDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, samplenotDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        notDEAneuChr <- subset(samp, chr==chrom)
return(notDEAneuChr)
}

# test it out on one of the ones I did already
# try S4
nonDEGCS4c5 <- getNonDEGCgenes(resGC4Dat, "chrV")
# write.csv(nonDEGCS4c5, file="nonDEGCS4c5.csv")
nonDEGCS7c1 <- getNonDEGCgenes(resGC7Dat, "chrI")
# write.csv(nonDEGCS7c1, file="nonDEGCS7c1.csv")
nonDEGCS8c16 <- getNonDEGCgenes(resGC8Dat, "chrXVI")
# write.csv(nonDEGCS8c16, file="nonDEGCS8c16.csv")
nonDEGCS11c1 <- getNonDEGCgenes(resGC11Dat, "chrI")
# write.csv(nonDEGCS11c1, file="nonDEGCS11c1.csv")
nonDEGCS11c15 <- getNonDEGCgenes(resGC11Dat, "chrXV")
# write.csv(nonDEGCS11c15, file="nonDEGCS11c15.csv")
nonDEGCS18c1 <- getNonDEGCgenes(resGC18Dat, "chrI")
# write.csv(nonDEGCS18c1, file="nonDEGCS18c1.csv")
nonDEGCS18c12 <- getNonDEGCgenes(resGC18Dat, "chrXII")
# write.csv(nonDEGCS18c12, file="nonDEGCS18c12.csv")
nonDEGCS59c7 <- getNonDEGCgenes(resGC59Dat, "chrVII")
# write.csv(nonDEGCS59c7, file="nonDEGCS59c7.csv")
nonDEGCS61c7 <- getNonDEGCgenes(resGC61Dat, "chrVII")
# write.csv(nonDEGCS61c7, file="nonDEGCS61c7.csv")
nonDEGCS76c9 <- getNonDEGCgenes(resGC76Dat, "chrIX")
# write.csv(nonDEGCS76c9, file="nonDEGCS76c9.csv")
nonDEGCS76c14 <- getNonDEGCgenes(resGC76Dat, "chrXIV")
# write.csv(nonDEGCS76c14, file="nonDEGCS76c14.csv")
nonDEGCS77c12 <- getNonDEGCgenes(resGC77Dat, "chrXII")
# write.csv(nonDEGCS77c12, file="nonDEGCS77c12.csv")

nonDEMA152c1 <- getNonDEGCgenes(resMA152Dat, "chrI")
nonDEMA117c5 <- getNonDEGCgenes(resMA117Dat, "chrV")
nonDEMA123c5 <- getNonDEGCgenes(resMA123Dat, "chrV")
nonDEMA108c7 <- getNonDEGCgenes(resMAold108Dat, "chrVII")
nonDEMA108c9 <- getNonDEGCgenes(resMAold108Dat, "chrIX")
nonDEMA15c9 <- getNonDEGCgenes(resMAold15Dat, "chrIX")
nonDEMA29c9 <- getNonDEGCgenes(resMA29Dat, "chrIX")
nonDEMA119c9 <- getNonDEGCgenes(resMAold119Dat, "chrIX")
nonDEMA9c14 <- getNonDEGCgenes(resMAold9Dat, "chrXIV")
nonDEMA112c16 <- getNonDEGCgenes(resMA112Dat, "chrXVI")


nonDE_upSet_listChrV <- list(MAS117=nonDEMA117c5$GENEID,
                       MAS123=nonDEMA123c5$GENEID
)


upset(fromList(nonDE_upSet_listChrV),nsets=2,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "Non DE ChrV Genes")
```




```{r DE_genes, include=TRUE, eval=TRUE, echo=FALSE}
#######################################################################################
# want to know WHAT genes are DE on the aneuploid chromosome in aneuploid samples

getDEGCgenes <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        DEAneuChr <- subset(samp, chr==chrom)
return(DEAneuChr)
}


# test it out on one of the ones I did already
# try S4
DEGCS4.c5 <- getDEGCgenes(resGC4Dat, "chrV")
# View(DEGCS4.c5)
# write.csv(DEGCS4.c5, file="DEGCS4c5.csv")
geneIDsGC4 <- DEGCS4.c5[,2]
str(geneIDsGC4)
geneIDsGC4 <- as.vector(geneIDsGC4)
# write.table(geneIDsGC4, file="DEgeneIDs4GC.txt")
DESGC7.c1 <- getDEGCgenes(resGC7Dat, "chrI")
# write.csv(DESGC7.c1, file="DESGC7c1.csv")
DESGC8.c16 <- getDEGCgenes(resGC8Dat, "chrXVI")
# write.csv(DESGC8.c16, file="DESGC8c16.csv")
DESGC11.c1 <- getDEGCgenes(resGC11Dat, "chrI")
# write.csv(DESGC11.c1, file="DESGC11c1.csv")
DESGC11.c15 <- getDEGCgenes(resGC11Dat, "chrXV")
# write.csv(DESGC11.c15, file="DESGC11c15.csv")
DESGC18.c1 <- getDEGCgenes(resGC18Dat, "chrI")
# write.csv(DESGC18.c1, file="DESGC18c1.csv")
DESGC18.c12 <- getDEGCgenes(resGC18Dat, "chrXII")
# write.csv(DESGC18.c12, file="DESGC18c12.csv")
DESGC59.c7 <- getDEGCgenes(resGC59Dat, "chrVII")
# write.csv(DESGC59.c7, file="DESGC59c7.csv")
DESGC61.c7 <- getDEGCgenes(resGC61Dat, "chrVII")
# write.csv(DESGC61.c7, file="DESGC61c7.csv")
DESGC76.c9 <- getDEGCgenes(resGC76Dat, "chrIX")
# write.csv(DESGC76.c9, file="DESGC76c9.csv")
DESGC76.c14 <- getDEGCgenes(resGC76Dat, "chrXIV")
# write.csv(DESGC76.c14, file="DESGC76c14.csv")
DESGC77.c12 <- getDEGCgenes(resGC77Dat, "chrXII")
# write.csv(DESGC77.c12, file="DESGC77c12.csv")



### DE genes in general (for euploid samples)
getDEgenes <- function(sample) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
return(samp)
}

GCS1 <- getDEgenes(resGC1Dat)
GCS2 <- getDEgenes(resGC2Dat)
GCS3 <- getDEgenes(resGC3Dat)
GCS5 <- getDEgenes(resGC5Dat)
GCS9 <- getDEgenes(resGC9Dat)

MAS1 <- getDEgenes(resMAold1Dat)
MAS2 <- getDEgenes(resMAold2Dat)
MAS3 <- getDEgenes(resMAold3Dat)
MAS4 <- getDEgenes(resMAold4Dat)
MAS5 <- getDEgenes(resMAold5Dat)
MAS7 <- getDEgenes(resMAold7Dat)
MAS8 <- getDEgenes(resMAold8Dat)
MAS28 <- getDEgenes(resMAold28Dat)
MAS50 <- getDEgenes(resMA50Dat)
MAS115 <- getDEgenes(resMA115Dat)


DE_upSet_listMAEups <- list(MAS1=MAS1$GENEID,
                      MAS2=MAS2$GENEID,
                      MAS3=MAS3$GENEID,
                      MAS4=MAS4$GENEID,
                      MAS5=MAS5$GENEID,
                      MAS7=MAS7$GENEID,
                      MAS8=MAS8$GENEID,
                      MAS28=MAS28$GENEID,
                      MAS115=MAS115$GENEID,
                      MAS50=MAS50$GENEID
)


library(nVennR)
upsetMAEusDE_contents <- listVennRegions(plotVenn(DE_upSet_listMAEups))
deTransGenesAneuHom <- upsetMAEusDE_contents$`0, 0, 1, 1, 1, 1, 0, 1, 0, 0 (MAS3, MAS4, MAS5, MAS7, MAS28)`
deTransGenesAneuHom <- as.character(deTransGenesAneuHom)


library(UpSetR)
upset(fromList(DE_upSet_listMAEups),nsets=10,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "Shared DE Genes in Euploid Lines")

GCS1 <- getDEgenes(resGC1Dat)
GCS2 <- getDEgenes(resGC2Dat)
GCS3 <- getDEgenes(resGC3Dat)
GCS5 <- getDEgenes(resGC5Dat)
GCS9 <- getDEgenes(resGC9Dat)

DE_upSet_listGCEups <- list(GCS1=GCS1$GENEID,
                      GCS2=GCS2$GENEID,
                      GCS3=GCS3$GENEID,
                      GCS5=GCS5$GENEID,
                      GCS9=GCS9$GENEID
)


upsetGCEusDE_contents <- listVennRegions(plotVenn(DE_upSet_listGCEups))
deTransGenesAneuHet <- upsetGCEusDE_contents$`1, 1, 1, 1, 1 (GCS1, GCS2, GCS3, GCS5, GCS9)`
deTransGenesAneuHet <- as.character(deTransGenesAneuHet)



upset(fromList(DE_upSet_listGCEups),nsets=5,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "Shared DE Genes in Euploid Lines")
```



```{r trans_de_genes,include=TRUE,eval=TRUE,echo=FALSE}
################################################################
# want to get DE genes in aneuploid lines that are NOT located on the aneuploid chromosome

getDEgenesGCElsewhere <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        DENotAneuChr <- subset(samp, chr!=chrom)
return(DENotAneuChr)
}

deGenesNotAneuGCS4 <- getDEgenesGCElsewhere(resGC4Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS7 <- getDEgenesGCElsewhere(resGC7Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS8 <- getDEgenesGCElsewhere(resGC8Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS18 <- getDEgenesGCElsewhere(resGC18Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS49 <- getDEgenesGCElsewhere(resGC49Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS59 <- getDEgenesGCElsewhere(resGC59Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS61 <- getDEgenesGCElsewhere(resGC61Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS76 <- getDEgenesGCElsewhere(resGC76Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS77 <- getDEgenesGCElsewhere(resGC77Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))
deGenesNotAneuGCS11 <- getDEgenesGCElsewhere(resGC11Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXV,chrXVI"))


# View(deGenesNotAneuGCS4)

DE_upSet_list <- list(GCS4=deGenesNotAneuGCS4$GENEID,
                      GCS7=deGenesNotAneuGCS7$GENEID,
                      GCS8=deGenesNotAneuGCS8$GENEID,
                      GCS11=deGenesNotAneuGCS11$GENEID,
                      GCS18=deGenesNotAneuGCS18$GENEID,
                      GCS49=deGenesNotAneuGCS49$GENEID,
                      GCS59=deGenesNotAneuGCS59$GENEID,
                      GCS61=deGenesNotAneuGCS61$GENEID,
                      GCS76=deGenesNotAneuGCS76$GENEID,
                      GCS77=deGenesNotAneuGCS77$GENEID
)


upsetGCAneusTransDE <- upset(fromList(DE_upSet_list),nsets=10,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "DE Trans Genes")

## Extract out the DE genes 
library(nVennR)
upsetGCAneusTransDE_contents <- listVennRegions(plotVenn(DE_upSet_list))
deTransGenesAneuHet <- upsetGCAneusTransDE_contents$`1, 1, 1, 0, 1, 0, 1, 1, 1, 1 (GCS4, GCS7, GCS8, GCS18, GCS59, GCS61, GCS76, GCS77)`
deTransGenesAneuHet <- as.character(deTransGenesAneuHet)





deGenesNotAneuMAS152 <- getDEgenesGCElsewhere(resMA152Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS112 <- getDEgenesGCElsewhere(resMA112Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS117 <- getDEgenesGCElsewhere(resMA117Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS123 <- getDEgenesGCElsewhere(resMA123Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS108 <- getDEgenesGCElsewhere(resMAold108Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS119 <- getDEgenesGCElsewhere(resMAold119Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS88 <- getDEgenesGCElsewhere(resMAold88Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS9 <- getDEgenesGCElsewhere(resMAold9Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS15 <- getDEgenesGCElsewhere(resMAold15Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))
deGenesNotAneuMAS29 <- getDEgenesGCElsewhere(resMA29Dat, chrom=c("chrI,chrV,chrVII,chrIX,chrXII,XIV,chrXVI"))


DE_upSet_list_MA <- list(MAS9=deGenesNotAneuMAS9$GENEID,
                      MAS15=deGenesNotAneuMAS15$GENEID,
                      MAS152=deGenesNotAneuMAS152$GENEID,
                      MAS112=deGenesNotAneuMAS112$GENEID,
                      MAS117=deGenesNotAneuMAS117$GENEID,
                      MAS108=deGenesNotAneuMAS108$GENEID,
                      MAS119=deGenesNotAneuMAS119$GENEID,
                      MAS88=deGenesNotAneuMAS88$GENEID,
                      MAS9=deGenesNotAneuMAS9$GENEID,
                      MAS19=deGenesNotAneuMAS29$GENEID
)


upset(fromList(DE_upSet_list_MA),nsets=10,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "DE Trans Genes")

## Extract out the DE genes 
library(nVennR)
upsetMAAneusTransDE_contents <- listVennRegions(plotVenn(DE_upSet_list_MA))
deTransGenesAneuHom <- upsetMAAneusTransDE_contents$`1, 1, 0, 0, 0, 1, 1, 1, 0 (MAS9, MAS15, MAS108, MAS119, MAS88)`
deTransGenesAneuHom <- as.character(deTransGenesAneuHom)



```


```{r cis_de_genes,include=TRUE,eval=TRUE,echo=FALSE}
#######################################################################################
# want to know WHAT genes are DE on the aneuploid chromosome in aneuploid samples

getDEgenesGCAll <- function(sample) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   

sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        DEAll  <- samp
return(DEAll)
}

# test it out on one of the ones I did already
# try S4
DEGCS1all <- getDEgenesGCAll(resGC1Dat)
DEGCS2all <- getDEgenesGCAll(resGC2Dat)
DEGCS3all <- getDEgenesGCAll(resGC3Dat)
DEGCS4all <- getDEgenesGCAll(resGC4Dat)
DEGCS5all <- getDEgenesGCAll(resGC5Dat)
DEGCS7all <- getDEgenesGCAll(resGC7Dat)
DEGCS8all <- getDEgenesGCAll(resGC8Dat)
DEGCS9all <- getDEgenesGCAll(resGC9Dat)
DEGCS11all <- getDEgenesGCAll(resGC11Dat)
DEGCS18all <- getDEgenesGCAll(resGC18Dat)
DEGCS49all <- getDEgenesGCAll(resGC49Dat)
DEGCS59all <- getDEgenesGCAll(resGC59Dat)
DEGCS61all <- getDEgenesGCAll(resGC61Dat)
DEGCS76all <- getDEgenesGCAll(resGC76Dat)
DEGCS77all <- getDEgenesGCAll(resGC77Dat)
```



```{r cis_non_de_genes,include=TRUE,eval=TRUE,echo=FALSE}
########### Non DE genes on aneuploid chromosome 
getnonDEGCgenes <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
sampleDE <- subset(sample, padj>0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        nonDEAneuChr <- subset(samp, chr==chrom)
return(nonDEAneuChr)
}

DEnonGCS4.c5 <- getnonDEGCgenes(resGC4Dat, "chrV")
# View(DEnonGCS4.c5)
# write.csv(DEGCS4.c5, file="DEGCS4c5.csv")

# write.table(geneIDsGC4, file="DEgeneIDs4GC.txt")

nonDESGC7.c1 <- getnonDEGCgenes(resGC7Dat, "chrI")
# write.csv(DESGC7.c1, file="DESGC7c1.csv")

nonDESGC8.c16 <- getnonDEGCgenes(resGC8Dat, "chrXVI")
# write.csv(DESGC8.c16, file="DESGC8c16.csv")

nonESGC11.c1 <- getnonDEGCgenes(resGC11Dat, "chrI")
# write.csv(DESGC11.c1, file="DESGC11c1.csv")

nonDESGC11.c15 <- getnonDEGCgenes(resGC11Dat, "chrXV")
# write.csv(DESGC11.c15, file="DESGC11c15.csv")

nonDESGC18.c1 <- getnonDEGCgenes(resGC18Dat, "chrI")
# write.csv(DESGC18.c1, file="DESGC18c1.csv")

nonDESGC18.c12 <- getnonDEGCgenes(resGC18Dat, "chrXII")
# write.csv(DESGC18.c12, file="DESGC18c12.csv")

nonDESGC49.c5 <- getnonDEGCgenes(resGC49Dat, "chrV")

nonDESGC59.c7 <- getnonDEGCgenes(resGC59Dat, "chrVII")
# write.csv(DESGC59.c7, file="DESGC59c7.csv")

nonDESGC61.c7 <- getnonDEGCgenes(resGC61Dat, "chrVII")
# write.csv(DESGC61.c7, file="DESGC61c7.csv")

nonDESGC76.c9 <- getnonDEGCgenes(resGC76Dat, "chrIX")
# write.csv(DESGC76.c9, file="DESGC76c9.csv")

nonDESGC76.c14 <- getnonDEGCgenes(resGC76Dat, "chrXIV")
# write.csv(DESGC76.c14, file="DESGC76c14.csv")

nonDESGC77.c12 <- getnonDEGCgenes(resGC77Dat, "chrXII")



nonDE_upSet_listChrI <- list(GCS7=nonDESGC7.c1$GENEID,
                         GCS18=nonDESGC18.c1$GENEID,
                         GCS11=nonESGC11.c1$GENEID
                     
)

upset(fromList(nonDE_upSet_listChrI),nsets=3,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "Non-DE ChrI Genes")

nonDE_upSet_listChrXII <- list(GCS18=nonDESGC18.c12$GENEID,
                         GCS77=nonDESGC77.c12$GENEID
                     
)

upset(fromList(nonDE_upSet_listChrXII),nsets=2,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "Non-DE ChrXII Genes")

nonDE_upSet_listChrVII <- list(GCS59=nonDESGC59.c7$GENEID,
                         GCS61=nonDESGC61.c7$GENEID
                     
)

upset(fromList(nonDE_upSet_listChrVII),nsets=2,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "Non-DE ChrVII Genes")
```


```{r common_de_cis_genes,include=TRUE,eval=TRUE,echo=FALSE}
########################################################################################
# what DE genes are shared between samples aneuploid with the same chromosome?
  # use data frames containing only DE genes in the given samples

library(dplyr)

getCommonDEGCgenes <- function(sample1, sample2) {
  # join them together by GENEID column
  chr.all <- inner_join(sample1, sample2, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matched <- intersect(chr.all$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepAll <- chr.all$GENEID %in% matched
          # only keep those genes in the data frame
          chr.all <- chr.all[keepAll,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenAll <- genesGC$GENEID %in% matched
              # put those genes in a vector
              GenAll <- genesGC[keepGenAll,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chr.all <- cbind(GenAll$TXCHROM, chr.all)
                return(chr.all)
}

chrVIIAllGC <- getCommonDEGCgenes(genesGC59DE, genesGC61DE)
# write.csv(chrVIIAllGC, file="chrVIIallDE_GC.csv")
chrXIIAllGC <- getCommonDEGCgenes(genesGC18DE, genesGC77DE)
# write.csv(chrXIIAllGC, file="chrXIIallDE_GC.csv")

DE_upSet_list <- list(GCS59=genesGC59DE$GENEID,
                      GCS61=genesGC61DE$GENEID
                     
)



upset(fromList(DE_upSet_list),nsets=2,keep.order = F,matrix.color = "darkorchid",main.bar.color = "deepskyblue",sets.bar.color = "deeppink",order.by = "freq",sets.x.label = "DE Trans Genes")
```

```{r cis_non_de_genes,include=TRUE,eval=TRUE,echo=FALSE}
########################################################################################
# what NON-DE genes are shared between samples aneuploid with the same chromosome?
  # use data frames containing only NON-DE genes in the given samples

getCommonNonDEGCgenes <- function(sample1, sample2) {
  # join them together by GENEID column
  chr.all <- inner_join(sample1, sample2, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matched <- intersect(chr.all$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepAll <- chr.all$GENEID %in% matched
          # only keep those genes in the data frame
          chr.all <- chr.all[keepAll,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenAll <- genesGC$GENEID %in% matched
              # put those genes in a vector
              GenAll <- genesGC[keepGenAll,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chr.all <- cbind(GenAll$TXCHROM, chr.all)
                return(chr.all)
}


chrVIIGCNon <- getCommonNonDEGCgenes(nonDEGCS59c7, nonDEGCS61c7)
# write.csv(chrVIIGCNon, file="chrVIIallNonDE_GC.csv")
chrXIIGCNon <- getCommonNonDEGCgenes(nonDEGCS18c12, nonDEGCS77c12)
# write.csv(chrXIIGCNon, file="chrXIIallNonDE_GC.csv")

## only trisomic samples
    # join them together by GENEID column
    chrItriGCNon <- inner_join(nonDEGCS7c1,nonDEGCS18c1, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matchedIGCNon <- intersect(chrItriGCNon$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepIGCNon <- chrItriGCNon$GENEID %in% matchedIGCNon
          # only keep those genes in the data frame
          chrItriGCNon <- chrItriGCNon[keepIGCNon,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenGCNon <- genesGC$GENEID %in% matchedIGCNon
              # put those genes in a vector
              GenNonGC <- genesGC[keepGenGCNon,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chrItriGCNon <- cbind(GenNonGC$TXCHROM, chrItriGCNon)
                  # save as .csv file
                  # write.csv(chrItriGCNon, file="chrITriNonDE_GC.csv")
#for chromosome I, which has 3 lines that are commonly aneuploid for it, so the function won't work properly
  # chrom I, all aneuploids (both trisomics plus the monosomic)
    chrI.NonGC <- inner_join(chrItriGCNon, nonDEGCS11c1, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matchedIallGCNon <- intersect(chrI.NonGC$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepIallGCNon <- chrI.NonGC$GENEID %in% matchedIallGCNon
          # only keep those genes in the data frame
          chrI.NonGC <- chrI.NonGC[keepIallGCNon,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenAllNonGC <- genesGC$GENEID %in% matchedIallGCNon
              # put those genes in a vector
              GenAllNonGC <- genesGC[keepGenAllNonGC,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chrI.NonGC <- cbind(GenAllNonGC$TXCHROM, chrI.NonGC)
                  # save as .csv file
                  # write.csv(chrI.NonGC, file="chrIAllNonDE_GC.csv")

eupLinesGC <- list(genesGC1DE, genesGC2DE, genesGC3DE, genesGC5DE, genesGC9DE)
# want to get list of common DE genes in all the euploid lines
commonDEeupGC <- Reduce(intersect, Map("[[", eupLinesGC, "GENEID"))
#Reduc
commonDEeupGC <- as.vector(commonDEeupGC)
commonDEeupGC <- as.data.frame(commonDEeupGC)
colnames(commonDEeupGC) <- c("names")
# write.csv(commonDEeupGC, file="commonDEeup_GC.csv")

# what genes are commonly DE in aneuploid lines, but NOT located on the aneuploid chromosomes?
# genes11DE,
aneuLinesGC <- 	list(deGenesNotAneuGCS4, deGenesNotAneuGCS7, deGenesNotAneuGCS8,deGenesNotAneuGCS11, deGenesNotAneuGCS18,
                     deGenesNotAneuGCS59, deGenesNotAneuGCS61, deGenesNotAneuGCS76, deGenesNotAneuGCS77)
# want to get list of common DE genes in all the euploid lines
commonDEaneuGC <- Reduce(intersect, Map("[[", aneuLinesGC, "GENEID"))
commonDEeupGC <- as.vector(commonDEeupGC)
# write.csv(commonDEeupGC, file="commonDEeup_GC.csv")

```



level `alpha` (we use the standard variable name for significance
level, though it is unrelated to the dispersion parameter $\alpha$).
The theory behind independent filtering is discussed in greater detail
[below](#indfilttheory). The adjusted *p* values for the genes
which do not pass the filter threshold are set to `NA`.

The default independent filtering is performed using the *filtered_p*
function of the [genefilter](http://bioconductor.org/packages/genefilter) package, and all of the
arguments of *filtered_p* can be passed to the *results* function.
The filter threshold value and the number of rejections at each
quantile of the filter statistic are available as metadata of the
object returned by *results*.

For example, we can visualize the optimization by plotting the
`filterNumRej` attribute of the results object. The *results* function
maximizes the number of rejections (adjusted *p* value less than a
significance level), over the quantiles of a filter statistic (the
mean of normalized counts). The threshold chosen (vertical line) is
the lowest quantile of the filter for which the number of rejections
is within 1 residual standard deviation to the peak of a curve fit to
the number of rejections over the filter quantiles:

```{r graphs, eval=TRUE,include=TRUE,echo=FALSE}
library(ggplot2)

size_v_numAneus <- read.delim("/Users/hollymcqueary/Dropbox/McQueary/DosageCompensationG3/size_vs_numAneus.txt")

# View(size_v_numAneus)

## Heterozygous Ancestor
sizeNumAneus <- ggplot(size_v_numAneus,aes(x=length,y=X.eventsGC))
sizeNumAneus + geom_point(color="turquoise") + theme_classic()  + geom_smooth(method = "lm", se=FALSE, color="turquoise") + labs(y="Aneuploidy Events", x="Chromosome Length", title="Heterozygous Ancestor") 


library(ggpmisc)
my.formula <- y ~ x
p <- ggplot(size_v_numAneus,aes(x=length,y=X.eventsGC))
p2 <- p + geom_point(color="turquoise") + theme_classic()  + geom_smooth(method = "lm", se=FALSE, color="turquoise",formula=my.formula) + stat_poly_eq(formula = my.formula, 
                aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), 
                parse = TRUE) + labs(y="Aneuploidy Events", x="Chromosome Length", title="Heterozygous Ancestor") 

pdf("het_anc_size_v_aneus.pdf")
p2
dev.off()
  

## Homozygous Ancestor
sizeNumAneus1 <- ggplot(size_v_numAneus,aes(x=length,y=X.eventsMA))
sizeNumAneus1 + geom_point(color="deeppink1") + theme_classic()  + geom_smooth(method = "lm", se=FALSE, color="deeppink1") + labs(y="Aneuploidy Events", x="Chromosome Length", title="Homozygous Ancestor") 

library(ggpmisc)
my.formula <- y ~ x
p1 <- ggplot(size_v_numAneus,aes(x=length,y=X.eventsMA))
p3 <- p1 + geom_point(color="deeppink1") + theme_classic()  + geom_smooth(method = "lm", se=FALSE, color="deeppink1",formula=my.formula) + stat_poly_eq(formula = my.formula, 
                aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), 
                parse = TRUE) + labs(y="Aneuploidy Events", x="Chromosome Length", title="Homozygous Ancestor") 

pdf("hom_anc_size_v_aneus.pdf")
p3
dev.off()


p1 <- ggplot(size_v_numAneus,aes(x=length,y=X.eventsMA))
p3 <- p1 + geom_point(color="deeppink1") + theme_classic()  + geom_smooth(method = "lm", se=FALSE, color="deeppink1",formula=my.formula) + stat_poly_eq(formula = my.formula, 
                aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), 
                parse = TRUE) + labs(y="Aneuploidy Events", x="Chromosome Length", title="Homozygous Ancestor") 


library(cowplot)
pdf("hom_het_size_v_aneu.pdf")
par(mar = c(2, 2, 2, 2))
plot_grid(p2,p3)
dev.off()

size_vAneusGC <- size_v_numAneus[,2:3]
library(data.table)
size_v_numAneus <- size_v_numAneus[,2:4]
colnames(size_v_numAneus) <- c("length","Heterozygous","Homozygous")
size_v_numAneusMelt <- melt(size_v_numAneus,measure.vars=c(2,3))
p1 <- ggplot(size_v_numAneusMelt,aes(x=length,y=value))
pdf("hom_het_size_v_aneu.pdf")
p1 + facet_wrap(vars(variable)) + geom_point(color="deeppink1") + theme_classic()  + geom_smooth(method = "lm", se=FALSE, color="deeppink1",formula=my.formula) + stat_poly_eq(formula = my.formula, 
                aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), 
                parse = TRUE) + labs(y="Aneuploidy Events", x="Chromosome Length") 
dev.off()



### Number of aneuploid events per chromosome 
aneuploidies <- read.table("/Users/hollymcqueary/Dropbox/McQueary/DosageCompensationG3/aneuploidies.txt",header=TRUE)
View(aneuploidies)

library(data.table)

numAneus <- reshape2::melt(aneuploidies, measure.vars=3:5)
View(numAneus)

pdf("anEvents.pdf")
gcEvents <- ggplot(numAneus, x=Chromosome, y=value,color=variable)
gcEvents + facet_wrap(vars(Ancestor)) + geom_col(aes(x=factor(Chromosome), y=value,fill=variable)) + theme_classic() + labs(y="Number",x="Chromosome") + scale_fill_discrete(name=NULL)
dev.off()

```


# Acknowledgments

We have benefited in the development of DESeq2 from the help and
feedback of many individuals, including but not limited to:

The Bionconductor Core Team,
Alejandro Reyes, Andrzej Oles, Aleksandra Pekowska, Felix Klein,
Nikolaos Ignatiadis (IHW),
Anqi Zhu (apeglm),
Joseph Ibrahim (apeglm),
Vince Carey,
Owen Solberg,
Ruping Sun,
Devon Ryan,
Steve Lianoglou, Jessica Larson, Christina Chaivorapol, Pan Du, Richard Bourgon,
Willem Talloen,
Elin Videvall, Hanneke van Deutekom,
Todd Burwell,
Jesse Rowley,
Igor Dolgalev,
Stephen Turner,
Ryan C Thompson,
Tyr Wiesner-Hanks,
Konrad Rudolph,
David Robinson,
Mingxiang Teng,
Mathias Lesche,
Sonali Arora,
Jordan Ramilowski,
Ian Dworkin,
Bjorn Gruning,
Ryan McMinds,
Paul Gordon,
Leonardo Collado Torres,
Enrico Ferrero,
Peter Langfelder,
Gavin Kelly,
Rob Patro,
Charlotte Soneson,
Koen Van den Berge,
Stephan Engelbrecht,
Nicolas Alcala.

# Session info

```{r look at variance,include=FALSE}

# GC_anc <- counts(ddsHTSeqGC_1)[,4:6]


# genes <- data.frame(x.GC$genes$GENEID, x.GC$genes$TXCHROM)
# genes <- genesGC
# rownames(genes) <- genes$x.GC.genes.GENEID
# 
# remove <- c("__no_feature", "__ambiguous","__alignment_not_unique","__too_low_aQual","__not_aligned")
# genes <- genes[setdiff(rownames(genes),remove),]
# removegenes <- c("YPL241C","YPL163C")
# genes <- genes[setdiff(rownames(genes),removegenes),]
# genes$x.GC.genes.GENEID <- droplevels(genes$x.GC.genes.GENEID)


#remove NAs from data

# lcpmGC.test <- merge(genes,lcpmGC, by.x="GENEID", by.y="names")
# lcpmGC <- lcpmGC.test
# 
# 
# cpmGC.test <- merge(genesGC,cpmGC, by.x="GENEID", by.y="names")
# cpmGC <- cpmGC.test
# write.csv(cpmGC,file="cpm_norm.csv")
# 
# nchrm <- nlevels(lcpmGC$x.GC.genes.TXCHROM)
# lcpmGC$x.GC.genes.TXCHROM
# lcpmGC
# x.GC$samples$group
# cpmGC
# length(x.GC$samples$group)
# names <- colnames(lcpmGC)
# length(lcpmGC[,1])
# nrow(lcpmGC)
# rm(i)
# lcpmGC <- na.omit(lcpmGC)
# View(lcpmGC)
# write.csv(lcpmGC,file="log_cpm_norm.csv")
# rm(i)
# # #testing to make sure this is doing what I want it to do: find the variance between replicates, not across all lines
# lineEl <- matrix(data=NA,ncol=3,nrow=length(lcpmGC))
# lineEl.ind <- grep(as.character(lines.GC[1]),names)
# ind.El <- lcpmGC[,lineEl.ind]
# test.M <- matrix(apply(ind.El,1,var))
# #yes it is working
# 
# colnames(lcpmGC) <- c("gene","chr", ".11.A",".11.B",".11.C",".18.A",".18.B",".18.C",".1.A", ".1.B", ".1.C", ".21.B",".21.C",".2.A", ".2.B", ".2.C", ".31.A",".31.B",".31.C",".3.A", ".3.B", ".3.C",".49.A",".49.B",".49.C",".4.A", ".4.B",".4.C", ".59.A",".59.B",".59.C",".5.A",".5.B", ".5.C", ".61.A",".61.B",".61.C",".66.B",".66.C",".68.A",".68.B",".68.C",".6.A", ".6.B", ".6.C", ".76.A",".76.B",".76.C",".77.A",".77.B",".77.C",".7.A",".7.B", ".7.C", ".8.A", ".8.B", ".8.C",".9.A", ".9.B", ".9.C", ".GCAnc.A" ,".GCAnc.B",".GCAnc.C")
# names <- colnames(lcpmGC)
# lines.G <- c(".11.",".18.",".1.",".21.",".2.",".31.",".3.",".49.",".4.",".59.",".5.",".61.",".66.",".68.",".6.", ".76.",".77.",".7.",".8.", ".9.",".GCAnc.")
# colnames.G <- c("GENE","CHRM", " .11.",".18.",".1.",".21.",".2.",".31.",".3.",".49.",".4.",".59.",".5.",".61.",".66.",".68.",".6.", ".76.",".77.",".7.",".8.", ".9.",".GCAnc.")
# 
# i=1
# rm(i)
# vari <- matrix(data=NA,nrow=nrow(lcpmGC),ncol=length(lines.G))
# is.list(vari)
# for (i in 1:length(lines.G)) {
#   ind <- grep(as.character(lines.G[i]),names)
#   cols <- lcpmGC[,ind]
#   cols <- matrix(apply(cols,1,var))
#   rownames(cols) <- c()
#   colnames(cols) <- c()
#   vari[,i] <- cols
# }
# 
# vari <- as.data.frame(vari)
# #vari <- cbind(as.factor(genes$x.GC.genes.TXCHROM),vari)
# View(vari)
# vari <- cbind(lcpmGC$gene,lcpmGC$chr,vari)
# colnames(vari) <- colnames.G
# class(vari[,3])
# vari$GENE <- droplevels(vari$GENE)
# rownames(vari) <- vari$GENE
# #vari2 <- subset(vari, vari[1:3386,]>=0.0001) # & vari[,3:23]>=0.0001)
# #summary(vari[,3:23]>=0.0001)
# #summary(vari[1:3386,]<=0.0001)
# table(rowSums(vari[3:23]>0.0001)>=21)
# keep.genes.GC <- rowSums(vari[,3:23]>=0.0001)>=21
# keep.genes.GC <- as.data.frame(keep.genes.GC)
# keep.genes.GC$genes <- rownames(keep.genes.GC)
# genes2 <- merge(keep.genes.GC,genes,by.x="genes",by.y="x.GC.genes.GENEID")
# genes2 <- genes2[!(genes2$keep.genes.GC=="FALSE"),]
# 
# #remove genes with variance less than 0.00001
# lcpmGC <- merge(genes2,lcpmGC, by.x="genes", by.y="gene")
# lcpmGC <- lcpmGC[,-2]
# lcpmGC <- lcpmGC[,-3]
# lcpmGC <- droplevels(lcpmGC)
# vari2 <- merge(vari,lcpmGC,by.x="GENE",by.y="genes")
# 
# mean <- matrix(data=NA,nrow=nrow(lcpmGC),ncol=length(lines.G))
# is.list(mean)
# for (i in 1:length(lines.G)) {
#   ind <- grep(as.character(lines.G[i]),names)
#   cols <- lcpmGC[,ind]
#   rownames(cols) <- c()
#   colnames(cols) <- c()
#   m <- matrix(apply(cols,1,mean))
#   mean[,i] <- m
# }
# mean <- as.data.frame(mean)
# mean <- cbind(lcpmGC$genes,lcpmGC$x.GC.genes.TXCHROM,mean)
# colnames(mean) <- colnames.G
# 
# 
# std <- matrix(data=NA,nrow=nrow(lcpmGC),ncol=length(lines.G))
# length(lcpmGC)
# is.list(std)
# for (i in 1:length(lines.G)) {
#   ind <- grep(as.character(lines.G[i]),names)
#   cols <- lcpmGC[,ind]
#   cols <- matrix(apply(cols,1,sd))
#   rownames(cols) <- c()
#   colnames(cols) <- c()
#   std[,i] <- cols
# }
# std <- as.data.frame(std)
# std <- cbind(lcpmGC$genes,lcpmGC$x.GC.genes.TXCHROM,std)
# colnames(std) <- colnames.G
# class(std[,3])
# 
# GCanc.stats <- data.frame(mean$GENE, mean$CHRM, mean$.GCAnc.,std$.GCAnc.)
# colnames(GCanc.stats) <- c("GENE","CHRM","mean","std")
# 
# #
# #
# k <- ggplot(GCanc.stats, aes(CHRM, mean))#ymin = mean-std, ymax = mean+std))
# print(k)
# 
# 
# 
# dim(mean)
# dim(vari)
# dim(std)
# #std[,3:23] <- as.numeric(std[,3:23])
# cv <- matrix(data=NA,nrow=nrow(mean),ncol=length(ncol(mean)))
# #log transform the coefficient of variance
# #cv <- log(cv)
# View(std)
# #mean[,3:23] <- as.numeric(mean[,3:23])
# View(mean)
# is.list(std)
# class(mean[,3:23])
# class(std[,3:23])
# class(mean[,3])
# is.list(mean)
# cv <- std[,3:23]/mean[,3:23]
# dim(cv)
# #cv <- transform_to_log_scale(cv)
# cv <- cbind(lcpmGC$genes,lcpmGC$x.GC.genes.TXCHROM,cv)
# colnames(cv) <- colnames.G
# #cv <- unlist(cv)
# #max(subset(cv,select="X3"))
# 
# par(mar=c(1,1,1,1))
# 
# #loop to plot mean-variance relationship for all lines
# lines.G
# line.G <- c("NA",".GCAnc.",".11.",".18.",".1.",".21.",".2.",".31.",".3.",".49.",".4.",".59.",".5.",".61.",".66.",".68.",".6.", ".76.",".77.",".7.",".8.", ".9.")
# i=3
# length(line.G)
# length(mean)
# rm(i)
# for (i in 3:23) {
#   pdf(paste("Mean-Variance Relationship",line.G[i],".pdf",sep=""))
#   print(ggplot () + geom_point(aes(x=mean[,i],y=vari2[,i],col=factor(mean[,2]))) + scale_y_log10() + labs(title=paste("Mean-Variance Relationship Line",line.G[i]), y="variance",x="mean"))
#   dev.off()
# }
# 
# #use plotly to view the graph interactively and see what points correspond to what genes/chromosomes
# length(vari2)
# class(vari2)
# class(mean)
# length(vari2[,1])
# length(mean[,1])
# 
# get.points <- function(line) {
#    ggplot () + geom_point(aes(x=mean[,line],y=vari2[,line],col=factor(mean[,2]))) + scale_y_log10() + labs(title=paste("Mean-Variance Relationship Line",line.G[line]), y="variance",x="mean")
# ggplotly()
# }
# #line=18
# get.points(18)
# 
# par(mfrow=c(1,1))
# 
# for (i in 3:23) {
#   pdf(paste("Dist_Variance",i,".pdf",sep=""))
#   plot(density(vari2[,i]),col=colGC[i],lwd=2,las=2,main="",xlab="")
#   title(main=paste("Distribution of Variance",line.G[i]),xlab="var of lcpm")
#   abline(v=0,lty=3)
#   dev.off()
# }

```

```{r sessionInfo}
sessionInfo()
```

# References
