---
title: "Analyzing RNA-seq data with DESeq2"
author: "Michael I. Love, Simon Anders, and Wolfgang Huber"
date: "`r format(Sys.Date(), '%m/%d/%Y')`"
abstract: >
  A basic task in the analysis of count data from RNA-seq is the
  detection of differentially expressed genes. The count data are
  presented as a table which reports, for each sample, the number of
  sequence fragments that have been assigned to each gene. Analogous
  data also arise for other assay types, including comparative ChIP-Seq,
  HiC, shRNA screening, mass spectrometry.  An important analysis
  question is the quantification and statistical inference of systematic
  changes between conditions, as compared to within-condition
  variability. The package DESeq2 provides methods to test for
  differential expression by use of negative binomial generalized linear
  models; the estimates of dispersion and logarithmic fold changes
  incorporate data-driven prior distributions This vignette explains the
  use of the package and demonstrates typical workflows.
  on the Bioconductor website covers similar material to this vignette
  but at a slower pace, including the generation of count matrices from
  FASTQ files.
  [An RNA-seq workflow](http://www.bioconductor.org/help/workflows/rnaseqGene/)
  DESeq2 package version: `r packageVersion("DESeq2")`
output:
  rmarkdown::html_document:
    highlight: pygments
    toc: true
    fig_width: 5
bibliography: library.bib
vignette: >
  %\VignetteIndexEntry{Analyzing RNA-seq data with DESeq2}
  %\VignetteEncoding{UTF-8}
  %\usepackage[utf8]{inputenc}
  %\VignetteEngine{knitr::rmarkdown}
editor_options:
  chunk_output_type: inline
---

```{r setup, eval=TRUE,include=TRUE}
library(knitr)
knitr::opts_chunk$set(tidy=FALSE, cache=TRUE,
                      dev="png",
                      message=FALSE, error=FALSE, warning=TRUE)
# source("https://bioconductor.org/biocLite.R")
#        biocLite("apeglm")
library(apeglm)
#source("https://bioconductor.org/biocLite.R")
 #       biocLite("parathyroidSE")
#library("parathyroidSE")
#       if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
#
# BiocManager::install("DESeq2")
library(DESeq2)


```

# Standard workflow

**Note:** if you use DESeq2 in published research, please cite:

> Love, M.I., Huber, W., Anders, S. (2014)
> Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2.
> *Genome Biology*, **15**:550.
> [10.1186/s13059-014-0550-8](http://dx.doi.org/10.1186/s13059-014-0550-8)

Other Bioconductor packages with similar aims are
[edgeR](http://bioconductor.org/packages/edgeR),
[limma](http://bioconductor.org/packages/limma),
[DSS](http://bioconductor.org/packages/DSS),
[EBSeq](http://bioconductor.org/packages/EBSeq), and
[baySeq](http://bioconductor.org/packages/baySeq).


### *htseq-count* input

You can use the function *DESeqDataSetFromHTSeqCount* if you
have used *htseq-count* from the
[HTSeq](http://www-huber.embl.de/users/anders/HTSeq)
python package [@Anders:2014:htseq].
For an example of using the python scripts, see the
[pasilla](http://bioconductor.org/packages/pasilla) data package. First you will want to specify a
variable which points to the directory in which the *htseq-count*
output files are located.

```{r htseqDirI, eval=TRUE,include=TRUE}
directoryGC_All <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC"

##individual lines
directoryGC_1 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_1vsAnc"
directoryGC_2 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_2vsAnc"
directoryGC_3 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_3vsAnc"
directoryGC_4 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_4vsAnc"
directoryGC_5 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_5vsAnc"
directoryGC_6 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_6vsAnc"
directoryGC_7 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_7vsAnc"
directoryGC_8 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_8vsAnc"
directoryGC_9 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_9vsAnc"
directoryGC_11 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_11vsAnc"
directoryGC_18 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_18vsAnc"
directoryGC_31 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_31vsAnc"
directoryGC_49 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_49vsAnc"
directoryGC_59 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_59vsAnc"
directoryGC_61 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_61vsAnc"
directoryGC_69 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_69vsAnc"
directoryGC_76 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_76vsAnc"
directoryGC_77 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_GC/GC_77vsAnc"
#########################################################
directoryMAold <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD"

directoryMAold_1 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/1vsAnc"
directoryMAold_2 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/2vsAnc"
directoryMAold_3 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/3vsAnc"
directoryMAold_4 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/4vsAnc"
directoryMAold_5 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/5vsAnc"
# directoryMAold_6 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/6vsAnc"
directoryMAold_7 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/7vsAnc"
directoryMAold_8 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/8vsAnc"
directoryMAold_9 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/9vsAnc"
directoryMAold_11 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/11vsAnc"
directoryMAold_15 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/15vsAnc"
directoryMAold_28 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/28vsAnc"
directoryMAold_88 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/88vsAnc"
directoryMAold_108 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/108vsAnc"
directoryMAold_119 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_OLD/119vsAnc"
########################################################
directoryMAnew <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW"

directoryMAnewS29 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/29vsAnc"

directoryMAnewS50 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/50vsAnc"

directoryMAnewS112 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/112vsAnc"

directoryMAnewS115 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/115vsAnc"

directoryMAnewS117 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/117vsAnc"

directoryMAnewS123 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/123vsAnc"

directoryMAnewS141 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/141vsAnc"

directoryMAnewS152 <- "/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/HTseq/HTseq_update_May2018/FINAL_DATA_TO_USE_MA_NEW/152vsAnc"
```

We specify which files to read in using *list.files*,
and select those files which contain the string `"treated"`
using *grep*. The *sub* function is used to
chop up the sample filename to obtain the condition status, or
you might alternatively read in a phenotypic table
using *read.table*.


```{r htseqInput,eval=TRUE,include=TRUE}
sampleFilesGC_All <- list.files(directoryGC_All)
sampleInfoGC_All <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/sample_data_GC.csv",header=TRUE)
sampleTableGC_All <- data.frame(sampleName = sampleInfoGC_All$sample,
                          fileName = sampleInfoGC_All$filename,
                          run = sampleInfoGC_All$seq_run ,
                           condition = sampleInfoGC_All$group)

sampleFilesGC_1 <- list.files(directoryGC_1)
sampleInfoGC_1 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S1vsAnc.csv",header=TRUE)
sampleTableGC_1 <- data.frame(sampleName = sampleInfoGC_1$sample,
                          fileName = sampleInfoGC_1$filename,
                          run = sampleInfoGC_1$seq_run ,
                           condition = sampleInfoGC_1$group)

sampleFilesGC_2 <- list.files(directoryGC_2)
sampleInfoGC_2 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S2vsAnc.csv",header=TRUE)
sampleTableGC_2 <- data.frame(sampleName = sampleInfoGC_2$sample,
                          fileName = sampleInfoGC_2$filename,
                          run = sampleInfoGC_2$seq_run ,
                           condition = sampleInfoGC_2$group)

sampleFilesGC_3 <- list.files(directoryGC_3)
sampleInfoGC_3 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S3vsAnc.csv",header=TRUE)
sampleTableGC_3 <- data.frame(sampleName = sampleInfoGC_3$sample,
                          fileName = sampleInfoGC_3$filename,
                          run = sampleInfoGC_3$seq_run ,
                           condition = sampleInfoGC_3$group)

sampleFilesGC_4 <- list.files(directoryGC_4)
sampleInfoGC_4 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S4vsAnc.csv",header=TRUE)
sampleTableGC_4 <- data.frame(sampleName = sampleInfoGC_4$sample,
                          fileName = sampleInfoGC_4$filename,
                          run = sampleInfoGC_4$seq_run ,
                           condition = sampleInfoGC_4$group)

sampleFilesGC_5 <- list.files(directoryGC_5)
sampleInfoGC_5 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S5vsAnc.csv",header=TRUE)
sampleTableGC_5 <- data.frame(sampleName = sampleInfoGC_5$sample,
                          fileName = sampleInfoGC_5$filename,
                          run = sampleInfoGC_5$seq_run ,
                           condition = sampleInfoGC_5$group)

sampleFilesGC_6 <- list.files(directoryGC_6)
sampleInfoGC_6 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S6vsAnc.csv",header=TRUE)
sampleTableGC_6 <- data.frame(sampleName = sampleInfoGC_6$sample,
                          fileName = sampleInfoGC_6$filename,
                          run = sampleInfoGC_6$seq_run ,
                           condition = sampleInfoGC_6$group)

sampleFilesGC_7 <- list.files(directoryGC_7)
sampleInfoGC_7 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S7vsAnc.csv",header=TRUE)
sampleTableGC_7 <- data.frame(sampleName = sampleInfoGC_7$sample,
                          fileName = sampleInfoGC_7$filename,
                          run = sampleInfoGC_7$seq_run ,
                           condition = sampleInfoGC_7$group)

sampleFilesGC_8 <- list.files(directoryGC_8)
sampleInfoGC_8 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S8vsAnc.csv",header=TRUE)
sampleTableGC_8 <- data.frame(sampleName = sampleInfoGC_8$sample,
                          fileName = sampleInfoGC_8$filename,
                          run = sampleInfoGC_8$seq_run ,
                           condition = sampleInfoGC_8$group)

sampleFilesGC_9 <- list.files(directoryGC_9)
sampleInfoGC_9 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S9vsAnc.csv",header=TRUE)
sampleTableGC_9 <- data.frame(sampleName = sampleInfoGC_9$sample,
                          fileName = sampleInfoGC_9$filename,
                          run = sampleInfoGC_9$seq_run ,
                           condition = sampleInfoGC_9$group)

sampleFilesGC_11 <- list.files(directoryGC_11)
sampleInfoGC_11 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S11vsAnc.csv",header=TRUE)
sampleTableGC_11 <- data.frame(sampleName = sampleInfoGC_11$sample,
                          fileName = sampleInfoGC_11$filename,
                          run = sampleInfoGC_11$seq_run ,
                           condition = sampleInfoGC_11$group)

sampleFilesGC_18 <- list.files(directoryGC_18)
sampleInfoGC_18 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S18vsAnc.csv",header=TRUE)
sampleTableGC_18 <- data.frame(sampleName = sampleInfoGC_18$sample,
                          fileName = sampleInfoGC_18$filename,
                          run = sampleInfoGC_18$seq_run ,
                           condition = sampleInfoGC_18$group)

sampleFilesGC_31 <- list.files(directoryGC_31)
sampleInfoGC_31 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S31vsAnc.csv",header=TRUE)
sampleTableGC_31 <- data.frame(sampleName = sampleInfoGC_31$sample,
                          fileName = sampleInfoGC_31$filename,
                          run = sampleInfoGC_31$seq_run ,
                           condition = sampleInfoGC_31$group)

sampleFilesGC_49 <- list.files(directoryGC_49)
sampleInfoGC_49 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S49vsAnc.csv",header=TRUE)
sampleTableGC_49 <- data.frame(sampleName = sampleInfoGC_49$sample,
                          fileName = sampleInfoGC_49$filename,
                          run = sampleInfoGC_49$seq_run ,
                           condition = sampleInfoGC_49$group)

sampleFilesGC_59 <- list.files(directoryGC_59)
sampleInfoGC_59 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S59vsAnc.csv",header=TRUE)
sampleTableGC_59 <- data.frame(sampleName = sampleInfoGC_59$sample,
                          fileName = sampleInfoGC_59$filename,
                          run = sampleInfoGC_59$seq_run ,
                           condition = sampleInfoGC_59$group)

sampleFilesGC_61 <- list.files(directoryGC_61)
sampleInfoGC_61 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S61vsAnc.csv",header=TRUE)
sampleTableGC_61 <- data.frame(sampleName = sampleInfoGC_61$sample,
                          fileName = sampleInfoGC_61$filename,
                          run = sampleInfoGC_61$seq_run ,
                           condition = sampleInfoGC_61$group)

sampleFilesGC_69 <- list.files(directoryGC_69)
sampleInfoGC_69 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S69vsAnc.csv",header=TRUE)
sampleTableGC_69 <- data.frame(sampleName = sampleInfoGC_69$sample,
                          fileName = sampleInfoGC_69$filename,
                          run = sampleInfoGC_69$seq_run ,
                           condition = sampleInfoGC_69$group)

sampleFilesGC_76 <- list.files(directoryGC_76)
sampleInfoGC_76 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S76vsAnc.csv",header=TRUE)
sampleTableGC_76 <- data.frame(sampleName = sampleInfoGC_76$sample,
                          fileName = sampleInfoGC_76$filename,
                          run = sampleInfoGC_76$seq_run ,
                           condition = sampleInfoGC_76$group)

sampleFilesGC_77 <- list.files(directoryGC_77)
sampleInfoGC_77 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/GC_data_S77vaAnc.csv",header=TRUE)
sampleTableGC_77 <- data.frame(sampleName = sampleInfoGC_77$sample,
                          fileName = sampleInfoGC_77$filename,
                          run = sampleInfoGC_77$seq_run ,
                           condition = sampleInfoGC_77$group)




##############################################################################
sampleFilesMAold <- list.files(directoryMAold)
sampleInfoMAold <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_sample_data.csv",header=TRUE)
sampleTableMAold <- data.frame(sampleName = sampleInfoMAold$sample,
                          fileName = sampleInfoMAold$filename,
                          condition = sampleInfoMAold$group)

sampleFilesMAold_1 <- list.files(directoryMAold_1)
sampleInfoMAold_1 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S1vsAnc.csv",header=TRUE)
sampleTableMAold_1 <- data.frame(sampleName = sampleInfoMAold_1$sample,
                          fileName = sampleInfoMAold_1$filename,
                          condition = sampleInfoMAold_1$group)

sampleFilesMAold_2 <- list.files(directoryMAold_2)
sampleInfoMAold_2 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S2vsAnc.csv",header=TRUE)
sampleTableMAold_2 <- data.frame(sampleName = sampleInfoMAold_2$sample,
                          fileName = sampleInfoMAold_2$filename,
                          condition = sampleInfoMAold_2$group)

sampleFilesMAold_3 <- list.files(directoryMAold_3)
sampleInfoMAold_3 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S3vsAnc.csv",header=TRUE)
sampleTableMAold_3 <- data.frame(sampleName = sampleInfoMAold_3$sample,
                          fileName = sampleInfoMAold_3$filename,
                          condition = sampleInfoMAold_3$group)

sampleFilesMAold_4 <- list.files(directoryMAold_4)
sampleInfoMAold_4 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S4vsAnc.csv",header=TRUE)
sampleTableMAold_4 <- data.frame(sampleName = sampleInfoMAold_4$sample,
                          fileName = sampleInfoMAold_4$filename,
                          condition = sampleInfoMAold_4$group)

sampleFilesMAold_5 <- list.files(directoryMAold_5)
sampleInfoMAold_5 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S5vsAnc.csv",header=TRUE)
sampleTableMAold_5 <- data.frame(sampleName = sampleInfoMAold_5$sample,
                          fileName = sampleInfoMAold_5$filename,
                          condition = sampleInfoMAold_5$group)

# sampleFilesMAold_6 <- list.files(directoryMAold_6)
# sampleInfoMAold_6 <- read.csv("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Paper_Results/MA_old_data_S6vsAnc.csv",header=TRUE)
# sampleTableMAold_6 <- data.frame(sampleName = sampleInfoMAold_6$sample,
#                           fileName = sampleInfoMAold_6$filename,
#                           condition = sampleInfoMAold_6$group)

sampleFilesMAold_7 <- list.files(directoryMAold_7)
sampleInfoMAold_7 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S7vsAnc.csv",header=TRUE)
sampleTableMAold_7 <- data.frame(sampleName = sampleInfoMAold_7$sample,
                          fileName = sampleInfoMAold_7$filename,
                          condition = sampleInfoMAold_7$group)

sampleFilesMAold_8 <- list.files(directoryMAold_8)
sampleInfoMAold_8 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S8vsAnc.csv",header=TRUE)
sampleTableMAold_8 <- data.frame(sampleName = sampleInfoMAold_8$sample,
                          fileName = sampleInfoMAold_8$filename,
                          condition = sampleInfoMAold_8$group)

sampleFilesMAold_9 <- list.files(directoryMAold_9)
sampleInfoMAold_9 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S9vsAnc.csv",header=TRUE)
sampleTableMAold_9 <- data.frame(sampleName = sampleInfoMAold_9$sample,
                          fileName = sampleInfoMAold_9$filename,
                          condition = sampleInfoMAold_9$group)

sampleFilesMAold_11 <- list.files(directoryMAold_11)
sampleInfoMAold_11 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S11vsAnc.csv",header=TRUE)
sampleTableMAold_11 <- data.frame(sampleName = sampleInfoMAold_11$sample,
                          fileName = sampleInfoMAold_11$filename,
                          condition = sampleInfoMAold_11$group)

sampleFilesMAold_15 <- list.files(directoryMAold_15)
sampleInfoMAold_15 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S15vsAnc.csv",header=TRUE)
sampleTableMAold_15 <- data.frame(sampleName = sampleInfoMAold_15$sample,
                          fileName = sampleInfoMAold_15$filename,
                          condition = sampleInfoMAold_15$group)

sampleFilesMAold_28 <- list.files(directoryMAold_28)
sampleInfoMAold_28 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S28vsAnc.csv",header=TRUE)
sampleTableMAold_28 <- data.frame(sampleName = sampleInfoMAold_28$sample,
                          fileName = sampleInfoMAold_28$filename,
                          condition = sampleInfoMAold_28$group)

sampleFilesMAold_88 <- list.files(directoryMAold_88)
sampleInfoMAold_88 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S88vsAnc.csv",header=TRUE)
sampleTableMAold_88 <- data.frame(sampleName = sampleInfoMAold_88$sample,
                          fileName = sampleInfoMAold_88$filename,
                          condition = sampleInfoMAold_88$group)

sampleFilesMAold_108 <- list.files(directoryMAold_108)
sampleInfoMAold_108 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S108vsAnc.csv",header=TRUE)
sampleTableMAold_108 <- data.frame(sampleName = sampleInfoMAold_108$sample,
                          fileName = sampleInfoMAold_108$filename,
                          condition = sampleInfoMAold_108$group)

sampleFilesMAold_119 <- list.files(directoryMAold_119)
sampleInfoMAold_119 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_old_data_S119vsAnc.csv",header=TRUE)
sampleTableMAold_119 <- data.frame(sampleName = sampleInfoMAold_119$sample,
                          fileName = sampleInfoMAold_119$filename,
                          condition = sampleInfoMAold_119$group)

###############################################################################
sampleFilesMAnew <- list.files(directoryMAnew)
sampleInfoMAnew <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_sample_data.csv",header=TRUE)
sampleTableMAnew <- data.frame(sampleName = sampleInfoMAnew$sample,
                          fileName = sampleInfoMAnew$filename,
                          run = sampleInfoMAnew$seq_run,
                           condition = sampleInfoMAnew$group)

sampleFilesMAnewS29 <- list.files(directoryMAnewS29)
sampleInfoMAnewS29 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S29.csv",header=TRUE)
sampleTableMAnewS29 <- data.frame(sampleName = sampleInfoMAnewS29$sample,
                          fileName = sampleInfoMAnewS29$filename,
                          run = sampleInfoMAnewS29$seq_run,
                           condition = sampleInfoMAnewS29$group)

sampleFilesMAnewS50 <- list.files(directoryMAnewS50)
sampleInfoMAnewS50 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S50.csv",header=TRUE)
sampleTableMAnewS50 <- data.frame(sampleName = sampleInfoMAnewS50$sample,
                          fileName = sampleInfoMAnewS50$filename,
                          run = sampleInfoMAnewS50$seq_run,
                           condition = sampleInfoMAnewS50$group)

sampleFilesMAnewS112 <- list.files(directoryMAnewS112)
sampleInfoMAnewS112 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S112.csv",header=TRUE)
sampleTableMAnewS112 <- data.frame(sampleName = sampleInfoMAnewS112$sample,
                          fileName = sampleInfoMAnewS112$filename,
                          run = sampleInfoMAnewS112$seq_run,
                           condition = sampleInfoMAnewS112$group)

sampleFilesMAnewS115 <- list.files(directoryMAnewS115)
sampleInfoMAnewS115 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S115.csv",header=TRUE)
sampleTableMAnewS115 <- data.frame(sampleName = sampleInfoMAnewS115$sample,
                          fileName = sampleInfoMAnewS115$filename,
                          run = sampleInfoMAnewS115$seq_run,
                           condition = sampleInfoMAnewS115$group)

sampleFilesMAnewS117 <- list.files(directoryMAnewS117)
sampleInfoMAnewS117 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S117.csv",header=TRUE)
sampleTableMAnewS117 <- data.frame(sampleName = sampleInfoMAnewS117$sample,
                          fileName = sampleInfoMAnewS117$filename,
                          run = sampleInfoMAnewS117$seq_run,
                           condition = sampleInfoMAnewS117$group)

sampleFilesMAnewS123 <- list.files(directoryMAnewS123)
sampleInfoMAnewS123 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S123.csv",header=TRUE)
sampleTableMAnewS123 <- data.frame(sampleName = sampleInfoMAnewS123$sample,
                          fileName = sampleInfoMAnewS123$filename,
                          run = sampleInfoMAnewS123$seq_run,
                           condition = sampleInfoMAnewS123$group)

# sampleFilesMAnewS141 <- list.files(directoryMAnewS141)
# sampleInfoMAnewS141 <- read.csv("/Users/hollymcqueary/Dropbox/McQueary/Dosage-Compensation/Paper_Results/MA_new_data_S141.csv",header=TRUE)
# sampleTableMAnewS141 <- data.frame(sampleName = sampleInfoMAnewS141$sample,
#                           fileName = sampleInfoMAnewS141$filename,
#                           run = sampleInfoMAnewS141$seq_run,
#                            condition = sampleInfoMAnewS141$group)

sampleFilesMAnewS152 <- list.files(directoryMAnewS152)
sampleInfoMAnewS152 <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Excel/MA_new_data_S152.csv",header=TRUE)
sampleTableMAnewS152 <- data.frame(sampleName = sampleInfoMAnewS152$sample,
                          fileName = sampleInfoMAnewS152$filename,
                          run = sampleInfoMAnewS152$seq_run,
                           condition = sampleInfoMAnewS152$group)

```

```{r other files,eval=TRUE,include=FALSE}
esrGenes <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/ESR_genes/ESR_genes_Gasch2000.csv",header=FALSE)
as.data.frame(esrGenes)
names(esrGenes)[1] <- "GENEID"

dosageSensitiveGenes <- read.csv("/Users/hollymcqueary/Dropbox/DosageCompensation/Indiv_Genes/DENonDEGenes/dosage_sensitive_genes.csv")
as.data.frame(dosageSensitiveGenes)
names(dosageSensitiveGenes)[1] <- "GENEID"

ASRgenes <- read.table("/Users/hollymcqueary/Dropbox/DosageCompensation/ESR_genes/ASRGenes.txt",header=TRUE)
as.data.frame(ASRgenes)


```
Then we build the *DESeqDataSet* using the following function:

```{r hsteqDds,eval=TRUE,include=TRUE}
library("DESeq2")
sampleTableGC_All$run <- as.factor(sampleTableGC_All$run)
sampleTableGC_All$sampleName <- as.factor(sampleTableGC_All$sampleName)
sampleTableGC_All$condition <- as.factor(sampleTableGC_All$condition)
ddsHTSeqGC_All <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_All,
                                       directory = directoryGC_All,
                                       design= ~ condition + run)
ddsHTSeqGC_All

write.csv(counts(ddsHTSeqGC_All),file="counts.DEseqGC.csv")

############################# Lines Individually ########################################
sampleTableGC_1$run <- as.factor(sampleTableGC_1$run)
sampleTableGC_1$sampleName <- as.factor(sampleTableGC_1$sampleName)
sampleTableGC_1$condition <- as.factor(sampleTableGC_1$condition)
ddsHTSeqGC_1 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_1,
                                       directory = directoryGC_1,
                                       design= ~ condition + run)
ddsHTSeqGC_1

# write.csv(counts(ddsHTSeqGC_1),file="counts.DEseqGC_S1.csv")
vsdGC_1 <- vst(ddsHTSeqGC_1,blind=FALSE)

# library(ggplot2)
# pdf("PCA_GC_by_run_1.pdf")
# ggplot(pcaDataGC_1,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_1[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_1[2],"% variance"))
#   # coord_fixed()
#   dev.off()
########## line 2
sampleTableGC_2$run <- as.factor(sampleTableGC_2$run)
sampleTableGC_2$sampleName <- as.factor(sampleTableGC_2$sampleName)
sampleTableGC_2$condition <- as.factor(sampleTableGC_2$condition)
ddsHTSeqGC_2 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_2,
                                       directory = directoryGC_2,
                                       design= ~ condition + run)
ddsHTSeqGC_2


#### line 3
sampleTableGC_3$run <- as.factor(sampleTableGC_3$run)
sampleTableGC_3$sampleName <- as.factor(sampleTableGC_3$sampleName)
sampleTableGC_3$condition <- as.factor(sampleTableGC_3$condition)
ddsHTSeqGC_3 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_3,
                                       directory = directoryGC_3,
                                       design= ~ condition + run)
ddsHTSeqGC_3

# write.csv(counts(ddsHTSeqGC_3),file="counts.DEseqGC_S1.csv")

######## line 4
sampleTableGC_4$run <- as.factor(sampleTableGC_4$run)
sampleTableGC_4$sampleName <- as.factor(sampleTableGC_4$sampleName)
sampleTableGC_4$condition <- as.factor(sampleTableGC_4$condition)
ddsHTSeqGC_4 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_4,
                                       directory = directoryGC_4,
                                       design= ~ condition + run)
ddsHTSeqGC_4


################## line 5
  sampleTableGC_5$run <- as.factor(sampleTableGC_5$run)
sampleTableGC_5$sampleName <- as.factor(sampleTableGC_5$sampleName)
sampleTableGC_5$condition <- as.factor(sampleTableGC_5$condition)
ddsHTSeqGC_5 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_5,
                                       directory = directoryGC_5,
                                       design= ~ condition)# + run)
ddsHTSeqGC_5

# write.csv(counts(ddsHTSeqGC_5),file="counts.DEseqGC_S1.csv")

################## line 6
#   sampleTableGC_6$run <- as.factor(sampleTableGC_6$run)
# sampleTableGC_6$sampleName <- as.factor(sampleTableGC_6$sampleName)
# sampleTableGC_6$condition <- as.factor(sampleTableGC_6$condition)
# ddsHTSeqGC_6 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_6,
#                                        directory = directoryGC_6,
#                                        design= ~ condition)# + run)
# ddsHTSeqGC_6
#
# write.csv(counts(ddsHTSeqGC_6),file="counts.DEseqGC_S6.csv")


################## line 7
sampleTableGC_7$run <- as.factor(sampleTableGC_7$run)
sampleTableGC_7$sampleName <- as.factor(sampleTableGC_7$sampleName)
sampleTableGC_7$condition <- as.factor(sampleTableGC_7$condition)
ddsHTSeqGC_7 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_7,
                                       directory = directoryGC_7,
                                       design= ~ condition + run)
ddsHTSeqGC_7


#################### line 8
  sampleTableGC_8$run <- as.factor(sampleTableGC_8$run)
sampleTableGC_8$sampleName <- as.factor(sampleTableGC_8$sampleName)
sampleTableGC_8$condition <- as.factor(sampleTableGC_8$condition)
ddsHTSeqGC_8 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_8,
                                       directory = directoryGC_8,
                                       design= ~ condition + run)
ddsHTSeqGC_8


################# line 9
  sampleTableGC_9$run <- as.factor(sampleTableGC_9$run)
sampleTableGC_9$sampleName <- as.factor(sampleTableGC_9$sampleName)
sampleTableGC_9$condition <- as.factor(sampleTableGC_9$condition)
ddsHTSeqGC_9 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_9,
                                       directory = directoryGC_9,
                                       design= ~ condition + run)
ddsHTSeqGC_9


################## line 11
  sampleTableGC_11$run <- as.factor(sampleTableGC_11$run)
sampleTableGC_11$sampleName <- as.factor(sampleTableGC_11$sampleName)
sampleTableGC_11$condition <- as.factor(sampleTableGC_11$condition)
ddsHTSeqGC_11 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_11,
                                       directory = directoryGC_11,
                                       design= ~ condition + run)
ddsHTSeqGC_11


##################### line 18
sampleTableGC_18$run <- as.factor(sampleTableGC_18$run)
sampleTableGC_18$sampleName <- as.factor(sampleTableGC_18$sampleName)
sampleTableGC_18$condition <- as.factor(sampleTableGC_18$condition)
ddsHTSeqGC_18 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_18,
                                       directory = directoryGC_18,
                                       design= ~ condition + run)
ddsHTSeqGC_18


  ##################### line 31
sampleTableGC_31$run <- as.factor(sampleTableGC_31$run)
sampleTableGC_31$sampleName <- as.factor(sampleTableGC_31$sampleName)
sampleTableGC_31$condition <- as.factor(sampleTableGC_31$condition)
ddsHTSeqGC_31 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_31,
                                       directory = directoryGC_31,
                                       design= ~ condition + run)
ddsHTSeqGC_31


############## line 49
  sampleTableGC_49$run <- as.factor(sampleTableGC_49$run)
sampleTableGC_49$sampleName <- as.factor(sampleTableGC_49$sampleName)
sampleTableGC_49$condition <- as.factor(sampleTableGC_49$condition)
ddsHTSeqGC_49 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_49,
                                       directory = directoryGC_49,
                                       design= ~ condition + run)
ddsHTSeqGC_49


#################### line 59
sampleTableGC_59$run <- as.factor(sampleTableGC_59$run)
sampleTableGC_59$sampleName <- as.factor(sampleTableGC_59$sampleName)
sampleTableGC_59$condition <- as.factor(sampleTableGC_59$condition)
ddsHTSeqGC_59 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_59,
                                       directory = directoryGC_59,
                                       design= ~ condition + run)
ddsHTSeqGC_59


####################### line 61
  sampleTableGC_61$run <- as.factor(sampleTableGC_61$run)
sampleTableGC_61$sampleName <- as.factor(sampleTableGC_61$sampleName)
sampleTableGC_61$condition <- as.factor(sampleTableGC_61$condition)
ddsHTSeqGC_61 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_61,
                                       directory = directoryGC_61,
                                       design= ~ condition + run)
ddsHTSeqGC_61


  ####################### line 69
  sampleTableGC_69$run <- as.factor(sampleTableGC_69$run)
sampleTableGC_69$sampleName <- as.factor(sampleTableGC_69$sampleName)
sampleTableGC_69$condition <- as.factor(sampleTableGC_69$condition)
ddsHTSeqGC_69 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_69,
                                       directory = directoryGC_69,
                                       design= ~ condition + run)
ddsHTSeqGC_69

################### line 76
  sampleTableGC_76$run <- as.factor(sampleTableGC_76$run)
sampleTableGC_76$sampleName <- as.factor(sampleTableGC_76$sampleName)
sampleTableGC_76$condition <- as.factor(sampleTableGC_76$condition)
ddsHTSeqGC_76 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_76,
                                       directory = directoryGC_76,
                                       design= ~ condition + run)
ddsHTSeqGC_76


################## line 77
sampleTableGC_77$run <- as.factor(sampleTableGC_77$run)
sampleTableGC_77$sampleName <- as.factor(sampleTableGC_77$sampleName)
sampleTableGC_77$condition <- as.factor(sampleTableGC_77$condition)
ddsHTSeqGC_77 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableGC_77,
                                       directory = directoryGC_77,
                                       design= ~ condition + run)
ddsHTSeqGC_77

# write.csv(counts(ddsHTSeqGC_77),file="counts.DEseqGC_S1.csv")
 


############################################################
  #### MA Old Samples ####
sampleTableMAold$sampleName <- as.factor(sampleTableMAold$sampleName)
sampleTableMAold$condition <- as.factor(sampleTableMAold$condition)
ddsHTSeqMAold <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold,
                                       directory = directoryMAold,
                                       design= ~ condition)

ddsHTSeqMAold
#
# write.csv(counts(ddsHTSeqMAold),file="counts.DEseqMAold.csv")



##################################### Lines Individually ########################
  ######## Line 1
sampleTableMAold_1$sampleName <- as.factor(sampleTableMAold_1$sampleName)
sampleTableMAold_1$condition <- as.factor(sampleTableMAold_1$condition)
ddsHTSeqMAold_1 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_1,
                                       directory = directoryMAold_1,
                                       design= ~ condition)

ddsHTSeqMAold_1
  ######## Line 2
sampleTableMAold_2$sampleName <- as.factor(sampleTableMAold_2$sampleName)
sampleTableMAold_2$condition <- as.factor(sampleTableMAold_2$condition)
ddsHTSeqMAold_2 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_2,
                                       directory = directoryMAold_2,
                                       design= ~ condition)

ddsHTSeqMAold_2
  ######## Line 3
sampleTableMAold_3$sampleName <- as.factor(sampleTableMAold_3$sampleName)
sampleTableMAold_3$condition <- as.factor(sampleTableMAold_3$condition)
ddsHTSeqMAold_3 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_3,
                                       directory = directoryMAold_3,
                                       design= ~ condition)

ddsHTSeqMAold_3
  ######## Line 4
sampleTableMAold_4$sampleName <- as.factor(sampleTableMAold_4$sampleName)
sampleTableMAold_4$condition <- as.factor(sampleTableMAold_4$condition)
ddsHTSeqMAold_4 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_4,
                                       directory = directoryMAold_4,
                                       design= ~ condition)

ddsHTSeqMAold_4

  ######## Line 5
sampleTableMAold_5$sampleName <- as.factor(sampleTableMAold_5$sampleName)
sampleTableMAold_5$condition <- as.factor(sampleTableMAold_5$condition)
ddsHTSeqMAold_5 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_5,
                                       directory = directoryMAold_5,
                                       design= ~ condition)

ddsHTSeqMAold_5

  ######## Line 6
# sampleTableMAold_6$sampleName <- as.factor(sampleTableMAold_6$sampleName)
# sampleTableMAold_6$condition <- as.factor(sampleTableMAold_6$condition)
# ddsHTSeqMAold_6 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_6,
#                                        directory = directoryMAold_6,
#                                        design= ~ condition)
#
# ddsHTSeqMAold_6

  ######## Line 7
sampleTableMAold_7$sampleName <- as.factor(sampleTableMAold_7$sampleName)
sampleTableMAold_7$condition <- as.factor(sampleTableMAold_7$condition)
ddsHTSeqMAold_7 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_7,
                                       directory = directoryMAold_7,
                                       design= ~ condition)

ddsHTSeqMAold_7

  ######## Line 8
sampleTableMAold_8$sampleName <- as.factor(sampleTableMAold_8$sampleName)
sampleTableMAold_8$condition <- as.factor(sampleTableMAold_8$condition)
ddsHTSeqMAold_8 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_8,
                                       directory = directoryMAold_8,
                                       design= ~ condition)

ddsHTSeqMAold_8

  ######## Line 9
sampleTableMAold_9$sampleName <- as.factor(sampleTableMAold_9$sampleName)
sampleTableMAold_9$condition <- as.factor(sampleTableMAold_9$condition)
ddsHTSeqMAold_9 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_9,
                                       directory = directoryMAold_9,
                                       design= ~ condition)

ddsHTSeqMAold_9

  ######## Line 11
sampleTableMAold_11$sampleName <- as.factor(sampleTableMAold_11$sampleName)
sampleTableMAold_11$condition <- as.factor(sampleTableMAold_11$condition)
ddsHTSeqMAold_11 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_11,
                                       directory = directoryMAold_11,
                                       design= ~ condition)

ddsHTSeqMAold_11

  ######## Line 15
sampleTableMAold_15$sampleName <- as.factor(sampleTableMAold_15$sampleName)
sampleTableMAold_15$condition <- as.factor(sampleTableMAold_15$condition)
ddsHTSeqMAold_15 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_15,
                                       directory = directoryMAold_15,
                                       design= ~ condition)

ddsHTSeqMAold_15

  ######## Line 28
sampleTableMAold_28$sampleName <- as.factor(sampleTableMAold_28$sampleName)
sampleTableMAold_28$condition <- as.factor(sampleTableMAold_28$condition)
ddsHTSeqMAold_28 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_28,
                                       directory = directoryMAold_28,
                                       design= ~ condition)

ddsHTSeqMAold_28

  ######## Line 88
sampleTableMAold_88$sampleName <- as.factor(sampleTableMAold_88$sampleName)
sampleTableMAold_88$condition <- as.factor(sampleTableMAold_88$condition)
ddsHTSeqMAold_88 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_88,
                                       directory = directoryMAold_88,
                                       design= ~ condition)

ddsHTSeqMAold_88

  ######## Line 108
sampleTableMAold_108$sampleName <- as.factor(sampleTableMAold_108$sampleName)
sampleTableMAold_108$condition <- as.factor(sampleTableMAold_108$condition)
ddsHTSeqMAold_108 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_108,
                                       directory = directoryMAold_108,
                                       design= ~ condition)

ddsHTSeqMAold_108

  ######## Line 119
sampleTableMAold_119$sampleName <- as.factor(sampleTableMAold_119$sampleName)
sampleTableMAold_119$condition <- as.factor(sampleTableMAold_119$condition)
ddsHTSeqMAold_119 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAold_119,
                                       directory = directoryMAold_119,
                                       design= ~ condition)

ddsHTSeqMAold_119

############################################################
# sampleTableMAnew$sampleName <- as.factor(sampleTableMAnew$sampleName)
# sampleTableMAnew$condition <- as.factor(sampleTableMAnew$condition)
# sampleTableMAnew$run <- as.factor(sampleTableMAnew$run)
# ddsHTSeqMAnew.run <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnew,
#                                        directory = directoryMAnew,
#                                        design= ~ condition + run)
#
# ddsHTSeqMAnew.run
# counts(ddsHTSeqMAnew.run)
# write.csv(counts(ddsHTSeqMAnew.run),file="counts.DEseqMAnew.run.csv")
ddsHTSeqMAnew <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnew,
                                       directory = directoryMAnew,
                                       design= ~ condition)

ddsHTSeqMAnew
# counts(ddsHTSeqMAnew)
# write.csv(counts(ddsHTSeqMAnew),file="counts.DEseqMAnew.csv")
#
#
# #### PCA
sampleTableMAnewS152$sampleName <- as.factor(sampleTableMAnewS152$sampleName)
sampleTableMAnewS152$condition <- as.factor(sampleTableMAnewS152$condition)
ddsHTSeqMAnewS152 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS152,
                                       directory = directoryMAnewS152,
                                       design= ~ condition + run)
#### subset the DESeq dataset into individual samples
sampleTableMAnewS29$sampleName <- as.factor(sampleTableMAnewS29$sampleName)
sampleTableMAnewS29$condition <- as.factor(sampleTableMAnewS29$condition)
ddsHTSeqMAnewS29 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS29,
                                       directory = directoryMAnewS29,
                                       design= ~ condition + run)

sampleTableMAnewS50$sampleName <- as.factor(sampleTableMAnewS50$sampleName)
sampleTableMAnewS50$condition <- as.factor(sampleTableMAnewS50$condition)
ddsHTSeqMAnewS50 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS50,
                                       directory = directoryMAnewS50,
                                       design= ~ condition + run)

sampleTableMAnewS112$sampleName <- as.factor(sampleTableMAnewS112$sampleName)
sampleTableMAnewS112$condition <- as.factor(sampleTableMAnewS112$condition)
ddsHTSeqMAnewS112 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS112,
                                       directory = directoryMAnewS112,
                                       design= ~ condition + run)

sampleTableMAnewS115$sampleName <- as.factor(sampleTableMAnewS115$sampleName)
sampleTableMAnewS115$condition <- as.factor(sampleTableMAnewS115$condition)
ddsHTSeqMAnewS115 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS115,
                                       directory = directoryMAnewS115,
                                       design= ~ condition + run)

sampleTableMAnewS117$sampleName <- as.factor(sampleTableMAnewS117$sampleName)
sampleTableMAnewS117$condition <- as.factor(sampleTableMAnewS117$condition)
ddsHTSeqMAnewS117 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS117,
                                       directory = directoryMAnewS117,
                                       design= ~ condition + run)

sampleTableMAnewS123$sampleName <- as.factor(sampleTableMAnewS123$sampleName)
sampleTableMAnewS123$condition <- as.factor(sampleTableMAnewS123$condition)
ddsHTSeqMAnewS123 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS123,
                                       directory = directoryMAnewS123,
                                       design= ~ condition)

sampleTableMAnewS152$sampleName <- as.factor(sampleTableMAnewS152$sampleName)
sampleTableMAnewS152$condition <- as.factor(sampleTableMAnewS152$condition)
ddsHTSeqMAnewS152 <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTableMAnewS152,
                                       directory = directoryMAnewS152,
                                       design= ~ condition)

```


```{r PCAs,include=FALSE,echo=FALSE}
#### PCA analysis of samples
# write.csv(counts(ddsHTSeqGC_2),file="counts.DEseqGC_S1.csv")
# vsdGC_2 <- vst(ddsHTSeqGC_2,blind=FALSE)
# rldGC_2 <- rlog(ddsHTSeqGC_2,blind=FALSE)
#
# ddsHTSeqGC_2$condition
# levels(ddsHTSeqGC_2$condition)
# head(assay(vsdGC_2),3)
# plotPCA(vsdGC_2, intgroup=c("condition"))
# pcaDataGC_2 <- plotPCA(vsdGC_2,intgroup=c("condition"),returnData=TRUE)
# percentVar_2 <- round(100 * attr(pcaDataGC_2,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_2.pdf")
# ggplot(pcaDataGC_2,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_2[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_2[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_2, intgroup=c("condition"))
# pcaDataGcRld_2 <- plotPCA(rldGC_2,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_2 <- round(100 * attr(pcaDataGcRld_2,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_2.pdf")
# ggplot(pcaDataGcRld_2,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_2[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_2[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_2)
# plotPCA(vsdGC_2, intgroup=c("run"))
# pcaDataGC_2 <- plotPCA(vsdGC_2,intgroup=c("run"),returnData=TRUE)
# percentVar_2 <- round(100 * attr(pcaDataGC_2,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_2.pdf")
# ggplot(pcaDataGC_2,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_2[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_2[2],"% variance"))
#   # coord_fixed()
#   dev.off()
# vsdGC_3 <- vst(ddsHTSeqGC_3,blind=FALSE)
# rldGC_3 <- rlog(ddsHTSeqGC_3,blind=FALSE)
#
# ddsHTSeqGC_3$condition
# levels(ddsHTSeqGC_3$condition)
# head(assay(vsdGC_3),3)
# plotPCA(vsdGC_3, intgroup=c("condition"))
# pcaDataGC_3 <- plotPCA(vsdGC_3,intgroup=c("condition"),returnData=TRUE)
# percentVar_3 <- round(100 * attr(pcaDataGC_3,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_3.pdf")
# ggplot(pcaDataGC_3,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_3[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_3[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_3, intgroup=c("condition"))
# pcaDataGcRld_3 <- plotPCA(rldGC_3,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_3 <- round(100 * attr(pcaDataGcRld_3,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_3.pdf")
# ggplot(pcaDataGcRld_3,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_3[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_3[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_3)
# plotPCA(vsdGC_3, intgroup=c("run"))
# pcaDataGC_3 <- plotPCA(vsdGC_3,intgroup=c("run"),returnData=TRUE)
# percentVar_3 <- round(100 * attr(pcaDataGC_3,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_3.pdf")
# ggplot(pcaDataGC_3,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_3[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_3[2],"% variance"))
#   # coord_fixed()
#   dev.off()
# write.csv(counts(ddsHTSeqGC_4),file="counts.DEseqGC_S1.csv")
# vsdGC_4 <- vst(ddsHTSeqGC_4,blind=FALSE)
# rldGC_4 <- rlog(ddsHTSeqGC_4,blind=FALSE)
#
# ddsHTSeqGC_4$condition
# levels(ddsHTSeqGC_4$condition)
# head(assay(vsdGC_4),3)
# plotPCA(vsdGC_4, intgroup=c("condition"))
# pcaDataGC_4 <- plotPCA(vsdGC_4,intgroup=c("condition"),returnData=TRUE)
# percentVar_4 <- round(100 * attr(pcaDataGC_4,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_4.pdf")
# ggplot(pcaDataGC_4,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_4[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_4[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_4, intgroup=c("condition"))
# pcaDataGcRld_4 <- plotPCA(rldGC_4,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_4 <- round(100 * attr(pcaDataGcRld_4,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_4.pdf")
# ggplot(pcaDataGcRld_4,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_4[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_4[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_4)
# plotPCA(vsdGC_4, intgroup=c("run"))
# pcaDataGC_4 <- plotPCA(vsdGC_4,intgroup=c("run"),returnData=TRUE)
# percentVar_4 <- round(100 * attr(pcaDataGC_4,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_4.pdf")
# ggplot(pcaDataGC_4,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_4[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_4[2],"% variance"))
#   # coord_fixed()
#   dev.off()
# vsdGC_5 <- vst(ddsHTSeqGC_5,blind=FALSE)
# rldGC_5 <- rlog(ddsHTSeqGC_5,blind=FALSE)
#
# ddsHTSeqGC_5$condition
# levels(ddsHTSeqGC_5$condition)
# head(assay(vsdGC_5),3)
# plotPCA(vsdGC_5, intgroup=c("condition"))
# pcaDataGC_5 <- plotPCA(vsdGC_5,intgroup=c("condition"),returnData=TRUE)
# percentVar_5 <- round(100 * attr(pcaDataGC_5,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_5.pdf")
# ggplot(pcaDataGC_5,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_5[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_5[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_5, intgroup=c("condition"))
# pcaDataGcRld_5 <- plotPCA(rldGC_5,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_5 <- round(100 * attr(pcaDataGcRld_5,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_5.pdf")
# ggplot(pcaDataGcRld_5,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_5[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_5[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_5)
# plotPCA(vsdGC_5, intgroup=c("run"))
# pcaDataGC_5 <- plotPCA(vsdGC_5,intgroup=c("run"),returnData=TRUE)
# percentVar_5 <- round(100 * attr(pcaDataGC_5,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_5.pdf")
# ggplot(pcaDataGC_5,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_5[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_5[2],"% variance"))
#   # coord_fixed()
#   dev.off()
# vsdGC_6 <- vst(ddsHTSeqGC_6,blind=FALSE)
# rldGC_6 <- rlog(ddsHTSeqGC_6,blind=FALSE)
#
# ddsHTSeqGC_6$condition
# levels(ddsHTSeqGC_6$condition)
# head(assay(vsdGC_6),3)
# plotPCA(vsdGC_6, intgroup=c("condition"))
# pcaDataGC_6 <- plotPCA(vsdGC_6,intgroup=c("condition"),returnData=TRUE)
# percentVar_6 <- round(100 * attr(pcaDataGC_6,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_6.pdf")
# ggplot(pcaDataGC_6,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_6[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_6[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_6, intgroup=c("condition"))
# pcaDataGcRld_6 <- plotPCA(rldGC_6,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_6 <- round(100 * attr(pcaDataGcRld_6,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_6.pdf")
# ggplot(pcaDataGcRld_6,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_6[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_6[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# write.csv(counts(ddsHTSeqGC_7),file="counts.DEseqGC_S1.csv")
# vsdGC_7 <- vst(ddsHTSeqGC_7,blind=FALSE)
# rldGC_7 <- rlog(ddsHTSeqGC_7,blind=FALSE)
#
# ddsHTSeqGC_7$condition
# levels(ddsHTSeqGC_7$condition)
# head(assay(vsdGC_7),3)
# plotPCA(vsdGC_7, intgroup=c("condition"))
# pcaDataGC_7 <- plotPCA(vsdGC_7,intgroup=c("condition"),returnData=TRUE)
# percentVar_7 <- round(100 * attr(pcaDataGC_7,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_7.pdf")
# ggplot(pcaDataGC_7,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_7[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_7[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_7, intgroup=c("condition"))
# pcaDataGcRld_7 <- plotPCA(rldGC_7,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_7 <- round(100 * attr(pcaDataGcRld_7,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_7.pdf")
# ggplot(pcaDataGcRld_7,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_7[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_7[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_7)
# plotPCA(vsdGC_7, intgroup=c("run"))
# pcaDataGC_7 <- plotPCA(vsdGC_7,intgroup=c("run"),returnData=TRUE)
# percentVar_7 <- round(100 * attr(pcaDataGC_7,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_7.pdf")
# ggplot(pcaDataGC_7,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_7[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_7[2],"% variance"))
#   # coord_fixed()
#   dev.off()
# write.csv(counts(ddsHTSeqGC_8),file="counts.DEseqGC_S1.csv")
# vsdGC_8 <- vst(ddsHTSeqGC_8,blind=FALSE)
# rldGC_8 <- rlog(ddsHTSeqGC_8,blind=FALSE)
#
# ddsHTSeqGC_8$condition
# levels(ddsHTSeqGC_8$condition)
# head(assay(vsdGC_8),3)
# plotPCA(vsdGC_8, intgroup=c("condition"))
# pcaDataGC_8 <- plotPCA(vsdGC_8,intgroup=c("condition"),returnData=TRUE)
# percentVar_8 <- round(100 * attr(pcaDataGC_8,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_8.pdf")
# ggplot(pcaDataGC_8,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_8[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_8[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_8, intgroup=c("condition"))
# pcaDataGcRld_8 <- plotPCA(rldGC_8,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_8 <- round(100 * attr(pcaDataGcRld_8,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_8.pdf")
# ggplot(pcaDataGcRld_8,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_8[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_8[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_8)
# plotPCA(vsdGC_8, intgroup=c("run"))
# pcaDataGC_8 <- plotPCA(vsdGC_8,intgroup=c("run"),returnData=TRUE)
# percentVar_8 <- round(100 * attr(pcaDataGC_8,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_8.pdf")
# ggplot(pcaDataGC_8,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_8[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_8[2],"% variance"))
#   # coord_fixed()
#   dev.off()
# write.csv(counts(ddsHTSeqGC_9),file="counts.DEseqGC_S1.csv")
# vsdGC_9 <- vst(ddsHTSeqGC_9,blind=FALSE)
# rldGC_9 <- rlog(ddsHTSeqGC_9,blind=FALSE)
#
# ddsHTSeqGC_9$condition
# levels(ddsHTSeqGC_9$condition)
# head(assay(vsdGC_9),3)
# plotPCA(vsdGC_9, intgroup=c("condition"))
# pcaDataGC_9 <- plotPCA(vsdGC_9,intgroup=c("condition"),returnData=TRUE)
# percentVar_9 <- round(100 * attr(pcaDataGC_9,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_9.pdf")
# ggplot(pcaDataGC_9,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_9[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_9[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_9, intgroup=c("condition"))
# pcaDataGcRld_9 <- plotPCA(rldGC_9,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_9 <- round(100 * attr(pcaDataGcRld_9,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_9.pdf")
# ggplot(pcaDataGcRld_9,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_9[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_9[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_9)
# plotPCA(vsdGC_9, intgroup=c("run"))
# pcaDataGC_9 <- plotPCA(vsdGC_9,intgroup=c("run"),returnData=TRUE)
# percentVar_9 <- round(100 * attr(pcaDataGC_9,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_9.pdf")
# ggplot(pcaDataGC_9,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_9[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_9[2],"% variance"))
#   # coord_fixed()
#   dev.off()
# write.csv(counts(ddsHTSeqGC_11),file="counts.DEseqGC_S1.csv")
# vsdGC_11 <- vst(ddsHTSeqGC_11,blind=FALSE)
# rldGC_11 <- rlog(ddsHTSeqGC_11,blind=FALSE)
#
# ddsHTSeqGC_11$condition
# levels(ddsHTSeqGC_11$condition)
# head(assay(vsdGC_11),3)
# plotPCA(vsdGC_11, intgroup=c("condition"))
# pcaDataGC_11 <- plotPCA(vsdGC_11,intgroup=c("condition"),returnData=TRUE)
# percentVar_11 <- round(100 * attr(pcaDataGC_11,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_11.pdf")
# ggplot(pcaDataGC_11,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_11[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_11[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_11, intgroup=c("condition"))
# pcaDataGcRld_11 <- plotPCA(rldGC_11,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_11 <- round(100 * attr(pcaDataGcRld_11,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_11.pdf")
# ggplot(pcaDataGcRld_11,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_11[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_11[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_11)
# plotPCA(vsdGC_11, intgroup=c("run"))
# pcaDataGC_11 <- plotPCA(vsdGC_11,intgroup=c("run"),returnData=TRUE)
# percentVar_11 <- round(100 * attr(pcaDataGC_11,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_11.pdf")
# ggplot(pcaDataGC_11,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_11[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_11[2],"% variance"))
#   # coord_fixed()
#   dev.off()
# write.csv(counts(ddsHTSeqGC_18),file="counts.DEseqGC_S1.csv")
# vsdGC_18 <- vst(ddsHTSeqGC_18,blind=FALSE)
# rldGC_18 <- rlog(ddsHTSeqGC_18,blind=FALSE)
#
# ddsHTSeqGC_18$condition
# levels(ddsHTSeqGC_18$condition)
# head(assay(vsdGC_18),3)
# plotPCA(vsdGC_18, intgroup=c("condition"))
# pcaDataGC_18 <- plotPCA(vsdGC_18,intgroup=c("condition"),returnData=TRUE)
# percentVar_18 <- round(100 * attr(pcaDataGC_18,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_18.pdf")
# ggplot(pcaDataGC_18,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_18[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_18[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_18, intgroup=c("condition"))
# pcaDataGcRld_18 <- plotPCA(rldGC_18,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_18 <- round(100 * attr(pcaDataGcRld_18,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_18.pdf")
# ggplot(pcaDataGcRld_18,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_18[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_18[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_18)
# plotPCA(vsdGC_18, intgroup=c("run"))
# pcaDataGC_18 <- plotPCA(vsdGC_18,intgroup=c("run"),returnData=TRUE)
# percentVar_18 <- round(100 * attr(pcaDataGC_18,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_18.pdf")
# ggplot(pcaDataGC_18,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_18[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_18[2],"% variance"))
#   # coord_fixed()
#   dev.off()
# write.csv(counts(ddsHTSeqGC_31),file="counts.DEseqGC_S31.csv")
# vsdGC_31 <- vst(ddsHTSeqGC_31,blind=FALSE)
# rldGC_31 <- rlog(ddsHTSeqGC_31,blind=FALSE)
#
# ddsHTSeqGC_31$condition
# levels(ddsHTSeqGC_31$condition)
# head(assay(vsdGC_31),3)
# plotPCA(vsdGC_31, intgroup=c("condition"))
# pcaDataGC_31 <- plotPCA(vsdGC_31,intgroup=c("condition"),returnData=TRUE)
# percentVar_31 <- round(100 * attr(pcaDataGC_31,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_31.pdf")
# ggplot(pcaDataGC_31,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_31[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_31[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_31, intgroup=c("condition"))
# pcaDataGcRld_31 <- plotPCA(rldGC_31,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_31 <- round(100 * attr(pcaDataGcRld_31,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_31.pdf")
# ggplot(pcaDataGcRld_31,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_31[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_31[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_31)
# plotPCA(vsdGC_31, intgroup=c("run"))
# pcaDataGC_31 <- plotPCA(vsdGC_31,intgroup=c("run"),returnData=TRUE)
# percentVar_31 <- round(100 * attr(pcaDataGC_31,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_31.pdf")
# ggplot(pcaDataGC_31,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_31[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_31[2],"% variance"))
#   # coord_fixed()
#   dev.off()
# write.csv(counts(ddsHTSeqGC_49),file="counts.DEseqGC_S1.csv")
# vsdGC_49 <- vst(ddsHTSeqGC_49,blind=FALSE)
# rldGC_49 <- rlog(ddsHTSeqGC_49,blind=FALSE)
#
# ddsHTSeqGC_49$condition
# levels(ddsHTSeqGC_49$condition)
# head(assay(vsdGC_49),3)
# plotPCA(vsdGC_49, intgroup=c("condition"))
# pcaDataGC_49 <- plotPCA(vsdGC_49,intgroup=c("condition"),returnData=TRUE)
# percentVar_49 <- round(100 * attr(pcaDataGC_49,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_49.pdf")
# ggplot(pcaDataGC_49,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_49[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_49[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_49, intgroup=c("condition"))
# pcaDataGcRld_49 <- plotPCA(rldGC_49,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_49 <- round(100 * attr(pcaDataGcRld_49,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_49.pdf")
# ggplot(pcaDataGcRld_49,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_49[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_49[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_49)
# plotPCA(vsdGC_49, intgroup=c("run"))
# pcaDataGC_49 <- plotPCA(vsdGC_49,intgroup=c("run"),returnData=TRUE)
# percentVar_49 <- round(100 * attr(pcaDataGC_49,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_49.pdf")
# ggplot(pcaDataGC_49,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_49[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_49[2],"% variance"))
#   # coord_fixed()
#   dev.off()
# write.csv(counts(ddsHTSeqGC_59),file="counts.DEseqGC_S1.csv")
# vsdGC_59 <- vst(ddsHTSeqGC_59,blind=FALSE)
# rldGC_59 <- rlog(ddsHTSeqGC_59,blind=FALSE)
#
# ddsHTSeqGC_59$condition
# levels(ddsHTSeqGC_59$condition)
# head(assay(vsdGC_59),3)
# plotPCA(vsdGC_59, intgroup=c("condition"))
# pcaDataGC_59 <- plotPCA(vsdGC_59,intgroup=c("condition"),returnData=TRUE)
# percentVar_59 <- round(100 * attr(pcaDataGC_59,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_59.pdf")
# ggplot(pcaDataGC_59,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_59[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_59[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_59, intgroup=c("condition"))
# pcaDataGcRld_59 <- plotPCA(rldGC_59,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_59 <- round(100 * attr(pcaDataGcRld_59,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_59.pdf")
# ggplot(pcaDataGcRld_59,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_59[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_59[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_59)
# plotPCA(vsdGC_59, intgroup=c("run"))
# pcaDataGC_59 <- plotPCA(vsdGC_59,intgroup=c("run"),returnData=TRUE)
# percentVar_59 <- round(100 * attr(pcaDataGC_59,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_59.pdf")
# ggplot(pcaDataGC_59,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_59[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_59[2],"% variance"))
#   # coord_fixed()
#   dev.off()
# transformation of count data
# vsdGC_All <- vst(ddsHTSeqGC_All,blind=FALSE)
# rldGC_All <- rlog(ddsHTSeqGC_All,blind=FALSE)
# 
# ddsHTSeqGC_All$condition
# levels(ddsHTSeqGC_All$condition)
# head(assay(vsdGC_All),3)
# plotPCA(vsdGC_All, intgroup=c("condition"))
# pcaDataGC_All <- plotPCA(vsdGC_All,intgroup=c("condition"),returnData=TRUE)
# percentVar_All <- round(100 * attr(pcaDataGC_All,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_All.pdf")
# ggplot(pcaDataGC_All,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_All[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_All[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()

# ## pca by sequencing run
# colData(ddsHTSeqGC_All)
# plotPCA(vsdGC_All, intgroup=c("run"))
# pcaDataGC_All <- plotPCA(vsdGC_All,intgroup=c("run"),returnData=TRUE)
# percentVar_All <- round(100 * attr(pcaDataGC_All,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_All.pdf")
# ggplot(pcaDataGC_All,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_All[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_All[2],"% variance"))
#   # coord_fixed()
#   dev.off()
#
### PCA with rld data
#   plotPCA(rldGC_All, intgroup=c("condition"))
# pcaDataGcRld_All <- plotPCA(rldGC_All,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_All <- round(100 * attr(pcaDataGcRld_All,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld.pdf")
# ggplot(pcaDataGcRld_All,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_All[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_All[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()


## PCA with rld data by run
#   plotPCA(rldGC.run, intgroup=c("run"))
# pcaDataGcRldRun <- plotPCA(rldGC.run,intgroup=c("run"),returnData=TRUE)
# percentVarRldRun <- round(100 * attr(pcaDataGcRldRun,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_Run.pdf")
# ggplot(pcaDataGcRldRun,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRldRun[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRldRun[2],"% variance")) +
#   scale_fill_hue(l=80)
# dev.off()

rldGC_1 <- rlog(ddsHTSeqGC_1,blind=FALSE)

ddsHTSeqGC_1$condition
levels(ddsHTSeqGC_1$condition)
head(assay(vsdGC_1),3)
plotPCA(vsdGC_1, intgroup=c("condition"))
pcaDataGC_1 <- plotPCA(vsdGC_1,intgroup=c("condition"),returnData=TRUE)
percentVar_1 <- round(100 * attr(pcaDataGC_1,"percentVar"))
library(ggplot2)
pdf("PCA_GC_1.pdf")
ggplot(pcaDataGC_1,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar_1[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_1[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
dev.off()
  plotPCA(rldGC_1, intgroup=c("condition"))
pcaDataGcRld_1 <- plotPCA(rldGC_1,intgroup=c("condition"),returnData=TRUE)
percentVarRld_1 <- round(100 * attr(pcaDataGcRld_1,"percentVar"))
library(ggplot2)
pdf("PCA_GC_rld_1.pdf")
ggplot(pcaDataGcRld_1,aes(PC1,PC2,shape=condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVarRld_1[1],"% variance")) +
  ylab(paste0("PC2: ",percentVarRld_1[2],"% variance")) +
  scale_fill_hue(l=80)+
  scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
  # coord_fixed()
dev.off()
colData(ddsHTSeqGC_1)
plotPCA(vsdGC_1, intgroup=c("run"))
pcaDataGC_1 <- plotPCA(vsdGC_1,intgroup=c("run"),returnData=TRUE)
percentVar_1 <- round(100 * attr(pcaDataGC_1,"percentVar"))

# vsdMAnew152 <- varianceStabilizingTransformation(ddsHTSeqMAnewS152,blind=FALSE)
# rldMAnew152 <- rlog(ddsHTSeqMAnewS152,blind=FALSE)
# head(assay(vsdMAnew152),3)
# DESeq2::plotPCA(vsdMAnew152, intgroup=c("condition"))
# pcaDataMAnewVsd152 <- DESeq2::plotPCA(vsdMAnew152,intgroup=c("condition"),returnData=TRUE)
# pcaDataMAnewRld152 <- DESeq2::plotPCA(rldMAnew152,intgroup=c("condition"),returnData=TRUE)
# # pcaDataMAnewRaw <- DESeq2::plotPCA(ddsHTSeqMAnew,intgroup=c("condition"),returnData=TRUE)
#
# percentVarMAnewVsd152 <- round(100 * attr(pcaDataMAnewVsd152,"percentVar"))
# percentVarMAnewRld152 <- round(100 * attr(pcaDataMAnewRld152,"percentVar"))

# vsdMAnew.run <- vst(ddsHTSeqMAnew.run,blind=FALSE)
# rldMAnew.run <- rlog(ddsHTSeqMAnew.run,blind=FALSE)
# head(assay(vsdMAnew.run),3)

# # library(ggplot2)
# pdf("MA_new_PCA_Line.pdf")
# ggplot(pcaDataMAnewRld,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarMAnew[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarMAnew[2],"% variance")) +
#   scale_fill_hue(l=80) +
#   scale_shape_manual(values=c(0,1,2,3,4,7,8,11))
# dev.off()

# pdf("MA_new_PCA_Line_vsd152.pdf")
# ggplot(pcaDataMAnewVsd152,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarMAnewVsd152[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarMAnewVsd152[2],"% variance")) +
#   scale_fill_hue(l=80) +
#   scale_shape_manual(values=c(0,1,2,3,4,7,8,11))
# dev.off()

# pdf("MA_new_PCA_Line_rld.pdf")
# ggplot(pcaDataMAnewRld,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarMAnewRld[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarMAnewRld[2],"% variance")) +
#   scale_fill_hue(l=80) +
#   scale_shape_manual(values=c(0,1,2,3,4,7,8,11))
# dev.off()

# ##can you do PCA on just the raw counts (no transformation)?
# ### apparently not
# # pdf("MA_new_PCA_Line.pdf")
# # ggplot(pcaDataMAnew,aes(PC1,PC2,shape=condition)) +
# #   geom_point(size=3) +
# #   xlab(paste0("PC1: ",percentVarMAnew[1],"% variance")) +
# #   ylab(paste0("PC2: ",percentVarMAnew[2],"% variance")) +
# #   scale_fill_hue(l=80) +
# #   scale_shape_manual(values=c(0,1,2,3,4,7,8,11))
# # dev.off()
#
# # by sequencing run
# # colData(ddsHTSeqMAnew.run)
# # DESeq2::plotPCA(vsdMAnew.run, intgroup=c("run"))
# # pcaDataMAnew.run <- DESeq2::plotPCA(vsdMAnew.run,intgroup=c("run"),returnData=TRUE)
# # percentVarMAnew.run <- round(100 * attr(pcaDataMAnew.run,"percentVar"))
# # # library(ggplot2)
# # pdf("MA_new_PCA_SeqRun.pdf")
# # ggplot(pcaDataMAnew.run,aes(PC1,PC2,color=run)) +
# #   geom_point(size=3) +
# #   xlab(paste0("PC1: ",percentVarMAnew.run[1],"% variance")) +
# #   ylab(paste0("PC2: ",percentVarMAnew.run[2],"% variance")) +
# #   scale_fill_hue(l=80)
# # dev.off()
#   # scale_shape_manual(values=c(0,1,2,3,4,7,8,11,15))
#   
# subsetMAnewS29 <- which(ddsHTSeqMAnew$condition=="29")
# subsetMAnewAnc <- which(ddsHTSeqMAnew$condition=="Anc")
# subsetMAnewS29 <- c(subsetMAnewS29,subsetMAnewAnc)
# ddsHTSeqMAnewS29 <- ddsHTSeqMAnew[,subsetMAnewS29]
# ddsHTSeqMAnewS29$condition
# ddsHTSeqMAnewS29$condition <- droplevels(ddsHTSeqMAnewS29$condition)
# ddsHTSeqMAnewS29$condition  
#   
# subsetMAnewS112 <- which(ddsHTSeqMAnew$condition=="112")
# subsetMAnewS112 <- c(subsetMAnewS112,subsetMAnewAnc)
# ddsHTSeqMAnewS112 <- ddsHTSeqMAnew[,subsetMAnewS112]
# ddsHTSeqMAnewS112$condition
# ddsHTSeqMAnewS112$condition <- droplevels(ddsHTSeqMAnewS112$condition)
# ddsHTSeqMAnewS112$condition
#
#
# subsetMAnewS50 <- which(ddsHTSeqMAnew$condition=="50")
# subsetMAnewS50 <- c(subsetMAnewS50,subsetMAnewAnc)
# ddsHTSeqMAnewS50 <- ddsHTSeqMAnew[,subsetMAnewS50]
# ddsHTSeqMAnewS50$condition
# ddsHTSeqMAnewS50$condition <- droplevels(ddsHTSeqMAnewS50$condition)
# ddsHTSeqMAnewS50$condition  
# # subsetMAnewS112 <- which(ddsHTSeqMAnew$condition=="112")
# subsetMAnewS115 <- which(ddsHTSeqMAnew$condition=="115")
# subsetMAnewS115 <- c(subsetMAnewS115,subsetMAnewAnc)
# ddsHTSeqMAnewS115 <- ddsHTSeqMAnew[,subsetMAnewS115]
# ddsHTSeqMAnewS115$condition
# ddsHTSeqMAnewS115$condition <- droplevels(ddsHTSeqMAnewS115$condition)
# ddsHTSeqMAnewS115$condition  
# subsetMAnewS117 <- which(ddsHTSeqMAnew$condition=="117")
# subsetMAnewS117 <- c(subsetMAnewS117,subsetMAnewAnc)
# ddsHTSeqMAnewS117 <- ddsHTSeqMAnew[,subsetMAnewS117]
# ddsHTSeqMAnewS117$condition
# ddsHTSeqMAnewS117$condition <- droplevels(ddsHTSeqMAnewS117$condition)
# ddsHTSeqMAnewS117$condition
# subsetMAnewS123 <- which(ddsHTSeqMAnew$condition=="123")
# subsetMAnewS123 <- c(subsetMAnewS123,subsetMAnewAnc)
# ddsHTSeqMAnewS123 <- ddsHTSeqMAnew[,subsetMAnewS123]
# ddsHTSeqMAnewS123$condition
# ddsHTSeqMAnewS123$condition <- droplevels(ddsHTSeqMAnewS123$condition)
# ddsHTSeqMAnewS123$condition
# subsetMAnewS152 <- which(ddsHTSeqMAnew$condition=="152")
# subsetMAnewS152 <- c(subsetMAnewS152,subsetMAnewAnc)
# ddsHTSeqMAnewS152 <- ddsHTSeqMAnew[,subsetMAnewS152]
# ddsHTSeqMAnewS152$condition
# ddsHTSeqMAnewS152$condition <- droplevels(ddsHTSeqMAnewS152$condition)
# ddsHTSeqMAnewS152$condition
#
#
# subsetMAnewS29.run <- which(ddsHTSeqMAnew.run$condition=="29")
# subsetMAnewAnc.run <- which(ddsHTSeqMAnew.run$condition=="Anc")
# subsetMAnewS29.run <- c(subsetMAnewS29.run,subsetMAnewAnc.run)
# ddsHTSeqMAnewS29.run <- ddsHTSeqMAnew.run[,subsetMAnewS29.run]
# ddsHTSeqMAnewS29.run$condition
# ddsHTSeqMAnewS29.run$condition <- droplevels(ddsHTSeqMAnewS29.run$condition)
# ddsHTSeqMAnewS29.run$condition  
#   
# subsetMAnewS112.run <- which(ddsHTSeqMAnew.run$condition=="112")
# subsetMAnewS112.run <- c(subsetMAnewS112.run,subsetMAnewAnc.run)
# ddsHTSeqMAnewS112.run <- ddsHTSeqMAnew.run[,subsetMAnewS112.run]
# ddsHTSeqMAnewS112.run$condition
# ddsHTSeqMAnewS112.run$condition <- droplevels(ddsHTSeqMAnewS112.run$condition)
# ddsHTSeqMAnewS112.run$condition
#
#
# subsetMAnewS50.run <- which(ddsHTSeqMAnew.run$condition=="50")
# subsetMAnewS50.run <- c(subsetMAnewS50.run,subsetMAnewAnc.run)
# ddsHTSeqMAnewS50.run <- ddsHTSeqMAnew.run[,subsetMAnewS50.run]
# ddsHTSeqMAnewS50.run$condition
# ddsHTSeqMAnewS50.run$condition <- droplevels(ddsHTSeqMAnewS50.run$condition)
# ddsHTSeqMAnewS50.run$condition  
# # subsetMAnewS112 <- which(ddsHTSeqMAnew$condition=="112")
# subsetMAnewS115.run <- which(ddsHTSeqMAnew.run$condition=="115")
# subsetMAnewS115.run <- c(subsetMAnewS115.run,subsetMAnewAnc.run)
# ddsHTSeqMAnewS115.run <- ddsHTSeqMAnew.run[,subsetMAnewS115.run]
# ddsHTSeqMAnewS115.run$condition
# ddsHTSeqMAnewS115.run$condition <- droplevels(ddsHTSeqMAnewS115.run$condition)
# ddsHTSeqMAnewS115.run$condition  
# subsetMAnewS117.run <- which(ddsHTSeqMAnew.run$condition=="117")
# subsetMAnewS117.run <- c(subsetMAnewS117.run,subsetMAnewAnc.run)
# ddsHTSeqMAnewS117.run <- ddsHTSeqMAnew.run[,subsetMAnewS117.run]
# ddsHTSeqMAnewS117.run$condition
# ddsHTSeqMAnewS117.run$condition <- droplevels(ddsHTSeqMAnewS117.run$condition)
# ddsHTSeqMAnewS117.run$condition
# subsetMAnewS123.run <- which(ddsHTSeqMAnew.run$condition=="123")
# subsetMAnewS123.run <- c(subsetMAnewS123.run,subsetMAnewAnc.run)
# ddsHTSeqMAnewS123.run <- ddsHTSeqMAnew.run[,subsetMAnewS123.run]
# ddsHTSeqMAnewS123.run$condition
# ddsHTSeqMAnewS123.run$condition <- droplevels(ddsHTSeqMAnewS123.run$condition)
# ddsHTSeqMAnewS123.run$condition
# subsetMAnewS152.run <- which(ddsHTSeqMAnew.run$condition=="152")
# subsetMAnewS152.run <- c(subsetMAnewS152.run,subsetMAnewAnc.run)
# ddsHTSeqMAnewS152.run <- ddsHTSeqMAnew.run[,subsetMAnewS152.run]
# ddsHTSeqMAnewS152.run$condition
# ddsHTSeqMAnewS152.run$condition <- droplevels(ddsHTSeqMAnewS152.run$condition)
# ddsHTSeqMAnewS152.run$condition

# subsetMAnewAll <- c(subsetMAnewS50,subsetMAnewS115, subsetMAnewS117,subsetMAnewS123,subsetMAnewS152,subsetMAnewAnc)
#
# ddsHTSeqMAnew.short <- ddsHTSeqMAnew[,subsetMAnewAll]
#find the loadings for PCA
# pcaTest <- prcomp(pcaDataMAnew)
# DESeq2:::plotPCA.DESeqTransform

##### PCA analysis of samples
## transformation of count data
# vsdMA_old <- vst(ddsHTSeqMAold,blind=FALSE)
# rldMA_old <- rlog(ddsHTSeqMAold,blind=FALSE)
# head(assay(vsdMA_old),3)
# plotPCA(vsdMA_old, intgroup=c("condition"))
# pcaDataMA_old <- plotPCA(vsdMA_old,intgroup=c("condition"),returnData=TRUE)
# percentVarMAold <- round(100 * attr(pcaDataMA_old,"percentVar"))
# # library(ggplot2)
# pdf("MA_old_PCA_Line.pdf")
# ggplot(pcaDataMA_old,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarMAold[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarMAold[2],"% variance")) +
#   scale_fill_hue(l=80) +
#   scale_shape_manual(values=c(0,1,2,3,4,7,8,11,15,16,17,6,22,9,12,10))
# #  coord_fixed()
#   dev.off()
#   
#   ### PCA with rld data
#   plotPCA(rldMA_old, intgroup=c("condition"))
# pcaDataMA_oldRld <- plotPCA(rldMA_old,intgroup=c("condition"),returnData=TRUE)
# percentVarMAoldRld <- round(100 * attr(pcaDataMA_oldRld,"percentVar"))
# # library(ggplot2)
# pdf("MA_old_PCA_Line_Rld.pdf")
# ggplot(pcaDataMA_oldRld,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarMAoldRld[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarMAoldRld[2],"% variance")) +
#   scale_fill_hue(l=80) +
#   scale_shape_manual(values=c(0,1,2,3,4,7,8,11,15,16,17,6,22,9,12,10))
# #  coord_fixed()
#   dev.off()
#   

# vsdGC_77 <- vst(ddsHTSeqGC_77,blind=FALSE)
# rldGC_77 <- rlog(ddsHTSeqGC_77,blind=FALSE)
#
# ddsHTSeqGC_77$condition
# levels(ddsHTSeqGC_77$condition)
# head(assay(vsdGC_77),3)
# plotPCA(vsdGC_77, intgroup=c("condition"))
# pcaDataGC_77 <- plotPCA(vsdGC_77,intgroup=c("condition"),returnData=TRUE)
# percentVar_77 <- round(100 * attr(pcaDataGC_77,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_77.pdf")
# ggplot(pcaDataGC_77,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_77[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_77[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_77, intgroup=c("condition"))
# pcaDataGcRld_77 <- plotPCA(rldGC_77,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_77 <- round(100 * attr(pcaDataGcRld_77,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_77.pdf")
# ggplot(pcaDataGcRld_77,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_77[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_77[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_77)
# plotPCA(vsdGC_77, intgroup=c("run"))
# pcaDataGC_77 <- plotPCA(vsdGC_77,intgroup=c("run"),returnData=TRUE)
# percentVar_77 <- round(100 * attr(pcaDataGC_77,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_77.pdf")
# ggplot(pcaDataGC_77,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_77[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_77[2],"% variance"))
#   # coord_fixed()
#   dev.off()  
#  

# write.csv(counts(ddsHTSeqGC_76),file="counts.DEseqGC_S1.csv")
# vsdGC_76 <- vst(ddsHTSeqGC_76,blind=FALSE)
# rldGC_76 <- rlog(ddsHTSeqGC_76,blind=FALSE)
#
# ddsHTSeqGC_76$condition
# levels(ddsHTSeqGC_76$condition)
# head(assay(vsdGC_76),3)
# plotPCA(vsdGC_76, intgroup=c("condition"))
# pcaDataGC_76 <- plotPCA(vsdGC_76,intgroup=c("condition"),returnData=TRUE)
# percentVar_76 <- round(100 * attr(pcaDataGC_76,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_76.pdf")
# ggplot(pcaDataGC_76,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_76[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_76[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_76, intgroup=c("condition"))
# pcaDataGcRld_76 <- plotPCA(rldGC_76,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_76 <- round(100 * attr(pcaDataGcRld_76,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_76.pdf")
# ggplot(pcaDataGcRld_76,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_76[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_76[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_76)
# plotPCA(vsdGC_76, intgroup=c("run"))
# pcaDataGC_76 <- plotPCA(vsdGC_76,intgroup=c("run"),returnData=TRUE)
# percentVar_76 <- round(100 * attr(pcaDataGC_76,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_76.pdf")
# ggplot(pcaDataGC_76,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_76[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_76[2],"% variance"))
#   # coord_fixed()
#   dev.off()
#
# write.csv(counts(ddsHTSeqGC_69),file="counts.DEseqGC_S69.csv")
# vsdGC_69 <- vst(ddsHTSeqGC_69,blind=FALSE)
# rldGC_69 <- rlog(ddsHTSeqGC_69,blind=FALSE)
#
# ddsHTSeqGC_69$condition
# levels(ddsHTSeqGC_69$condition)
# head(assay(vsdGC_69),3)
# plotPCA(vsdGC_69, intgroup=c("condition"))
# pcaDataGC_69 <- plotPCA(vsdGC_69,intgroup=c("condition"),returnData=TRUE)
# percentVar_69 <- round(100 * attr(pcaDataGC_69,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_69.pdf")
# ggplot(pcaDataGC_69,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_69[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_69[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_69, intgroup=c("condition"))
# pcaDataGcRld_69 <- plotPCA(rldGC_69,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_69 <- round(100 * attr(pcaDataGcRld_69,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_69.pdf")
# ggplot(pcaDataGcRld_69,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_69[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_69[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_69)
# plotPCA(vsdGC_69, intgroup=c("run"))
# pcaDataGC_69 <- plotPCA(vsdGC_69,intgroup=c("run"),returnData=TRUE)
# percentVar_69 <- round(100 * attr(pcaDataGC_69,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_69.pdf")
# ggplot(pcaDataGC_69,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_69[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_69[2],"% variance"))
#   # coord_fixed()
#   dev.off()
# write.csv(counts(ddsHTSeqGC_61),file="counts.DEseqGC_S1.csv")
# vsdGC_61 <- vst(ddsHTSeqGC_61,blind=FALSE)
# rldGC_61 <- rlog(ddsHTSeqGC_61,blind=FALSE)
#
# ddsHTSeqGC_61$condition
# levels(ddsHTSeqGC_61$condition)
# head(assay(vsdGC_61),3)
# plotPCA(vsdGC_61, intgroup=c("condition"))
# pcaDataGC_61 <- plotPCA(vsdGC_61,intgroup=c("condition"),returnData=TRUE)
# percentVar_61 <- round(100 * attr(pcaDataGC_61,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_61.pdf")
# ggplot(pcaDataGC_61,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_61[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_61[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
#   plotPCA(rldGC_61, intgroup=c("condition"))
# pcaDataGcRld_61 <- plotPCA(rldGC_61,intgroup=c("condition"),returnData=TRUE)
# percentVarRld_61 <- round(100 * attr(pcaDataGcRld_61,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_rld_61.pdf")
# ggplot(pcaDataGcRld_61,aes(PC1,PC2,shape=condition)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVarRld_61[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVarRld_61[2],"% variance")) +
#   scale_fill_hue(l=80)+
#   scale_shape_manual(values=c(0,1,2,3,4,5,6,7,8,9,10,11,12,13,15,16,17,18,19))
#   # coord_fixed()
# dev.off()
# colData(ddsHTSeqGC_61)
# plotPCA(vsdGC_61, intgroup=c("run"))
# pcaDataGC_61 <- plotPCA(vsdGC_61,intgroup=c("run"),returnData=TRUE)
# percentVar_61 <- round(100 * attr(pcaDataGC_61,"percentVar"))
# library(ggplot2)
# pdf("PCA_GC_by_run_61.pdf")
# ggplot(pcaDataGC_61,aes(PC1,PC2,color=run)) +
#   geom_point(size=3) +
#   xlab(paste0("PC1: ",percentVar_61[1],"% variance")) +
#   ylab(paste0("PC2: ",percentVar_61[2],"% variance"))
#   # coord_fixed()
#   dev.off()
```

### Note on factor levels

By default, R will choose a *reference level* for factors based on
alphabetical order. Then, if you never tell the DESeq2 functions which
level you want to compare against (e.g. which level represents the
control group), the comparisons will be based on the alphabetical
order of the levels. There are two solutions: you can either
explicitly tell *results* which comparison to make using the
`contrast` argument (this will be shown later), or you can explicitly
set the factors levels. You should only change the factor levels
of variables in the design before running the DESeq2 analysis, not
during or afterward.

```{r relevel,eval=TRUE,include=TRUE,echo=FALSE}
ddsHTSeqGCcondition <- relevel(ddsHTSeqGC_All$condition, ref = "Holly_Anc")
# ddsHTSeqGC.short$condition <- relevel(ddsHTSeqGC.short$condition,ref="Holly_Anc")

#filtered genes
# ddsHTSeqGC.fil$condition <- relevel(ddsHTSeqGC.fil$condition, ref = "Holly_Anc")
# ## individual samples
ddsHTSeqGC_1$condition <- relevel(ddsHTSeqGC_1$condition, ref = "Holly_Anc")
ddsHTSeqGC_2$condition <- relevel(ddsHTSeqGC_2$condition, ref = "Holly_Anc")
ddsHTSeqGC_3$condition <- relevel(ddsHTSeqGC_3$condition, ref = "Holly_Anc")
ddsHTSeqGC_4$condition <- relevel(ddsHTSeqGC_4$condition, ref = "Holly_Anc")
ddsHTSeqGC_5$condition <- relevel(ddsHTSeqGC_5$condition, ref = "Holly_Anc")
# ddsHTSeqGC_6$condition <- relevel(ddsHTSeqGC_6$condition, ref = "Holly_Anc")
ddsHTSeqGC_7$condition <- relevel(ddsHTSeqGC_7$condition, ref = "Holly_Anc")
ddsHTSeqGC_8$condition <- relevel(ddsHTSeqGC_8$condition, ref = "Holly_Anc")
ddsHTSeqGC_9$condition <- relevel(ddsHTSeqGC_9$condition, ref = "Holly_Anc")
ddsHTSeqGC_11$condition <- relevel(ddsHTSeqGC_11$condition, ref = "Holly_Anc")
ddsHTSeqGC_18$condition <- relevel(ddsHTSeqGC_18$condition, ref = "Holly_Anc")
ddsHTSeqGC_31$condition <- relevel(ddsHTSeqGC_31$condition, ref = "Holly_Anc")
ddsHTSeqGC_49$condition <- relevel(ddsHTSeqGC_49$condition, ref = "Holly_Anc")
ddsHTSeqGC_59$condition <- relevel(ddsHTSeqGC_59$condition, ref = "Holly_Anc")
ddsHTSeqGC_61$condition <- relevel(ddsHTSeqGC_61$condition, ref = "Holly_Anc")
ddsHTSeqGC_69$condition <- relevel(ddsHTSeqGC_69$condition, ref = "Holly_Anc")
ddsHTSeqGC_76$condition <- relevel(ddsHTSeqGC_76$condition, ref = "Holly_Anc")
ddsHTSeqGC_77$condition <- relevel(ddsHTSeqGC_77$condition, ref = "Holly_Anc")

# ddsHTSeqGCS1.run$condition <- relevel(ddsHTSeqGCS1.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS2.run$condition <- relevel(ddsHTSeqGCS2.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS3.run$condition <- relevel(ddsHTSeqGCS3.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS4.run$condition <- relevel(ddsHTSeqGCS4.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS5.run$condition <- relevel(ddsHTSeqGCS5.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS6.run$condition <- relevel(ddsHTSeqGCS6.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS7.run$condition <- relevel(ddsHTSeqGCS7.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS8.run$condition <- relevel(ddsHTSeqGCS8.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS9.run$condition <- relevel(ddsHTSeqGCS9.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS11.run$condition <- relevel(ddsHTSeqGCS11.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS18.run$condition <- relevel(ddsHTSeqGCS18.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS31.run$condition <- relevel(ddsHTSeqGCS31.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS49.run$condition <- relevel(ddsHTSeqGCS49.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS59.run$condition <- relevel(ddsHTSeqGCS59.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS61.run$condition <- relevel(ddsHTSeqGCS61.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS69.run$condition <- relevel(ddsHTSeqGCS69.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS76.run$condition <- relevel(ddsHTSeqGCS76.run$condition, ref = "Holly_Anc")
# ddsHTSeqGCS77.run$condition <- relevel(ddsHTSeqGCS77.run$condition, ref = "Holly_Anc")
# ddsHTSeqGC.short$condition <- relevel(ddsHTSeqGC.short$condition, ref="Holly_Anc")
###############################################################################
ddsHTSeqMAold$condition <- relevel(ddsHTSeqMAold$condition, ref = "SCA")
# ddsHTSeqMAold.short$condition <- relevel(ddsHTSeqMAold.short$condition, ref = "SCA")
ddsHTSeqMAold_1$condition <- relevel(ddsHTSeqMAold_1$condition, ref = "SCA")
ddsHTSeqMAold_2$condition <- relevel(ddsHTSeqMAold_2$condition, ref = "SCA")
ddsHTSeqMAold_3$condition <- relevel(ddsHTSeqMAold_3$condition, ref = "SCA")
ddsHTSeqMAold_4$condition <- relevel(ddsHTSeqMAold_4$condition, ref = "SCA")
ddsHTSeqMAold_5$condition <- relevel(ddsHTSeqMAold_5$condition, ref = "SCA")
# ddsHTSeqMAold_6$condition <- relevel(ddsHTSeqMAold_6$condition, ref = "SCA")
ddsHTSeqMAold_7$condition <- relevel(ddsHTSeqMAold_7$condition, ref = "SCA")
ddsHTSeqMAold_8$condition <- relevel(ddsHTSeqMAold_8$condition, ref = "SCA")
ddsHTSeqMAold_9$condition <- relevel(ddsHTSeqMAold_9$condition, ref = "SCA")
ddsHTSeqMAold_11$condition <- relevel(ddsHTSeqMAold_11$condition, ref = "SCA")
ddsHTSeqMAold_15$condition <- relevel(ddsHTSeqMAold_15$condition, ref = "SCA")
ddsHTSeqMAold_28$condition <- relevel(ddsHTSeqMAold_28$condition, ref = "SCA")
ddsHTSeqMAold_88$condition <- relevel(ddsHTSeqMAold_88$condition, ref = "SCA")
ddsHTSeqMAold_108$condition <- relevel(ddsHTSeqMAold_108$condition, ref = "SCA")
ddsHTSeqMAold_119$condition <- relevel(ddsHTSeqMAold_119$condition, ref = "SCA")
###############################################################################
ddsHTSeqMAnew$condition <- relevel(ddsHTSeqMAnew$condition, ref = "Anc")
# ddsHTSeqMAnew.short$condition <- relevel(ddsHTSeqMAnew.short$condition, ref = "Anc")
ddsHTSeqMAnewS29$condition <- relevel(ddsHTSeqMAnewS29$condition, ref = "Anc")
ddsHTSeqMAnewS50$condition <- relevel(ddsHTSeqMAnewS50$condition, ref = "Anc")
ddsHTSeqMAnewS112$condition <- relevel(ddsHTSeqMAnewS112$condition, ref = "Anc")
ddsHTSeqMAnewS115$condition <- relevel(ddsHTSeqMAnewS115$condition, ref = "Anc")
ddsHTSeqMAnewS117$condition <- relevel(ddsHTSeqMAnewS117$condition, ref = "Anc")
ddsHTSeqMAnewS123$condition <- relevel(ddsHTSeqMAnewS123$condition, ref = "Anc")
# ddsHTSeqMAnewS141$condition <- relevel(ddsHTSeqMAnewS141$condition, ref = "Anc")
ddsHTSeqMAnewS152$condition <- relevel(ddsHTSeqMAnewS152$condition, ref = "Anc")

# ddsHTSeqMAnew$condition <- relevel(ddsHTSeqMAnew$condition, ref = "Anc")
# ddsHTSeqMAnew.short$condition <- relevel(ddsHTSeqMAnew.short$condition, ref = "Anc")
# ddsHTSeqMAnewS29.run$condition <- relevel(ddsHTSeqMAnewS29.run$condition, ref = "Anc")
# ddsHTSeqMAnewS50.run$condition <- relevel(ddsHTSeqMAnewS50.run$condition, ref = "Anc")
# ddsHTSeqMAnewS112.run$condition <- relevel(ddsHTSeqMAnewS112.run$condition, ref = "Anc")
# ddsHTSeqMAnewS115.run$condition <- relevel(ddsHTSeqMAnewS115.run$condition, ref = "Anc")
# ddsHTSeqMAnewS117.run$condition <- relevel(ddsHTSeqMAnewS117.run$condition, ref = "Anc")
# ddsHTSeqMAnewS123.run$condition <- relevel(ddsHTSeqMAnewS123.run$condition, ref = "Anc")
# ddsHTSeqMAnewS152.run$condition <- relevel(ddsHTSeqMAnewS152.run$condition, ref = "Anc")

```


### Pre-filtering
We filtered based on counts in the ancestor of GC: we kept only those genes with more than 10 counts in all 3 ancestor replicates of GC. We used the same metric for old and new MA datasets, and then found the common genes between all 3 for continuity, for a total of 5532 genes analyzed. 

```{r prefilter,eval=TRUE,include=TRUE,echo=FALSE}

## separate lines
ddsHTSeqGC_1
counts(ddsHTSeqGC_1)
dim(counts(ddsHTSeqGC_1))
ancGC <-  counts(ddsHTSeqGC_1)[,4:6]
#subset it based on cpm > 1 in all 3 replicates
table(rowSums(ancGC>10)==3)
keepGC <- rowSums(ancGC>10)==3

# need to make sure the samples don't have 0s either
ddsHTSeqGC_1 <- ddsHTSeqGC_1[keepGC,]

ddsHTSeqGC_11 <- ddsHTSeqGC_11[keepGC,]
ddsHTSeqGC_11
counts(ddsHTSeqGC_11)
dim(counts(ddsHTSeqGC_11))

ddsHTSeqGC_2 <- ddsHTSeqGC_2[keepGC,]

ddsHTSeqGC_3 <- ddsHTSeqGC_3[keepGC,]

ddsHTSeqGC_4 <- ddsHTSeqGC_4[keepGC,]

ddsHTSeqGC_5 <- ddsHTSeqGC_5[keepGC,]

# ddsHTSeqGC_6 <- ddsHTSeqGC_6[keepGC,]
# dim(ddsHTSeqGC_6)
# ddsHTSeqGC_6$condition
ddsHTSeqGC_7 <- ddsHTSeqGC_7[keepGC,]

ddsHTSeqGC_8 <- ddsHTSeqGC_8[keepGC,]

ddsHTSeqGC_9 <- ddsHTSeqGC_9[keepGC,]

ddsHTSeqGC_18 <- ddsHTSeqGC_18[keepGC,]

# ddsHTSeqGC_31 <- ddsHTSeqGC_31[keepGC,]
# dim(ddsHTSeqGC_31)
# ddsHTSeqGC_31$condition
ddsHTSeqGC_49 <- ddsHTSeqGC_49[keepGC,]

ddsHTSeqGC_59 <- ddsHTSeqGC_59[keepGC,]

ddsHTSeqGC_61 <- ddsHTSeqGC_61[keepGC,]

# ddsHTSeqGC_69 <- ddsHTSeqGC_69[keepGC,]
# dim(ddsHTSeqGC_69)
# ddsHTSeqGC_69$condition
ddsHTSeqGC_76 <- ddsHTSeqGC_76[keepGC,]

ddsHTSeqGC_77 <- ddsHTSeqGC_77[keepGC,]

###############################################################################
keepMAold <- rowSums(counts(ddsHTSeqMAold)) >= 20
ddsHTSeqMAold <- ddsHTSeqMAold[keepMAold,]
ddsHTSeqMAold
counts(ddsHTSeqMAold)
dim(counts(ddsHTSeqMAold))

#make a separate data frame with just the ancestor replicates
ancMAold <-  counts(ddsHTSeqMAold)[,43:45]
#subset it based on cpm > 1 in all 3 replicates
table(rowSums(ancMAold>10)==3)
keepMAold <- rowSums(ancMAold>10)==3
# try it in the full count set
# make a test file first
ddsHTSeqMAoldtest <- ddsHTSeqMAold
ddsHTSeqMAoldtest <- ddsHTSeqMAoldtest[keepMAold,]
ddsHTSeqMAoldtest
counts(ddsHTSeqMAoldtest)
dim(counts(ddsHTSeqMAoldtest))
View(counts(ddsHTSeqMAoldtest))
# # test file was fine so go ahead and make it the real thing
ddsHTSeqMAold <- ddsHTSeqMAoldtest
## get rid of any NAs
ddsHTSeqMAold <- na.omit(ddsHTSeqMAold)
dim(ddsHTSeqMAold)

ddsHTSeqMAold_1 <- ddsHTSeqMAold_1[keepGC,]

ddsHTSeqMAold_2 <- ddsHTSeqMAold_2[keepGC,]

ddsHTSeqMAold_2$condition

ddsHTSeqMAold_3 <- ddsHTSeqMAold_3[keepGC,]

ddsHTSeqMAold_4 <- ddsHTSeqMAold_4[keepGC,]

ddsHTSeqMAold_5 <- ddsHTSeqMAold_5[keepGC,]

ddsHTSeqMAold_7 <- ddsHTSeqMAold_7[keepGC,]

ddsHTSeqMAold_8 <- ddsHTSeqMAold_8[keepGC,]

ddsHTSeqMAold_9 <- ddsHTSeqMAold_9[keepGC,]

ddsHTSeqMAold_11 <- ddsHTSeqMAold_11[keepGC,]

ddsHTSeqMAold_15 <- ddsHTSeqMAold_15[keepGC,]

ddsHTSeqMAold_28 <- ddsHTSeqMAold_28[keepGC,]

ddsHTSeqMAold_88 <- ddsHTSeqMAold_88[keepGC,]

ddsHTSeqMAold_108 <- ddsHTSeqMAold_108[keepGC,]

ddsHTSeqMAold_119 <- ddsHTSeqMAold_119[keepGC,]

###############################################################################
keepMA <- rownames(ddsHTSeqMAnewS29)
keepGC1 <- rownames(ddsHTSeqGC_1)
keep1 <- keepMA %in% keepGC1
summary(keep1)
length(keep1)
keep <- keep1

ddsHTSeqMAnewS29 <- ddsHTSeqMAnewS29[keep,]
length(ddsHTSeqMAnewS29)

ddsHTSeqMAnewS50 <- ddsHTSeqMAnewS50[keep,]
length(ddsHTSeqMAnewS50)

ddsHTSeqMAnewS112 <- ddsHTSeqMAnewS112[keep,]

ddsHTSeqMAnewS115 <- ddsHTSeqMAnewS115[keep,]

ddsHTSeqMAnewS117 <- ddsHTSeqMAnewS117[keep,]

ddsHTSeqMAnewS123 <- ddsHTSeqMAnewS123[keep,]

# ddsHTSeqMAnewS141 <- ddsHTSeqMAnewS141[keepMAnew,]
# ddsHTSeqMAnewS141

ddsHTSeqMAnewS152 <- ddsHTSeqMAnewS152[keep,]

#now need to make sure all the GC and old MA samples match the new MA samples since got rid of a couple
keepMA.1 <- rownames(ddsHTSeqMAnewS152)
keepGC1.1 <- rownames(ddsHTSeqGC_77)
# keep <- keepGC1 %in% keepMA
keep2 <- keepGC1.1 %in% keepMA.1
summary(keep2)
length(keep2)

ddsHTSeqGC_1 <- ddsHTSeqGC_1[keep2,]
ddsHTSeqGC_2 <- ddsHTSeqGC_2[keep2,]
ddsHTSeqGC_3 <- ddsHTSeqGC_3[keep2,]
ddsHTSeqGC_4 <- ddsHTSeqGC_4[keep2,]
ddsHTSeqGC_5 <- ddsHTSeqGC_5[keep2,]
ddsHTSeqGC_7 <- ddsHTSeqGC_7[keep2,]
ddsHTSeqGC_8 <- ddsHTSeqGC_8[keep2,]
ddsHTSeqGC_9 <- ddsHTSeqGC_9[keep2,]
ddsHTSeqGC_11 <- ddsHTSeqGC_11[keep2,]
ddsHTSeqGC_18 <- ddsHTSeqGC_18[keep2,]
ddsHTSeqGC_49 <- ddsHTSeqGC_49[keep2,]
ddsHTSeqGC_59 <- ddsHTSeqGC_59[keep2,]
ddsHTSeqGC_61 <- ddsHTSeqGC_61[keep2,]
ddsHTSeqGC_76 <- ddsHTSeqGC_76[keep2,]
ddsHTSeqGC_77 <- ddsHTSeqGC_77[keep2,]

##################################################################################
### Left off here, trying to get MA old genes to match GC and MA new genes..
keepMA.1 <- rownames(ddsHTSeqMAnewS152)
keepGC1.2 <- rownames(ddsHTSeqMAold_2)
# keep <- keepGC1 %in% keepMA
keep3 <- keepGC1.2 %in% keepMA.1
summary(keep3)
length(keep3)

ddsHTSeqMAold_1 <- ddsHTSeqMAold_1[keep3,]
ddsHTSeqMAold_2 <- ddsHTSeqMAold_2[keep3,]
ddsHTSeqMAold_3 <- ddsHTSeqMAold_3[keep3,]
ddsHTSeqMAold_4 <- ddsHTSeqMAold_4[keep3,]
ddsHTSeqMAold_5 <- ddsHTSeqMAold_5[keep3,]
ddsHTSeqMAold_7 <- ddsHTSeqMAold_7[keep3,]
ddsHTSeqMAold_8 <- ddsHTSeqMAold_8[keep3,]
ddsHTSeqMAold_9 <- ddsHTSeqMAold_9[keep3,]
ddsHTSeqMAold_11 <- ddsHTSeqMAold_11[keep3,]
ddsHTSeqMAold_15 <- ddsHTSeqMAold_15[keep3,]
ddsHTSeqMAold_28 <- ddsHTSeqMAold_28[keep3,]
ddsHTSeqMAold_88 <- ddsHTSeqMAold_88[keep3,]
ddsHTSeqMAold_108 <- ddsHTSeqMAold_108[keep3,]
ddsHTSeqMAold_119 <- ddsHTSeqMAold_119[keep3,]
```

<a name="factorlevels"/>

## Differential expression analysis

The standard differential expression analysis steps are wrapped
into a single function, *DESeq*. The estimation steps performed
by this function are described [below](#theory), in the manual page for
`?DESeq` and in the Methods section of the DESeq2 publication [@Love2014].

Results tables are generated using the function *results*, which
extracts a results table with log2 fold changes, *p* values and adjusted
*p* values. With no additional arguments to *results*, the log2 fold change and
Wald test *p* value will be for the **last variable** in the design
formula, and if this is a factor, the comparison will be the **last
level** of this variable over the **reference level**
(see previous [note on factor levels](#factorlevels)).
However, the order of the variables of the design do not matter
so long as the user specifies the comparison to build a results table
for, using the `name` or `contrast` arguments of *results*.

Details about the comparison are printed to the console, directly above the
results table. The text, `condition treated vs untreated`, tells you that the
estimates are of the logarithmic fold change log2(treated/untreated).



```{r full dataset, eval=TRUE,include=TRUE}
#####################################################
#default
## add annotations
# geneIDgC <- rownames(ddsHTSeqGC.short)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesGC <- ensembldb::select(txdb, keys=geneIDgC, columnns="GENEID","TXCHROM" , keytype="GENEID")
#
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGC <- na.omit(genesGC)
# genesGC <- as.data.frame(genesGC)
# remove1 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGC.short) == genesGC$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS1 <- intersect(rownames(ddsHTSeqGC.short), genesGC$GENEID)
# keep3 <- rownames(ddsHTSeqGC.short) %in% matchedS1
# ddsHTSeqGC.short <- ddsHTSeqGC.short[keep3,]
# length(ddsHTSeqGC.short)
# all(rownames(ddsHTSeqGC.short) == genesGC$GENEID)
# mcols(ddsHTSeqGC.short) <- cbind(mcols(ddsHTSeqGC.short), genesGC)
#### filtered genes
#default
## add annotations
# geneIDgC.fil <- rownames(ddsHTSeqGC.short)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesGC.fil <- ensembldb::select(txdb, keys=geneIDgC.fil, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGC.fil <- na.omit(genesGC.fil)
# genesGC.fil <- as.data.frame(genesGC.fil)
# remove1 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGC.fil) == genesGC.fil$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS1.fil <- intersect(rownames(ddsHTSeqGC.fil), genesGC.fil$GENEID)
# keep3.fil <- rownames(ddsHTSeqGC.fil) %in% matchedS1.fil
# ddsHTSeqGC.fil <- ddsHTSeqGC.fil[keep3.fil,]
# length(ddsHTSeqGC.fil)
# all(rownames(ddsHTSeqGC.fil) == genesGC.fil$GENEID)
# mcols(ddsHTSeqGC.fil) <- cbind(mcols(ddsHTSeqGC.fil), genesGC.fil)
#
#
# geneIDgC.short <- rownames(ddsHTSeqGC.short)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesGC.short <- ensembldb::select(txdb, keys=geneIDgC.short, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGC.short <- na.omit(genesGC.short)
# genesGC.short <- as.data.frame(genesGC.short)
# remove1 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGC.short) == genesGC.short$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS1.short <- intersect(rownames(ddsHTSeqGC.short), genesGC.short$GENEID)
# keep3.short <- rownames(ddsHTSeqGC.short) %in% matchedS1.short
# ddsHTSeqGC.short <- ddsHTSeqGC.short[keep3.short,]
# length(ddsHTSeqGC.short)
# all(rownames(ddsHTSeqGC.short) == genesGC.short$GENEID)
# mcols(ddsHTSeqGC.short) <- cbind(mcols(ddsHTSeqGC.short), genesGC.short)
#

######### individual samples
## add annotations
# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
#
# BiocManager::install("TxDb.Scerevisiae.UCSC.sacCer2.sgdGene")
geneIDgC_1 <- rownames(ddsHTSeqGC_1)
library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
#
# BiocManager::install("ensembldb")
columns(txdb)
genesGC_1 <- ensembldb::select(txdb, keys=geneIDgC_1, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_1 <- na.omit(genesGC_1)
genesGC_1 <- as.data.frame(genesGC_1)
remove1 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_1) == genesGC_1$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS1 <- intersect(rownames(ddsHTSeqGC_1), genesGC_1$GENEID)
keep3S1 <- rownames(ddsHTSeqGC_1) %in% matchedS1
ddsHTSeqGC_1 <- ddsHTSeqGC_1[keep3S1,]
length(ddsHTSeqGC_1)
all(rownames(ddsHTSeqGC_1) == genesGC_1$GENEID)
mcols(ddsHTSeqGC_1) <- cbind(mcols(ddsHTSeqGC_1), genesGC_1)

geneIDgC_11 <- rownames(ddsHTSeqGC_11)
# library(TxDb.Scerevisiae.UC_C.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UC_C.sacCer2.sgdGene
genesGC_11 <- ensembldb::select(txdb, keys=geneIDgC_11, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_11 <- na.omit(genesGC_11)
genesGC_11 <- as.data.frame(genesGC_11)
remove1 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_11) == genesGC_11$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS11 <- intersect(rownames(ddsHTSeqGC_11), genesGC_11$GENEID)
keep3S11 <- rownames(ddsHTSeqGC_11) %in% matchedS11
ddsHTSeqGC_11 <- ddsHTSeqGC_11[keep3S11,]
length(ddsHTSeqGC_11)
all(rownames(ddsHTSeqGC_11) == genesGC_11$GENEID)
mcols(ddsHTSeqGC_11) <- cbind(mcols(ddsHTSeqGC_11), genesGC_11)
geneIDgC_2 <- rownames(ddsHTSeqGC_2)
genesGC_2 <- ensembldb::select(txdb, keys=geneIDgC_2, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_2 <- na.omit(genesGC_2)
genesGC_2 <- as.data.frame(genesGC_2)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_2) == genesGC_2$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqGC_2), genesGC_2$GENEID)
keep3S2 <- rownames(ddsHTSeqGC_2) %in% matchedS2
ddsHTSeqGC_2 <- ddsHTSeqGC_2[keep3S2,]
length(ddsHTSeqGC_2)
all(rownames(ddsHTSeqGC_2) == genesGC_2$GENEID)
mcols(ddsHTSeqGC_2) <- cbind(mcols(ddsHTSeqGC_2), genesGC_2)

all(rownames(ddsHTSeqGC_3) == genesGC_2$GENEID)

geneIDgC_1 <- rownames(ddsHTSeqGC_1)
genesGC_1 <- ensembldb::select(txdb, keys=geneIDgC_1, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_1 <- na.omit(genesGC_1)
genesGC_1 <- as.data.frame(genesGC_1)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_1) == genesGC_1$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS1 <- intersect(rownames(ddsHTSeqGC_1), genesGC_1$GENEID)
keep3S1 <- rownames(ddsHTSeqGC_1) %in% matchedS1
ddsHTSeqGC_1 <- ddsHTSeqGC_1[keep3S1,]
length(ddsHTSeqGC_1)
all(rownames(ddsHTSeqGC_1) == genesGC_1$GENEID)
mcols(ddsHTSeqGC_1) <- cbind(mcols(ddsHTSeqGC_1), genesGC_1)
mcols(ddsHTSeqGC_2) <- cbind(mcols(ddsHTSeqGC_2), genesGC_2)

geneIDgC_3 <- rownames(ddsHTSeqGC_3)
genesGC_3 <- ensembldb::select(txdb, keys=geneIDgC_3, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_3 <- na.omit(genesGC_3)
genesGC_3 <- as.data.frame(genesGC_3)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove3,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_3) == genesGC_3$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS3 <- intersect(rownames(ddsHTSeqGC_3), genesGC_3$GENEID)
keep3S3 <- rownames(ddsHTSeqGC_3) %in% matchedS3
ddsHTSeqGC_3 <- ddsHTSeqGC_3[keep3S3,]
length(ddsHTSeqGC_3)
all(rownames(ddsHTSeqGC_3) == genesGC_3$GENEID)
mcols(ddsHTSeqGC_3) <- cbind(mcols(ddsHTSeqGC_3), genesGC_3)

geneIDgC_4 <- rownames(ddsHTSeqGC_4)
genesGC_4 <- ensembldb::select(txdb, keys=geneIDgC_4, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_4 <- na.omit(genesGC_4)
genesGC_4 <- as.data.frame(genesGC_4)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove4,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_4) == genesGC_4$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS4 <- intersect(rownames(ddsHTSeqGC_4), genesGC_4$GENEID)
keep3S4 <- rownames(ddsHTSeqGC_4) %in% matchedS4
ddsHTSeqGC_4 <- ddsHTSeqGC_4[keep3S4,]
length(ddsHTSeqGC_4)
all(rownames(ddsHTSeqGC_4) == genesGC_4$GENEID)
mcols(ddsHTSeqGC_4) <- cbind(mcols(ddsHTSeqGC_4), genesGC_4)
# mcols(ddsHTSeqGC_2.fil) <- cbind(mcols(ddsHTSeqGC_2.fil), genesGC_2)

{geneIDgC_5 <- rownames(ddsHTSeqGC_5)
genesGC_5 <- ensembldb::select(txdb, keys=geneIDgC_5, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_5 <- na.omit(genesGC_5)
genesGC_5 <- as.data.frame(genesGC_5)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove4,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_5) == genesGC_5$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS5 <- intersect(rownames(ddsHTSeqGC_5), genesGC_5$GENEID)
keep3S5 <- rownames(ddsHTSeqGC_5) %in% matchedS4
ddsHTSeqGC_5 <- ddsHTSeqGC_5[keep3S5,]
length(ddsHTSeqGC_5)
all(rownames(ddsHTSeqGC_5) == genesGC_5$GENEID)
mcols(ddsHTSeqGC_5) <- cbind(mcols(ddsHTSeqGC_5), genesGC_5)}
# mcols(ddsHTSeqGC_2.fil) <- cbind(mcols(ddsHTSeqGC_2.fil), genesGC_2)

# geneIDgC_6 <- rownames(ddsHTSeqGC_6)
# genesGC_6 <- ensembldb::select(txdb, keys=geneIDgC_6, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGC_6 <- na.omit(genesGC_6)
# genesGC_6 <- as.data.frame(genesGC_6)
# remove2 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove4,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGC_6) == genesGC_6$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS6 <- intersect(rownames(ddsHTSeqGC_6), genesGC_6$GENEID)
# keep3S6 <- rownames(ddsHTSeqGC_6) %in% matchedS6
# ddsHTSeqGC_6 <- ddsHTSeqGC_6[keep3S6,]
# length(ddsHTSeqGC_6)
# all(rownames(ddsHTSeqGC_6) == genesGC_6$GENEID)
# mcols(ddsHTSeqGC_6) <- cbind(mcols(ddsHTSeqGC_6), genesGC_6)


geneIDgC_7 <- rownames(ddsHTSeqGC_7)
genesGC_7 <- ensembldb::select(txdb, keys=geneIDgC_7, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_7 <- na.omit(genesGC_7)
genesGC_7 <- as.data.frame(genesGC_7)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove7,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_7) == genesGC_7$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS7 <- intersect(rownames(ddsHTSeqGC_7), genesGC_7$GENEID)
keep3S7 <- rownames(ddsHTSeqGC_7) %in% matchedS7
ddsHTSeqGC_7 <- ddsHTSeqGC_7[keep3S7,]
length(ddsHTSeqGC_7)
all(rownames(ddsHTSeqGC_7) == genesGC_7$GENEID)
mcols(ddsHTSeqGC_7) <- cbind(mcols(ddsHTSeqGC_7), genesGC_7)

geneIDgC_8 <- rownames(ddsHTSeqGC_8)
genesGC_8 <- ensembldb::select(txdb, keys=geneIDgC_8, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_8 <- na.omit(genesGC_8)
genesGC_8 <- as.data.frame(genesGC_8)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove8,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_8) == genesGC_8$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS8 <- intersect(rownames(ddsHTSeqGC_8), genesGC_8$GENEID)
keep3S8 <- rownames(ddsHTSeqGC_8) %in% matchedS8
ddsHTSeqGC_8 <- ddsHTSeqGC_8[keep3S8,]
length(ddsHTSeqGC_8)
all(rownames(ddsHTSeqGC_8) == genesGC_8$GENEID)
mcols(ddsHTSeqGC_8) <- cbind(mcols(ddsHTSeqGC_8), genesGC_8)

geneIDgC_9 <- rownames(ddsHTSeqGC_9)
genesGC_9 <- ensembldb::select(txdb, keys=geneIDgC_9, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_9 <- na.omit(genesGC_9)
genesGC_9 <- as.data.frame(genesGC_9)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove9,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_9) == genesGC_9$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS9 <- intersect(rownames(ddsHTSeqGC_9), genesGC_9$GENEID)
keep3S9 <- rownames(ddsHTSeqGC_9) %in% matchedS9
ddsHTSeqGC_9 <- ddsHTSeqGC_9[keep3S9,]
length(ddsHTSeqGC_9)
all(rownames(ddsHTSeqGC_9) == genesGC_9$GENEID)
mcols(ddsHTSeqGC_9) <- cbind(mcols(ddsHTSeqGC_9), genesGC_9)

geneIDgC_11 <- rownames(ddsHTSeqGC_11)
genesGC_11 <- ensembldb::select(txdb, keys=geneIDgC_11, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_11 <- na.omit(genesGC_11)
genesGC_11 <- as.data.frame(genesGC_11)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove11,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_11) == genesGC_11$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS11 <- intersect(rownames(ddsHTSeqGC_11), genesGC_11$GENEID)
keep3S11 <- rownames(ddsHTSeqGC_11) %in% matchedS11
ddsHTSeqGC_11 <- ddsHTSeqGC_11[keep3S11,]
length(ddsHTSeqGC_11)
all(rownames(ddsHTSeqGC_11) == genesGC_11$GENEID)
mcols(ddsHTSeqGC_11) <- cbind(mcols(ddsHTSeqGC_11), genesGC_11)

geneIDgC_18 <- rownames(ddsHTSeqGC_18)
genesGC_18 <- ensembldb::select(txdb, keys=geneIDgC_18, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_18 <- na.omit(genesGC_18)
genesGC_18 <- as.data.frame(genesGC_18)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove18,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_18) == genesGC_18$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS18 <- intersect(rownames(ddsHTSeqGC_18), genesGC_18$GENEID)
keep3S18 <- rownames(ddsHTSeqGC_18) %in% matchedS18
ddsHTSeqGC_18 <- ddsHTSeqGC_18[keep3S18,]
length(ddsHTSeqGC_18)
all(rownames(ddsHTSeqGC_18) == genesGC_18$GENEID)
mcols(ddsHTSeqGC_18) <- cbind(mcols(ddsHTSeqGC_18), genesGC_18)

geneIDgC_31 <- rownames(ddsHTSeqGC_31)
genesGC_31 <- ensembldb::select(txdb, keys=geneIDgC_31, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_31 <- na.omit(genesGC_31)
genesGC_31 <- as.data.frame(genesGC_31)
remove2 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove18,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_31) == genesGC_31$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS31 <- intersect(rownames(ddsHTSeqGC_31), genesGC_31$GENEID)
keep3S31 <- rownames(ddsHTSeqGC_31) %in% matchedS31
ddsHTSeqGC_31 <- ddsHTSeqGC_31[keep3S31,]
length(ddsHTSeqGC_31)
all(rownames(ddsHTSeqGC_31) == genesGC_31$GENEID)
mcols(ddsHTSeqGC_31) <- cbind(mcols(ddsHTSeqGC_31), genesGC_31)


# mcols(ddsHTSeqGC_31) <- cbind(mcols(ddsHTSeqGC_31), genesGC_2)
geneIDgC_49 <- rownames(ddsHTSeqGC_49)
genesGC_49 <- ensembldb::select(txdb, keys=geneIDgC_49, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_49 <- na.omit(genesGC_49)
genesGC_49 <- as.data.frame(genesGC_49)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove49,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_49) == genesGC_49$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS49 <- intersect(rownames(ddsHTSeqGC_49), genesGC_49$GENEID)
keep3S49 <- rownames(ddsHTSeqGC_49) %in% matchedS49
ddsHTSeqGC_49 <- ddsHTSeqGC_49[keep3S49,]
length(ddsHTSeqGC_49)
all(rownames(ddsHTSeqGC_49) == genesGC_49$GENEID)
mcols(ddsHTSeqGC_49) <- cbind(mcols(ddsHTSeqGC_49), genesGC_49)

geneIDgC_59 <- rownames(ddsHTSeqGC_59)
genesGC_59 <- ensembldb::select(txdb, keys=geneIDgC_59, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_59 <- na.omit(genesGC_59)
genesGC_59 <- as.data.frame(genesGC_59)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove59,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_59) == genesGC_59$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS59 <- intersect(rownames(ddsHTSeqGC_59), genesGC_59$GENEID)
keep3S59 <- rownames(ddsHTSeqGC_59) %in% matchedS59
ddsHTSeqGC_59 <- ddsHTSeqGC_59[keep3S59,]
length(ddsHTSeqGC_59)
all(rownames(ddsHTSeqGC_59) == genesGC_59$GENEID)
mcols(ddsHTSeqGC_59) <- cbind(mcols(ddsHTSeqGC_59), genesGC_59)

geneIDgC_61 <- rownames(ddsHTSeqGC_61)
genesGC_61 <- ensembldb::select(txdb, keys=geneIDgC_61, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_61 <- na.omit(genesGC_61)
genesGC_61 <- as.data.frame(genesGC_61)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove61,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_61) == genesGC_61$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS61 <- intersect(rownames(ddsHTSeqGC_61), genesGC_61$GENEID)
keep3S61 <- rownames(ddsHTSeqGC_61) %in% matchedS61
ddsHTSeqGC_61 <- ddsHTSeqGC_61[keep3S61,]
length(ddsHTSeqGC_61)
all(rownames(ddsHTSeqGC_61) == genesGC_61$GENEID)
mcols(ddsHTSeqGC_61) <- cbind(mcols(ddsHTSeqGC_61), genesGC_61)
# mcols(ddsHTSeqGC_69) <- cbind(mcols(ddsHTSeqGC_69), genesGC_2)

# geneIDgC_69 <- rownames(ddsHTSeqGC_69)
# genesGC_69 <- ensembldb::select(txdb, keys=geneIDgC_69, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGC_69 <- na.omit(genesGC_69)
# genesGC_69 <- as.data.frame(genesGC_69)
# remove2 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove61,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGC_69) == genesGC_69$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS69 <- intersect(rownames(ddsHTSeqGC_69), genesGC_69$GENEID)
# keep3S69 <- rownames(ddsHTSeqGC_69) %in% matchedS69
# ddsHTSeqGC_69 <- ddsHTSeqGC_69[keep3S69,]
# length(ddsHTSeqGC_69)
# all(rownames(ddsHTSeqGC_69) == genesGC_69$GENEID)
# mcols(ddsHTSeqGC_69) <- cbind(mcols(ddsHTSeqGC_69), genesGC_69)
# # mcols(ddsHTSeqGC_69) <- cbind(mcols(ddsHTSeqGC_69), genesGC_2)


geneIDgC_76 <- rownames(ddsHTSeqGC_76)
genesGC_76 <- ensembldb::select(txdb, keys=geneIDgC_76, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_76 <- na.omit(genesGC_76)
genesGC_76 <- as.data.frame(genesGC_76)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove76,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_76) == genesGC_76$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS76 <- intersect(rownames(ddsHTSeqGC_76), genesGC_76$GENEID)
keep3S76 <- rownames(ddsHTSeqGC_76) %in% matchedS76
ddsHTSeqGC_76 <- ddsHTSeqGC_76[keep3S76,]
length(ddsHTSeqGC_76)
all(rownames(ddsHTSeqGC_76) == genesGC_76$GENEID)
mcols(ddsHTSeqGC_76) <- cbind(mcols(ddsHTSeqGC_76), genesGC_76)

geneIDgC_77 <- rownames(ddsHTSeqGC_77)
genesGC_77 <- ensembldb::select(txdb, keys=geneIDgC_77, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesGC_77 <- na.omit(genesGC_77)
genesGC_77 <- as.data.frame(genesGC_77)
remove2 <- "chrM"
# genesGC <- genesGC[-(grep(paste0(remove77,"$"),genesGC$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqGC_77) == genesGC_77$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS77 <- intersect(rownames(ddsHTSeqGC_77), genesGC_77$GENEID)
keep3S77 <- rownames(ddsHTSeqGC_77) %in% matchedS77
ddsHTSeqGC_77 <- ddsHTSeqGC_77[keep3S77,]
length(ddsHTSeqGC_77)
all(rownames(ddsHTSeqGC_77) == genesGC_77$GENEID)
mcols(ddsHTSeqGC_77) <- cbind(mcols(ddsHTSeqGC_77), genesGC_77)



### with run as factor
# geneIDgCS1.run <- rownames(ddsHTSeqGCS1.run)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesGCS1.run <- ensembldb::select(txdb, keys=geneIDgCS1.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGCS1.run <- na.omit(genesGCS1.run)
# genesGCS1.run <- as.data.frame(genesGCS1.run)
# remove1 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGCS1.run) == genesGCS1.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS1.run <- intersect(rownames(ddsHTSeqGCS1.run), genesGCS1.run$GENEID)
# keep3S1.run <- rownames(ddsHTSeqGCS1.run) %in% matchedS1.run
# ddsHTSeqGCS1.run <- ddsHTSeqGCS1.run[keep3S1.run,]
# length(ddsHTSeqGCS1.run)
# all(rownames(ddsHTSeqGCS1.run) == genesGCS1.run$GENEID)
# mcols(ddsHTSeqGCS1.run) <- cbind(mcols(ddsHTSeqGCS1.run), genesGCS1.run)
#
# geneIDgCS11.run <- rownames(ddsHTSeqGCS11.run)
# # library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# # txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesGCS11.run <- ensembldb::select(txdb, keys=geneIDgCS11.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGCS11.run <- na.omit(genesGCS11.run)
# genesGCS11.run <- as.data.frame(genesGCS11.run)
# remove1 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGCS11.run) == genesGCS11.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS11.run <- intersect(rownames(ddsHTSeqGCS11.run), genesGCS11.run$GENEID)
# keep3S11.run <- rownames(ddsHTSeqGCS11.run) %in% matchedS11.run
# ddsHTSeqGCS11.run <- ddsHTSeqGCS11.run[keep3S11.run,]
# length(ddsHTSeqGCS11.run)
# all(rownames(ddsHTSeqGCS11.run) == genesGCS11.run$GENEID)
# mcols(ddsHTSeqGCS11.run) <- cbind(mcols(ddsHTSeqGCS11.run), genesGCS11.run)
# geneIDgCS2.run <- rownames(ddsHTSeqGCS2.run)
# genesGCS2.run <- ensembldb::select(txdb, keys=geneIDgCS2.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGCS2.run <- na.omit(genesGCS2.run)
# genesGCS2.run <- as.data.frame(genesGCS2.run)
# remove2 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGCS2.run) == genesGCS2.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqGCS2.run), genesGCS2.run$GENEID)
# keep3S2.run <- rownames(ddsHTSeqGCS2.run) %in% matchedS2.run
# ddsHTSeqGCS2.run <- ddsHTSeqGCS2.run[keep3S2.run,]
# length(ddsHTSeqGCS2.run)
# all(rownames(ddsHTSeqGCS2.run) == genesGCS2.run$GENEID)
# mcols(ddsHTSeqGCS2.run) <- cbind(mcols(ddsHTSeqGCS2.run), genesGCS2.run)
#
# all(rownames(ddsHTSeqGCS3.run) == genesGCS2.run$GENEID)
# # mcols(ddsHTSeqGCS1.run) <- cbind(mcols(ddsHTSeqGCS1.run), genesGCS2.run)
# mcols(ddsHTSeqGCS3.run) <- cbind(mcols(ddsHTSeqGCS3.run), genesGCS2.run)
# mcols(ddsHTSeqGCS4.run) <- cbind(mcols(ddsHTSeqGCS4.run), genesGCS2.run)
# # mcols(ddsHTSeqGCS2.fil) <- cbind(mcols(ddsHTSeqGCS2.fil), genesGCS2)
# mcols(ddsHTSeqGCS7.run) <- cbind(mcols(ddsHTSeqGCS7.run), genesGCS2.run)
# mcols(ddsHTSeqGCS8.run) <- cbind(mcols(ddsHTSeqGCS8.run), genesGCS2.run)
# mcols(ddsHTSeqGCS9.run) <- cbind(mcols(ddsHTSeqGCS9.run), genesGCS2.run)
# # mcols(ddsHTSeqGCS11.run) <- cbind(mcols(ddsHTSeqGCS11.run), genesGCS2.run)
# mcols(ddsHTSeqGCS18.run) <- cbind(mcols(ddsHTSeqGCS18.run), genesGCS2.run)
# mcols(ddsHTSeqGCS31.run) <- cbind(mcols(ddsHTSeqGCS31.run), genesGCS2.run)
# mcols(ddsHTSeqGCS49.run) <- cbind(mcols(ddsHTSeqGCS49.run), genesGCS2.run)
# mcols(ddsHTSeqGCS59.run) <- cbind(mcols(ddsHTSeqGCS59.run), genesGCS2.run)
# mcols(ddsHTSeqGCS61.run) <- cbind(mcols(ddsHTSeqGCS61.run), genesGCS2.run)
# mcols(ddsHTSeqGCS69.run) <- cbind(mcols(ddsHTSeqGCS69.run), genesGCS2.run)
# mcols(ddsHTSeqGCS76.run) <- cbind(mcols(ddsHTSeqGCS76.run), genesGCS2.run)
# mcols(ddsHTSeqGCS77.run) <- cbind(mcols(ddsHTSeqGCS77.run), genesGCS2.run)

#### filtered genes
#default
# ## add annotations
# geneIDgC.fil <- rownames(ddsHTSeqGC.fil)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesGC.fil <- ensembldb::select(txdb, keys=geneIDgC.fil, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesGC.fil <- na.omit(genesGC.fil)
# genesGC.fil <- as.data.frame(genesGC.fil)
# remove1 <- "chrM"
# # genesGC <- genesGC[-(grep(paste0(remove1,"$"),genesGC$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqGC.fil) == genesGC.fil$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS1.fil <- intersect(rownames(ddsHTSeqGC.fil), genesGC.fil$GENEID)
# keep3.fil <- rownames(ddsHTSeqGC.fil) %in% matchedS1.fil
# ddsHTSeqGC.fil <- ddsHTSeqGC.fil[keep3.fil,]
# length(ddsHTSeqGC.fil)
# all(rownames(ddsHTSeqGC.fil) == genesGC.fil$GENEID)
# mcols(ddsHTSeqGC.fil) <- cbind(mcols(ddsHTSeqGC.fil), genesGC.fil)
#####################################################
#default
## add annotations
# geneIDMAold <- rownames(ddsHTSeqMAold)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAold <- ensembldb::select(txdb, keys=geneIDMAold, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAold <- na.omit(genesMAold)
# genesMAold <- as.data.frame(genesMAold)
# remove1 <- "chrM"
# # genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAold) == genesMAold$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2 <- intersect(rownames(ddsHTSeqMAold), genesMAold$GENEID)
# keep4 <- rownames(ddsHTSeqMAold) %in% matchedS2
# ddsHTSeqMAold <- ddsHTSeqMAold[keep4,]
# # length(ddsHTSeqMAold)
# all(rownames(ddsHTSeqMAold) == genesMAold$GENEID)
# mcols(ddsHTSeqMAold) <- cbind(mcols(ddsHTSeqMAold), genesMAold)

###### individual samples
geneIDMAold_1 <- rownames(ddsHTSeqMAold_1)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_1 <- ensembldb::select(txdb, keys=geneIDMAold_1, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_1 <- na.omit(genesMAold_1)
genesMAold_1 <- as.data.frame(genesMAold_1)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_1) == genesMAold_1$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_1), genesMAold_1$GENEID)
keep4 <- rownames(ddsHTSeqMAold_1) %in% matched_2
ddsHTSeqMAold_1 <- ddsHTSeqMAold_1[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_1) == genesMAold_1$GENEID)
mcols(ddsHTSeqMAold_1) <- cbind(mcols(ddsHTSeqMAold_1), genesMAold_1)

geneIDMAold_2 <- rownames(ddsHTSeqMAold_2)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_2 <- ensembldb::select(txdb, keys=geneIDMAold_2, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_2 <- na.omit(genesMAold_2)
genesMAold_2 <- as.data.frame(genesMAold_2)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_2) == genesMAold_2$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_2), genesMAold_2$GENEID)
keep4 <- rownames(ddsHTSeqMAold_2) %in% matched_2
ddsHTSeqMAold_2 <- ddsHTSeqMAold_2[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_2) == genesMAold_2$GENEID)
mcols(ddsHTSeqMAold_2) <- cbind(mcols(ddsHTSeqMAold_2), genesMAold_2)

geneIDMAold_3 <- rownames(ddsHTSeqMAold_3)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_3 <- ensembldb::select(txdb, keys=geneIDMAold_3, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_3 <- na.omit(genesMAold_3)
genesMAold_3 <- as.data.frame(genesMAold_3)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_3) == genesMAold_3$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_3), genesMAold_3$GENEID)
keep4 <- rownames(ddsHTSeqMAold_3) %in% matched_2
ddsHTSeqMAold_3 <- ddsHTSeqMAold_3[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_3) == genesMAold_3$GENEID)
mcols(ddsHTSeqMAold_3) <- cbind(mcols(ddsHTSeqMAold_3), genesMAold_3)

geneIDMAold_4 <- rownames(ddsHTSeqMAold_4)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_4 <- ensembldb::select(txdb, keys=geneIDMAold_4, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_4 <- na.omit(genesMAold_4)
genesMAold_4 <- as.data.frame(genesMAold_4)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_4) == genesMAold_4$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_4), genesMAold_4$GENEID)
keep4 <- rownames(ddsHTSeqMAold_4) %in% matched_2
ddsHTSeqMAold_4 <- ddsHTSeqMAold_4[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_4) == genesMAold_4$GENEID)
mcols(ddsHTSeqMAold_4) <- cbind(mcols(ddsHTSeqMAold_4), genesMAold_4)

geneIDMAold_5 <- rownames(ddsHTSeqMAold_5)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_5 <- ensembldb::select(txdb, keys=geneIDMAold_5, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_5 <- na.omit(genesMAold_5)

genesMAold_5 <- as.data.frame(genesMAold_5)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_5) == genesMAold_5$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_5), genesMAold_5$GENEID)
keep4 <- rownames(ddsHTSeqMAold_5) %in% matched_2
ddsHTSeqMAold_5 <- ddsHTSeqMAold_5[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_5) == genesMAold_5$GENEID)
mcols(ddsHTSeqMAold_5) <- cbind(mcols(ddsHTSeqMAold_5), genesMAold_5)

# geneIDMAold_6 <- rownames(ddsHTSeqMAold_6)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAold_6 <- ensembldb::select(txdb, keys=geneIDMAold_6, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAold_6 <- na.omit(genesMAold_6)
# genesMAold_6 <- as.data.frame(genesMAold_6)
# remove1 <- "chrM"
# # genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAold_6) == genesMAold_6$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matched_2 <- intersect(rownames(ddsHTSeqMAold_6), genesMAold_6$GENEID)
# keep4 <- rownames(ddsHTSeqMAold_6) %in% matched_2
# ddsHTSeqMAold_6 <- ddsHTSeqMAold_6[keep4,]
# # length(ddsHTSeqMAold)
# all(rownames(ddsHTSeqMAold_6) == genesMAold_6$GENEID)
# mcols(ddsHTSeqMAold_6) <- cbind(mcols(ddsHTSeqMAold_6), genesMAold_6)

geneIDMAold_7 <- rownames(ddsHTSeqMAold_7)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_7 <- ensembldb::select(txdb, keys=geneIDMAold_7, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_7 <- na.omit(genesMAold_7)
genesMAold_7 <- as.data.frame(genesMAold_7)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_7) == genesMAold_7$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_7), genesMAold_7$GENEID)
keep4 <- rownames(ddsHTSeqMAold_7) %in% matched_2
ddsHTSeqMAold_7 <- ddsHTSeqMAold_7[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_7) == genesMAold_7$GENEID)
mcols(ddsHTSeqMAold_7) <- cbind(mcols(ddsHTSeqMAold_7), genesMAold_7)

geneIDMAold_8 <- rownames(ddsHTSeqMAold_8)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_8 <- ensembldb::select(txdb, keys=geneIDMAold_8, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_8 <- na.omit(genesMAold_8)
genesMAold_8 <- as.data.frame(genesMAold_8)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_8) == genesMAold_8$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_8), genesMAold_8$GENEID)
keep4 <- rownames(ddsHTSeqMAold_8) %in% matched_2
ddsHTSeqMAold_8 <- ddsHTSeqMAold_8[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_8) == genesMAold_8$GENEID)
mcols(ddsHTSeqMAold_8) <- cbind(mcols(ddsHTSeqMAold_8), genesMAold_8)

geneIDMAold_9 <- rownames(ddsHTSeqMAold_9)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_9 <- ensembldb::select(txdb, keys=geneIDMAold_9, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_9 <- na.omit(genesMAold_9)
genesMAold_9 <- as.data.frame(genesMAold_9)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_9) == genesMAold_9$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_9), genesMAold_9$GENEID)
keep4 <- rownames(ddsHTSeqMAold_9) %in% matched_2
ddsHTSeqMAold_9 <- ddsHTSeqMAold_9[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_9) == genesMAold_9$GENEID)
mcols(ddsHTSeqMAold_9) <- cbind(mcols(ddsHTSeqMAold_9), genesMAold_9)

geneIDMAold_11 <- rownames(ddsHTSeqMAold_11)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_11 <- ensembldb::select(txdb, keys=geneIDMAold_11, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_11 <- na.omit(genesMAold_11)
genesMAold_11 <- as.data.frame(genesMAold_11)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_11) == genesMAold_11$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_11), genesMAold_11$GENEID)
keep4 <- rownames(ddsHTSeqMAold_11) %in% matched_2
ddsHTSeqMAold_11 <- ddsHTSeqMAold_11[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_11) == genesMAold_11$GENEID)
mcols(ddsHTSeqMAold_11) <- cbind(mcols(ddsHTSeqMAold_11), genesMAold_11)

geneIDMAold_15 <- rownames(ddsHTSeqMAold_15)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_15 <- ensembldb::select(txdb, keys=geneIDMAold_15, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_15 <- na.omit(genesMAold_15)
genesMAold_15 <- as.data.frame(genesMAold_15)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_15) == genesMAold_15$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_15), genesMAold_15$GENEID)
keep4 <- rownames(ddsHTSeqMAold_15) %in% matched_2
ddsHTSeqMAold_15 <- ddsHTSeqMAold_15[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_15) == genesMAold_15$GENEID)
mcols(ddsHTSeqMAold_15) <- cbind(mcols(ddsHTSeqMAold_15), genesMAold_15)

geneIDMAold_28 <- rownames(ddsHTSeqMAold_28)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_28 <- ensembldb::select(txdb, keys=geneIDMAold_28, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_28 <- na.omit(genesMAold_28)
genesMAold_28 <- as.data.frame(genesMAold_28)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_28) == genesMAold_28$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_28), genesMAold_28$GENEID)
keep4 <- rownames(ddsHTSeqMAold_28) %in% matched_2
ddsHTSeqMAold_28 <- ddsHTSeqMAold_28[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_28) == genesMAold_28$GENEID)
mcols(ddsHTSeqMAold_28) <- cbind(mcols(ddsHTSeqMAold_28), genesMAold_28)

geneIDMAold_88 <- rownames(ddsHTSeqMAold_88)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_88 <- ensembldb::select(txdb, keys=geneIDMAold_88, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_88 <- na.omit(genesMAold_88)
genesMAold_88 <- as.data.frame(genesMAold_88)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_88) == genesMAold_88$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_88), genesMAold_88$GENEID)
keep4 <- rownames(ddsHTSeqMAold_88) %in% matched_2
ddsHTSeqMAold_88 <- ddsHTSeqMAold_88[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_88) == genesMAold_88$GENEID)
mcols(ddsHTSeqMAold_88) <- cbind(mcols(ddsHTSeqMAold_88), genesMAold_88)

geneIDMAold_108 <- rownames(ddsHTSeqMAold_108)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_108 <- ensembldb::select(txdb, keys=geneIDMAold_108, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_108 <- na.omit(genesMAold_108)
genesMAold_108 <- as.data.frame(genesMAold_108)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_108) == genesMAold_108$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_108), genesMAold_108$GENEID)
keep4 <- rownames(ddsHTSeqMAold_108) %in% matched_2
ddsHTSeqMAold_108 <- ddsHTSeqMAold_108[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_108) == genesMAold_108$GENEID)
mcols(ddsHTSeqMAold_108) <- cbind(mcols(ddsHTSeqMAold_108), genesMAold_108)

geneIDMAold_119 <- rownames(ddsHTSeqMAold_119)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAold_119 <- ensembldb::select(txdb, keys=geneIDMAold_119, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAold_119 <- na.omit(genesMAold_119)
genesMAold_119 <- as.data.frame(genesMAold_119)
remove1 <- "chrM"
# genesMAold <- genesMAold[-(grep(paste0(remove1,"$"),genesMAold$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAold_119) == genesMAold_119$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matched_2 <- intersect(rownames(ddsHTSeqMAold_119), genesMAold_119$GENEID)
keep4 <- rownames(ddsHTSeqMAold_119) %in% matched_2
ddsHTSeqMAold_119 <- ddsHTSeqMAold_119[keep4,]
# length(ddsHTSeqMAold)
all(rownames(ddsHTSeqMAold_119) == genesMAold_119$GENEID)
mcols(ddsHTSeqMAold_119) <- cbind(mcols(ddsHTSeqMAold_119), genesMAold_119)





#####################################################
#default
## add annotations
# geneIDMAnew <- rownames(ddsHTSeqMAnew)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnew <- ensembldb::select(txdb, keys=geneIDMAnew, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnew <- na.omit(genesMAnew)
# genesMAnew <- as.data.frame(genesMAnew)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnew) == genesMAnew$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedMAnew <- intersect(rownames(ddsHTSeqMAnew), genesMAnew$GENEID)
# keepMAnew<- rownames(ddsHTSeqMAnew) %in% matchedMAnew
# ddsHTSeqMAnew <- ddsHTSeqMAnew[keepMAnew,]
# length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnew) == genesMAnew$GENEID)
# mcols(ddsHTSeqMAnew) <- cbind(mcols(ddsHTSeqMAnew), genesMAnew)


#### individual samples
geneIDMAnewS29 <- rownames(ddsHTSeqMAnewS29)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAnewS29 <- ensembldb::select(txdb, keys=geneIDMAnewS29, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS29 <- na.omit(genesMAnewS29)
genesMAnewS29 <- as.data.frame(genesMAnewS29)
remove1 <- "chrM"
# genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAnewS29) == genesMAnewS29$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS29), genesMAnewS29$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS29) %in% matchedS2
ddsHTSeqMAnewS29 <- ddsHTSeqMAnewS29[keep4,]
# length(ddsHTSeqMAnew)
all(rownames(ddsHTSeqMAnewS29) == genesMAnewS29$GENEID)
mcols(ddsHTSeqMAnewS29) <- cbind(mcols(ddsHTSeqMAnewS29), genesMAnewS29)

geneIDMAnewS50 <- rownames(ddsHTSeqMAnewS50)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAnewS50 <- ensembldb::select(txdb, keys=geneIDMAnewS50, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS50 <- na.omit(genesMAnewS50)
genesMAnewS50 <- as.data.frame(genesMAnewS50)
remove1 <- "chrM"
# genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAnewS50) == genesMAnewS50$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS50), genesMAnewS50$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS50) %in% matchedS2
ddsHTSeqMAnewS50 <- ddsHTSeqMAnewS50[keep4,]
# length(ddsHTSeqMAnew)
all(rownames(ddsHTSeqMAnewS50) == genesMAnewS50$GENEID)
mcols(ddsHTSeqMAnewS50) <- cbind(mcols(ddsHTSeqMAnewS50), genesMAnewS50)

geneIDMAnewS112 <- rownames(ddsHTSeqMAnewS112)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAnewS112 <- ensembldb::select(txdb, keys=geneIDMAnewS112, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS112 <- na.omit(genesMAnewS112)
genesMAnewS112 <- as.data.frame(genesMAnewS112)
remove1 <- "chrM"
# genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAnewS112) == genesMAnewS112$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS112), genesMAnewS112$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS112) %in% matchedS2
ddsHTSeqMAnewS112 <- ddsHTSeqMAnewS112[keep4,]
# length(ddsHTSeqMAnew)
all(rownames(ddsHTSeqMAnewS112) == genesMAnewS112$GENEID)
mcols(ddsHTSeqMAnewS112) <- cbind(mcols(ddsHTSeqMAnewS112), genesMAnewS112)
#
geneIDMAnewS115 <- rownames(ddsHTSeqMAnewS115)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAnewS115 <- ensembldb::select(txdb, keys=geneIDMAnewS115, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS115 <- na.omit(genesMAnewS115)
genesMAnewS115 <- as.data.frame(genesMAnewS115)
remove1 <- "chrM"
# genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAnewS115) == genesMAnewS115$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS115), genesMAnewS115$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS115) %in% matchedS2
ddsHTSeqMAnewS115 <- ddsHTSeqMAnewS115[keep4,]
# length(ddsHTSeqMAnew)
all(rownames(ddsHTSeqMAnewS115) == genesMAnewS115$GENEID)
mcols(ddsHTSeqMAnewS115) <- cbind(mcols(ddsHTSeqMAnewS115), genesMAnewS115)

geneIDMAnewS117 <- rownames(ddsHTSeqMAnewS117)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAnewS117 <- ensembldb::select(txdb, keys=geneIDMAnewS117, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS117 <- na.omit(genesMAnewS117)
genesMAnewS117 <- as.data.frame(genesMAnewS117)
remove1 <- "chrM"
# genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAnewS117) == genesMAnewS117$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS117), genesMAnewS117$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS117) %in% matchedS2
ddsHTSeqMAnewS117 <- ddsHTSeqMAnewS117[keep4,]
# length(ddsHTSeqMAnew)
all(rownames(ddsHTSeqMAnewS117) == genesMAnewS117$GENEID)
mcols(ddsHTSeqMAnewS117) <- cbind(mcols(ddsHTSeqMAnewS117), genesMAnewS117)

geneIDMAnewS123 <- rownames(ddsHTSeqMAnewS123)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAnewS123 <- ensembldb::select(txdb, keys=geneIDMAnewS123, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS123 <- na.omit(genesMAnewS123)
genesMAnewS123 <- as.data.frame(genesMAnewS123)
remove1 <- "chrM"
# genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAnewS123) == genesMAnewS123$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS123), genesMAnewS123$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS123) %in% matchedS2
ddsHTSeqMAnewS123 <- ddsHTSeqMAnewS123[keep4,]
# length(ddsHTSeqMAnew)
all(rownames(ddsHTSeqMAnewS123) == genesMAnewS123$GENEID)
mcols(ddsHTSeqMAnewS123) <- cbind(mcols(ddsHTSeqMAnewS123), genesMAnewS123)

# geneIDMAnewS141 <- rownames(ddsHTSeqMAnewS141)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS141 <- ensembldb::select(txdb, keys=geneIDMAnewS141, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS141 <- na.omit(genesMAnewS141)
# genesMAnewS141 <- as.data.frame(genesMAnewS141)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS141) == genesMAnewS141$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2 <- intersect(rownames(ddsHTSeqMAnewS141), genesMAnewS141$GENEID)
# keep4 <- rownames(ddsHTSeqMAnewS141) %in% matchedS2
# ddsHTSeqMAnewS141 <- ddsHTSeqMAnewS141[keep4,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS141) == genesMAnewS141$GENEID)
# mcols(ddsHTSeqMAnewS141) <- cbind(mcols(ddsHTSeqMAnewS141), genesMAnewS141)

geneIDMAnewS152 <- rownames(ddsHTSeqMAnewS152)
#library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
#txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
genesMAnewS152 <- ensembldb::select(txdb, keys=geneIDMAnewS152, columnns="GENEID","TXCHROM" , keytype="GENEID")
#remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
genesMAnewS152 <- na.omit(genesMAnewS152)
genesMAnewS152 <- as.data.frame(genesMAnewS152)
remove1 <- "chrM"
# genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
all(rownames(ddsHTSeqMAnewS152) == genesMAnewS152$GENEID)
#need to remove the ones in rownames of dds that don't match genes$GENEID
matchedS2 <- intersect(rownames(ddsHTSeqMAnewS152), genesMAnewS152$GENEID)
keep4 <- rownames(ddsHTSeqMAnewS152) %in% matchedS2
ddsHTSeqMAnewS152 <- ddsHTSeqMAnewS152[keep4,]
# length(ddsHTSeqMAnew)
all(rownames(ddsHTSeqMAnewS152) == genesMAnewS152$GENEID)
mcols(ddsHTSeqMAnewS152) <- cbind(mcols(ddsHTSeqMAnewS152), genesMAnewS152)

### with run included as factor in design
# geneIDMAnewS29.run <- rownames(ddsHTSeqMAnewS29.run)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS29.run <- ensembldb::select(txdb, keys=geneIDMAnewS29.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS29.run <- na.omit(genesMAnewS29.run)
# genesMAnewS29.run <- as.data.frame(genesMAnewS29.run)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS29.run) == genesMAnewS29.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqMAnewS29.run), genesMAnewS29.run$GENEID)
# keep4.run <- rownames(ddsHTSeqMAnewS29.run) %in% matchedS2.run
# ddsHTSeqMAnewS29.run <- ddsHTSeqMAnewS29.run[keep4.run,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS29.run) == genesMAnewS29.run$GENEID)
# mcols(ddsHTSeqMAnewS29.run) <- cbind(mcols(ddsHTSeqMAnewS29.run), genesMAnewS29.run)
#
# geneIDMAnewS50.run <- rownames(ddsHTSeqMAnewS50.run)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS50.run <- ensembldb::select(txdb, keys=geneIDMAnewS50.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS50.run <- na.omit(genesMAnewS50.run)
# genesMAnewS50.run <- as.data.frame(genesMAnewS50.run)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS50.run) == genesMAnewS50.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqMAnewS50.run), genesMAnewS50.run$GENEID)
# keep4.run <- rownames(ddsHTSeqMAnewS50.run) %in% matchedS2.run
# ddsHTSeqMAnewS50.run <- ddsHTSeqMAnewS50.run[keep4.run,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS50.run) == genesMAnewS50.run$GENEID)
# mcols(ddsHTSeqMAnewS50.run) <- cbind(mcols(ddsHTSeqMAnewS50.run), genesMAnewS50.run)
#
# geneIDMAnewS112.run <- rownames(ddsHTSeqMAnewS112.run)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS112.run <- ensembldb::select(txdb, keys=geneIDMAnewS112.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS112.run <- na.omit(genesMAnewS112.run)
# genesMAnewS112.run <- as.data.frame(genesMAnewS112.run)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS112.run) == genesMAnewS112.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqMAnewS112.run), genesMAnewS112.run$GENEID)
# keep4.run <- rownames(ddsHTSeqMAnewS112.run) %in% matchedS2.run
# ddsHTSeqMAnewS112.run <- ddsHTSeqMAnewS112.run[keep4.run,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS112.run) == genesMAnewS112.run$GENEID)
# mcols(ddsHTSeqMAnewS112.run) <- cbind(mcols(ddsHTSeqMAnewS112.run), genesMAnewS112.run)
# #
# geneIDMAnewS115.run <- rownames(ddsHTSeqMAnewS115.run)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS115.run <- ensembldb::select(txdb, keys=geneIDMAnewS115.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS115.run <- na.omit(genesMAnewS115.run)
# genesMAnewS115.run <- as.data.frame(genesMAnewS115.run)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS115.run) == genesMAnewS115.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqMAnewS115.run), genesMAnewS115.run$GENEID)
# keep4.run <- rownames(ddsHTSeqMAnewS115.run) %in% matchedS2.run
# ddsHTSeqMAnewS115.run <- ddsHTSeqMAnewS115.run[keep4.run,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS115.run) == genesMAnewS115.run$GENEID)
# mcols(ddsHTSeqMAnewS115.run) <- cbind(mcols(ddsHTSeqMAnewS115.run), genesMAnewS115.run)
#
# geneIDMAnewS117.run <- rownames(ddsHTSeqMAnewS117.run)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS117.run <- ensembldb::select(txdb, keys=geneIDMAnewS117.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS117.run <- na.omit(genesMAnewS117.run)
# genesMAnewS117.run <- as.data.frame(genesMAnewS117.run)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS117.run) == genesMAnewS117.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqMAnewS117.run), genesMAnewS117.run$GENEID)
# keep4.run <- rownames(ddsHTSeqMAnewS117.run) %in% matchedS2.run
# ddsHTSeqMAnewS117.run <- ddsHTSeqMAnewS117.run[keep4.run,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS117.run) == genesMAnewS117.run$GENEID)
# mcols(ddsHTSeqMAnewS117.run) <- cbind(mcols(ddsHTSeqMAnewS117.run), genesMAnewS117.run)
#
# geneIDMAnewS123.run <- rownames(ddsHTSeqMAnewS123.run)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS123.run <- ensembldb::select(txdb, keys=geneIDMAnewS123.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS123.run <- na.omit(genesMAnewS123.run)
# genesMAnewS123.run <- as.data.frame(genesMAnewS123.run)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS123.run) == genesMAnewS123.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqMAnewS123.run), genesMAnewS123.run$GENEID)
# keep4.run <- rownames(ddsHTSeqMAnewS123.run) %in% matchedS2.run
# ddsHTSeqMAnewS123.run <- ddsHTSeqMAnewS123.run[keep4.run,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS123.run) == genesMAnewS123.run$GENEID)
# mcols(ddsHTSeqMAnewS123.run) <- cbind(mcols(ddsHTSeqMAnewS123.run), genesMAnewS123.run)
#
# # geneIDMAnewS141 <- rownames(ddsHTSeqMAnewS141)
# # #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# # #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# # genesMAnewS141 <- ensembldb::select(txdb, keys=geneIDMAnewS141, columnns="GENEID","TXCHROM" , keytype="GENEID")
# # #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# # genesMAnewS141 <- na.omit(genesMAnewS141)
# # genesMAnewS141 <- as.data.frame(genesMAnewS141)
# # remove1 <- "chrM"
# # # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# # all(rownames(ddsHTSeqMAnewS141) == genesMAnewS141$GENEID)
# # #need to remove the ones in rownames of dds that don't match genes$GENEID
# # matchedS2 <- intersect(rownames(ddsHTSeqMAnewS141), genesMAnewS141$GENEID)
# # keep4 <- rownames(ddsHTSeqMAnewS141) %in% matchedS2
# # ddsHTSeqMAnewS141 <- ddsHTSeqMAnewS141[keep4,]
# # # length(ddsHTSeqMAnew)
# # all(rownames(ddsHTSeqMAnewS141) == genesMAnewS141$GENEID)
# # mcols(ddsHTSeqMAnewS141) <- cbind(mcols(ddsHTSeqMAnewS141), genesMAnewS141)
# #
# geneIDMAnewS152.run <- rownames(ddsHTSeqMAnewS152.run)
# #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesMAnewS152.run <- ensembldb::select(txdb, keys=geneIDMAnewS152.run, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesMAnewS152.run <- na.omit(genesMAnewS152.run)
# genesMAnewS152.run <- as.data.frame(genesMAnewS152.run)
# remove1 <- "chrM"
# # genesMAnew <- genesMAnew[-(grep(paste0(remove1,"$"),genesMAnew$TXCHROM,perl=TRUE)),]
# all(rownames(ddsHTSeqMAnewS152.run) == genesMAnewS152.run$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedS2.run <- intersect(rownames(ddsHTSeqMAnewS152.run), genesMAnewS152.run$GENEID)
# keep4.run <- rownames(ddsHTSeqMAnewS152.run) %in% matchedS2.run
# ddsHTSeqMAnewS152.run <- ddsHTSeqMAnewS152.run[keep4.run,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsHTSeqMAnewS152.run) == genesMAnewS152.run$GENEID)
# mcols(ddsHTSeqMAnewS152.run) <- cbind(mcols(ddsHTSeqMAnewS152.run), genesMAnewS152.run)
#
#
# #################################################
# #default DESeq run
# # ddsDefGC.short <- DESeq(ddsHTSeqGC.short)
# # ddsDefGC.short
# # attr(dispersionFunction(ddsDefGC.short),"dispPriorVar")
# # norm.factors.with.aneuGC <- as.data.frame(sizeFactors(ddsDefGC.short))
# # dispersions.with.aneuGC <- dispersions(ddsDefGC.short)
# ## with filtered genes
# #default DESeq run
# # ddsDefGC.fil <- DESeq(ddsHTSeqGC.fil)
# # ddsDefGC.fil
# # attr(dispersionFunction(ddsDefGC.fil),"dispPriorVar")
# # norm.factors.with.aneuGC.fil <- as.data.frame(sizeFactors(ddsDefGC.fil))
# # dispersions.with.aneuGC.fil <- dispersions(ddsDefGC.fil)
#
# ## with filtered genes
# #default DESeq run
# # with each sample individually
# # ddsHTSeqGCS1$run <- droplevels(ddsHTSeqGCS1$run)
# ddsDefGCS1.run <- DESeq(ddsHTSeqGCS1.run)
#
# ddsDefGCS1.run
# ddsDefGCS2.run <- DESeq(ddsHTSeqGCS2.run)
# ddsDefGCS2.run
# ddsDefGCS3.run <- DESeq(ddsHTSeqGCS3.run)
# ddsDefGCS3.run
# ddsDefGCS4.run <- DESeq(ddsHTSeqGCS4.run)
# ddsDefGCS4.run
# ddsDefGCS5.run <- DESeq(ddsHTSeqGCS5.run)
# ddsDefGCS5.run
# ddsDefGCS6.run <-DESeq(ddsHTSeqGCS6.run)
# ddsDefGCS6.run
# ddsDefGCS7.run <- DESeq(ddsHTSeqGCS7.run)
# ddsDefGCS7.run
# ddsDefGCS8.run <- DESeq(ddsHTSeqGCS8.run)
# ddsDefGCS8.run
# ddsDefGCS9.run <- DESeq(ddsHTSeqGCS9.run)
# ddsDefGCS9.run
# # ddsHTSeqGCS11$run <- droplevels(ddsHTSeqGCS11$run)
# ddsDefGCS11.run <- DESeq(ddsHTSeqGCS11.run)
# ddsDefGCS11.run
# ddsDefGCS18.run <- DESeq(ddsHTSeqGCS18.run)
# ddsDefGCS18.run
# ddsDefGCS31.run <- DESeq(ddsHTSeqGCS31.run)
# ddsDefGCS31.run
# ddsDefGCS49.run <- DESeq(ddsHTSeqGCS49.run)
# ddsDefGCS49.run
# ddsDefGCS59.run <- DESeq(ddsHTSeqGCS59.run)
# ddsDefGCS59.run
# ddsDefGCS61.run <- DESeq(ddsHTSeqGCS61.run)
# ddsDefGCS61.run
# ddsDefGCS69.run <- DESeq(ddsHTSeqGCS69.run)
# ddsDefGCS69.run
# ddsDefGCS76.run <- DESeq(ddsHTSeqGCS76.run)
# ddsDefGCS76.run
# ddsDefGCS77.run <- DESeq(ddsHTSeqGCS77.run)
# ddsDefGCS77.run


### with run included as factor in design matrix

ddsDefGC_1 <- DESeq(ddsHTSeqGC_1)
ddsDefGC_1
ddsDefGC_2 <- DESeq(ddsHTSeqGC_2)
ddsDefGC_2
ddsDefGC_3 <- DESeq(ddsHTSeqGC_3)
ddsDefGC_3
ddsDefGC_4 <- DESeq(ddsHTSeqGC_4)
ddsDefGC_4
ddsDefGC_5 <- DESeq(ddsHTSeqGC_5)
ddsDefGC_5
# ddsDefGC_6 <- DESeq(ddsHTSeqGC_6)
# ddsDefGC_6
ddsDefGC_7 <- DESeq(ddsHTSeqGC_7)
ddsDefGC_7
ddsDefGC_8 <- DESeq(ddsHTSeqGC_8)
ddsDefGC_8
ddsDefGC_9 <- DESeq(ddsHTSeqGC_9)
ddsDefGC_9
# ddsHTSeqGC_11$run <- droplevels(ddsHTSeqGC_11$run)
ddsDefGC_11 <- DESeq(ddsHTSeqGC_11)
ddsDefGC_11
ddsDefGC_18 <- DESeq(ddsHTSeqGC_18)
ddsDefGC_18
# ddsDefGC_31 <- DESeq(ddsHTSeqGC_31)
# ddsDefGC_31
ddsDefGC_49 <- DESeq(ddsHTSeqGC_49)
ddsDefGC_49
ddsDefGC_59 <- DESeq(ddsHTSeqGC_59)
ddsDefGC_59
ddsDefGC_61 <- DESeq(ddsHTSeqGC_61)
ddsDefGC_61
# ddsDefGC_69 <- DESeq(ddsHTSeqGC_69)
# ddsDefGC_69
ddsDefGC_76 <- DESeq(ddsHTSeqGC_76)
ddsDefGC_76
ddsDefGC_77 <- DESeq(ddsHTSeqGC_77)
ddsDefGC_77
#################################################
#default DESeq run
# ddsDefMAold <- DESeq(ddsHTSeqMAold)
# ddsDefMAold
# attr(dispersionFunction(ddsDefMAold),"dispPriorVar")
# norm.factors.with.aneuMAold <- as.data.frame(sizeFactors(ddsDefMAold))
# dispersions.with.aneuMAold <- dispersions(ddsDefMAold)

### individual lines
ddsDefMAold_1 <- DESeq(ddsHTSeqMAold_1)
ddsDefMAold_1
ddsDefMAold_2 <- DESeq(ddsHTSeqMAold_2)
ddsDefMAold_2
ddsDefMAold_3 <- DESeq(ddsHTSeqMAold_3)
ddsDefMAold_3
ddsDefMAold_4 <- DESeq(ddsHTSeqMAold_4)
ddsDefMAold_4
ddsDefMAold_5 <- DESeq(ddsHTSeqMAold_5)
ddsDefMAold_5
# ddsDefMAold_6 <- DESeq(ddsHTSeqMAold_6)
# ddsDefMAold_6
ddsDefMAold_7 <- DESeq(ddsHTSeqMAold_7)
ddsDefMAold_7
ddsDefMAold_8 <- DESeq(ddsHTSeqMAold_8)
ddsDefMAold_8
ddsDefMAold_9 <- DESeq(ddsHTSeqMAold_9)
ddsDefMAold_9
ddsDefMAold_11 <- DESeq(ddsHTSeqMAold_11)
ddsDefMAold_11
ddsDefMAold_15 <- DESeq(ddsHTSeqMAold_15)
ddsDefMAold_15
ddsDefMAold_28 <- DESeq(ddsHTSeqMAold_28)
ddsDefMAold_28
ddsDefMAold_88 <- DESeq(ddsHTSeqMAold_88)
ddsDefMAold_88
ddsDefMAold_108 <- DESeq(ddsHTSeqMAold_108)
ddsDefMAold_108
ddsDefMAold_119 <- DESeq(ddsHTSeqMAold_119)
ddsDefMAold_119
#################################################
#default DESeq run
# ddsHTSeqMAnew.short$run <- droplevels(ddsHTSeqMAnew.short$run)
# ddsHTSeqMAnew$condition <- droplevels(ddsHTSeqMAnew$condition)

# ddsDefMAnew <- DESeq(ddsHTSeqMAnew)
# ddsDefMAnew
# attr(dispersionFunction(ddsDefMAnew),"dispPriorVar")
# norm.factors.with.aneuMAnew <- as.data.frame(sizeFactors(ddsDefMAnew))
# dispersions.with.aneuMAnew <- dispersions(ddsDefMAnew)


##### individual samples
ddsDefMAnewS29 <- DESeq(ddsHTSeqMAnewS29)
ddsDefMAnewS50 <- DESeq(ddsHTSeqMAnewS50)
ddsDefMAnewS112 <- DESeq(ddsHTSeqMAnewS112)
ddsDefMAnewS115 <- DESeq(ddsHTSeqMAnewS115)
ddsDefMAnewS117 <- DESeq(ddsHTSeqMAnewS117)
ddsDefMAnewS123 <- DESeq(ddsHTSeqMAnewS123)
# ddsDefMAnewS141 <- DESeq(ddsHTSeqMAnewS141)
ddsDefMAnewS152 <- DESeq(ddsHTSeqMAnewS152)

# ## with run included as factor
# ddsDefMAnewS29.run <- DESeq(ddsHTSeqMAnewS29.run)
# ddsDefMAnewS50.run <- DESeq(ddsHTSeqMAnewS50.run)
# ddsDefMAnewS112.run <- DESeq(ddsHTSeqMAnewS112.run)
# ddsDefMAnewS115.run <- DESeq(ddsHTSeqMAnewS115.run)
# ddsDefMAnewS117.run <- DESeq(ddsHTSeqMAnewS117.run)
# ddsDefMAnewS123.run <- DESeq(ddsHTSeqMAnewS123.run)
# # ddsDefMAnewS141 <- DESeq(ddsHTSeqMAnewS141)
# ddsDefMAnewS152.run <- DESeq(ddsHTSeqMAnewS152.run)
#

#######################################################
#### results with aneuploid chromosomes included ######
# resultsNames(ddsDefGC.short)
# resultsS1GC <- results(ddsDefGC,contrast=c("condition","Holly_1","Holly_Anc"))
# resultsS1GC
# sigGenesGC1 <- rownames(subset(resultsS1GC,padj<0.1))
# summary(resultsS1GC)

# resultsNames(ddsDefGC.short)
# resultsS1GC.sh <- results(ddsDefGC.short,contrast=c("condition","Holly_1","Holly_Anc"))
# resultsS1GC.sh
# # sigGenesGC1 <- rownames(subset(resultsS1GC,padj<0.1))
# summary(resultsS1GC.sh)

# resultsNames(ddsDefGC.short)
# resultsS2GC <- results(ddsDefGC.short,contrast=c("condition","Holly_2","Holly_Anc"))
# resultsS2GC.sh
# sigGenesGC1 <- rownames(subset(resultsS1GC,padj<0.1))
# summary(resultsS2GC)

# resultsS2GC <- results(ddsDefGC,contrast=c("condition","Holly_2","Holly_Anc"))
# resultsS2GC
# sigGenesGC2 <- rownames(subset(resultsS2GC,padj<0.1))
# summary(resultsS2GC)

# resultsS3GC <- results(ddsDefGC.short,contrast=c("condition","Holly_3","Holly_Anc"))
# resultsS3GC
# sigGenesGC3 <- rownames(subset(resultsS3GC,padj<0.1))
# summary(resultsS3GC)
#
# resultsS4GC <- results(ddsDefGC.short,contrast=c("condition","Holly_4","Holly_Anc"))
# resultsS4GC
# sigGenesGC4 <- rownames(subset(resultsS4GC,padj<0.1))
# summary(resultsS4GC)
#
# resultsS5GC <- results(ddsDefGC.short,contrast=c("condition","Holly_5","Holly_Anc"))
# resultsS5GC
# sigGenesGC5 <- rownames(subset(resultsS5GC,padj<0.1))
# summary(resultsS5GC)
#
# resultsS7GC <- results(ddsDefGC.short,contrast=c("condition","Holly_7","Holly_Anc"))
# resultsS7GC
# sigGenesGC7 <- rownames(subset(resultsS7GC,padj<0.1))
# summary(resultsS7GC)
#
# resultsS8GC <- results(ddsDefGC.short,contrast=c("condition","Holly_8","Holly_Anc"))
# resultsS8GC
# sigGenesGC8 <- rownames(subset(resultsS8GC,padj<0.1))
# summary(resultsS8GC)
#
#
# resultsS9GC <- results(ddsDefGC.short,contrast=c("condition","Holly_9","Holly_Anc"))
# resultsS9GC
# sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS9GC)
#
#   resultsS1GCDat <- as.data.frame(resultsS1GC)
#     resultsS2GCDat <- as.data.frame(resultsS2GC)
#       resultsS3GCDat <- as.data.frame(resultsS3GC)
#         resultsS5GCDat <- as.data.frame(resultsS5GC)
#           resultsS9GCDat <- as.data.frame(resultsS9GC)

  # add gene names as first column ( which is currently rownames)
  # resS1GCDat <- cbind(GENEID = rownames(resultsS1GCDat), resultsS1GCDat)
  # resS2GCDat <- cbind(GENEID = rownames(resultsS2GCDat), resultsS2GCDat)
  # resS3GCDat <- cbind(GENEID = rownames(resultsS3GCDat), resultsS3GCDat)
  # resS5GCDat <- cbind(GENEID = rownames(resultsS5GCDat), resultsS5GCDat)
  # resS9GCDat <- cbind(GENEID = rownames(resultsS9GCDat), resultsS9GCDat)

  # need to make a column identifying the sample
    # resS1GCDat$Sample <- "1"
    # resS2GCDat$Sample <- "2"
    # resS3GCDat$Sample <- "3"
    # resS5GCDat$Sample <- "5"
    # resS9GCDat$Sample <- "9"

## with filtered genes #################
    #### results with aneuploid chromosomes included ######
# resultsNames(ddsDefGC.fil)
# resultsS1GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_1","Holly_Anc"))
# resultsS1GC.fil
# sigGenesGC1 <- rownames(subset(resultsS1GC,padj<0.1))
# summary(resultsS1GC.fil)
# pdf("resultsS1GC.filMAplot.pdf")
# plotMA(resultsS1GC.fil,ylim=c(-2,2))
# dev.off()

# resultsS2GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_2","Holly_Anc"))
# resultsS2GC.fil
# sigGenesGC2 <- rownames(subset(resultsS2GC,padj<0.1))
# summary(resultsS2GC.fil)
# pdf("resultsS2GC.filMAplot.pdf")
# plotMA(resultsS2GC.fil,ylim=c(-2,2))
# dev.off()

# resultsS3GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_3","Holly_Anc"))
# resultsS3GC.fil
# # sigGenesGC3 <- rownames(subset(resultsS3GC,padj<0.1))
# summary(resultsS3GC.fil)
# pdf("resultsS3GC.filMAplot.pdf")
# plotMA(resultsS3GC.fil,ylim=c(-2,2))
# dev.off()

# resultsS5GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_5","Holly_Anc"))
# resultsS5GC.fil
# # sigGenesGC5 <- rownames(subset(resultsS5GC,padj<0.1))
# summary(resultsS5GC.fil)
# pdf("resultsS5GC.filMAplot.pdf")
# plotMA(resultsS5GC.fil)#,ylim=c(-2,2))
# dev.off()
#
# resultsS9GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_9","Holly_Anc"))
# resultsS9GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS9GC.fil)

# resultsS4GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_4","Holly_Anc"))
# resultsS4GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS4GC.fil)
#
# resultsS7GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_7","Holly_Anc"))
# resultsS7GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS7GC.fil)

# resultsS8GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_8","Holly_Anc"))
# resultsS8GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS8GC.fil)
#
# resultsS11GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_11","Holly_Anc"))
# resultsS11GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS11GC.fil)

# resultsS18GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_18","Holly_Anc"))
# resultsS18GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS18GC.fil)
#
# resultsS49GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_49","Holly_Anc"))
# resultsS49GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS49GC.fil)

# resultsS59GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_59","Holly_Anc"))
# resultsS59GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS59GC.fil)
#
# resultsS61GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_61","Holly_Anc"))
# resultsS61GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS61GC.fil)

# resultsS76GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_76","Holly_Anc"))
# resultsS76GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS76GC.fil)
#
# resultsS77GC.fil <- results(ddsDefGC.fil,contrast=c("condition","Holly_77","Holly_Anc"))
# resultsS77GC.fil
# # sigGenesGC9 <- rownames(subset(resultsS9GC,padj<0.1))
# summary(resultsS77GC.fil)

# with individual samples
resultsNames(ddsDefGC_1)
resultsS1GC <- results(ddsDefGC_1,contrast=c("condition","Holly_1","Holly_Anc"),alpha=0.05)
summary(resultsS1GC)
# alpha set to 0.05, does it make a huge difference?
# resultsS1GC <- results(ddsDefGC_1,contrast=c("condition","Holly_1","Holly_Anc"))
# summary(resultsS1GC)
resultsS2GC <- results(ddsDefGC_2,contrast=c("condition","Holly_2","Holly_Anc"))
summary(resultsS2GC)
resultsS2GC <- results(ddsDefGC_2,contrast=c("condition","Holly_2","Holly_Anc"),alpha=0.05)
summary(resultsS2GC)
resultsS3GC <- results(ddsDefGC_3,contrast=c("condition","Holly_3","Holly_Anc"),alpha=0.05)
summary(resultsS3GC)
resultsS4GC <- results(ddsDefGC_4,contrast=c("condition","Holly_4","Holly_Anc"),alpha=0.05)
summary(resultsS4GC)
resultsS5GC <- results(ddsDefGC_5,contrast=c("condition","Holly_5","Holly_Anc"),alpha=0.05)
summary(resultsS5GC)
# resultsS6GC <- results(ddsDefGC_6,contrast=c("condition","Holly_6","Holly_Anc"),alpha=0.05)
# summary(resultsS6GC)
resultsS7GC <- results(ddsDefGC_7,contrast=c("condition","Holly_7","Holly_Anc"),alpha=0.05)
summary(resultsS7GC)
resultsS8GC <- results(ddsDefGC_8,contrast=c("condition","Holly_8","Holly_Anc"),alpha=0.05)
summary(resultsS8GC)
resultsS9GC <- results(ddsDefGC_9,contrast=c("condition","Holly_9","Holly_Anc"),alpha=0.05)
summary(resultsS9GC)

normCountsGC_9 <- counts(ddsDefGC_9,normalized=TRUE)
write.csv(normCountsGC_9,file="normCountsGC_9.csv")

resultsS11GC <- results(ddsDefGC_11,contrast=c("condition","Holly_11","Holly_Anc"),alpha=0.05)
summary(resultsS11GC)

normCountsGC_11 <- counts(ddsDefGC_11,normalized=TRUE)

resultsS18GC <- results(ddsDefGC_18,contrast=c("condition","Holly_18","Holly_Anc"),alpha=0.05)
summary(resultsS18GC)
# resultsS31GC <- results(ddsDefGC_31,contrast=c("condition","Holly_31","Holly_Anc"),alpha=0.05)
# summary(resultsS31GC)
# # resultsS49GC <- results(ddsDefGC_49,contrast=c("condition","Holly_49","Holly_Anc"),alpha=0.05)
# summary(resultsS49GC)
##without independent filtering
# resultsS49GC <- results(ddsDefGC_49,contrast=c("condition","Holly_49","Holly_Anc"),alpha=0.05)
# summary(resultsS49GC)
# resultsS49GC <- as.data.frame(resultsS49GC)
# resultsS49GC$padj
# normCountsGC_49 <- counts(ddsDefGC_49,normalized=TRUE)
# write.csv(normCountsGC_49,file="normCountsGC_49.csv")

resultsS59GC <- results(ddsDefGC_59,contrast=c("condition","Holly_59","Holly_Anc"),alpha=0.05)#,independentFiltering = FALSE)
summary(resultsS59GC)
normCountsGC_59 <- counts(ddsDefGC_59,normalized=TRUE)
write.csv(normCountsGC_59,file="normCountsGC_59.csv")

resultsS61GC <- results(ddsDefGC_61,contrast=c("condition","Holly_61","Holly_Anc"),alpha=0.05)
summary(resultsS61GC)
# resultsS69GC <- results(ddsDefGC_69,contrast=c("condition","Holly_69","Holly_Anc"))#,alpha=0.05)
# summary(resultsS69GC)
resultsS76GC <- results(ddsDefGC_76,contrast=c("condition","Holly_76","Holly_Anc"),alpha=0.05)
summary(resultsS76GC)
resultsS77GC <- results(ddsDefGC_77,contrast=c("condition","Holly_77","Holly_Anc"),alpha=0.05)
summary(resultsS77GC)

## with run included in design formula
# resultsNames(ddsDefGCS1.run)
# resultsS1GC.run <- results(ddsDefGCS1.run,contrast=c("condition","Holly_1","Holly_Anc"))
# summary(resultsS1GC.run)
# # alpha set to 0.05, does it make a huge difference?
# # resultsS1GC <- results(ddsDefGCS1.fil,contrast=c("condition","Holly_1","Holly_Anc"),alpha = 0.05)
# # summary(resultsS1GC)
# resultsS2GC.run <- results(ddsDefGCS2.run,contrast=c("condition","Holly_2","Holly_Anc"))
# summary(resultsS2GC.run)
# resultsS3GC.run <- results(ddsDefGCS3.run,contrast=c("condition","Holly_3","Holly_Anc"))#,alpha=0.05)
# summary(resultsS3GC.run)
# resultsS4GC.run <- results(ddsDefGCS4.run,contrast=c("condition","Holly_4","Holly_Anc"))#,alpha=0.05)
# summary(resultsS4GC.run)
# resultsS5GC.run <- results(ddsDefGCS5.run,contrast=c("condition","Holly_5","Holly_Anc"))#,alpha=0.05)
# summary(resultsS5GC.run)
# resultsS6GC.run <- results(ddsDefGCS6.run,contrast=c("condition","Holly_6","Holly_Anc"))#,alpha=0.05)
# summary(resultsS6GC.run)
# resultsS7GC.run <- results(ddsDefGCS7.run,contrast=c("condition","Holly_7","Holly_Anc"))#,alpha=0.05)
# summary(resultsS7GC.run)
# resultsS8GC.run <- results(ddsDefGCS8.run,contrast=c("condition","Holly_8","Holly_Anc"))#,alpha=0.05)
# summary(resultsS8GC.run)
# resultsS9GC.run <- results(ddsDefGCS9.run,contrast=c("condition","Holly_9","Holly_Anc"))#,alpha=0.05)#,independentFiltering = FALSE)
# summary(resultsS9GC.run)
# resultsS11GC.run <- results(ddsDefGCS11.run,contrast=c("condition","Holly_11","Holly_Anc"))#,alpha=0.05)
# summary(resultsS11GC.run)
#
# resultsS18GC.run <- results(ddsDefGCS18.run,contrast=c("condition","Holly_18","Holly_Anc"))#,alpha=0.05)
# summary(resultsS18GC.run)
# resultsS31GC.run <- results(ddsDefGCS31.run,contrast=c("condition","Holly_31","Holly_Anc"))#,alpha=0.05)
# summary(resultsS31GC.run)
# resultsS49GC.run <- results(ddsDefGCS49.run,contrast=c("condition","Holly_49","Holly_Anc"))#,alpha=0.05)
# summary(resultsS49GC.run)
# resultsS59GC.run <- results(ddsDefGCS59.run,contrast=c("condition","Holly_59","Holly_Anc"))#,alpha=0.05)#,independentFiltering = FALSE)
# summary(resultsS59GC.run)
# resultsS61GC.run <- results(ddsDefGCS61.run,contrast=c("condition","Holly_61","Holly_Anc"))#,alpha=0.05)
# summary(resultsS61GC.run)
# resultsS69GC.run <- results(ddsDefGCS69.run,contrast=c("condition","Holly_69","Holly_Anc"))#,alpha=0.05)
# summary(resultsS69GC.run)
# resultsS76GC.run <- results(ddsDefGCS76.run,contrast=c("condition","Holly_76","Holly_Anc"))#,alpha=0.05)
# summary(resultsS76GC.run)
# resultsS77GC.run <- results(ddsDefGCS77.run,contrast=c("condition","Holly_77","Holly_Anc"))#,alpha=0.05)
# summary(resultsS77GC.run)

 # put all sample expression data in the same data frame
# euploidGCSamples <- rbind(resS1GCDat, resS2GCDat, resS3GCDat, resS5GCDat, resS9GCDat)
# dim(euploidGCSamples)
# View(euploidGCSamples)
  # # # add chromosome numbers to this file
      # match <- intersect(euploidGCSamples$GENEID, genesGC$GENEID)
      # length(match)
  # #   # tell R which genes to keep
      # rm(keep)
      # keep <- euploidGCSamples$GENEID %in% match
      # length(keep)
      # View(keep)
      # View(euploidGCSamples$GENEID)
# #   # subset list of genes in analysis to include only the ones matching those found in          sample file
#     genesGCeu <- genesGC[keep,]
# # #   # add the correct chromosomes in
#     length(genesGCeu)
#     length(euploidGCSamples)
#     dim(genesGCeu)
#     dim(euploidGCSamples)
#     euploidGCSamples <- cbind(genesGCeu$GENEID, genesGCeu$TXCHROM, euploidGCSamples)
#     # chr1.all <- subset(euploidGCSamplesDE, euploidGCSamplesDE$`genesNoAneu$GENEID`=="chrI")
#     sample1GC <- subset(euploidGCSamples, Sample==1)
#     View(sample1GC)
#     # sample7 <- subset(allSamplesExpr, Sample==7)
# # # first need to filter allSamplesExpr based on p-adj level
#     euploidGCSamplesDiffExpr <- subset(euploidGCSamples, padj<0.1)
#     
    # genesS1GCDE <- subset(euploidGCSamplesDiffExpr, Sample=="1")
    # genesS2GCDE <- subset(euploidGCSamplesDiffExpr, Sample=="2")
    # genesS3GCDE <- subset(euploidGCSamplesDiffExpr, Sample=="3")
    # genesS5GCDE <- subset(euploidGCSamplesDiffExpr, Sample=="5")
    # genesS9GCDE <- subset(euploidGCSamplesDiffExpr, Sample=="9")

#     library(VennDiagram)
# venn.plot <- venn.diagram(
# 	x = list(
# 		A = genesS1GCDE$GENEID,
# 		B = genesS2GCDE$GENEID,
# 		C = genesS3GCDE$GENEID,
# 		D = genesS5GCDE$GENEID,
# 		E = genesS9GCDE$GENEID
# 		),
# 	filename = NULL, #"euploid_commonDE.tiff"
# 	col = "black",
# 	fill = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
# 	alpha = 0.50,
# 	cex = c(1.5, 1.5, 1.5, 1.5, 1.5, 1, 0.8, 1, 0.8, 1, 0.8, 1, 0.8,
# 	 1, 0.8, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 1, 1, 1, 1, 1.5),
# 	cat.col = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
# 	cat.cex = 1.5,
# 	cat.fontface = "bold",
# 	margin = 0.05
# 	)
# pdf("euploidGC_commonDE.pdf")
# grid.draw(venn.plot)

# eupGCLinesDE <- list(genesS1GCDE, genesS2GCDE, genesS3GCDE, genesS5GCDE, genesS9GCDE)
# # want to get list of common DE genes in all the euploid lines
# commonDEGCeup <- Reduce(intersect, Map("[[", eupGCLinesDE, "GENEID"))
# #Reduc
# commonDEGCeup <- as.vector(commonDEGCeup)
# commonDEGCeup <- as.data.frame(commonDEGCeup)
# colnames(commonDEGCeup) <- c("names")
# write.csv(commonDEGCeup, file="commonDEeup.csv")


#######################################################
#### results with aneuploid chromosomes included ######
########## MA old ######################################
# resultsNames(ddsDefMAold)
# resultsS1MAold <- results(ddsDefMAold,contrast=c("condition","SC001","SCA"))
# class(resultsS1MAold)
# sigGenesMAold1 <- rownames(subset(resultsS1MAold,padj<0.1))
# summary(resultsS1MAold)
#
# resultsS2MAold <- results(ddsDefMAold,contrast=c("condition","SC002","SCA"))
# resultsS2MAold
# sigGenesMAold2 <- rownames(subset(resultsS2MAold,padj<0.1))
# summary(resultsS2MAold)

# resultsS3MAold <- results(ddsDefMAold,contrast=c("condition","SC003","SCA"))
# resultsS3MAold
# sigGenesMAold3 <- rownames(subset(resultsS3MAold,padj<0.1))
# summary(resultsS3MAold)
#
# resultsS4MAold <- results(ddsDefMAold,contrast=c("condition","SC004","SCA"))
# resultsS4MAold
# sigGenesMAold4 <- rownames(subset(resultsS4MAold,padj<0.1))
# summary(resultsS4MAold)

# resultsS5MAold <- results(ddsDefMAold,contrast=c("condition","SC005","SCA"))
# resultsS5MAold
# sigGenesMAold5 <- rownames(subset(resultsS5MAold,padj<0.1))
# summary(resultsS5MAold)
#
# resultsS6MAold <- results(ddsDefMAold,contrast=c("condition","SC006","SCA"))
# resultsS6MAold
# sigGenesMAold6 <- rownames(subset(resultsS6MAold,padj<0.1))
# summary(resultsS6MAold)

# resultsS7MAold <- results(ddsDefMAold,contrast=c("condition","SC007","SCA"))
# resultsS7MAold
# sigGenesMAold7 <- rownames(subset(resultsS7MAold,padj<0.1))
# summary(resultsS7MAold)
#
# resultsS8MAold <- results(ddsDefMAold,contrast=c("condition","SC008","SCA"))
# resultsS8MAold
# sigGenesMAold8 <- rownames(subset(resultsS8MAold,padj<0.1))
# summary(resultsS8MAold)

# resultsS9MAold <- results(ddsDefMAold,contrast=c("condition","SC009","SCA"))
# resultsS9MAold
# sigGenesMAold9 <- rownames(subset(resultsS9MAold,padj<0.1))
# summary(resultsS9MAold)

# resultsS11MAold <- results(ddsDefMAold,contrast=c("condition","SC011","SCA"))
# resultsS11MAold
# sigGenesMAold11 <- rownames(subset(resultsS11MAold,padj<0.1))
# summary(resultsS11MAold)
#
# resultsS15MAold <- results(ddsDefMAold,contrast=c("condition","SC015","SCA"))
# resultsS15MAold
# sigGenesMAold15 <- rownames(subset(resultsS15MAold,padj<0.1))
# summary(resultsS15MAold)

# resultsS88MAold <- results(ddsDefMAold,contrast=c("condition","SC088","SCA"))
# resultsS88MAold
# sigGenesMAold88 <- rownames(subset(resultsS88MAold,padj<0.1))
# summary(resultsS88MAold)
#
# resultsS108MAold <- results(ddsDefMAold,contrast=c("condition","SC108","SCA"))
# resultsS108MAold
# sigGenesMAold108 <- rownames(subset(resultsS108MAold,padj<0.1))
# summary(resultsS108MAold)

# resultsS119MAold <- results(ddsDefMAold,contrast=c("condition","SC119","SCA"))
# resultsS119MAold
# sigGenesMAold119 <- rownames(subset(resultsS119MAold,padj<0.1))
# summary(resultsS119MAold)
#
# resultsS28MAold <- results(ddsDefMAold,contrast=c("condition","SC028","SCA"))
# resultsS28MAold
# sigGenesMAold28 <- rownames(subset(resultsS28MAold,padj<0.1))
# summary(resultsS28MAold)

  # resultsS1MAoldDat <- as.data.frame(resultsS1MAold)
  #   resultsS2MAoldDat <- as.data.frame(resultsS2MAold)
  #     resultsS3MAoldDat <- as.data.frame(resultsS3MAold)
  #     resultsS4MAoldDat <- as.data.frame(resultsS4MAold)
  #       resultsS5MAoldDat <- as.data.frame(resultsS5MAold)
  #       resultsS6MAoldDat <- as.data.frame(resultsS6MAold)
  #       resultsS7MAoldDat <- as.data.frame(resultsS7MAold)
  #       resultsS8MAoldDat <- as.data.frame(resultsS8MAold)
  #         # resultsS9MAoldDat <- as.data.frame(resultsS9MAold)
  #          resultsS28MAoldDat <- as.data.frame(resultsS28MAold)
  #         
  # add gene names as first column ( which is currently rownames)
  # resS1MAoldDat <- cbind(GENEID = rownames(resultsS1MAoldDat), resultsS1MAoldDat)
  # write.csv(resS1MAoldDat, file="resS1MAoldDat.csv")
  # resS2MAoldDat <- cbind(GENEID = rownames(resultsS2MAoldDat), resultsS2MAoldDat)
  # write.csv(resS2MAoldDat, file="resS2MAoldDat.csv")
  # resS3MAoldDat <- cbind(GENEID = rownames(resultsS3MAoldDat), resultsS3MAoldDat)
  # write.csv(resS3MAoldDat, file="resS3MAoldDat.csv")
  # resS4MAoldDat <- cbind(GENEID = rownames(resultsS4MAoldDat), resultsS4MAoldDat)
  # write.csv(resS4MAoldDat, file="resS4MAoldDat.csv")
  # resS5MAoldDat <- cbind(GENEID = rownames(resultsS5MAoldDat), resultsS5MAoldDat)
  # write.csv(resS5MAoldDat, file="resS5MAoldDat.csv")
  # resS6MAoldDat <- cbind(GENEID = rownames(resultsS6MAoldDat), resultsS6MAoldDat)
  # write.csv(resS6MAoldDat, file="resS6MAoldDat.csv")
  # resS7MAoldDat <- cbind(GENEID = rownames(resultsS7MAoldDat), resultsS7MAoldDat)
  # write.csv(resS7MAoldDat, file="resS7MAoldDat.csv")
  # resS8MAoldDat <- cbind(GENEID = rownames(resultsS8MAoldDat), resultsS8MAoldDat)
  # write.csv(resS8MAoldDat, file="resS8MAoldDat.csv")
  # # resS9MAoldDat <- cbind(GENEID = rownames(resultsS9MAoldDat), resultsS9MAoldDat)
  #   resS28MAoldDat <- cbind(GENEID = rownames(resultsS28MAoldDat), resultsS28MAoldDat)
  #   write.csv(resS28MAoldDat, file="resS28MAoldDat.csv")
  #
  # need to make a column identifying the sample
    # resS1MAoldDat$Sample <- "1"
    # resS2MAoldDat$Sample <- "2"
    # resS3MAoldDat$Sample <- "3"
    # resS4MAoldDat$Sample <- "4"
    # resS5MAoldDat$Sample <- "5"
    # resS6MAoldDat$Sample <- "6"
    # resS7MAoldDat$Sample <- "7"
    # resS8MAoldDat$Sample <- "8"
    # # resS9MAoldDat$Sample <- "9"
    # resS28MAoldDat$Sample <- "28"
    #

# put all sample expression data in the same data frame

# euploidMAoldSamples <- rbind(resS1MAoldDat, resS2MAoldDat, resS3MAoldDat, resS4MAoldDat, resS5MAoldDat,resS6MAoldDat,resS7MAoldDat, resS8MAoldDat, resS28MAoldDat)
#
#
# dim(euploidMAoldSamples)

# View(euploidMAoldSamples)
#   # # # add chromosome numbers to this file
#       match <- intersect(euploidMAoldSamples$GENEID, genesMAold$GENEID)
#       length(match)
#   # #   # tell R which genes to keep
#       rm(keep)
#       keep <- euploidMAoldSamples$GENEID %in% match
#       length(keep)
#       View(keep)
#       View(euploidMAoldSamples$GENEID)
# # #   # subset list of genes in analysis to include only the ones matching those found in          sample file
#     genesMAoldeu <- genesMAold[keep,]
# # #   # add the correct chromosomes in
#     length(genesMAoldeu)
    # length(euploidMAoldSamples)
    # dim(genesMAoldeu)
    # dim(euploidMAoldSamples)
    #
    #
    # euploidMAoldSamples <- cbind(genesMAoldeu$GENEID, genesMAoldeu$TXCHROM, euploidMAoldSamples)
    # class(euploidMAoldSamples)
    # class(euploidGCSamples)
#     # chr1.all <- subset(euploidMAoldSamplesDE, euploidMAoldSamplesDE$`genesNoAneu$GENEID`=="chrI")
#     sample1MAold <- subset(euploidMAoldSamples, Sample==1)
#     View(sample1MAold)
#     # sample7 <- subset(allSamplesExpr, Sample==7)
# # # first need to filter allSamplesExpr based on p-adj level
#     euploidMAoldSamplesDiffExpr <- subset(euploidMAoldSamples, padj<0.1)
#     

    # genesS1MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="1")
    # genesS2MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="2")
    # genesS3MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="3")
    # genesS4MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="4")
    # genesS5MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="5")
    # genesS6MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="6")
    # genesS7MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="7")
    # genesS8MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="8")
    # # line 9 is potentially aneuploid
    # # genesS9MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="9")
    # # line 28 is euploid
    # genesS28MAoldDE <- subset(euploidMAoldSamplesDiffExpr, Sample=="28")
    # View(genesS28MAoldDE)
#     library(VennDiagram)
# venn.plot <- venn.diagram(
# 	x = list(
# 		A = genesS1GCDE$GENEID,
# 		B = genesS2GCDE$GENEID,
# 		C = genesS3GCDE$GENEID,
# 		D = genesS5GCDE$GENEID,
# 		E = genesS9GCDE$GENEID
# 		),
# 	filename = NULL, #"euploid_commonDE.tiff"
# 	col = "black",
# 	fill = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
# 	alpha = 0.50,
# 	cex = c(1.5, 1.5, 1.5, 1.5, 1.5, 1, 0.8, 1, 0.8, 1, 0.8, 1, 0.8,
# 	 1, 0.8, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 1, 1, 1, 1, 1.5),
# 	cat.col = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
# 	cat.cex = 1.5,
# 	cat.fontface = "bold",
# 	margin = 0.05
# 	)
# pdf("euploidGC_commonDE.pdf")
# grid.draw(venn.plot)

# eupMAoldLinesDE <- list(genesS1MAoldDE, genesS2MAoldDE, genesS3MAoldDE, genesS4MAoldDE,genesS5MAoldDE,genesS6MAoldDE,genesS7MAoldDE,genesS8MAoldDE, genesS28MAoldDE)
# # want to get list of common DE genes in all the euploid lines
# # class(eupMAoldLines)
# class(eupMAoldLinesDE)
# commonDEMAoldeup <- Reduce(intersect, Map("[[", eupMAoldLinesDE, "GENEID"))
# #Reduc
# commonDEMAoldeup <- as.data.frame(commonDEMAoldeup)
# colnames(commonDEMAoldeup) <- c("names")
# write.csv(commonDEMAoldeup, file="commonDEeupMAold.csv")

###### individual samples
resultsNames(ddsDefMAold_1)
resultsS1MAold <- results(ddsDefMAold_1,contrast=c("condition","SC001","SCA"))
class(resultsS1MAold)
summary(resultsS1MAold)

resultsNames(ddsDefMAold_2)
resultsS2MAold <- results(ddsDefMAold_2,contrast=c("condition","SC002","SCA"))
class(resultsS2MAold)
summary(resultsS2MAold)

resultsNames(ddsDefMAold_3)
resultsS3MAold <- results(ddsDefMAold_3,contrast=c("condition","SC003","SCA"))
class(resultsS3MAold)
summary(resultsS3MAold)

resultsNames(ddsDefMAold_4)
resultsS4MAold <- results(ddsDefMAold_4,contrast=c("condition","SC004","SCA"))
class(resultsS4MAold)
summary(resultsS4MAold)

resultsNames(ddsDefMAold_5)
resultsS5MAold <- results(ddsDefMAold_5,contrast=c("condition","SC005","SCA"))
class(resultsS5MAold)
summary(resultsS5MAold)

# resultsNames(ddsDefMAold_6)
# resultsS6MAold <- results(ddsDefMAold_6,contrast=c("condition","SC006","SCA"))
# class(resultsS6MAold)
# summary(resultsS6MAold)

# normCountsMAold_6 <- counts(ddsDefMAold_6,normalized=TRUE)
# write.csv(normCountsMAold_6,file="normCountsMAold_6.csv")

resultsNames(ddsDefMAold_7)
resultsS7MAold <- results(ddsDefMAold_7,contrast=c("condition","SC007","SCA"))
class(resultsS7MAold)
summary(resultsS7MAold)

resultsNames(ddsDefMAold_8)
resultsS8MAold <- results(ddsDefMAold_8,contrast=c("condition","SC008","SCA"))
class(resultsS8MAold)
summary(resultsS8MAold)

resultsNames(ddsDefMAold_9)
resultsS9MAold <- results(ddsDefMAold_9,contrast=c("condition","SC009","SCA"))
class(resultsS9MAold)
summary(resultsS9MAold)

resultsNames(ddsDefMAold_11)
resultsS11MAold <- results(ddsDefMAold_11,contrast=c("condition","SC011","SCA"))
class(resultsS11MAold)
summary(resultsS11MAold)

resultsNames(ddsDefMAold_15)
resultsS15MAold <- results(ddsDefMAold_15,contrast=c("condition","SC015","SCA"))
class(resultsS15MAold)
summary(resultsS15MAold)

resultsNames(ddsDefMAold_28)
resultsS28MAold <- results(ddsDefMAold_28,contrast=c("condition","SC028","SCA"))
class(resultsS28MAold)
summary(resultsS28MAold)

resultsNames(ddsDefMAold_88)
resultsS88MAold <- results(ddsDefMAold_88,contrast=c("condition","SC088","SCA"))
class(resultsS88MAold)
summary(resultsS88MAold)

resultsNames(ddsDefMAold_108)
resultsS108MAold <- results(ddsDefMAold_108,contrast=c("condition","SC108","SCA"))
class(resultsS108MAold)
summary(resultsS108MAold)

resultsNames(ddsDefMAold_119)
resultsS119MAold <- results(ddsDefMAold_119,contrast=c("condition","SC119","SCA"))
class(resultsS119MAold)
summary(resultsS119MAold)


################################################################################

#aneuploid chromosomes removed from normalization
# allGenesGC <- as.data.frame(mcols(ddsHTSeqGC))
# View(allGenesGC)
# ddsGCNoAneuInNorm <- estimateSizeFactors(ddsHTSeqGC, controlGenes=c(85:481,482:629,630:1373,1647:1761,3090:3403,3881:4340))
# genesGCnoAn <- rownames(ddsGCNoAneuInNorm)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesNoAneuGC <- ensembldb::select(txdb, keys=genesGCnoAn, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesNoAneuGC <- na.omit(genesNoAneuGC)
# genesNoAneuGC <- as.data.frame(genesNoAneuGC)
# # remove1 <- "chrM"
# # genesNoAneu <- genesNoAneu[-(grep(paste0(remove1,"$"),genesNoAneu$TXCHROM,perl=TRUE)),]
# ddsGCNoAneuInNorm <- estimateDispersions(ddsGCNoAneuInNorm)
# attr(dispersionFunction(ddsGCNoAneuInNorm),"dispPriorVar")
# ddsGCNoAneuInNorm <- nbinomWaldTest(ddsGCNoAneuInNorm,useT=TRUE,df=(nrow(ddsGCNoAneuInNorm)))
# norm.factors.no.aneuGC <- as.data.frame(sizeFactors(ddsGCNoAneuInNorm))
# design(ddsGCNoAneuInNorm)
# dispersions.no.aneuGC <- dispersions(ddsGCNoAneuInNorm)
# #add annotation to DESeq2 data set
# all(rownames(ddsGCNoAneuInNorm) == genesNoAneuGC$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedNoAneuGC <- intersect(rownames(ddsGCNoAneuInNorm), genesNoAneuGC$GENEID)
# keep3 <- rownames(ddsGCNoAneuInNorm) %in% matchedNoAneuGC
# ddsGCNoAneuInNorm <- ddsGCNoAneuInNorm[keep3,]
# length(ddsGCNoAneuInNorm)
# all(rownames(ddsGCNoAneuInNorm) == genesNoAneuGC$GENEID)
# mcols(ddsGCNoAneuInNorm)
# #then check again
# all(rownames(ddsGCNoAneuInNorm) == genesNoAneuGC$GENEID)
# #nrow(ddsNoAneuInNorm)
# #nrow(genesNoAneu)
# #mcols(ddsNoAneuInNorm) <- cbind(mcols(ddsNoAneuInNorm), genesNoAneu)
# #mcols(ddsNoAneuInNorm)
# #write.csv(counts(dds), file="counts.Rsubread.csv")
# #mcols(dds)
# # write.csv(genes,file="gene_anno.csv")
# # write.csv(counts(ddsHTSeq),file="counts.filtered.csv")

############################################################################
#aneuploid chromosomes removed
# allGenesMAold <- as.data.frame(mcols(ddsHTSeqMAold))
# View(allGenesMAold)
# ddsMAoldNoAneuInNorm <- estimateSizeFactors(ddsHTSeqMAold, controlGenes=c(1:2268,2740:5667))
# genesMAoldnoAn <- rownames(ddsMAoldNoAneuInNorm)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesNoAneuMAold <- ensembldb::select(txdb, keys=genesMAoldnoAn, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesNoAneuMAold <- na.omit(genesNoAneuMAold)
# genesNoAneuMAold <- as.data.frame(genesNoAneuMAold)
# # remove1 <- "chrM"
# # genesNoAneu <- genesNoAneu[-(grep(paste0(remove1,"$"),genesNoAneu$TXCHROM,perl=TRUE)),]
# ddsMAoldNoAneuInNorm <- estimateDispersions(ddsMAoldNoAneuInNorm)
# attr(dispersionFunction(ddsMAoldNoAneuInNorm),"dispPriorVar")
# ddsMAoldNoAneuInNorm <- nbinomWaldTest(ddsMAoldNoAneuInNorm,useT=TRUE,df=(nrow(ddsMAoldNoAneuInNorm)))
# norm.factors.no.aneuMAold <- as.data.frame(sizeFactors(ddsMAoldNoAneuInNorm))
# design(ddsMAoldNoAneuInNorm)
# dispersions.no.aneuMAold <- dispersions(ddsMAoldNoAneuInNorm)
# #add annotation to DESeq2 data set
# all(rownames(ddsMAoldNoAneuInNorm) == genesNoAneuMAold$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedNoAneuMAold <- intersect(rownames(ddsMAoldNoAneuInNorm), genesNoAneuMAold$GENEID)
# keep3 <- rownames(ddsMAoldNoAneuInNorm) %in% matchedNoAneuMAold
# ddsMAoldNoAneuInNorm <- ddsMAoldNoAneuInNorm[keep3,]
# length(ddsMAoldNoAneuInNorm)
# all(rownames(ddsMAoldNoAneuInNorm) == genesNoAneuMAold$GENEID)
# mcols(ddsMAoldNoAneuInNorm)
# #then check again
# all(rownames(ddsMAoldNoAneuInNorm) == genesNoAneuMAold$GENEID)
#
# ############################################################################
# #aneuploid chromosomes removed
# allGenesMAnew <- as.data.frame(mcols(ddsHTSeqMAnew))
# View(allGenesMAnew)
# ddsMAnewNoAneuInNorm <- estimateSizeFactors(ddsHTSeqMAnew, controlGenes=c(89:485,638:1385,1657:1769,2549:3412,3886:4337,4723:5237))
# genesMAnewnoAn <- rownames(ddsMAnewNoAneuInNorm)
# library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
# txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
# genesNoAneuMAnew <- ensembldb::select(txdb, keys=genesMAnewnoAn, columnns="GENEID","TXCHROM" , keytype="GENEID")
# #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
# genesNoAneuMAnew <- na.omit(genesNoAneuMAnew)
# genesNoAneuMAnew <- as.data.frame(genesNoAneuMAnew)
# # remove1 <- "chrM"
# # genesNoAneu <- genesNoAneu[-(grep(paste0(remove1,"$"),genesNoAneu$TXCHROM,perl=TRUE)),]
# ddsMAnewNoAneuInNorm <- estimateDispersions(ddsMAnewNoAneuInNorm)
# attr(dispersionFunction(ddsMAnewNoAneuInNorm),"dispPriorVar")
# ddsMAnewNoAneuInNorm <- nbinomWaldTest(ddsMAnewNoAneuInNorm,useT=TRUE,df=(nrow(ddsMAnewNoAneuInNorm)))
# norm.factors.no.aneuMAnew <- as.data.frame(sizeFactors(ddsMAnewNoAneuInNorm))
# design(ddsMAnewNoAneuInNorm)
# dispersions.no.aneuMAnew <- dispersions(ddsMAnewNoAneuInNorm)
# #add annotation to DESeq2 data set
# all(rownames(ddsMAnewNoAneuInNorm) == genesNoAneuMAnew$GENEID)
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# matchedNoAneuMAnew <- intersect(rownames(ddsMAnewNoAneuInNorm), genesNoAneuMAnew$GENEID)
# keep3 <- rownames(ddsMAnewNoAneuInNorm) %in% matchedNoAneuMAnew
# ddsMAnewNoAneuInNorm <- ddsMAnewNoAneuInNorm[keep3,]
# length(ddsMAnewNoAneuInNorm)
# all(rownames(ddsMAnewNoAneuInNorm) == genesNoAneuMAnew$GENEID)
# mcols(ddsMAnewNoAneuInNorm)
# #then check again
# all(rownames(ddsMAnewNoAneuInNorm) == genesNoAneuMAnew$GENEID)

```

Now want to make distributions of ratios for cis and trans genes in aneuploid samples

```{r ratioDistributions,eval=TRUE,include=TRUE}

############ Euploid distributions
ratioDistributionsEuploids <- function(dds, genes, sample) {
  
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)
library(ggplot2)

allGenes <- qplot(ratios$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios$sampleOverAnc)), main=sample, xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios$sampleOverAnc, na.rm = T)), colour = "red", linetype ="longdash", size = .5 ) +  geom_vline(aes(xintercept = median(ratios$sampleOverAnc, na.rm = T)), colour = "blue", linetype ="longdash", size = .5 ) + theme_classic() + geom_vline(xintercept=1.5, col="magenta") + geom_vline(xintercept=0.5,col="green") + geom_vline(xintercept=2,col="orange") 

print(allGenes)

}

ratioDistGC1 <- ratioDistributionsEuploids(ddsDefGC_1,genesGC_1,"Het Sample 1")
ratioDistGC2 <- ratioDistributionsEuploids(ddsDefGC_2,genesGC_2,"Het Sample 2")
ratioDistGC3 <- ratioDistributionsEuploids(ddsDefGC_3,genesGC_3,"Het Sample 3")
ratioDistGC5 <- ratioDistributionsEuploids(ddsDefGC_5,genesGC_5,"Het Sample 5")
# ratioDistributionsEuploids(ddsDefGC_6,genesGC_6,"Sample_6_GC")
ratioDistGC9 <- ratioDistributionsEuploids(ddsDefGC_9,genesGC_9,"Het Sample 9")


library(cowplot)

hetEups <- plot_grid(ratioDistGC1, ratioDistGC2, ratioDistGC3, ratioDistGC5, ratioDistGC9, ncol = 2)
pdf("hetEups.pdf")
hetEups
dev.off()

library(car)
ks.test(ratioDistGC2,ratioDistGC3,alternative="two.sided")

ratioDistMA1 <- ratioDistributionsEuploids(ddsDefMAold_1,genesMAold_1,"Sample_1_MA")
ratioDistMA2 <- ratioDistributionsEuploids(ddsDefMAold_2,genesMAold_2,"Sample_2_MA")
ratioDistMA3 <- ratioDistributionsEuploids(ddsDefMAold_3,genesMAold_3,"Sample_3_MA")
ratioDistMA4 <- ratioDistributionsEuploids(ddsDefMAold_4,genesMAold_4,"Sample_4_MA")
ratioDistMA5 <- ratioDistributionsEuploids(ddsDefMAold_5,genesMAold_5,"Sample_5_MA")
ratioDistMA7 <- ratioDistributionsEuploids(ddsDefMAold_7,genesMAold_7,"Sample_7_MA")
ratioDistMA8 <- ratioDistributionsEuploids(ddsDefMAold_8,genesMAold_8,"Sample_8_MA")

ratioDistributionsEuploidsMANew <- function(dds, genes, sample) {
  
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:5)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)

allGenes <- qplot(ratios$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios$sampleOverAnc)), main=paste(sample), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1)+ coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic() + theme_classic() + geom_vline(xintercept=1.5, col="magenta") + geom_vline(xintercept=0.5,col="green") + geom_vline(xintercept=2,col="orange") 

print(allGenes)

}

ratioDistMA50 <- ratioDistributionsEuploidsMANew(ddsDefMAnewS50,genesMAnewS50,"Sample_50_MA")
ratioDistMA115 <- ratioDistributionsEuploidsMANew(ddsDefMAnewS115,genesMAnewS115,"Sample_115_MA")

homEups <- plot_grid(ratioDistMA1, ratioDistMA2, ratioDistMA3, ratioDistMA4, ratioDistMA5, ratioDistMA7, ratioDistMA8, ratioDistMA50, ratioDistMA115 , ncol = 2)
pdf("homEups.pdf")
homEups
dev.off()
#####################################################################
#####################################################################

dds=ddsDefGC_4
genes=genesGC_4
anChr="chrV"
sample="Sample_4_GC"

ratioDistributionsTrisomy <- function(dds, genes, anChr,sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]

# cisHist <- hist(ratiosGC4.cis$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy,Cis Genes")
#
# transHist <- hist(ratiosGC4.Trans$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy, Trans Genes")
### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Trisomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic()
#+ theme(plot.margin = margin(t=2,r=4,b=2,l=4,unit="cm"))
{pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
dev.off()}

trans <- qplot(ratios.Trans$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)), main=paste("Trisomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5)
{pdf(paste("transGenes",sample,".pdf",sep=""))
  print(trans)
dev.off()}
# meanTrans <- mean(ratios.Trans$sampleOverAnc,na.rm=T)
# medianTrans <- median(ratios.Trans$sampleOverAnc,na.rm=T)
# modeTrans <- mode(ratios.Trans$sampleOverAnc)
# return(ratios.cis$sampleOverAnc)
write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
return(ratios.Trans$sampleOverAnc)
# return(meanTrans)
# return(medianTrans)
# return(modeTrans)
# write.csv(ratios.Trans,file=paste("trans",sample,anChr,".csv",sep=""))
}

#count # of occurences of each chromosome in genesGC dataset
# table(genesGC_1$TXCHROM)

# dev.off()
### GC lines

#euploid lines
ratioDistributionsTrisomy(ddsDefGC_1,genesGC_1,"chrI","Sample_1_GC_ChrI")

ratioDistributionsTrisomy(ddsDefGC_7,genesGC_7,"chrI","Sample_7_GC_ChrI")
ratioDistGC7 <- ratioDistributionsTrisomy(ddsDefGC_7,genesGC_7,"chrI","Sample_7_GC_ChrI")
meanGC7 <- mean(ratioDistGC7)
medianGC7 <- median(ratioDistGC7)
ratioDistributionsTrisomy(ddsDefGC_4,genesGC_4,"chrV","Sample_4_GC_ChrV")
ratioDistGC4 <- ratioDistributionsTrisomy(ddsDefGC_4,genesGC_4,"chrV","Sample_4_GC_ChrV")
meanGC4 <- mean(ratioDistGC4)
medianGC4 <- median(ratioDistGC4)
stdevGC4 <- sd(ratioDistGC4)
ratioDistGC1 <- ratioDistributionsTrisomy(ddsDefGC_1,genesGC_1,"chrI","Sample_1_GC_ChrI")
ratioDistGC7 <- ratioDistributionsTrisomy(ddsDefGC_7,genesGC_7,"chrI","Sample_7_GC_ChrI")
ks.test(ratioDistGC1,ratioDistGC7)


ratioDistGC49 <- ratioDistributionsTrisomy(ddsDefGC_49,genesGC_49,"chrV","Sample_49_GC_ChrV")
meanGC49 <- mean(ratioDistGC49)
medianGC49 <- median(ratioDistGC49)
stdevGC49 <- sd(ratioDistGC49)
ratioDistGC59 <- ratioDistributionsTrisomy(ddsDefGC_59,genesGC_59,"chrVII","Sample_59_GC_ChrVII")
meanGC59 <- mean(ratioDistGC59)
medianGC59 <- median(ratioDistGC59)
stdevGC59 <- sd(ratioDistGC59)
ratioDistGC61 <- ratioDistributionsTrisomy(ddsDefGC_61,genesGC_61,"chrVII","Sample_61_GC_ChrVII")
meanGC61 <- mean(ratioDistGC61)
medianGC61 <- median(ratioDistGC61)
stdevGC61 <- sd(ratioDistGC61)
ratioDistGC77 <- ratioDistributionsTrisomy(ddsDefGC_77,genesGC_77,"chrXII","Sample_77_GC_ChrXII")
meanGC77 <- mean(ratioDistGC77)
medianGC77 <- median(ratioDistGC77)
stdevGC77 <- sd(ratioDistGC77)

# ratioDistributionsTrisomy(ddsDefGC_18,genesGC_18,"chrXII","Sample_18_GC_ChrXII")
# ratioDistributionsTrisomy(ddsDefGC_11,genesGC_11,"chrXV","Sample_11_GC_ChrXV")


### MA old lines trisomic
# ratioDistributionsTrisomy(ddsDefMAold_108,genesMAold_108,"chrVIII","Sample_108_MA_ChrVIII")
ratioDistMA15 <- ratioDistributionsTrisomy(ddsDefMAold_15,genesMAold_15,"chrIX","Sample_15_MA_ChrIX")
meanMA15 <- mean(ratioDistMA15)
medianMA15 <- median(ratioDistMA15)
stdevMA15 <- sd(ratioDistMA15)
ratioDistMA88 <- ratioDistributionsTrisomy(ddsDefMAold_88,genesMAold_88,"chrIX","Sample_88_MA_ChrIX")
meanMA88 <- mean(ratioDistMA88)
medianMA88 <- median(ratioDistMA88)
stdevMA88 <- sd(ratioDistMA88)
ratioDistMA119 <- ratioDistributionsTrisomy(ddsDefMAold_119,genesMAold_119,"chrIX","Sample_119_MA_ChrIX")
meanMA119 <- mean(ratioDistMA119)
medianMA119 <- median(ratioDistMA119)
stdevMA119 <- sd(ratioDistMA119)
ratioDistMA9 <- ratioDistributionsTrisomy(ddsDefMAold_9,genesMAold_9,"chrXIV","Sample_9_MA_ChrXIV")
meanMA9 <- mean(ratioDistMA9)
medianMA9 <- median(ratioDistMA9)
stdevMA9 <- sd(ratioDistMA9)


#### for those lines with multiple aneuploidies
ratioDistributionsTrisomyMulti <- function(dds, genes, anChr, anChr2, sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]

ratios.keepTrans2 <- ratios$chr!=anChr2
ratios.Trans2 <- ratios.Trans[ratios.keepTrans2,]

# cisHist <- hist(ratiosGC4.cis$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy,Cis Genes")
#
# transHist <- hist(ratiosGC4.Trans$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy, Trans Genes")
### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Trisomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic()
{pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
dev.off()}


trans <- qplot(ratios.Trans2$sampleOverAnc,geom="histogram",breaks=c(seq(0,6,by=0.05),max(ratios.Trans2$sampleOverAnc)), main=paste("Trisomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans2$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans2$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic()
{pdf(paste("transGenes",sample,".pdf",sep=""))
  print(trans)
dev.off()}
write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
return(ratios.Trans2$sampleOverAnc)
}

ratioDistGC18 <- ratioDistributionsTrisomyMulti(ddsDefGC_18,genesGC_18,"chrI","chrXII", "Sample_18_GC_ChrI")
meanGC18 <- mean(ratioDistGC18)
medianGC18 <- median(ratioDistGC18)
stdevGC18 <- sd(ratioDistGC18)
ratioDistributionsTrisomyMulti(ddsDefGC_11,genesGC_11,"chrXV","chrI", "Sample_11_GC_ChrXV")


ratioDistributionsTrisomyMulti(ddsDefMAold_108,genesMAold_108,"chrVIII","chrIX", "Sample_108_MA_ChrVIII")

### the chromosomes reversed
ratioDistributionsTrisomyMulti(ddsDefGC_18,genesGC_18,"chrXII","chrI", "Sample_18_GC_ChrXII")


### line 76 has 2 fully duplicated chromosomes and one partial duplication


ratioDistributionsTrisomyMulti2 <- function(dds, genes, anChr, anChr2, anChr3, sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
class(ratios)
class(ratios$sampleOverAnc)
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]

ratios.keepTrans2 <- ratios$chr!=anChr2
ratios.Trans2 <- ratios.Trans[ratios.keepTrans2,]

ratios.keepTrans3 <- ratios$chr!=anChr3
ratios.Trans3 <- ratios.Trans2[ratios.keepTrans3,]

# cisHist <- hist(ratiosGC4.cis$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy,Cis Genes")
#
# transHist <- hist(ratiosGC4.Trans$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy, Trans Genes")
### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Trisomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic()
{pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
dev.off()}


trans <- qplot(ratios.Trans3$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans3$sampleOverAnc)), main=paste("Trisomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="magenta") + geom_vline(xintercept=0.67,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans3$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans3$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic()
{pdf(paste("transGenes",sample,".pdf",sep=""))
  print(trans)
dev.off()}
write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
}
ratioDistributionsTrisomyMulti2(ddsDefGC_76,genesGC_76,"chrIX","chrXIV","chrX", "Sample_76_GC_ChIX")
ratioDistributionsTrisomyMulti2(ddsDefGC_76,genesGC_76,"chrXIV","chrIX","chrX", "Sample_76_GC_ChrXIV")
ratioDistributionsTrisomyMulti2(ddsDefGC_76,genesGC_76,"chrX","chrXIV","chrIX",  "Sample_76_GC_ChrX")


ratioDistributionsTrisomyMultiMANew <- function(dds, genes, anChr,anChri2,sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:5)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]


# cisHist <- hist(ratiosGC4.cis$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy,Cis Genes")
#
# transHist <- hist(ratiosGC4.Trans$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy, Trans Genes")
### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Trisomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic()
{pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
dev.off()}


trans <- qplot(ratios.Trans$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)), main=paste("Trisomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green")+
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic()
{pdf(paste("transGenes",sample,".pdf",sep="")) 
  print(trans)
dev.off()}
write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
return(ratios.cis$sampleOverAnc)
return(ratios.Trans$sampleOverAnc)
}


ratioDistributionsTrisomyMultiMANew(ddsDefMAnewS152,genesMAnewS152,"chrI","chrVIII", "Sample_152_MA_ChrI")
# other chromosome

ratioDistributionsTrisomyMultiMANew(ddsDefMAnewS152,genesMAnewS152,"chrVIII","chrI", "Sample_152_MA_ChrVIII")

### need a function for monosomy
ratioDistributionsMonosomy <- function(dds, genes, anChr,anChr2, sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]
ratios.keepTrans2 <- ratios$chr!=anChr2
ratios.Trans2 <- ratios.Trans[ratios.keepTrans2,]

# cisHist <- hist(ratiosGC4.cis$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy,Cis Genes")
#
# transHist <- hist(ratiosGC4.Trans$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy, Trans Genes")
### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Monosomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.5,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic()
{pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
dev.off()}


trans <- qplot(ratios.Trans2$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)), main=paste("Monosomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.5,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans2$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans2$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic()
{pdf(paste("transGenes",sample,".pdf",sep=""))
  print(trans)
dev.off()}
write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
return(ratios.Trans2$sampleOverAnc)
}


# for GC line 11

ratioDistributionsMonosomy(ddsDefGC_11,genesGC_11,"chrI","chrXV", "Sample_11_GC_I")


# for MA old line 108
ratioDistributionsMonosomy(ddsDefMAold_108,genesMAold_108,"chrIX","chrVIII" ,"Sample_108_MA_IX")


### need to do this for tetrasomy not trisomy
ratioDistributionsTetrasomy <- function(dds, genes, anChr,sample) {

counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:6)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]

# cisHist <- hist(ratiosGC4.cis$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy,Cis Genes")
#
# transHist <- hist(ratiosGC4.Trans$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy, Trans Genes")
### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Tetrasomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=2, col="darkmagenta") + geom_vline(xintercept=0.5,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic()
{pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
  mean(cis)
dev.off()}


trans <- qplot(ratios.Trans$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)), main=paste("Tetrasomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=2, col="darkmagenta") + geom_vline(xintercept=0.5,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic()
{pdf(paste("transGenes",sample,".pdf",sep=""))
  print(trans)
dev.off()}
write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
return(ratios.Trans$sampleOverAnc)
}
ratioDistGC8 <- ratioDistributionsTetrasomy(ddsDefGC_8,genesGC_8,"chrXVI","Sample_8_GC_XVI")
meanGC8 <- mean(ratioDistGC8)
medianGC8 <- median(ratioDistGC8)
stdevGC8 <- sd(ratioDistGC8)

### will need to do MA New lines separately because only 2 ancestor replicates
ratioDistributionsTrisomyMANew <- function(dds, genes, anChr,sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:5)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]

# cisHist <- hist(ratiosGC4.cis$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy,Cis Genes")
#
# transHist <- hist(ratiosGC4.Trans$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy, Trans Genes")
### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Trisomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic()
{pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
dev.off()}


trans <- qplot(ratios.Trans$sampleOverAnc,geom="histogram", breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)), main=paste("Trisomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=1.5, col="darkmagenta") + geom_vline(xintercept=0.67,col="green")+
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic()
{pdf(paste("transGenes",sample,".pdf",sep=""))
  print(trans)
dev.off()}
write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
return(ratios.Trans$sampleOverAnc)
}

# ratioDistributionsTrisomyMANew(ddsDefMAnewS152,genesMAnewS152,"chrI","Sample_152_MA_ChrI")
# ratioDistributionsTrisomyMANew(ddsDefMAnewS152,genesMAnewS152,"chrVIII","Sample_152_MA_ChrVIII")
ratioDistMA112 <- ratioDistributionsTrisomyMANew(ddsDefMAnewS112,genesMAnewS112,"chrXVI","Sample_112_MA_ChrXVI")
meanMA112 <- mean(ratioDistMA112)
medianMA112 <- median(ratioDistMA112)
stdevMA112 <- sd(ratioDistMA112)
ratioDistMA117 <- ratioDistributionsTrisomyMANew(ddsDefMAnewS117,genesMAnewS117,"chrV","Sample_117_MA_ChrV")
meanMA117 <- mean(ratioDistMA117)
medianMA117 <- median(ratioDistMA117)
stdevMA117 <- sd(ratioDistMA117)
ratioDistMA123 <- ratioDistributionsTrisomyMANew(ddsDefMAnewS123,genesMAnewS123,"chrV","Sample_123_MA_ChrV")
meanMA123 <- mean(ratioDistMA123)
medianMA123 <- median(ratioDistMA123)
stdevMA123 <- sd(ratioDistMA123)

## need monosomy one for MA new 29
ratioDistributionsMonosomyMANew <- function(dds, genes, anChr,sample) {
counts <- counts(dds,normalized=TRUE)
 ratios <- c()
 as.data.frame(ratios)
ratios$genes <- rownames(counts)
ratios$chr <- genes$TXCHROM
ratios$sampleMean <- rowMeans2(counts, cols = 1:3)
ratios$ancMean <- rowMeans2(counts,cols=4:5)
ratios$sampleOverAnc <- ratios$sampleMean/ratios$ancMean
ratios <- as.data.frame(ratios)

ratios.keepCis <- ratios$chr==anChr
ratios.cis <- ratios[ratios.keepCis,]

ratios.keepTrans <- ratios$chr!=anChr
ratios.Trans <- ratios[ratios.keepTrans,]

# cisHist <- hist(ratiosGC4.cis$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy,Cis Genes")
#
# transHist <- hist(ratiosGC4.Trans$sampleOverAnc,freq=TRUE,xlab="Ratio",xlim=range(0,10),main="Chrom V Trisomy, Trans Genes")
### using ggplot2

cis <- qplot(ratios.cis$sampleOverAnc,geom="histogram",breaks=c(seq(0,6,by=0.05),max(ratios.cis$sampleOverAnc)), main=paste("Monosomy",anChr,"Cis Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise"))+ geom_vline(xintercept = 1) + geom_vline(xintercept=0.5, col="green") + geom_vline(xintercept=1.5,col="darkmagenta") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.cis$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic()

{pdf(paste("cisGenes",sample,".pdf",sep=""))
  print(cis)
dev.off()}


trans <- qplot(ratios.Trans$sampleOverAnc,geom="histogram",breaks=c(seq(0,6,by=0.05),max(ratios.Trans$sampleOverAnc)),main=paste("Monosomy",anChr, "Trans Genes",sample,sep=" "), xlab = "Ratio", ylab="Number of Genes",fill=I("turquoise"),col=I("darkturquoise")) + geom_vline(xintercept = 1) + geom_vline(xintercept=0.5, col="green") + geom_vline(xintercept=1.5,col="darkmagenta") +
  coord_cartesian(xlim=c(0,6)) + geom_vline(aes(xintercept = mean(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "red", linetype ="longdash", size = .5) + geom_vline(aes(xintercept = median(ratios.Trans$sampleOverAnc, na.rm = T)),
             colour = "blue", linetype ="longdash", size = .5) + theme_classic()
{pdf(paste("transGenes",sample,".pdf",sep=""))
  print(trans)
dev.off()}
write.csv(ratios,file=paste("ratios",sample,".csv",sep=""))
return(ratios.Trans$sampleOverAnc)
}

ratioDistMA29 <- ratioDistributionsMonosomyMANew(ddsDefMAnewS29,genesMAnewS29,"chrIX","Sample_29_MA_ChrIX")
meanMA29 <- mean(ratioDistMA29)
medianMA29 <- median(ratioDistMA29)
stdevMA29 <- sd(ratioDistMA29)


#### using ggplot to make figures 


library(ggplot2)










```


Note that we could have specified the coefficient or contrast we want
to build a results table for, using either of the following equivalent
commands:

```{r results, eval=TRUE,include=TRUE}
###########################################
#without aneuploid chromosomes
# resGC <- results(ddsGCNoAneuInNorm)
# resGC
# resultsNames(ddsGCNoAneuInNorm)
# #euploid line
# resGC2 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_2","Holly_Anc"))
# summary(resGC2)
# sigGenesGC2 <- rownames(subset(resGC2,padj<0.1))
# #aneuploid line
# resultsNames(ddsGCNoAneuInNorm)
# resGC7 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_7","Holly_Anc"))#,cooksCutoff=FALSE)
# summary(resGC7)
# # testing with no indepdent filtering - does it change the results
# resGC7.test <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_7","Holly_Anc"), independentFiltering = FALSE)#,cooksCutoff=FALSE)
# summary(resGC7.test) # no difference
# # what about without cooksCutoff
# resGC7.test2 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_7","Holly_Anc"), independentFiltering = FALSE ,cooksCutoff=FALSE)
# summary(resGC7.test2) # no difference
#
# sigGenesGC7 <- rownames(subset(resGC7,padj<0.1))
# avgRatioGC7 <- mean(resGC7$log2FoldChange)
# avgRatioGC7
# resGC18 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_18","Holly_Anc"), independentFiltering=FALSE) #, cooksCutoff = FALSE)
# summary(resGC18)
# sigGenesGC18 <- rownames(subset(resGC18,padj<0.1))
# avgRatioGC18 <- mean(resGC18$log2FoldChange)
# resGC11 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_11","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC11)
# sigGenesGC11 <- rownames(subset(resGC11,padj<0.1))
# resGC1 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_1","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC1)
# avgRatioGC11 <- mean(resGC11$log2FoldChange)
# sigGenesGC1 <- rownames(subset(resGC1,padj<0.1))
# resGC3 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_3","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC3)
# sigGenesGC3 <- rownames(subset(resGC3,padj<0.1))
# resGC4 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_4","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC4)
# sigGenesGC4 <- rownames(subset(resGC4,padj<0.1))
# resGC49 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_49","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC49)
# sigGenesGC49 <- rownames(subset(resGC49,padj<0.1))
# resGC59 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_59","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC59)
# sigGenesGC59 <- rownames(subset(resGC59,padj<0.1))
# resGC61 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_61","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC61)
# sigGenesGC61 <- rownames(subset(resGC61,padj<0.1))
# resGC8 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_8","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC8)
# sigGenesGC8 <- rownames(subset(resGC8,padj<0.1))
# resGC9 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_9","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC9)
# sigGenesGC9 <- rownames(subset(resGC9,padj<0.1))
# #res69 <- results(ddsNoAneuInNorm, contrast=c("condition","Holly_69","Holly_Anc"))#, cooksCutoff = FALSE)
# #summary(res69)
# resGC76 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_76","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC76)
# sigGenesGC76 <- rownames(subset(resGC76,padj<0.1))
# resGC77 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_77","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC77)
# sigGenesGC77 <- rownames(subset(resGC77,padj<0.1))
#
# resGC5 <- results(ddsGCNoAneuInNorm, contrast=c("condition","Holly_5","Holly_Anc"))#, cooksCutoff = FALSE)
# summary(resGC5)
# sigGenesGC5 <- rownames(subset(resGC5,padj<0.1))

###########################################
#with aneuploid chromosomes
# resDefGC <- results(ddsDefGC)
# resDefGC
# resultsNames(ddsDefGC)
#euploid line
# resDefGC1 <- results(ddsDefGC, contrast=c("condition","Holly_1","Holly_Anc"))
# summary(resDefGC1)
# resDefGC2 <- results(ddsDefGC, contrast=c("condition","Holly_2","Holly_Anc"))
# summary(resDefGC2)
# resDefGC3 <- results(ddsDefGC, contrast=c("condition","Holly_3","Holly_Anc"))
# summary(resDefGC3)
# resDefGC4 <- results(ddsDefGC, contrast=c("condition","Holly_4","Holly_Anc"))
# summary(resDefGC4)
# resDefGC7 <- results(ddsDefGC, contrast=c("condition","Holly_7","Holly_Anc"))
# summary(resDefGC7)
#
#
# ###########################################
# #without aneuploid chromosomes
# resMAold <- results(ddsMAoldNoAneuInNorm)
# resMAold
# resultsNames(ddsMAoldNoAneuInNorm)
# #euploid line
# resMAold1 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC001","SCA"))
# summary(resMAold1)
# resMAold2 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC002","SCA"))
# summary(resMAold2)
# resMAold3 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC003","SCA"))
# summary(resMAold3)
# resMAold4 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC004","SCA"))
# summary(resMAold4)
# resMAold5 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC005","SCA"))
# summary(resMAold5)
# resMAold6 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC006","SCA"))
# summary(resMAold6)
# resMAold7 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC007","SCA"))
# summary(resMAold7)
# resMAold8 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC008","SCA"))
# summary(resMAold8)
# resMAold9 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC009","SCA"))
# summary(resMAold9)
# resMAold11 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC011","SCA"))
# summary(resMAold11)
# resMAold15 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC015","SCA"))
# summary(resMAold15)
# resMAold28 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC028","SCA"))
# summary(resMAold28)
# resMAold88 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC088","SCA"))
# summary(resMAold88)
# resMAold108 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC108","SCA"))
# summary(resMAold108)
# resMAold119 <- results(ddsMAoldNoAneuInNorm, contrast=c("condition","SC119","SCA"))
# summary(resMAold119)
#
# #############################################
# ### MA New no aneuploid chromosomes in normalization ####################
#
# resMAnew <- results(ddsMAnewNoAneuInNorm)
# resMAnew
# resultsNames(ddsMAnewNoAneuInNorm)
# #euploid line
# resMAnew29 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","29","Anc"))
# summary(resMAnew29)
# resMAnew50 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","50","Anc"))
# summary(resMAnew50)
# resMAnew112 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","112","Anc"))
# summary(resMAnew112)
# resMAnew115 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","115","Anc"))
# summary(resMAnew115)
# resMAnew117 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","117","Anc"))
# summary(resMAnew117)
# resMAnew123 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","123","Anc"))
# summary(resMAnew123)
# resMAnew141 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","141","Anc"))
# summary(resMAnew141)
# resMAnew152 <- results(ddsMAnewNoAneuInNorm, contrast=c("condition","152","Anc"))
# summary(resMAnew152)
#

### DESeq ran on samples individually
resMAnewS29 <- results(ddsDefMAnewS29)
resultsNames(ddsDefMAnewS29)
#euploid line
resMAnew29 <- results(ddsDefMAnewS29, contrast=c("condition","29","Anc"))
summary(resMAnew29)

resMAnew50 <- results(ddsDefMAnewS50, contrast=c("condition","50","Anc"))
summary(resMAnew50)

resMAnew112 <- results(ddsDefMAnewS112, contrast=c("condition","112","Anc"))
summary(resMAnew112)
resMAnew112$pvalue

resMAnew115 <- results(ddsDefMAnewS115, contrast=c("condition","115","Anc"))
summary(resMAnew115)

resMAnew117 <- results(ddsDefMAnewS117, contrast=c("condition","117","Anc"))
summary(resMAnew117)

resMAnew123 <- results(ddsDefMAnewS123, contrast=c("condition","123","Anc"))
summary(resMAnew123)

# resMAnew141 <- results(ddsDefMAnewS141, contrast=c("condition","141","Anc"))
# summary(resMAnew141)

resMAnew152 <- results(ddsDefMAnewS152, contrast=c("condition","152","Anc"))
summary(resMAnew152)

```

```{r common sig genes, eval=TRUE,include=TRUE}
## Big Problem: How many genes are differentially expressed in ALL samples?
# need to get results from DESeq into a more easily workable format (ie data frame)
  # make results from DESeq a data frame to work with
  resGC1Dat <- as.data.frame(resultsS1GC)
  resGC2Dat <- as.data.frame(resultsS2GC)
  resGC3Dat <- as.data.frame(resultsS3GC)
  resGC5Dat <- as.data.frame(resultsS5GC)
  resGC9Dat <- as.data.frame(resultsS9GC)
  resGC4Dat <- as.data.frame(resultsS4GC)
  resGC7Dat <- as.data.frame(resultsS7GC)
  resGC8Dat <- as.data.frame(resultsS8GC)
  resGC11Dat <- as.data.frame(resultsS11GC)
  resGC18Dat <- as.data.frame(resultsS18GC)
  resGC59Dat <- as.data.frame(resultsS59GC)
  resGC61Dat <- as.data.frame(resultsS61GC)
  resGC76Dat <- as.data.frame(resultsS76GC)
  resGC77Dat <- as.data.frame(resultsS77GC)
  # add gene names as first column ( which is currently rownames)
  resGC1Dat <- cbind(GENEID = rownames(resGC1Dat), resGC1Dat)
  resGC2Dat <- cbind(GENEID = rownames(resGC2Dat), resGC2Dat)
  resGC3Dat <- cbind(GENEID = rownames(resGC3Dat), resGC3Dat)
  resGC4Dat <- cbind(GENEID = rownames(resGC4Dat), resGC4Dat)
  resGC5Dat <- cbind(GENEID = rownames(resGC5Dat), resGC5Dat)
  resGC7Dat <- cbind(GENEID = rownames(resGC7Dat), resGC7Dat)
  resGC8Dat <- cbind(GENEID = rownames(resGC8Dat), resGC8Dat)
  resGC9Dat <- cbind(GENEID = rownames(resGC9Dat), resGC9Dat)
  resGC11Dat <- cbind(GENEID = rownames(resGC11Dat), resGC11Dat)
  resGC18Dat <- cbind(GENEID = rownames(resGC18Dat), resGC18Dat)
  # res49Dat <- cbind(GENEID = rownames(res49Dat), res49Dat)
  resGC59Dat <- cbind(GENEID = rownames(resGC59Dat), resGC59Dat)
  resGC61Dat <- cbind(GENEID = rownames(resGC61Dat), resGC61Dat)
  resGC76Dat <- cbind(GENEID = rownames(resGC76Dat), resGC76Dat)
  resGC77Dat <- cbind(GENEID = rownames(resGC77Dat), resGC77Dat)

  # need to make a column identifying the sample
    resGC1Dat$Sample <- "1"
    resGC2Dat$Sample <- "2"
    resGC3Dat$Sample <- "3"
    resGC4Dat$Sample <- "4"
    resGC5Dat$Sample <- "5"
    resGC7Dat$Sample <- "7"
    resGC8Dat$Sample <- "8"
    resGC9Dat$Sample <- "9"
    resGC11Dat$Sample <- "11"
    resGC18Dat$Sample <- "18"
    # res49Dat$Sample <- "49"
    resGC59Dat$Sample <- "59"
    resGC61Dat$Sample <- "61"
    resGC76Dat$Sample <- "76"
    resGC77Dat$Sample <- "77"

View(resGC7Dat)


      # put all sample expression data in the same data frame
      allGCSamplesExpr <- rbind(resGC1Dat, resGC2Dat, resGC3Dat, resGC4Dat, resGC5Dat, resGC7Dat, resGC8Dat, resGC9Dat, resGC11Dat, resGC18Dat, resGC59Dat, resGC61Dat, resGC76Dat, resGC77Dat)
      str(allGCSamplesExpr)
    # # # add chromosome numbers to this file
        matchGC <- intersect(allGCSamplesExpr$GENEID, genesGC_1$GENEID)
        str(matchGC)
    # #   # tell R which genes to keep
       keepGC <- genesGC_1$GENEID %in% matchGC
       str(keepGC)
    # #   # subset list of genes in analysis to include only the ones matching those found in          sample file
       genesGC <- genesGC_1[keepGC,]
       str(genesGC)
       View(genesGC)
       View(genesGC_1)
       dim(allGCSamplesExpr)
       dim(genesGC)
       str(genesGC_1)
       str(genesGC)
    # #   # add the correct chromosomes in
        allGCSamplesExpr <- cbind(genesGC_1$GENEID, genesGC_1$TXCHROM, allGCSamplesExpr)
        str(allGCSamplesExpr)
        View(allGCSamplesExpr)
       chr1.all <- subset(allGCSamplesExpr, allGCSamplesExpr$`genesNoAneu$GENEID`=="chrI")
       sample1 <- subset(allGCSamplesExpr, Sample==1)
       View(sample1)
       sample7 <- subset(allGCSamplesExpr, Sample==7)

    #    { plot(sample1$log2FoldChange,sample7$log2FoldChange)
    #    # want to draw two lines to indicate differential expression cutoffs
    #    # one for monosomy
    #    abline(-1, 1, col='purple', lty="dashed")
    #    #one for trisomy
    #     abline(0.583, 1, col="purple", lty="dashed")
    #     # # one for tetrasomy
    #     # abline(1, 1, col="purple", lty="dashed")
    #     # tell R to color dots above or below these lines certain colors
    #     above <- sample7$log2FoldChange-sample1$log2FoldChange > 1
    #     points(sample7$log2FoldChange[above], sample1$log2FoldChange[above], col="red", cex=0.5)
    #     below <- sample1$log2FoldChange-sample7$log2FoldChange < -1
    #     points(sample1$log2FoldChange[below], sample7$log2FoldChange[below], col="blue", cex=0.5) }
    #            
# want to find how many genes are differentially expressed in 2 samples, 3 samples, 4 samples, etc
#
  #     cbind(names = rownames(df), df)
  # # first need to filter allSamplesExpr based on p-adj level
    allGCSamplesDiffExpr <- subset(allGCSamplesExpr, padj<0.1)
    write.csv(allGCSamplesDiffExpr,file="allGCDiffExpr.csv")
  #   # then ask how many times each GENEID is found in allSamplesExpr
    library(plyr)
    geneGCFreq <- plyr::count(allGCSamplesDiffExpr, c("GENEID"))
      # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
      numGCGenesDE <- plyr::count(geneGCFreq,"freq")
        # rename columns so that I know what is going on: column 1 is how many samples a gene is DE in, column 2 is how many           genes are in that category
        colnames(numGCGenesDE) <- c("numSamples", "numGenes")
          # export as .csv to use in excel or whatever
          write.csv(numGCGenesDE, file="numb_comm_DE_genesALL_GC.csv")

##########################
          #### Same thing but for MA New Lines #####

  resMA29Dat <- as.data.frame(resMAnew29)
  resMA50Dat <- as.data.frame(resMAnew50)
  resMA112Dat <- as.data.frame(resMAnew112)
  resMA115Dat <- as.data.frame(resMAnew115)
  resMA117Dat <- as.data.frame(resMAnew117)
  resMA123Dat <- as.data.frame(resMAnew123)
  resMA152Dat <- as.data.frame(resMAnew152)

  # add gene names as first column ( which is currently rownames)
  resMA29Dat <- cbind(GENEID = rownames(resMA29Dat), resMA29Dat)
  resMA50Dat <- cbind(GENEID = rownames(resMA50Dat), resMA50Dat)
  resMA112Dat <- cbind(GENEID = rownames(resMA112Dat), resMA112Dat)
  resMA115Dat <- cbind(GENEID = rownames(resMA115Dat), resMA115Dat)
  resMA117Dat <- cbind(GENEID = rownames(resMA117Dat), resMA117Dat)
  resMA123Dat <- cbind(GENEID = rownames(resMA123Dat), resMA123Dat)
  resMA152Dat <- cbind(GENEID = rownames(resMA152Dat), resMA152Dat)
# need to make a column identifying the sample
    resMA29Dat$Sample <- "29"
    resMA50Dat$Sample <- "50"
    resMA112Dat$Sample <- "112"
    resMA115Dat$Sample <- "115"
    resMA117Dat$Sample <- "117"
    resMA123Dat$Sample <- "123"
    resMA152Dat$Sample <- "152"


View(resMA29Dat)

##### LEFT OFF HERE
## ** struggling to cbind genes with allMAnewSamplesExpr
## worked for GC samples and nothing has changed so why is it not working for these samples??


   allMAnewSamplesExpr <- rbind(resMA29Dat, resMA50Dat, resMA112Dat, resMA115Dat, resMA117Dat, resMA123Dat, resMA152Dat)
      str(allGCSamplesExpr)
    # # # add chromosome numbers to this file
        matchMAnew<- intersect(allMAnewSamplesExpr$GENEID, genesGC_1$GENEID)
        str(matchMAnew)
    # #   # tell R which genes to keep
       keepMAnew <- genesGC_1$GENEID %in% matchMAnew
    str(keepMAnew)
       # #   # subset list of genes in analysis to include only the ones matching those found in          sample file
       genesMAnew <- genesGC_1[keepMAnew,]
       str(genesMAnew)
       View(genesMAnew)
       View(genesGC_1)
       dim(allMAnewSamplesExpr)
       dim(genesMAnew)
       str(genesGC_1)
       str(genesGC)
    # #   # add the correct chromosomes in
        allMAnewSamplesExpr <- cbind(genesGC_1$GENEID, genesGC_1$TXCHROM, allMAnewSamplesExpr)
        str(allMAnewSamplesExpr)
        View(allMAnewSamplesExpr)
       chr1.all <- subset(allMAnewSamplesExpr, allMAnewSamplesExpr$`genesNoAneu$GENEID`=="chrI")
       sample29MAnew <- subset(allMAnewSamplesExpr, Sample==29)
       View(sample29MAnew)
       sample50MAnew <- subset(allMAnewSamplesExpr, Sample==50)

    #    { plot(sample1$log2FoldChange,sample7$log2FoldChange)
    #    # want to draw two lines to indicate differential expression cutoffs
    #    # one for monosomy
    #    abline(-1, 1, col='purple', lty="dashed")
    #    #one for trisomy
    #     abline(0.583, 1, col="purple", lty="dashed")
    #     # # one for tetrasomy
    #     # abline(1, 1, col="purple", lty="dashed")
    #     # tell R to color dots above or below these lines certain colors
    #     above <- sample7$log2FoldChange-sample1$log2FoldChange > 1
    #     points(sample7$log2FoldChange[above], sample1$log2FoldChange[above], col="red", cex=0.5)
    #     below <- sample1$log2FoldChange-sample7$log2FoldChange < -1
    #     points(sample1$log2FoldChange[below], sample7$log2FoldChange[below], col="blue", cex=0.5) }
    #            
# want to find how many genes are differentially expressed in 2 samples, 3 samples, 4 samples, etc
#
  #     cbind(names = rownames(df), df)
  # # first need to filter allSamplesExpr based on p-adj level
    allMAnewSamplesDiffExpr <- subset(allMAnewSamplesExpr, padj<0.1)
    write.csv(allMAnewSamplesDiffExpr,file="allMAnewDiffExpr.csv")
  #   # then ask how many times each GENEID is found in allSamplesExpr
    library(plyr)
    geneMAnewFreq <- plyr::count(allMAnewSamplesDiffExpr, c("GENEID"))
      # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
      numMAnewGenesDE <- plyr::count(geneMAnewFreq,"freq")
        # rename columns so that I know what is going on: column 1 is how many samples a gene is DE in, column 2 is how many           genes are in that category
        colnames(numMAnewGenesDE) <- c("numSamples", "numGenes")
          # export as .csv to use in excel or whatever
          write.csv(numMAnewGenesDE, file="numb_comm_DE_genesALL_MAnew.csv")
          ################## MA old Lines
  ########################################################################
  # make results from DESeq a data frame to work with
  resMAold1Dat <- as.data.frame(resultsS1MAold)
  resMAold2Dat <- as.data.frame(resultsS2MAold)
  resMAold3Dat <- as.data.frame(resultsS3MAold)
  resMAold4Dat <- as.data.frame(resultsS4MAold)
  resMAold5Dat <- as.data.frame(resultsS5MAold)
  resMAold7Dat <- as.data.frame(resultsS7MAold)
  resMAold8Dat <- as.data.frame(resultsS8MAold)
  resMAold9Dat <- as.data.frame(resultsS9MAold)
  resMAold15Dat <- as.data.frame(resultsS15MAold)
  resMAold28Dat <- as.data.frame(resultsS28MAold)
  resMAold88Dat <- as.data.frame(resultsS88MAold)
  resMAold108Dat <- as.data.frame(resultsS108MAold)
  resMAold119Dat <- as.data.frame(resultsS119MAold)
  # add gene names as first column ( which is currently rownames)
  resMAold1Dat <- cbind(GENEID = rownames(resMAold1Dat), resMAold1Dat)
  resMAold2Dat <- cbind(GENEID = rownames(resMAold2Dat), resMAold2Dat)
  resMAold3Dat <- cbind(GENEID = rownames(resMAold3Dat), resMAold3Dat)
  resMAold4Dat <- cbind(GENEID = rownames(resMAold4Dat), resMAold4Dat)
  resMAold5Dat <- cbind(GENEID = rownames(resMAold5Dat), resMAold5Dat)
  resMAold7Dat <- cbind(GENEID = rownames(resMAold7Dat), resMAold7Dat)
  resMAold8Dat <- cbind(GENEID = rownames(resMAold8Dat), resMAold8Dat)
  resMAold9Dat <- cbind(GENEID = rownames(resMAold9Dat), resMAold9Dat)
  resMAold15Dat <- cbind(GENEID = rownames(resMAold15Dat), resMAold15Dat)
  resMAold28Dat <- cbind(GENEID = rownames(resMAold28Dat), resMAold28Dat)
  # res49Dat <- cbind(GENEID = rownames(res49Dat), res49Dat)
  resMAold88Dat <- cbind(GENEID = rownames(resMAold88Dat), resMAold88Dat)
  resMAold108Dat <- cbind(GENEID = rownames(resMAold108Dat), resMAold108Dat)
  resMAold119Dat <- cbind(GENEID = rownames(resMAold119Dat), resMAold119Dat)


  # need to make a column identifying the sample
    resMAold1Dat$Sample <- "1"
    resMAold2Dat$Sample <- "2"
    resMAold3Dat$Sample <- "3"
    resMAold4Dat$Sample <- "4"
    resMAold5Dat$Sample <- "5"
    resMAold7Dat$Sample <- "7"
    resMAold8Dat$Sample <- "8"
    resMAold9Dat$Sample <- "9"
    resMAold15Dat$Sample <- "15"
    resMAold28Dat$Sample <- "28"
    resMAold88Dat$Sample <- "88"
    resMAold108Dat$Sample <- "108"
    resMAold119Dat$Sample <- "119"

      # put all sample expression data in the same data frame
allMAoldSamplesExpr <- rbind(resMAold1Dat, resMAold2Dat, resMAold3Dat, resMAold4Dat, resMAold5Dat, resMAold7Dat, resMAold8Dat, resMAold9Dat, resMAold15Dat, resMAold28Dat, resMAold88Dat, resMAold108Dat, resMAold119Dat)
      str(allMAoldSamplesExpr)
      View(allMAoldSamplesExpr)
    # # # add chromosome numbers to this file
        matchMAold <- intersect(allMAoldSamplesExpr$GENEID, genesGC_1$GENEID)
        str(matchMAold)

    # #   # tell R which genes to keep
       keepMAold <- genesGC_1$GENEID %in% matchMAold
       str(keepMAold)

    # #   # subset list of genes in analysis to include only the ones matching those found in          sample file
       genesMAold <- genesGC_1[keepMAold,]
       str(genesMAold)
       View(genesMAold)
       View(genesGC_1)
       dim(allMAoldSamplesExpr)
       dim(genesMAold)
       str(genesGC_1)
       str(genesMAold)
    # #   # add the correct chromosomes in
        allMAoldSamplesExpr <- cbind(genesMAold$GENEID, genesMAold$TXCHROM, allMAoldSamplesExpr)

        str(allGCSamplesExpr)
        View(allGCSamplesExpr)
       chr1.all <- subset(allGCSamplesExpr, allGCSamplesExpr$`genesNoAneu$GENEID`=="chrI")
       sample1 <- subset(allGCSamplesExpr, Sample==1)
       View(sample1)
       sample7 <- subset(allGCSamplesExpr, Sample==7)

    #    { plot(sample1$log2FoldChange,sample7$log2FoldChange)
    #    # want to draw two lines to indicate differential expression cutoffs
    #    # one for monosomy
    #    abline(-1, 1, col='purple', lty="dashed")
    #    #one for trisomy
    #     abline(0.583, 1, col="purple", lty="dashed")
    #     # # one for tetrasomy
    #     # abline(1, 1, col="purple", lty="dashed")
    #     # tell R to color dots above or below these lines certain colors
    #     above <- sample7$log2FoldChange-sample1$log2FoldChange > 1
    #     points(sample7$log2FoldChange[above], sample1$log2FoldChange[above], col="red", cex=0.5)
    #     below <- sample1$log2FoldChange-sample7$log2FoldChange < -1
    #     points(sample1$log2FoldChange[below], sample7$log2FoldChange[below], col="blue", cex=0.5) }
    #            
# want to find how many genes are differentially expressed in 2 samples, 3 samples, 4 samples, etc
#
  #     cbind(names = rownames(df), df)
  # # first need to filter allSamplesExpr based on p-adj level
    allGCSamplesDiffExpr <- subset(allGCSamplesExpr, padj<0.1)
    write.csv(allGCSamplesDiffExpr,file="allGCDiffExpr.csv")
  #   # then ask how many times each GENEID is found in allSamplesExpr
    library(plyr)
    geneGCFreq <- plyr::count(allGCSamplesDiffExpr, c("GENEID"))
      # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
      numGCGenesDE <- plyr::count(geneGCFreq,"freq")
        # rename columns so that I know what is going on: column 1 is how many samples a gene is DE in, column 2 is how many           genes are in that category
        colnames(numGCGenesDE) <- c("numSamples", "numGenes")
          # export as .csv to use in excel or whatever
          write.csv(numGCGenesDE, file="numb_comm_DE_genesALL_GC.csv")          


##############################################################################################          
# want to know how many genes are differentially expressed commonly in 1,2,3 etc EUPLOID lines
#
  # put all EUPLOID samples in same data frame
    eupGCSamplesExpr <- rbind(resGC1Dat, resGC2Dat, resGC3Dat, resGC5Dat, resGC9Dat)
      # first need to filter allSamplesExpr based on p-adj level
      eupGCSamplesDiffExpr <- subset(eupGCSamplesExpr, padj<0.1)
      #

        # then ask how many times each GENEID  is found in eupSamplesExpr
        library(plyr)
        geneGCFreqEu <- plyr::count(eupGCSamplesDiffExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numGCEupGenesDE <- plyr::count(geneGCFreqEu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numGCEupGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numGCEupGenesDE, file="numb_comm_DE__GC_genesEUP.csv")
##############################################################################################          
# want to know how many genes are differentially expressed commonly in 1,2,3 etc EUPLOID lines
#
  # put all EUPLOID samples in same data frame
    eupMAnewSamplesExpr <- rbind(resMA50Dat, resMA115Dat)
      # first need to filter allSamplesExpr based on p-adj level
      eupMAnewSamplesDiffExpr <- subset(eupMAnewSamplesExpr, padj<0.1)
      #

        # then ask how many times each GENEID  is found in eupSamplesExpr
        library(plyr)
        geneMAnewFreqEu <- plyr::count(eupMAnewSamplesDiffExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numMAnewEupGenesDE <- plyr::count(geneMAnewFreqEu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numMAnewEupGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numMAnewEupGenesDE, file="numb_comm_DE_MAnew_genesEUP.csv")              
##############################################################################################          
# want to know how many genes are differentially expressed commonly in 1,2,3 etc ANEUPLOID lines
#
  # put all ANEUPLOID samples in same data frame
    aneuGCSamplesExpr <- rbind(resGC4Dat, resGC7Dat, resGC8Dat,  resGC11Dat, resGC18Dat, resGC59Dat, resGC61Dat, resGC76Dat, resGC77Dat)
      # first need to filter allSamplesExpr based on p-adj level
      aneuGCSamplesDiffExpr <- subset(aneuGCSamplesExpr, padj<0.1)
      #
        # then ask how many times each GENEID  is found in eupSamplesExpr
        library(plyr)
        geneGCFreqAneu <- plyr::count(aneuGCSamplesDiffExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numGCAneuGenesDE <- plyr::count(geneGCFreqAneu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numGCAneuGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numGCAneuGenesDE, file="numb_comm_DE_genesANEU.csv")     

##############################################################################################          
# want to know how many genes are differentially expressed commonly in 1,2,3 etc ANEUPLOID lines
# with the SAME ANEUPLOID CHROMOSOME
  # put all samples in same data frame
    aneuChrIGCSamplesExpr <- rbind(resGC7Dat,  resGC11Dat, resGC18Dat)
      aneuChrItriGCExpr <- rbind(resGC7Dat, resGC18Dat)
      aneuChrVtriGCExpr <- rbind(resGC4Dat)
      aneuChrVIItriGCExpr <- rbind(resGC59Dat, resGC61Dat)
      aneuChrXIItriGCExpr <- rbind(resGC18Dat, resGC77Dat)
      aneuLine76GC <- resGC76Dat
      aneuLine76GC <- merge(genesGC,resGC76Dat,by="GENEID")
      write.csv(aneuLine76GC,file="aneuLine76GC.csv")
      aneuLine76GCdiffExpr <- subset(resGC76Dat,padj<0.1)
      aneuLine76GCdiffExpr <- merge(genesGC,aneuLine76GCdiffExpr,by.x="GENEID",by.y="GENEID")
      write.csv(aneuLine76GCdiffExpr,file="aneuLine76GCdiffExpr.csv")

      # first need to filter allSamplesExpr based on p-adj level
      aneuChrIGCSamplesDiffExpr <- subset(aneuChrIGCSamplesExpr, padj<0.1)
        triChrIGCSamplesDiffExpr <- subset(aneuChrItriGCExpr, padj<0.1)
        triChrVGCSamplesDiffExpr <- subset(aneuChrVtriGCExpr, padj<0.1)
        triChrVIIGCSamplesDiffExpr <- subset(aneuChrVIItriGCExpr, padj<0.1)
        triChrXIIGCSamplesDiffExpr <- subset(aneuChrXIItriGCExpr, padj<0.1)
      #
        # then ask how many times each GENEID  is found in eupSamplesExpr
        library(plyr)
        geneFreqGCAneuChrI <- plyr::count(aneuChrIGCSamplesDiffExpr, c("GENEID"))
          geneFreqGCtriChrI <- plyr::count(triChrIGCSamplesDiffExpr, c("GENEID"))
          geneFreqGCtriChrV <- plyr::count(triChrVGCSamplesDiffExpr, c("GENEID"))
          geneFreqGCtriChrVII <- plyr::count(triChrVIIGCSamplesDiffExpr, c("GENEID"))
          geneFreqtriGCChrXII <- plyr::count(triChrXIIGCSamplesDiffExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numGCAneuChrIGenesDE <- plyr::count(geneFreqGCAneuChrI,"freq")
            numtriGCChrIGenesDE <- plyr::count(geneFreqGCtriChrI,"freq")
            numtriGCChrVGenesDE <- plyr::count(geneFreqGCtriChrV,"freq")
            numtriGCChrVIIGenesDE <- plyr::count(geneFreqGCtriChrVII,"freq")
            numtriGCChrXIIGenesDE <- plyr::count(geneFreqtriGCChrXII,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numGCAneuChrIGenesDE) <- c("numSamples", "numGenes")
              colnames(numtriGCChrIGenesDE) <- c("numSamples", "numGenes")
              colnames(numtriGCChrVGenesDE) <- c("numSamples", "numGenes")
              colnames(numtriGCChrVIIGenesDE) <- c("numSamples", "numGenes")
              colnames(numtriGCChrXIIGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numGCAneuChrIGenesDE, file="numb_comm_DE_GC_genesANEUchrI.csv")  
                write.csv(numtriGCChrIGenesDE, file="numb_comm_DE_GC_genesTRIchrI.csv")
                write.csv(numtriGCChrVGenesDE, file="numb_comm_DE_GC_genesTRIchrV.csv")
                write.csv(numtriGCChrVIIGenesDE, file="numb_comm_DE_GC_genesTRIchrVII.csv")
                write.csv(numtriGCChrXIIGenesDE, file="numb_comm_DE_GC_genesTRIchrXII.csv")

# want to know how many genes are NOT DE on the aneuploid chromosome in aneuploid samples
#
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1
  # #
  #   sample4notDE <- subset(res4Dat, padj>0.1)
  #   #
  #   sample49notDE <- subset(res49Dat, padj>0.1)  
  #   #
  #   sample7notDE <- subset(res7Dat, padj>0.1)
  #   #
  #   sample11notDE <- subset(res11Dat, padj>0.1)
  #   #
  #   sample18notDE <- subset(res18Dat, padj>0.1)
  #   #
  #   sample59notDE <- subset(res59Dat, padj>0.1)
  #   # then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
  #   #
  #   # add column with chromosome numbers matching the GENEIDs
  #     matchedS4 <- intersect(sample4notDE$GENEID, genesNoAneu$GENEID)
  #     keep4 <- genesNoAneu$GENEID %in% matchedS4
  #     s4 <- genesNoAneu[keep4,]
  #     test4 <- cbind(s4$GENEID, s4$TXCHROM, sample4notDE)
  #     sample4 <- test4[,2:10]
  #     #
  #     matchedS49 <- intersect(sample49notDE$GENEID, genesNoAneu$GENEID)
  #     keep49 <- genesNoAneu$GENEID %in% matchedS49
  #     s49 <- genesNoAneu[keep49,]
  #     test49 <- cbind(s49$GENEID, s49$TXCHROM, sample49notDE)
  #     sample49 <- test49[,2:10]
  #     #
  #     matchedS7 <- intersect(sample7notDE$GENEID, genesNoAneu$GENEID)
  #     keep7 <- genesNoAneu$GENEID %in% matchedS7
  #     s7 <- genesNoAneu[keep7,]
  #     test7 <- cbind(s7$GENEID, s7$TXCHROM, sample7notDE)
  #     sample7 <- test7[,2:10]
  #     #
  #     matchedS11 <- intersect(sample11notDE$GENEID, genesNoAneu$GENEID)
  #     keep11 <- genesNoAneu$GENEID %in% matchedS11
  #     s11 <- genesNoAneu[keep11,]
  #     test11 <- cbind(s11$GENEID, s11$TXCHROM, sample11notDE)
  #     sample11 <- test11[,2:10]
  #     #
  #     matchedS18 <- intersect(sample18notDE$GENEID, genesNoAneu$GENEID)
  #     keep18 <- genesNoAneu$GENEID %in% matchedS18
  #     s18 <- genesNoAneu[keep18,]
  #     test18 <- cbind(s18$GENEID, s18$TXCHROM, sample18notDE)
  #     sample18 <- test18[,2:10]
  #     #
  #     matchedS59 <- intersect(sample59notDE$GENEID, genesNoAneu$GENEID)
  #     keep59 <- genesNoAneu$GENEID %in% matchedS59
  #     s59 <- genesNoAneu[keep59,]
  #     test59 <- cbind(s59$GENEID, s59$TXCHROM, sample59notDE)
  #     sample59 <- test59[,2:10]
  #     # add a column name to the chromosome column
  #     colnames(sample4)[1] <- "chr"
  #     #
  #     colnames(sample49)[1] <- "chr"
  #     #
  #     colnames(sample7)[1] <- "chr"
  #     #
  #     colnames(sample11)[1] <- "chr"
  #     #
  #     colnames(sample18)[1] <- "chr"
  #     #
  #     colnames(sample59)[1] <- "chr"
  #     # take out all chromosomes except the aneuploid one(s)
  #       sample4notDE <- subset(sample4, chr=="chrV")
  #       #
  #       sample49notDE <- subset(sample49, chr=="chrV")
  #       #
  #       sample7notDE <- subset(sample7, chr=="chrI")
  #       #
  #       sample11notDE <- subset(sample11, chr=="chrI")
  #       #
  #       sample18notDE <- subset(sample18, chr=="chrI")
  #       # count how many genes are not DE on this chrom in this sample
  #         countNotDeSample4 <- nrow(sample4notDE)
  #         #
  #         countNotDeSample49 <- nrow(sample49notDE)
  #         #
  #         countNotDeSample7 <- nrow(sample7notDE)
  #         #
  #         countNotDeSample11 <- nrow(sample11notDE)
  #         #
  #         countNotDeSample18 <- nrow(sample18notDE)
  #         # find how many genes are analyzed total on this chromosome
  #         chrVgenes <- subset(genesNoAneu, TXCHROM=="chrV")
  #         countChrV <- nrow(chrVgenes)
  #         #
  #         chrIgenes <- subset(genesNoAneu, TXCHROM=="chrI")
  #         countChrI <- nrow(chrIgenes)
  #         
  #           # what percent of genes on chromosome V are not DE in sample 4?
  #             percNotDeS4 = countNotDeSample4/countChrV * 100
  #             # 49% in sample 4
  #             #
  #             percNotDeS49 = countNotDeSample49/countChrV * 100
  #             # 68% in sample 49
  #             #
  #             percNotDeS7 = countNotDeSample7/countChrI * 100
  #             # 77% in sample 7 for chr I
  #             #
  #             percNotDeS11 = countNotDeSample11/countChrI * 100
  #             # 39% not differentially expressed in chr I for line 11
  #             #
  #             percNotDeS18 = countNotDeSample18/countChrI * 100
              # 65% not DE on chr I for line 18

#####################################################################################
# make  this into a function
# want to know how many genes are NOT DE on the aneuploid chromosome in aneuploid samples
# #
# sample = res4Dat
# chrom = "chrV"

getPercNonDEgenesGC <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
samplenotDE <- subset(sample, padj>0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(samplenotDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, samplenotDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        notDEAneuChr <- subset(samp, chr==chrom)
      # count how many genes are not DE on this chrom in this sample
          countNotDeSample <- nrow(notDEAneuChr)
         # find how many genes are analyzed total on this chromosome
          chrgenes <- subset(genesGC, TXCHROM==chrom)
          countChr <- nrow(chrgenes)     
            # what percent of genes on chromosome V are not DE in sample 4?
              percNotDeS <- setNames(data.frame(matrix(ncol = 3, nrow = 1)), c("Sample", "Chromosome", "percentNotDE"))
              percNotDeS$Sample <- sample[1,8]
              percNotDeS$Chromosome <- chrom
              percNotDeS$percentNotDE <- countNotDeSample/countChr * 100
              # percNotDeS <- c(sample[1,8], chrom, percNotDeS)
              # percNotDeS =
              # # 49% in sample 4
              # percNotDeS <- as.data.frame(percNotDeS)
              # colnames(percNotDeS) <- c("Sample","Chromosome","percentNotDE")
return(percNotDeS)
}

# test it out on one of the ones I did already
# try S4
percNonDEGCS4c5 <- getPercNonDEgenesGC(resGC4Dat, "chrV")
percNonDEGCS7c1 <- getPercNonDEgenesGC(resGC7Dat, "chrI")
percNonDEGCS8c16 <- getPercNonDEgenesGC(resGC8Dat, "chrXVI")
percNonDEGCS11c1 <- getPercNonDEgenesGC(resGC11Dat, "chrI")
percNonDEGCS11c15 <- getPercNonDEgenesGC(resGC11Dat, "chrXV")
percNonDEGCS18c1 <- getPercNonDEgenesGC(resGC18Dat, "chrI")
percNonDEGCS18c12 <- getPercNonDEgenesGC(resGC18Dat, "chrXII")
# percNonDEGCS49c5 <- getPercNonDEgenesGC(res49Dat, "chrV")
percNonDEGCS59c7 <- getPercNonDEgenesGC(resGC59Dat, "chrVII")
percNonDEGCS61c7 <- getPercNonDEgenesGC(resGC61Dat, "chrVII")
percNonDEGCS76c9 <- getPercNonDEgenesGC(resGC76Dat, "chrIX")
percNonDEGCS76c14 <- getPercNonDEgenesGC(resGC76Dat, "chrXIV")
percNonDEGCS77c12 <- getPercNonDEgenesGC(resGC77Dat, "chrXII")

percNonDEAllGCSamples <- rbind(percNonDEGCS4c5,percNonDEGCS7c1, percNonDEGCS8c16, percNonDEGCS11c1, percNonDEGCS11c15, percNonDEGCS18c1, percNonDEGCS18c12, percNonDEGCS59c7, percNonDEGCS61c7, percNonDEGCS76c9, percNonDEGCS76c14, percNonDEGCS77c12)

write.csv(percNonDEAllGCSamples, file="percNonDEAllgcLines.csv")

#########################################################################################

getNumNonDEGCgenes <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
samplenotDE <- subset(sample, padj>0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(samplenotDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, samplenotDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        notDEAneuChr <- subset(samp, chr==chrom)
      # count how many genes are not DE on this chrom in this sample
          countNotDeSample <- nrow(notDEAneuChr)
         # find how many genes are analyzed total on this chromosome
          chrgenes <- subset(genesGC, TXCHROM==chrom)
          countChr <- nrow(chrgenes)     
            # what percent of genes on chromosome V are not DE in sample 4?
              numNotDeS <- setNames(data.frame(matrix(ncol = 2, nrow = 1)), c("Sample", "Chromosome"))
              numNotDeS$Sample <- sample[1,8]
              numNotDeS$Chromosome <- chrom
              numNotDeS$countNotDe <- countNotDeSample
            numNotDeS$countChrom <- countChr
              # percNotDeS <- c(sample[1,8], chrom, percNotDeS)
              # percNotDeS =
              # # 49% in sample 4
              # percNotDeS <- as.data.frame(percNotDeS)
              # colnames(percNotDeS) <- c("Sample","Chromosome","percentNotDE")
return(numNotDeS)
}

numNonDEGCS4c5 <- getNumNonDEGCgenes(resGC4Dat, "chrV")
numNonDEGCS7c1 <- getNumNonDEGCgenes(resGC7Dat, "chrI")
numNonDEGCS8c16 <- getNumNonDEGCgenes(resGC8Dat, "chrXVI")
numNonDEGCS11c1 <- getNumNonDEGCgenes(resGC11Dat, "chrI")
numNonDEGCS11c15 <- getNumNonDEGCgenes(resGC11Dat, "chrXV")
numNonDEGCS18c1 <- getNumNonDEGCgenes(resGC18Dat, "chrI")
numNonDEGCS18c12 <- getNumNonDEGCgenes(resGC18Dat, "chrXII")
# numNonDEGCS49c5 <- getNumNonDEGCgenes(resGC49Dat, "chrV")
numNonDEGCS59c7 <- getNumNonDEGCgenes(resGC59Dat, "chrVII")
numNonDEGCS61c7 <- getNumNonDEGCgenes(resGC61Dat, "chrVII")
numNonDEGCS76c9 <- getNumNonDEGCgenes(resGC76Dat, "chrIX")
numNonDEGCS76c14 <- getNumNonDEGCgenes(resGC76Dat, "chrXIV")
numNonDEGCS77c12 <- getNumNonDEGCgenes(resGC77Dat, "chrXII")

numNonDEAllGCSamples <- rbind(numNonDEGCS4c5,numNonDEGCS7c1, numNonDEGCS8c16, numNonDEGCS11c1, numNonDEGCS11c15, numNonDEGCS18c1, numNonDEGCS18c12, numNonDEGCS59c7, numNonDEGCS61c7, numNonDEGCS76c9, numNonDEGCS76c14, numNonDEGCS77c12)

write.csv(numNonDEAllGCSamples, file="numNonDEAllgcLines.csv")







#######################################################################################
# want to know WHAT genes are NOT DE on the aneuploid chromosome in aneuploid samples
#
sample = resGC4Dat
chrom = "chrV"

getNonDEGCgenes <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
samplenotDE <- subset(sample, padj>0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(samplenotDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, samplenotDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        notDEAneuChr <- subset(samp, chr==chrom)
return(notDEAneuChr)
}

# test it out on one of the ones I did already
# try S4
nonDEGCS4c5 <- getNonDEGCgenes(resGC4Dat, "chrV")
write.csv(nonDEGCS4c5, file="nonDEGCS4c5.csv")
nonDEGCS7c1 <- getNonDEGCgenes(resGC7Dat, "chrI")
write.csv(nonDEGCS7c1, file="nonDEGCS7c1.csv")
nonDEGCS8c16 <- getNonDEGCgenes(resGC8Dat, "chrXVI")
write.csv(nonDEGCS8c16, file="nonDEGCS8c16.csv")
nonDEGCS11c1 <- getNonDEGCgenes(resGC11Dat, "chrI")
write.csv(nonDEGCS11c1, file="nonDEGCS11c1.csv")
nonDEGCS11c15 <- getNonDEGCgenes(resGC11Dat, "chrXV")
write.csv(nonDEGCS11c15, file="nonDEGCS11c15.csv")
nonDEGCS18c1 <- getNonDEGCgenes(resGC18Dat, "chrI")
write.csv(nonDEGCS18c1, file="nonDEGCS18c1.csv")
nonDEGCS18c12 <- getNonDEGCgenes(resGC18Dat, "chrXII")
write.csv(nonDEGCS18c12, file="nonDEGCS18c12.csv")
# nonDEGCS49c5 <- getNonDEGCgenes(resGC49Dat, "chrV")
# write.csv(nonDES49c5, file="nonDES49c5.csv")
nonDEGCS59c7 <- getNonDEGCgenes(resGC59Dat, "chrVII")
write.csv(nonDEGCS59c7, file="nonDEGCS59c7.csv")
nonDEGCS61c7 <- getNonDEGCgenes(resGC61Dat, "chrVII")
write.csv(nonDEGCS61c7, file="nonDEGCS61c7.csv")
nonDEGCS76c9 <- getNonDEGCgenes(resGC76Dat, "chrIX")
write.csv(nonDEGCS76c9, file="nonDEGCS76c9.csv")
nonDEGCS76c14 <- getNonDEGCgenes(resGC76Dat, "chrXIV")
write.csv(nonDEGCS76c14, file="nonDEGCS76c14.csv")
nonDEGCS77c12 <- getNonDEGCgenes(resGC77Dat, "chrXII")
write.csv(nonDEGCS77c12, file="nonDEGCS77c12.csv")

############################################################################
# want to know how many genes are DE on the aneuploid chromosome in aneuploid samples
# #
# sample = res4Dat
# chrom = "chrV"

getPercDEGCgenes <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        DEAneuChr <- subset(samp, chr==chrom)
      # count how many genes are DE on this chrom in this sample
          countDeSample <- nrow(DEAneuChr)
         # find how many genes are analyzed total on this chromosome
          chrgenes <- subset(genesGC, TXCHROM==chrom)
          countChr <- nrow(chrgenes)     
            # what percent of genes on chromosome V are DE in sample 4?
              # percDeS = countDeSample/countChr * 100
              # # 49% in sample 4
              # percDeS <- c(sample[1,8], chrom, percDeS)
              # percDeS <- as.data.frame(percDeS)
              # colnames(percDeS) <- c("Sample","Chromosome","percentNotDE")
              percDeS <- setNames(data.frame(matrix(ncol = 3, nrow = 1)), c("Sample", "Chromosome", "percentNotDE"))
              percDeS$Sample <- sample[1,8]
              percDeS$Chromosome <- chrom
              percDeS$percentNotDE <- countDeSample/countChr * 100

return(percDeS)
}

# test it out on one of the ones I did already
# try S4
percDEGCS4c5 <- getPercDEGCgenes(resGC4Dat, "chrV")
percDEGCS7c1 <- getPercDEGCgenes(resGC7Dat, "chrI")
percDEGCS8c16 <- getPercDEGCgenes(resGC8Dat, "chrXVI")
percDEGCS11c1 <- getPercDEGCgenes(resGC11Dat, "chrI")
percDEGCS11c15 <- getPercDEGCgenes(resGC11Dat, "chrXV")
percDEGCS18c1 <- getPercDEGCgenes(resGC18Dat, "chrI")
percDEGCS18c12 <- getPercDEGCgenes(resGC18Dat, "chrXII")
# percDEGCS49c5 <- getPercDEGCgenes(resGC49Dat, "chrV")
percDEGCS59c7 <- getPercDEGCgenes(resGC59Dat, "chrVII")
percDEGCS61c7 <- getPercDEGCgenes(resGC61Dat, "chrVII")
percDEGCS76c9 <- getPercDEGCgenes(resGC76Dat, "chrIX")
percDEGCS76c14 <- getPercDEGCgenes(resGC76Dat, "chrXIV")
percDEGCS77c12 <- getPercDEGCgenes(resGC77Dat, "chrXII")

percDEAllGCSamples <- rbind(percDEGCS4c5,percDEGCS7c1, percDEGCS8c16, percDEGCS11c1, percDEGCS11c15, percDEGCS18c1, percDEGCS18c12, percDEGCS59c7, percDEGCS61c7, percDEGCS76c9, percDEGCS76c14, percDEGCS77c12)

write.csv(percDEAllGCSamples, file="percDEAllgcLines.csv")


###################################################################

#######################################################################################
# want to know WHAT genes are DE on the aneuploid chromosome in aneuploid samples
#
# sample = res4Dat
# chrom = "chrV"
getDEGCgenes <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        DEAneuChr <- subset(samp, chr==chrom)
return(DEAneuChr)
}


# test it out on one of the ones I did already
# try S4
DEGCS4.c5 <- getDEGCgenes(resGC4Dat, "chrV")


write.csv(DEGCS4.c5, file="DEGCS4c5.csv")
# View(geneIDs4)
geneIDsGC4 <- DEGCS4.c5[,2]
str(geneIDsGC4)
# geneIDs4 <- geneIDs4$GENEID
geneIDsGC4 <- as.vector(geneIDsGC4)
write.table(geneIDsGC4, file="DEgeneIDs4GC.txt")

DESGC7.c1 <- getDEGCgenes(resGC7Dat, "chrI")
write.csv(DESGC7.c1, file="DESGC7c1.csv")

DESGC8.c16 <- getDEGCgenes(resGC8Dat, "chrXVI")
write.csv(DESGC8.c16, file="DESGC8c16.csv")

DESGC11.c1 <- getDEGCgenes(resGC11Dat, "chrI")
write.csv(DESGC11.c1, file="DESGC11c1.csv")

DESGC11.c15 <- getDEGCgenes(resGC11Dat, "chrXV")
write.csv(DESGC11.c15, file="DESGC11c15.csv")

DESGC18.c1 <- getDEGCgenes(resGC18Dat, "chrI")
write.csv(DESGC18.c1, file="DESGC18c1.csv")

DESGC18.c12 <- getDEGCgenes(resGC18Dat, "chrXII")
write.csv(DESGC18.c12, file="DESGC18c12.csv")

# DESGC49.c5 <- getDEGCgenes(resGC49Dat, "chrV")
# write.csv(DES49.c5, file="DES18c5.csv")

DESGC59.c7 <- getDEGCgenes(resGC59Dat, "chrVII")
write.csv(DESGC59.c7, file="DESGC59c7.csv")

DESGC61.c7 <- getDEGCgenes(resGC61Dat, "chrVII")
write.csv(DESGC61.c7, file="DESGC61c7.csv")

DESGC76.c9 <- getDEGCgenes(resGC76Dat, "chrIX")
write.csv(DESGC76.c9, file="DESGC76c9.csv")

DESGC76.c14 <- getDEGCgenes(resGC76Dat, "chrXIV")
write.csv(DESGC76.c14, file="DESGC76c14.csv")

DESGC77.c12 <- getDEGCgenes(resGC77Dat, "chrXII")
write.csv(DESGC77.c12, file="DESGC77c12.csv")

################################################################
# want to get DE genes in aneuploid lines that are NOT located on the aneuploid chromosome
# sample = resGC4Dat
# chrom= "chrV"

getDEgenesGCElsewhere <- function(sample,chrom) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        DENotAneuChr <- subset(samp, chr!=chrom)
return(DENotAneuChr)
}

deGenesNotAneuGCS4 <- getDEgenesGCElsewhere(resGC4Dat, chrom="chrV")
deGenesNotAneuGCS7 <- getDEgenesGCElsewhere(resGC7Dat, chrom="chrI")
deGenesNotAneuGCS8 <- getDEgenesGCElsewhere(resGC8Dat, chrom="chrXVI")
deGenesNotAneuGCS18 <- getDEgenesGCElsewhere(resGC18Dat, chrom=c("chrXII","chrI"))
# deGenesNotAneuGCS49 <- getDEgenesGCElsewhere(resGC49Dat, chrom="chrV")
deGenesNotAneuGCS59 <- getDEgenesGCElsewhere(resGC59Dat, chrom="chrVII")
deGenesNotAneuGCS61 <- getDEgenesGCElsewhere(resGC61Dat, chrom="chrVII")
deGenesNotAneuGCS76 <- getDEgenesGCElsewhere(resGC76Dat, chrom=c("chrIX","XIV"))
deGenesNotAneuGCS77 <- getDEgenesGCElsewhere(resGC77Dat, chrom="chrXII")
deGenesNotAneuGCS11 <- getDEgenesGCElsewhere(resGC11Dat, chrom=c("chrI","chrXV"))

genesGC1DE <- subset(allGCSamplesDiffExpr, Sample=="1")
        write.table(genesGC1DE, file="genesGC1DE.txt")
        genesGC2DE <- subset(allGCSamplesDiffExpr, Sample=="2")
        write.table(genesGC2DE, file="genesGC2DE.txt")
        genesGC3DE <- subset(allGCSamplesDiffExpr, Sample=="3")
        write.table(genesGC3DE, file="genesGC3DE.txt")
        genesGC4DE <- subset(allGCSamplesDiffExpr, Sample=="4")
        write.table(genesGC4DE, file="genesGC4DE.txt")
        genesGC5DE <- subset(allGCSamplesDiffExpr, Sample=="5")
        write.table(genesGC5DE, file="genesGC5DE.txt")
        genesGC7DE <- subset(allGCSamplesDiffExpr, Sample=="7")
        write.table(genesGC7DE, file="genesGC7DE.txt")
        genesGC8DE <- subset(allGCSamplesDiffExpr, Sample=="8")
        write.table(genesGC8DE, file="genesGC8DE.txt")
        genesGC9DE <- subset(allGCSamplesDiffExpr, Sample=="9")
        write.table(genesGC9DE, file="genesGC9DE.txt")
        genesGC11DE <- subset(allGCSamplesDiffExpr, Sample=="11")
        write.table(genesGC11DE, file="genesGC11DE.txt")
        genesGC18DE <- subset(allGCSamplesDiffExpr, Sample=="18")
        write.table(genesGC18DE, file="genesGC18DE.txt")
        # genes49DE <- subset(allSamplesDiffExpr, Sample=="49")
        # write.table(genes49DE, file="genes49DE.txt")
        genesGC59DE <- subset(allGCSamplesDiffExpr, Sample=="59")
        write.table(genesGC59DE, file="genesGC59DE.txt")
        genesGC61DE <- subset(allGCSamplesDiffExpr, Sample=="61")
        write.table(genesGC61DE, file="genesGC61DE.txt")
        genesGC76DE <- subset(allGCSamplesDiffExpr, Sample=="76")
        write.table(genesGC76DE, file="genesGC76DE.txt")
        genesGC77DE <- subset(allGCSamplesDiffExpr, Sample=="77")
        write.table(genesGC77DE, file="genesGC77DE.txt")
      # trying different things to make a venn diagram
        library(VennDiagram)
          # pdf("venn_diagram.pdf")
          # venn.plot <- venn.diagram(list(genes1DE$GENEID, genes2DE$GENEID,genes3DE$GENEID,genes4DE$GENEID, genes5DE$GENEID, genes7DE$GENEID, genes8DE$GENEID, genes9DE$GENEID, genes11DE$GENEID, genes18DE$GENEID, genes49DE$GENEID, genes59DE$GENEID, genes61DE$GENEID, genes76DE$GENEID, genes77DE$GENEID), NULL, fill=c("red", "green","blue","pink","orange","turquoise","yellow","purple","lightblue","lightyellow","lightred","lightgreen","lightpink","lightorange"), alpha=c(0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5), cex = 2, cat.fontface=3, category.names=c("Sample1", "Sample2","Sample3", "Sample4","Sample5","Sample7","Sample8","Sample9","Sample11","Sample18","Sample49","Sample59","Sample61","Sample76","Sample77"))
          # grid.draw(venn.plot)
          # dev.off()


venn.plotGCAneu <- venn.diagram(
	x = list(
		A = deGenesNotAneuGCS4$GENEID,
		B = deGenesNotAneuGCS7$GENEID,
		C = deGenesNotAneuGCS8$GENEID,
		D = deGenesNotAneuGCS59$GENEID,
		E = deGenesNotAneuGCS77$GENEID
		),
	filename = NULL, #"euploid_commonDE.tiff"
	col = "black",
	fill = c("lightblue", "goldenrod1", "lavender", "turquoise", "lightpink"),
	category.names = c("Chr V 3n","Chr I 3n","Chr XVI 4n","Chr VII 3n","Chr XII 3n"),
	alpha = 0.50,
	cex = c(1.5, 1.5, 1.5, 1.5, 1.5, 1, 0.8, 1, 0.8, 1, 0.8, 1, 0.8,
	 1, 0.8, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 1, 1, 1, 1, 1.5),
	cat.col = c("mediumblue", "darkgoldenrod", "purple", "darkturquoise", "deeppink"),
	cat.cex = 1,
	cat.default.pos ="outer",
	cat.fontface = "bold",
	main="Common DE Genes Not on Aneuploid Chromosome in Aneuploid Samples",
	main.fontface = "bold",
	main.cex = 1,
	margin = 0.05
	)

grid.draw(venn.plotGCAneu)

# venn.plotGCChrV <- venn.diagram(
# 	x = list(
# 		A = deGenesNotAneuGCS4$GENEID,
# 		B = deGenesNotAneuGCS49$GENEID
# 		),
# 	filename = NULL, #"euploid_commonDE.tiff",
# 	fill = c("turquoise", "lightpink")
# 	)
#
# pdf("commonDE_chrV.pdf")
# grid.draw(venn.plotChrV)

pdf("aneuploid_commonDE_GC.pdf")
grid.draw(venn.plotGCAneu)
dev.off()
# aneuLines <- list(genes4DE, genes7DE, genes8DE, genes18DE, genes49DE)
# want to get list of common DE genes in all the euploid lines
#commonDEaneu <- Reduce(intersect, Map("[[", aneuLines, "GENEID"))
#Reduc
#commonDEaneu <- as.vector(commonDEaneu)
#write.csv(commonDEaneu, file="commonDEaneu.csv")

venn.plotGCChrI <- venn.diagram(
	x = list(
		A = deGenesNotAneuGCS7$GENEID,
		B = deGenesNotAneuGCS18$GENEID,
		C = deGenesNotAneuGCS11$GENEID
		),
	filename = NULL,
	fill = c("orange", "deeppink1","turquoise"),
	cex=2,
	cat.col = c("orange", "deeppink1", "darkturquoise"),
	category.names = c("Sample 7 (3n)","Sample 18 (3n)","Sample 11 (1n)"),
	cat.cex = 1,
	cat.default.pos ="outer",
	cat.fontface = "bold",
	main= "Common DE Genes Not on Chromosome 1 in Aneuploid Samples",
	main.fontface = "bold",
	main.cex = 1.5,
	margin = 0.05
	)

pdf("commonGC.DE_chrI.pdf")
grid.draw(venn.plotGCChrI)

venn.plotGCChrXII <- venn.diagram(
	x = list(
		A = deGenesNotAneuGCS18$GENEID,
		B = deGenesNotAneuGCS77$GENEID
		),
	filename = NULL,
	fill = c("orange", "deeppink1"),
	category.names = c("Sample 18 (3n)","Sample 77 (3n)"),
	cat.cex = 1,
	cat.col = c("orange","deeppink"),
	main= "Common DE Genes Not on Chrom XII in Aneuploid Samples ",
	main.fontface = "bold",
	main.cex = 1.5,
	margin=0.05
	)

pdf("commonGC.DE_chrVXII.pdf")
grid.draw(venn.plotGCChrXII)

venn.plotGCChrVII <- venn.diagram(
	x = list(
		A = deGenesNotAneuGCS59$GENEID,
		B = deGenesNotAneuGCS61$GENEID
		),
	filename = NULL,
	fill = c("turquoise", "purple"),
	category.names = c("Sample 59 (3n)","Sample 61 (3n)"),
	cat.cex = 1,
	cat.col=c("darkturquoise","purple"),
	main="Common DE Genes Not on Chrom VII in Aneuploid Samples",
	main.fontface = "bold",
	main.cex = 1.5,
	margin=0.05
	)

pdf("commonGC.DE_chrVII.pdf")
grid.draw(venn.plotGCChrVII)


############################################################################
# want to know what genes are DE in all samples
# getDEgenes <- function(sample) {              
#   # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   
# allSampleDE <- subset(sample, padj<0.1)
# # then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
#     #
#     # add column with chromosome numbers matching the GENEIDs
#       # find matches of GENEID in sample file to GENEID in list of genes included in analysis
#       matched <- intersect(sampleDE$GENEID, genesNoAneu$GENEID)
#       # tell R which genes to keep
#       keep <- genesNoAneu$GENEID %in% matched
#       # subset list of genes in analysis to include only the ones matching those found in          sample file
#       s <- genesNoAneu[keep,]
#       # use c(olumn)bind to stick the chromosome #s to the sample analysis file
#       test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
#       # subset to exclude first column (duplicate)
#       samp <- test[,2:10]
#       # add a column name to the chromosome column
#       colnames(samp)[1] <- "chr"
# return(deAll)
# }



############################################################################
# want to know how many genes are DE on the aneuploid chromosome in multiple aneuploid samples sharing the same aneuploid
#
# chr I
  # lines 7,11,and 18
  # deChrI <-

#######################################################################################
# want to know WHAT genes are DE on the aneuploid chromosome in aneuploid samples
#
# sample = res4Dat
#chrom = "chrV"

getDEgenesGCAll <- function(sample) {              
  # first need to subset the data frame of gene expression for each aneuploid sample to include those genes that have a p-value above 0.1   

sampleDE <- subset(sample, padj<0.1)
# then subset the data frame of genes with a p-value above 0.1 to only include genes that are located on the aneuploid chromosome
    #
    # add column with chromosome numbers matching the GENEIDs
      # find matches of GENEID in sample file to GENEID in list of genes included in analysis
      matched <- intersect(sampleDE$GENEID, genesGC$GENEID)
      # tell R which genes to keep
      keep <- genesGC$GENEID %in% matched
      # subset list of genes in analysis to include only the ones matching those found in          sample file
      s <- genesGC[keep,]
      # use c(olumn)bind to stick the chromosome #s to the sample analysis file
      test <- cbind(s$GENEID, s$TXCHROM, sampleDE)
      # subset to exclude first column (duplicate)
      samp <- test[,2:10]
      # add a column name to the chromosome column
      colnames(samp)[1] <- "chr"
      # take out all chromosomes except the aneuploid one(s)
      # chrom is designated when you call the function
        DEAll  <- samp
return(DEAll)
}

# test it out on one of the ones I did already
# try S4
DEGCS1all <- getDEgenesGCAll(resGC1Dat)
DEGCS2all <- getDEgenesGCAll(resGC2Dat)
DEGCS3all <- getDEgenesGCAll(resGC3Dat)
DEGCS4all <- getDEgenesGCAll(resGC4Dat)
DEGCS5all <- getDEgenesGCAll(resGC5Dat)
DEGCS7all <- getDEgenesGCAll(resGC7Dat)
DEGCS8all <- getDEgenesGCAll(resGC8Dat)
DEGCS9all <- getDEgenesGCAll(resGC9Dat)
DEGCS11all <- getDEgenesGCAll(resGC11Dat)
DEGCS18all <- getDEgenesGCAll(resGC18Dat)
DEGCS59all <- getDEgenesGCAll(resGC59Dat)
DEGCS61all <- getDEgenesGCAll(resGC61Dat)
DEGCS76all <- getDEgenesGCAll(resGC76Dat)
DEGCS77all <- getDEgenesGCAll(resGC77Dat)

########################################################################################
# what DE genes are shared between samples aneuploid with the same chromosome?
  # use data frames containing only DE genes in the given samples
# sample1 = genes4DE
# sample2 = genes49DE
library(dplyr)

getCommonDEGCgenes <- function(sample1, sample2) {
  # join them together by GENEID column
  chr.all <- inner_join(sample1, sample2, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matched <- intersect(chr.all$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepAll <- chr.all$GENEID %in% matched
          # only keep those genes in the data frame
          chr.all <- chr.all[keepAll,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenAll <- genesGC$GENEID %in% matched
              # put those genes in a vector
              GenAll <- genesGC[keepGenAll,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chr.all <- cbind(GenAll$TXCHROM, chr.all)
                return(chr.all)
}

# chrVAll <- getCommonDEGCgenes(genesGC4DE, genes49DE)
# write.csv(chrVAll, file="chrVallDE_GC.csv")
chrVIIAllGC <- getCommonDEGCgenes(genesGC59DE, genesGC61DE)
write.csv(chrVIIAllGC, file="chrVIIallDE_GC.csv")
chrXIIAllGC <- getCommonDEGCgenes(genesGC18DE, genesGC77DE)
write.csv(chrXIIAllGC, file="chrXIIallDE_GC.csv")

## only trisomic samples
    # # join them together by GENEID column
    # chrI.tri <- inner_join(genes7DE,genes18DE, by="GENEID")
    #   # match the genes to the genes in genes table that includes chromosome number
    #   matchedI <- intersect(chrI.tri$GENEID, genesNoAneu$GENEID)
    #     # tell R which genes to keep from the chrI.tri dataframe
    #     keepI <- chrI.tri$GENEID %in% matchedI
    #       # only keep those genes in the data frame
    #       chrI.tri <- chrI.tri[keepI,]
    #         # tell R which genes to keep from the genesNoAneu dataframe
    #         keepGen <- genesNoAneu$GENEID %in% matchedI
    #           # put those genes in a vector
    #           Gen <- genesNoAneu[keepGen,]
    #             # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
    #             chrI.tri <- cbind(Gen$TXCHROM, chrI.tri)
    #               # save as .csv file
    #               write.csv(chrI.tri, file="chrITriDE.csv")

#for chromosome I, which has 3 lines that are commonly aneuploid for it, so the function won't work properly
#chrom I, all aneuploids (both trisomics plus the monosomic)
# chrI.all <- inner_join(chrI.tri, genes11DE, by="GENEID")
#   # match the genes to the genes in genes table that includes chromosome number
#   matchedIall <- intersect(chrI.all$GENEID, genesNoAneu$GENEID)
#     # tell R which genes to keep from the chrI.tri dataframe
#     keepIall <- chrI.all$GENEID %in% matchedI
#       # only keep those genes in the data frame
#       chrI.all <- chrI.all[keepIall,]
#         # tell R which genes to keep from the genesNoAneu dataframe
#         keepGenAll <- genesNoAneu$GENEID %in% matchedIall
#           # put those genes in a vector
#           GenAll <- genesNoAneu[keepGenAll,]
#             # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
#             chrI.all <- cbind(GenAll$TXCHROM, chrI.all)
#               # save as .csv file
#               write.csv(chrI.all, file="chrIAllDE.csv")

########################################################################################
# what NON-DE genes are shared between samples aneuploid with the same chromosome?
  # use data frames containing only NON-DE genes in the given samples
 # sample1 = nonDES4c5
 # sample2 = nonDES49c5
 #
getCommonNonDEGCgenes <- function(sample1, sample2) {
  # join them together by GENEID column
  chr.all <- inner_join(sample1, sample2, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matched <- intersect(chr.all$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepAll <- chr.all$GENEID %in% matched
          # only keep those genes in the data frame
          chr.all <- chr.all[keepAll,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenAll <- genesGC$GENEID %in% matched
              # put those genes in a vector
              GenAll <- genesGC[keepGenAll,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chr.all <- cbind(GenAll$TXCHROM, chr.all)
                return(chr.all)
}

# chrVNon <- getCommonNonDEgenes(nonDES4c5, nonDES49c5)
# write.csv(chrVNon, file="chrVallDNonE_GC.csv")
chrVIIGCNon <- getCommonNonDEGCgenes(nonDEGCS59c7, nonDEGCS61c7)
write.csv(chrVIIGCNon, file="chrVIIallNonDE_GC.csv")
chrXIIGCNon <- getCommonNonDEGCgenes(nonDEGCS18c12, nonDEGCS77c12)
write.csv(chrXIIGCNon, file="chrXIIallNonDE_GC.csv")

## only trisomic samples
    # join them together by GENEID column
    chrItriGCNon <- inner_join(nonDEGCS7c1,nonDEGCS18c1, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matchedIGCNon <- intersect(chrItriGCNon$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepIGCNon <- chrItriGCNon$GENEID %in% matchedIGCNon
          # only keep those genes in the data frame
          chrItriGCNon <- chrItriGCNon[keepIGCNon,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenGCNon <- genesGC$GENEID %in% matchedIGCNon
              # put those genes in a vector
              GenNonGC <- genesGC[keepGenGCNon,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chrItriGCNon <- cbind(GenNonGC$TXCHROM, chrItriGCNon)
                  # save as .csv file
                  write.csv(chrItriGCNon, file="chrITriNonDE_GC.csv")
#for chromosome I, which has 3 lines that are commonly aneuploid for it, so the function won't work properly
  # chrom I, all aneuploids (both trisomics plus the monosomic)
    chrI.NonGC <- inner_join(chrItriGCNon, nonDEGCS11c1, by="GENEID")
      # match the genes to the genes in genes table that includes chromosome number
      matchedIallGCNon <- intersect(chrI.NonGC$GENEID, genesGC$GENEID)
        # tell R which genes to keep from the chrI.tri dataframe
        keepIallGCNon <- chrI.NonGC$GENEID %in% matchedIallGCNon
          # only keep those genes in the data frame
          chrI.NonGC <- chrI.NonGC[keepIallGCNon,]
            # tell R which genes to keep from the genesNoAneu dataframe
            keepGenAllNonGC <- genesGC$GENEID %in% matchedIallGCNon
              # put those genes in a vector
              GenAllNonGC <- genesGC[keepGenAllNonGC,]
                # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
                chrI.NonGC <- cbind(GenAllNonGC$TXCHROM, chrI.NonGC)
                  # save as .csv file
                  write.csv(chrI.NonGC, file="chrIAllNonDE_GC.csv")

{
  venn.plotChrINonDEGC <- venn.diagram(
	x = list(
		"L11" = nonDEGCS11c1$GENEID,
		"L18" =nonDEGCS18c1$GENEID,
		"L7" = nonDEGCS7c1$GENEID
		),
	filename = NULL,
	fill = c("orange", "deeppink1","turquoise"),
	cex=2
	)
pdf("commonNonDE_chrI_GC.pdf")
grid.draw(venn.plotChrINonDEGC)
dev.off()
}


{venn.plotChrINonDETrisGC <- venn.diagram(
	x = list(
		"L18" =nonDEGCS18c1$GENEID,
		"L7" = nonDEGCS7c1$GENEID
		),
	filename = NULL,
	fill = c( "deeppink1","turquoise"),
	cex=2
	)
pdf("commonNonDE_chrITri_GC.pdf")
grid.draw(venn.plotChrINonDETrisGC)
dev.off()}

# venn.plotChrVNonDE <- venn.diagram(
# 	x = list(
# 		"L4" =nonDES4c5$GENEID,
# 		"L49" = nonDES49c5$GENEID
# 		),
# 	filename = NULL,
# 	fill = c( "goldenrod","turquoise"),
# 	cex=2
# 	)
# pdf("commonNonDE_chrVTri.pdf")
# grid.draw(venn.plotChrVNonDE)
# dev.off()

# then compare how many of these genes are shared between samples aneuploid for the same chromosome       #
# want to know WHAT genes are not DE - GO analysis?           

# put this information into a venn diagram
  #
  # subset the allSamplesDiffExpr data frame into vectors by sample with gene names

# A <- sample(1:1000, 400, replace = FALSE);
# B <- sample(1:1000, 600, replace = FALSE);
# C <- sample(1:1000, 350, replace = FALSE);
# D <- sample(1:1000, 550, replace = FALSE);
# E <- sample(1:1000, 375, replace = FALSE);

{
  venn.plot.EuGC <- venn.diagram(
	x = list(
		A = genesGC1DE$GENEID,
		B = genesGC2DE$GENEID,
		C = genesGC3DE$GENEID,
		D = genesGC5DE$GENEID,
		E = genesGC9DE$GENEID
		),
	filename = NULL, #"euploid_commonDE.tiff"
	col = "black",
	fill = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
	alpha = 0.50,
	cex = c(1.5, 1.5, 1.5, 1.5, 1.5, 1, 0.8, 1, 0.8, 1, 0.8, 1, 0.8,
	 1, 0.8, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 0.55, 1, 1, 1, 1, 1, 1.5),
	cat.col = c("dodgerblue", "goldenrod1", "darkorange1", "seagreen3", "orchid3"),
	cat.cex = 1.5,
	cat.fontface = "bold",
	margin = 0.05
	)
pdf("euploid_commonDE.pdf")
grid.draw(venn.plot.EuGC)
dev.off()
}

eupLinesGC <- list(genesGC1DE, genesGC2DE, genesGC3DE, genesGC5DE, genesGC9DE)
# want to get list of common DE genes in all the euploid lines
commonDEeupGC <- Reduce(intersect, Map("[[", eupLinesGC, "GENEID"))
#Reduc
commonDEeupGC <- as.vector(commonDEeupGC)
commonDEeupGC <- as.data.frame(commonDEeupGC)
colnames(commonDEeupGC) <- c("names")
write.csv(commonDEeupGC, file="commonDEeup_GC.csv")

# what genes are commonly DE in aneuploid lines, but NOT located on the aneuploid chromosomes?
# genes11DE,
aneuLinesGC <- 	list(deGenesNotAneuGCS4, deGenesNotAneuGCS7, deGenesNotAneuGCS8,deGenesNotAneuGCS11, deGenesNotAneuGCS18,
                     deGenesNotAneuGCS59, deGenesNotAneuGCS61, deGenesNotAneuGCS76, deGenesNotAneuGCS77)
# want to get list of common DE genes in all the euploid lines
commonDEaneuGC <- Reduce(intersect, Map("[[", aneuLinesGC, "GENEID"))
commonDEeupGC <- as.vector(commonDEeupGC)
write.csv(commonDEeupGC, file="commonDEeup_GC.csv")

# aneuLines4GC <- 	list(deGenesNotAneuGCS4, deGenesNotAneuGCS7, deGenesNotAneuGCS8,deGenesNotAneuGCS59)
# # want to get list of common DE genes in all the euploid lines
# commonDEaneu4 <- Reduce(intersect, Map("[[", aneuLines4, "GENEID"))
#
# aneuLines3 <- 	list(deGenesNotAneuS4, deGenesNotAneuS7, deGenesNotAneuS8)
# # want to get list of common DE genes in all the euploid lines
# commonDEaneu3 <- Reduce(intersect, Map("[[", aneuLines3, "GENEID"))




# later want to ask how many genes are DE in only euploid samples or only aneuploid samples
#
  # then query which samples the gene is DE in

```

```{r common DE genes Aneuploids, eval=TRUE,include=TRUE}

# setDT(res4Dat, keep.rownames = TRUE)[]
# setnames(res4Dat, 1, "GENEID")
# genes4DE <- subset(res4Dat,padj<0.1)


# setDT(res7Dat, keep.rownames = TRUE)[]
# setnames(res7Dat, 1, "GENEID")
# genes7DE <- subset(res7Dat,padj<0.1)


# setDT(res8Dat, keep.rownames = TRUE)[]
# setnames(res8Dat, 1, "GENEID")
# genes8DE <- subset(res8Dat,padj<0.1)


# setDT(res11Dat, keep.rownames = TRUE)[]
# setnames(res11Dat, 1, "GENEID")
# genes11DE <- subset(res11Dat,padj<0.1)


# setDT(res18Dat, keep.rownames = TRUE)[]
# setnames(res18Dat, 1, "GENEID")
# genes18DE <- subset(res18Dat,padj<0.1)


# setDT(res49Dat, keep.rownames = TRUE)[]
# setnames(res49Dat, 1, "GENEID")
# genes49DE <- subset(res49Dat,padj<0.1)


# setDT(res59Dat, keep.rownames = TRUE)[]
# setnames(res59Dat, 1, "GENEID")
# genes59DE <- subset(res59Dat,padj<0.1)


# setDT(res61Dat, keep.rownames = TRUE)[]
# setnames(res61Dat, 1, "GENEID")
# genes61DE <- subset(res61Dat,padj<0.1)


# setDT(res76Dat, keep.rownames = TRUE)[]
# setnames(res76Dat, 1, "GENEID")
# genes76DE <- subset(res76Dat,padj<0.1)


# setDT(res77Dat, keep.rownames = TRUE)[]
# setnames(res77Dat, 1, "GENEID")
# genes77DE <- subset(res77Dat,padj<0.1)

#remove genes that are commonly DE in euploid lines
# genes4DE.r <- anti_join(genes4DE, common1235.9,by="GENEID")
# genes7DE.r <- anti_join(genes7DE, common1235.9,by="GENEID")
# genes8DE.r <- anti_join(genes8DE, common1235.9,by="GENEID")
# genes11DE.r <- anti_join(genes11DE, common1235.9,by="GENEID")
# genes18DE.r <- anti_join(genes18DE, common1235.9,by="GENEID")
# genes49DE.r <- anti_join(genes49DE, common1235.9,by="GENEID")
# genes59DE.r <- anti_join(genes59DE, common1235.9,by="GENEID")
# genes61DE.r <- anti_join(genes61DE, common1235.9,by="GENEID")
# genes76DE.r <- anti_join(genes76DE, common1235.9,by="GENEID")
# genes77DE.r <- anti_join(genes77DE, common1235.9,by="GENEID")

#find common DE genes in aneuploids with same chromosome aneuploid
# chrI.tri <- inner_join(genes7DE,genes18DE, by="GENEID")
# matchedI <- intersect(chrI.tri$GENEID, genes$GENEID)
# keepI <- chrI.tri$GENEID %in% matchedI
# chrI.tri <- chrI.tri[keepI,]
# keepGen <- genes$GENEID %in% matchedI
# Gen <- genes[keepGen,]
# chrI.tri <- cbind(Gen$TXCHROM, chrI.tri)
#
# chrI.all <- inner_join(chrI.tri, genes11DE, by="GENEID")
# matchedI <- intersect(chrI.all$GENEID, genes$GENEID)
# keepI <- chrI.all$GENEID %in% matchedI
# chrI.all <- chrI.all[keepI,]
#
# chrV.tri <- inner_join(genes4DE, genes49DE, by="GENEID")
# matchedV <- intersect(chrV.tri$GENEID, genes$GENEID)
# keepV <- chrV.tri$GENEID %in% matchedV
# chrV.tri <- chrV.tri[keepI,]
# keepGen <- genes$GENEID %in% matchedV
# Gen <- genes[keepGen,]
# chrV.tri <- cbind(Gen$TXCHROM, chrV.tri)
#
# chrVII.tri <- inner_join(genes59DE, genes61DE, by="GENEID")
# matchedVII <- intersect(chrVII.tri$GENEID, genes$GENEID)
# keepVII <- chrVII.tri$GENEID %in% matchedVII
# chrVII.tri <- chrVII.tri[keepVII,]
# keepGen <- genes$GENEID %in% matchedVII
# Gen <- genes[keepGen,]
# chrVII.tri <- cbind(Gen$TXCHROM, chrVII.tri)
#
# chrXII.tri <- inner_join(genes18DE, genes77DE, by="GENEID")
# matchedXII <- intersect(chrXII.tri$GENEID, genes$GENEID)
# keepXII <- chrXII.tri$GENEID %in% matchedXII
# chrXII.tri <- chrXII.tri[keepXII,]
# keepGen <- genes$GENEID %in% matchedXII
# Gen <- genes[keepGen,]
# chrXII.tri <- cbind(Gen$TXCHROM, chrXII.tri)

##any commonly DE genes between 2+ aneuploids?

#that are NOT commonly DE in euploid lines
# dupRows <- dupsBetweenGroups(allSamplesDE, "Sample")

#find any genes on aneuploid chromosomes that are NOT DE
# notDE4 <- subset(res4Dat,padj>0.1)
# notDE7 <- subset(res7Dat,padj>0.1)
# notDE8 <- subset(res8Dat,padj>0.1)
# notDE11 <- subset(res11Dat,padj>0.1)
# notDE18 <- subset(res18Dat,padj>0.1)
# notDE49 <- subset(res49Dat,padj>0.1)
# notDE59 <- subset(res59Dat,padj>0.1)
# notDE61 <- subset(res61Dat,padj>0.1)
# notDE76 <- subset(res76Dat,padj>0.1)
# notDE77 <- subset(res77Dat,padj>0.1)

# notDECom <- inner_join(notDE4, notDE7, by="GENEID")
# notDECom <- inner_join(notDECom, notDE8, by="GENEID")
# notDECom <- inner_join(notDECom, notDE11, by="GENEID")
# notDECom <- inner_join(notDECom, notDE18, by="GENEID")
# notDECom <- inner_join(notDECom, notDE49, by="GENEID")
# notDECom <- inner_join(notDECom, notDE59, by="GENEID")
# notDECom <- inner_join(notDECom, notDE61, by="GENEID")
# notDECom <- inner_join(notDECom, notDE76, by="GENEID")
# notDECom <- inner_join(notDECom, notDE77, by="GENEID")

```

Below is a script for differentiating genes that are differentially expressed from those that are not dosage compensated, dosage compensated, or partially dosage compensated. I used the log fold change threshold function of the results function to isolate these genes. FDR p values < 0.1 were considered statistically significant.

```{r lfcThresh2,eval=TRUE}
# par(mfrow=c(2,2),mar=c(2,2,1,1))
# ylim <- c(-2.5,2.5)
# resGA <- results(ddsNoAneuInNorm, lfcThreshold=.5, altHypothesis="greaterAbs", test="Wald")
# resLA <- results(ddsNoAneuInNorm, lfcThreshold=.5, altHypothesis="lessAbs", test="Wald")
# resG <- results(ddsNoAneuInNorm, lfcThreshold=.5, altHypothesis="greater", test="Wald")
# resL <- results(ddsNoAneuInNorm, lfcThreshold=.5, altHypothesis="less", test="Wald")
# drawLines <- function() abline(h=c(-.5,.5),col="dodgerblue",lwd=2)
# plotMA(resGA, ylim=ylim); drawLines()
# plotMA(resLA, ylim=ylim); drawLines()
# plotMA(resG, ylim=ylim); drawLines()
# plotMA(resL, ylim=ylim); drawLines()

# Test for DC
##### GC Lines
# ddsNoAneuInNorm$condition
# res11DC <- results(ddsDefGC_11,contrast=c("condition","Holly_11","Holly_Anc"),lfcThreshold = 0,altHypothesis = "greaterAbs",test="Wald")
# # resultsS11GC <- results(ddsDefGC_11,contrast=c("condition","Holly_11","Holly_Anc"),alpha=0.05)
# summary(res11DC)
# View(resultsS11GC$padj)
# write.csv(resultsS11GC,file="res11GC.csv")
# res11DCnS <- as.data.frame(subset(res11DC,padj>0.1))
# res11DCnSmon <- as.data.frame(subset(res11DCnS,log2FoldChange>-1))
# res11DCnStri <- as.data.frame(subset(res11DCnSmon,log2FoldChange<0.585))

# res11DCnS <- cbind(names = rownames(res11DCnS), res11DCnS)

# res11DCS <- as.data.frame(subset(res11DC,padj<0.1))
# res11DCS <- cbind(names = rownames(res11DCS), res11DCS)
# write.csv(res11DCS,file="res11SigDiff.csv")
#
# res11Tri <- results(ddsDefGC_11,contrast=c("condition","Holly_11","Holly_Anc"),lfcThreshold = 0.585,altHypothesis = "greaterAbs",test="Wald")
# View(res11noDCTri$padj)
# write.csv(res11noDCTri,file="res11noDCTri.csv")
# res11TriS <- as.data.frame(subset(res11noDCTri,padj<0.1))
# res11TriS <- cbind(names = rownames(res11noDCTriS), res11noDCTriS)
# write.csv(res11TriS,file="res11TriOVER.csv")
#
# res11TriNS <- as.data.frame(subset(res11noDCTri,padj>0.1))
# res11TriNS <- cbind(names = rownames(res11noDCTriNS), res11noDCTriNS)
# write.csv(res11TriNS,file="res11TriNON.csv")
#
# res11noDCMon <- results(ddsDefGC_11,contrast=c("condition","Holly_11","Holly_Anc"),lfcThreshold = 1,altHypothesis = "less",test="Wald")
# View(res11noDCMon$padj)
# write.csv(res11noDCMon,file="res11Mon.csv")

############################################################
# Testing hypothesis with lfcThreshold other than 0
############################################################


### calculating p values manually
# for beta < -1
# rm(res11TEST)
# res11TEST <- results(ddsDefGC_11,contrast=c("condition","Holly_11","Holly_Anc"))
# # lfc threshold for monosomic genes is -1
# lfcThreshold <- -1
# # set lower.tail to FALSE when you want genes that are > -1
# res11TEST$pvalue <- pnorm(res11TEST$log2FoldChange,mean=lfcThreshold,sd=res11TEST$lfcSE,lower.tail=FALSE)
# ### directly compute adjusted p values
# library(genefilter)
# # set alpha to desired cutoff
# alpha <- 0.1
# # use baseMean as filter
# filter <- res11TEST$baseMean
# theta <- seq(mean(filter==0),1,length=20)
# filtPadj <- filtered_p(filter=filter,test=res11TEST$pvalue,theta=theta,method="BH")
# numRej <- colSums(filtPadj < alpha, na.rm = TRUE)
# # add in new adjusted p values based on your chosen lfcThreshold
# res11TEST$padj <- filtPadj[,which.max(numRej),drop=TRUE]
#
# rm(res11MonSig)
# separate the genes that are statistically significant
# meaning lfc > -1, Potentially compensated
# res11MonSig <- as.data.frame(subset(res11TEST,padj<0.1))
# res11MonSig <- cbind(names = rownames(res11MonSig), res11MonSig)
# write.csv(res11MonSig,file="res11MonSig.csv")
#
# rm(res11TESTnonSig)
# # separate the genes that are NOT statistically significant
# # meaning NO DC
# res11TESTnonSig <- as.data.frame(subset(res11TEST,padj>0.1))
# res11TESTnonSig <- cbind(names = rownames(res11TESTnonSig), res11TESTnonSig)
# write.csv(res11TESTnonSig,file="res11MonNonSig.csv")
#
# # separate the genes that are statistically significant but also <= -1
# # meaning lfc <=1, ANTI DC
# # there are none here
# res11MonAnti <- as.data.frame(subset(res11MonSig,log2FoldChange<=(-1)))
# res11MonAnti <- cbind(names = rownames(res11MonAnti), res11MonAnti)
# write.csv(res11MonAnti,file="res11MonAnti.csv")

# res11TESTanti <- results(ddsDefGC_11,contrast=c("condition","Holly_11","Holly_Anc"))
# lfcThreshold <- -1
# res11TESTanti$pvalue <- pnorm(res11TESTanti$log2FoldChange,mean=lfcThreshold,sd=res11TESTanti$lfcSE,lower.tail=TRUE)
# ### directly compute adjusted p values
# library(genefilter)
# alpha <- 0.1
# filter <- res11TESTanti$baseMean
# theta <- seq(mean(filter==0),1,length=20)
# filtPadj <- filtered_p(filter=filter,test=res11TESTanti$pvalue,theta=theta,method="BH")
# numRej <- colSums(filtPadj < alpha, na.rm = TRUE)
# res11TESTanti$padj <- filtPadj[,which.max(numRej),drop=TRUE]
#
# rm(res11MonAnti)
# res11MonAnti <- as.data.frame(subset(res11TESTanti,log2FoldChange<(-1)))
# res11MonAnti <- cbind(names = rownames(res11MonAnti), res11MonAnti)
# write.csv(res11MonAnti,file="res11MonAnti.csv")

#### for trisomy
# res11Tri <- results(ddsDefGC_11,contrast=c("condition","Holly_11","Holly_Anc"),lfcThreshold = 0.585,altHypothesis = "greaterAbs",test="Wald")
#
# # separate the genes that are statistically significant
# # meaning lfc > 0.585 > lfc, Potentially compensated
# res11TriSig <- as.data.frame(subset(res11Tri,padj<0.1))
# res11TriSig <- cbind(names = rownames(res11TriSig), res11TriSig)
# write.csv(res11TriSig,file="res11TriSig.csv")
#
# # meaning lfc > 0.585 > lfc, Potentially compensated
# res11TriNonSig <- as.data.frame(subset(res11Tri,padj>0.1))
# res11TriNonSig <- cbind(names = rownames(res11TriNonSig), res11TriNonSig)
# write.csv(res11TriNonSig,file="res11TriSig.csv")


############       SAMPLE 18 GC        ####################
# res18TEST <- results(ddsDefGC_18,contrast=c("condition","Holly_18","Holly_Anc"))
# resGC18 <- results(ddsDefGC_18,contrast=c("condition","Holly_18","Holly_Anc"),lfcThreshold = 0.585)
# summary(resGC18)
# lfc threshold for trisomic genes is 0.585
# lfcThreshold <- .585
# set lower.tail to FALSE when you want genes that are > -1
# res18TEST$pvalue <- pnorm(res18TEST$log2FoldChange,mean=lfcThreshold,sd=res18TEST$lfcSE,lower.tail=FALSE)
### directly compute adjusted p values
# library(genefilter)
# set alpha to desired cutoff
# alpha <- 0.1
# use baseMean as filter
# filter <- res18TEST$baseMean
# theta <- seq(mean(filter==0),1,length=20)
# filtPadj <- filtered_p(filter=filter,test=res18TEST$pvalue,theta=theta,method="BH")
# numRej <- colSums(filtPadj < alpha, na.rm = TRUE)
# # add in new adjusted p values based on your chosen lfcThreshold
# res18TEST$padj <- filtPadj[,which.max(numRej),drop=TRUE]

# resGC18nonSig <- as.data.frame(subset(resGC18,padj>0.1))
# resGC18nonSig <- cbind(names = rownames(resGC18nonSig), resGC18nonSig)
# resGC18nonSig <- dplyr::inner_join(genesGC_18,resGC18nonSig,by=c("GENEID"="names"))
#
# resGC18Sig <- as.data.frame(subset(resGC18,padj<0.1))
# resGC18Sig <- cbind(names = rownames(resGC18Sig), resGC18Sig)
# resGC18Sig <- dplyr::inner_join(genesGC_18,resGC18Sig,by=c("GENEID"="names"))
#
# ##### AGAINST LOG2(1) (lfcThreshold=0)
# res18Eu <- results(ddsDefGC_18,contrast=c("condition","Holly_18","Holly_Anc"),lfcThreshold = 0)
#
# res18EunonSig <- as.data.frame(subset(res18Eu,padj>0.1))
# res18EunonSig <- cbind(names = rownames(res18EunonSig), res18EunonSig)
# res18EunonSig <- dplyr::inner_join(genesGC_18,res18EunonSig,by=c("GENEID"="names"))
#
# res18EuSig <- as.data.frame(subset(res18Eu,padj<0.1))
# res18EuSig <- cbind(names = rownames(res18EuSig), res18EuSig)
# res18EuSig <- dplyr::inner_join(genesGC_18,res18EuSig,by=c("GENEID"="names"))
#
#
# ##### NOW FOR THE INTERSECTIONS
# res18NoDC <- dplyr::inner_join(resGC18nonSig,res18EuSig,by="GENEID")
#
# res18PartOvDC <- dplyr::inner_join(resGC18Sig,res18EuSig,by="GENEID")
#
# res18FullDC <- dplyr::inner_join(resGC18Sig,res18EunonSig,by="GENEID")
#
#
#
#
#
#
# # rm(res11TESTnonSig)
# # # separate the genes that are NOT statistically significant
# # # meaning NO DC
# # res11TESTnonSig <- as.data.frame(subset(res11TEST,padj>0.1))
# # res11TESTnonSig <- cbind(names = rownames(res11TESTnonSig), res11TESTnonSig)
# # write.csv(res11TESTnonSig,file="res11MonNonSig.csv")
# #
# # # separate the genes that are statistically significant but also <= -1
# # meaning lfc <=1, ANTI DC
# # there are none here
# # res11MonAnti <- as.data.frame(subset(res11MonSig,log2FoldChange<=(-1)))
# # res11MonAnti <- cbind(names = rownames(res11MonAnti), res11MonAnti)
# # write.csv(res11MonAnti,file="res11MonAnti.csv")
# #
# #
# # # now for the intersections
# # # No DC:
# #   # intersection of no DC sig and DC test sig
# #
# #     res11MonNoDC <- dplyr::inner_join(res11noDCMonS,res11DCS,by="names")
# #     # res11MonNoDC <- dplyr::anti_join(res11MonNoDC, commonDEeup,by="names")
# #     res11MonNoDC <- dplyr::inner_join(genesGC_11,res11MonNoDC,by=c("GENEID"="names"))
# #     write.csv(res11MonNoDC,file="Sample11NoDCMon.csv")
# #     res11TriNoDC <- dplyr::inner_join(res11noDCTriS,res11DCS,by="names")
# #     # res11TriNoDC <- dplyr::anti_join(res11TriNoDC,commonDEeup,by="names")
# #       # add chromosome number
# #       res11TriNoDC <- dplyr::inner_join(genesGC_11,res11TriNoDC,by=c("GENEID"="names"))
# #           write.csv(res11TriNoDC,file="Sample11NoDCTri.csv")
# #     
# # DC
# #     # intersection of DC test non significant and no DC test non significnant
# #     res11MonDC <- dplyr::inner_join(res11noDCMonNS,res11DCnS,by="names")
# #     # res11MonDC <- dplyr::anti_join(res11MonDC,commonDEeup,by="names")
# #         res11MonDC <- dplyr::inner_join(genesGC_11,res11MonDC,by=c("GENEID"="names"))
# #     write.csv(res11MonDC,file="Sample11DCMon.csv")
# #     res11TriDC <- dplyr::inner_join(res11noDCTriS,res11DCnS,by="names")
# #     # res11TriDC <- dplyr::anti_join(res11TriDC,commonDEeup,by="names")
# #     res11TriDC <- dplyr::inner_join(genesGC_11,res11TriDC,by=c("GENEID"="names"))
# #     write.csv(res11TriDC,file="Sample11DCTri.csv")
# #     View(res11MonDC)
# # # partial?
# #     # intersection of no DC test sig and DC test sig
# #     res11MonPartial <- dplyr::inner_join(res11noDCMonS,res11DCS,by="names")
# #     # res11MonPartial <- dplyr::anti_join(res11MonPartial,commonDEeup,by="names")
# #       res11MonPartial <- dplyr::inner_join(genesGC_11,res11MonPartial,by=c("GENEID"="names"))
# #     write.csv(res11MonPartial,file="Sample11PartialDCMon.csv")
# #     res11TriPartial <- dplyr::inner_join(res11noDCTriS,res11DCS,by="names")
# #     # res11TriPartial <- dplyr::anti_join(res11TriPartial,commonDEeup,by="names")
# #     res11TriPartial <- dplyr::inner_join(genesGC_11,res11TriPartial,by=c("GENEID"="names"))
# #     write.csv(res11TriPartial,file="Sample11PartialDCTri.csv")
# #     
#
#
# # can I make this into a function
#
# # testForDCSig <- function(sample1, sample2) {
# #   # first get the results for the given sample, and include the lfcThreshold -- 0 in this case as we are testing for DC
# #     # always testing against the ancestor so that can stay the same.
# #   res11DC <- results(ddsNoAneuInNorm,contrast=c("condition",sample,"Holly_Anc"),lfcThreshold = 0,altHypothesis = "greaterAbs",test="Wald")
# #   # next turn this dataset into a dataframe (from results object), and subset these results by p values greater than 0.1 (NOT SIGNIFICANT)
# #   res11DCnS <- as.data.frame(subset(res11DC,padj>0.1))
# #   # make the rownames (gene names) a new column called names
# #   res11DCnS <- cbind(names = rownames(res11DCnS), res11DCnS)
# #   # do the same thing, subset the dataset but this time pull out p values > or = 0.1 (SIGNIFICANT)
# #   res11DCS <- as.data.frame(subset(res11DC,padj<=0.1))
# #   # once again make the rownames (gene names) a column called names
# # res11DCS <- cbind(names = rownames(res11DCS), res11DCS)
# # # repeat the above, but this time testing for no DC in trisomics
# # # lfcThreshold = 0.538 (log2(1.5)), once again always testing against the ancestor
# # res11noDCTri <- results(ddsNoAneuInNorm,contrast=c("condition",sample,"Holly_Anc"),lfcThreshold = 0.538,altHypothesis = "greater",test="Wald")
# # # subset into significant p values
# # res11noDCTriS <- as.data.frame(subset(res11noDCTri,padj<=0.1))
# # # add gene names
# # res11noDCTriS <- cbind(names = rownames(res11noDCTriS), res11noDCTriS)
# # # subset into non significant values
# # res11noDCTriNS <- as.data.frame(subset(res11noDCTri,padj>0.1))
# # # add gene names
# # res11noDCTriNS <- cbind(names = rownames(res11noDCTriNS), res11noDCTriNS)
#
# ############       TRYING A LOOP     ####################
# # dds = ddsDefGC_18
# # sample = "Holly_18"
# # genes = genesGC_18
# # res <- results(dds,contrast=c("condition",sample,"Holly_Anc"))
# # summary(res)
# # resultsNames(ddsDefGC_18)
# # res18 <- results(ddsDefGC_18,contrast=c("condition","Holly_18","Holly_Anc"))
# # summary(res18)
# # summary(res18Eu)
# #
# # ddsGC18TEST <- ddsHTSeqGC_18
# # all(rownames(ddsGC18TEST) == genesGC_18$GENEID)
# # chrOfInt <- "chrXII"
# # #need to remove the ones in rownames of dds that don't match genes$GENEID
# # genesOfInt18 <- genesGC_18[(grep(paste0(chrOfInt,"$"),genesGC_18$TXCHROM,perl=TRUE)),]
# # matchedS2 <- intersect(rownames(ddsGC18TEST), genesOfInt18$GENEID)
# # keep4 <- rownames(ddsGC18TEST) %in% matchedS2
# # ddsGC18TEST <- ddsGC18TEST[keep4,]
# # # length(ddsHTSeqMAnew)
# # all(rownames(ddsGC18TEST) == genesGC_18$GENEID)
# # mcols(ddsGC18TEST) <- cbind(mcols(ddsGC18TEST), genesOfInt18)
# # ddsGC18TEST <- DESeq(ddsGC18TEST)
# # chrOfInt <- "chrXII"
# # genesOfInt <- genesGC_18[(grep(paste0(chrOfInt,"$"),genesGC_18$TXCHROM,perl=TRUE)),]
#
# resultsTrisomy <- function(dds,sample,anc,genes) {
# # make a new dds data set that we can use in future analyses
#   ddsCopy <- dds
#   # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
#   resTri <- results(dds,contrast=c("condition",sample,anc),lfcThreshold=0.585)
#   #subset data frame to get nonSignificant p values (indicating gene does is as expected (NO DC))
#   resTriNonSig <- as.data.frame(subset(resTri,padj>0.1))
#   resTriNonSig <- cbind(names = rownames(resTriNonSig), resTriNonSig)
#   resTriNonSig <- dplyr::inner_join(genes,resTriNonSig,by=c("GENEID"="names"))
#   # subset data frame to get Significant p values (indicting gene does is NOT as expected)
#   resTriSig <- as.data.frame(subset(resTri,padj<0.1))
#   resTriSig <- cbind(names = rownames(resTriSig), resTriSig)
#   resTriSig <- dplyr::inner_join(genes,resTriSig,by=c("GENEID"="names"))
# # put non-significantly different genes in a data table
#   results <- list()
#   results$resNoDC <- resTriNonSig
#   # results$resPotDC <- resTriSig
#   # need to pull out genes from resPotDC and use them to subset our ddsDataSet so we can run DESeq on only those genes
#     genesPotDC <- resTriNonSig$GENEID
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# # genesPotDC <- genes[(grep(paste0(genesPotDC,"$"),genesGC_18$GENEID,perl=TRUE)),]
# matchedS2 <- intersect(rownames(ddsCopy), genesPotDC)
# keep4 <- rownames(ddsCopy) %in% matchedS2
# ddsCopy <- ddsCopy[keep4,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsCopy) == genesPotDC)
# mcols(ddsCopy) <- cbind(mcols(ddsCopy), genesPotDC)
# ddsCopy <- DESeq(ddsCopy)
# View(counts(ddsCopy))
#   #### run same analyses using new ddsCopy dataset and new parameters
#   # run results function with log fold change threshold at 0 (expectation for disomic genes)
#   resDi <- results(ddsCopy,contrast=c("condition",sample,anc),lfcThreshold = 0)
#   # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
#   resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
#   resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
#   resDiNonSig <- dplyr::inner_join(genes,resDiNonSig,by=c("GENEID"="names"))
#   # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
#   resDiSig <- as.data.frame(subset(resDi,padj<0.1))
#   resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
#   resDiSig <- dplyr::inner_join(genes,resDiSig,by=c("GENEID"="names"))
#   ### Find genes that are Signficiantly different than Trisomic expectation and are Significantly different from Disomic expectation
#   resPartOvDC <- dplyr::inner_join(resTriSig,resDiSig,by="GENEID")
#   results$resPartOvDC <- resPartOvDC
#   ### Find genes that are Signficiantly different than Trisomic expectation but NotSignificantly different from Disomic expectation
#   resFullDC <- dplyr::inner_join(resTriSig,resDiNonSig,by="GENEID")
#   results$resFullDC <- resFullDC
#   return(results)
# }
#
# # View(results$resPartOvDC)
# # View(results$resFullDC)
# #this is with only genes from chromosome 12 (which is trisomic in sample 18)
# # testGC18ChrXII <- resultsTrisomy(ddsGC18TEST,sample,genesOfInt18)
# # View(testGC18ChrXII$resNoDC)
# # View(testGC18ChrXII$resPotDC)
#
# #this is including all of the genes in sample 18
# testGC18 <- resultsTrisomy(ddsDefGC_18,sample,anc="Holly_Anc",genesGC_18)
# testGC11 <- resultsTrisomy(ddsDefGC_11,sample="Holly_11",anc="Holly_Anc",genesGC_11)
# View(testGC11$resNoDC)
# View(testGC11$resPotDC)
# View(testGC11$resPartOvDC)
# View(testGC11$resFullDC)
#
# ddsDefMAold_15
# testMAOld15 <- resultsTrisomy(ddsDefMAold_15,sample="SC015",anc="SCA",genesMAold_15)
# View(testMAOld15$resNoDC)
# View(testMAOld15$resFullDC)
# View(testMAOld15$resPartOvDC)
# View(testMAOld15$resPotDC)
# View(testGC18$resNoDC)
# View(testGC18$resPotDC)
# View(testGC18$resPartOvDC)
# View(testGC18$resFullDC)
# dds = ddsHTSeqGC_18
# dds = DESeq(ddsHTSeqGC_18)
# testGC18NoDC <- resultsTrisomyNoDC(dds,sample,genes)

############## Slightly different way of testing -- different order ######
## for euploid lines
resultsEuploid <- function(dds,sample,anc,genes) {
# make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  all(rownames(dds) == genes$GENEID)
  # genes <- genes$GENEID
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  # ddsGenes <- intersect(rownames(dds),genes)
  # as.data.frame(genes)
  # ddsGenes <- genes[,ddsGenes]
  # length(ddsHTSeqMAold)
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes,resDiSig,by=c("GENEID"="names"))
   results <- list()
  results$Sig <- resDiSig
  results$nonSig <- resDiNonSig
  return(results)
}

sample2GC <- resultsEuploid(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2)
sample2GC$Sig
write.csv(sample2GC$Sig,file="DEgenesSample2GC.csv")

sample3GC <- resultsEuploid(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3)
sample3GC$Sig
write.csv(sample3GC$Sig,file="DEgenesSample3GC.csv")

sample5GC <- resultsEuploid(ddsDefGC_5,"Holly_5","Holly_Anc",genesGC_5)
sample5GC$Sig
write.csv(sample5GC$Sig,file="DEgenesSample5GC.csv")

# sample6GC <- resultsEuploid(ddsDefGC_6,"Holly_6","Holly_Anc",genesGC_6)
# sample6GC$Sig
# write.csv(sample6GC$Sig,file="DEgenesSample6GC.csv")

sample9GC <- resultsEuploid(ddsDefGC_9,"Holly_9","Holly_Anc",genesGC_9)
sample9GC$Sig
write.csv(sample9GC$Sig,file="DEgenesSample9GC.csv")

# DE genes in aneuploid lines

sample1MA <- resultsEuploid(ddsDefMAold_1,"SC001","SCA",genesMAold_1)
sample1MA$Sig
write.csv(sample1MA$Sig,file="DEgenesSample1MA.csv")

sample2MA <- resultsEuploid(ddsDefMAold_2,"SC002","SCA",genesMAold_2)
sample2MA$Sig
write.csv(sample2MA$Sig,file="DEgenesSample2MA.csv")

sample3MA <- resultsEuploid(ddsDefMAold_3,"SC003","SCA",genesMAold_3)
sample3MA$Sig
write.csv(sample3MA$Sig,file="DEgenesSample3MA.csv")

sample4MA <- resultsEuploid(ddsDefMAold_4,"SC004","SCA",genesMAold_4)
sample4MA$Sig
write.csv(sample4MA$Sig,file="DEgenesSample4MA.csv")

sample5MA <- resultsEuploid(ddsDefMAold_5,"SC005","SCA",genesMAold_5)
sample5MA$Sig
write.csv(sample5MA$Sig,file="DEgenesSample5MA.csv")

sample7MA <- resultsEuploid(ddsDefMAold_7,"SC007","SCA",genesMAold_7)
sample7MA$Sig
write.csv(sample7MA$Sig,file="DEgenesSample7MA.csv")

sample8MA <- resultsEuploid(ddsDefMAold_8,"SC008","SCA",genesMAold_8)
sample8MA$Sig
write.csv(sample8MA$Sig,file="DEgenesSample8MA.csv")

sample50MA <- resultsEuploid(ddsDefMAnewS50,"50","Anc",genesMAnew_50)
sample50MA$Sig
write.csv(sample50MA$Sig,file="DEgenesSample50MA.csv")

sample115MA <- resultsEuploid(ddsDefMAnewS115,"115","Anc",genesMAnewS115)
sample115MA$Sig
write.csv(sample115MA$Sig,file="DEgenesSample115MA.csv")

rm(dds)


resultsTrisomy <- function(dds,sample,anc,genes,anChr) {
# make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  keep <- anChr
  genes <- genes[(grep(paste0(keep,"$"),genes$TXCHROM,perl=TRUE)),]
  all(rownames(dds) == genes$GENEID)
  # genes <- genes$GENEID
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  # ddsGenes <- intersect(rownames(dds),genes)
  # as.data.frame(genes)
  # ddsGenes <- genes[,ddsGenes]
  # length(ddsHTSeqMAold)
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  resTri <- results(dds,contrast=c("condition",sample,anc),lfcThreshold=0.585)
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (NO DC))
  resTriNonSig <- as.data.frame(subset(resTri,padj>0.1))
  resTriNonSig <- cbind(names = rownames(resTriNonSig), resTriNonSig)
  resTriNonSig <- dplyr::inner_join(genes,resTriNonSig,by=c("GENEID"="names"))
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resTriSig <- as.data.frame(subset(resTri,padj<0.1))
  resTriSig <- cbind(names = rownames(resTriSig), resTriSig)
  resTriSig <- dplyr::inner_join(genes,resTriSig,by=c("GENEID"="names"))
# put non-significantly different genes in a data table
  results <- list()
  results$resNoDC <- resTriNonSig
  results$resPotDC <- resTriSig
#     genesPotDC <- resTriNonSig$GENEID
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# # genesPotDC <- genes[(grep(paste0(genesPotDC,"$"),genesGC_18$GENEID,perl=TRUE)),]
# matchedS2 <- intersect(rownames(ddsCopy), genesPotDC)
# keep4 <- rownames(ddsCopy) %in% matchedS2
# ddsCopy <- ddsCopy[keep4,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsCopy) == genesPotDC)
# mcols(ddsCopy) <- cbind(mcols(ddsCopy), genesPotDC)
# ddsCopy <- DESeq(ddsCopy)
# View(counts(ddsCopy))
  #### run same analyses using new ddsCopy dataset and new parameters
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(ddsCopy,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get Significant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  ### Find genes that are Signficiantly different than Trisomic expectation and are Significantly different from Disomic expectation
  resPartOvDC <- dplyr::inner_join(resTriSig,resDiSig,by="GENEID")
  results$resPartOvDC <- resPartOvDC
  ### Find genes that are Signficiantly different than Trisomic expectation but NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resTriSig,resDiNonSig,by="GENEID")
  results$resFullDC <- resFullDC
  transGenesDE <- dplyr::anti_join(resDiSig,genes,by=c("GENEID"="TXCHROM"))
  results$transGenesDE <- transGenesDE
  return(results)
}

resultsNames(ddsDefMAnewS152)
maOld152.1 <- resultsTrisomy(ddsDefMAnewS152,"152","Anc",genesMAnewS152,"chrI")
View(maOld152.1)


maOldTest <- resultsTrisomy(ddsDefMAold_108,"SC108","SCA",genesMAold_108,"chrVIII")

View(maOldTest$resNoDC)
View(maOldTest$resFullDC)
View(maOldTest$resPartOvDC)

### get trans genes in aneuploid lines that are differentially expressed
# then analyze if any of them are ESR genes


resultsESR <- function(dds,sample,anc,genes) {
# make a new dds data set that we can use in future analyses
   geneID <- rownames(dds)
  # #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  # #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
   genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  # #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  # genes <- na.omit(genes1)
  # genes <- as.data.frame(genes1)
  # remove1 <- anChr
  # genes <- genes[-(grep(paste0(remove1,"$"),genes$TXCHROM,perl=TRUE)),]
  # all(rownames(dds) == genes$GENEID)
  # # genes <- genes$GENEID
  # #need to remove the ones in rownames of dds that don't match genes$GENEID
  # matched_2 <- intersect(rownames(dds), genes$GENEID)
  # keep4 <- rownames(dds) %in% matched_2
  # dds <- dds[keep4,]
  # # ddsGenes <- intersect(rownames(dds),genes)
  # # as.data.frame(genes)
  # # ddsGenes <- genes[,ddsGenes]
  # # length(ddsHTSeqMAold)
  # all(rownames(dds) == genes$GENEID)
  # mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
# determine if any significantly differentially expressed genes match those identified by gasch et al 2000
  ESR <- list()
  ESR$Sig <- dplyr::inner_join(ASRgenes,resDiSig,by=c("GENEID"="GENEID"))
  ESR$NonSig <- dplyr::inner_join(ASRgenes,resDiNonSig,by=c("GENEID"="GENEID"))
  return(ESR)
}
#### these two need to be done separately bc they have 2 aneuploid chromosomes
esrGenesMA152 <- resultsESR(ddsDefMAnewS152,"152","Anc",genesMAnewS152)#,"chrI")
esrGenesMA152Sig <- esrGenesMA152$Sig
esrGenesMA152Up <- subset(esrGenesMA152Sig,log2FoldChange>1)
View(esrGenesMA152Up)
str(esrGenesMA152Up)
esrGenesMA152Down <- subset(esrGenesMA152Sig,log2FoldChange<1)
str(esrGenesMA152Down)


# write.csv(esrGenesMA152$Sig,file="sigESRGenesMA152.csv")
# write.csv(esrGenesMA152$NonSig,file="nonSigESRGenesMA152.csv")
esrGenesGC11 <- resultsESR(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11)#,"chrI")
esrGenesGC11Sig <- esrGenesGC11$Sig
esrGenesGC11Up <- subset(esrGenesGC11Sig,log2FoldChange>1)
View(esrGenesGC11Up)
str(esrGenesGC11Up)
esrGenesGC11Down <- subset(esrGenesGC11Sig,log2FoldChange<1)
str(esrGenesGC11Down)

# write.csv(esrGenesGC11$Sig,file="sigESRGenesGC11.csv")
# write.csv(esrGenesGC11$NonSig,file="nonSigESRGenesGC11.csv")

esrGenesGC18chrI <- resultsESR(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18)#,"chrI")
write.csv(esrGenesGC18chrI$Sig,file="sigESRGenesGC18.csv")
write.csv(esrGenesGC18chrI$NonSig,file="nonSigESRGenesGC18.csv")

esrGenesGC7chrI <- resultsESR(ddsDefGC_7,"Holly_7","Holly_Anc",genesGC_7)#,"chrI")
write.csv(esrGenesGC7chrI$Sig,file="sigESRGenesGC7.csv")
write.csv(esrGenesGC7chrI$NonSig,file="nonSigESRGenesGC7.csv")

esrGenesMA123chrV <- resultsESR(ddsDefMAnewS123,"123","Anc",genesMAnewS123)#,"chrV")
write.csv(esrGenesMA123chrV$Sig,file="sigESRGenesMA123.csv")
write.csv(esrGenesMA123chrV$NonSig,file="nonSigESRGenesMA123.csv")

esrGenesGC4chrV <- resultsESR(ddsDefGC_4,"Holly_4","Holly_Anc",genesGC_4)#,"chrV")
write.csv(esrGenesGC4chrV$Sig,file="sigESRGenesGC4.csv")
write.csv(esrGenesGC4chrV$NonSig,file="nonSigESRGenesGC4.csv")

esrGenesGC49chrV <- resultsESR(ddsDefGC_49,"Holly_49","Holly_Anc",genesGC_49)#,"chrV")
write.csv(esrGenesGC49chrV$Sig,file="sigESRGenesGC49.csv")
write.csv(esrGenesGC49chrV$NonSig,file="nonSigESRGenesGC49.csv")

esrGenesMA117chrV <- resultsESR(ddsDefMAnewS117,"117","Anc",genesMAnewS117)#,"chrV")
write.csv(esrGenesMA117chrV$Sig,file="sigESRGenesMA117.csv")
write.csv(esrGenesMA117chrV$NonSig,file="nonSigESRGenesMA117.csv")

esrGenesGC59chrVII <- resultsESR(ddsDefGC_59,"Holly_59","Holly_Anc",genesGC_59)#,"chrVII")
write.csv(esrGenesGC59chrVII$Sig,file="sigESRGenesGC59.csv")
write.csv(esrGenesGC59chrVII$NonSig,file="nonSigESRGenesGC59.csv")

esrGenesGC61chrVII <- resultsESR(ddsDefGC_61,"Holly_61","Holly_Anc",genesGC_61)#,"chrVII")
write.csv(esrGenesGC61chrVII$Sig,file="sigESRGenesGC61.csv")
write.csv(esrGenesGC61chrVII$NonSig,file="nonSigESRGenesGC61.csv")

esrGenesMA29chrIX <- resultsESR(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
write.csv(esrGenesMA29chrIX$Sig,file="sigESRGenesMA29.csv")
write.csv(esrGenesMA29chrIX$NonSig,file="nonSigESRGenesMA29.csv")

esrGenesMA119chrIX <- resultsESR(ddsDefMAold_119,"SC119","SCA",genesMAoldS119)#,"chrIX")
write.csv(esrGenesMA119chrIX$Sig,file="sigESRGenesMA119.csv")
write.csv(esrGenesMA119chrIX$NonSig,file="nonSigESRGenesMA119.csv")

esrGenesMA88chrIX <- resultsESR(ddsDefMAold_88,"SC088","SCA",genesMAoldS88)#,"chrIX")
write.csv(esrGenesMA88chrIX$Sig,file="sigESRGenesMA88.csv")
write.csv(esrGenesMA88chrIX$NonSig,file="nonSigESRGenesMA88.csv")

esrGenesMA108chrVIII <- resultsESR(ddsDefMAold_108,"SC108","SCA",genesMAoldS108)#,"chrIX")
write.csv(esrGenesMA108chrVIII$Sig,file="sigESRGenesMA108.csv")
write.csv(esrGenesMA108chrVIII$NonSig,file="nonSigESRGenesMA108.csv")


esrGenesMA15chrIX <- resultsESR(ddsDefMAold_15,"SC015","SCA",genesMAoldS15)#,"chrIX")
write.csv(esrGenesMA15chrIX$Sig,file="sigESRGenesMA15.csv")
write.csv(esrGenesMA15chrIX$NonSig,file="nonSigESRGenesMA15.csv")

esrGenesMA29chrIX <- resultsESR(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
write.csv(esrGenesMA29chrIX$Sig,file="sigESRGenesMA29.csv")
write.csv(esrGenesMA29chrIX$NonSig,file="nonSigESRGenesMA29.csv")

esrGenesGC76chrIX <- resultsESR(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76)#,"chrIX")
write.csv(esrGenesGC76chrIX$Sig,file="sigESRGenesGC76.csv")
write.csv(esrGenesGC76chrIX$NonSig,file="nonSigESRGenesGC76.csv")

esrGenesGC77chrXII <- resultsESR(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_76)#,"chrIX")
write.csv(esrGenesGC77chrXII$Sig,file="sigESRGenesGC77.csv")
write.csv(esrGenesGC77chrXII$NonSig,file="nonSigESRGenesGC77.csv")

esrGenesMA9chrXIV <- resultsESR(ddsDefMAold_9,"SC009","SCA",genesMAoldS9)#,"chrXIV")
write.csv(esrGenesMA9chrXIV$Sig,file="sigESRGenesMA9.csv")
write.csv(esrGenesMA9chrXIV$NonSig,file="nonSigESRGenesMA9.csv")

esrGenesMA11 <- resultsESR(ddsDefMAold_11,"SC011","SCA",genesMAoldS11)#,"chrXIV")
write.csv(esrGenesMA11$Sig,file="sigESRGenesMA11.csv")
write.csv(esrGenesMA11$NonSig,file="nonSigESRGenesMA11.csv")

esrGenesMA112chrXVI <- resultsESR(ddsDefMAnewS112,"112","Anc",genesMAnewS112)#,"chrXVI")
write.csv(esrGenesMA112chrXVI$Sig,file="sigESRGenesMA112.csv")
write.csv(esrGenesMA112chrXVI$NonSig,file="nonSigESRGenesMA112.csv")

esrGenesGC8chrXVI <- resultsESR(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8)#,"chrXVI")
write.csv(esrGenesGC8chrXVI$Sig,file="sigESRGenesGC8.csv")
write.csv(esrGenesGC8chrXVI$NonSig,file="nonSigESRGenesGC8.csv")

### need to do the euploid samples next - ESR genes
esrGenesGC2 <- resultsESR(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2)
write.csv(esrGenesGC2$Sig,file="sigESRGenesGC2.csv")
write.csv(esrGenesGC2$NonSig,file="nonSigESRGenesGC2.csv")

esrGenesGC3 <- resultsESR(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3)
write.csv(esrGenesGC3$Sig,file="sigESRGenesGC3.csv")
write.csv(esrGenesGC3$NonSig,file="nonSigESRGenesGC3.csv")

esrGenesGC5 <- resultsESR(ddsDefGC_5,"Holly_5","Holly_Anc",genesGC_5)
write.csv(esrGenesGC5$Sig,file="sigESRGenesGC5.csv")
write.csv(esrGenesGC5$NonSig,file="nonSigESRGenesGC5.csv")

# esrGenesGC6 <- resultsESR(ddsDefGC_6,"Holly_6","Holly_Anc",genesGC_6)
# write.csv(esrGenesGC6$Sig,file="sigESRGenesGC6.csv")
# write.csv(esrGenesGC6$NonSig,file="nonSigESRGenesGC6.csv")

esrGenesGC9 <- resultsESR(ddsDefGC_9,"Holly_9","Holly_Anc",genesGC_9)
write.csv(esrGenesGC9$Sig,file="sigESRGenesGC9.csv")
write.csv(esrGenesGC9$NonSig,file="nonSigESRGenesGC9.csv")

esrGenesMA1 <- resultsESR(ddsDefMAold_1,"SC001","SCA",genesMAold_1)
write.csv(esrGenesMA1$Sig,file="sigESRGenesMA1.csv")
write.csv(esrGenesMA1$NonSig,file="nonSigESRGenesMA1.csv")

esrGenesMA2 <- resultsESR(ddsDefMAold_2,"SC002","SCA",genesMAold_2)
write.csv(esrGenesMA2$Sig,file="sigESRGenesMA2.csv")
write.csv(esrGenesMA2$NonSig,file="nonSigESRGenesMA2.csv")

esrGenesMA3 <- resultsESR(ddsDefMAold_3,"SC003","SCA",genesMAold_3)
write.csv(esrGenesMA3$Sig,file="sigESRGenesMA3.csv")
write.csv(esrGenesMA3$NonSig,file="nonSigESRGenesMA3.csv")

esrGenesMA4 <- resultsESR(ddsDefMAold_4,"SC004","SCA",genesMAold_4)
write.csv(esrGenesMA4$Sig,file="sigESRGenesMA4.csv")
write.csv(esrGenesMA4$NonSig,file="nonSigESRGenesMA4.csv")

esrGenesMA5 <- resultsESR(ddsDefMAold_5,"SC005","SCA",genesMAold_5)
write.csv(esrGenesMA5$Sig,file="sigESRGenesMA5.csv")
write.csv(esrGenesMA5$NonSig,file="nonSigESRGenesMA5.csv")

esrGenesMA7 <- resultsESR(ddsDefMAold_7,"SC007","SCA",genesMAold_7)
write.csv(esrGenesMA7$Sig,file="sigESRGenesMA7.csv")
write.csv(esrGenesMA7$NonSig,file="nonSigESRGenesMA7.csv")

esrGenesMA8 <- resultsESR(ddsDefMAold_8,"SC008","SCA",genesMAold_8)
write.csv(esrGenesMA8$Sig,file="sigESRGenesMA8.csv")
write.csv(esrGenesMA8$NonSig,file="nonSigESRGenesMA8.csv")

esrGenesMA50 <- resultsESR(ddsDefMAnewS50,"50","Anc",genesMAnewS50)
write.csv(esrGenesMA50$Sig,file="sigESRGenesMA50.csv")
write.csv(esrGenesMA50$NonSig,file="nonSigESRGenesMA50.csv")

esrGenesMA115 <- resultsESR(ddsDefMAnewS115,"115","Anc",genesMAnewS115)
write.csv(esrGenesMA115$Sig,file="sigESRGenesMA115.csv")
write.csv(esrGenesMA115$NonSig,file="nonSigESRGenesMA115.csv")

### see if they are in ASR
resultsASR <- function(dds,sample,anc,genes) {
# make a new dds data set that we can use in future analyses
   geneID <- rownames(dds)
  # #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  # #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
   genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  # #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  # genes <- na.omit(genes1)
  # genes <- as.data.frame(genes1)
  # remove1 <- anChr
  # genes <- genes[-(grep(paste0(remove1,"$"),genes$TXCHROM,perl=TRUE)),]
  # all(rownames(dds) == genes$GENEID)
  # # genes <- genes$GENEID
  # #need to remove the ones in rownames of dds that don't match genes$GENEID
  # matched_2 <- intersect(rownames(dds), genes$GENEID)
  # keep4 <- rownames(dds) %in% matched_2
  # dds <- dds[keep4,]
  # # ddsGenes <- intersect(rownames(dds),genes)
  # # as.data.frame(genes)
  # # ddsGenes <- genes[,ddsGenes]
  # # length(ddsHTSeqMAold)
  # all(rownames(dds) == genes$GENEID)
  # mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
# determine if any significantly differentially expressed genes match those identified by gasch et al 2000
  ASR <- list()
  ASR$Sig <- dplyr::inner_join(ASRgenes,resDiSig,by=c("GENEID"="GENEID"))
  ASR$NonSig <- dplyr::inner_join(ASRgenes,resDiNonSig,by=c("GENEID"="GENEID"))
  return(ASR)
}

ASRGenesMA152 <- resultsASR(ddsDefMAnewS152,"152","Anc",genesMAnewS152)#,"chrI")
asrGenesSigMA152 <- ASRGenesMA152$Sig
write.csv(ASRGenesMA152$Sig,file="sigASRGenesMA152.csv")
write.csv(ASRGenesMA152$NonSig,file="nonSigASRGenesMA152.csv")
ASRGenesGC11 <- resultsASR(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11)#,"chrI")
asrGenesSigGC11 <- ASRGenesGC11$Sig
write.csv(ASRGenesGC11$Sig,file="sigASRGenesGC11.csv")
write.csv(ASRGenesGC11$NonSig,file="nonSigASRGenesGC11.csv")

ASRGenesGC18chrI <- resultsASR(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18)#,"chrI")
asrGenesSigGC18 <- ASRGenesGC18chrI$Sig
write.csv(ASRGenesGC18chrI$Sig,file="sigASRGenesGC18.csv")
write.csv(ASRGenesGC18chrI$NonSig,file="nonSigASRGenesGC18.csv")

ASRGenesGC7chrI <- resultsASR(ddsDefGC_7,"Holly_7","Holly_Anc",genesGC_7)#,"chrI")
asrGenesSigGC7 <- ASRGenesGC7chrI$Sig
write.csv(ASRGenesGC7chrI$Sig,file="sigASRGenesGC7.csv")
write.csv(ASRGenesGC7chrI$NonSig,file="nonSigASRGenesGC7.csv")

ASRGenesMA123chrV <- resultsASR(ddsDefMAnewS123,"123","Anc",genesMAnewS123)#,"chrV")
asrGenesSigMA123 <- ASRGenesMA123chrV$Sig
write.csv(ASRGenesMA123chrV$Sig,file="sigASRGenesMA123.csv")
write.csv(ASRGenesMA123chrV$NonSig,file="nonSigASRGenesMA123.csv")

ASRGenesGC4chrV <- resultsASR(ddsDefGC_4,"Holly_4","Holly_Anc",genesGC_4)#,"chrV")
asrGenesSigGC4 <- ASRGenesGC4chrV$Sig
write.csv(ASRGenesGC4chrV$Sig,file="sigASRGenesGC4.csv")
write.csv(ASRGenesGC4chrV$NonSig,file="nonSigASRGenesGC4.csv")

ASRGenesGC49chrV <- resultsASR(ddsDefGC_49,"Holly_49","Holly_Anc",genesGC_49)#,"chrV")
asrGenesSigGC49 <- ASRGenesGC49chrV$Sig
write.csv(ASRGenesGC49chrV$Sig,file="sigASRGenesGC49.csv")
write.csv(ASRGenesGC49chrV$NonSig,file="nonSigASRGenesGC49.csv")

ASRGenesMA117chrV <- resultsASR(ddsDefMAnewS117,"117","Anc",genesMAnewS117)#,"chrV")
asrGenesSigMA117 <- ASRGenesMA117chrV$Sig
write.csv(ASRGenesMA117chrV$Sig,file="sigASRGenesMA117.csv")
write.csv(ASRGenesMA117chrV$NonSig,file="nonSigASRGenesMA117.csv")

ASRGenesGC59chrVII <- resultsASR(ddsDefGC_59,"Holly_59","Holly_Anc",genesGC_59)#,"chrVII")
asrGenesSigGC59 <- ASRGenesGC59chrVII$Sig
write.csv(ASRGenesGC59chrVII$Sig,file="sigASRGenesGC59.csv")
write.csv(ASRGenesGC59chrVII$NonSig,file="nonSigASRGenesGC59.csv")

ASRGenesGC61chrVII <- resultsASR(ddsDefGC_61,"Holly_61","Holly_Anc",genesGC_61)#,"chrVII")
asrGenesSigGC61 <- ASRGenesGC61chrVII$Sig
write.csv(ASRGenesGC61chrVII$Sig,file="sigASRGenesGC61.csv")
write.csv(ASRGenesGC61chrVII$NonSig,file="nonSigASRGenesGC61.csv")

ASRGenesMA29chrIX <- resultsASR(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
asrGenesSigMA29 <- ASRGenesMA29chrIX$Sig
write.csv(ASRGenesMA29chrIX$Sig,file="sigASRGenesMA29.csv")
write.csv(ASRGenesMA29chrIX$NonSig,file="nonSigASRGenesMA29.csv")

ASRGenesMA119chrIX <- resultsASR(ddsDefMAold_119,"SC119","SCA",genesMAoldS119)#,"chrIX")
asrGenesSigMA119 <- ASRGenesMA119chrIX$Sig
write.csv(ASRGenesMA119chrIX$Sig,file="sigASRGenesMA119.csv")
write.csv(ASRGenesMA119chrIX$NonSig,file="nonSigASRGenesMA119.csv")

ASRGenesMA88chrIX <- resultsASR(ddsDefMAold_88,"SC088","SCA",genesMAoldS88)#,"chrIX")
asrGenesSigMA88 <- ASRGenesMA88chrIX$Sig
write.csv(ASRGenesMA88chrIX$Sig,file="sigASRGenesMA88.csv")
write.csv(ASRGenesMA88chrIX$NonSig,file="nonSigASRGenesMA88.csv")

ASRGenesMA108chrVIII <- resultsASR(ddsDefMAold_108,"SC108","SCA",genesMAoldS108)#,"chrIX")
asrGenesSigMA108 <- ASRGenesMA108chrVIII$Sig
write.csv(ASRGenesMA108chrVIII$Sig,file="sigASRGenesMA108.csv")
write.csv(ASRGenesMA108chrVIII$NonSig,file="nonSigASRGenesMA108.csv")


ASRGenesMA15chrIX <- resultsASR(ddsDefMAold_15,"SC015","SCA",genesMAoldS15)#,"chrIX")
asrGenesSigMA15 <- ASRGenesMA15chrIX$Sig
write.csv(ASRGenesMA15chrIX$Sig,file="sigASRGenesMA15.csv")
write.csv(ASRGenesMA15chrIX$NonSig,file="nonSigASRGenesMA15.csv")

ASRGenesMA29chrIX <- resultsASR(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
asrGenesSigMA29 <- ASRGenesMA29chrIX$Sig
write.csv(ASRGenesMA29chrIX$Sig,file="sigASRGenesMA29.csv")
write.csv(ASRGenesMA29chrIX$NonSig,file="nonSigASRGenesMA29.csv")

ASRGenesGC76chrIX <- resultsASR(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76)#,"chrIX")
asrGenesSigGC76 <- ASRGenesGC76chrIX$Sig
write.csv(ASRGenesGC76chrIX$Sig,file="sigASRGenesGC76.csv")
write.csv(ASRGenesGC76chrIX$NonSig,file="nonSigASRGenesGC76.csv")

ASRGenesGC77chrXII <- resultsASR(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_76)#,"chrIX")
asrGenesSigGC77 <- ASRGenesGC77chrXII$Sig
write.csv(ASRGenesGC77chrXII$Sig,file="sigASRGenesGC77.csv")
write.csv(ASRGenesGC77chrXII$NonSig,file="nonSigASRGenesGC77.csv")

ASRGenesMA9chrXIV <- resultsASR(ddsDefMAold_9,"SC009","SCA",genesMAoldS9)#,"chrXIV")
asrGenesSigMA9 <- ASRGenesMA9chrXIV$Sig
write.csv(ASRGenesMA9chrXIV$Sig,file="sigASRGenesMA9.csv")
write.csv(ASRGenesMA9chrXIV$NonSig,file="nonSigASRGenesMA9.csv")

ASRGenesMA11 <- resultsASR(ddsDefMAold_11,"SC011","SCA",genesMAoldS11)#,"chrXIV")
asrGenesSigMA11 <- ASRGenesMA11$Sig
write.csv(ASRGenesMA11$Sig,file="sigASRGenesMA11.csv")
write.csv(ASRGenesMA11$NonSig,file="nonSigASRGenesMA11.csv")

ASRGenesMA112chrXVI <- resultsASR(ddsDefMAnewS112,"112","Anc",genesMAnewS112)#,"chrXVI")
asrGenesSigMA112 <- ASRGenesMA112chrXVI$Sig
write.csv(ASRGenesMA112chrXVI$Sig,file="sigASRGenesMA112.csv")
write.csv(ASRGenesMA112chrXVI$NonSig,file="nonSigASRGenesMA112.csv")

ASRGenesGC8chrXVI <- resultsASR(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8)#,"chrXVI")
asrGenesSigGC8 <- ASRGenesGC8chrXVI$Sig
write.csv(ASRGenesGC8chrXVI$Sig,file="sigASRGenesGC8.csv")
write.csv(ASRGenesGC8chrXVI$NonSig,file="nonSigASRGenesGC8.csv")

### need to do the euploid samples next - ASR genes

ASRGenesGC1 <- resultsASR(ddsDefGC_1,"Holly_1","Holly_Anc",genesGC_1)
asrGenesSigGC1 <- ASRGenesGC1$Sig
write.csv(ASRGenesGC1$Sig,file="sigASRGenesGC1.csv")
write.csv(ASRGenesGC1$NonSig,file="nonSigASRGenesGC1.csv")

ASRGenesGC2 <- resultsASR(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2)
asrGenesSigGC2 <- ASRGenesGC2$Sig
write.csv(ASRGenesGC2$Sig,file="sigASRGenesGC2.csv")
write.csv(ASRGenesGC2$NonSig,file="nonSigASRGenesGC2.csv")

ASRGenesGC3 <- resultsASR(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3)
asrGenesSigGC3 <- ASRGenesGC3$Sig
write.csv(ASRGenesGC3$Sig,file="sigASRGenesGC3.csv")
write.csv(ASRGenesGC3$NonSig,file="nonSigASRGenesGC3.csv")

ASRGenesGC5 <- resultsASR(ddsDefGC_5,"Holly_5","Holly_Anc",genesGC_5)
asrGenesSigGC5 <- ASRGenesGC5$Sig
write.csv(ASRGenesGC5$Sig,file="sigASRGenesGC5.csv")
write.csv(ASRGenesGC5$NonSig,file="nonSigASRGenesGC5.csv")

# ASRGenesGC6 <- resultsASR(ddsDefGC_6,"Holly_6","Holly_Anc",genesGC_6)
# write.csv(ASRGenesGC6$Sig,file="sigASRGenesGC6.csv")
# write.csv(ASRGenesGC6$NonSig,file="nonSigASRGenesGC6.csv")

ASRGenesGC9 <- resultsASR(ddsDefGC_9,"Holly_9","Holly_Anc",genesGC_9)
asrGenesSigGC9 <- ASRGenesGC9$Sig
write.csv(ASRGenesGC9$Sig,file="sigASRGenesGC9.csv")
write.csv(ASRGenesGC9$NonSig,file="nonSigASRGenesGC9.csv")

ASRGenesMA1 <- resultsASR(ddsDefMAold_1,"SC001","SCA",genesMAold_1)
asrGenesSigMA1 <- ASRGenesMA1$Sig
write.csv(ASRGenesMA1$Sig,file="sigASRGenesMA1.csv")
write.csv(ASRGenesMA1$NonSig,file="nonSigASRGenesMA1.csv")

ASRGenesMA2 <- resultsASR(ddsDefMAold_2,"SC002","SCA",genesMAold_2)
asrGenesSigMA2 <- ASRGenesMA2$Sig
write.csv(ASRGenesMA2$Sig,file="sigASRGenesMA2.csv")
write.csv(ASRGenesMA2$NonSig,file="nonSigASRGenesMA2.csv")

ASRGenesMA3 <- resultsASR(ddsDefMAold_3,"SC003","SCA",genesMAold_3)
asrGenesSigMA3 <- ASRGenesMA3$Sig
write.csv(ASRGenesMA3$Sig,file="sigASRGenesMA3.csv")
write.csv(ASRGenesMA3$NonSig,file="nonSigASRGenesMA3.csv")

ASRGenesMA4 <- resultsASR(ddsDefMAold_4,"SC004","SCA",genesMAold_4)
asrGenesSigMA4 <- ASRGenesMA4$Sig
write.csv(ASRGenesMA4$Sig,file="sigASRGenesMA4.csv")
write.csv(ASRGenesMA4$NonSig,file="nonSigASRGenesMA4.csv")

ASRGenesMA5 <- resultsASR(ddsDefMAold_5,"SC005","SCA",genesMAold_5)
asrGenesSigMA5 <- ASRGenesMA5$Sig
write.csv(ASRGenesMA5$Sig,file="sigASRGenesMA5.csv")
write.csv(ASRGenesMA5$NonSig,file="nonSigASRGenesMA5.csv")

ASRGenesMA7 <- resultsASR(ddsDefMAold_7,"SC007","SCA",genesMAold_7)
asrGenesSigMA7 <- ASRGenesMA7$Sig
write.csv(ASRGenesMA7$Sig,file="sigASRGenesMA7.csv")
write.csv(ASRGenesMA7$NonSig,file="nonSigASRGenesMA7.csv")

ASRGenesMA8 <- resultsASR(ddsDefMAold_8,"SC008","SCA",genesMAold_8)
asrGenesSigMA8 <- ASRGenesMA8$Sig
write.csv(ASRGenesMA8$Sig,file="sigASRGenesMA8.csv")
write.csv(ASRGenesMA8$NonSig,file="nonSigASRGenesMA8.csv")

ASRGenesMA50 <- resultsASR(ddsDefMAnewS50,"50","Anc",genesMAnewS50)
asrGenesSigMA50 <- ASRGenesMA50$Sig
write.csv(ASRGenesMA50$Sig,file="sigASRGenesMA50.csv")
write.csv(ASRGenesMA50$NonSig,file="nonSigASRGenesMA50.csv")

ASRGenesMA115 <- resultsASR(ddsDefMAnewS115,"115","Anc",genesMAnewS115)
asrGenesSigMA115 <- ASRGenesMA115$Sig
write.csv(ASRGenesMA115$Sig,file="sigASRGenesMA115.csv")
write.csv(ASRGenesMA115$NonSig,file="nonSigASRGenesMA115.csv")

##find common sig DE ASR genes in GC euploids
# l1 <- c(1,2,3,4,5,6,7)
# l2 <- c(1,2,3,4,5,6)
# l3 <- c(1,2,3,4,5)
# l4 <- c(1,2,3,4)
#
# Reduce(intersect, list(l1,l2,l3,l4))
asrGenesCommGCEup <- Reduce(intersect, list(asrGenesSigGC1,asrGenesSigGC2,asrGenesSigGC3,asrGenesSigGC5,asrGenesSigGC9))
asrGenesCommGCEup
## no commonly found sig DE ASR genes in all GC euploids

asrGenesCommMAoldEup <- Reduce(intersect, list(asrGenesSigMA1,asrGenesSigMA2,asrGenesSigMA3,asrGenesSigMA4,asrGenesSigMA5,asrGenesSigMA7,asrGenesSigMA8))
asrGenesCommMAoldEup

asrGenesCommMAnewEup <- Reduce(intersect,list(asrGenesSigMA50,asrGenesSigMA115))
# also 0

### see if there are any ASR genes commonly found in All Aneuploid GC samples
asrGenesCommGCAneu <- Reduce(intersect,list(asrGenesSigGC11,asrGenesSigGC18,asrGenesSigGC4,asrGenesSigGC49,asrGenesSigGC59,asrGenesSigGC61,asrGenesSigGC7,asrGenesSigGC76,asrGenesSigGC77,asrGenesSigGC8))
# no commonly sig DE ASR genes in aneuploid GC samples

##old ma samples
asrGenesCommMAOldAneu <- Reduce(intersect,list(asrGenesSigMA108,asrGenesSigMA119,asrGenesSigMA15,asrGenesSigMA88))

asrGenesCommMANewAneu <- Reduce(intersect,list(asrGenesSigMA112,asrGenesSigMA117,asrGenesSigMA123,asrGenesSigMA152,asrGenesSigMA29))

# 0 in MA old and new samples too

### command to find common elements in several files: 1,2,3,4, + files
# want to know how many genes are differentially expressed commonly in 1,2,3 etc EUPLOID lines
#
  # put all ANEUPLOID samples in same data frame
    asrGenesAneuGCSamplesExpr <- rbind(asrGenesSigGC11,asrGenesSigGC18,asrGenesSigGC4,asrGenesSigGC49,asrGenesSigGC59,asrGenesSigGC61,asrGenesSigGC7,asrGenesSigGC76,asrGenesSigGC77,asrGenesSigGC8)
      # # first need to filter allSamplesExpr based on p-adj level
      # eupMAnewSamplesDiffExpr <- subset(eupMAnewSamplesExpr, padj<0.1)
      # #
      #
        # then ask how many times each GENEID  is found in eupSamplesExpr
        library(plyr)
        asrGeneGCFreqAneu <- plyr::count(asrGenesAneuGCSamplesExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numGCAneuASRGenesDE <- plyr::count(asrGeneGCFreqAneu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numGCAneuASRGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numGCAneuASRGenesDE, file="numb_comm_DE_GC_ASR_genesANEU.csv")   

### command to find common elements in several files: 1,2,3,4, + files
# want to know how many genes are differentially expressed commonly in 1,2,3 etc ANEUPLOID lines
#
  # put all ANEUPLOID samples in same data frame
    asrGenesAneuMAoldSamplesExpr <- rbind(asrGenesSigMA108,asrGenesSigMA119,asrGenesSigMA15,asrGenesSigMA88)
      # # first need to filter allSamplesExpr based on p-adj level
      # eupMAnewSamplesDiffExpr <- subset(eupMAnewSamplesExpr, padj<0.1)
      # #
      #
        # then ask how many times each GENEID  is found in eupSamplesExpr
        # library(plyr)
        asrGeneMAoldFreqAneu <- plyr::count(asrGenesAneuMAoldSamplesExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numMAoldAneuASRGenesDE <- plyr::count(asrGeneMAoldFreqAneu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numMAoldAneuASRGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numMAoldAneuASRGenesDE, file="numb_comm_DE_MAold_ASR_genesANEU.csv")


### command to find common elements in several files: 1,2,3,4, + files
# want to know how many genes are differentially expressed commonly in 1,2,3 etc ANEUPLOID lines
#
  # put all ANEUPLOID samples in same data frame
    asrGenesAneuMAnewSamplesExpr <- rbind(asrGenesSigMA112,asrGenesSigMA29,asrGenesSigMA117,asrGenesSigMA123,asrGenesSigMA152)
      # # first need to filter allSamplesExpr based on p-adj level
      # eupMAnewSamplesDiffExpr <- subset(eupMAnewSamplesExpr, padj<0.1)
      # #
      #
        # then ask how many times each GENEID  is found in eupSamplesExpr
        # library(plyr)
        asrGeneMAnewFreqAneu <- plyr::count(asrGenesAneuMAnewSamplesExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numMAnewAneuASRGenesDE <- plyr::count(asrGeneMAnewFreqAneu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numMAnewAneuASRGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numMAnewAneuASRGenesDE, file="numb_comm_DE_MAnew_ASR_genesANEU.csv")

## then do all samples - dosage Sensitive Genes


resultsDosSens <- function(dds,sample,anc,genes) {
# make a new dds data set that we can use in future analyses
   geneID <- rownames(dds)
  # #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  # #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
   genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  # #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  # genes <- na.omit(genes1)
  # genes <- as.data.frame(genes1)
  # remove1 <- anChr
  # genes <- genes[-(grep(paste0(remove1,"$"),genes$TXCHROM,perl=TRUE)),]
  # all(rownames(dds) == genes$GENEID)
  # # genes <- genes$GENEID
  # #need to remove the ones in rownames of dds that don't match genes$GENEID
  # matched_2 <- intersect(rownames(dds), genes$GENEID)
  # keep4 <- rownames(dds) %in% matched_2
  # dds <- dds[keep4,]
  # # ddsGenes <- intersect(rownames(dds),genes)
  # # as.data.frame(genes)
  # # ddsGenes <- genes[,ddsGenes]
  # # length(ddsHTSeqMAold)
  # all(rownames(dds) == genes$GENEID)
  # mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
# subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
# determine if any significantly differentially expressed genes match those identified by gasch et al 2000
  dosSensRes <- list()
  dosSensRes$Sig <- dplyr::inner_join(dosageSensitiveGenes,resDiSig,by=c("GENEID"="GENEID"))
  dosSensRes$NonSig <- dplyr::inner_join(dosageSensitiveGenes,resDiNonSig,by=c("GENEID"="GENEID"))
  return(dosSensRes)
}


dosSensGenesMA152 <- resultsDosSens(ddsDefMAnewS152,"152","Anc",genesMAnewS152)#,"chrI")
dosSensGenesSigMA152 <- dosSensGenesMA152$Sig
dosSensGenesNonSigMA152 <- dosSensGenesMA152$NonSig

write.csv(dosSensGenesMA152$Sig,file="sigEdosSensGenesMA152.csv")
write.csv(dosSensGenesMA152$NonSig,file="nonSigdosSensGenesMA152.csv")

dosSensGenesGC11 <- resultsDosSens(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11)#,"chrI")
dosSensGenesSigGC11 <- dosSensGenesGC11$Sig
write.csv(dosSensGenesGC11$Sig,file="sigdosSensGenesGC11.csv")
write.csv(dosSensGenesGC11$NonSig,file="nondosSensGenesGC11.csv")

dosSensGenesGC18chrI <- resultsDosSens(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18)#,"chrI")
dosSensGenesSigGC18 <- dosSensGenesGC18chrI$Sig
write.csv(dosSensGenesGC18chrI$Sig,file="sigdosSensGenesGC18.csv")
write.csv(dosSensGenesGC18chrI$NonSig,file="nonSigdosSensGenesGC18.csv")

dosSensGenesGC7chrI <- resultsDosSens(ddsDefGC_7,"Holly_7","Holly_Anc",genesGC_7)#,"chrI")
dosSensGenesSigGC7 <- dosSensGenesGC7chrI$Sig
write.csv(dosSensGenesGC7chrI$Sig,file="sigdosSensGenesGC7.csv")
write.csv(dosSensGenesGC7chrI$NonSig,file="nonSigdosSensGenesGC7.csv")

dosSensGenesMA123chrV <- resultsDosSens(ddsDefMAnewS123,"123","Anc",genesMAnewS123)#,"chrV")
dosSensGenesSigMA123 <- dosSensGenesMA123chrV$Sig
write.csv(dosSensGenesMA123chrV$Sig,file="sigdosSensGenesMA123.csv")
write.csv(dosSensGenesMA123chrV$NonSig,file="nonSigdosSensGenesMA123.csv")

dosSensGenesGC4chrV <- resultsDosSens(ddsDefGC_4,"Holly_4","Holly_Anc",genesGC_4)#,"chrV")
dosSensGenesSigGC4 <- dosSensGenesGC4chrV$Sig
write.csv(dosSensGenesGC4chrV$Sig,file="sigdosSensGenesGC4.csv")
write.csv(dosSensGenesGC4chrV$NonSig,file="nonSigdosSensGenesGC4.csv")

dosSensGenesGC49chrV <- resultsDosSens(ddsDefGC_49,"Holly_49","Holly_Anc",genesGC_49)#,"chrV")
dosSensGenesSigGC49 <- dosSensGenesGC49chrV$Sig
write.csv(dosSensGenesGC49chrV$Sig,file="sigdosSensGenesGC49.csv")
write.csv(dosSensGenesGC49chrV$NonSig,file="nonSigdosSensGenesGC49.csv")

dosSensGenesMA117chrV <- resultsDosSens(ddsDefMAnewS117,"117","Anc",genesMAnewS117)#,"chrV")
dosSensGenesSigMA117 <- dosSensGenesMA117chrV$Sig
write.csv(dosSensGenesMA117chrV$Sig,file="sigdosSensGenesMA117.csv")
write.csv(dosSensGenesMA117chrV$NonSig,file="nonSigdosSensGenesMA117.csv")

dosSensGenesGC59chrVII <- resultsDosSens(ddsDefGC_59,"Holly_59","Holly_Anc",genesGC_59)#,"chrVII")
dosSensGenesSigGC59 <- dosSensGenesGC59chrVII$Sig
write.csv(dosSensGenesGC59chrVII$Sig,file="sigdosSensGenesGC59.csv")
write.csv(dosSensGenesGC59chrVII$NonSig,file="nonSigdosSensGenesGC59.csv")

dosSensGenesGC61chrVII <- resultsDosSens(ddsDefGC_61,"Holly_61","Holly_Anc",genesGC_61)#,"chrVII")
dosSensGenesSigGC61 <- dosSensGenesGC61chrVII$Sig
write.csv(dosSensGenesGC61chrVII$Sig,file="sigdosSensGenesGC61.csv")
write.csv(dosSensGenesGC61chrVII$NonSig,file="nonSigdosSensGenesGC61.csv")

dosSensGenesMA29chrIX <- resultsDosSens(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
dosSensGenesSigMA29 <- dosSensGenesMA29chrIX$Sig
write.csv(dosSensGenesMA29chrIX$Sig,file="sigdosSensGenesMA29.csv")
write.csv(dosSensGenesMA29chrIX$NonSig,file="nonSigdosSensGenesMA29.csv")

dosSensGenesMA119chrIX <- resultsDosSens(ddsDefMAold_119,"SC119","SCA",genesMAoldS119)#,"chrIX")
dosSensGenesSigMA119 <- dosSensGenesMA119chrIX$Sig
write.csv(dosSensGenesMA119chrIX$Sig,file="sigdosSensGenesMA119.csv")
write.csv(dosSensGenesMA119chrIX$NonSig,file="nonSigdosSensGenesMA119.csv")

dosSensGenesMA88chrIX <- resultsDosSens(ddsDefMAold_88,"SC088","SCA",genesMAoldS88)#,"chrIX")
dosSensGenesSigMA88 <- dosSensGenesMA88chrIX$Sig
write.csv(dosSensGenesMA88chrIX$Sig,file="sigdosSensGenesMA88.csv")
write.csv(dosSensGenesMA88chrIX$NonSig,file="nonSigdosSensGenesMA88.csv")

dosSensGenesMA108chrVIII <- resultsDosSens(ddsDefMAold_108,"SC108","SCA",genesMAoldS108)#,"chrIX")
dosSensGenesSigMA108 <- dosSensGenesMA108chrVIII$Sig
write.csv(dosSensGenesMA108chrVIII$Sig,file="sigdosSensGenesMA108.csv")
write.csv(dosSensGenesMA108chrVIII$NonSig,file="nonSigdosSensGenesMA108.csv")

dosSensGenesMA15chrIX <- resultsDosSens(ddsDefMAold_15,"SC015","SCA",genesMAoldS15)#,"chrIX")
dosSensGenesSigMA15 <- dosSensGenesMA15chrIX$Sig
write.csv(dosSensGenesMA15chrIX$Sig,file="sigdosSensGenesMA15.csv")
write.csv(dosSensGenesMA15chrIX$NonSig,file="nonSigdosSensGenesMA15.csv")

dosSensGenesMA29chrIX <- resultsDosSens(ddsDefMAnewS29,"29","Anc",genesMAnewS29)#,"chrIX")
dosSensGenesSigMA29 <- dosSensGenesMA29chrIX$Sig
write.csv(dosSensGenesMA29chrIX$Sig,file="sigdosSensGenesMA29.csv")
write.csv(dosSensGenesMA29chrIX$NonSig,file="nonSigdosSensGenesMA29.csv")

dosSensGenesGC76chrIX <- resultsDosSens(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76)#,"chrIX")
dosSensGenesSigGC76 <- dosSensGenesGC76chrIX$Sig
write.csv(dosSensGenesGC76chrIX$Sig,file="sigdosSensGenesGC76.csv")
write.csv(dosSensGenesGC76chrIX$NonSig,file="nonSigdosSensGenesGC76.csv")

dosSensGenesGC77chrXII <- resultsDosSens(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_76)#,"chrIX")
dosSensGenesSigGC77 <- dosSensGenesGC77chrXII$Sig
write.csv(dosSensGenesGC77chrXII$Sig,file="sigdosSensGenesGC77.csv")
write.csv(dosSensGenesGC77chrXII$NonSig,file="nonSigdosSensGenesGC77.csv")

dosSensGenesMA9chrXIV <- resultsDosSens(ddsDefMAold_9,"SC009","SCA",genesMAoldS9)#,"chrXIV")
dosSensGenesSigMA9 <- dosSensGenesMA9chrXIV$Sig
write.csv(dosSensGenesMA9chrXIV$Sig,file="sigdosSensGenesMA9.csv")
write.csv(dosSensGenesMA9chrXIV$NonSig,file="nonSigdosSensGenesMA9.csv")

dosSensGenesMA11 <- resultsDosSens(ddsDefMAold_11,"SC011","SCA",genesMAoldS11)#,"chrXIV")
dosSensGenesSigMA11 <- dosSensGenesMA11$Sig
write.csv(dosSensGenesMA11$Sig,file="sigdosSensGenesMA11.csv")
write.csv(dosSensGenesMA11$NonSig,file="nonSigdosSensGenesMA11.csv")

dosSensGenesMA112chrXVI <- resultsDosSens(ddsDefMAnewS112,"112","Anc",genesMAnewS112)#,"chrXVI")
dosSensGenesSigMA112 <- dosSensGenesMA112chrXVI$Sig
write.csv(dosSensGenesMA112chrXVI$Sig,file="sigdosSensGenesMA112.csv")
write.csv(dosSensGenesMA112chrXVI$NonSig,file="nonSigdosSensGenesMA112.csv")

dosSensGenesGC8chrXVI <- resultsDosSens(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8)#,"chrXVI")
dosSensGenesSigGC8 <- dosSensGenesGC8chrXVI$Sig
write.csv(dosSensGenesGC8chrXVI$Sig,file="sigdosSensGenesGC8.csv")
write.csv(dosSensGenesGC8chrXVI$NonSig,file="nonSigdosSensGenesGC8.csv")

### need to do the euploid samples next - dosage sensitive genes
dosSensGenesGC1 <- resultsDosSens(ddsDefGC_1,"Holly_1","Holly_Anc",genesGC_1)
dosSensGenesSigGC1 <- dosSensGenesGC1$Sig

dosSensGenesGC2 <- resultsDosSens(ddsDefGC_2,"Holly_2","Holly_Anc",genesGC_2)
dosSensGenesSigGC2 <- dosSensGenesGC2$Sig
write.csv(dosSensGenesGC2$Sig,file="sigdosSensGenesGC2.csv")
write.csv(dosSensGenesGC2$NonSig,file="nonSigdosSensGenesGC2.csv")

dosSensGenesGC3 <- resultsDosSens(ddsDefGC_3,"Holly_3","Holly_Anc",genesGC_3)
dosSensGenesSigGC3 <- dosSensGenesGC3$Sig
write.csv(dosSensGenesGC3$Sig,file="sigdosSensGenesGC3.csv")
write.csv(dosSensGenesGC3$NonSig,file="nonSigdosSensGenesGC3.csv")

dosSensGenesGC5 <- resultsDosSens(ddsDefGC_5,"Holly_5","Holly_Anc",genesGC_5)
dosSensGenesSigGC5 <- dosSensGenesGC5$Sig
write.csv(dosSensGenesGC5$Sig,file="sigdosSensGenesGC5.csv")
write.csv(dosSensGenesGC5$NonSig,file="nonSigdosSensGenesGC5.csv")

# esrGenesGC6 <- resultsESR(ddsDefGC_6,"Holly_6","Holly_Anc",genesGC_6)
# write.csv(esrGenesGC6$Sig,file="sigESRGenesGC6.csv")
# write.csv(esrGenesGC6$NonSig,file="nonSigESRGenesGC6.csv")

dosSensGenesGC9 <- resultsDosSens(ddsDefGC_9,"Holly_9","Holly_Anc",genesGC_9)
dosSensGenesSigGC9 <- dosSensGenesGC9$Sig
write.csv(dosSensGenesGC9$Sig,file="sigdosSensGenesGC9.csv")
write.csv(dosSensGenesGC9$NonSig,file="nonSigdosSensGenesGC9.csv")

dosSensGenesMA1 <- resultsDosSens(ddsDefMAold_1,"SC001","SCA",genesMAold_1)
dosSensGenesSigMA1 <- dosSensGenesMA1$Sig
write.csv(dosSensGenesMA1$Sig,file="sigdosSensGenesMA1.csv")
write.csv(dosSensGenesMA1$NonSig,file="nonSigdosSensGenesMA1.csv")

dosSensGenesMA2 <- resultsDosSens(ddsDefMAold_2,"SC002","SCA",genesMAold_2)
dosSensGenesSigMA2 <- dosSensGenesMA2$Sig
write.csv(dosSensGenesMA2$Sig,file="sigdosSensGenesMA2.csv")
write.csv(dosSensGenesMA2$NonSig,file="nonSigdosSensGenesMA2.csv")

dosSensGenesMA3 <- resultsDosSens(ddsDefMAold_3,"SC003","SCA",genesMAold_3)
dosSensGenesSigMA3 <- dosSensGenesMA3$Sig
write.csv(dosSensGenesMA3$Sig,file="sigdosSensGenesMA3.csv")
write.csv(dosSensGenesMA3$NonSig,file="nonSigdosSensGenesMA3.csv")

dosSensGenesMA4 <- resultsDosSens(ddsDefMAold_4,"SC004","SCA",genesMAold_4)
dosSensGenesSigMA4 <- dosSensGenesMA4$Sig
write.csv(dosSensGenesMA4$Sig,file="sigdosSensGenesMA4.csv")
write.csv(dosSensGenesMA4$NonSig,file="nonSigdosSensGenesMA4.csv")

dosSensGenesMA5 <- resultsDosSens(ddsDefMAold_5,"SC005","SCA",genesMAold_5)
dosSensGenesSigMA5 <- dosSensGenesMA5$Sig
write.csv(dosSensGenesMA5$Sig,file="sigdosSensGenesMA5.csv")
write.csv(dosSensGenesMA5$NonSig,file="nonSigdosSensGenesMA5.csv")

dosSensGenesMA7 <- resultsDosSens(ddsDefMAold_7,"SC007","SCA",genesMAold_7)
dosSensGenesSigMA7 <- dosSensGenesMA7$Sig
write.csv(dosSensGenesMA7$Sig,file="sigdosSensGenesMA7.csv")
write.csv(dosSensGenesMA7$NonSig,file="nonSigdosSensGenesMA7.csv")

dosSensGenesMA8 <- resultsDosSens(ddsDefMAold_8,"SC008","SCA",genesMAold_8)
dosSensGenesSigMA8 <- dosSensGenesMA8$Sig
write.csv(dosSensGenesMA8$Sig,file="sigdosSensGenesMA8.csv")
write.csv(dosSensGenesMA8$NonSig,file="nonSigdosSensGenesMA8.csv")

dosSensGenesMA50 <- resultsDosSens(ddsDefMAnewS50,"50","Anc",genesMAnewS50)
dosSensGenesSigMA50 <-  dosSensGenesMA50$Sig
write.csv(dosSensGenesMA50$Sig,file="sigdosSensGenesMA50.csv")
write.csv(dosSensGenesMA50$NonSig,file="nonSigdosSensGenesMA50.csv")

dosSensResMA115 <- resultsDosSens(ddsDefMAnewS115,"115","Anc",genesMAnewS115)
dosSensGenesSigMA115 <- dosSensResMA115$Sig
write.csv(dosSensResMA115$Sig,file="sigDosSensGenesMA115.csv")
write.csv(dosSensResMA115$NonSig,file="nonSigDosSensGenesMA115.csv")


### find commonly DE dosage Sensitive genes in euploid GC samples
dosSensGenesCommGCEup <- Reduce(intersect, list(dosSensGenesSigGC1,dosSensGenesSigGC2,dosSensGenesSigGC3,dosSensGenesSigGC5,dosSensGenesSigGC9))
dosSensGenesCommGCEup
## no commonly found sig DE dosage Sensitive genes in all GC euploids

dosSensGenesCommMAoldEup <- Reduce(intersect, list(dosSensGenesSigMA1,dosSensGenesSigMA2,dosSensGenesSigMA3,dosSensGenesSigMA4,dosSensGenesSigMA5,dosSensGenesSigMA7,dosSensGenesSigMA8))
dosSensGenesCommMAoldEup

dosSensGenesCommMAnewEup <- Reduce(intersect,list(dosSensGenesSigMA50,dosSensGenesSigMA115))
# also 0

### see if there are any ASR genes commonly found in All Aneuploid GC samples
dosSensGenesCommGCAneu <- Reduce(intersect,list(dosSensGenesSigGC11,dosSensGenesSigGC18,dosSensGenesSigGC4,dosSensGenesSigGC49,dosSensGenesSigGC59,dosSensGenesSigGC61,dosSensGenesSigGC7,dosSensGenesSigGC76,dosSensGenesSigGC77,dosSensGenesSigGC8))
# no commonly sig DE ASR genes in aneuploid GC samples

##old ma samples
dosSensGenesCommMAOldAneu <- Reduce(intersect,list(dosSensGenesSigMA108,dosSensGenesSigMA119,dosSensGenesSigMA15,dosSensGenesSigMA88))

dosSensGenesCommMANewAneu <- Reduce(intersect,list(dosSensGenesSigMA112,dosSensGenesSigMA117,dosSensGenesSigMA123,dosSensGenesSigMA152,dosSensGenesSigMA29))

# 0 in MA old and new samples too

### command to find common elements in several files: 1,2,3,4, + files
# want to know how many genes are differentially expressed commonly in 1,2,3 etc EUPLOID lines
#
  # put all ANEUPLOID samples in same data frame
    dosSensGenesAneuGCSamplesExpr <- rbind(dosSensGenesSigGC11,dosSensGenesSigGC18,dosSensGenesSigGC4,dosSensGenesSigGC49,dosSensGenesSigGC59,dosSensGenesSigGC61,dosSensGenesSigGC7,dosSensGenesSigGC76,dosSensGenesSigGC77,dosSensGenesSigGC8)
      # # first need to filter allSamplesExpr based on p-adj level
      # eupMAnewSamplesDiffExpr <- subset(eupMAnewSamplesExpr, padj<0.1)
      # #
      #
        # then ask how many times each GENEID  is found in eupSamplesExpr
        # library(plyr)
        dosSensGeneGCFreqAneu <- plyr::count(dosSensGenesAneuGCSamplesExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numGCAneuDosSensGenesDE <- plyr::count(dosSensGeneGCFreqAneu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numGCAneuDosSensGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numGCAneuDosSensGenesDE, file="numb_comm_DE_GC_DosSens_genesANEU.csv")   

### command to find common elements in several files: 1,2,3,4, + files
# want to know how many genes are differentially expressed commonly in 1,2,3 etc ANEUPLOID lines
#
  # put all ANEUPLOID samples in same data frame
    dosSensGenesAneuMAoldSamplesExpr <- rbind(dosSensGenesSigMA108,dosSensGenesSigMA119,dosSensGenesSigMA15,dosSensGenesSigMA88)
      # # first need to filter allSamplesExpr based on p-adj level
      # eupMAnewSamplesDiffExpr <- subset(eupMAnewSamplesExpr, padj<0.1)
      # #
      #
        # then ask how many times each GENEID  is found in eupSamplesExpr
        # library(plyr)
        dosSensGeneMAoldFreqAneu <- plyr::count(dosSensGenesAneuMAoldSamplesExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numMAoldAneudosSensGenesDE <- plyr::count(dosSensGeneMAoldFreqAneu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numMAoldAneudosSensGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numMAoldAneudosSensGenesDE, file="numb_comm_DE_MAold_dosSens_genesANEU.csv")


### command to find common elements in several files: 1,2,3,4, + files
# want to know how many genes are differentially expressed commonly in 1,2,3 etc ANEUPLOID lines
#
  # put all ANEUPLOID samples in same data frame
    dosSensGenesAneuMAnewSamplesExpr <- rbind(dosSensGenesSigMA112,dosSensGenesSigMA29,dosSensGenesSigMA117,dosSensGenesSigMA123,dosSensGenesSigMA152)
      # # first need to filter allSamplesExpr based on p-adj level
      # eupMAnewSamplesDiffExpr <- subset(eupMAnewSamplesExpr, padj<0.1)
      # #
      #
        # then ask how many times each GENEID  is found in eupSamplesExpr
        # library(plyr)
        dosSensGeneMAnewFreqAneu <- plyr::count(dosSensGenesAneuMAnewSamplesExpr, c("GENEID"))
        #
          # want to know how many genes are DE in 1, 2, 3, 4 , etc samples
          numMAnewAneudosSensGenesDE <- plyr::count(dosSensGeneMAnewFreqAneu,"freq")
          #
            # rename columns so that I know what is going on: column 1 is how many samples a gene is DE               in, column 2 is how many genes are in that category
            colnames(numMAnewAneudosSensGenesDE) <- c("numSamples", "numGenes")
            #
              # export as .csv to use in excel or whatever
              write.csv(numMAnewAneudosSensGenesDE, file="numb_comm_DE_MAnew_dosSens_genesANEU.csv")



anChr = "chrXII"
dds=ddsDefGC_18
anc="Holly_Anc"
genes="genesGC_18"

resultsMonosomy <- function(dds,sample,anc,genes,anChr) {
# make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  remove1 <- anChr
  genes <- genes[(grep(paste0(remove1,"$"),genes$TXCHROM,perl=TRUE)),]
  all(rownames(dds) == genes$GENEID)
  # genes <- genes$GENEID
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  # ddsGenes <- intersect(rownames(dds),genes)
  # as.data.frame(genes)
  # ddsGenes <- genes[,ddsGenes]
  # length(ddsHTSeqMAold)
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  resMon <- results(dds,contrast=c("condition",sample,anc),lfcThreshold=1, altHypothesis="less")
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (NO DC))
  resMonNonSig <- as.data.frame(subset(resMon,padj>0.1))
  resMonNonSig <- cbind(names = rownames(resMonNonSig), resMonNonSig)
  resMonNonSig <- dplyr::inner_join(genes,resMonNonSig,by=c("GENEID"="names"))
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resMonSig <- as.data.frame(subset(resMon,padj<0.1))
  resMonSig <- cbind(names = rownames(resMonSig), resMonSig)
  resMonSig <- dplyr::inner_join(genes,resMonSig,by=c("GENEID"="names"))
# put non-significantly different genes in a data table
  results <- list()
  results$resNoDC <- resMonNonSig
  results$resPotDC <- resMonSig
  # need to pull out genes from resPotDC and use them to subset our ddsDataSet so we can run DESeq on only those genes
#     genesPotDC <- resTriNonSig$GENEID
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# # genesPotDC <- genes[(grep(paste0(genesPotDC,"$"),genesGC_18$GENEID,perl=TRUE)),]
# matchedS2 <- intersect(rownames(ddsCopy), genesPotDC)
# keep4 <- rownames(ddsCopy) %in% matchedS2
# ddsCopy <- ddsCopy[keep4,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsCopy) == genesPotDC)
# mcols(ddsCopy) <- cbind(mcols(ddsCopy), genesPotDC)
# ddsCopy <- DESeq(ddsCopy)
# View(counts(ddsCopy))
  #### run same analyses using new ddsCopy dataset and new parameters
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(ddsCopy,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  ### Find genes that are Signficiantly different than Trisomic expectation and are Significantly different from Disomic expectation
  resPartOvDC <- dplyr::inner_join(resMonSig,resDiSig,by="GENEID")
  results$resPartOvDC <- resPartOvDC
  ### Find genes that are Signficiantly different than Trisomic expectation but NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resMonSig,resDiNonSig,by="GENEID")
  results$resFullDC <- resFullDC
  transGenesDE <- dplyr::anti_join(resDiSig,genes,by=c("GENEID"="TXCHROM"))
  results$transGenesDE <- transGenesDE
  return(results)
}

resultsTetrasomy <- function(dds,sample,anc,genes,anChr) {
# make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  #library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  #txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  remove1 <- anChr
  genes <- genes[(grep(paste0(remove1,"$"),genes$TXCHROM,perl=TRUE)),]
  all(rownames(dds) == genes$GENEID)
  # genes <- genes$GENEID
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  # ddsGenes <- intersect(rownames(dds),genes)
  # as.data.frame(genes)
  # ddsGenes <- genes[,ddsGenes]
  # length(ddsHTSeqMAold)
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  resTetra <- results(dds,contrast=c("condition",sample,anc),lfcThreshold=1, altHypothesis="greater")
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (NO DC))
  resTetraNonSig <- as.data.frame(subset(resTetra,padj>0.1))
  resTetraNonSig <- cbind(names = rownames(resTetraNonSig), resTetraNonSig)
  resTetraNonSig <- dplyr::inner_join(genes,resTetraNonSig,by=c("GENEID"="names"))
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resTetraSig <- as.data.frame(subset(resTetra,padj<0.1))
  resTetraSig <- cbind(names = rownames(resTetraSig), resTetraSig)
  resTetraSig <- dplyr::inner_join(genes,resTetraSig,by=c("GENEID"="names"))
# put non-significantly different genes in a data table
  results <- list()
  results$resNoDC <- resTetraNonSig
  results$resPotDC <- resTetraSig
  # need to pull out genes from resPotDC and use them to subset our ddsDataSet so we can run DESeq on only those genes
#     genesPotDC <- resTriNonSig$GENEID
# #need to remove the ones in rownames of dds that don't match genes$GENEID
# # genesPotDC <- genes[(grep(paste0(genesPotDC,"$"),genesGC_18$GENEID,perl=TRUE)),]
# matchedS2 <- intersect(rownames(ddsCopy), genesPotDC)
# keep4 <- rownames(ddsCopy) %in% matchedS2
# ddsCopy <- ddsCopy[keep4,]
# # length(ddsHTSeqMAnew)
# all(rownames(ddsCopy) == genesPotDC)
# mcols(ddsCopy) <- cbind(mcols(ddsCopy), genesPotDC)
# ddsCopy <- DESeq(ddsCopy)
# View(counts(ddsCopy))
  #### run same analyses using new ddsCopy dataset and new parameters
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(ddsCopy,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  ### Find genes that are Signficiantly different than Trisomic expectation and are Significantly different from Disomic expectation
  resPartOvDC <- dplyr::inner_join(resTetraSig,resDiSig,by="GENEID")
  results$resPartOvDC <- resPartOvDC
  ### Find genes that are Signficiantly different than Trisomic expectation but NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resTetraSig,resDiNonSig,by="GENEID")
  results$resFullDC <- resFullDC
  transGenesDE <- dplyr::anti_join(resDiSig,genes,by=c("GENEID"="TXCHROM"))
  results$transGenesDE <- transGenesDE
  return(results)
}


#### GC Samples
# resultsNames(ddsDefGC_11)
sampleResults11GCXV <- resultsTrisomy(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11,"chrXV")
write.csv(sampleResults11GCXV$resNoDC,file="sample11GCchrXVnoDCgenes.csv")
# want % of genes on this chromosome that are not dosage compensated
write.csv(sampleResults11GCXV$resPartOvDC,file="sample11GCchrXVPotDCgenes.csv")
write.csv(sampleResults11GCXV$resFullDC,file="sample11GCchrXVFullDCgenes.csv")
write.csv(sampleResults11GCXV$transGenesDE,file="sample11GCchrXVtransDEgenes.csv")

sampleResults4V <- resultsTrisomy(ddsDefGC_4,"Holly_4","Holly_Anc",genesGC_4,"chrV")
write.csv(sampleResults4V$resNoDC,file="sample4GCchrVnoDCgenes.csv")
# want % of genes on this chromosome that are not dosage compensated
write.csv(sampleResults4V$resPartOvDC,file="sample4GCchrVPotDCgenes.csv")
write.csv(sampleResults4V$resFullDC,file="sample4GCchr4VFullDCgenes.csv")
write.csv(sampleResults4V$transGenesDE,file="sample4GCchrVtransDEgenes.csv")


sampleResults18GCI <- resultsTrisomy(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18,"chrI")
  # perNotDCsample18GC <- 81/82
  # perPartDCsample18GC <- 1/82
  # rm(partDCgenesSample18GC)
write.csv(sampleResults18GCI$resNoDC,file="sample18GCnoDCgenesI.csv")
write.csv(sampleResults18GCI$resPartOvDC,file="sample18GCPartOverDCgenesI.csv")
write.csv(sampleResults18GCI$resFullDC,file="sample18GCfullDCgenesI.csv")
write.csv(sampleResults18GCI$transGenesDE,file="sample18GCtransDEgenesI.csv")

sampleResults18GCXII <- resultsTrisomy(ddsDefGC_18,"Holly_18","Holly_Anc",genesGC_18,"chrXII")
  # perNotDCsample18GC <- 81/82
  # perPartDCsample18GC <- 1/82
  # rm(partDCgenesSample18GC)
write.csv(sampleResults18GCXII$resNoDC,file="sample18GCnoDCgenesXII.csv")
write.csv(sampleResults18GCXII$resPartOvDC,file="sample18GCPartOverDCgenesXII.csv")
write.csv(sampleResults18GCXII$resFullDC,file="sample18GCfullDCgenesXII.csv")
write.csv(sampleResults18GCXII$transGenesDE,file="sample18GCtransDEgenesXII.csv")

sampleResults49GC <- resultsTrisomy(ddsDefGC_49,"Holly_49","Holly_Anc",genesGC_49,"chrV")
  # perNotDCsample49GC <- 272/272
write.csv(sampleResults49GC$resNoDC,file="sample49GCnoDCgenesV.csv")
write.csv(sampleResults49GC$resPartOvDC,file="sample49PartOverDCgenesV.csv")
write.csv(sampleResults49GC$resFullDC,file="sample49FullDCgenesV.csv")
write.csv(sampleResults49GC$transGenesDE,file="sample49GCtransDEgenesV.csv")

sampleResults59GC <- resultsTrisomy(ddsDefGC_59,"Holly_59","Holly_Anc",genesGC_59,"chrVII")
  # perNotDCsample59GC <- 497/(497+14)
  # perPartDCsample59GC <- 14/(497+14)
write.csv(sampleResults59GC$resNoDC,file="sample59GCnoDCgenesVII.csv")
write.csv(sampleResults59GC$resPartOvDC,file="sample59GCPartOverGenesVII.csv")
write.csv(sampleResults59GC$resFullDC,file="sample59GCFullDCGenesVII.csv")
write.csv(sampleResults59GC$transGenesDE,file="sample59GCtransDEgenesVII.csv")

sampleResults61GC <- resultsTrisomy(ddsDefGC_61,"Holly_61","Holly_Anc",genesGC_61,"chrVII")
  # percNotDCsample61GC <- 489/(489+20)
  # perPartDCsample61GC <- 20/(489+20)
  # partDCgenesSample61GC <-cbind(sampleResults61GC$resPartOvDC$GENEID,sampleResults61GC$resPartOvDC$log2FoldChange.x)
# colnames(partDCgenesSample61GC) <- c("GENEID","log2FoldChange")
write.csv(sampleResults61GC$resNoDC,file="sample61GCnoDCgenesVII.csv")
write.csv(sampleResults61GC$resPartOvDC, file="sample61GCpartOverDCgenesVII.csv")
write.csv(sampleResults61GC$resFullDC,file="sample61GCfullDCgenesVII.csv")
write.csv(sampleResults61GC$transGenesDE,file="sample61GCtransDEgenesVII.csv")

sampleResults76GCIX <- resultsTrisomy(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76,"chrIX")
  # perNotDCsample76GCIX <- 200/203
  # perPartDCsample76GCIX <- 3/203
write.csv(sampleResults76GCIX$resNoDC,file="sample76GCnoDCgenesIX.csv")
write.csv(sampleResults76GCIX$resPartOvDC, file="sample76GCpartOverDCgenesIX.csv")
write.csv(sampleResults76GCIX$resFullDC,file="sample76GCfullDCgenesIX.csv")
write.csv(sampleResults76GCIX$transGenesDE,file="sample76GCtransDEgenesIX.csv")

sampleResults76GCX <- resultsTrisomy(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76,"chrX")
  # perNotDCsample76GCXII <- 471/473
  # perPartDCsample76GCXII <- 2/471
write.csv(sampleResults76GCX$resNoDC,file="sample76GCnoDCgenesX.csv")
write.csv(sampleResults76GCX$resPartOvDC, file="sample76GCpartOverDCgenesX.csv")
write.csv(sampleResults76GCX$resFullDC,file="sample76GCfullDCgenesX.csv")
write.csv(sampleResults76GCX$transGenesDE,file="sample76GCtransDEgenesX.csv")

sampleResults76GCXIV <- resultsTrisomy(ddsDefGC_76,"Holly_76","Holly_Anc",genesGC_76,"chrXIV")
  # perNotDCsample76GCXII <- 471/473
  # perPartDCsample76GCXII <- 2/471
write.csv(sampleResults76GCXIV$resNoDC,file="sample76GCnoDCgenesXIV.csv")
write.csv(sampleResults76GCXIV$resPartOvDC, file="sample76GCpartOverDCgenesXIV.csv")
write.csv(sampleResults76GCXIV$resFullDC,file="sample76GCfullDCgenesXIV.csv")
write.csv(sampleResults76GCXIV$transGenesDE,file="sample76GCtransDEgenesXIV.csv")

sampleResults7GC <- resultsTrisomy(ddsDefGC_7,"Holly_7","Holly_Anc",genesGC_7,"chrI")
  # perNotDCsample7GC <- 83/83
write.csv(sampleResults7GC$resNoDC,file="sample7GCnoDCgenesI.csv")
write.csv(sampleResults7GC$resPartOvDC, file="sample7GCpartOverDCgenesI.csv")
write.csv(sampleResults7GC$resFullDC,file="sample7GCfullDCgenesI.csv")
write.csv(sampleResults7GC$transGenesDE,file="sample7GCtransDEgenesI.csv")

sampleResults77GC <- resultsTrisomy(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_77,"chrXIV")
  # perNotDCsample77GC <- 383/384
  # perPartDCsample77GC <- 1/384
write.csv(sampleResults77GC$resNoDC,file="sample77GCnoDCgenesXIV.csv")
write.csv(sampleResults77GC$resPartOvDC, file="sample77GCpartOverDCgenesXIV.csv")
write.csv(sampleResults77GC$resFullDC,file="sample77GCfullDCgenesXIV.csv")
write.csv(sampleResults77GC$transGenesDE,file="sample177GCtransDEgenesXIV.csv")

sampleResults8GC <- resultsTetrasomy(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8,"chrXVI")
write.csv(sampleResults8GC$resNoDC,file="sample8GCnoDCgenesXVI.csv")
write.csv(sampleResults8GC$resPartOvDC, file="sample8GCpartOverDCgenesXVI.csv")
write.csv(sampleResults8GC$resFullDC,file="sample8GCfullDCgenesXVI.csv")
write.csv(sampleResults8GC$transGenesDE,file="sample8GCtransDEgenesXVI.csv")

sampleResults11GCI <- resultsMonosomy(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11,"chrI")
  # perNoDCsample11GCI <- 78/79
  # perPartDCsample11GCI <- 1/79
write.csv(sampleResults11GCI$resNoDC,file="sample11GCnoDCgenesI.csv")
write.csv(sampleResults11GCI$resPartOvDC, file="sample11GCpartOverDCgenesI.csv")
write.csv(sampleResults11GCI$resFullDC,file="sample11GCfullDCgenesI.csv")
write.csv(sampleResults11GCI$transGenesDE,file="sample11GCtransDEgenesI.csv")
# perNoDCGCSamples <- c(perNoDCsample11GCI,perNotDCsample11GCXV)
# View(sampleResults11GCI$resPartOvDC)
# View(sampleResults11GCI$resFullDC)
# View(sampleResults11GCI$resNoDC)


#### MA Old Samples
# resultsNames(ddsDefMAold)
sampleResults9MAoldXIV <- resultsTrisomy(ddsDefMAold_9,"SC009","SCA",genesMAold_9,"chrXIV")
write.csv(sampleResults9MAoldXIV$resNoDC,file="sample9MAnoDCgenesXIV.csv")
write.csv(sampleResults9MAoldXIV$resPartOvDC, file="sample9MApartOverDCgenesXIV.csv")
write.csv(sampleResults9MAoldXIV$resFullDC,file="sample9MAfullDCgenesXIV.csv")
write.csv(sampleResults9MAoldXIV$transGenesDE,file="sample9MAtransDEgenesXIV.csv")

sampleResults15MAoldIX <- resultsTrisomy(ddsDefMAold_15,"SC015","SCA",genesMAold_15,"chrIX")
write.csv(sampleResults15MAoldIX$resNoDC,file="sample15MAnoDCgenesIX.csv")
write.csv(sampleResults15MAoldIX$resPartOvDC, file="sample15MApartOverDCgenesIX.csv")
write.csv(sampleResults15MAoldIX$resFullDC,file="sample15MAfullDCgenesIX.csv")
write.csv(sampleResults15MAoldIX$transGenesDE,file="sample15MAtransDEgenesIX.csv")

sampleResults88MAoldIX <- resultsTrisomy(ddsDefMAold_88,"SC088","SCA",genesMAold_88,"chrIX")
write.csv(sampleResults88MAoldIX$resNoDC,file="sample88MAnoDCgenesIX.csv")
write.csv(sampleResults88MAoldIX$resPartOvDC, file="sample88MApartOverDCgenesIX.csv")
write.csv(sampleResults88MAoldIX$resFullDC,file="sample88MAfullDCgenesIX.csv")
write.csv(sampleResults88MAoldIX$transGenesDE,file="sample88MAtransDEgenesIX.csv")

sampleResults108MAoldVIII <- resultsTrisomy(ddsDefMAold_108,"SC108","SCA",genesMAold_108,"chrVIII")
write.csv(sampleResults108MAoldVIII$resNoDC,file="sample108MAnoDCgenesVIII.csv")
write.csv(sampleResults108MAoldVIII$resPartOvDC, file="sample108MApartOverDCgenesVIII.csv")
write.csv(sampleResults108MAoldVIII$resFullDC,file="sample108MAfullDCgenesVIII.csv")
write.csv(sampleResults108MAoldVIII$transGenesDE,file="sample108MAtransDEgenesVIII.csv")

sampleResults119MAoldIX <- resultsTrisomy(ddsDefMAold_119,"SC119","SCA",genesMAold_119,"chrIX")
write.csv(sampleResults119MAoldIX$resNoDC,file="sample119MAnoDCgenesIX.csv")
write.csv(sampleResults119MAoldIX$resPartOvDC, file="sample119MApartOverDCgenesIX.csv")
write.csv(sampleResults119MAoldIX$resFullDC,file="sample119MAfullDCgenesIX.csv")
write.csv(sampleResults119MAoldIX$transGenesDE,file="sample119MAtransDEgenesIX.csv")

sampleResults108MAoldIX <- resultsMonosomy(ddsDefMAold_108,"SC108","SCA",genesMAold_108,"chrIX")
write.csv(sampleResults108MAoldIX$resNoDC,file="sample108MAnoDCgenesIX.csv")
write.csv(sampleResults108MAoldIX$resPartOvDC, file="sample108MApartOverDCgenesIX.csv")
write.csv(sampleResults108MAoldIX$resFullDC,file="sample108MAfullDCgenesIX.csv")
write.csv(sampleResults108MAoldIX$transGenesDE,file="sample108MAtransDEgenesIX.csv")

### MA New Samples
resultsNames(ddsDefMAnewS29)
sampleResults29MANewIX <- resultsMonosomy(ddsDefMAnewS29,"29","Anc",genesMAnewS29,"chrIX")
write.csv(sampleResults29MANewIX$resNoDC,file="sample29MAnoDCgenesIX.csv")
write.csv(sampleResults29MANewIX$resPartOvDC, file="sample29MApartOverDCgenesIX.csv")
write.csv(sampleResults29MANewIX$resFullDC,file="sample29MAfullDCgenesIX.csv")
write.csv(sampleResults29MANewIX$transGenesDE,file="sample29MAtransDEgenesIX.csv")

sampleResults112MANewXVI <- resultsTrisomy(ddsDefMAnewS112,"112","Anc",genesMAnewS112,"chrXVI")
write.csv(sampleResults112MANewXVI$resNoDC,file="sample112MAnoDCgenesXVI.csv")
write.csv(sampleResults112MANewXVI$resPartOvDC, file="sample112MApartOverDCgenesXVI.csv")
write.csv(sampleResults112MANewXVI$resFullDC,file="sample112MAfullDCgenesXVI.csv")
write.csv(sampleResults112MANewXVI$transGenesDE,file="sample112MAtransDEgenesXVI.csv")

sampleResults117MANewV <- resultsTrisomy(ddsDefMAnewS117,"117","Anc",genesMAnewS117,"chrV")
write.csv(sampleResults117MANewV$resNoDC,file="sample117MAnoDCgenesV.csv")
write.csv(sampleResults117MANewV$resPartOvDC, file="sample117MApartOverDCgenesV.csv")
write.csv(sampleResults117MANewV$resFullDC,file="sample117MAfullDCgenesV.csv")
write.csv(sampleResults117MANewV$transGenesDE,file="sample117MAtransDEgenesV.csv")

sampleResults123MANewV <- resultsTrisomy(ddsDefMAnewS123,"123","Anc",genesMAnewS123,"chrV")
write.csv(sampleResults123MANewV$resNoDC,file="sample123MAnoDCgenesV.csv")
write.csv(sampleResults123MANewV$resPartOvDC, file="sample123MApartOverDCgenesV.csv")
write.csv(sampleResults123MANewV$resFullDC,file="sample123MAfullDCgenesV.csv")
write.csv(sampleResults123MANewV$transGenesDE,file="sample123MAtransDEgenesV.csv")

sampleResults152MANewI <- resultsTrisomy(ddsDefMAnewS152,"152","Anc",genesMAnewS152,"chrI")
write.csv(sampleResults152MANewI$resNoDC,file="sample152MAnoDCgenesI.csv")
write.csv(sampleResults152MANewI$resPartOvDC, file="sample152MApartOverDCgenesI.csv")
write.csv(sampleResults152MANewI$resFullDC,file="sample152MAfullDCgenesI.csv")
write.csv(sampleResults152MANewI$transGenesDE,file="sample152MAtransDEgenesI.csv")

sampleResults152MANewVIII <- resultsTrisomy(ddsDefMAnewS152,"152","Anc",genesMAnewS152,"chrVIII")
write.csv(sampleResults152MANewVIII$resNoDC,file="sample152MAnoDCgenesVIII.csv")
write.csv(sampleResults152MANewVIII$resPartOvDC, file="sample152MApartOverDCgenesVIII.csv")
write.csv(sampleResults152MANewVIII$resFullDC,file="sample152MAfullDCgenesVII.csv")
write.csv(sampleResults152MANewVIII$transGenesDE,file="sample152MAtransDEgenesVII.csv")


# resultsTrisomyPartDC <- function(dds,sample,genes) {
#   # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
#   resTri <- results(dds,contrast=c("condition",sample,"Holly_Anc"),lfcThreshold=0.585)
#   #subset data frame to get nonSignificant p values (indicating gene does is as expected)
#   resTriNonSig <- as.data.frame(subset(resTri,padj>0.1))
#   resTriNonSig <- cbind(names = rownames(resTriNonSig), resTriNonSig)
#   resTriNonSig <- dplyr::inner_join(genes,resTriNonSig,by=c("GENEID"="names"))
#   # subset data frame to get Significant p values (indicting gene does is NOT as expected)
#   resTriSig <- as.data.frame(subset(resTri,padj<0.1))
#   resTriSig <- cbind(names = rownames(resTriSig), resTriSig)
#   resTriSig <- dplyr::inner_join(genes,resTriSig,by=c("GENEID"="names"))
#   # run results function with log fold change threshold at 0 (expectation for disomic genes)
#   resDi <- results(dds,contrast=c("condition",sample,"Holly_Anc"),lfcThreshold = 0)
#   # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
#   resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
#   resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
#   resDiNonSig <- dplyr::inner_join(genes,resDiNonSig,by=c("GENEID"="names"))
#   # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
#   resDiSig <- as.data.frame(subset(resDi,padj<0.1))
#   resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
#   resDiSig <- dplyr::inner_join(genes,resDiSig,by=c("GENEID"="names"))
#   ### Find genes that are Signficiantly different than Trisomic expectation and are Significantly different from Disomic expectation
#   resPartOvDC <- dplyr::inner_join(resTriSig,resDiSig,by="GENEID")
#   return(resPartOvDC)
# }
#
# testGC18PartDC <- resultsTrisomyPartDC(dds,sample,genes)
#
# resultsTrisomyFullDC <- function(dds,sample,genes) {
# # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
# resTri <- results(dds,contrast=c("condition",sample,"Holly_Anc"),lfcThreshold=0.585)
# #subset data frame to get nonSignificant p values (indicating gene does is as expected)
# resTriNonSig <- as.data.frame(subset(resTri,padj>0.1))
# resTriNonSig <- cbind(names = rownames(resTriNonSig), resTriNonSig)
# resTriNonSig <- dplyr::inner_join(genes,resTriNonSig,by=c("GENEID"="names"))
# # subset data frame to get Significant p values (indicting gene does is NOT as expected)
# resTriSig <- as.data.frame(subset(resTri,padj<0.1))
# resTriSig <- cbind(names = rownames(resTriSig), resTriSig)
# resTriSig <- dplyr::inner_join(genes,resTriSig,by=c("GENEID"="names"))
# # run results function with log fold change threshold at 0 (expectation for disomic genes)
# resDi <- results(dds,contrast=c("condition",sample,"Holly_Anc"),lfcThreshold = 0)
# # subset data frame to get nonSignificant p values (indicating gene dose is as expected for Disomic genes)
# resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
# resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
# resDiNonSig <- dplyr::inner_join(genes,resDiNonSig,by=c("GENEID"="names"))
# # subset data frame to get nonSignificant p values (indicating gene dose is NOT as expected for Disomic genes)
# resDiSig <- as.data.frame(subset(resDi,padj<0.1))
# resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
# resDiSig <- dplyr::inner_join(genes,resDiSig,by=c("GENEID"="names"))
# ### Find genes that are Signficiantly different than Trisomic expectation but NotSignificantly different from Disomic expectation
# resFullDC <- dplyr::inner_join(resTriSig,resDiNonSig,by="GENEID")
# return(resFullDC)
# }
#
# testGC18FullDC <- resultsTrisomyFullDC(dds,sample,genes)
#
#
#
#
#
#






# res18TESTnonSig <- as.data.frame(subset(res18TEST,padj>0.1))
# res18TESTnonSig <- cbind(names = rownames(res18TESTnonSig), res18TESTnonSig)
# res18TESTnonSig <- dplyr::inner_join(genesGC_18,res18TESTnonSig,by=c("GENEID"="names"))
#
# res18TESTSig <- as.data.frame(subset(res18TEST,padj<0.1))
# res18TESTSig <- cbind(names = rownames(res18TESTSig), res18TESTSig)
# res18TESTSig <- dplyr::inner_join(genesGC_18,res18TESTSig,by=c("GENEID"="names"))
#
# ##### AGAINST LOG2(1) (lfcThreshold=0)
# res18Eu <- results(ddsDefGC_18,contrast=c("condition","Holly_18","Holly_Anc"))
#
# res18EunonSig <- as.data.frame(subset(res18Eu,padj>0.1))
# res18EunonSig <- cbind(names = rownames(res18EunonSig), res18EunonSig)
# res18EunonSig <- dplyr::inner_join(genesGC_18,res18EunonSig,by=c("GENEID"="names"))
#
# res18EuSig <- as.data.frame(subset(res18Eu,padj<0.1))
# res18EuSig <- cbind(names = rownames(res18EuSig), res18EuSig)
# res18EuSig <- dplyr::inner_join(genesGC_18,res18EuSig,by=c("GENEID"="names"))
#
#
# ##### NOW FOR THE INTERSECTIONS
# res18NoDC <- dplyr::inner_join(res18TESTnonSig,res18EuSig,by="GENEID")
#
# res18PartOvDC <- dplyr::inner_join(res18TESTSig,res18EuSig,by="GENEID")
#
# res18FullDC <- dplyr::inner_join(res18TESTSig,res18EunonSig,by="GENEID")



#
#   chr.all <- inner_join(sample1, sample2, by="GENEID")
#       # match the genes to the genes in genes table that includes chromosome number
#       matched <- intersect(chr.all$GENEID, genesNoAneu$GENEID)
#         # tell R which genes to keep from the chrI.tri dataframe
#         keepAll <- chr.all$GENEID %in% matched
#           # only keep those genes in the data frame
#           chr.all <- chr.all[keepAll,]
#             # tell R which genes to keep from the genesNoAneu dataframe
#             keepGenAll <- genesNoAneu$GENEID %in% matched
#               # put those genes in a vector
#               GenAll <- genesNoAneu[keepGenAll,]
#                 # tell R to bind Gen chromosome column to chrI.tri data so we know what chromosome the genes are on
#                 chr.all <- cbind(GenAll$TXCHROM, chr.all)
#                 return(chr.all)
# }
#
#     ddsNoAneuInNorm$condition
#
# res11DC <- results(ddsNoAneuInNorm,contrast=c("condition","Holly_11","Holly_Anc"),lfcThreshold = 0,altHypothesis = "greaterAbs",test="Wald")
# View(res11DC$padj)
# write.csv(res11DC,file="res11DC.csv")
# res11DCnS <- as.data.frame(subset(res11DC,padj>0.1))
# res11DCnS <- cbind(names = rownames(res11DCnS), res11DCnS)
# res11DCS <- as.data.frame(subset(res11DC,padj<=0.1))
# res11DCS <- cbind(names = rownames(res11DCS), res11DCS)
#
# res11noDCTri <- results(ddsNoAneuInNorm,contrast=c("condition","Holly_11","Holly_Anc"),lfcThreshold = 0.538,altHypothesis = "greater",test="Wald")
# View(res11noDCTri$padj)
# write.csv(res11noDCTri,file="res11noDCTri.csv")
# res11noDCTriS <- as.data.frame(subset(res11noDCTri,padj<=0.1))
# res11noDCTriS <- cbind(names = rownames(res11noDCTriS), res11noDCTriS)
# res11noDCTriNS <- as.data.frame(subset(res11noDCTri,padj>0.1))
# res11noDCTriNS <- cbind(names = rownames(res11noDCTriNS), res11noDCTriNS)
#
#
# res11noDCMon <- results(ddsNoAneuInNorm,contrast=c("condition","Holly_11","Holly_Anc"),lfcThreshold = 1,altHypothesis = "less",test="Wald")
# View(res11noDCMon$padj)
# write.csv(res11noDCMon,file="res11noDCMon.csv")
# res11noDCMonS <- as.data.frame(subset(res11noDCMon,padj<=0.1))
# res11noDCMonS <- cbind(names = rownames(res11noDCMonS), res11noDCMonS)
# res11noDCMonNS <- as.data.frame(subset(res11noDCMon,padj>0.1))
# res11noDCMonNS <- cbind(names = rownames(res11noDCMonNS), res11noDCMonNS)
#
#
# # now for the intersections
# # No DC:
#   # intersection of no DC test not sig and DC test sig
#
#     res11MonNoDC <- dplyr::inner_join(res11noDCMonNS,res11DCnS,by="names")
#     res11MonNoDC <- dplyr::anti_join(res11MonNoDC, commonDEeup,by="names")
#     res11MonNoDC <- dplyr::inner_join(genes,res11MonNoDC,by=c("GENEID"="names"))
#     write.csv(res11MonNoDC,file="Sample11NoDCMon.csv")
#     res11TriNoDC <- dplyr::inner_join(res11noDCTriNS,res11DCnS,by="names")
#     res11TriNoDC <- dplyr::anti_join(res11TriNoDC,commonDEeup,by="names")
#       # add chromosome number
#       res11TriNoDC <- dplyr::inner_join(genes,res11TriNoDC,by=c("GENEID"="names"))
#           write.csv(res11TriNoDC,file="Sample11NoDCTri.csv")
#     
# # DC
#     # intersection of no DC test sig and DC test not sig
#     res11MonDC <- dplyr::inner_join(res11noDCMonS,res11DCnS,by="names")
#     res11MonDC <- dplyr::anti_join(res11MonDC,commonDEeup,by="names")
#         res11MonDC <- dplyr::inner_join(genes,res11MonDC,by=c("GENEID"="names"))
#     write.csv(res11MonDC,file="Sample11DCMon.csv")
#     res11TriDC <- dplyr::inner_join(res11noDCTriS,res11DCnS,by="names")
#     res11TriDC <- dplyr::anti_join(res11TriDC,commonDEeup,by="names")
#     res11TriDC <- dplyr::inner_join(genes,res11TriDC,by=c("GENEID"="names"))
#     write.csv(res11TriDC,file="Sample11DCTri.csv")
#     View(res11MonDC)
# # partial?
#     # intersection of no DC test sig and DC test sig
#     res11MonPartial <- dplyr::inner_join(res11noDCMonS,res11DCS,by="names")
#     res11MonPartial <- dplyr::anti_join(res11MonPartial,commonDEeup,by="names")
#       res11MonPartial <- dplyr::inner_join(genes,res11MonPartial,by=c("GENEID"="names"))
#     write.csv(res11MonPartial,file="Sample11PartialDCMon.csv")
#     res11TriPartial <- dplyr::inner_join(res11noDCTriS,res11DCS,by="names")
#     res11TriPartial <- dplyr::anti_join(res11TriPartial,commonDEeup,by="names")
#     res11TriPartial <- dplyr::inner_join(genes,res11TriPartial,by=c("GENEID"="names"))
#     write.csv(res11TriPartial,file="Sample11PartialDCTri.csv")
#     
```



Below is a script for differentiating genes that are differentially expressed from those that are not dosage compensated, dosage compensated, anti DC, over DC, unknown, or partially dosage compensated. I used the log fold change threshold function of the results function to isolate these genes. FDR p values < 0.1 were considered statistically significant.
```{r trisomy, eval=TRUE, echo=FALSE}


resultsTrisomy <- function(dds,sample,anc,genes,anChr) {
  # make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  genes
  keep <- anChr
  genes <- genes[(grep(paste0(keep,"$"),genes$TXCHROM,perl=TRUE)),]
  genes
  all(rownames(dds) == genes$GENEID)
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  ddsGenes <- intersect(rownames(dds),genes)
  as.data.frame(genes)
  str(genes)
  # ddsGenes <- genes[,ddsGenes]
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 0.585 (expectation for trisomic genes)
  resTri <- results(dds,contrast=c("condition",sample,anc),lfcThreshold=0.585)
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (either no DC or unknown))
  resTriNonSig <- as.data.frame(subset(resTri,padj>0.1))
  length(resTriNonSig)
  resTriNonSig <- cbind(names = rownames(resTriNonSig), resTriNonSig)
  resTriNonSig <- dplyr::inner_join(genes,resTriNonSig,by=c("GENEID"="names"))
  str(resTriNonSig)
  # View(resTriNonSig)
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resTriSig <- as.data.frame(subset(resTri,padj<0.1))
  resTriSig <- cbind(names = rownames(resTriSig), resTriSig)
  resTriSig <- dplyr::inner_join(genes,resTriSig,by=c("GENEID"="names"))
  str(resTriSig)
  #### run same analyses using new ddsCopy dataset and new parameters
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (either Full DC or unknown genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  str(resDiNonSig)
  # subset data frame to get Significant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  str(resDiSig)
  #####################################
  ### Find genes that are Signficiantly different than Trisomic expectation and 
  # are Significantly different from Disomic expectation
  results <- list()
  resSigBoth<- dplyr::inner_join(resTriSig,resDiSig,by="GENEID")
  ## over DC genes for trisomy : less than 0 (1x)
  over.4 <- resSigBoth[which(resSigBoth$log2FoldChange.x<0),]
  ## anti DC genes for trisomy: more than 0.585 (1.5x)
  anti.5 <- resSigBoth[which(resSigBoth$log2FoldChange.x > 0.585),]
  results$over.4 <- over.4
  results$anti.5 <- anti.5
  ### Find genes that are Signficiantly different than Trisomic expectation but
  # NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resTriSig,resDiNonSig,by="GENEID")
  results$full.3 <- resFullDC
  ## Find genes that are not significantly diff from trisomic exp but sig diff from disomic
  resPartNo <- dplyr::inner_join(resTriNonSig,resDiSig,by="GENEID")
  resPart <- resPartNo[which(resPartNo$log2FoldChange.x < 0.585),]
  results$Part.2 <- resPart
  resNo <- resPartNo[which(resPartNo$log2FoldChange.x >= 0.585),]
  results$No.1 <- resNo
  ## Get genes that are not significantly different from either Trisomic or Disomic expectation
  resUnknown <- dplyr::inner_join(resTriNonSig,resDiNonSig,by="GENEID")
  results$Unknown.0 <- resUnknown
  return(results)
}

# input is dds for that sample, line #, ancestor name, genes matrix for that sample, and chromosome of interest
# output is list with full, part, over, anti, no, and unknown DC genes 
rm(dds)
rm(sample)
rm(anc)
rm(genes)
rm(anChr)

L <- resultsTrisomy(ddsDefMAold_9,"SC009","SCA",genesMAold_9,"chrXIV")
View(L)


L <- resultsTrisomy(ddsDefMAnewS112,"112","Anc",genesMAnewS112,"chrXVI")
View(L)

resultsNames(ddsDefGC_77)
L <- resultsTrisomy(ddsDefGC_77,"Holly_77","Holly_Anc",genesGC_77,"chrXII")
View(L)




```

```{r tetrasomy, eval=TRUE, echo=FALSE}

dds=ddsDefGC_8
sample="Holly_8"
anc="Holly_Anc"
genes=genesGC_8
anChr="chrXVI"

resultsTetrasomy <- function(dds,sample,anc,genes,anChr) {
  # make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  genes
  keep <- anChr
  genes <- genes[(grep(paste0(keep,"$"),genes$TXCHROM,perl=TRUE)),]
  genes
  all(rownames(dds) == genes$GENEID)
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  ddsGenes <- intersect(rownames(dds),genes)
  as.data.frame(genes)
  str(genes)
  # ddsGenes <- genes[,ddsGenes]
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at 1 (expectation for tetrasomic genes)
  resTet <- results(dds,contrast=c("condition",sample,anc), lfcThreshold=1 )
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (either no DC or unknown))
  resTetNonSig <- as.data.frame(subset(resTet,padj>0.1))
  length(resTetNonSig)
 resTetNonSig <- cbind(names = rownames(resTetNonSig), resTetNonSig)
  resTetNonSig <- dplyr::inner_join(genes,resTetNonSig,by=c("GENEID"="names"))
  str(resTetNonSig)
  # View(resTriNonSig)
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resTetSig <- as.data.frame(subset(resTet,padj<0.1))
  resTetSig <- cbind(names = rownames(resTetSig), resTetSig)
  resTetSig <- dplyr::inner_join(genes,resTetSig,by=c("GENEID"="names"))
  str(resTetSig)
  #### run same analyses using new ddsCopy dataset and new parameters
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (either Full DC or unknown genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  str(resDiNonSig)
  # subset data frame to get Significant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  str(resDiSig)
  #####################################
  ### Find genes that are Signficiantly different than Trisomic expectation and 
  # are Significantly different from Disomic expectation
  results <- list()
  resSigBoth<- dplyr::inner_join(resTetSig,resDiSig,by="GENEID")
  ## over DC genes for tetrasomy : less than 0 (1x)
  over.4 <- resSigBoth[which(resSigBoth$log2FoldChange.x<0),]
  ## anti DC genes for trisomy: more than 1 (2x)
  anti.5 <- resSigBoth[which(resSigBoth$log2FoldChange.x > 1),]
  results$over.4 <- over.4
  results$anti.5 <- anti.5
  ### Find genes that are Signficiantly different than Trisomic expectation but
  # NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resTetSig,resDiNonSig,by="GENEID")
  results$full.3 <- resFullDC
  ## Find genes that are not significantly diff from trisomic exp but sig diff from disomic
  resPartNo <- dplyr::inner_join(resTetNonSig,resDiSig,by="GENEID")
  resPart <- resPartNo[which(resPartNo$log2FoldChange.x < 1),]
  resPart1 <- resPart[which(resPart$log2FoldChange.x > 0),]
  results$Part.2 <- resPart1
  resNo <- resPartNo[which(resPartNo$log2FoldChange.x >= 1),]
  results$No.1 <- resNo
  ## Get genes that are not significantly different from either Trisomic or Disomic expectation
  resUnknown <- dplyr::inner_join(resTriNonSig,resDiNonSig,by="GENEID")
  results$Unknown.0 <- resUnknown
  return(results)
}

# input is dds for that sample, line #, ancestor name, genes matrix for that sample, and chromosome of interest
# output is list with full, part, over, anti, no, and unknown DC genes 
rm(dds)
rm(sample)
rm(anc)
rm(genes)
rm(anChr)

L <- resultsTrisomy(ddsDefGC_8,"Holly_8","Holly_Anc",genesGC_8,"chrXVI")
View(L)




```

```{r monosomy, eval=TRUE,echo=FALSE}


dds=ddsDefMAold_108
sample="SC108"
anc="SCA"
genes=genesMAold_108
anChr="chrIX"

resultsMonosomy <- function(dds,sample,anc,genes,anChr) {
  # make a new dds data set that we can use in future analyses
  ddsCopy <- dds
  geneID <- rownames(dds)
  library(TxDb.Scerevisiae.UCSC.sacCer2.sgdGene)
  txdb <- TxDb.Scerevisiae.UCSC.sacCer2.sgdGene
  genes1 <- ensembldb::select(txdb, keys=geneID, columnns="GENEID","TXCHROM" , keytype="GENEID")
  #remove the chromosomes that I don't want (i.e. anything that is NA for TXCHROM)
  genes <- na.omit(genes1)
  genes <- as.data.frame(genes1)
  genes
  keep <- anChr
  genes <- genes[(grep(paste0(keep,"$"),genes$TXCHROM,perl=TRUE)),]
  genes
  all(rownames(dds) == genes$GENEID)
  #need to remove the ones in rownames of dds that don't match genes$GENEID
  matched_2 <- intersect(rownames(dds), genes$GENEID)
  keep4 <- rownames(dds) %in% matched_2
  dds <- dds[keep4,]
  ddsGenes <- intersect(rownames(dds),genes)
  as.data.frame(genes)
  str(genes)
  # ddsGenes <- genes[,ddsGenes]
  all(rownames(dds) == genes$GENEID)
  mcols(dds) <- cbind(mcols(dds), genes)
  # run results function with log fold change threshold at -1 (expectation for monosomic genes)
  resMon <- results(dds,contrast=c("condition",sample,anc),lfcThreshold=1, altHypothesis="less")
  #subset data frame to get nonSignificant p values (indicating gene does is as expected (either no DC or unknown))
  resMonNonSig <- as.data.frame(subset(resMon,padj>0.1))
  length(resMonNonSig)
  resMonNonSig <- cbind(names = rownames(resMonNonSig), resMonNonSig)
  resMonNonSig <- dplyr::inner_join(genes,resMonNonSig,by=c("GENEID"="names"))
  str(resMonNonSig)
  # View(resTriNonSig)
  # subset data frame to get Significant p values (indicting gene does is NOT as expected)
  resMonSig <- as.data.frame(subset(resMon,padj<0.1))
  resMonSig <- cbind(names = rownames(resMonSig), resMonSig)
  resMonSig <- dplyr::inner_join(genes,resMonSig,by=c("GENEID"="names"))
  str(resMonSig)
  #### run same analyses using new ddsCopy dataset and new parameters
  # run results function with log fold change threshold at 0 (expectation for disomic genes)
  resDi <- results(dds,contrast=c("condition",sample,anc),lfcThreshold = 0)
  # subset data frame to get nonSignificant p values (either Full DC or unknown genes)
  resDiNonSig <- as.data.frame(subset(resDi,padj>0.1))
  resDiNonSig <- cbind(names = rownames(resDiNonSig), resDiNonSig)
  resDiNonSig <- dplyr::inner_join(genes1,resDiNonSig,by=c("GENEID"="names"))
  str(resDiNonSig)
  # subset data frame to get Significant p values (indicating gene dose is NOT as expected for Disomic genes)
  resDiSig <- as.data.frame(subset(resDi,padj<0.1))
  resDiSig <- cbind(names = rownames(resDiSig), resDiSig)
  resDiSig <- dplyr::inner_join(genes1,resDiSig,by=c("GENEID"="names"))
  str(resDiSig)
  #####################################
  ### Find genes that are Signficiantly different than Monosomic expectation and 
  # are Significantly different from Disomic expectation
  results <- list()
  resSigBoth<- dplyr::inner_join(resMonSig,resDiSig,by="GENEID")
  ## over DC genes for monosomy : greater than 0 (1x)
  over.4 <- resSigBoth[which(resSigBoth$log2FoldChange.x>0),]
  ## anti DC genes for monosomy: less than -1 (0.5)
  anti.5 <- resSigBoth[which(resSigBoth$log2FoldChange.x < -1),]
  results$over.4 <- over.4
  results$anti.5 <- anti.5
  ### Find genes that are Signficiantly different than Trisomic expectation but
  # NotSignificantly different from Disomic expectation
  resFullDC <- dplyr::inner_join(resTriSig,resDiNonSig,by="GENEID")
  results$full.3 <- resFullDC
  ## Find genes that are not significantly diff from trisomic exp but sig diff from disomic
  resPartNo <- dplyr::inner_join(resTriNonSig,resDiSig,by="GENEID")
  resPart <- resPartNo[which(resPartNo$log2FoldChange.x > -1),]
  results$Part.2 <- resPart
  resNo <- resPartNo[which(resPartNo$log2FoldChange.x <= -1),]
  results$No.1 <- resNo
  ## Get genes that are not significantly different from either Trisomic or Disomic expectation
  resUnknown <- dplyr::inner_join(resMonNonSig,resDiNonSig,by="GENEID")
  results$Unknown.0 <- resUnknown
  return(results)
}

L <- resultsTrisomy(ddsDefGC_11,"Holly_11","Holly_Anc",genesGC_11,"chrI")
View(L)



```


level `alpha` (we use the standard variable name for significance
level, though it is unrelated to the dispersion parameter $\alpha$).
The theory behind independent filtering is discussed in greater detail
[below](#indfilttheory). The adjusted *p* values for the genes
which do not pass the filter threshold are set to `NA`.

The default independent filtering is performed using the *filtered_p*
function of the [genefilter](http://bioconductor.org/packages/genefilter) package, and all of the
arguments of *filtered_p* can be passed to the *results* function.
The filter threshold value and the number of rejections at each
quantile of the filter statistic are available as metadata of the
object returned by *results*.

For example, we can visualize the optimization by plotting the
`filterNumRej` attribute of the results object. The *results* function
maximizes the number of rejections (adjusted *p* value less than a
significance level), over the quantiles of a filter statistic (the
mean of normalized counts). The threshold chosen (vertical line) is
the lowest quantile of the filter for which the number of rejections
is within 1 residual standard deviation to the peak of a curve fit to
the number of rejections over the filter quantiles:

```{r filtByMean,include=FALSE}
# metadata(res)$alpha
# metadata(res)$filterThreshold
# {plot(metadata(res)$filterNumRej,
#      type="b", ylab="number of rejections",
#      xlab="quantiles of filter")
# lines(metadata(res)$lo.fit, col="red")
# abline(v=metadata(res)$filterTheta)}
```



```{r varGroup, echo=FALSE,include=FALSE}
# set.seed(3)
# dds1 <- makeExampleDESeqDataSet(n=1000,m=12,betaSD=.3,dispMeanRel=function(x) 0.01)
# dds2 <- makeExampleDESeqDataSet(n=1000,m=12,
#                                 betaSD=.3,
#                                 interceptMean=mcols(dds1)$trueIntercept,
#                                 interceptSD=0,
#                                 dispMeanRel=function(x) 0.2)
# dds2 <- dds2[,7:12]
# dds2$condition <- rep("C",6)
# mcols(dds2) <- NULL
# dds12 <- cbind(dds1, dds2)
# rld <- rlog(dds12, blind=FALSE, fitType="mean")
# plotPCA(rld)
```

## Can I run DESeq2 to contrast the levels of many groups?

DESeq2 will work with any kind of design specified using the R
formula. We enourage users to consider exploratory data analysis such
as principal components analysis rather than performing statistical
testing of all pairs of many groups of samples. Statistical testing is
one of many ways of describing differences between samples.

As a speed concern with fitting very large models,
note that each additional level of a factor in the
design formula adds another parameter to the GLM which is fit by
DESeq2. Users might consider first removing genes with very few
reads, as this will speed up the fitting procedure.

## Can I use DESeq2 to analyze a dataset without replicates?

If a *DESeqDataSet* is provided with an experimental design without replicates,
a warning is printed, that the samples are treated as replicates
for estimation of dispersion. This kind of analysis is
only useful for exploring the data, but will not provide the kind of
proper statistical inference on differences between groups.
Without biological replicates, it is not possible to estimate the biological
variability of each gene.
More details can be found in the manual page for `?DESeq`.

## How can I include a continuous covariate in the design formula?

Continuous covariates can be included in the design formula in exactly
the same manner as factorial covariates, and then *results* for the
continuous covariate can be extracted by specifying `name`.
Continuous covariates might make sense in certain experiments, where a
constant fold change might be
expected for each unit of the covariate.  However, in many cases, more
meaningful results can be obtained by cutting continuous covariates
into a factor defined over a small number of bins (e.g. 3-5).  In this
way, the average effect of each group is controlled for, regardless of
the trend over the continuous covariates.  In R, *numeric*
vectors can be converted into *factors* using the function *cut*.

## I ran a likelihood ratio test, but results() only gives me one comparison.

"... How do I get the *p* values for all of the variables/levels
that were removed in the reduced design?"

This is explained in the help page for `?results` in the
section about likelihood ratio test p-values, but we will restate the
answer here. When one performs a likelihood ratio test, the *p* values and
the test statistic (the `stat` column) are values for the test
that removes all of the variables which are present in the full
design and not in the reduced design. This tests the null hypothesis
that all the coefficients from these variables and levels of these factors
are equal to zero.

The likelihood ratio test *p* values therefore
represent a test of *all the variables and all the levels of factors*
which are among these variables. However, the results table only has space for
one column of log fold change, so a single variable and a single
comparison is shown (among the potentially multiple log fold changes
which were tested in the likelihood ratio test).
This is indicated at the top of the results table
with the text, e.g., log2 fold change (MLE): condition C vs A, followed
by, LRT p-value: '~ batch + condition' vs '~ batch'.
This indicates that the *p* value is for the likelihood ratio test of
*all the variables and all the levels*, while the log fold change is a single
comparison from among those variables and levels.
See the help page for *results* for more details.

## What are the exact steps performed by DESeq()?

See the manual page for *DESeq*, which links to the
subfunctions which are called in order, where complete details are
listed. Also you can read the three steps listed in the
[DESeq2 model](#theory) in this document.


## Is there an official Galaxy tool for DESeq2?

Yes. The repository for the DESeq2 tool is

<https://github.com/galaxyproject/tools-iuc/tree/master/tools/deseq2>

and a link to its location in the Tool Shed is

<https://toolshed.g2.bx.psu.edu/view/iuc/deseq2/d983d19fbbab>.

## I want to benchmark DESeq2 comparing to other DE tools.

One aspect which can cause problems for comparison is that, by default,
DESeq2 outputs `NA` values for adjusted *p* values based on
independent filtering of genes which have low counts.
This is a way for the DESeq2 to give extra
information on why the adjusted *p* value for this gene is not small.
Additionally, *p* values can be set to `NA` based on extreme
count outlier detection. These `NA` values should be considered
*negatives* for purposes of estimating sensitivity and specificity. The
easiest way to work with the adjusted *p* values in a benchmarking
context is probably to convert these `NA` values to 1:

```{r convertNA, eval=FALSE,include=FALSE}
# res$padj <- ifelse(is.na(res$padj), 1, res$padj)
```

## I have trouble installing DESeq2 on Ubuntu/Linux...

"*I try to install DESeq2, but I get an error trying to
install the R packages XML and/or RCurl:*"

`ERROR: configuration failed for package XML`

`ERROR: configuration failed for package RCurl`

You need to install the following devel versions of packages using
your standard package manager, e.g. `sudo apt-get install` or
`sudo apt install`

* libxml2-dev
* libcurl4-openssl-dev

# Acknowledgments

We have benefited in the development of DESeq2 from the help and
feedback of many individuals, including but not limited to:

The Bionconductor Core Team,
Alejandro Reyes, Andrzej Oles, Aleksandra Pekowska, Felix Klein,
Nikolaos Ignatiadis (IHW),
Anqi Zhu (apeglm),
Joseph Ibrahim (apeglm),
Vince Carey,
Owen Solberg,
Ruping Sun,
Devon Ryan,
Steve Lianoglou, Jessica Larson, Christina Chaivorapol, Pan Du, Richard Bourgon,
Willem Talloen,
Elin Videvall, Hanneke van Deutekom,
Todd Burwell,
Jesse Rowley,
Igor Dolgalev,
Stephen Turner,
Ryan C Thompson,
Tyr Wiesner-Hanks,
Konrad Rudolph,
David Robinson,
Mingxiang Teng,
Mathias Lesche,
Sonali Arora,
Jordan Ramilowski,
Ian Dworkin,
Bjorn Gruning,
Ryan McMinds,
Paul Gordon,
Leonardo Collado Torres,
Enrico Ferrero,
Peter Langfelder,
Gavin Kelly,
Rob Patro,
Charlotte Soneson,
Koen Van den Berge,
Stephan Engelbrecht,
Nicolas Alcala.

# Session info


```{r sva,include=FALSE}
# #source("https://bioconductor.org/biocLite.R")
# #biocLite("rnaseqGene")
# library(rnaseqGene)
# #browseVignettes(rnaseqGene)
# source("https://bioconductor.org/biocLite.R")
# biocLite("RUVSeq")
# library(sva)
# dds <- DESeq(ddsHTSeq)
#
# data <- counts(dds,normalized=TRUE)
# idx <- rowMeans(data) > 1
# data <- data[idx,]
# mod <- model.matrix(~condition, colData(dds))
# mod0 <- model.matrix(~1, colData(dds))
# svseq <- svaseq(data,mod,mod0,n.sv = 2 )
# svseq$sv
#
# #see how good sva was at recovering the condition (sample group) variable:
# par(mfrow=c(2,1), mar=c(3,5,3,1))
# for (i in 1:2) {
#   stripchart(svseq$sv[,i] ~dds$condition, vertical=TRUE, main= paste0("SV",i))
#   abline(h=0)
# }
#
# #add surrogate variables to the design
# ddssva <- dds
# ddssva$SV1 <- svseq$sv[,1]
# ddssva$SV2 <- svseq$sv[,2]
# design(ddssva) <- ~ SV1 + SV2 + condition
# ddsSva <- DESeq(ddssva)
#
# #testing on lines that work fine
# #2
# resSV2 <- results(ddsSva, contrast=c("condition","Holly_2","Holly_Anc"))
# summary(resSV2)
# hist(resSV2$pvalue,col="green", main="2 vs Anc, with aneuploid c'somes", xlab="p values")
#
# #lines that dont work fine
# #7
# resSV7 <- results(ddsSva, contrast=c("condition","Holly_7","Holly_Anc"))
# summary(resSV7)
# hist(resSV7$pvalue,col="green", main="7 vs Anc, with aneuploid c'somes", xlab="p values")



```

```{r look at variance,include=FALSE}

# GC_anc <- counts(ddsHTSeqGC_1)[,4:6]


# genes <- data.frame(x.GC$genes$GENEID, x.GC$genes$TXCHROM)
# genes <- genesGC
# rownames(genes) <- genes$x.GC.genes.GENEID
# 
# remove <- c("__no_feature", "__ambiguous","__alignment_not_unique","__too_low_aQual","__not_aligned")
# genes <- genes[setdiff(rownames(genes),remove),]
# removegenes <- c("YPL241C","YPL163C")
# genes <- genes[setdiff(rownames(genes),removegenes),]
# genes$x.GC.genes.GENEID <- droplevels(genes$x.GC.genes.GENEID)


#remove NAs from data

# lcpmGC.test <- merge(genes,lcpmGC, by.x="GENEID", by.y="names")
# lcpmGC <- lcpmGC.test
# 
# 
# cpmGC.test <- merge(genesGC,cpmGC, by.x="GENEID", by.y="names")
# cpmGC <- cpmGC.test
# write.csv(cpmGC,file="cpm_norm.csv")
# 
# nchrm <- nlevels(lcpmGC$x.GC.genes.TXCHROM)
# lcpmGC$x.GC.genes.TXCHROM
# lcpmGC
# x.GC$samples$group
# cpmGC
# length(x.GC$samples$group)
# names <- colnames(lcpmGC)
# length(lcpmGC[,1])
# nrow(lcpmGC)
# rm(i)
# lcpmGC <- na.omit(lcpmGC)
# View(lcpmGC)
# write.csv(lcpmGC,file="log_cpm_norm.csv")
# rm(i)
# # #testing to make sure this is doing what I want it to do: find the variance between replicates, not across all lines
# lineEl <- matrix(data=NA,ncol=3,nrow=length(lcpmGC))
# lineEl.ind <- grep(as.character(lines.GC[1]),names)
# ind.El <- lcpmGC[,lineEl.ind]
# test.M <- matrix(apply(ind.El,1,var))
# #yes it is working
# 
# colnames(lcpmGC) <- c("gene","chr", ".11.A",".11.B",".11.C",".18.A",".18.B",".18.C",".1.A", ".1.B", ".1.C", ".21.B",".21.C",".2.A", ".2.B", ".2.C", ".31.A",".31.B",".31.C",".3.A", ".3.B", ".3.C",".49.A",".49.B",".49.C",".4.A", ".4.B",".4.C", ".59.A",".59.B",".59.C",".5.A",".5.B", ".5.C", ".61.A",".61.B",".61.C",".66.B",".66.C",".68.A",".68.B",".68.C",".6.A", ".6.B", ".6.C", ".76.A",".76.B",".76.C",".77.A",".77.B",".77.C",".7.A",".7.B", ".7.C", ".8.A", ".8.B", ".8.C",".9.A", ".9.B", ".9.C", ".GCAnc.A" ,".GCAnc.B",".GCAnc.C")
# names <- colnames(lcpmGC)
# lines.G <- c(".11.",".18.",".1.",".21.",".2.",".31.",".3.",".49.",".4.",".59.",".5.",".61.",".66.",".68.",".6.", ".76.",".77.",".7.",".8.", ".9.",".GCAnc.")
# colnames.G <- c("GENE","CHRM", " .11.",".18.",".1.",".21.",".2.",".31.",".3.",".49.",".4.",".59.",".5.",".61.",".66.",".68.",".6.", ".76.",".77.",".7.",".8.", ".9.",".GCAnc.")
# 
# i=1
# rm(i)
# vari <- matrix(data=NA,nrow=nrow(lcpmGC),ncol=length(lines.G))
# is.list(vari)
# for (i in 1:length(lines.G)) {
#   ind <- grep(as.character(lines.G[i]),names)
#   cols <- lcpmGC[,ind]
#   cols <- matrix(apply(cols,1,var))
#   rownames(cols) <- c()
#   colnames(cols) <- c()
#   vari[,i] <- cols
# }
# 
# vari <- as.data.frame(vari)
# #vari <- cbind(as.factor(genes$x.GC.genes.TXCHROM),vari)
# View(vari)
# vari <- cbind(lcpmGC$gene,lcpmGC$chr,vari)
# colnames(vari) <- colnames.G
# class(vari[,3])
# vari$GENE <- droplevels(vari$GENE)
# rownames(vari) <- vari$GENE
# #vari2 <- subset(vari, vari[1:3386,]>=0.0001) # & vari[,3:23]>=0.0001)
# #summary(vari[,3:23]>=0.0001)
# #summary(vari[1:3386,]<=0.0001)
# table(rowSums(vari[3:23]>0.0001)>=21)
# keep.genes.GC <- rowSums(vari[,3:23]>=0.0001)>=21
# keep.genes.GC <- as.data.frame(keep.genes.GC)
# keep.genes.GC$genes <- rownames(keep.genes.GC)
# genes2 <- merge(keep.genes.GC,genes,by.x="genes",by.y="x.GC.genes.GENEID")
# genes2 <- genes2[!(genes2$keep.genes.GC=="FALSE"),]
# 
# #remove genes with variance less than 0.00001
# lcpmGC <- merge(genes2,lcpmGC, by.x="genes", by.y="gene")
# lcpmGC <- lcpmGC[,-2]
# lcpmGC <- lcpmGC[,-3]
# lcpmGC <- droplevels(lcpmGC)
# vari2 <- merge(vari,lcpmGC,by.x="GENE",by.y="genes")
# 
# mean <- matrix(data=NA,nrow=nrow(lcpmGC),ncol=length(lines.G))
# is.list(mean)
# for (i in 1:length(lines.G)) {
#   ind <- grep(as.character(lines.G[i]),names)
#   cols <- lcpmGC[,ind]
#   rownames(cols) <- c()
#   colnames(cols) <- c()
#   m <- matrix(apply(cols,1,mean))
#   mean[,i] <- m
# }
# mean <- as.data.frame(mean)
# mean <- cbind(lcpmGC$genes,lcpmGC$x.GC.genes.TXCHROM,mean)
# colnames(mean) <- colnames.G
# 
# 
# std <- matrix(data=NA,nrow=nrow(lcpmGC),ncol=length(lines.G))
# length(lcpmGC)
# is.list(std)
# for (i in 1:length(lines.G)) {
#   ind <- grep(as.character(lines.G[i]),names)
#   cols <- lcpmGC[,ind]
#   cols <- matrix(apply(cols,1,sd))
#   rownames(cols) <- c()
#   colnames(cols) <- c()
#   std[,i] <- cols
# }
# std <- as.data.frame(std)
# std <- cbind(lcpmGC$genes,lcpmGC$x.GC.genes.TXCHROM,std)
# colnames(std) <- colnames.G
# class(std[,3])
# 
# GCanc.stats <- data.frame(mean$GENE, mean$CHRM, mean$.GCAnc.,std$.GCAnc.)
# colnames(GCanc.stats) <- c("GENE","CHRM","mean","std")
# 
# #
# #
# k <- ggplot(GCanc.stats, aes(CHRM, mean))#ymin = mean-std, ymax = mean+std))
# print(k)
# 
# 
# 
# dim(mean)
# dim(vari)
# dim(std)
# #std[,3:23] <- as.numeric(std[,3:23])
# cv <- matrix(data=NA,nrow=nrow(mean),ncol=length(ncol(mean)))
# #log transform the coefficient of variance
# #cv <- log(cv)
# View(std)
# #mean[,3:23] <- as.numeric(mean[,3:23])
# View(mean)
# is.list(std)
# class(mean[,3:23])
# class(std[,3:23])
# class(mean[,3])
# is.list(mean)
# cv <- std[,3:23]/mean[,3:23]
# dim(cv)
# #cv <- transform_to_log_scale(cv)
# cv <- cbind(lcpmGC$genes,lcpmGC$x.GC.genes.TXCHROM,cv)
# colnames(cv) <- colnames.G
# #cv <- unlist(cv)
# #max(subset(cv,select="X3"))
# 
# par(mar=c(1,1,1,1))
# 
# #loop to plot mean-variance relationship for all lines
# lines.G
# line.G <- c("NA",".GCAnc.",".11.",".18.",".1.",".21.",".2.",".31.",".3.",".49.",".4.",".59.",".5.",".61.",".66.",".68.",".6.", ".76.",".77.",".7.",".8.", ".9.")
# i=3
# length(line.G)
# length(mean)
# rm(i)
# for (i in 3:23) {
#   pdf(paste("Mean-Variance Relationship",line.G[i],".pdf",sep=""))
#   print(ggplot () + geom_point(aes(x=mean[,i],y=vari2[,i],col=factor(mean[,2]))) + scale_y_log10() + labs(title=paste("Mean-Variance Relationship Line",line.G[i]), y="variance",x="mean"))
#   dev.off()
# }
# 
# #use plotly to view the graph interactively and see what points correspond to what genes/chromosomes
# length(vari2)
# class(vari2)
# class(mean)
# length(vari2[,1])
# length(mean[,1])
# 
# get.points <- function(line) {
#    ggplot () + geom_point(aes(x=mean[,line],y=vari2[,line],col=factor(mean[,2]))) + scale_y_log10() + labs(title=paste("Mean-Variance Relationship Line",line.G[line]), y="variance",x="mean")
# ggplotly()
# }
# #line=18
# get.points(18)
# 
# par(mfrow=c(1,1))
# 
# for (i in 3:23) {
#   pdf(paste("Dist_Variance",i,".pdf",sep=""))
#   plot(density(vari2[,i]),col=colGC[i],lwd=2,las=2,main="",xlab="")
#   title(main=paste("Distribution of Variance",line.G[i]),xlab="var of lcpm")
#   abline(v=0,lty=3)
#   dev.off()
# }

```

```{r sessionInfo}
sessionInfo()
```

# References
