---
title: "DESeq2_practice"
author: "Holly"
date: "6/29/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r Deseq2 practice pipeline, include=TRUE}
#upgrade bioconductor
#source("https://bioconductor.org/biocLite.R")
#    biocLite("BiocUpgrade")
#source("http://bioconductor.org/biocLite.R")
#biocLite("DESeq2")
library(DESeq2)
## try http:// if https:// URLs are not supported
#source("https://bioconductor.org/biocLite.R")
#biocLite("parathyroidSE")
#source("https://bioconductor.org/biocLite.R")
#        biocLite("apeglm")
library(apeglm)
library("parathyroidSE")

data("parathyroidGenesSE")
se <- parathyroidGenesSE
colnames(se) <- se$run
#metadata already here, but if using own data, will probably have to add it (recommended in a .csv format)
## My data: import a .csv file of metadata from experiment: run, line #, treatment, etc 
colData(se)[1:5,1:4]

#build DESeqDataSet from Summarized Experiment, specify design formula 
ddsFull <- DESeqDataSet(se , design= ~ patient + treatment)

#####starting from count tables 
####more likely what I will be using 

#to look at count data in SummarizedExperiment, use accessor function
countdata <- assay(parathyroidGenesSE)
head(countdata)
#row: Ensembl gene
#column: sequenced RNA library
#values: raw numbers of sequencing reads mapped to each gene 

#metadata on samples ("columns" of count table)
coldata <- colData(parathyroidGenesSE)
rownames(coldata) <- coldata$run
colnames(countdata) <- coldata$run
head(coldata[,c("patient","treatment","time")])

#construct data object from matrix of counts and metadata table 
ddsFullCountTable <- DESeqDataSetFromMatrix(
  countData = countdata,
  colData = coldata,
 design = ~ patient + treatment)

#ddsFullCountTable

###############
#continuing with object generated from summarizedExperiment

#collapsing technical replicates (not necessary for mine)
as.data.frame( colData(ddsFull) [ , c("sample","patient","treatment","time") ])

ddsCollapsed <- collapseReplicates(ddsFull,
                                   groupby = ddsFull$sample,
                                   run = ddsFull$run )
head(as.data.frame(colData(ddsCollapsed)[,c("sample","runsCollapsed")]),12)
#confirm counts for new object are equal to summed up counts of columns with same value for grouping factor: 
original <- rowSums( counts(ddsFull)[ , ddsFull$sample == "SRS308873"] )
all( original == counts (ddsCollapsed[ , "SRS308873"] ) )

str(ddsCollapsed)
rownames(ddsCollapsed)
levels(ddsCollapsed$time)

####Running DESeq2 Pipeline
#prep data object for analysis of interest
#subset relevant columns from the full dataset:
ddsCollapsed$time
str(ddsCollapsed)
dds2
dds2$time
colnames(ddsCollapsed)

dds2 <- subset(ddsCollapsed, time == "48h" )
#drop levels if needed 
#not necessary unless time or whatever variable is included in the design formula 
dds2$time <- droplevels(dds2$time)
dds2
dds2$time 
#relevel so that control is the first level in the treatment factor 
#For Mine I would want this to be the Ancestor!!
dds2$treatment <- relevel( dds2$treatment , "Control" )

#check to make sure we have the right samples
as.data.frame(colData(dds2))

dds

#########################################################################################
#Run DESeq2 analysis with single call to the function DESeq
dds <- DESeq(dds)

#inspecting results table
#if you call results with no arguments, it will extract the estimated log2 fold changes and p values for the last variable in the design formula 
#if there are more than 2 levels for this variable, results will extract the results table for a comparison of the last level over the first level 
#how to extract other comparisons: 
res <- results(dds)
res

#res is a DataFrame object and has metadata information describing the columns:
mcols(res,use.names=TRUE)
#baseMean: average of the normalized count values, dividing by size factors, taken over all samples
#log2FoldChange: effect size estimate - how much the gene's expression seems to have changed with treatment in comparison to control 
#lfcSE: standard error estimate for log2foldchange estimate 
#DESeq2 performs a hypothesis test on EACH GENE: whether evidence is sufficient to decide against the null hypothesis (there is no effect of treatment on the gene and difference between treatment and control was purely from experimental variation)
#p value indicates that a fold change as strong/stronger than the observed fold change would be seen under the situation described by the null hypothesis

# a subset of p values in res are NA because either: all counts for the gene were 0 OR the gene was excluded from analysis because it contained an extreme count outlier 

######################################
#other comparisons 
#use contrast argument to results 
#specify:
##name of variable (i.e. treatment)
##name of level in numerator (i.e. DPN)
##name of level in denominator (i.e. Control)

res2 <- results(dds, contrast = c("treatment","DPN","Control"))
res2
resultsNames(dds)
#trying to do log fold change shrink because apparently it doesn't do this by default anymore
#coef: # of comparison you are looking at in resultsNames (this case: 5 )
#normal is the the original DESeq2 shrinkage estimator, an adaptive Normal distribution as prior. This is currently the default, although the default will likely change to apeglm in the October 2018 release given apeglmâ€™s superior performance.

#ashr says it might be unreliable because it is missing something 
#also it is recommended to use apeglm anyway 

lfcShrink(dds=dds,coef=5, type="ashr")
resAsh <- lfcShrink(dds=dds, coef=5, type="ashr")
resApe <- lfcShrink(dds=dds, coef=5, type="apeglm")

#if you want to get results from an interaction term, use the name argument of results 
resAsh
resApe

#plot MA to make sure the fold changes were really shrunk
plotMA(resAsh,ylim=c(-1,1)) 
plotMA(resApe,ylim=c(-1,1))

sum(resAsh$pvalue < 0.01, na.rm = TRUE )
table(is.na(resAsh$pvalue))
sum(resApe$pvalue < 0.01, na.rm = TRUE )
table(is.na(resApe$pvalue))

#adjusted p values
sum(resAsh$padj <0.1, na.rm = TRUE)
sum(resApe$padj <0.1, na.rm = TRUE)

#subset results table to get the genes with padjust less than 0.1, and sort by log2 fold change to get the genes that are most down-regulated
resAshSig <- resAsh[which(resAsh$padj < 0.1) , ]
head(resAshSig[order(resAshSig$log2FoldChange) , ] )
#and the ones most up-regulated
tail(resAshSig[order(resAshSig$log2FoldChange) , ] )

resApeSig <- resApe[which(resApe$padj < 0.1) , ]
head(resApeSig[order(resApeSig$log2FoldChange) , ] )
#and the ones most up-regulated
tail(resApeSig[order(resApeSig$log2FoldChange) , ] )

####################################
#multiple testing 
#how many genes with p value below 0.01
sum(res2$pvalue <0.01,na.rm=TRUE)
table(is.na(res2$pvalue))
#adjusted p values
sum(res2$padj <0.1, na.rm = TRUE)
#subset results table to get the genes with padjust less than 0.1, and sort by log2 fold change to get the genes that are most down-regulated
res2Sig <- res2[which(res2$padj <0.1),]
head(res2Sig[order(res2Sig$log2FoldChange),])
#and the ones most up-regulated
tail(res2Sig[order(res2Sig$log2FoldChange),])

###############################################
#diagnostic plots 
#MA plot
#useful overview for two-group comparison:
plotMA(res2,ylim=c(-1,1))

#plot dispersion estimates
plotDispEsts(dds,ylim=c(1e-6,1e1))
#the blue circles above the main "cloud" of points are genes which have high gene-wise dispersion estimates and are labeled as dispersion outliers
#these are not shrunk toward the fitted trend line 

# histogram of p-values
hist(resNorm$pvalue,breaks=20,col="grey")

#with default settings
hist(res2$pvalue,breaks=20,col="purple")

##############################################
#independent filtering
#weakly-expressed genes have no chance of being seen as differentially expressed because of Poisson noise 
#look at ratio of small p values for genes binned by mean normalized count:
#create bins using the quantile function
qs <- c(0,quantile(resNorm$baseMean[resNorm$baseMean>0],0:7/7))
#"cut" the genes into the bins
bins <- cut(resNorm$baseMean,qs)
#rename the levels of the bins using the middle point
levels(bins) <- paste0("~",round(.5*qs[-1]+.5*qs[-length(qs)]))
#calculate the ratio of p values less than .01 for each bin
ratios <- tapply(resNorm$pvalue,bins,function(p)mean(p<.01, na.rm=TRUE))
#plot these ratios
barplot(ratios,xlab="mean normalized count",ylab="ratio of small p values")

#can get rid of these lowly-expressed genes through independent filtering
#results function does this automatically and we can control it through it 
attr(resNorm,"filterThreshold")
#apparently it is not on by default
res2 <- results(dds, independentFiltering = TRUE, alpha=0.1, contrast = c("treatment","DPN","Control"))
attr(res2, "filterThreshold")
metadata(res2)$filterThreshold
metadata(res2)$alpha
resNorm <- lfcShrink(dds=dds, coef=5,res=res2)
attr(resNorm,"filterThreshold")
metadata(resNorm)$filterThreshold

plot(metadata(resNorm)$filterNumRej,type="b",xlab="quantiles of baseMean",ylab="number of rejections",xlim=c(0,1),ylim=c(0,300))




###
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
